<!DOCTYPE html>

<html lang="en" xml:lang="en" xmlns="http://www.w3.org/1999/xhtml"><head>
<meta charset="utf-8"/>
<meta content="quarto-1.5.55" name="generator"/>
<meta content="width=device-width, initial-scale=1.0, user-scalable=yes" name="viewport"/>
<meta content="Alejandro Fontal" name="author"/>
<meta content="2024-07-28" name="dcterms.date"/>
<title>Microbial Richness in the Troposphere: Appeal Reanalysis – Microbial Richness in the Troposphere</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>
<script crossorigin="anonymous" integrity="sha512-bLT0Qm9VnAYZDflyKcBaQ2gg0hSYNQrJ8RilYldYQ1FxQYoCLtUjuuRuZo+fjqhx/qtq/1itJ0C2ejDxltZVFg==" src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.5.1/jquery.min.js"></script><script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta content="./" name="quarto:offset"/>
<link href="./assets/favicon.ico" rel="icon"/>
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet"/>
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" id="quarto-text-highlighting-styles" rel="stylesheet"/>
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet"/>
<link data-mode="light" href="site_libs/bootstrap/bootstrap.min.css" id="quarto-bootstrap" rel="stylesheet"/>
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
<style>html{ scroll-behavior: smooth; }</style>
<script crossorigin="anonymous" integrity="sha512-c3Nl8+7g4LMSTdrm621y7kf9v3SDPnhxLNhcjFJbKECVnmZHTdo+IRO05sNLTH/D3vA6u1X32ehoLC7WFVdheg==" src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.6/require.min.js"></script>
<script type="application/javascript">define('jquery', [],function() {return window.jQuery;})</script>
<link href="custom.css" rel="stylesheet"/>
</head>
<body class="nav-fixed">
<div id="quarto-search-results"></div>
<header class="headroom fixed-top" id="quarto-header">
<nav class="navbar navbar-expand-lg" data-bs-theme="dark">
<div class="navbar-container container-fluid">
<div class="navbar-brand-container mx-auto">
<a class="navbar-brand" href="./index.html">
<span class="navbar-title">Microbial Richness in the Troposphere</span>
</a>
</div>
<div class="" id="quarto-search" title="Search"></div>
<button aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" class="navbar-toggler" data-bs-target="#navbarCollapse" data-bs-toggle="collapse" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }" role="menu" type="button">
<span class="navbar-toggler-icon"></span>
</button>
<div class="collapse navbar-collapse" id="navbarCollapse">
<ul class="navbar-nav navbar-nav-scroll me-auto">
<li class="nav-item">
<a class="nav-link" href="./index.qmd">
<span class="menu-text">Home</span></a>
</li>
<li class="nav-item dropdown">
<a aria-expanded="false" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" href="#" id="nav-menu-reports" role="link">
<span class="menu-text">Reports</span>
</a>
<ul aria-labelledby="nav-menu-reports" class="dropdown-menu">
<li>
<a class="dropdown-item" href="./microbial_diversity_analysis.html">
<span class="dropdown-text">Diversity analysis</span></a>
</li>
<li>
<a class="dropdown-item" href="./flights.html">
<span class="dropdown-text">Flights GPS</span></a>
</li>
</ul>
</li>
<li class="nav-item dropdown">
<a aria-expanded="false" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" href="#" id="nav-menu-about" role="link">
<span class="menu-text">About</span>
</a>
<ul aria-labelledby="nav-menu-about" class="dropdown-menu">
<li>
<a class="dropdown-item" href="https://www.isglobal.org/en">
<span class="dropdown-text">ISGlobal</span></a>
</li>
<li class="dropdown-header">Climate &amp; Health Programme</li>
<li>
<a class="dropdown-item" href="https://ginys.cerca.cat/en/facility/airlab/">
<span class="dropdown-text">AIRLAB</span></a>
</li>
</ul>
</li>
</ul>
<ul class="navbar-nav navbar-nav-scroll ms-auto">
<li class="nav-item dropdown">
<a aria-expanded="false" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" href="#" id="nav-menu-bi-github" role="link">
<i class="bi bi-github" role="img">
</i>
<span class="menu-text"></span>
</a>
<ul aria-labelledby="nav-menu-bi-github" class="dropdown-menu dropdown-menu-end">
<li>
<a class="dropdown-item" href="https://github.com/Airlabbcn/microbial-richness-troposphere">
<span class="dropdown-text">GitHub Repository</span></a>
</li>
<li>
<a class="dropdown-item" href="https://github.com/Airlabbcn/microbial-richness-troposphere/issues/new">
<span class="dropdown-text">Raise an issue</span></a>
</li>
</ul>
</li>
<li class="nav-item compact">
<a class="nav-link" href="mailto:alejandro.fontal.92@gmail.com"> <i aria-label="Mail" class="bi bi-envelope" role="img">
</i>
<span class="menu-text"></span></a>
</li>
</ul>
</div> <!-- /navcollapse -->
<div class="quarto-navbar-tools">
</div>
</div> <!-- /container-fluid -->
</nav>
</header>
<!-- content -->
<div class="quarto-container page-columns page-rows-contents page-layout-article page-navbar" id="quarto-content">
<!-- sidebar -->
<!-- margin-sidebar -->
<div class="sidebar margin-sidebar" id="quarto-margin-sidebar">
<nav class="toc-active" id="TOC" role="doc-toc">
<h2 id="toc-title">CONTENTS</h2>
<ul>
<li><a class="nav-link active" data-scroll-target="#preamble" href="#preamble" id="toc-preamble">Preamble</a>
<ul class="collapse">
<li><a class="nav-link" data-scroll-target="#imports" href="#imports" id="toc-imports">Imports</a></li>
<li><a class="nav-link" data-scroll-target="#pre-sets" href="#pre-sets" id="toc-pre-sets">Pre-sets</a></li>
</ul></li>
<li><a class="nav-link" data-scroll-target="#data-loading-and-wrangling" href="#data-loading-and-wrangling" id="toc-data-loading-and-wrangling">Data Loading and Wrangling</a></li>
<li><a class="nav-link" data-scroll-target="#analysis-figures" href="#analysis-figures" id="toc-analysis-figures">Analysis &amp; Figures</a>
<ul class="collapse">
<li><a class="nav-link" data-scroll-target="#sampling-schedule" href="#sampling-schedule" id="toc-sampling-schedule">Sampling Schedule</a></li>
<li><a class="nav-link" data-scroll-target="#abundance-comparisons" href="#abundance-comparisons" id="toc-abundance-comparisons">Abundance comparisons</a></li>
<li><a class="nav-link" data-scroll-target="#metagenomics-identification-rate" href="#metagenomics-identification-rate" id="toc-metagenomics-identification-rate">Metagenomics identification rate</a></li>
<li><a class="nav-link" data-scroll-target="#exclusive-otus-per-sample-type" href="#exclusive-otus-per-sample-type" id="toc-exclusive-otus-per-sample-type">Exclusive OTUs per sample type</a></li>
<li><a class="nav-link" data-scroll-target="#filtering-for-correlation-analysis" href="#filtering-for-correlation-analysis" id="toc-filtering-for-correlation-analysis">Filtering for correlation analysis</a></li>
<li><a class="nav-link" data-scroll-target="#diversity-per-sample" href="#diversity-per-sample" id="toc-diversity-per-sample">Diversity per sample</a></li>
<li><a class="nav-link" data-scroll-target="#distance-between-samples-nmds-and-permanova" href="#distance-between-samples-nmds-and-permanova" id="toc-distance-between-samples-nmds-and-permanova">Distance between samples (nMDS and PERMANOVA)</a></li>
<li><a class="nav-link" data-scroll-target="#barplots" href="#barplots" id="toc-barplots">Barplots</a></li>
<li><a class="nav-link" data-scroll-target="#correlation-with-heavy-metals" href="#correlation-with-heavy-metals" id="toc-correlation-with-heavy-metals">Correlation with heavy metals</a></li>
<li><a class="nav-link" data-scroll-target="#pathogen-bubble-plots" href="#pathogen-bubble-plots" id="toc-pathogen-bubble-plots">Pathogen bubble plots</a></li>
<li><a class="nav-link" data-scroll-target="#re-sequencing-nov23" href="#re-sequencing-nov23" id="toc-re-sequencing-nov23">Re-sequencing Nov’23</a></li>
</ul></li>
</ul>
</nav>
</div>
<!-- main -->
<main class="content" id="quarto-document-content">
<header class="quarto-title-block default" id="title-block-header">
<div class="quarto-title">
<div class="quarto-title-block"><div><h1 class="title">Microbial Richness in the Troposphere: Appeal Reanalysis</h1><button aria-expanded="false" class="btn code-tools-button dropdown-toggle" data-bs-toggle="dropdown" id="quarto-code-tools-menu" type="button"><i class="bi"></i> Code</button><ul aria-labelelledby="quarto-code-tools-menu" class="dropdown-menu dropdown-menu-end"><li><a class="dropdown-item" href="javascript:void(0)" id="quarto-show-all-code" role="button">Show All Code</a></li><li><a class="dropdown-item" href="javascript:void(0)" id="quarto-hide-all-code" role="button">Hide All Code</a></li></ul></div></div>
<p class="subtitle lead">Data Analysis and Figures</p>
</div>
<div class="quarto-title-meta-author">
<div class="quarto-title-meta-heading">Author</div>
<div class="quarto-title-meta-heading">Affiliation</div>
<div class="quarto-title-meta-contents">
<p class="author">Alejandro Fontal <a class="quarto-title-author-email" href="mailto:alejandro.fontal@isglobal.org"><i class="bi bi-envelope"></i></a> <a class="quarto-title-author-orcid" href="https://orcid.org/0000-0003-1138-2158"> <img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAA2ZpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADw/eHBhY2tldCBiZWdpbj0i77u/IiBpZD0iVzVNME1wQ2VoaUh6cmVTek5UY3prYzlkIj8+IDx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IkFkb2JlIFhNUCBDb3JlIDUuMC1jMDYwIDYxLjEzNDc3NywgMjAxMC8wMi8xMi0xNzozMjowMCAgICAgICAgIj4gPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4gPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIgeG1sbnM6eG1wTU09Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9tbS8iIHhtbG5zOnN0UmVmPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvc1R5cGUvUmVzb3VyY2VSZWYjIiB4bWxuczp4bXA9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC8iIHhtcE1NOk9yaWdpbmFsRG9jdW1lbnRJRD0ieG1wLmRpZDo1N0NEMjA4MDI1MjA2ODExOTk0QzkzNTEzRjZEQTg1NyIgeG1wTU06RG9jdW1lbnRJRD0ieG1wLmRpZDozM0NDOEJGNEZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wTU06SW5zdGFuY2VJRD0ieG1wLmlpZDozM0NDOEJGM0ZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wOkNyZWF0b3JUb29sPSJBZG9iZSBQaG90b3Nob3AgQ1M1IE1hY2ludG9zaCI+IDx4bXBNTTpEZXJpdmVkRnJvbSBzdFJlZjppbnN0YW5jZUlEPSJ4bXAuaWlkOkZDN0YxMTc0MDcyMDY4MTE5NUZFRDc5MUM2MUUwNEREIiBzdFJlZjpkb2N1bWVudElEPSJ4bXAuZGlkOjU3Q0QyMDgwMjUyMDY4MTE5OTRDOTM1MTNGNkRBODU3Ii8+IDwvcmRmOkRlc2NyaXB0aW9uPiA8L3JkZjpSREY+IDwveDp4bXBtZXRhPiA8P3hwYWNrZXQgZW5kPSJyIj8+84NovQAAAR1JREFUeNpiZEADy85ZJgCpeCB2QJM6AMQLo4yOL0AWZETSqACk1gOxAQN+cAGIA4EGPQBxmJA0nwdpjjQ8xqArmczw5tMHXAaALDgP1QMxAGqzAAPxQACqh4ER6uf5MBlkm0X4EGayMfMw/Pr7Bd2gRBZogMFBrv01hisv5jLsv9nLAPIOMnjy8RDDyYctyAbFM2EJbRQw+aAWw/LzVgx7b+cwCHKqMhjJFCBLOzAR6+lXX84xnHjYyqAo5IUizkRCwIENQQckGSDGY4TVgAPEaraQr2a4/24bSuoExcJCfAEJihXkWDj3ZAKy9EJGaEo8T0QSxkjSwORsCAuDQCD+QILmD1A9kECEZgxDaEZhICIzGcIyEyOl2RkgwAAhkmC+eAm0TAAAAABJRU5ErkJggg=="/></a></p>
</div>
<div class="quarto-title-meta-contents">
<p class="affiliation">
            Climate &amp; Health Program, ISGlobal
          </p>
</div>
</div>
<div class="quarto-title-meta">
<div>
<div class="quarto-title-meta-heading">Published</div>
<div class="quarto-title-meta-contents">
<p class="date">July 28, 2024</p>
</div>
</div>
</div>
</header>
<details>
<summary>
Click to show library imports and pre-sets
</summary>
<section class="level2" id="preamble">
<h2 class="anchored" data-anchor-id="preamble">Preamble</h2>
<section class="level3" id="imports">
<h3 class="anchored" data-anchor-id="imports">Imports</h3>
<div class="cell" data-execution_count="1" id="cell-4">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb1-1"><a aria-hidden="true" href="#cb1-1" tabindex="-1"></a><span class="im">import</span> numpy <span class="im">as</span> np</span>
<span id="cb1-2"><a aria-hidden="true" href="#cb1-2" tabindex="-1"></a><span class="im">import</span> pandas <span class="im">as</span> pd</span>
<span id="cb1-3"><a aria-hidden="true" href="#cb1-3" tabindex="-1"></a><span class="im">import</span> plotnine <span class="im">as</span> p9</span>
<span id="cb1-4"><a aria-hidden="true" href="#cb1-4" tabindex="-1"></a><span class="im">from</span> itertools <span class="im">import</span> product</span>
<span id="cb1-5"><a aria-hidden="true" href="#cb1-5" tabindex="-1"></a><span class="im">import</span> skbio.diversity <span class="im">as</span> div</span>
<span id="cb1-6"><a aria-hidden="true" href="#cb1-6" tabindex="-1"></a><span class="im">from</span> skbio <span class="im">import</span> DistanceMatrix</span>
<span id="cb1-7"><a aria-hidden="true" href="#cb1-7" tabindex="-1"></a><span class="im">from</span> skbio.stats.distance <span class="im">import</span> permanova</span>
<span id="cb1-8"><a aria-hidden="true" href="#cb1-8" tabindex="-1"></a><span class="im">from</span> IPython.display <span class="im">import</span> Image</span>
<span id="cb1-9"><a aria-hidden="true" href="#cb1-9" tabindex="-1"></a><span class="im">import</span> matplotlib.pyplot <span class="im">as</span> plt</span>
<span id="cb1-10"><a aria-hidden="true" href="#cb1-10" tabindex="-1"></a><span class="im">from</span> mizani.formatters <span class="im">import</span> percent_format</span>
<span id="cb1-11"><a aria-hidden="true" href="#cb1-11" tabindex="-1"></a><span class="im">from</span> matplotlib_inline.backend_inline <span class="im">import</span> set_matplotlib_formats</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
</div>
</section>
<section class="level3" id="pre-sets">
<h3 class="anchored" data-anchor-id="pre-sets">Pre-sets</h3>
<div class="cell" data-execution_count="2" id="cell-6">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb2-1"><a aria-hidden="true" href="#cb2-1" tabindex="-1"></a><span class="co"># Matplotlib settings</span></span>
<span id="cb2-2"><a aria-hidden="true" href="#cb2-2" tabindex="-1"></a><span class="im">from</span> matplotlib_inline.backend_inline <span class="im">import</span> set_matplotlib_formats</span>
<span id="cb2-3"><a aria-hidden="true" href="#cb2-3" tabindex="-1"></a>plt.rcParams[<span class="st">'svg.fonttype'</span>] <span class="op">=</span> <span class="st">'none'</span></span>
<span id="cb2-4"><a aria-hidden="true" href="#cb2-4" tabindex="-1"></a>set_matplotlib_formats(<span class="st">'retina'</span>)</span>
<span id="cb2-5"><a aria-hidden="true" href="#cb2-5" tabindex="-1"></a>plt.rcParams[<span class="st">'figure.dpi'</span>] <span class="op">=</span> <span class="dv">300</span></span>
<span id="cb2-6"><a aria-hidden="true" href="#cb2-6" tabindex="-1"></a><span class="co"># Plotnine settings (for figures)</span></span>
<span id="cb2-7"><a aria-hidden="true" href="#cb2-7" tabindex="-1"></a>p9.theme_set(</span>
<span id="cb2-8"><a aria-hidden="true" href="#cb2-8" tabindex="-1"></a>    p9.theme_bw()</span>
<span id="cb2-9"><a aria-hidden="true" href="#cb2-9" tabindex="-1"></a>    <span class="op">+</span> p9.theme(panel_grid<span class="op">=</span>p9.element_blank(),</span>
<span id="cb2-10"><a aria-hidden="true" href="#cb2-10" tabindex="-1"></a>               legend_background<span class="op">=</span>p9.element_blank(),</span>
<span id="cb2-11"><a aria-hidden="true" href="#cb2-11" tabindex="-1"></a>               panel_grid_major<span class="op">=</span>p9.element_line(size<span class="op">=</span><span class="fl">.5</span>, linetype<span class="op">=</span><span class="st">'dashed'</span>,</span>
<span id="cb2-12"><a aria-hidden="true" href="#cb2-12" tabindex="-1"></a>                                                alpha<span class="op">=</span><span class="fl">.15</span>, color<span class="op">=</span><span class="st">'black'</span>),</span>
<span id="cb2-13"><a aria-hidden="true" href="#cb2-13" tabindex="-1"></a>               dpi<span class="op">=</span><span class="dv">300</span></span>
<span id="cb2-14"><a aria-hidden="true" href="#cb2-14" tabindex="-1"></a>    )</span>
<span id="cb2-15"><a aria-hidden="true" href="#cb2-15" tabindex="-1"></a>)</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="3" id="cell-7">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb3-1"><a aria-hidden="true" href="#cb3-1" tabindex="-1"></a>bac_df <span class="op">=</span> (pd.read_csv(</span>
<span id="cb3-2"><a aria-hidden="true" href="#cb3-2" tabindex="-1"></a>    <span class="st">'../data/nov23/16S/Boostraping RDP_NBC_90/reduced_16S_genus_(ASV+taxonomy+S1-S22)_90_NB.csv'</span>)</span>
<span id="cb3-3"><a aria-hidden="true" href="#cb3-3" tabindex="-1"></a>          .query(<span class="st">'Order != "Chloroplast"'</span>)</span>
<span id="cb3-4"><a aria-hidden="true" href="#cb3-4" tabindex="-1"></a>          .query(<span class="st">'Kingdom.notna()'</span>)</span>
<span id="cb3-5"><a aria-hidden="true" href="#cb3-5" tabindex="-1"></a>)</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="4" id="cell-8">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb4-1"><a aria-hidden="true" href="#cb4-1" tabindex="-1"></a>sample_columns <span class="op">=</span> bac_df.columns[<span class="dv">6</span>:]</span>
<span id="cb4-2"><a aria-hidden="true" href="#cb4-2" tabindex="-1"></a>zfill_map <span class="op">=</span> {s: <span class="st">'S'</span> <span class="op">+</span> s[<span class="dv">1</span>:].zfill(<span class="dv">2</span>) <span class="cf">for</span> s <span class="kw">in</span> sample_columns}</span>
<span id="cb4-3"><a aria-hidden="true" href="#cb4-3" tabindex="-1"></a>sample_columns <span class="op">=</span> sample_columns.<span class="bu">map</span>(zfill_map)</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="5" id="cell-9">
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb5-1"><a aria-hidden="true" href="#cb5-1" tabindex="-1"></a>taxon_names <span class="op">=</span> [</span>
<span id="cb5-2"><a aria-hidden="true" href="#cb5-2" tabindex="-1"></a>    <span class="st">'Kingdom'</span>, </span>
<span id="cb5-3"><a aria-hidden="true" href="#cb5-3" tabindex="-1"></a>    <span class="st">'Phylum'</span>, </span>
<span id="cb5-4"><a aria-hidden="true" href="#cb5-4" tabindex="-1"></a>    <span class="st">'Class'</span>, </span>
<span id="cb5-5"><a aria-hidden="true" href="#cb5-5" tabindex="-1"></a>    <span class="st">'Order'</span>, </span>
<span id="cb5-6"><a aria-hidden="true" href="#cb5-6" tabindex="-1"></a>    <span class="st">'Family'</span>, </span>
<span id="cb5-7"><a aria-hidden="true" href="#cb5-7" tabindex="-1"></a>    <span class="st">'Genus'</span>]</span>
<span id="cb5-8"><a aria-hidden="true" href="#cb5-8" tabindex="-1"></a></span>
<span id="cb5-9"><a aria-hidden="true" href="#cb5-9" tabindex="-1"></a>taxon_plurals <span class="op">=</span> {</span>
<span id="cb5-10"><a aria-hidden="true" href="#cb5-10" tabindex="-1"></a>    <span class="st">'Phylum'</span>: <span class="st">'Phyla'</span>,</span>
<span id="cb5-11"><a aria-hidden="true" href="#cb5-11" tabindex="-1"></a>    <span class="st">'Class'</span>: <span class="st">'Classes'</span>,</span>
<span id="cb5-12"><a aria-hidden="true" href="#cb5-12" tabindex="-1"></a>    <span class="st">'Order'</span>: <span class="st">'Orders'</span>,</span>
<span id="cb5-13"><a aria-hidden="true" href="#cb5-13" tabindex="-1"></a>    <span class="st">'Family'</span>: <span class="st">'Families'</span>,</span>
<span id="cb5-14"><a aria-hidden="true" href="#cb5-14" tabindex="-1"></a>    <span class="st">'Genus'</span>: <span class="st">'Genera'</span>}</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="6" id="cell-10">
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb6-1"><a aria-hidden="true" href="#cb6-1" tabindex="-1"></a>bac_df <span class="op">=</span> bac_df.rename(columns<span class="op">=</span>zfill_map).fillna(<span class="st">'unknown'</span>)</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="7" id="cell-11">
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb7-1"><a aria-hidden="true" href="#cb7-1" tabindex="-1"></a>fun_df <span class="op">=</span> (pd.read_csv(<span class="st">'../data/nov23/ITS/bootstrap90/reduced_ITS_genus_(ASV+taxonomy+S1-S22)_90_NB.csv'</span>)</span>
<span id="cb7-2"><a aria-hidden="true" href="#cb7-2" tabindex="-1"></a>          .query(<span class="st">'Kingdom == "k__Fungi"'</span>)</span>
<span id="cb7-3"><a aria-hidden="true" href="#cb7-3" tabindex="-1"></a>)</span>
<span id="cb7-4"><a aria-hidden="true" href="#cb7-4" tabindex="-1"></a>fun_df[taxon_names] <span class="op">=</span> fun_df[taxon_names].<span class="bu">apply</span>(<span class="kw">lambda</span> x: x.<span class="bu">str</span>.split(<span class="st">'__'</span>).<span class="bu">str</span>[<span class="op">-</span><span class="dv">1</span>])</span>
<span id="cb7-5"><a aria-hidden="true" href="#cb7-5" tabindex="-1"></a>fun_df <span class="op">=</span> fun_df.rename(columns<span class="op">=</span>zfill_map).fillna(<span class="st">'unknown'</span>)</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="8" id="cell-12">
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb8-1"><a aria-hidden="true" href="#cb8-1" tabindex="-1"></a>abundance_df_genera <span class="op">=</span> pd.concat([bac_df, fun_df])</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="9" id="cell-13">
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb9-1"><a aria-hidden="true" href="#cb9-1" tabindex="-1"></a>samples_info <span class="op">=</span> (pd.read_csv(<span class="st">'../data/samples_info.csv'</span>)</span>
<span id="cb9-2"><a aria-hidden="true" href="#cb9-2" tabindex="-1"></a>        .assign(sample_id <span class="op">=</span> <span class="kw">lambda</span> dd:  <span class="st">"S"</span> <span class="op">+</span> dd.sample_id.<span class="bu">str</span>[<span class="op">-</span><span class="dv">2</span>:])</span>
<span id="cb9-3"><a aria-hidden="true" href="#cb9-3" tabindex="-1"></a>)</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="10" id="cell-14">
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb10-1"><a aria-hidden="true" href="#cb10-1" tabindex="-1"></a>inconclusive_list <span class="op">=</span> [<span class="st">'unknown'</span>,</span>
<span id="cb10-2"><a aria-hidden="true" href="#cb10-2" tabindex="-1"></a>                     <span class="st">'Cystobasidiomycetes_ord_Incertae_sedis'</span>,</span>
<span id="cb10-3"><a aria-hidden="true" href="#cb10-3" tabindex="-1"></a>                     <span class="st">'uncultured bacterium'</span>, <span class="st">'uncultured'</span>,</span>
<span id="cb10-4"><a aria-hidden="true" href="#cb10-4" tabindex="-1"></a>                     <span class="st">'uncultured Armatimonadetes bacterium'</span>,</span>
<span id="cb10-5"><a aria-hidden="true" href="#cb10-5" tabindex="-1"></a>                     <span class="st">'uncultured phototrophic eukaryote'</span>,</span>
<span id="cb10-6"><a aria-hidden="true" href="#cb10-6" tabindex="-1"></a>                     <span class="st">'Ambiguous_taxa'</span>,</span>
<span id="cb10-7"><a aria-hidden="true" href="#cb10-7" tabindex="-1"></a>                     <span class="st">'Hypocreales_fam_Incertae_sedis'</span>,</span>
<span id="cb10-8"><a aria-hidden="true" href="#cb10-8" tabindex="-1"></a>                     <span class="st">'Polyporales_fam_Incertae_sedis'</span>,</span>
<span id="cb10-9"><a aria-hidden="true" href="#cb10-9" tabindex="-1"></a>                     <span class="st">'Medicago truncatula (barrel medic)'</span>,</span>
<span id="cb10-10"><a aria-hidden="true" href="#cb10-10" tabindex="-1"></a>                     <span class="st">'Cantharellales_fam_Incertae_sedis'</span>,</span>
<span id="cb10-11"><a aria-hidden="true" href="#cb10-11" tabindex="-1"></a>                     <span class="st">'Russulales_fam_Incertae_sedis'</span>,</span>
<span id="cb10-12"><a aria-hidden="true" href="#cb10-12" tabindex="-1"></a>                     <span class="st">'uncultured Chloroflexi bacterium'</span>,</span>
<span id="cb10-13"><a aria-hidden="true" href="#cb10-13" tabindex="-1"></a>                     <span class="st">'uncultured soil bacterium'</span>,</span>
<span id="cb10-14"><a aria-hidden="true" href="#cb10-14" tabindex="-1"></a>                     <span class="st">'uncultured Chlorophyta'</span>,</span>
<span id="cb10-15"><a aria-hidden="true" href="#cb10-15" tabindex="-1"></a>                     <span class="st">'Pleosporales_fam_Incertae_sedis'</span>,</span>
<span id="cb10-16"><a aria-hidden="true" href="#cb10-16" tabindex="-1"></a>                     <span class="st">'Xylariales_fam_Incertae_sedis'</span>,</span>
<span id="cb10-17"><a aria-hidden="true" href="#cb10-17" tabindex="-1"></a>                     <span class="st">'Hymenochaetales_fam_Incertae_sedis'</span>,</span>
<span id="cb10-18"><a aria-hidden="true" href="#cb10-18" tabindex="-1"></a>                     <span class="st">'Entylomatales_fam_Incertae_sedis'</span>,</span>
<span id="cb10-19"><a aria-hidden="true" href="#cb10-19" tabindex="-1"></a>                     <span class="st">'Saccharomycetales_fam_Incertae_sedis'</span>,</span>
<span id="cb10-20"><a aria-hidden="true" href="#cb10-20" tabindex="-1"></a>                     <span class="st">'Phytophthora lateralis MPF4'</span>,</span>
<span id="cb10-21"><a aria-hidden="true" href="#cb10-21" tabindex="-1"></a>                     <span class="st">'Helotiales_fam_Incertae_sedis'</span>,</span>
<span id="cb10-22"><a aria-hidden="true" href="#cb10-22" tabindex="-1"></a>                     <span class="st">'candidate division SR1 bacterium taxon 345'</span>,</span>
<span id="cb10-23"><a aria-hidden="true" href="#cb10-23" tabindex="-1"></a>                     <span class="st">'Neo-b11'</span>,</span>
<span id="cb10-24"><a aria-hidden="true" href="#cb10-24" tabindex="-1"></a>                     <span class="st">'gut'</span>,</span>
<span id="cb10-25"><a aria-hidden="true" href="#cb10-25" tabindex="-1"></a>                     <span class="st">'candidate'</span>,</span>
<span id="cb10-26"><a aria-hidden="true" href="#cb10-26" tabindex="-1"></a>                     <span class="st">'Chloroplast'</span></span>
<span id="cb10-27"><a aria-hidden="true" href="#cb10-27" tabindex="-1"></a>                     ]</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="11" id="cell-15">
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb11-1"><a aria-hidden="true" href="#cb11-1" tabindex="-1"></a>(abundance_df_genera</span>
<span id="cb11-2"><a aria-hidden="true" href="#cb11-2" tabindex="-1"></a> .replace(<span class="st">'unknown'</span>, np.nan)</span>
<span id="cb11-3"><a aria-hidden="true" href="#cb11-3" tabindex="-1"></a> .assign(total_counts<span class="op">=</span><span class="kw">lambda</span> dd: dd[sample_columns].<span class="bu">sum</span>(axis<span class="op">=</span><span class="dv">1</span>))</span>
<span id="cb11-4"><a aria-hidden="true" href="#cb11-4" tabindex="-1"></a> .query(<span class="st">'total_counts &gt; 0'</span>)</span>
<span id="cb11-5"><a aria-hidden="true" href="#cb11-5" tabindex="-1"></a> .assign(Genus<span class="op">=</span><span class="kw">lambda</span> dd: </span>
<span id="cb11-6"><a aria-hidden="true" href="#cb11-6" tabindex="-1"></a>         np.where(dd[<span class="st">'Genus'</span>].isin(inconclusive_list), np.nan, dd[<span class="st">'Genus'</span>]))</span>
<span id="cb11-7"><a aria-hidden="true" href="#cb11-7" tabindex="-1"></a> .drop(columns<span class="op">=</span><span class="st">'total_counts'</span>)</span>
<span id="cb11-8"><a aria-hidden="true" href="#cb11-8" tabindex="-1"></a> .melt(id_vars<span class="op">=</span>taxon_names, var_name<span class="op">=</span><span class="st">'sample_id'</span>)</span>
<span id="cb11-9"><a aria-hidden="true" href="#cb11-9" tabindex="-1"></a> .merge(samples_info, on<span class="op">=</span><span class="st">'sample_id'</span>)</span>
<span id="cb11-10"><a aria-hidden="true" href="#cb11-10" tabindex="-1"></a> .assign(date<span class="op">=</span><span class="kw">lambda</span> dd: pd.to_datetime(dd.date))</span>
<span id="cb11-11"><a aria-hidden="true" href="#cb11-11" tabindex="-1"></a> .assign(month<span class="op">=</span><span class="kw">lambda</span> dd: dd.date.dt.month_name())</span>
<span id="cb11-12"><a aria-hidden="true" href="#cb11-12" tabindex="-1"></a> .query(<span class="st">'value &gt; 0'</span>)</span>
<span id="cb11-13"><a aria-hidden="true" href="#cb11-13" tabindex="-1"></a> .query(<span class="st">'month=="February"'</span>)</span>
<span id="cb11-14"><a aria-hidden="true" href="#cb11-14" tabindex="-1"></a> .groupby([<span class="st">'Kingdom'</span>])</span>
<span id="cb11-15"><a aria-hidden="true" href="#cb11-15" tabindex="-1"></a> .<span class="bu">apply</span>(<span class="kw">lambda</span> dd: dd[taxon_names[<span class="dv">1</span>:]].nunique())</span>
<span id="cb11-16"><a aria-hidden="true" href="#cb11-16" tabindex="-1"></a> .T</span>
<span id="cb11-17"><a aria-hidden="true" href="#cb11-17" tabindex="-1"></a>)</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="11">
<div>
<table class="dataframe caption-top table table-sm table-striped small" data-border="1" data-quarto-postprocess="true">
<thead>
<tr class="header">
<th data-quarto-table-cell-role="th">Kingdom</th>
<th data-quarto-table-cell-role="th">Bacteria</th>
<th data-quarto-table-cell-role="th">Fungi</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td data-quarto-table-cell-role="th">Phylum</td>
<td>17</td>
<td>2</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">Class</td>
<td>34</td>
<td>19</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">Order</td>
<td>72</td>
<td>60</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">Family</td>
<td>110</td>
<td>155</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">Genus</td>
<td>160</td>
<td>234</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="cell" data-execution_count="12" id="cell-16">
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb12-1"><a aria-hidden="true" href="#cb12-1" tabindex="-1"></a>abundance_df_phyla <span class="op">=</span> (abundance_df_genera</span>
<span id="cb12-2"><a aria-hidden="true" href="#cb12-2" tabindex="-1"></a> .groupby([<span class="st">'Kingdom'</span>, <span class="st">'Phylum'</span>])</span>
<span id="cb12-3"><a aria-hidden="true" href="#cb12-3" tabindex="-1"></a> [sample_columns]</span>
<span id="cb12-4"><a aria-hidden="true" href="#cb12-4" tabindex="-1"></a> .<span class="bu">sum</span>()</span>
<span id="cb12-5"><a aria-hidden="true" href="#cb12-5" tabindex="-1"></a> .astype(<span class="bu">int</span>)</span>
<span id="cb12-6"><a aria-hidden="true" href="#cb12-6" tabindex="-1"></a> .reset_index()</span>
<span id="cb12-7"><a aria-hidden="true" href="#cb12-7" tabindex="-1"></a>)</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="13" id="cell-17">
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb13-1"><a aria-hidden="true" href="#cb13-1" tabindex="-1"></a>abundance_df_phyla</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="13">
<div>
<table class="dataframe caption-top table table-sm table-striped small" data-border="1" data-quarto-postprocess="true">
<thead>
<tr class="header">
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th">Kingdom</th>
<th data-quarto-table-cell-role="th">Phylum</th>
<th data-quarto-table-cell-role="th">S01</th>
<th data-quarto-table-cell-role="th">S02</th>
<th data-quarto-table-cell-role="th">S03</th>
<th data-quarto-table-cell-role="th">S04</th>
<th data-quarto-table-cell-role="th">S05</th>
<th data-quarto-table-cell-role="th">S06</th>
<th data-quarto-table-cell-role="th">S07</th>
<th data-quarto-table-cell-role="th">S08</th>
<th data-quarto-table-cell-role="th">...</th>
<th data-quarto-table-cell-role="th">S12</th>
<th data-quarto-table-cell-role="th">S13</th>
<th data-quarto-table-cell-role="th">S15</th>
<th data-quarto-table-cell-role="th">S16</th>
<th data-quarto-table-cell-role="th">S17</th>
<th data-quarto-table-cell-role="th">S18</th>
<th data-quarto-table-cell-role="th">S19</th>
<th data-quarto-table-cell-role="th">S20</th>
<th data-quarto-table-cell-role="th">S21</th>
<th data-quarto-table-cell-role="th">S22</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td data-quarto-table-cell-role="th">0</td>
<td>Bacteria</td>
<td>Acidobacteriota</td>
<td>9</td>
<td>9</td>
<td>423</td>
<td>4</td>
<td>6</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>...</td>
<td>11</td>
<td>11</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>64</td>
<td>0</td>
<td>0</td>
<td>0</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">1</td>
<td>Bacteria</td>
<td>Actinobacteriota</td>
<td>508</td>
<td>282</td>
<td>457</td>
<td>101</td>
<td>144</td>
<td>138</td>
<td>182</td>
<td>39</td>
<td>...</td>
<td>168</td>
<td>396</td>
<td>78</td>
<td>246</td>
<td>45</td>
<td>696</td>
<td>438</td>
<td>493</td>
<td>678</td>
<td>423</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">2</td>
<td>Bacteria</td>
<td>Armatimonadota</td>
<td>5</td>
<td>0</td>
<td>77</td>
<td>0</td>
<td>0</td>
<td>11</td>
<td>0</td>
<td>0</td>
<td>...</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>5</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">3</td>
<td>Bacteria</td>
<td>Bacteroidota</td>
<td>401</td>
<td>54</td>
<td>385</td>
<td>6</td>
<td>13</td>
<td>14</td>
<td>0</td>
<td>8</td>
<td>...</td>
<td>62</td>
<td>3135</td>
<td>1134</td>
<td>2685</td>
<td>664</td>
<td>3930</td>
<td>3635</td>
<td>3338</td>
<td>8735</td>
<td>2549</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">4</td>
<td>Bacteria</td>
<td>Campylobacterota</td>
<td>6</td>
<td>4</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>...</td>
<td>0</td>
<td>19</td>
<td>0</td>
<td>44</td>
<td>0</td>
<td>0</td>
<td>33</td>
<td>32</td>
<td>98</td>
<td>9</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">5</td>
<td>Bacteria</td>
<td>Chloroflexi</td>
<td>11</td>
<td>0</td>
<td>23</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>6</td>
<td>0</td>
<td>...</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>38</td>
<td>0</td>
<td>0</td>
<td>0</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">6</td>
<td>Bacteria</td>
<td>Cyanobacteria</td>
<td>0</td>
<td>11</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>9</td>
<td>0</td>
<td>0</td>
<td>...</td>
<td>0</td>
<td>49</td>
<td>11</td>
<td>33</td>
<td>0</td>
<td>0</td>
<td>13</td>
<td>36</td>
<td>94</td>
<td>0</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">7</td>
<td>Bacteria</td>
<td>Deinococcota</td>
<td>6</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>...</td>
<td>38</td>
<td>19</td>
<td>12</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">8</td>
<td>Bacteria</td>
<td>Dependentiae</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>10</td>
<td>0</td>
<td>0</td>
<td>...</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">9</td>
<td>Bacteria</td>
<td>Desulfobacterota</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>...</td>
<td>0</td>
<td>59</td>
<td>0</td>
<td>66</td>
<td>0</td>
<td>177</td>
<td>0</td>
<td>25</td>
<td>124</td>
<td>25</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">10</td>
<td>Bacteria</td>
<td>Firmicutes</td>
<td>1712</td>
<td>449</td>
<td>720</td>
<td>189</td>
<td>185</td>
<td>191</td>
<td>218</td>
<td>120</td>
<td>...</td>
<td>190</td>
<td>5332</td>
<td>2430</td>
<td>4700</td>
<td>1165</td>
<td>8791</td>
<td>7617</td>
<td>7467</td>
<td>17434</td>
<td>6486</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">11</td>
<td>Bacteria</td>
<td>Fusobacteriota</td>
<td>0</td>
<td>13</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>10</td>
<td>0</td>
<td>0</td>
<td>...</td>
<td>0</td>
<td>37</td>
<td>30</td>
<td>18</td>
<td>9</td>
<td>151</td>
<td>54</td>
<td>28</td>
<td>167</td>
<td>48</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">12</td>
<td>Bacteria</td>
<td>Gemmatimonadota</td>
<td>0</td>
<td>16</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>...</td>
<td>0</td>
<td>5</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">13</td>
<td>Bacteria</td>
<td>Myxococcota</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>26</td>
<td>0</td>
<td>0</td>
<td>...</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>0</td>
<td>0</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">14</td>
<td>Bacteria</td>
<td>Patescibacteria</td>
<td>75</td>
<td>0</td>
<td>3</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>0</td>
<td>6</td>
<td>...</td>
<td>3</td>
<td>9</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>9</td>
<td>0</td>
<td>3</td>
<td>0</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">15</td>
<td>Bacteria</td>
<td>Planctomycetota</td>
<td>18</td>
<td>0</td>
<td>64</td>
<td>6</td>
<td>0</td>
<td>7</td>
<td>0</td>
<td>5</td>
<td>...</td>
<td>4</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>15</td>
<td>0</td>
<td>0</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">16</td>
<td>Bacteria</td>
<td>Proteobacteria</td>
<td>163942</td>
<td>163508</td>
<td>167209</td>
<td>80363</td>
<td>67083</td>
<td>64785</td>
<td>47424</td>
<td>40914</td>
<td>...</td>
<td>63862</td>
<td>51290</td>
<td>42165</td>
<td>54696</td>
<td>27010</td>
<td>86805</td>
<td>77241</td>
<td>42186</td>
<td>33958</td>
<td>64411</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">17</td>
<td>Bacteria</td>
<td>Verrucomicrobiota</td>
<td>24</td>
<td>0</td>
<td>50</td>
<td>0</td>
<td>0</td>
<td>7</td>
<td>0</td>
<td>0</td>
<td>...</td>
<td>0</td>
<td>96</td>
<td>0</td>
<td>40</td>
<td>0</td>
<td>120</td>
<td>93</td>
<td>118</td>
<td>201</td>
<td>86</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">18</td>
<td>Bacteria</td>
<td>unknown</td>
<td>6</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>5</td>
<td>0</td>
<td>2</td>
<td>0</td>
<td>...</td>
<td>2</td>
<td>258</td>
<td>159</td>
<td>300</td>
<td>39</td>
<td>369</td>
<td>381</td>
<td>257</td>
<td>683</td>
<td>237</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">19</td>
<td>Fungi</td>
<td>Ascomycota</td>
<td>161470</td>
<td>168168</td>
<td>259986</td>
<td>104351</td>
<td>111155</td>
<td>0</td>
<td>108411</td>
<td>75315</td>
<td>...</td>
<td>49801</td>
<td>45344</td>
<td>146977</td>
<td>73190</td>
<td>41023</td>
<td>41262</td>
<td>62190</td>
<td>66732</td>
<td>114059</td>
<td>109389</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">20</td>
<td>Fungi</td>
<td>Basidiomycota</td>
<td>12998</td>
<td>13292</td>
<td>24392</td>
<td>8527</td>
<td>7078</td>
<td>0</td>
<td>4612</td>
<td>12230</td>
<td>...</td>
<td>4712</td>
<td>3255</td>
<td>6415</td>
<td>4777</td>
<td>1346</td>
<td>18650</td>
<td>6893</td>
<td>27544</td>
<td>13425</td>
<td>28489</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">21</td>
<td>Fungi</td>
<td>unknown</td>
<td>403</td>
<td>929</td>
<td>1581</td>
<td>0</td>
<td>63</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>...</td>
<td>424</td>
<td>326</td>
<td>3808</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>305</td>
<td>482</td>
<td>0</td>
<td>0</td>
</tr>
</tbody>
</table>
<p>22 rows × 23 columns</p>
</div>
</div>
</div>
<div class="cell" data-execution_count="14" id="cell-18">
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb14-1"><a aria-hidden="true" href="#cb14-1" tabindex="-1"></a>abundance_df_genera.to_csv(<span class="st">'../data/nov23/16S/Boostraping RDP_NBC_90/full_abundance_genera.csv'</span>, index<span class="op">=</span><span class="va">False</span>)</span>
<span id="cb14-2"><a aria-hidden="true" href="#cb14-2" tabindex="-1"></a>abundance_df_phyla.to_csv(<span class="st">'../data/nov23/16S/Boostraping RDP_NBC_90/full_abundance_phyla.csv'</span>, index<span class="op">=</span><span class="va">False</span>)</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="15" id="cell-19">
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb15-1"><a aria-hidden="true" href="#cb15-1" tabindex="-1"></a>only_assigned_bac_df <span class="op">=</span> bac_df.query(<span class="st">'Genus.notna()'</span>)</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="16" id="cell-20">
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb16-1"><a aria-hidden="true" href="#cb16-1" tabindex="-1"></a>shannon_only_assigned <span class="op">=</span> (div.alpha_diversity(metric<span class="op">=</span><span class="st">'shannon'</span>,</span>
<span id="cb16-2"><a aria-hidden="true" href="#cb16-2" tabindex="-1"></a>                    counts<span class="op">=</span>only_assigned_bac_df[sample_columns].T.values,</span>
<span id="cb16-3"><a aria-hidden="true" href="#cb16-3" tabindex="-1"></a>                    ids<span class="op">=</span>sample_columns</span>
<span id="cb16-4"><a aria-hidden="true" href="#cb16-4" tabindex="-1"></a>                    ).rename(<span class="st">'shannon_only_assigned'</span>)</span>
<span id="cb16-5"><a aria-hidden="true" href="#cb16-5" tabindex="-1"></a>)</span>
<span id="cb16-6"><a aria-hidden="true" href="#cb16-6" tabindex="-1"></a></span>
<span id="cb16-7"><a aria-hidden="true" href="#cb16-7" tabindex="-1"></a>shannon_all <span class="op">=</span> (div.alpha_diversity(metric<span class="op">=</span><span class="st">'shannon'</span>,</span>
<span id="cb16-8"><a aria-hidden="true" href="#cb16-8" tabindex="-1"></a>                    counts<span class="op">=</span>bac_df[sample_columns].T.values,</span>
<span id="cb16-9"><a aria-hidden="true" href="#cb16-9" tabindex="-1"></a>                    ids<span class="op">=</span>sample_columns</span>
<span id="cb16-10"><a aria-hidden="true" href="#cb16-10" tabindex="-1"></a>                    ).rename(<span class="st">'shannon_all'</span>)</span>
<span id="cb16-11"><a aria-hidden="true" href="#cb16-11" tabindex="-1"></a>)</span>
<span id="cb16-12"><a aria-hidden="true" href="#cb16-12" tabindex="-1"></a></span>
<span id="cb16-13"><a aria-hidden="true" href="#cb16-13" tabindex="-1"></a>chao_only_assigned <span class="op">=</span> (div.alpha_diversity(metric<span class="op">=</span><span class="st">'chao1'</span>,</span>
<span id="cb16-14"><a aria-hidden="true" href="#cb16-14" tabindex="-1"></a>                    counts<span class="op">=</span>only_assigned_bac_df[sample_columns].T.values,</span>
<span id="cb16-15"><a aria-hidden="true" href="#cb16-15" tabindex="-1"></a>                    ids<span class="op">=</span>sample_columns</span>
<span id="cb16-16"><a aria-hidden="true" href="#cb16-16" tabindex="-1"></a>                    ).rename(<span class="st">'chao_only_assigned'</span>)</span>
<span id="cb16-17"><a aria-hidden="true" href="#cb16-17" tabindex="-1"></a>)</span>
<span id="cb16-18"><a aria-hidden="true" href="#cb16-18" tabindex="-1"></a></span>
<span id="cb16-19"><a aria-hidden="true" href="#cb16-19" tabindex="-1"></a>chao_all <span class="op">=</span> (div.alpha_diversity(metric<span class="op">=</span><span class="st">'chao1'</span>,</span>
<span id="cb16-20"><a aria-hidden="true" href="#cb16-20" tabindex="-1"></a>                    counts<span class="op">=</span>bac_df[sample_columns].T.values,</span>
<span id="cb16-21"><a aria-hidden="true" href="#cb16-21" tabindex="-1"></a>                    ids<span class="op">=</span>sample_columns</span>
<span id="cb16-22"><a aria-hidden="true" href="#cb16-22" tabindex="-1"></a>                    ).rename(<span class="st">'chao_all'</span>)</span>
<span id="cb16-23"><a aria-hidden="true" href="#cb16-23" tabindex="-1"></a>)</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="17" id="cell-21">
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb17-1"><a aria-hidden="true" href="#cb17-1" tabindex="-1"></a>(shannon_all</span>
<span id="cb17-2"><a aria-hidden="true" href="#cb17-2" tabindex="-1"></a> .reset_index()</span>
<span id="cb17-3"><a aria-hidden="true" href="#cb17-3" tabindex="-1"></a> .merge(shannon_only_assigned.reset_index())</span>
<span id="cb17-4"><a aria-hidden="true" href="#cb17-4" tabindex="-1"></a> .assign(index<span class="op">=</span><span class="kw">lambda</span> df: <span class="st">'S'</span> <span class="op">+</span> df[<span class="st">'index'</span>].<span class="bu">str</span>[<span class="dv">1</span>:].<span class="bu">str</span>.zfill(<span class="dv">2</span>))</span>
<span id="cb17-5"><a aria-hidden="true" href="#cb17-5" tabindex="-1"></a> .melt(id_vars<span class="op">=</span><span class="st">'index'</span>)</span>
<span id="cb17-6"><a aria-hidden="true" href="#cb17-6" tabindex="-1"></a> .replace({<span class="st">'shannon_all'</span>: <span class="st">'Keeping all unassigned as different'</span>,</span>
<span id="cb17-7"><a aria-hidden="true" href="#cb17-7" tabindex="-1"></a>           <span class="st">'shannon_only_assigned'</span>: <span class="st">'Dropping unassigned genera'</span>})</span>
<span id="cb17-8"><a aria-hidden="true" href="#cb17-8" tabindex="-1"></a> .pipe(<span class="kw">lambda</span> dd: p9.ggplot(dd) <span class="op">+</span> p9.aes(x<span class="op">=</span><span class="st">'index'</span>, y<span class="op">=</span><span class="st">'value'</span>, fill<span class="op">=</span><span class="st">'variable'</span>)</span>
<span id="cb17-9"><a aria-hidden="true" href="#cb17-9" tabindex="-1"></a>       <span class="op">+</span> p9.geom_col(position<span class="op">=</span><span class="st">'dodge'</span>)</span>
<span id="cb17-10"><a aria-hidden="true" href="#cb17-10" tabindex="-1"></a>       <span class="op">+</span> p9.theme(legend_position<span class="op">=</span>(<span class="fl">.5</span>, <span class="fl">.875</span>),</span>
<span id="cb17-11"><a aria-hidden="true" href="#cb17-11" tabindex="-1"></a>                  figure_size<span class="op">=</span>(<span class="dv">6</span>, <span class="dv">3</span>)</span>
<span id="cb17-12"><a aria-hidden="true" href="#cb17-12" tabindex="-1"></a>                  )</span>
<span id="cb17-13"><a aria-hidden="true" href="#cb17-13" tabindex="-1"></a>       <span class="op">+</span> p9.labs(x<span class="op">=</span><span class="st">''</span>, y<span class="op">=</span><span class="st">'Shannon diversity'</span>, fill<span class="op">=</span><span class="st">''</span>)</span>
<span id="cb17-14"><a aria-hidden="true" href="#cb17-14" tabindex="-1"></a>       )</span>
<span id="cb17-15"><a aria-hidden="true" href="#cb17-15" tabindex="-1"></a>)</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img class="figure-img" height="900" src="microbial_diversity_analysis_files/figure-html/cell-18-output-1.png" width="1800"/></p>
</figure>
</div>
</div>
</div>
<div class="cell" data-execution_count="18" id="cell-22">
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb18-1"><a aria-hidden="true" href="#cb18-1" tabindex="-1"></a>shannon_16s <span class="op">=</span> (abundance_df_genera</span>
<span id="cb18-2"><a aria-hidden="true" href="#cb18-2" tabindex="-1"></a> .query(<span class="st">'Kingdom=="Bacteria"'</span>)</span>
<span id="cb18-3"><a aria-hidden="true" href="#cb18-3" tabindex="-1"></a> .query(<span class="st">'Genus!="unknown"'</span>)</span>
<span id="cb18-4"><a aria-hidden="true" href="#cb18-4" tabindex="-1"></a> .pipe(<span class="kw">lambda</span> dd: div.alpha_diversity(metric<span class="op">=</span><span class="st">'shannon'</span>,</span>
<span id="cb18-5"><a aria-hidden="true" href="#cb18-5" tabindex="-1"></a>                                      counts<span class="op">=</span>dd[sample_columns].T.values,</span>
<span id="cb18-6"><a aria-hidden="true" href="#cb18-6" tabindex="-1"></a>                    ids<span class="op">=</span>sample_columns</span>
<span id="cb18-7"><a aria-hidden="true" href="#cb18-7" tabindex="-1"></a>                    ).rename(<span class="st">'shannon'</span>)</span>
<span id="cb18-8"><a aria-hidden="true" href="#cb18-8" tabindex="-1"></a> )</span>
<span id="cb18-9"><a aria-hidden="true" href="#cb18-9" tabindex="-1"></a> .reset_index()</span>
<span id="cb18-10"><a aria-hidden="true" href="#cb18-10" tabindex="-1"></a> .rename(columns<span class="op">=</span>{<span class="st">'index'</span>: <span class="st">'sample_id'</span>})</span>
<span id="cb18-11"><a aria-hidden="true" href="#cb18-11" tabindex="-1"></a> .assign(source<span class="op">=</span><span class="st">'16S'</span>)</span>
<span id="cb18-12"><a aria-hidden="true" href="#cb18-12" tabindex="-1"></a>)</span>
<span id="cb18-13"><a aria-hidden="true" href="#cb18-13" tabindex="-1"></a></span>
<span id="cb18-14"><a aria-hidden="true" href="#cb18-14" tabindex="-1"></a>shannon_its <span class="op">=</span> (abundance_df_genera</span>
<span id="cb18-15"><a aria-hidden="true" href="#cb18-15" tabindex="-1"></a>    .query(<span class="st">'Kingdom=="Fungi"'</span>)</span>
<span id="cb18-16"><a aria-hidden="true" href="#cb18-16" tabindex="-1"></a>    .query(<span class="st">'Genus!="unknown"'</span>)</span>
<span id="cb18-17"><a aria-hidden="true" href="#cb18-17" tabindex="-1"></a>    .drop(columns<span class="op">=</span><span class="st">'S06'</span>)</span>
<span id="cb18-18"><a aria-hidden="true" href="#cb18-18" tabindex="-1"></a>    .pipe(<span class="kw">lambda</span> dd: div.alpha_diversity(metric<span class="op">=</span><span class="st">'shannon'</span>,</span>
<span id="cb18-19"><a aria-hidden="true" href="#cb18-19" tabindex="-1"></a>          counts<span class="op">=</span>dd[[col <span class="cf">for</span> col <span class="kw">in</span> sample_columns <span class="cf">if</span> col <span class="op">!=</span> <span class="st">'S06'</span>]].T.values,</span>
<span id="cb18-20"><a aria-hidden="true" href="#cb18-20" tabindex="-1"></a>                        ids<span class="op">=</span>[col <span class="cf">for</span> col <span class="kw">in</span> sample_columns <span class="cf">if</span> col <span class="op">!=</span> <span class="st">'S06'</span>]</span>
<span id="cb18-21"><a aria-hidden="true" href="#cb18-21" tabindex="-1"></a>                        ).rename(<span class="st">'shannon'</span>)</span>
<span id="cb18-22"><a aria-hidden="true" href="#cb18-22" tabindex="-1"></a>    )</span>
<span id="cb18-23"><a aria-hidden="true" href="#cb18-23" tabindex="-1"></a>    .reset_index()</span>
<span id="cb18-24"><a aria-hidden="true" href="#cb18-24" tabindex="-1"></a>    .rename(columns<span class="op">=</span>{<span class="st">'index'</span>: <span class="st">'sample_id'</span>})</span>
<span id="cb18-25"><a aria-hidden="true" href="#cb18-25" tabindex="-1"></a>    .assign(source<span class="op">=</span><span class="st">'ITS'</span>)</span>
<span id="cb18-26"><a aria-hidden="true" href="#cb18-26" tabindex="-1"></a>    )</span>
<span id="cb18-27"><a aria-hidden="true" href="#cb18-27" tabindex="-1"></a></span>
<span id="cb18-28"><a aria-hidden="true" href="#cb18-28" tabindex="-1"></a>chao_16s <span class="op">=</span> (abundance_df_genera</span>
<span id="cb18-29"><a aria-hidden="true" href="#cb18-29" tabindex="-1"></a>    .query(<span class="st">'Kingdom=="Bacteria"'</span>)</span>
<span id="cb18-30"><a aria-hidden="true" href="#cb18-30" tabindex="-1"></a>    .query(<span class="st">'Genus!="unknown"'</span>)</span>
<span id="cb18-31"><a aria-hidden="true" href="#cb18-31" tabindex="-1"></a>    .pipe(<span class="kw">lambda</span> dd: div.alpha_diversity(metric<span class="op">=</span><span class="st">'chao1'</span>,</span>
<span id="cb18-32"><a aria-hidden="true" href="#cb18-32" tabindex="-1"></a>                                        counts<span class="op">=</span>dd[sample_columns].T.values,</span>
<span id="cb18-33"><a aria-hidden="true" href="#cb18-33" tabindex="-1"></a>                        ids<span class="op">=</span>sample_columns</span>
<span id="cb18-34"><a aria-hidden="true" href="#cb18-34" tabindex="-1"></a>                        ).rename(<span class="st">'chao'</span>)</span>
<span id="cb18-35"><a aria-hidden="true" href="#cb18-35" tabindex="-1"></a>    )</span>
<span id="cb18-36"><a aria-hidden="true" href="#cb18-36" tabindex="-1"></a>    .reset_index()</span>
<span id="cb18-37"><a aria-hidden="true" href="#cb18-37" tabindex="-1"></a>    .rename(columns<span class="op">=</span>{<span class="st">'index'</span>: <span class="st">'sample_id'</span>})</span>
<span id="cb18-38"><a aria-hidden="true" href="#cb18-38" tabindex="-1"></a>    .assign(source<span class="op">=</span><span class="st">'16S'</span>)</span>
<span id="cb18-39"><a aria-hidden="true" href="#cb18-39" tabindex="-1"></a>    )</span>
<span id="cb18-40"><a aria-hidden="true" href="#cb18-40" tabindex="-1"></a></span>
<span id="cb18-41"><a aria-hidden="true" href="#cb18-41" tabindex="-1"></a>chao_its <span class="op">=</span> (abundance_df_genera</span>
<span id="cb18-42"><a aria-hidden="true" href="#cb18-42" tabindex="-1"></a>    .query(<span class="st">'Kingdom=="Fungi"'</span>)</span>
<span id="cb18-43"><a aria-hidden="true" href="#cb18-43" tabindex="-1"></a>    .drop(columns<span class="op">=</span><span class="st">'S06'</span>)</span>
<span id="cb18-44"><a aria-hidden="true" href="#cb18-44" tabindex="-1"></a>    .query(<span class="st">'Genus!="unknown"'</span>)</span>
<span id="cb18-45"><a aria-hidden="true" href="#cb18-45" tabindex="-1"></a>    .pipe(<span class="kw">lambda</span> dd: div.alpha_diversity(metric<span class="op">=</span><span class="st">'chao1'</span>,</span>
<span id="cb18-46"><a aria-hidden="true" href="#cb18-46" tabindex="-1"></a>            counts<span class="op">=</span>dd[[col <span class="cf">for</span> col <span class="kw">in</span> sample_columns <span class="cf">if</span> col <span class="op">!=</span> <span class="st">'S06'</span>]].T.values,</span>
<span id="cb18-47"><a aria-hidden="true" href="#cb18-47" tabindex="-1"></a>                        ids<span class="op">=</span>[col <span class="cf">for</span> col <span class="kw">in</span> sample_columns <span class="cf">if</span> col <span class="op">!=</span> <span class="st">'S06'</span>]</span>
<span id="cb18-48"><a aria-hidden="true" href="#cb18-48" tabindex="-1"></a>                        ).rename(<span class="st">'chao'</span>)</span>
<span id="cb18-49"><a aria-hidden="true" href="#cb18-49" tabindex="-1"></a>    )</span>
<span id="cb18-50"><a aria-hidden="true" href="#cb18-50" tabindex="-1"></a>    .reset_index()</span>
<span id="cb18-51"><a aria-hidden="true" href="#cb18-51" tabindex="-1"></a>    .rename(columns<span class="op">=</span>{<span class="st">'index'</span>: <span class="st">'sample_id'</span>})</span>
<span id="cb18-52"><a aria-hidden="true" href="#cb18-52" tabindex="-1"></a>    .assign(source<span class="op">=</span><span class="st">'ITS'</span>)</span>
<span id="cb18-53"><a aria-hidden="true" href="#cb18-53" tabindex="-1"></a>    )</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="19" id="cell-23">
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb19-1"><a aria-hidden="true" href="#cb19-1" tabindex="-1"></a>diversity_df <span class="op">=</span> pd.concat(</span>
<span id="cb19-2"><a aria-hidden="true" href="#cb19-2" tabindex="-1"></a>    [chao_16s.merge(shannon_16s),</span>
<span id="cb19-3"><a aria-hidden="true" href="#cb19-3" tabindex="-1"></a>     chao_its.merge(shannon_its)]</span>
<span id="cb19-4"><a aria-hidden="true" href="#cb19-4" tabindex="-1"></a>)     [[<span class="st">'sample_id'</span>, <span class="st">'chao'</span>, <span class="st">'shannon'</span>, <span class="st">'source'</span>]]</span>
<span id="cb19-5"><a aria-hidden="true" href="#cb19-5" tabindex="-1"></a></span>
<span id="cb19-6"><a aria-hidden="true" href="#cb19-6" tabindex="-1"></a>diversity_df.to_csv(<span class="st">'../data/nov23/16S/Boostraping RDP_NBC_90/diversity_df.csv'</span>, index<span class="op">=</span><span class="va">False</span>)</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="20" id="cell-24">
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb20-1"><a aria-hidden="true" href="#cb20-1" tabindex="-1"></a>(chao_all</span>
<span id="cb20-2"><a aria-hidden="true" href="#cb20-2" tabindex="-1"></a> .reset_index()</span>
<span id="cb20-3"><a aria-hidden="true" href="#cb20-3" tabindex="-1"></a> .merge(chao_only_assigned.reset_index())</span>
<span id="cb20-4"><a aria-hidden="true" href="#cb20-4" tabindex="-1"></a> .assign(index<span class="op">=</span><span class="kw">lambda</span> df: <span class="st">'S'</span> <span class="op">+</span> df[<span class="st">'index'</span>].<span class="bu">str</span>[<span class="dv">1</span>:].<span class="bu">str</span>.zfill(<span class="dv">2</span>))</span>
<span id="cb20-5"><a aria-hidden="true" href="#cb20-5" tabindex="-1"></a> .melt(id_vars<span class="op">=</span><span class="st">'index'</span>)</span>
<span id="cb20-6"><a aria-hidden="true" href="#cb20-6" tabindex="-1"></a> .replace({<span class="st">'chao_all'</span>: <span class="st">'Keeping all unassigned as different'</span>,</span>
<span id="cb20-7"><a aria-hidden="true" href="#cb20-7" tabindex="-1"></a>           <span class="st">'chao_only_assigned'</span>: <span class="st">'Dropping unassigned genera'</span>})</span>
<span id="cb20-8"><a aria-hidden="true" href="#cb20-8" tabindex="-1"></a> .pipe(<span class="kw">lambda</span> dd: p9.ggplot(dd) <span class="op">+</span> p9.aes(x<span class="op">=</span><span class="st">'index'</span>, y<span class="op">=</span><span class="st">'value'</span>, fill<span class="op">=</span><span class="st">'variable'</span>)</span>
<span id="cb20-9"><a aria-hidden="true" href="#cb20-9" tabindex="-1"></a>       <span class="op">+</span> p9.geom_col(position<span class="op">=</span><span class="st">'dodge'</span>)</span>
<span id="cb20-10"><a aria-hidden="true" href="#cb20-10" tabindex="-1"></a>       <span class="op">+</span> p9.theme(legend_position<span class="op">=</span>(<span class="fl">.5</span>, <span class="fl">.9</span>),</span>
<span id="cb20-11"><a aria-hidden="true" href="#cb20-11" tabindex="-1"></a>                  figure_size<span class="op">=</span>(<span class="dv">6</span>, <span class="dv">3</span>)</span>
<span id="cb20-12"><a aria-hidden="true" href="#cb20-12" tabindex="-1"></a>                  )</span>
<span id="cb20-13"><a aria-hidden="true" href="#cb20-13" tabindex="-1"></a>       <span class="op">+</span> p9.labs(x<span class="op">=</span><span class="st">''</span>, y<span class="op">=</span><span class="st">'CHAO1 diversity'</span>, fill<span class="op">=</span><span class="st">''</span>)</span>
<span id="cb20-14"><a aria-hidden="true" href="#cb20-14" tabindex="-1"></a>       )</span>
<span id="cb20-15"><a aria-hidden="true" href="#cb20-15" tabindex="-1"></a>)</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img class="figure-img" height="900" src="microbial_diversity_analysis_files/figure-html/cell-21-output-1.png" width="1800"/></p>
</figure>
</div>
</div>
</div>
</section></section></details>
<section class="level2" id="data-loading-and-wrangling">
<h2 class="anchored" data-anchor-id="data-loading-and-wrangling">Data Loading and Wrangling</h2>
<div class="cell" data-execution_count="21" id="cell-27">
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb21-1"><a aria-hidden="true" href="#cb21-1" tabindex="-1"></a>abundance_long_genera <span class="op">=</span> (abundance_df_genera</span>
<span id="cb21-2"><a aria-hidden="true" href="#cb21-2" tabindex="-1"></a>      .melt(id_vars<span class="op">=</span>taxon_names,</span>
<span id="cb21-3"><a aria-hidden="true" href="#cb21-3" tabindex="-1"></a>            var_name<span class="op">=</span><span class="st">'sample_id'</span>,</span>
<span id="cb21-4"><a aria-hidden="true" href="#cb21-4" tabindex="-1"></a>            value_name<span class="op">=</span><span class="st">'counts'</span>)</span>
<span id="cb21-5"><a aria-hidden="true" href="#cb21-5" tabindex="-1"></a>)</span>
<span id="cb21-6"><a aria-hidden="true" href="#cb21-6" tabindex="-1"></a></span>
<span id="cb21-7"><a aria-hidden="true" href="#cb21-7" tabindex="-1"></a>abundance_long_phyla <span class="op">=</span> (abundance_df_phyla</span>
<span id="cb21-8"><a aria-hidden="true" href="#cb21-8" tabindex="-1"></a>        .melt(id_vars<span class="op">=</span>taxon_names[:<span class="dv">2</span>],</span>
<span id="cb21-9"><a aria-hidden="true" href="#cb21-9" tabindex="-1"></a>                var_name<span class="op">=</span><span class="st">'sample_id'</span>,</span>
<span id="cb21-10"><a aria-hidden="true" href="#cb21-10" tabindex="-1"></a>                value_name<span class="op">=</span><span class="st">'counts'</span>)</span>
<span id="cb21-11"><a aria-hidden="true" href="#cb21-11" tabindex="-1"></a>)</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
</div>
</section>
<section class="level2" id="analysis-figures">
<h2 class="anchored" data-anchor-id="analysis-figures">Analysis &amp; Figures</h2>
<section class="level3" id="sampling-schedule">
<h3 class="anchored" data-anchor-id="sampling-schedule">Sampling Schedule</h3>
<div class="cell" data-execution_count="22" id="cell-30">
<details class="code-fold">
<summary>Show Code</summary>
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb22-1"><a aria-hidden="true" href="#cb22-1" tabindex="-1"></a>f <span class="op">=</span> (samples_info</span>
<span id="cb22-2"><a aria-hidden="true" href="#cb22-2" tabindex="-1"></a>.assign(date<span class="op">=</span><span class="kw">lambda</span> dd: pd.to_datetime(dd.date))</span>
<span id="cb22-3"><a aria-hidden="true" href="#cb22-3" tabindex="-1"></a> .assign(weekday<span class="op">=</span><span class="kw">lambda</span> dd: dd.date.dt.weekday)</span>
<span id="cb22-4"><a aria-hidden="true" href="#cb22-4" tabindex="-1"></a> .assign(week<span class="op">=</span><span class="kw">lambda</span> dd: dd.date.dt.isocalendar().week)</span>
<span id="cb22-5"><a aria-hidden="true" href="#cb22-5" tabindex="-1"></a> .assign(week<span class="op">=</span><span class="kw">lambda</span> dd: np.where(dd.sample_mode<span class="op">==</span><span class="st">'Flight'</span>, dd.week <span class="op">-</span><span class="fl">.25</span>, dd.week <span class="op">+</span> <span class="fl">.25</span>))</span>
<span id="cb22-6"><a aria-hidden="true" href="#cb22-6" tabindex="-1"></a> .assign(week<span class="op">=</span><span class="kw">lambda</span> dd: np.where(dd.sample_id<span class="op">==</span><span class="st">'S11'</span>, dd.week <span class="op">-</span> <span class="fl">.5</span>, dd.week).astype(<span class="bu">float</span>))</span>
<span id="cb22-7"><a aria-hidden="true" href="#cb22-7" tabindex="-1"></a> .pipe(<span class="kw">lambda</span> dd: p9.ggplot(dd)</span>
<span id="cb22-8"><a aria-hidden="true" href="#cb22-8" tabindex="-1"></a>       <span class="op">+</span> p9.aes(<span class="st">'weekday'</span>, <span class="st">'week'</span>)</span>
<span id="cb22-9"><a aria-hidden="true" href="#cb22-9" tabindex="-1"></a>       <span class="op">+</span> p9.geom_tile(p9.aes(fill<span class="op">=</span><span class="st">'sample_mode'</span>), color<span class="op">=</span><span class="st">'black'</span>)</span>
<span id="cb22-10"><a aria-hidden="true" href="#cb22-10" tabindex="-1"></a>       <span class="op">+</span> p9.geom_text(p9.aes(label<span class="op">=</span><span class="st">'sample_id'</span>), size<span class="op">=</span><span class="dv">9</span>, nudge_y<span class="op">=-</span><span class="fl">.05</span>)</span>
<span id="cb22-11"><a aria-hidden="true" href="#cb22-11" tabindex="-1"></a>       <span class="op">+</span> p9.scale_y_reverse(labels<span class="op">=</span>[<span class="st">'02/17-02/23'</span>, <span class="st">'02/24-03/02'</span>,</span>
<span id="cb22-12"><a aria-hidden="true" href="#cb22-12" tabindex="-1"></a>                                    <span class="st">'03/03-03/09'</span>, <span class="st">'03/10-03/16'</span>,</span>
<span id="cb22-13"><a aria-hidden="true" href="#cb22-13" tabindex="-1"></a>                                    <span class="st">'03/17-03/23'</span>, <span class="st">'03/24-03/30'</span>,</span>
<span id="cb22-14"><a aria-hidden="true" href="#cb22-14" tabindex="-1"></a>                                    <span class="st">'03/31-04/06'</span>, <span class="st">'04/07-04/13'</span>], </span>
<span id="cb22-15"><a aria-hidden="true" href="#cb22-15" tabindex="-1"></a>                            breaks<span class="op">=</span><span class="bu">range</span>(<span class="dv">8</span>, <span class="dv">16</span>))</span>
<span id="cb22-16"><a aria-hidden="true" href="#cb22-16" tabindex="-1"></a>       <span class="op">+</span> p9.scale_x_continuous(labels<span class="op">=</span>[<span class="st">'Mon'</span>, <span class="st">'Tue'</span>, <span class="st">'Wed'</span>, <span class="st">'Thu'</span>, <span class="st">'Fri'</span>, <span class="st">'Sat'</span>, <span class="st">'Sun'</span>],</span>
<span id="cb22-17"><a aria-hidden="true" href="#cb22-17" tabindex="-1"></a>                               breaks<span class="op">=</span>[<span class="dv">0</span>, <span class="dv">1</span>, <span class="dv">2</span>, <span class="dv">3</span>, <span class="dv">4</span>, <span class="dv">5</span>, <span class="dv">6</span>])</span>
<span id="cb22-18"><a aria-hidden="true" href="#cb22-18" tabindex="-1"></a>       <span class="op">+</span> p9.labs(x<span class="op">=</span><span class="st">''</span>, y<span class="op">=</span><span class="st">''</span>, title<span class="op">=</span><span class="st">'Sampling Schedule (2014)'</span>, fill<span class="op">=</span><span class="st">''</span>)</span>
<span id="cb22-19"><a aria-hidden="true" href="#cb22-19" tabindex="-1"></a>       <span class="op">+</span> p9.theme(figure_size<span class="op">=</span>(<span class="dv">4</span>, <span class="dv">3</span>),</span>
<span id="cb22-20"><a aria-hidden="true" href="#cb22-20" tabindex="-1"></a>                  plot_title<span class="op">=</span>p9.element_text(ha<span class="op">=</span><span class="st">'center'</span>),</span>
<span id="cb22-21"><a aria-hidden="true" href="#cb22-21" tabindex="-1"></a>                  legend_position<span class="op">=</span>(<span class="fl">.6</span>, <span class="fl">.51</span>),</span>
<span id="cb22-22"><a aria-hidden="true" href="#cb22-22" tabindex="-1"></a>                  )</span>
<span id="cb22-23"><a aria-hidden="true" href="#cb22-23" tabindex="-1"></a>      )</span>
<span id="cb22-24"><a aria-hidden="true" href="#cb22-24" tabindex="-1"></a>)</span>
<span id="cb22-25"><a aria-hidden="true" href="#cb22-25" tabindex="-1"></a></span>
<span id="cb22-26"><a aria-hidden="true" href="#cb22-26" tabindex="-1"></a>f.save(<span class="st">'../output/figures/sampling_schedule.svg'</span>)</span>
<span id="cb22-27"><a aria-hidden="true" href="#cb22-27" tabindex="-1"></a>f</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img class="figure-img" height="900" src="microbial_diversity_analysis_files/figure-html/cell-23-output-1.png" width="1200"/></p>
</figure>
</div>
</div>
</div>
</section>
<section class="level3" id="abundance-comparisons">
<h3 class="anchored" data-anchor-id="abundance-comparisons">Abundance comparisons</h3>
<section class="level4" id="read-count-boxplots-flight-vs-ground">
<h4 class="anchored" data-anchor-id="read-count-boxplots-flight-vs-ground">Read Count boxplots (Flight vs Ground)</h4>
<div class="cell" data-execution_count="23" id="cell-33">
<details class="code-fold">
<summary>Show Code</summary>
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb23-1"><a aria-hidden="true" href="#cb23-1" tabindex="-1"></a>f <span class="op">=</span> (abundance_long_phyla</span>
<span id="cb23-2"><a aria-hidden="true" href="#cb23-2" tabindex="-1"></a> .groupby([<span class="st">'Kingdom'</span>, <span class="st">'sample_id'</span>])</span>
<span id="cb23-3"><a aria-hidden="true" href="#cb23-3" tabindex="-1"></a> [<span class="st">'counts'</span>]</span>
<span id="cb23-4"><a aria-hidden="true" href="#cb23-4" tabindex="-1"></a> .<span class="bu">sum</span>()</span>
<span id="cb23-5"><a aria-hidden="true" href="#cb23-5" tabindex="-1"></a> .reset_index()</span>
<span id="cb23-6"><a aria-hidden="true" href="#cb23-6" tabindex="-1"></a> .merge(samples_info)</span>
<span id="cb23-7"><a aria-hidden="true" href="#cb23-7" tabindex="-1"></a> .query(<span class="st">'counts &gt; 0'</span>)</span>
<span id="cb23-8"><a aria-hidden="true" href="#cb23-8" tabindex="-1"></a> .pipe(<span class="kw">lambda</span> dd: p9.ggplot(dd)</span>
<span id="cb23-9"><a aria-hidden="true" href="#cb23-9" tabindex="-1"></a>    <span class="op">+</span> p9.aes(<span class="st">'sample_mode'</span>, <span class="st">'counts.astype(float)'</span>, fill<span class="op">=</span><span class="st">'sample_mode'</span>)</span>
<span id="cb23-10"><a aria-hidden="true" href="#cb23-10" tabindex="-1"></a>    <span class="op">+</span> p9.facet_wrap(<span class="st">'Kingdom'</span>)</span>
<span id="cb23-11"><a aria-hidden="true" href="#cb23-11" tabindex="-1"></a>    <span class="op">+</span> p9.geom_boxplot(outlier_alpha<span class="op">=</span><span class="dv">0</span>, alpha<span class="op">=</span><span class="fl">.7</span>)</span>
<span id="cb23-12"><a aria-hidden="true" href="#cb23-12" tabindex="-1"></a>    <span class="op">+</span> p9.geom_point(alpha<span class="op">=</span><span class="fl">.5</span>, color<span class="op">=</span><span class="st">'black'</span>, fill<span class="op">=</span><span class="st">'black'</span>)</span>
<span id="cb23-13"><a aria-hidden="true" href="#cb23-13" tabindex="-1"></a>    <span class="op">+</span> p9.scale_fill_manual(values<span class="op">=</span>[<span class="st">'#D3894C'</span>, <span class="st">'#435B97'</span>])</span>
<span id="cb23-14"><a aria-hidden="true" href="#cb23-14" tabindex="-1"></a>    <span class="op">+</span> p9.labs(x<span class="op">=</span><span class="st">''</span>, y<span class="op">=</span><span class="st">'Read Counts'</span>)</span>
<span id="cb23-15"><a aria-hidden="true" href="#cb23-15" tabindex="-1"></a>    <span class="op">+</span> p9.guides(fill<span class="op">=</span><span class="va">False</span>)</span>
<span id="cb23-16"><a aria-hidden="true" href="#cb23-16" tabindex="-1"></a>    <span class="op">+</span> p9.theme(figure_size<span class="op">=</span>(<span class="dv">5</span>, <span class="dv">2</span>))</span>
<span id="cb23-17"><a aria-hidden="true" href="#cb23-17" tabindex="-1"></a> )</span>
<span id="cb23-18"><a aria-hidden="true" href="#cb23-18" tabindex="-1"></a> )</span>
<span id="cb23-19"><a aria-hidden="true" href="#cb23-19" tabindex="-1"></a>f.save(<span class="st">'../output/figures/abundance_by_sample_type.svg'</span>)</span>
<span id="cb23-20"><a aria-hidden="true" href="#cb23-20" tabindex="-1"></a>f.draw()</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-display" data-execution_count="23">
<div>
<figure class="figure">
<p><img class="figure-img" height="614" src="microbial_diversity_analysis_files/figure-html/cell-24-output-1.png" width="1529"/></p>
</figure>
</div>
</div>
</div>
</section>
<section class="level4" id="read-counts-per-sample-type-and-date">
<h4 class="anchored" data-anchor-id="read-counts-per-sample-type-and-date">Read Counts per sample type and date</h4>
<div class="cell" data-execution_count="24" id="cell-35">
<details class="code-fold">
<summary>Show Code</summary>
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb24-1"><a aria-hidden="true" href="#cb24-1" tabindex="-1"></a>f <span class="op">=</span> (abundance_long_phyla</span>
<span id="cb24-2"><a aria-hidden="true" href="#cb24-2" tabindex="-1"></a> .groupby([<span class="st">'Kingdom'</span>, <span class="st">'sample_id'</span>])</span>
<span id="cb24-3"><a aria-hidden="true" href="#cb24-3" tabindex="-1"></a> [<span class="st">'counts'</span>]</span>
<span id="cb24-4"><a aria-hidden="true" href="#cb24-4" tabindex="-1"></a> .<span class="bu">sum</span>()</span>
<span id="cb24-5"><a aria-hidden="true" href="#cb24-5" tabindex="-1"></a> .reset_index()</span>
<span id="cb24-6"><a aria-hidden="true" href="#cb24-6" tabindex="-1"></a> .merge(samples_info)</span>
<span id="cb24-7"><a aria-hidden="true" href="#cb24-7" tabindex="-1"></a> .assign(date_label<span class="op">=</span><span class="kw">lambda</span> dd: pd.to_datetime(dd.date).dt.strftime(<span class="st">'%m/</span><span class="sc">%d</span><span class="st">'</span>))</span>
<span id="cb24-8"><a aria-hidden="true" href="#cb24-8" tabindex="-1"></a>    .pipe(<span class="kw">lambda</span> dd: p9.ggplot(dd)</span>
<span id="cb24-9"><a aria-hidden="true" href="#cb24-9" tabindex="-1"></a>  <span class="op">+</span> p9.aes(<span class="st">'date_label'</span>, <span class="st">'counts.astype(float)'</span>, fill<span class="op">=</span><span class="st">'Kingdom'</span>)</span>
<span id="cb24-10"><a aria-hidden="true" href="#cb24-10" tabindex="-1"></a>  <span class="op">+</span> p9.facet_wrap(<span class="st">'sample_mode'</span>, ncol<span class="op">=</span><span class="dv">1</span>)</span>
<span id="cb24-11"><a aria-hidden="true" href="#cb24-11" tabindex="-1"></a>  <span class="op">+</span> p9.geom_col(position<span class="op">=</span><span class="st">'dodge'</span>)</span>
<span id="cb24-12"><a aria-hidden="true" href="#cb24-12" tabindex="-1"></a>  <span class="op">+</span> p9.scale_fill_manual(values<span class="op">=</span>[<span class="st">'#D3894C'</span>, <span class="st">'#435B97'</span>])</span>
<span id="cb24-13"><a aria-hidden="true" href="#cb24-13" tabindex="-1"></a>  <span class="op">+</span> p9.labs(x<span class="op">=</span><span class="st">''</span>, y<span class="op">=</span><span class="st">'Read Counts'</span>, fill<span class="op">=</span><span class="st">''</span>)</span>
<span id="cb24-14"><a aria-hidden="true" href="#cb24-14" tabindex="-1"></a>  <span class="op">+</span> p9.theme(legend_position<span class="op">=</span>(<span class="fl">.55</span>, <span class="fl">.8</span>),</span>
<span id="cb24-15"><a aria-hidden="true" href="#cb24-15" tabindex="-1"></a>             figure_size<span class="op">=</span>(<span class="dv">7</span>, <span class="dv">3</span>),</span>
<span id="cb24-16"><a aria-hidden="true" href="#cb24-16" tabindex="-1"></a>             legend_key_size<span class="op">=</span><span class="dv">8</span>,</span>
<span id="cb24-17"><a aria-hidden="true" href="#cb24-17" tabindex="-1"></a>             )</span>
<span id="cb24-18"><a aria-hidden="true" href="#cb24-18" tabindex="-1"></a>    )</span>
<span id="cb24-19"><a aria-hidden="true" href="#cb24-19" tabindex="-1"></a>)</span>
<span id="cb24-20"><a aria-hidden="true" href="#cb24-20" tabindex="-1"></a></span>
<span id="cb24-21"><a aria-hidden="true" href="#cb24-21" tabindex="-1"></a>f.save(<span class="st">'../output/figures/abundance_by_sample_type_time.svg'</span>)</span>
<span id="cb24-22"><a aria-hidden="true" href="#cb24-22" tabindex="-1"></a>f.draw()</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-display" data-execution_count="24">
<div>
<figure class="figure">
<p><img class="figure-img" height="902" src="microbial_diversity_analysis_files/figure-html/cell-25-output-1.png" width="2118"/></p>
</figure>
</div>
</div>
</div>
</section>
<section class="level4" id="total-counts-per-phylum">
<h4 class="anchored" data-anchor-id="total-counts-per-phylum">Total counts per Phylum</h4>
<div class="cell" data-execution_count="25" id="cell-37">
<details class="code-fold">
<summary>Show Code</summary>
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb25-1"><a aria-hidden="true" href="#cb25-1" tabindex="-1"></a>(abundance_long_phyla</span>
<span id="cb25-2"><a aria-hidden="true" href="#cb25-2" tabindex="-1"></a> .groupby([<span class="st">'Kingdom'</span>, <span class="st">'Phylum'</span>])</span>
<span id="cb25-3"><a aria-hidden="true" href="#cb25-3" tabindex="-1"></a> .counts</span>
<span id="cb25-4"><a aria-hidden="true" href="#cb25-4" tabindex="-1"></a> .<span class="bu">sum</span>()</span>
<span id="cb25-5"><a aria-hidden="true" href="#cb25-5" tabindex="-1"></a> .sort_values(ascending<span class="op">=</span><span class="va">False</span>)</span>
<span id="cb25-6"><a aria-hidden="true" href="#cb25-6" tabindex="-1"></a> .loc[<span class="kw">lambda</span> x: x <span class="op">&gt;</span> <span class="dv">0</span>]</span>
<span id="cb25-7"><a aria-hidden="true" href="#cb25-7" tabindex="-1"></a> .reset_index()</span>
<span id="cb25-8"><a aria-hidden="true" href="#cb25-8" tabindex="-1"></a> .style</span>
<span id="cb25-9"><a aria-hidden="true" href="#cb25-9" tabindex="-1"></a> .hide(axis<span class="op">=</span><span class="st">'index'</span>)</span>
<span id="cb25-10"><a aria-hidden="true" href="#cb25-10" tabindex="-1"></a> .<span class="bu">apply</span>(<span class="kw">lambda</span> row: [<span class="st">'color: #435B97'</span>] <span class="op">*</span> <span class="bu">len</span>(row)   <span class="cf">if</span> <span class="st">'Fungi'</span> <span class="kw">in</span> row.Kingdom <span class="cf">else</span></span>
<span id="cb25-11"><a aria-hidden="true" href="#cb25-11" tabindex="-1"></a>        [<span class="st">'color: #D3894C'</span>] <span class="op">*</span> <span class="bu">len</span>(row), axis<span class="op">=</span><span class="dv">1</span>)</span>
<span id="cb25-12"><a aria-hidden="true" href="#cb25-12" tabindex="-1"></a> .background_gradient(cmap<span class="op">=</span><span class="st">'viridis'</span>)</span>
<span id="cb25-13"><a aria-hidden="true" href="#cb25-13" tabindex="-1"></a> .applymap(<span class="kw">lambda</span> x: <span class="st">'color: red'</span> <span class="cf">if</span> x<span class="op">==</span><span class="st">'unknown'</span> <span class="cf">else</span> <span class="st">''</span>)</span>
<span id="cb25-14"><a aria-hidden="true" href="#cb25-14" tabindex="-1"></a>)</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-display" data-execution_count="25">
<style type="text/css">
#T_51306_row0_col0, #T_51306_row0_col1, #T_51306_row2_col0, #T_51306_row2_col1, #T_51306_row5_col0 {
  color: #435B97;
}
#T_51306_row0_col2 {
  color: #435B97;
  background-color: #fde725;
  color: #000000;
}
#T_51306_row1_col0, #T_51306_row1_col1, #T_51306_row3_col0, #T_51306_row3_col1, #T_51306_row4_col0, #T_51306_row4_col1, #T_51306_row6_col0, #T_51306_row6_col1, #T_51306_row7_col0, #T_51306_row8_col0, #T_51306_row8_col1, #T_51306_row9_col0, #T_51306_row9_col1, #T_51306_row10_col0, #T_51306_row10_col1, #T_51306_row11_col0, #T_51306_row11_col1, #T_51306_row12_col0, #T_51306_row12_col1, #T_51306_row13_col0, #T_51306_row13_col1, #T_51306_row14_col0, #T_51306_row14_col1, #T_51306_row15_col0, #T_51306_row15_col1, #T_51306_row16_col0, #T_51306_row16_col1, #T_51306_row17_col0, #T_51306_row17_col1, #T_51306_row18_col0, #T_51306_row18_col1, #T_51306_row19_col0, #T_51306_row19_col1, #T_51306_row20_col0, #T_51306_row20_col1, #T_51306_row21_col0, #T_51306_row21_col1 {
  color: #D3894C;
}
#T_51306_row1_col2 {
  color: #D3894C;
  background-color: #77d153;
  color: #000000;
}
#T_51306_row2_col2 {
  color: #435B97;
  background-color: #482878;
  color: #f1f1f1;
}
#T_51306_row3_col2 {
  color: #D3894C;
  background-color: #470d60;
  color: #f1f1f1;
}
#T_51306_row4_col2 {
  color: #D3894C;
  background-color: #46075a;
  color: #f1f1f1;
}
#T_51306_row5_col1 {
  color: #435B97;
  color: red;
}
#T_51306_row5_col2 {
  color: #435B97;
  background-color: #440256;
  color: #f1f1f1;
}
#T_51306_row6_col2, #T_51306_row7_col2, #T_51306_row8_col2, #T_51306_row9_col2, #T_51306_row10_col2, #T_51306_row11_col2, #T_51306_row12_col2, #T_51306_row13_col2, #T_51306_row14_col2, #T_51306_row15_col2, #T_51306_row16_col2, #T_51306_row17_col2, #T_51306_row18_col2, #T_51306_row19_col2, #T_51306_row20_col2, #T_51306_row21_col2 {
  color: #D3894C;
  background-color: #440154;
  color: #f1f1f1;
}
#T_51306_row7_col1 {
  color: #D3894C;
  color: red;
}
</style>
<table class="caption-top table table-sm table-striped small" data-quarto-postprocess="true" id="T_51306">
<thead>
<tr class="header">
<th class="col_heading level0 col0" data-quarto-table-cell-role="th" id="T_51306_level0_col0">Kingdom</th>
<th class="col_heading level0 col1" data-quarto-table-cell-role="th" id="T_51306_level0_col1">Phylum</th>
<th class="col_heading level0 col2" data-quarto-table-cell-role="th" id="T_51306_level0_col2">counts</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td class="data row0 col0" id="T_51306_row0_col0">Fungi</td>
<td class="data row0 col1" id="T_51306_row0_col1">Ascomycota</td>
<td class="data row0 col2" id="T_51306_row0_col2">1964855</td>
</tr>
<tr class="even">
<td class="data row1 col0" id="T_51306_row1_col0">Bacteria</td>
<td class="data row1 col1" id="T_51306_row1_col1">Proteobacteria</td>
<td class="data row1 col2" id="T_51306_row1_col2">1562108</td>
</tr>
<tr class="odd">
<td class="data row2 col0" id="T_51306_row2_col0">Fungi</td>
<td class="data row2 col1" id="T_51306_row2_col1">Basidiomycota</td>
<td class="data row2 col2" id="T_51306_row2_col2">222314</td>
</tr>
<tr class="even">
<td class="data row3 col0" id="T_51306_row3_col0">Bacteria</td>
<td class="data row3 col1" id="T_51306_row3_col1">Firmicutes</td>
<td class="data row3 col2" id="T_51306_row3_col2">66403</td>
</tr>
<tr class="odd">
<td class="data row4 col0" id="T_51306_row4_col0">Bacteria</td>
<td class="data row4 col1" id="T_51306_row4_col1">Bacteroidota</td>
<td class="data row4 col2" id="T_51306_row4_col2">30860</td>
</tr>
<tr class="even">
<td class="data row5 col0" id="T_51306_row5_col0">Fungi</td>
<td class="data row5 col1" id="T_51306_row5_col1">unknown</td>
<td class="data row5 col2" id="T_51306_row5_col2">9247</td>
</tr>
<tr class="odd">
<td class="data row6 col0" id="T_51306_row6_col0">Bacteria</td>
<td class="data row6 col1" id="T_51306_row6_col1">Actinobacteriota</td>
<td class="data row6 col2" id="T_51306_row6_col2">6048</td>
</tr>
<tr class="even">
<td class="data row7 col0" id="T_51306_row7_col0">Bacteria</td>
<td class="data row7 col1" id="T_51306_row7_col1">unknown</td>
<td class="data row7 col2" id="T_51306_row7_col2">2702</td>
</tr>
<tr class="odd">
<td class="data row8 col0" id="T_51306_row8_col0">Bacteria</td>
<td class="data row8 col1" id="T_51306_row8_col1">Verrucomicrobiota</td>
<td class="data row8 col2" id="T_51306_row8_col2">835</td>
</tr>
<tr class="even">
<td class="data row9 col0" id="T_51306_row9_col0">Bacteria</td>
<td class="data row9 col1" id="T_51306_row9_col1">Acidobacteriota</td>
<td class="data row9 col2" id="T_51306_row9_col2">581</td>
</tr>
<tr class="odd">
<td class="data row10 col0" id="T_51306_row10_col0">Bacteria</td>
<td class="data row10 col1" id="T_51306_row10_col1">Fusobacteriota</td>
<td class="data row10 col2" id="T_51306_row10_col2">565</td>
</tr>
<tr class="even">
<td class="data row11 col0" id="T_51306_row11_col0">Bacteria</td>
<td class="data row11 col1" id="T_51306_row11_col1">Desulfobacterota</td>
<td class="data row11 col2" id="T_51306_row11_col2">476</td>
</tr>
<tr class="odd">
<td class="data row12 col0" id="T_51306_row12_col0">Bacteria</td>
<td class="data row12 col1" id="T_51306_row12_col1">Cyanobacteria</td>
<td class="data row12 col2" id="T_51306_row12_col2">273</td>
</tr>
<tr class="even">
<td class="data row13 col0" id="T_51306_row13_col0">Bacteria</td>
<td class="data row13 col1" id="T_51306_row13_col1">Campylobacterota</td>
<td class="data row13 col2" id="T_51306_row13_col2">245</td>
</tr>
<tr class="odd">
<td class="data row14 col0" id="T_51306_row14_col0">Bacteria</td>
<td class="data row14 col1" id="T_51306_row14_col1">Chloroflexi</td>
<td class="data row14 col2" id="T_51306_row14_col2">123</td>
</tr>
<tr class="even">
<td class="data row15 col0" id="T_51306_row15_col0">Bacteria</td>
<td class="data row15 col1" id="T_51306_row15_col1">Planctomycetota</td>
<td class="data row15 col2" id="T_51306_row15_col2">119</td>
</tr>
<tr class="odd">
<td class="data row16 col0" id="T_51306_row16_col0">Bacteria</td>
<td class="data row16 col1" id="T_51306_row16_col1">Patescibacteria</td>
<td class="data row16 col2" id="T_51306_row16_col2">110</td>
</tr>
<tr class="even">
<td class="data row17 col0" id="T_51306_row17_col0">Bacteria</td>
<td class="data row17 col1" id="T_51306_row17_col1">Armatimonadota</td>
<td class="data row17 col2" id="T_51306_row17_col2">98</td>
</tr>
<tr class="odd">
<td class="data row18 col0" id="T_51306_row18_col0">Bacteria</td>
<td class="data row18 col1" id="T_51306_row18_col1">Deinococcota</td>
<td class="data row18 col2" id="T_51306_row18_col2">79</td>
</tr>
<tr class="even">
<td class="data row19 col0" id="T_51306_row19_col0">Bacteria</td>
<td class="data row19 col1" id="T_51306_row19_col1">Myxococcota</td>
<td class="data row19 col2" id="T_51306_row19_col2">28</td>
</tr>
<tr class="odd">
<td class="data row20 col0" id="T_51306_row20_col0">Bacteria</td>
<td class="data row20 col1" id="T_51306_row20_col1">Gemmatimonadota</td>
<td class="data row20 col2" id="T_51306_row20_col2">27</td>
</tr>
<tr class="even">
<td class="data row21 col0" id="T_51306_row21_col0">Bacteria</td>
<td class="data row21 col1" id="T_51306_row21_col1">Dependentiae</td>
<td class="data row21 col2" id="T_51306_row21_col2">10</td>
</tr>
</tbody>
</table>
</div>
</div>
</section>
</section>
<section class="level3" id="metagenomics-identification-rate">
<h3 class="anchored" data-anchor-id="metagenomics-identification-rate">Metagenomics identification rate</h3>
<p>Here we are going to take a look at the rate at which the different samples have been resolved up to the Genus level.</p>
<div class="cell" data-execution_count="26" id="cell-40">
<details class="code-fold">
<summary>Show Code</summary>
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb26-1"><a aria-hidden="true" href="#cb26-1" tabindex="-1"></a>reads_per_source <span class="op">=</span> (abundance_long_genera</span>
<span id="cb26-2"><a aria-hidden="true" href="#cb26-2" tabindex="-1"></a> .merge(samples_info)</span>
<span id="cb26-3"><a aria-hidden="true" href="#cb26-3" tabindex="-1"></a> .query(<span class="st">'counts &gt; 0'</span>)</span>
<span id="cb26-4"><a aria-hidden="true" href="#cb26-4" tabindex="-1"></a> .groupby([<span class="st">'Kingdom'</span> , <span class="st">'sample_mode'</span>], as_index<span class="op">=</span><span class="va">False</span>)</span>
<span id="cb26-5"><a aria-hidden="true" href="#cb26-5" tabindex="-1"></a> .counts.<span class="bu">sum</span>()</span>
<span id="cb26-6"><a aria-hidden="true" href="#cb26-6" tabindex="-1"></a>)</span>
<span id="cb26-7"><a aria-hidden="true" href="#cb26-7" tabindex="-1"></a></span>
<span id="cb26-8"><a aria-hidden="true" href="#cb26-8" tabindex="-1"></a>identification_df <span class="op">=</span> <span class="op">\</span></span>
<span id="cb26-9"><a aria-hidden="true" href="#cb26-9" tabindex="-1"></a>(abundance_long_genera</span>
<span id="cb26-10"><a aria-hidden="true" href="#cb26-10" tabindex="-1"></a> .merge(samples_info)</span>
<span id="cb26-11"><a aria-hidden="true" href="#cb26-11" tabindex="-1"></a> .query(<span class="st">'counts &gt; 0'</span>)</span>
<span id="cb26-12"><a aria-hidden="true" href="#cb26-12" tabindex="-1"></a> .groupby(taxon_names <span class="op">+</span> [<span class="st">'sample_mode'</span>], as_index<span class="op">=</span><span class="va">False</span>)</span>
<span id="cb26-13"><a aria-hidden="true" href="#cb26-13" tabindex="-1"></a> .counts.<span class="bu">sum</span>()</span>
<span id="cb26-14"><a aria-hidden="true" href="#cb26-14" tabindex="-1"></a> .groupby([<span class="st">'Kingdom'</span>, <span class="st">'sample_mode'</span>])</span>
<span id="cb26-15"><a aria-hidden="true" href="#cb26-15" tabindex="-1"></a> .<span class="bu">apply</span>(<span class="kw">lambda</span> dd: dd.assign(relative_abundance<span class="op">=</span> dd.counts <span class="op">/</span> dd.counts.<span class="bu">sum</span>()))</span>
<span id="cb26-16"><a aria-hidden="true" href="#cb26-16" tabindex="-1"></a> .melt([<span class="st">'Kingdom'</span>, <span class="st">'sample_mode'</span>, <span class="st">'counts'</span>, <span class="st">'relative_abundance'</span>])</span>
<span id="cb26-17"><a aria-hidden="true" href="#cb26-17" tabindex="-1"></a> .query(<span class="st">'value == "unknown"'</span>)</span>
<span id="cb26-18"><a aria-hidden="true" href="#cb26-18" tabindex="-1"></a> .assign(variable<span class="op">=</span><span class="kw">lambda</span> dd: pd.Categorical(dd.variable, categories<span class="op">=</span>taxon_names, ordered<span class="op">=</span><span class="va">True</span>))</span>
<span id="cb26-19"><a aria-hidden="true" href="#cb26-19" tabindex="-1"></a> .groupby([<span class="st">'Kingdom'</span>, <span class="st">'sample_mode'</span>, <span class="st">'variable'</span>])</span>
<span id="cb26-20"><a aria-hidden="true" href="#cb26-20" tabindex="-1"></a> .relative_abundance.<span class="bu">sum</span>()</span>
<span id="cb26-21"><a aria-hidden="true" href="#cb26-21" tabindex="-1"></a> .rename(<span class="st">'freq'</span>)</span>
<span id="cb26-22"><a aria-hidden="true" href="#cb26-22" tabindex="-1"></a> .astype(<span class="bu">float</span>)</span>
<span id="cb26-23"><a aria-hidden="true" href="#cb26-23" tabindex="-1"></a> .reset_index()</span>
<span id="cb26-24"><a aria-hidden="true" href="#cb26-24" tabindex="-1"></a> .assign(rate_discovered<span class="op">=</span><span class="kw">lambda</span> dd: <span class="dv">1</span> <span class="op">-</span> dd.freq)</span>
<span id="cb26-25"><a aria-hidden="true" href="#cb26-25" tabindex="-1"></a> .groupby([<span class="st">'Kingdom'</span>, <span class="st">'sample_mode'</span>])</span>
<span id="cb26-26"><a aria-hidden="true" href="#cb26-26" tabindex="-1"></a> .<span class="bu">apply</span>(<span class="kw">lambda</span> dd: dd</span>
<span id="cb26-27"><a aria-hidden="true" href="#cb26-27" tabindex="-1"></a> .assign(rate_diff<span class="op">=</span>dd.rate_discovered.diff().shift(<span class="op">-</span><span class="dv">1</span>) <span class="op">*</span> <span class="op">-</span><span class="dv">1</span>)</span>
<span id="cb26-28"><a aria-hidden="true" href="#cb26-28" tabindex="-1"></a> .assign(rate_diff<span class="op">=</span><span class="kw">lambda</span> ddd: </span>
<span id="cb26-29"><a aria-hidden="true" href="#cb26-29" tabindex="-1"></a>         np.where(ddd.rate_diff.isna(), ddd.rate_discovered, ddd.rate_diff)))</span>
<span id="cb26-30"><a aria-hidden="true" href="#cb26-30" tabindex="-1"></a> )</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>If we go by maximum depth of identification, ordered by Genus &gt; Family &gt; Order &gt; Class &gt; Phylum &gt; Kingdom:</p>
<div class="cell" data-execution_count="27" id="cell-43">
<details class="code-fold">
<summary>Show Code</summary>
<div class="sourceCode cell-code" id="cb27"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb27-1"><a aria-hidden="true" href="#cb27-1" tabindex="-1"></a>f <span class="op">=</span> (identification_df</span>
<span id="cb27-2"><a aria-hidden="true" href="#cb27-2" tabindex="-1"></a> .assign(source_label<span class="op">=</span><span class="kw">lambda</span> dd: dd.sample_mode <span class="op">+</span> <span class="st">'-'</span> <span class="op">+</span> dd.Kingdom)</span>
<span id="cb27-3"><a aria-hidden="true" href="#cb27-3" tabindex="-1"></a> .assign(percent<span class="op">=</span><span class="kw">lambda</span> dd: (dd.rate_diff <span class="op">*</span> <span class="dv">100</span>).<span class="bu">round</span>(<span class="dv">2</span>).astype(<span class="bu">str</span>) <span class="op">+</span> <span class="st">'%'</span>)</span>
<span id="cb27-4"><a aria-hidden="true" href="#cb27-4" tabindex="-1"></a> .pipe(<span class="kw">lambda</span> dd: p9.ggplot(dd) </span>
<span id="cb27-5"><a aria-hidden="true" href="#cb27-5" tabindex="-1"></a>       <span class="op">+</span> p9.aes(<span class="st">'source_label'</span>, <span class="st">'rate_diff'</span>, fill<span class="op">=</span><span class="st">'variable'</span>)</span>
<span id="cb27-6"><a aria-hidden="true" href="#cb27-6" tabindex="-1"></a>       <span class="op">+</span> p9.geom_col()</span>
<span id="cb27-7"><a aria-hidden="true" href="#cb27-7" tabindex="-1"></a>       <span class="op">+</span> p9.scale_y_continuous(labels<span class="op">=</span>percent_format())</span>
<span id="cb27-8"><a aria-hidden="true" href="#cb27-8" tabindex="-1"></a>       <span class="op">+</span> p9.labs(x<span class="op">=</span><span class="st">''</span>, y<span class="op">=</span><span class="st">'Percentage of all reads'</span>, fill<span class="op">=</span><span class="st">'Taxonomic level'</span>,</span>
<span id="cb27-9"><a aria-hidden="true" href="#cb27-9" tabindex="-1"></a>                 title<span class="op">=</span><span class="st">'Maximum depth of identification for all reads'</span>)</span>
<span id="cb27-10"><a aria-hidden="true" href="#cb27-10" tabindex="-1"></a>       <span class="op">+</span> p9.coord_flip()</span>
<span id="cb27-11"><a aria-hidden="true" href="#cb27-11" tabindex="-1"></a>       <span class="op">+</span> p9.geom_text(p9.aes(label<span class="op">=</span><span class="st">'percent'</span>, y<span class="op">=</span><span class="fl">.025</span>), ha<span class="op">=</span><span class="st">'left'</span>, size<span class="op">=</span><span class="dv">8</span>,</span>
<span id="cb27-12"><a aria-hidden="true" href="#cb27-12" tabindex="-1"></a>                      data<span class="op">=</span>dd.loc[dd.variable<span class="op">==</span><span class="st">'Genus'</span>])</span>
<span id="cb27-13"><a aria-hidden="true" href="#cb27-13" tabindex="-1"></a>       <span class="op">+</span> p9.theme(figure_size<span class="op">=</span>(<span class="dv">6</span>, <span class="dv">3</span>), title<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">10</span>))</span>
<span id="cb27-14"><a aria-hidden="true" href="#cb27-14" tabindex="-1"></a>)</span>
<span id="cb27-15"><a aria-hidden="true" href="#cb27-15" tabindex="-1"></a>)</span>
<span id="cb27-16"><a aria-hidden="true" href="#cb27-16" tabindex="-1"></a>f.save(<span class="st">'../output/figures/identification_depth.svg'</span>)</span>
<span id="cb27-17"><a aria-hidden="true" href="#cb27-17" tabindex="-1"></a>f.draw()</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-display" data-execution_count="27">
<div>
<figure class="figure">
<p><img class="figure-img" height="924" src="microbial_diversity_analysis_files/figure-html/cell-29-output-1.png" width="1824"/></p>
</figure>
</div>
</div>
</div>
<p>We see how the bacterial reads have really good resolution even at the Genus level. Fungal reads, from ITS, hover around the 94% rate at the Genus level.</p>
<p>Let’s look at this the other way around, with the % of unidentified reads at each level per sample type and kingdom:</p>
<div class="cell" data-execution_count="28" id="cell-45">
<details class="code-fold">
<summary>Show Code</summary>
<div class="sourceCode cell-code" id="cb28"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb28-1"><a aria-hidden="true" href="#cb28-1" tabindex="-1"></a>reversed_taxon_names <span class="op">=</span> taxon_names.copy()</span>
<span id="cb28-2"><a aria-hidden="true" href="#cb28-2" tabindex="-1"></a>reversed_taxon_names.reverse()</span>
<span id="cb28-3"><a aria-hidden="true" href="#cb28-3" tabindex="-1"></a>f <span class="op">=</span> (identification_df</span>
<span id="cb28-4"><a aria-hidden="true" href="#cb28-4" tabindex="-1"></a> .reset_index(drop<span class="op">=</span><span class="va">True</span>)</span>
<span id="cb28-5"><a aria-hidden="true" href="#cb28-5" tabindex="-1"></a> .assign(variable<span class="op">=</span><span class="kw">lambda</span> dd: </span>
<span id="cb28-6"><a aria-hidden="true" href="#cb28-6" tabindex="-1"></a>         pd.Categorical(dd.variable,</span>
<span id="cb28-7"><a aria-hidden="true" href="#cb28-7" tabindex="-1"></a>                        categories<span class="op">=</span>reversed_taxon_names,</span>
<span id="cb28-8"><a aria-hidden="true" href="#cb28-8" tabindex="-1"></a>                        ordered<span class="op">=</span><span class="va">True</span>))</span>
<span id="cb28-9"><a aria-hidden="true" href="#cb28-9" tabindex="-1"></a>  .pipe(<span class="kw">lambda</span> dd: p9.ggplot(dd)</span>
<span id="cb28-10"><a aria-hidden="true" href="#cb28-10" tabindex="-1"></a>        <span class="op">+</span> p9.aes(<span class="st">'variable'</span>, <span class="st">'freq'</span>)</span>
<span id="cb28-11"><a aria-hidden="true" href="#cb28-11" tabindex="-1"></a>        <span class="op">+</span> p9.geom_col()</span>
<span id="cb28-12"><a aria-hidden="true" href="#cb28-12" tabindex="-1"></a>        <span class="op">+</span> p9.coord_flip()</span>
<span id="cb28-13"><a aria-hidden="true" href="#cb28-13" tabindex="-1"></a>        <span class="op">+</span> p9.facet_grid(<span class="st">'sample_mode ~ Kingdom'</span>)</span>
<span id="cb28-14"><a aria-hidden="true" href="#cb28-14" tabindex="-1"></a>        <span class="op">+</span> p9.scale_y_continuous(labels<span class="op">=</span>percent_format())</span>
<span id="cb28-15"><a aria-hidden="true" href="#cb28-15" tabindex="-1"></a>        <span class="op">+</span> p9.labs(x<span class="op">=</span><span class="st">'Taxonomic level'</span>, y<span class="op">=</span><span class="st">'Unidentified reads (%)'</span>)</span>
<span id="cb28-16"><a aria-hidden="true" href="#cb28-16" tabindex="-1"></a>        <span class="op">+</span> p9.theme(figure_size<span class="op">=</span>(<span class="dv">5</span>, <span class="fl">3.5</span>))</span>
<span id="cb28-17"><a aria-hidden="true" href="#cb28-17" tabindex="-1"></a>        )</span>
<span id="cb28-18"><a aria-hidden="true" href="#cb28-18" tabindex="-1"></a></span>
<span id="cb28-19"><a aria-hidden="true" href="#cb28-19" tabindex="-1"></a>)</span>
<span id="cb28-20"><a aria-hidden="true" href="#cb28-20" tabindex="-1"></a>f.save(<span class="st">'../output/figures/unidentified_reads_bars.svg'</span>)</span>
<span id="cb28-21"><a aria-hidden="true" href="#cb28-21" tabindex="-1"></a>f.draw()</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-display" data-execution_count="28">
<div>
<figure class="figure">
<p><img class="figure-img" height="1064" src="microbial_diversity_analysis_files/figure-html/cell-30-output-1.png" width="1514"/></p>
</figure>
</div>
</div>
</div>
<p>And if we look at the numbers on a sample per sample basis, we can observe how there is quite a lot of variability, specially on the fungal side. This is probably a product of the higher diversity found in February samples compared to the April ones in the ITS database:</p>
<div class="cell" data-execution_count="29" id="cell-47">
<div class="sourceCode cell-code" id="cb29"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb29-1"><a aria-hidden="true" href="#cb29-1" tabindex="-1"></a>sample_map <span class="op">=</span> (samples_info</span>
<span id="cb29-2"><a aria-hidden="true" href="#cb29-2" tabindex="-1"></a>              [[<span class="st">'sample_id'</span>, <span class="st">'sample_mode'</span>]]</span>
<span id="cb29-3"><a aria-hidden="true" href="#cb29-3" tabindex="-1"></a>              .set_index(<span class="st">'sample_id'</span>)</span>
<span id="cb29-4"><a aria-hidden="true" href="#cb29-4" tabindex="-1"></a>              .to_dict()</span>
<span id="cb29-5"><a aria-hidden="true" href="#cb29-5" tabindex="-1"></a>              [<span class="st">'sample_mode'</span>]</span>
<span id="cb29-6"><a aria-hidden="true" href="#cb29-6" tabindex="-1"></a>             )</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="30" id="cell-48">
<details class="code-fold">
<summary>Show Code</summary>
<div class="sourceCode cell-code" id="cb30"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb30-1"><a aria-hidden="true" href="#cb30-1" tabindex="-1"></a>sample_map <span class="op">=</span> (samples_info</span>
<span id="cb30-2"><a aria-hidden="true" href="#cb30-2" tabindex="-1"></a>              [[<span class="st">'sample_id'</span>, <span class="st">'sample_mode'</span>]]</span>
<span id="cb30-3"><a aria-hidden="true" href="#cb30-3" tabindex="-1"></a>              .set_index(<span class="st">'sample_id'</span>)</span>
<span id="cb30-4"><a aria-hidden="true" href="#cb30-4" tabindex="-1"></a>              .to_dict()</span>
<span id="cb30-5"><a aria-hidden="true" href="#cb30-5" tabindex="-1"></a>              [<span class="st">'sample_mode'</span>]</span>
<span id="cb30-6"><a aria-hidden="true" href="#cb30-6" tabindex="-1"></a>             )</span>
<span id="cb30-7"><a aria-hidden="true" href="#cb30-7" tabindex="-1"></a></span>
<span id="cb30-8"><a aria-hidden="true" href="#cb30-8" tabindex="-1"></a>unknown_per_sample <span class="op">=</span> (abundance_long_genera</span>
<span id="cb30-9"><a aria-hidden="true" href="#cb30-9" tabindex="-1"></a> .merge(samples_info)</span>
<span id="cb30-10"><a aria-hidden="true" href="#cb30-10" tabindex="-1"></a> .query(<span class="st">'counts &gt; 0'</span>)</span>
<span id="cb30-11"><a aria-hidden="true" href="#cb30-11" tabindex="-1"></a> .groupby(taxon_names <span class="op">+</span> [<span class="st">'sample_mode'</span>, <span class="st">'sample_id'</span>], as_index<span class="op">=</span><span class="va">False</span>)</span>
<span id="cb30-12"><a aria-hidden="true" href="#cb30-12" tabindex="-1"></a> .counts.<span class="bu">sum</span>()</span>
<span id="cb30-13"><a aria-hidden="true" href="#cb30-13" tabindex="-1"></a> .groupby([<span class="st">'Kingdom'</span>, <span class="st">'sample_mode'</span>, <span class="st">'sample_id'</span>])</span>
<span id="cb30-14"><a aria-hidden="true" href="#cb30-14" tabindex="-1"></a> .<span class="bu">apply</span>(<span class="kw">lambda</span> df: df.assign(relative_abundance<span class="op">=</span>df.counts <span class="op">/</span> df.counts.<span class="bu">sum</span>()))</span>
<span id="cb30-15"><a aria-hidden="true" href="#cb30-15" tabindex="-1"></a> .melt([<span class="st">'Kingdom'</span>, <span class="st">'sample_mode'</span>, <span class="st">'counts'</span>, <span class="st">'relative_abundance'</span>, <span class="st">'sample_id'</span>])</span>
<span id="cb30-16"><a aria-hidden="true" href="#cb30-16" tabindex="-1"></a> .query(<span class="st">'value == "unknown"'</span>)</span>
<span id="cb30-17"><a aria-hidden="true" href="#cb30-17" tabindex="-1"></a> .assign(variable<span class="op">=</span><span class="kw">lambda</span> dd: pd.Categorical(dd.variable, categories<span class="op">=</span>taxon_names, ordered<span class="op">=</span><span class="va">True</span>))</span>
<span id="cb30-18"><a aria-hidden="true" href="#cb30-18" tabindex="-1"></a> .groupby([<span class="st">'Kingdom'</span>, <span class="st">'sample_mode'</span>, <span class="st">'variable'</span>, <span class="st">'sample_id'</span>])</span>
<span id="cb30-19"><a aria-hidden="true" href="#cb30-19" tabindex="-1"></a> .relative_abundance.<span class="bu">sum</span>()</span>
<span id="cb30-20"><a aria-hidden="true" href="#cb30-20" tabindex="-1"></a> .rename(<span class="st">'freq'</span>)</span>
<span id="cb30-21"><a aria-hidden="true" href="#cb30-21" tabindex="-1"></a> .reset_index()</span>
<span id="cb30-22"><a aria-hidden="true" href="#cb30-22" tabindex="-1"></a> .loc[<span class="kw">lambda</span> dd: dd.variable<span class="op">!=</span><span class="st">'Kingdom'</span>]</span>
<span id="cb30-23"><a aria-hidden="true" href="#cb30-23" tabindex="-1"></a>  .replace(<span class="dv">0</span>, np.NaN)</span>
<span id="cb30-24"><a aria-hidden="true" href="#cb30-24" tabindex="-1"></a>  .dropna()</span>
<span id="cb30-25"><a aria-hidden="true" href="#cb30-25" tabindex="-1"></a> .drop(columns<span class="op">=</span><span class="st">'sample_mode'</span>)</span>
<span id="cb30-26"><a aria-hidden="true" href="#cb30-26" tabindex="-1"></a> .pivot(index<span class="op">=</span>[<span class="st">'variable'</span>, <span class="st">'Kingdom'</span>], columns<span class="op">=</span><span class="st">'sample_id'</span>, values<span class="op">=</span><span class="st">'freq'</span>)</span>
<span id="cb30-27"><a aria-hidden="true" href="#cb30-27" tabindex="-1"></a> .fillna(<span class="dv">0</span>)</span>
<span id="cb30-28"><a aria-hidden="true" href="#cb30-28" tabindex="-1"></a> .T</span>
<span id="cb30-29"><a aria-hidden="true" href="#cb30-29" tabindex="-1"></a> .reset_index()</span>
<span id="cb30-30"><a aria-hidden="true" href="#cb30-30" tabindex="-1"></a> .assign(sample_mode<span class="op">=</span><span class="kw">lambda</span> dd: dd.sample_id.<span class="bu">map</span>(sample_map))</span>
<span id="cb30-31"><a aria-hidden="true" href="#cb30-31" tabindex="-1"></a> .set_index([<span class="st">'sample_id'</span>, <span class="st">'sample_mode'</span>])</span>
<span id="cb30-32"><a aria-hidden="true" href="#cb30-32" tabindex="-1"></a>)</span>
<span id="cb30-33"><a aria-hidden="true" href="#cb30-33" tabindex="-1"></a>unknown_per_sample.to_csv(<span class="st">'../output/tables/unknown_per_sample.csv'</span>)</span>
<span id="cb30-34"><a aria-hidden="true" href="#cb30-34" tabindex="-1"></a></span>
<span id="cb30-35"><a aria-hidden="true" href="#cb30-35" tabindex="-1"></a>(unknown_per_sample.pipe(<span class="kw">lambda</span> dd: dd</span>
<span id="cb30-36"><a aria-hidden="true" href="#cb30-36" tabindex="-1"></a>     .style</span>
<span id="cb30-37"><a aria-hidden="true" href="#cb30-37" tabindex="-1"></a>     .background_gradient(cmap<span class="op">=</span><span class="st">'inferno'</span>, axis<span class="op">=</span><span class="va">None</span>)</span>
<span id="cb30-38"><a aria-hidden="true" href="#cb30-38" tabindex="-1"></a>     .<span class="bu">format</span>(<span class="st">"</span><span class="sc">{:.2%}</span><span class="st">"</span>)</span>
<span id="cb30-39"><a aria-hidden="true" href="#cb30-39" tabindex="-1"></a>     .set_caption(<span class="st">'Percentage of reads that could not be identified'</span>)</span>
<span id="cb30-40"><a aria-hidden="true" href="#cb30-40" tabindex="-1"></a>))</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-display" data-execution_count="30">
<style type="text/css">
#T_e904a_row0_col0, #T_e904a_row0_col2, #T_e904a_row0_col4, #T_e904a_row1_col0, #T_e904a_row1_col2, #T_e904a_row1_col4, #T_e904a_row1_col6, #T_e904a_row1_col8, #T_e904a_row2_col0, #T_e904a_row2_col2, #T_e904a_row2_col4, #T_e904a_row3_col0, #T_e904a_row3_col1, #T_e904a_row3_col2, #T_e904a_row3_col3, #T_e904a_row3_col4, #T_e904a_row3_col5, #T_e904a_row3_col6, #T_e904a_row3_col8, #T_e904a_row4_col0, #T_e904a_row4_col1, #T_e904a_row4_col2, #T_e904a_row4_col4, #T_e904a_row4_col6, #T_e904a_row4_col8, #T_e904a_row5_col0, #T_e904a_row5_col1, #T_e904a_row5_col2, #T_e904a_row5_col3, #T_e904a_row5_col4, #T_e904a_row5_col5, #T_e904a_row5_col6, #T_e904a_row5_col7, #T_e904a_row5_col9, #T_e904a_row6_col0, #T_e904a_row6_col1, #T_e904a_row6_col2, #T_e904a_row6_col3, #T_e904a_row6_col4, #T_e904a_row6_col6, #T_e904a_row7_col0, #T_e904a_row7_col1, #T_e904a_row7_col2, #T_e904a_row7_col4, #T_e904a_row7_col6, #T_e904a_row7_col8, #T_e904a_row8_col0, #T_e904a_row8_col1, #T_e904a_row8_col2, #T_e904a_row8_col3, #T_e904a_row8_col4, #T_e904a_row8_col5, #T_e904a_row8_col6, #T_e904a_row8_col7, #T_e904a_row9_col0, #T_e904a_row9_col2, #T_e904a_row9_col4, #T_e904a_row9_col6, #T_e904a_row10_col0, #T_e904a_row10_col2, #T_e904a_row10_col4, #T_e904a_row11_col0, #T_e904a_row11_col2, #T_e904a_row11_col4, #T_e904a_row11_col6, #T_e904a_row14_col1, #T_e904a_row14_col3, #T_e904a_row14_col5, #T_e904a_row14_col7, #T_e904a_row15_col1, #T_e904a_row16_col1, #T_e904a_row16_col3, #T_e904a_row16_col5, #T_e904a_row19_col1, #T_e904a_row20_col1, #T_e904a_row20_col3, #T_e904a_row20_col5, #T_e904a_row20_col7, #T_e904a_row20_col9 {
  background-color: #000004;
  color: #f1f1f1;
}
#T_e904a_row0_col1, #T_e904a_row6_col5 {
  background-color: #010108;
  color: #f1f1f1;
}
#T_e904a_row0_col3, #T_e904a_row2_col1, #T_e904a_row12_col2, #T_e904a_row13_col4, #T_e904a_row17_col2 {
  background-color: #040314;
  color: #f1f1f1;
}
#T_e904a_row0_col5, #T_e904a_row15_col9, #T_e904a_row16_col4, #T_e904a_row18_col2 {
  background-color: #060419;
  color: #f1f1f1;
}
#T_e904a_row0_col6, #T_e904a_row2_col6, #T_e904a_row4_col3, #T_e904a_row4_col5, #T_e904a_row5_col8, #T_e904a_row6_col8, #T_e904a_row8_col8, #T_e904a_row10_col6, #T_e904a_row11_col8 {
  background-color: #010005;
  color: #f1f1f1;
}
#T_e904a_row0_col7 {
  background-color: #0c0826;
  color: #f1f1f1;
}
#T_e904a_row0_col8, #T_e904a_row9_col8, #T_e904a_row10_col8, #T_e904a_row15_col0 {
  background-color: #010106;
  color: #f1f1f1;
}
#T_e904a_row0_col9 {
  background-color: #fcffa4;
  color: #000000;
}
#T_e904a_row1_col1, #T_e904a_row4_col9, #T_e904a_row13_col2, #T_e904a_row16_col2, #T_e904a_row18_col1, #T_e904a_row18_col3, #T_e904a_row18_col5, #T_e904a_row18_col7 {
  background-color: #040312;
  color: #f1f1f1;
}
#T_e904a_row1_col3, #T_e904a_row1_col5, #T_e904a_row17_col4, #T_e904a_row18_col9 {
  background-color: #09061f;
  color: #f1f1f1;
}
#T_e904a_row1_col7, #T_e904a_row9_col3, #T_e904a_row11_col5, #T_e904a_row17_col6 {
  background-color: #150b37;
  color: #f1f1f1;
}
#T_e904a_row1_col9 {
  background-color: #1c0c43;
  color: #f1f1f1;
}
#T_e904a_row2_col3 {
  background-color: #3b0964;
  color: #f1f1f1;
}
#T_e904a_row2_col5 {
  background-color: #550f6d;
  color: #f1f1f1;
}
#T_e904a_row2_col7, #T_e904a_row18_col8 {
  background-color: #88226a;
  color: #f1f1f1;
}
#T_e904a_row2_col8, #T_e904a_row13_col0, #T_e904a_row20_col0 {
  background-color: #02020c;
  color: #f1f1f1;
}
#T_e904a_row2_col9 {
  background-color: #ef6c23;
  color: #f1f1f1;
}
#T_e904a_row3_col7, #T_e904a_row12_col4 {
  background-color: #08051d;
  color: #f1f1f1;
}
#T_e904a_row3_col9 {
  background-color: #ed6925;
  color: #f1f1f1;
}
#T_e904a_row4_col7, #T_e904a_row9_col1, #T_e904a_row12_col0, #T_e904a_row16_col0, #T_e904a_row17_col0, #T_e904a_row20_col2 {
  background-color: #02020e;
  color: #f1f1f1;
}
#T_e904a_row6_col7, #T_e904a_row6_col9, #T_e904a_row11_col1, #T_e904a_row20_col4 {
  background-color: #07051b;
  color: #f1f1f1;
}
#T_e904a_row7_col3, #T_e904a_row7_col5 {
  background-color: #3e0966;
  color: #f1f1f1;
}
#T_e904a_row7_col7, #T_e904a_row9_col7, #T_e904a_row17_col8 {
  background-color: #490b6a;
  color: #f1f1f1;
}
#T_e904a_row7_col9 {
  background-color: #e8602d;
  color: #f1f1f1;
}
#T_e904a_row8_col9 {
  background-color: #ee6a24;
  color: #f1f1f1;
}
#T_e904a_row9_col5 {
  background-color: #340a5f;
  color: #f1f1f1;
}
#T_e904a_row9_col9 {
  background-color: #a82e5f;
  color: #f1f1f1;
}
#T_e904a_row10_col1, #T_e904a_row12_col1, #T_e904a_row14_col2, #T_e904a_row15_col6, #T_e904a_row19_col7 {
  background-color: #050417;
  color: #f1f1f1;
}
#T_e904a_row10_col3, #T_e904a_row17_col3 {
  background-color: #120a32;
  color: #f1f1f1;
}
#T_e904a_row10_col5 {
  background-color: #1b0c41;
  color: #f1f1f1;
}
#T_e904a_row10_col7, #T_e904a_row12_col8 {
  background-color: #420a68;
  color: #f1f1f1;
}
#T_e904a_row10_col9 {
  background-color: #bc3754;
  color: #f1f1f1;
}
#T_e904a_row11_col3 {
  background-color: #110a30;
  color: #f1f1f1;
}
#T_e904a_row11_col7 {
  background-color: #260c51;
  color: #f1f1f1;
}
#T_e904a_row11_col9 {
  background-color: #932667;
  color: #f1f1f1;
}
#T_e904a_row12_col3, #T_e904a_row12_col5 {
  background-color: #0e092b;
  color: #f1f1f1;
}
#T_e904a_row12_col6, #T_e904a_row15_col8, #T_e904a_row17_col5 {
  background-color: #190c3e;
  color: #f1f1f1;
}
#T_e904a_row12_col7 {
  background-color: #180c3c;
  color: #f1f1f1;
}
#T_e904a_row12_col9, #T_e904a_row19_col6 {
  background-color: #5a116e;
  color: #f1f1f1;
}
#T_e904a_row13_col1, #T_e904a_row13_col3, #T_e904a_row13_col5, #T_e904a_row13_col7, #T_e904a_row13_col9 {
  background-color: #2d0b59;
  color: #f1f1f1;
}
#T_e904a_row13_col6, #T_e904a_row14_col4, #T_e904a_row18_col4 {
  background-color: #0b0724;
  color: #f1f1f1;
}
#T_e904a_row13_col8, #T_e904a_row18_col6 {
  background-color: #280b53;
  color: #f1f1f1;
}
#T_e904a_row14_col0, #T_e904a_row15_col4, #T_e904a_row17_col1, #T_e904a_row18_col0 {
  background-color: #030210;
  color: #f1f1f1;
}
#T_e904a_row14_col6, #T_e904a_row19_col2 {
  background-color: #160b39;
  color: #f1f1f1;
}
#T_e904a_row14_col8 {
  background-color: #470b6a;
  color: #f1f1f1;
}
#T_e904a_row14_col9, #T_e904a_row15_col2, #T_e904a_row15_col3, #T_e904a_row15_col5, #T_e904a_row15_col7, #T_e904a_row19_col3, #T_e904a_row19_col5 {
  background-color: #02010a;
  color: #f1f1f1;
}
#T_e904a_row16_col6 {
  background-color: #0a0722;
  color: #f1f1f1;
}
#T_e904a_row16_col7 {
  background-color: #380962;
  color: #f1f1f1;
}
#T_e904a_row16_col8 {
  background-color: #2b0b57;
  color: #f1f1f1;
}
#T_e904a_row16_col9 {
  background-color: #972766;
  color: #f1f1f1;
}
#T_e904a_row17_col7 {
  background-color: #440a68;
  color: #f1f1f1;
}
#T_e904a_row17_col9 {
  background-color: #771c6d;
  color: #f1f1f1;
}
#T_e904a_row19_col0, #T_e904a_row20_col6 {
  background-color: #0d0829;
  color: #f1f1f1;
}
#T_e904a_row19_col4 {
  background-color: #230c4c;
  color: #f1f1f1;
}
#T_e904a_row19_col8 {
  background-color: #e05536;
  color: #f1f1f1;
}
#T_e904a_row19_col9 {
  background-color: #7d1e6d;
  color: #f1f1f1;
}
#T_e904a_row20_col8 {
  background-color: #320a5e;
  color: #f1f1f1;
}
</style>
<div class="quarto-float quarto-figure quarto-figure-center anchored" data-quarto-postprocess="true" id="T_e904a">
<figure class="quarto-float quarto-float-tbl figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-tbl" id="T_e904a-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Table 1: Percentage of reads that could not be identified
</figcaption>
<div aria-describedby="T_e904a-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<table class="caption-top table table-sm table-striped small" data-quarto-postprocess="true" id="T_e904a">
<thead>
<tr class="header">
<th class="blank" data-quarto-table-cell-role="th"> </th>
<th class="index_name level0" data-quarto-table-cell-role="th">variable</th>
<th class="col_heading level0 col0" colspan="2" data-quarto-table-cell-role="th" id="T_e904a_level0_col0">Phylum</th>
<th class="col_heading level0 col2" colspan="2" data-quarto-table-cell-role="th" id="T_e904a_level0_col2">Class</th>
<th class="col_heading level0 col4" colspan="2" data-quarto-table-cell-role="th" id="T_e904a_level0_col4">Order</th>
<th class="col_heading level0 col6" colspan="2" data-quarto-table-cell-role="th" id="T_e904a_level0_col6">Family</th>
<th class="col_heading level0 col8" colspan="2" data-quarto-table-cell-role="th" id="T_e904a_level0_col8">Genus</th>
</tr>
<tr class="odd">
<th class="blank" data-quarto-table-cell-role="th"> </th>
<th class="index_name level1" data-quarto-table-cell-role="th">Kingdom</th>
<th class="col_heading level1 col0" data-quarto-table-cell-role="th" id="T_e904a_level1_col0">Bacteria</th>
<th class="col_heading level1 col1" data-quarto-table-cell-role="th" id="T_e904a_level1_col1">Fungi</th>
<th class="col_heading level1 col2" data-quarto-table-cell-role="th" id="T_e904a_level1_col2">Bacteria</th>
<th class="col_heading level1 col3" data-quarto-table-cell-role="th" id="T_e904a_level1_col3">Fungi</th>
<th class="col_heading level1 col4" data-quarto-table-cell-role="th" id="T_e904a_level1_col4">Bacteria</th>
<th class="col_heading level1 col5" data-quarto-table-cell-role="th" id="T_e904a_level1_col5">Fungi</th>
<th class="col_heading level1 col6" data-quarto-table-cell-role="th" id="T_e904a_level1_col6">Bacteria</th>
<th class="col_heading level1 col7" data-quarto-table-cell-role="th" id="T_e904a_level1_col7">Fungi</th>
<th class="col_heading level1 col8" data-quarto-table-cell-role="th" id="T_e904a_level1_col8">Bacteria</th>
<th class="col_heading level1 col9" data-quarto-table-cell-role="th" id="T_e904a_level1_col9">Fungi</th>
</tr>
<tr class="header">
<th class="index_name level0" data-quarto-table-cell-role="th">sample_id</th>
<th class="index_name level1" data-quarto-table-cell-role="th">sample_mode</th>
<th class="blank col0" data-quarto-table-cell-role="th"> </th>
<th class="blank col1" data-quarto-table-cell-role="th"> </th>
<th class="blank col2" data-quarto-table-cell-role="th"> </th>
<th class="blank col3" data-quarto-table-cell-role="th"> </th>
<th class="blank col4" data-quarto-table-cell-role="th"> </th>
<th class="blank col5" data-quarto-table-cell-role="th"> </th>
<th class="blank col6" data-quarto-table-cell-role="th"> </th>
<th class="blank col7" data-quarto-table-cell-role="th"> </th>
<th class="blank col8" data-quarto-table-cell-role="th"> </th>
<th class="blank col9" data-quarto-table-cell-role="th"> </th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td class="row_heading level0 row0" data-quarto-table-cell-role="th" id="T_e904a_level0_row0">S01</td>
<td class="row_heading level1 row0" data-quarto-table-cell-role="th" id="T_e904a_level1_row0">Flight</td>
<td class="data row0 col0" id="T_e904a_row0_col0">0.00%</td>
<td class="data row0 col1" id="T_e904a_row0_col1">0.23%</td>
<td class="data row0 col2" id="T_e904a_row0_col2">0.00%</td>
<td class="data row0 col3" id="T_e904a_row0_col3">0.56%</td>
<td class="data row0 col4" id="T_e904a_row0_col4">0.01%</td>
<td class="data row0 col5" id="T_e904a_row0_col5">0.70%</td>
<td class="data row0 col6" id="T_e904a_row0_col6">0.07%</td>
<td class="data row0 col7" id="T_e904a_row0_col7">1.09%</td>
<td class="data row0 col8" id="T_e904a_row0_col8">0.13%</td>
<td class="data row0 col9" id="T_e904a_row0_col9">15.56%</td>
</tr>
<tr class="even">
<td class="row_heading level0 row1" data-quarto-table-cell-role="th" id="T_e904a_level0_row1">S02</td>
<td class="row_heading level1 row1" data-quarto-table-cell-role="th" id="T_e904a_level1_row1">Ground</td>
<td class="data row1 col0" id="T_e904a_row1_col0">0.00%</td>
<td class="data row1 col1" id="T_e904a_row1_col1">0.51%</td>
<td class="data row1 col2" id="T_e904a_row1_col2">0.00%</td>
<td class="data row1 col3" id="T_e904a_row1_col3">0.86%</td>
<td class="data row1 col4" id="T_e904a_row1_col4">0.00%</td>
<td class="data row1 col5" id="T_e904a_row1_col5">0.86%</td>
<td class="data row1 col6" id="T_e904a_row1_col6">0.01%</td>
<td class="data row1 col7" id="T_e904a_row1_col7">1.47%</td>
<td class="data row1 col8" id="T_e904a_row1_col8">0.04%</td>
<td class="data row1 col9" id="T_e904a_row1_col9">1.80%</td>
</tr>
<tr class="odd">
<td class="row_heading level0 row2" data-quarto-table-cell-role="th" id="T_e904a_level0_row2">S03</td>
<td class="row_heading level1 row2" data-quarto-table-cell-role="th" id="T_e904a_level1_row2">Flight</td>
<td class="data row2 col0" id="T_e904a_row2_col0">0.00%</td>
<td class="data row2 col1" id="T_e904a_row2_col1">0.55%</td>
<td class="data row2 col2" id="T_e904a_row2_col2">0.00%</td>
<td class="data row2 col3" id="T_e904a_row2_col3">2.88%</td>
<td class="data row2 col4" id="T_e904a_row2_col4">0.02%</td>
<td class="data row2 col5" id="T_e904a_row2_col5">3.85%</td>
<td class="data row2 col6" id="T_e904a_row2_col6">0.10%</td>
<td class="data row2 col7" id="T_e904a_row2_col7">5.79%</td>
<td class="data row2 col8" id="T_e904a_row2_col8">0.33%</td>
<td class="data row2 col9" id="T_e904a_row2_col9">10.46%</td>
</tr>
<tr class="even">
<td class="row_heading level0 row3" data-quarto-table-cell-role="th" id="T_e904a_level0_row3">S04</td>
<td class="row_heading level1 row3" data-quarto-table-cell-role="th" id="T_e904a_level1_row3">Ground</td>
<td class="data row3 col0" id="T_e904a_row3_col0">0.00%</td>
<td class="data row3 col1" id="T_e904a_row3_col1">0.00%</td>
<td class="data row3 col2" id="T_e904a_row3_col2">0.00%</td>
<td class="data row3 col3" id="T_e904a_row3_col3">0.01%</td>
<td class="data row3 col4" id="T_e904a_row3_col4">0.00%</td>
<td class="data row3 col5" id="T_e904a_row3_col5">0.01%</td>
<td class="data row3 col6" id="T_e904a_row3_col6">0.00%</td>
<td class="data row3 col7" id="T_e904a_row3_col7">0.82%</td>
<td class="data row3 col8" id="T_e904a_row3_col8">0.01%</td>
<td class="data row3 col9" id="T_e904a_row3_col9">10.36%</td>
</tr>
<tr class="odd">
<td class="row_heading level0 row4" data-quarto-table-cell-role="th" id="T_e904a_level0_row4">S05</td>
<td class="row_heading level1 row4" data-quarto-table-cell-role="th" id="T_e904a_level1_row4">Flight</td>
<td class="data row4 col0" id="T_e904a_row4_col0">0.01%</td>
<td class="data row4 col1" id="T_e904a_row4_col1">0.05%</td>
<td class="data row4 col2" id="T_e904a_row4_col2">0.01%</td>
<td class="data row4 col3" id="T_e904a_row4_col3">0.08%</td>
<td class="data row4 col4" id="T_e904a_row4_col4">0.01%</td>
<td class="data row4 col5" id="T_e904a_row4_col5">0.08%</td>
<td class="data row4 col6" id="T_e904a_row4_col6">0.02%</td>
<td class="data row4 col7" id="T_e904a_row4_col7">0.37%</td>
<td class="data row4 col8" id="T_e904a_row4_col8">0.05%</td>
<td class="data row4 col9" id="T_e904a_row4_col9">0.50%</td>
</tr>
<tr class="even">
<td class="row_heading level0 row5" data-quarto-table-cell-role="th" id="T_e904a_level0_row5">S06</td>
<td class="row_heading level1 row5" data-quarto-table-cell-role="th" id="T_e904a_level1_row5">Ground</td>
<td class="data row5 col0" id="T_e904a_row5_col0">0.00%</td>
<td class="data row5 col1" id="T_e904a_row5_col1">0.00%</td>
<td class="data row5 col2" id="T_e904a_row5_col2">0.00%</td>
<td class="data row5 col3" id="T_e904a_row5_col3">0.00%</td>
<td class="data row5 col4" id="T_e904a_row5_col4">0.00%</td>
<td class="data row5 col5" id="T_e904a_row5_col5">0.00%</td>
<td class="data row5 col6" id="T_e904a_row5_col6">0.04%</td>
<td class="data row5 col7" id="T_e904a_row5_col7">0.00%</td>
<td class="data row5 col8" id="T_e904a_row5_col8">0.11%</td>
<td class="data row5 col9" id="T_e904a_row5_col9">0.00%</td>
</tr>
<tr class="odd">
<td class="row_heading level0 row6" data-quarto-table-cell-role="th" id="T_e904a_level0_row6">S07</td>
<td class="row_heading level1 row6" data-quarto-table-cell-role="th" id="T_e904a_level1_row6">Flight</td>
<td class="data row6 col0" id="T_e904a_row6_col0">0.00%</td>
<td class="data row6 col1" id="T_e904a_row6_col1">0.00%</td>
<td class="data row6 col2" id="T_e904a_row6_col2">0.00%</td>
<td class="data row6 col3" id="T_e904a_row6_col3">0.00%</td>
<td class="data row6 col4" id="T_e904a_row6_col4">0.00%</td>
<td class="data row6 col5" id="T_e904a_row6_col5">0.18%</td>
<td class="data row6 col6" id="T_e904a_row6_col6">0.00%</td>
<td class="data row6 col7" id="T_e904a_row6_col7">0.74%</td>
<td class="data row6 col8" id="T_e904a_row6_col8">0.06%</td>
<td class="data row6 col9" id="T_e904a_row6_col9">0.74%</td>
</tr>
<tr class="even">
<td class="row_heading level0 row7" data-quarto-table-cell-role="th" id="T_e904a_level0_row7">S08</td>
<td class="row_heading level1 row7" data-quarto-table-cell-role="th" id="T_e904a_level1_row7">Ground</td>
<td class="data row7 col0" id="T_e904a_row7_col0">0.00%</td>
<td class="data row7 col1" id="T_e904a_row7_col1">0.00%</td>
<td class="data row7 col2" id="T_e904a_row7_col2">0.00%</td>
<td class="data row7 col3" id="T_e904a_row7_col3">2.99%</td>
<td class="data row7 col4" id="T_e904a_row7_col4">0.00%</td>
<td class="data row7 col5" id="T_e904a_row7_col5">2.99%</td>
<td class="data row7 col6" id="T_e904a_row7_col6">0.00%</td>
<td class="data row7 col7" id="T_e904a_row7_col7">3.37%</td>
<td class="data row7 col8" id="T_e904a_row7_col8">0.05%</td>
<td class="data row7 col9" id="T_e904a_row7_col9">10.02%</td>
</tr>
<tr class="odd">
<td class="row_heading level0 row8" data-quarto-table-cell-role="th" id="T_e904a_level0_row8">S09</td>
<td class="row_heading level1 row8" data-quarto-table-cell-role="th" id="T_e904a_level1_row8">Flight</td>
<td class="data row8 col0" id="T_e904a_row8_col0">0.00%</td>
<td class="data row8 col1" id="T_e904a_row8_col1">0.00%</td>
<td class="data row8 col2" id="T_e904a_row8_col2">0.00%</td>
<td class="data row8 col3" id="T_e904a_row8_col3">0.00%</td>
<td class="data row8 col4" id="T_e904a_row8_col4">0.00%</td>
<td class="data row8 col5" id="T_e904a_row8_col5">0.00%</td>
<td class="data row8 col6" id="T_e904a_row8_col6">0.02%</td>
<td class="data row8 col7" id="T_e904a_row8_col7">0.00%</td>
<td class="data row8 col8" id="T_e904a_row8_col8">0.08%</td>
<td class="data row8 col9" id="T_e904a_row8_col9">10.43%</td>
</tr>
<tr class="even">
<td class="row_heading level0 row9" data-quarto-table-cell-role="th" id="T_e904a_level0_row9">S10</td>
<td class="row_heading level1 row9" data-quarto-table-cell-role="th" id="T_e904a_level1_row9">Ground</td>
<td class="data row9 col0" id="T_e904a_row9_col0">0.00%</td>
<td class="data row9 col1" id="T_e904a_row9_col1">0.39%</td>
<td class="data row9 col2" id="T_e904a_row9_col2">0.00%</td>
<td class="data row9 col3" id="T_e904a_row9_col3">1.50%</td>
<td class="data row9 col4" id="T_e904a_row9_col4">0.03%</td>
<td class="data row9 col5" id="T_e904a_row9_col5">2.64%</td>
<td class="data row9 col6" id="T_e904a_row9_col6">0.04%</td>
<td class="data row9 col7" id="T_e904a_row9_col7">3.35%</td>
<td class="data row9 col8" id="T_e904a_row9_col8">0.17%</td>
<td class="data row9 col9" id="T_e904a_row9_col9">7.03%</td>
</tr>
<tr class="odd">
<td class="row_heading level0 row10" data-quarto-table-cell-role="th" id="T_e904a_level0_row10">S11</td>
<td class="row_heading level1 row10" data-quarto-table-cell-role="th" id="T_e904a_level1_row10">Ground</td>
<td class="data row10 col0" id="T_e904a_row10_col0">0.00%</td>
<td class="data row10 col1" id="T_e904a_row10_col1">0.64%</td>
<td class="data row10 col2" id="T_e904a_row10_col2">0.00%</td>
<td class="data row10 col3" id="T_e904a_row10_col3">1.36%</td>
<td class="data row10 col4" id="T_e904a_row10_col4">0.02%</td>
<td class="data row10 col5" id="T_e904a_row10_col5">1.76%</td>
<td class="data row10 col6" id="T_e904a_row10_col6">0.08%</td>
<td class="data row10 col7" id="T_e904a_row10_col7">3.15%</td>
<td class="data row10 col8" id="T_e904a_row10_col8">0.18%</td>
<td class="data row10 col9" id="T_e904a_row10_col9">7.80%</td>
</tr>
<tr class="even">
<td class="row_heading level0 row11" data-quarto-table-cell-role="th" id="T_e904a_level0_row11">S12</td>
<td class="row_heading level1 row11" data-quarto-table-cell-role="th" id="T_e904a_level1_row11">Ground</td>
<td class="data row11 col0" id="T_e904a_row11_col0">0.00%</td>
<td class="data row11 col1" id="T_e904a_row11_col1">0.77%</td>
<td class="data row11 col2" id="T_e904a_row11_col2">0.01%</td>
<td class="data row11 col3" id="T_e904a_row11_col3">1.33%</td>
<td class="data row11 col4" id="T_e904a_row11_col4">0.01%</td>
<td class="data row11 col5" id="T_e904a_row11_col5">1.50%</td>
<td class="data row11 col6" id="T_e904a_row11_col6">0.03%</td>
<td class="data row11 col7" id="T_e904a_row11_col7">2.18%</td>
<td class="data row11 col8" id="T_e904a_row11_col8">0.09%</td>
<td class="data row11 col9" id="T_e904a_row11_col9">6.23%</td>
</tr>
<tr class="odd">
<td class="row_heading level0 row12" data-quarto-table-cell-role="th" id="T_e904a_level0_row12">S13</td>
<td class="row_heading level1 row12" data-quarto-table-cell-role="th" id="T_e904a_level1_row12">Flight</td>
<td class="data row12 col0" id="T_e904a_row12_col0">0.42%</td>
<td class="data row12 col1" id="T_e904a_row12_col1">0.67%</td>
<td class="data row12 col2" id="T_e904a_row12_col2">0.56%</td>
<td class="data row12 col3" id="T_e904a_row12_col3">1.18%</td>
<td class="data row12 col4" id="T_e904a_row12_col4">0.80%</td>
<td class="data row12 col5" id="T_e904a_row12_col5">1.18%</td>
<td class="data row12 col6" id="T_e904a_row12_col6">1.65%</td>
<td class="data row12 col7" id="T_e904a_row12_col7">1.60%</td>
<td class="data row12 col8" id="T_e904a_row12_col8">3.12%</td>
<td class="data row12 col9" id="T_e904a_row12_col9">4.02%</td>
</tr>
<tr class="even">
<td class="row_heading level0 row13" data-quarto-table-cell-role="th" id="T_e904a_level0_row13">S15</td>
<td class="row_heading level1 row13" data-quarto-table-cell-role="th" id="T_e904a_level1_row13">Flight</td>
<td class="data row13 col0" id="T_e904a_row13_col0">0.35%</td>
<td class="data row13 col1" id="T_e904a_row13_col1">2.42%</td>
<td class="data row13 col2" id="T_e904a_row13_col2">0.51%</td>
<td class="data row13 col3" id="T_e904a_row13_col3">2.42%</td>
<td class="data row13 col4" id="T_e904a_row13_col4">0.58%</td>
<td class="data row13 col5" id="T_e904a_row13_col5">2.42%</td>
<td class="data row13 col6" id="T_e904a_row13_col6">0.98%</td>
<td class="data row13 col7" id="T_e904a_row13_col7">2.42%</td>
<td class="data row13 col8" id="T_e904a_row13_col8">2.19%</td>
<td class="data row13 col9" id="T_e904a_row13_col9">2.42%</td>
</tr>
<tr class="odd">
<td class="row_heading level0 row14" data-quarto-table-cell-role="th" id="T_e904a_level0_row14">S16</td>
<td class="row_heading level1 row14" data-quarto-table-cell-role="th" id="T_e904a_level1_row14">Ground</td>
<td class="data row14 col0" id="T_e904a_row14_col0">0.48%</td>
<td class="data row14 col1" id="T_e904a_row14_col1">0.00%</td>
<td class="data row14 col2" id="T_e904a_row14_col2">0.62%</td>
<td class="data row14 col3" id="T_e904a_row14_col3">0.00%</td>
<td class="data row14 col4" id="T_e904a_row14_col4">1.01%</td>
<td class="data row14 col5" id="T_e904a_row14_col5">0.00%</td>
<td class="data row14 col6" id="T_e904a_row14_col6">1.58%</td>
<td class="data row14 col7" id="T_e904a_row14_col7">0.00%</td>
<td class="data row14 col8" id="T_e904a_row14_col8">3.29%</td>
<td class="data row14 col9" id="T_e904a_row14_col9">0.26%</td>
</tr>
<tr class="even">
<td class="row_heading level0 row15" data-quarto-table-cell-role="th" id="T_e904a_level0_row15">S17</td>
<td class="row_heading level1 row15" data-quarto-table-cell-role="th" id="T_e904a_level1_row15">Flight</td>
<td class="data row15 col0" id="T_e904a_row15_col0">0.13%</td>
<td class="data row15 col1" id="T_e904a_row15_col1">0.00%</td>
<td class="data row15 col2" id="T_e904a_row15_col2">0.28%</td>
<td class="data row15 col3" id="T_e904a_row15_col3">0.26%</td>
<td class="data row15 col4" id="T_e904a_row15_col4">0.43%</td>
<td class="data row15 col5" id="T_e904a_row15_col5">0.26%</td>
<td class="data row15 col6" id="T_e904a_row15_col6">0.63%</td>
<td class="data row15 col7" id="T_e904a_row15_col7">0.26%</td>
<td class="data row15 col8" id="T_e904a_row15_col8">1.67%</td>
<td class="data row15 col9" id="T_e904a_row15_col9">0.67%</td>
</tr>
<tr class="odd">
<td class="row_heading level0 row16" data-quarto-table-cell-role="th" id="T_e904a_level0_row16">S18</td>
<td class="row_heading level1 row16" data-quarto-table-cell-role="th" id="T_e904a_level1_row16">Ground</td>
<td class="data row16 col0" id="T_e904a_row16_col0">0.37%</td>
<td class="data row16 col1" id="T_e904a_row16_col1">0.00%</td>
<td class="data row16 col2" id="T_e904a_row16_col2">0.49%</td>
<td class="data row16 col3" id="T_e904a_row16_col3">0.00%</td>
<td class="data row16 col4" id="T_e904a_row16_col4">0.68%</td>
<td class="data row16 col5" id="T_e904a_row16_col5">0.00%</td>
<td class="data row16 col6" id="T_e904a_row16_col6">0.91%</td>
<td class="data row16 col7" id="T_e904a_row16_col7">2.76%</td>
<td class="data row16 col8" id="T_e904a_row16_col8">2.33%</td>
<td class="data row16 col9" id="T_e904a_row16_col9">6.34%</td>
</tr>
<tr class="even">
<td class="row_heading level0 row17" data-quarto-table-cell-role="th" id="T_e904a_level0_row17">S19</td>
<td class="row_heading level1 row17" data-quarto-table-cell-role="th" id="T_e904a_level1_row17">Flight</td>
<td class="data row17 col0" id="T_e904a_row17_col0">0.43%</td>
<td class="data row17 col1" id="T_e904a_row17_col1">0.44%</td>
<td class="data row17 col2" id="T_e904a_row17_col2">0.57%</td>
<td class="data row17 col3" id="T_e904a_row17_col3">1.39%</td>
<td class="data row17 col4" id="T_e904a_row17_col4">0.86%</td>
<td class="data row17 col5" id="T_e904a_row17_col5">1.66%</td>
<td class="data row17 col6" id="T_e904a_row17_col6">1.47%</td>
<td class="data row17 col7" id="T_e904a_row17_col7">3.19%</td>
<td class="data row17 col8" id="T_e904a_row17_col8">3.35%</td>
<td class="data row17 col9" id="T_e904a_row17_col9">5.14%</td>
</tr>
<tr class="odd">
<td class="row_heading level0 row18" data-quarto-table-cell-role="th" id="T_e904a_level0_row18">S20</td>
<td class="row_heading level1 row18" data-quarto-table-cell-role="th" id="T_e904a_level1_row18">Ground</td>
<td class="data row18 col0" id="T_e904a_row18_col0">0.48%</td>
<td class="data row18 col1" id="T_e904a_row18_col1">0.51%</td>
<td class="data row18 col2" id="T_e904a_row18_col2">0.69%</td>
<td class="data row18 col3" id="T_e904a_row18_col3">0.51%</td>
<td class="data row18 col4" id="T_e904a_row18_col4">1.01%</td>
<td class="data row18 col5" id="T_e904a_row18_col5">0.51%</td>
<td class="data row18 col6" id="T_e904a_row18_col6">2.19%</td>
<td class="data row18 col7" id="T_e904a_row18_col7">0.51%</td>
<td class="data row18 col8" id="T_e904a_row18_col8">5.80%</td>
<td class="data row18 col9" id="T_e904a_row18_col9">0.86%</td>
</tr>
<tr class="even">
<td class="row_heading level0 row19" data-quarto-table-cell-role="th" id="T_e904a_level0_row19">S21</td>
<td class="row_heading level1 row19" data-quarto-table-cell-role="th" id="T_e904a_level1_row19">Flight</td>
<td class="data row19 col0" id="T_e904a_row19_col0">1.10%</td>
<td class="data row19 col1" id="T_e904a_row19_col1">0.00%</td>
<td class="data row19 col2" id="T_e904a_row19_col2">1.53%</td>
<td class="data row19 col3" id="T_e904a_row19_col3">0.25%</td>
<td class="data row19 col4" id="T_e904a_row19_col4">2.04%</td>
<td class="data row19 col5" id="T_e904a_row19_col5">0.25%</td>
<td class="data row19 col6" id="T_e904a_row19_col6">4.06%</td>
<td class="data row19 col7" id="T_e904a_row19_col7">0.66%</td>
<td class="data row19 col8" id="T_e904a_row19_col8">9.49%</td>
<td class="data row19 col9" id="T_e904a_row19_col9">5.37%</td>
</tr>
<tr class="odd">
<td class="row_heading level0 row20" data-quarto-table-cell-role="th" id="T_e904a_level0_row20">S22</td>
<td class="row_heading level1 row20" data-quarto-table-cell-role="th" id="T_e904a_level1_row20">Ground</td>
<td class="data row20 col0" id="T_e904a_row20_col0">0.32%</td>
<td class="data row20 col1" id="T_e904a_row20_col1">0.00%</td>
<td class="data row20 col2" id="T_e904a_row20_col2">0.42%</td>
<td class="data row20 col3" id="T_e904a_row20_col3">0.00%</td>
<td class="data row20 col4" id="T_e904a_row20_col4">0.75%</td>
<td class="data row20 col5" id="T_e904a_row20_col5">0.00%</td>
<td class="data row20 col6" id="T_e904a_row20_col6">1.15%</td>
<td class="data row20 col7" id="T_e904a_row20_col7">0.00%</td>
<td class="data row20 col8" id="T_e904a_row20_col8">2.58%</td>
<td class="data row20 col9" id="T_e904a_row20_col9">0.00%</td>
</tr>
</tbody>
</table>
</div>
</figure>
</div>
</div>
</div>
</section>
<section class="level3" id="exclusive-otus-per-sample-type">
<h3 class="anchored" data-anchor-id="exclusive-otus-per-sample-type">Exclusive OTUs per sample type</h3>
<p>In this section we will compute the number of OTUs, resolved to at least the <strong>Genus</strong> level, that belong to each group of sample types.</p>
<p>We will be looking at mainly two variables:</p>
<ul>
<li>Sample source: Ground or Flight level</li>
<li>Month of the sample: February or April</li>
</ul>
<div class="cell" data-execution_count="31" id="cell-51">
<div class="sourceCode cell-code" id="cb31"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb31-1"><a aria-hidden="true" href="#cb31-1" tabindex="-1"></a>sample_types <span class="op">=</span> [<span class="st">'February-flight'</span>,</span>
<span id="cb31-2"><a aria-hidden="true" href="#cb31-2" tabindex="-1"></a>                <span class="st">'February-ground'</span>,</span>
<span id="cb31-3"><a aria-hidden="true" href="#cb31-3" tabindex="-1"></a>                <span class="st">'April-flight'</span>,</span>
<span id="cb31-4"><a aria-hidden="true" href="#cb31-4" tabindex="-1"></a>                <span class="st">'April-ground'</span>]</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
</div>
<p>Now that we are looking into resolved OTUs, we will be removing the unknown Genera. On top of that, there are several resolved taxons that either refer to plants or to unresolved Genus-level values, so we’ll make a list and remove all of those from the posterior analysis:</p>
<p>If we only consider an OTU present in a sample if <em>AT LEAST 2 READS</em> match with it, the distribution of number of samples in which each Genera-level OTU is found looks as the following:</p>
<div class="cell" data-execution_count="33" id="cell-54">
<details class="code-fold">
<summary>Show Code</summary>
<div class="sourceCode cell-code" id="cb32"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb32-1"><a aria-hidden="true" href="#cb32-1" tabindex="-1"></a>samples_per_otu <span class="op">=</span> (abundance_long_genera</span>
<span id="cb32-2"><a aria-hidden="true" href="#cb32-2" tabindex="-1"></a> .query(<span class="st">'Genus not in @inconclusive_list'</span>)</span>
<span id="cb32-3"><a aria-hidden="true" href="#cb32-3" tabindex="-1"></a> .query(<span class="st">'counts &gt;= 2'</span>)</span>
<span id="cb32-4"><a aria-hidden="true" href="#cb32-4" tabindex="-1"></a> .groupby(taxon_names)</span>
<span id="cb32-5"><a aria-hidden="true" href="#cb32-5" tabindex="-1"></a> .size()</span>
<span id="cb32-6"><a aria-hidden="true" href="#cb32-6" tabindex="-1"></a> .rename(<span class="st">'n_samples'</span>)</span>
<span id="cb32-7"><a aria-hidden="true" href="#cb32-7" tabindex="-1"></a> .reset_index()</span>
<span id="cb32-8"><a aria-hidden="true" href="#cb32-8" tabindex="-1"></a>)</span>
<span id="cb32-9"><a aria-hidden="true" href="#cb32-9" tabindex="-1"></a></span>
<span id="cb32-10"><a aria-hidden="true" href="#cb32-10" tabindex="-1"></a>(samples_per_otu</span>
<span id="cb32-11"><a aria-hidden="true" href="#cb32-11" tabindex="-1"></a> .pipe(<span class="kw">lambda</span> dd: </span>
<span id="cb32-12"><a aria-hidden="true" href="#cb32-12" tabindex="-1"></a>       p9.ggplot(dd)</span>
<span id="cb32-13"><a aria-hidden="true" href="#cb32-13" tabindex="-1"></a>       <span class="op">+</span> p9.aes(<span class="st">'n_samples'</span>, fill<span class="op">=</span><span class="st">'Kingdom'</span>)</span>
<span id="cb32-14"><a aria-hidden="true" href="#cb32-14" tabindex="-1"></a>       <span class="op">+</span> p9.geom_histogram(binwidth<span class="op">=</span><span class="dv">1</span>)</span>
<span id="cb32-15"><a aria-hidden="true" href="#cb32-15" tabindex="-1"></a>       <span class="op">+</span> p9.scale_fill_manual(values<span class="op">=</span>[<span class="st">'#D3894C'</span>, <span class="st">'#435B97'</span>])</span>
<span id="cb32-16"><a aria-hidden="true" href="#cb32-16" tabindex="-1"></a>       <span class="op">+</span> p9.facet_wrap(<span class="st">'Kingdom'</span>, ncol<span class="op">=</span><span class="dv">1</span>)</span>
<span id="cb32-17"><a aria-hidden="true" href="#cb32-17" tabindex="-1"></a>       <span class="op">+</span> p9.geom_text(p9.aes(label<span class="op">=</span><span class="st">'"n = " + size'</span>), x<span class="op">=</span><span class="dv">18</span>, y<span class="op">=</span><span class="dv">130</span>, size<span class="op">=</span><span class="dv">8</span>,</span>
<span id="cb32-18"><a aria-hidden="true" href="#cb32-18" tabindex="-1"></a>                                 data<span class="op">=</span>dd.groupby(<span class="st">'Kingdom'</span>, as_index<span class="op">=</span><span class="va">False</span>).size().astype(<span class="bu">str</span>))</span>
<span id="cb32-19"><a aria-hidden="true" href="#cb32-19" tabindex="-1"></a>       <span class="op">+</span> p9.theme(figure_size<span class="op">=</span>(<span class="dv">4</span>, <span class="dv">3</span>))</span>
<span id="cb32-20"><a aria-hidden="true" href="#cb32-20" tabindex="-1"></a>       <span class="op">+</span> p9.guides(fill<span class="op">=</span><span class="va">False</span>)</span>
<span id="cb32-21"><a aria-hidden="true" href="#cb32-21" tabindex="-1"></a>       <span class="op">+</span> p9.labs(x<span class="op">=</span><span class="st">'Present in $n$ samples'</span>, y<span class="op">=</span><span class="st">'Number of unique Genera'</span>)</span>
<span id="cb32-22"><a aria-hidden="true" href="#cb32-22" tabindex="-1"></a>       )</span>
<span id="cb32-23"><a aria-hidden="true" href="#cb32-23" tabindex="-1"></a>)</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img class="figure-img" height="900" src="microbial_diversity_analysis_files/figure-html/cell-34-output-1.png" width="1200"/></p>
</figure>
</div>
</div>
</div>
<p>If we look at the number of unique Genera which are present in just a single sample, they represent:</p>
<ul>
<li>29.4% (94) of all bacterial genera.</li>
<li>58.6% (143) of all fungal genera.</li>
</ul>
<p>Counting all the resolved OTUs at the Genus level, there are a total of <strong>564</strong> distinct genera present at least in one sample with at least two reads (320 bacterial, 244 fungal).</p>
<p>If we are a bit more strict and focus only on those genera that appear in at least 2 distinct samples, the distribution looks like the following:</p>
<div class="cell" data-execution_count="30" id="cell-56">
<details class="code-fold">
<summary>Show Code</summary>
<div class="sourceCode cell-code" id="cb33"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb33-1"><a aria-hidden="true" href="#cb33-1" tabindex="-1"></a>(samples_per_otu</span>
<span id="cb33-2"><a aria-hidden="true" href="#cb33-2" tabindex="-1"></a> .query(<span class="st">'n_samples &gt;= 2'</span>)</span>
<span id="cb33-3"><a aria-hidden="true" href="#cb33-3" tabindex="-1"></a> .pipe(<span class="kw">lambda</span> dd: </span>
<span id="cb33-4"><a aria-hidden="true" href="#cb33-4" tabindex="-1"></a>       p9.ggplot(dd)</span>
<span id="cb33-5"><a aria-hidden="true" href="#cb33-5" tabindex="-1"></a>       <span class="op">+</span> p9.aes(<span class="st">'n_samples'</span>, fill<span class="op">=</span><span class="st">'Kingdom'</span>)</span>
<span id="cb33-6"><a aria-hidden="true" href="#cb33-6" tabindex="-1"></a>       <span class="op">+</span> p9.geom_histogram(binwidth<span class="op">=</span><span class="dv">1</span>)</span>
<span id="cb33-7"><a aria-hidden="true" href="#cb33-7" tabindex="-1"></a>       <span class="op">+</span> p9.scale_fill_manual(values<span class="op">=</span>[<span class="st">'#D3894C'</span>, <span class="st">'#435B97'</span>])</span>
<span id="cb33-8"><a aria-hidden="true" href="#cb33-8" tabindex="-1"></a>       <span class="op">+</span> p9.facet_wrap(<span class="st">'Kingdom'</span>, ncol<span class="op">=</span><span class="dv">1</span>)</span>
<span id="cb33-9"><a aria-hidden="true" href="#cb33-9" tabindex="-1"></a>       <span class="op">+</span> p9.geom_text(p9.aes(label<span class="op">=</span><span class="st">'"n = " + size'</span>), x<span class="op">=</span><span class="dv">18</span>, y<span class="op">=</span><span class="dv">35</span>, size<span class="op">=</span><span class="dv">8</span>,</span>
<span id="cb33-10"><a aria-hidden="true" href="#cb33-10" tabindex="-1"></a>                                 data<span class="op">=</span>dd.groupby(<span class="st">'Kingdom'</span>, as_index<span class="op">=</span><span class="va">False</span>).size().astype(<span class="bu">str</span>))</span>
<span id="cb33-11"><a aria-hidden="true" href="#cb33-11" tabindex="-1"></a>       <span class="op">+</span> p9.theme(figure_size<span class="op">=</span>(<span class="dv">4</span>, <span class="dv">3</span>))</span>
<span id="cb33-12"><a aria-hidden="true" href="#cb33-12" tabindex="-1"></a>       <span class="op">+</span> p9.guides(fill<span class="op">=</span><span class="va">False</span>)</span>
<span id="cb33-13"><a aria-hidden="true" href="#cb33-13" tabindex="-1"></a>       <span class="op">+</span> p9.labs(x<span class="op">=</span><span class="st">'Present in $n$ samples'</span>, y<span class="op">=</span><span class="st">'Number of unique Genera'</span>)</span>
<span id="cb33-14"><a aria-hidden="true" href="#cb33-14" tabindex="-1"></a>       )</span>
<span id="cb33-15"><a aria-hidden="true" href="#cb33-15" tabindex="-1"></a>)</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img class="figure-img" height="900" src="microbial_diversity_analysis_files/figure-html/cell-35-output-1.png" width="1200"/></p>
</figure>
</div>
</div>
</div>
<p>The total number of distinct genera is therefore 327, 226 of them bacterial and 101 fungal.</p>
<p>Take into account that fungal genera can appear at most in 20 samples since S06 didn’t work for ITS. Bacterial genera can appear in a maximum of 21 samples.</p>
<p>IF we look at the <em>ubiquitous</em> genera, those that appear in every single sample, this is the list:</p>
<div class="cell" data-execution_count="31" id="cell-58">
<div class="sourceCode cell-code" id="cb34"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb34-1"><a aria-hidden="true" href="#cb34-1" tabindex="-1"></a>samples_per_otu.query(<span class="st">'n_samples &gt;=20'</span>).set_index(<span class="st">'Kingdom'</span>)</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="31">
<div>
<table class="dataframe caption-top table table-sm table-striped small" data-border="1" data-quarto-postprocess="true">
<thead>
<tr class="header">
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th">Phylum</th>
<th data-quarto-table-cell-role="th">Class</th>
<th data-quarto-table-cell-role="th">Order</th>
<th data-quarto-table-cell-role="th">Family</th>
<th data-quarto-table-cell-role="th">Genus</th>
<th data-quarto-table-cell-role="th">n_samples</th>
</tr>
<tr class="odd">
<th data-quarto-table-cell-role="th">Kingdom</th>
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th"></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td data-quarto-table-cell-role="th">Bacteria</td>
<td>Firmicutes</td>
<td>Bacilli</td>
<td>Lactobacillales</td>
<td>Streptococcaceae</td>
<td>Streptococcus</td>
<td>20</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">Bacteria</td>
<td>Firmicutes</td>
<td>Bacilli</td>
<td>Staphylococcales</td>
<td>Staphylococcaceae</td>
<td>Staphylococcus</td>
<td>20</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">Bacteria</td>
<td>Proteobacteria</td>
<td>Alphaproteobacteria</td>
<td>Rhizobiales</td>
<td>Beijerinckiaceae</td>
<td>Methylobacterium-Methylorubrum</td>
<td>21</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">Bacteria</td>
<td>Proteobacteria</td>
<td>Alphaproteobacteria</td>
<td>Rhizobiales</td>
<td>Rhizobiaceae</td>
<td>Ochrobactrum</td>
<td>20</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">Bacteria</td>
<td>Proteobacteria</td>
<td>Alphaproteobacteria</td>
<td>Sphingomonadales</td>
<td>Sphingomonadaceae</td>
<td>Porphyrobacter</td>
<td>20</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">Bacteria</td>
<td>Proteobacteria</td>
<td>Alphaproteobacteria</td>
<td>Sphingomonadales</td>
<td>Sphingomonadaceae</td>
<td>Sphingomonas</td>
<td>21</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">Bacteria</td>
<td>Proteobacteria</td>
<td>Gammaproteobacteria</td>
<td>Burkholderiales</td>
<td>Comamonadaceae</td>
<td>Delftia</td>
<td>21</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">Bacteria</td>
<td>Proteobacteria</td>
<td>Gammaproteobacteria</td>
<td>Salinisphaerales</td>
<td>Solimonadaceae</td>
<td>Nevskia</td>
<td>21</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">Fungi</td>
<td>Ascomycota</td>
<td>Eurotiomycetes</td>
<td>Chaetothyriales</td>
<td>Herpotrichiellaceae</td>
<td>Exophiala</td>
<td>20</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">Fungi</td>
<td>Basidiomycota</td>
<td>Malasseziomycetes</td>
<td>Malasseziales</td>
<td>Malasseziaceae</td>
<td>Malassezia</td>
<td>20</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="cell" data-execution_count="46" id="cell-60">
<div class="sourceCode cell-code" id="cb35"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb35-1"><a aria-hidden="true" href="#cb35-1" tabindex="-1"></a>(counts_per_sample_type</span>
<span id="cb35-2"><a aria-hidden="true" href="#cb35-2" tabindex="-1"></a> .fillna(<span class="dv">0</span>)</span>
<span id="cb35-3"><a aria-hidden="true" href="#cb35-3" tabindex="-1"></a>  .melt(ignore_index<span class="op">=</span><span class="va">False</span>,</span>
<span id="cb35-4"><a aria-hidden="true" href="#cb35-4" tabindex="-1"></a>      value_name<span class="op">=</span><span class="st">'counts'</span>)</span>
<span id="cb35-5"><a aria-hidden="true" href="#cb35-5" tabindex="-1"></a></span>
<span id="cb35-6"><a aria-hidden="true" href="#cb35-6" tabindex="-1"></a>)</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="46">
<div>
<table class="dataframe caption-top table table-sm table-striped small" data-border="1" data-quarto-postprocess="true">
<thead>
<tr class="header">
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th">sample_mode</th>
<th data-quarto-table-cell-role="th">month</th>
<th data-quarto-table-cell-role="th">counts</th>
</tr>
<tr class="odd">
<th data-quarto-table-cell-role="th">Kingdom</th>
<th data-quarto-table-cell-role="th">Phylum</th>
<th data-quarto-table-cell-role="th">Class</th>
<th data-quarto-table-cell-role="th">Order</th>
<th data-quarto-table-cell-role="th">Family</th>
<th data-quarto-table-cell-role="th">Genus</th>
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th"></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td data-quarto-table-cell-role="th" data-valign="top" rowspan="5">Bacteria</td>
<td data-quarto-table-cell-role="th" data-valign="top" rowspan="5">Acidobacteriota</td>
<td data-quarto-table-cell-role="th" data-valign="top" rowspan="3">Acidobacteriae</td>
<td data-quarto-table-cell-role="th" data-valign="top" rowspan="3">Acidobacteriales</td>
<td data-quarto-table-cell-role="th" data-valign="top" rowspan="3">Acidobacteriaceae (Subgroup 1)</td>
<td data-quarto-table-cell-role="th">Bryocella</td>
<td>Flight</td>
<td>April</td>
<td>10.0</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">Granulicella</td>
<td>Flight</td>
<td>April</td>
<td>0.0</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">Terriglobus</td>
<td>Flight</td>
<td>April</td>
<td>0.0</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th" data-valign="top" rowspan="2">Blastocatellia</td>
<td data-quarto-table-cell-role="th">Blastocatellales</td>
<td data-quarto-table-cell-role="th">Blastocatellaceae</td>
<td data-quarto-table-cell-role="th">Blastocatella</td>
<td>Flight</td>
<td>April</td>
<td>0.0</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">Pyrinomonadales</td>
<td data-quarto-table-cell-role="th">Pyrinomonadaceae</td>
<td data-quarto-table-cell-role="th">RB41</td>
<td>Flight</td>
<td>April</td>
<td>26.0</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">...</td>
<td data-quarto-table-cell-role="th">...</td>
<td data-quarto-table-cell-role="th">...</td>
<td data-quarto-table-cell-role="th">...</td>
<td data-quarto-table-cell-role="th">...</td>
<td data-quarto-table-cell-role="th">...</td>
<td>...</td>
<td>...</td>
<td>...</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th" data-valign="top" rowspan="5">Fungi</td>
<td data-quarto-table-cell-role="th" data-valign="top" rowspan="5">Basidiomycota</td>
<td data-quarto-table-cell-role="th">Tremellomycetes</td>
<td data-quarto-table-cell-role="th">Trichosporonales</td>
<td data-quarto-table-cell-role="th">Trichosporonaceae</td>
<td data-quarto-table-cell-role="th">Trichosporon</td>
<td>Ground</td>
<td>February</td>
<td>0.0</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th" data-valign="top" rowspan="3">Ustilaginomycetes</td>
<td data-quarto-table-cell-role="th" data-valign="top" rowspan="3">Ustilaginales</td>
<td data-quarto-table-cell-role="th" data-valign="top" rowspan="3">Ustilaginaceae</td>
<td data-quarto-table-cell-role="th">Anthracocystis</td>
<td>Ground</td>
<td>February</td>
<td>0.0</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">Moesziomyces</td>
<td>Ground</td>
<td>February</td>
<td>0.0</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">Ustilago</td>
<td>Ground</td>
<td>February</td>
<td>89.0</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">Wallemiomycetes</td>
<td data-quarto-table-cell-role="th">Wallemiales</td>
<td data-quarto-table-cell-role="th">Wallemiaceae</td>
<td data-quarto-table-cell-role="th">Wallemia</td>
<td>Ground</td>
<td>February</td>
<td>202.0</td>
</tr>
</tbody>
</table>
<p>2284 rows × 3 columns</p>
</div>
</div>
</div>
<div class="cell" data-execution_count="47" id="cell-61">
<details class="code-fold">
<summary>Show Code</summary>
<div class="sourceCode cell-code" id="cb36"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb36-1"><a aria-hidden="true" href="#cb36-1" tabindex="-1"></a>min_n_samples <span class="op">=</span> <span class="dv">1</span></span>
<span id="cb36-2"><a aria-hidden="true" href="#cb36-2" tabindex="-1"></a>min_n_counts <span class="op">=</span> <span class="dv">2</span></span>
<span id="cb36-3"><a aria-hidden="true" href="#cb36-3" tabindex="-1"></a></span>
<span id="cb36-4"><a aria-hidden="true" href="#cb36-4" tabindex="-1"></a>over_threshold_otus <span class="op">=</span> (samples_per_otu</span>
<span id="cb36-5"><a aria-hidden="true" href="#cb36-5" tabindex="-1"></a>        .query(<span class="ss">f"n_samples &gt;= </span><span class="sc">{</span>min_n_samples<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb36-6"><a aria-hidden="true" href="#cb36-6" tabindex="-1"></a>        .Genus</span>
<span id="cb36-7"><a aria-hidden="true" href="#cb36-7" tabindex="-1"></a>        .unique()</span>
<span id="cb36-8"><a aria-hidden="true" href="#cb36-8" tabindex="-1"></a>         )</span>
<span id="cb36-9"><a aria-hidden="true" href="#cb36-9" tabindex="-1"></a></span>
<span id="cb36-10"><a aria-hidden="true" href="#cb36-10" tabindex="-1"></a>counts_per_sample_type <span class="op">=</span> <span class="op">\</span></span>
<span id="cb36-11"><a aria-hidden="true" href="#cb36-11" tabindex="-1"></a>(abundance_long_genera</span>
<span id="cb36-12"><a aria-hidden="true" href="#cb36-12" tabindex="-1"></a> .merge(samples_info, on<span class="op">=</span><span class="st">'sample_id'</span>)</span>
<span id="cb36-13"><a aria-hidden="true" href="#cb36-13" tabindex="-1"></a> .query(<span class="ss">f'counts &gt;= </span><span class="sc">{</span>min_n_counts<span class="sc">}</span><span class="ss">'</span>)</span>
<span id="cb36-14"><a aria-hidden="true" href="#cb36-14" tabindex="-1"></a> .query(<span class="st">'Genus not in @inconclusive_list'</span>)</span>
<span id="cb36-15"><a aria-hidden="true" href="#cb36-15" tabindex="-1"></a> .query(<span class="st">'Genus in @over_threshold_otus'</span>)</span>
<span id="cb36-16"><a aria-hidden="true" href="#cb36-16" tabindex="-1"></a> .assign(month<span class="op">=</span><span class="kw">lambda</span> dd: pd.to_datetime(dd.date).dt.month_name())</span>
<span id="cb36-17"><a aria-hidden="true" href="#cb36-17" tabindex="-1"></a> .groupby(taxon_names <span class="op">+</span> [<span class="st">'sample_mode'</span>, <span class="st">'month'</span>], as_index<span class="op">=</span><span class="va">False</span>)</span>
<span id="cb36-18"><a aria-hidden="true" href="#cb36-18" tabindex="-1"></a> .counts.<span class="bu">sum</span>()</span>
<span id="cb36-19"><a aria-hidden="true" href="#cb36-19" tabindex="-1"></a> .pivot(index<span class="op">=</span>taxon_names, columns<span class="op">=</span>[<span class="st">'sample_mode'</span>, <span class="st">'month'</span>], values<span class="op">=</span><span class="st">'counts'</span>)</span>
<span id="cb36-20"><a aria-hidden="true" href="#cb36-20" tabindex="-1"></a>)</span>
<span id="cb36-21"><a aria-hidden="true" href="#cb36-21" tabindex="-1"></a></span>
<span id="cb36-22"><a aria-hidden="true" href="#cb36-22" tabindex="-1"></a>presence_by_sample_type <span class="op">=</span> (counts_per_sample_type</span>
<span id="cb36-23"><a aria-hidden="true" href="#cb36-23" tabindex="-1"></a> .melt(ignore_index<span class="op">=</span><span class="va">False</span>, value_name<span class="op">=</span><span class="st">'counts'</span>)</span>
<span id="cb36-24"><a aria-hidden="true" href="#cb36-24" tabindex="-1"></a> .assign(present_in<span class="op">=</span><span class="kw">lambda</span> dd: dd.month <span class="op">+</span> <span class="st">'-'</span> <span class="op">+</span> dd.sample_mode)</span>
<span id="cb36-25"><a aria-hidden="true" href="#cb36-25" tabindex="-1"></a> .dropna()</span>
<span id="cb36-26"><a aria-hidden="true" href="#cb36-26" tabindex="-1"></a> .groupby(taxon_names)</span>
<span id="cb36-27"><a aria-hidden="true" href="#cb36-27" tabindex="-1"></a> .agg({<span class="st">'counts'</span>: <span class="st">'sum'</span>, <span class="st">'present_in'</span>: <span class="st">'unique'</span>})</span>
<span id="cb36-28"><a aria-hidden="true" href="#cb36-28" tabindex="-1"></a> .assign(presence_type<span class="op">=</span><span class="kw">lambda</span> dd: </span>
<span id="cb36-29"><a aria-hidden="true" href="#cb36-29" tabindex="-1"></a>         dd.present_in.<span class="bu">apply</span>(<span class="kw">lambda</span> x: <span class="st">'All sample types'</span> <span class="cf">if</span> <span class="bu">len</span>(x) <span class="op">==</span> <span class="dv">4</span> <span class="cf">else</span> (</span>
<span id="cb36-30"><a aria-hidden="true" href="#cb36-30" tabindex="-1"></a>         <span class="ss">f'Only </span><span class="sc">{</span>x[<span class="dv">0</span>]<span class="sc">}</span><span class="ss">'</span> <span class="cf">if</span> <span class="bu">len</span>(x) <span class="op">==</span> <span class="dv">1</span> <span class="cf">else</span> (</span>
<span id="cb36-31"><a aria-hidden="true" href="#cb36-31" tabindex="-1"></a>         <span class="st">'All except '</span> <span class="op">+</span> <span class="bu">list</span>(<span class="bu">set</span>(sample_types).difference(<span class="bu">set</span>(x)))[<span class="dv">0</span>] <span class="cf">if</span> <span class="bu">len</span>(x) <span class="op">==</span> <span class="dv">3</span> <span class="cf">else</span> (</span>
<span id="cb36-32"><a aria-hidden="true" href="#cb36-32" tabindex="-1"></a>         <span class="st">'Only February'</span> <span class="cf">if</span> <span class="st">'April'</span> <span class="kw">not</span> <span class="kw">in</span> <span class="st">''</span>.join(x) <span class="cf">else</span> (</span>
<span id="cb36-33"><a aria-hidden="true" href="#cb36-33" tabindex="-1"></a>         <span class="st">'Only April'</span> <span class="cf">if</span> <span class="st">'February'</span> <span class="kw">not</span> <span class="kw">in</span> <span class="st">''</span>.join(x) <span class="cf">else</span> (</span>
<span id="cb36-34"><a aria-hidden="true" href="#cb36-34" tabindex="-1"></a>         <span class="st">'Only Flight'</span> <span class="cf">if</span> <span class="st">'Ground'</span> <span class="kw">not</span> <span class="kw">in</span> <span class="st">''</span>.join(x) <span class="cf">else</span> (</span>
<span id="cb36-35"><a aria-hidden="true" href="#cb36-35" tabindex="-1"></a>         <span class="st">'Only Ground'</span> <span class="cf">if</span> <span class="st">'Flight'</span> <span class="kw">not</span> <span class="kw">in</span> <span class="st">''</span>.join(x) <span class="cf">else</span> (</span>
<span id="cb36-36"><a aria-hidden="true" href="#cb36-36" tabindex="-1"></a>         <span class="st">' and '</span>.join(x))</span>
<span id="cb36-37"><a aria-hidden="true" href="#cb36-37" tabindex="-1"></a>         ))))))</span>
<span id="cb36-38"><a aria-hidden="true" href="#cb36-38" tabindex="-1"></a>        ))</span>
<span id="cb36-39"><a aria-hidden="true" href="#cb36-39" tabindex="-1"></a>  .rename(columns<span class="op">=</span>{<span class="st">'counts'</span>: <span class="st">'total_counts'</span>})</span>
<span id="cb36-40"><a aria-hidden="true" href="#cb36-40" tabindex="-1"></a>.assign(month_presence<span class="op">=</span><span class="kw">lambda</span> dd: dd.presence_type.<span class="bu">apply</span>(<span class="kw">lambda</span> x: </span>
<span id="cb36-41"><a aria-hidden="true" href="#cb36-41" tabindex="-1"></a>    <span class="st">'Both months'</span> <span class="cf">if</span> <span class="st">'All'</span> <span class="kw">in</span> x <span class="kw">or</span> <span class="st">'and'</span> <span class="kw">in</span> x <span class="kw">or</span> x <span class="op">==</span> <span class="st">'Only Ground'</span> <span class="kw">or</span> x <span class="op">==</span> <span class="st">'Only Flight'</span> </span>
<span id="cb36-42"><a aria-hidden="true" href="#cb36-42" tabindex="-1"></a>                                                          <span class="cf">else</span> x.split(<span class="st">'-'</span>)[<span class="dv">0</span>]))</span>
<span id="cb36-43"><a aria-hidden="true" href="#cb36-43" tabindex="-1"></a> .assign(area_presence<span class="op">=</span><span class="kw">lambda</span> dd: dd.presence_type.<span class="bu">apply</span>(<span class="kw">lambda</span> x: </span>
<span id="cb36-44"><a aria-hidden="true" href="#cb36-44" tabindex="-1"></a>    <span class="st">'Both sites'</span> <span class="cf">if</span> <span class="st">'All'</span> <span class="kw">in</span> x <span class="kw">or</span> <span class="st">'and'</span> <span class="kw">in</span> x <span class="kw">or</span> x <span class="op">==</span> <span class="st">'Only February'</span> <span class="kw">or</span> x <span class="op">==</span> <span class="st">'Only April'</span> <span class="cf">else</span></span>
<span id="cb36-45"><a aria-hidden="true" href="#cb36-45" tabindex="-1"></a>                            <span class="st">'Only '</span> <span class="op">+</span> x.split(<span class="st">'-'</span>)[<span class="dv">1</span>].capitalize() <span class="cf">if</span> <span class="st">'-'</span> <span class="kw">in</span> x <span class="cf">else</span> x))        </span>
<span id="cb36-46"><a aria-hidden="true" href="#cb36-46" tabindex="-1"></a>)</span>
<span id="cb36-47"><a aria-hidden="true" href="#cb36-47" tabindex="-1"></a></span>
<span id="cb36-48"><a aria-hidden="true" href="#cb36-48" tabindex="-1"></a>counts_per_sample_type.columns <span class="op">=</span> counts_per_sample_type.columns.<span class="bu">map</span>(<span class="kw">lambda</span> x: <span class="st">'-'</span>.join(x))</span>
<span id="cb36-49"><a aria-hidden="true" href="#cb36-49" tabindex="-1"></a>presence_by_sample_type <span class="op">=</span> (presence_by_sample_type</span>
<span id="cb36-50"><a aria-hidden="true" href="#cb36-50" tabindex="-1"></a>      .merge(counts_per_sample_type, left_index<span class="op">=</span><span class="va">True</span>, right_index<span class="op">=</span><span class="va">True</span>)</span>
<span id="cb36-51"><a aria-hidden="true" href="#cb36-51" tabindex="-1"></a>      [[<span class="st">'total_counts'</span>, <span class="st">'Flight-February'</span>, <span class="st">'Flight-April'</span>, <span class="st">'Ground-February'</span>, <span class="st">'Ground-April'</span>,</span>
<span id="cb36-52"><a aria-hidden="true" href="#cb36-52" tabindex="-1"></a>        <span class="st">'month_presence'</span>, <span class="st">'area_presence'</span>, <span class="st">'presence_type'</span>]]</span>
<span id="cb36-53"><a aria-hidden="true" href="#cb36-53" tabindex="-1"></a>        .fillna(<span class="dv">0</span>)</span>
<span id="cb36-54"><a aria-hidden="true" href="#cb36-54" tabindex="-1"></a>        .applymap(<span class="kw">lambda</span> x: <span class="bu">int</span>(x) <span class="cf">if</span> <span class="bu">isinstance</span>(x, <span class="bu">float</span>) <span class="cf">else</span> x)</span>
<span id="cb36-55"><a aria-hidden="true" href="#cb36-55" tabindex="-1"></a>)</span>
<span id="cb36-56"><a aria-hidden="true" href="#cb36-56" tabindex="-1"></a></span>
<span id="cb36-57"><a aria-hidden="true" href="#cb36-57" tabindex="-1"></a>presence_by_sample_type.to_csv(<span class="st">'../output/tables/genera_presence_by_sample_type.csv'</span>)</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell" data-execution_count="50" id="cell-62">
<div class="sourceCode cell-code" id="cb37"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb37-1"><a aria-hidden="true" href="#cb37-1" tabindex="-1"></a>genera_long_df <span class="op">=</span> (abundance_df_genera</span>
<span id="cb37-2"><a aria-hidden="true" href="#cb37-2" tabindex="-1"></a> .drop(columns<span class="op">=</span>taxon_names[<span class="dv">1</span>:<span class="op">-</span><span class="dv">1</span>])</span>
<span id="cb37-3"><a aria-hidden="true" href="#cb37-3" tabindex="-1"></a> .groupby([<span class="st">'Kingdom'</span>, <span class="st">'Genus'</span>], as_index<span class="op">=</span><span class="va">False</span>)</span>
<span id="cb37-4"><a aria-hidden="true" href="#cb37-4" tabindex="-1"></a> .<span class="bu">sum</span>()</span>
<span id="cb37-5"><a aria-hidden="true" href="#cb37-5" tabindex="-1"></a> .melt([<span class="st">'Kingdom'</span>, <span class="st">'Genus'</span>], var_name<span class="op">=</span><span class="st">'sample_id'</span>, value_name<span class="op">=</span><span class="st">'counts'</span>)</span>
<span id="cb37-6"><a aria-hidden="true" href="#cb37-6" tabindex="-1"></a>)</span>
<span id="cb37-7"><a aria-hidden="true" href="#cb37-7" tabindex="-1"></a></span>
<span id="cb37-8"><a aria-hidden="true" href="#cb37-8" tabindex="-1"></a>top_10_genera <span class="op">=</span> (genera_long_df</span>
<span id="cb37-9"><a aria-hidden="true" href="#cb37-9" tabindex="-1"></a> .query(<span class="st">'Genus != "unknown"'</span>)</span>
<span id="cb37-10"><a aria-hidden="true" href="#cb37-10" tabindex="-1"></a> .groupby([<span class="st">'Kingdom'</span>, <span class="st">'Genus'</span>], as_index<span class="op">=</span><span class="va">False</span>)</span>
<span id="cb37-11"><a aria-hidden="true" href="#cb37-11" tabindex="-1"></a> .agg({<span class="st">'counts'</span>: <span class="st">'sum'</span>})</span>
<span id="cb37-12"><a aria-hidden="true" href="#cb37-12" tabindex="-1"></a> .groupby(<span class="st">'Kingdom'</span>, as_index<span class="op">=</span><span class="va">False</span>)</span>
<span id="cb37-13"><a aria-hidden="true" href="#cb37-13" tabindex="-1"></a> .<span class="bu">apply</span>(<span class="kw">lambda</span> dd: dd.sort_values(<span class="st">'counts'</span>, ascending<span class="op">=</span><span class="va">False</span>).head(<span class="dv">10</span>))</span>
<span id="cb37-14"><a aria-hidden="true" href="#cb37-14" tabindex="-1"></a> .reset_index(drop<span class="op">=</span><span class="va">True</span>)</span>
<span id="cb37-15"><a aria-hidden="true" href="#cb37-15" tabindex="-1"></a> [<span class="st">'Genus'</span>]</span>
<span id="cb37-16"><a aria-hidden="true" href="#cb37-16" tabindex="-1"></a>)</span>
<span id="cb37-17"><a aria-hidden="true" href="#cb37-17" tabindex="-1"></a></span>
<span id="cb37-18"><a aria-hidden="true" href="#cb37-18" tabindex="-1"></a>f <span class="op">=</span> (genera_long_df</span>
<span id="cb37-19"><a aria-hidden="true" href="#cb37-19" tabindex="-1"></a> .merge(samples_info)</span>
<span id="cb37-20"><a aria-hidden="true" href="#cb37-20" tabindex="-1"></a> .replace({<span class="st">'Genus'</span>: {<span class="st">'unknown'</span>: pd.NA}})</span>
<span id="cb37-21"><a aria-hidden="true" href="#cb37-21" tabindex="-1"></a> .assign(Genus<span class="op">=</span><span class="kw">lambda</span> dd: np.where(dd.Genus.isin(top_10_genera) <span class="op">|</span> dd.Genus.isna(), dd.Genus, <span class="st">'Other'</span>))</span>
<span id="cb37-22"><a aria-hidden="true" href="#cb37-22" tabindex="-1"></a> .query(<span class="st">'Kingdom=="Bacteria"'</span>)</span>
<span id="cb37-23"><a aria-hidden="true" href="#cb37-23" tabindex="-1"></a> .assign(date_short<span class="op">=</span><span class="kw">lambda</span> dd: pd.to_datetime(dd.date).dt.strftime(<span class="st">'%m/</span><span class="sc">%d</span><span class="st">'</span>))</span>
<span id="cb37-24"><a aria-hidden="true" href="#cb37-24" tabindex="-1"></a> .assign(date_short<span class="op">=</span><span class="kw">lambda</span> dd: np.where(dd.sample_id<span class="op">==</span><span class="st">'S12'</span>, <span class="st">'02/28.5'</span>, dd.date_short))</span>
<span id="cb37-25"><a aria-hidden="true" href="#cb37-25" tabindex="-1"></a> .pipe(<span class="kw">lambda</span> dd: p9.ggplot(dd)</span>
<span id="cb37-26"><a aria-hidden="true" href="#cb37-26" tabindex="-1"></a>       <span class="op">+</span> p9.aes(<span class="st">'date_short'</span>, <span class="st">'counts'</span>, fill<span class="op">=</span><span class="st">'Genus'</span>, group<span class="op">=</span><span class="st">'sample_id'</span>)</span>
<span id="cb37-27"><a aria-hidden="true" href="#cb37-27" tabindex="-1"></a>       <span class="op">+</span> p9.geom_col()</span>
<span id="cb37-28"><a aria-hidden="true" href="#cb37-28" tabindex="-1"></a>       <span class="op">+</span> p9.facet_wrap(<span class="st">'Kingdom + " " + sample_mode'</span>, ncol<span class="op">=</span><span class="dv">1</span>)</span>
<span id="cb37-29"><a aria-hidden="true" href="#cb37-29" tabindex="-1"></a>       <span class="op">+</span> p9.labs(x<span class="op">=</span><span class="st">''</span>, y<span class="op">=</span><span class="st">'Read Counts'</span>, fill<span class="op">=</span><span class="st">''</span>)</span>
<span id="cb37-30"><a aria-hidden="true" href="#cb37-30" tabindex="-1"></a>       <span class="op">+</span> p9.theme(figure_size<span class="op">=</span>(<span class="dv">4</span>, <span class="dv">5</span>),</span>
<span id="cb37-31"><a aria-hidden="true" href="#cb37-31" tabindex="-1"></a>                  axis_text_x<span class="op">=</span>p9.element_text(angle<span class="op">=</span><span class="dv">45</span>),</span>
<span id="cb37-32"><a aria-hidden="true" href="#cb37-32" tabindex="-1"></a>                  legend_key_size<span class="op">=</span><span class="dv">7</span>,</span>
<span id="cb37-33"><a aria-hidden="true" href="#cb37-33" tabindex="-1"></a>                  legend_text<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">7</span>, face<span class="op">=</span><span class="st">'italic'</span>),</span>
<span id="cb37-34"><a aria-hidden="true" href="#cb37-34" tabindex="-1"></a>                  legend_position<span class="op">=</span><span class="st">'bottom'</span></span>
<span id="cb37-35"><a aria-hidden="true" href="#cb37-35" tabindex="-1"></a>                  )</span>
<span id="cb37-36"><a aria-hidden="true" href="#cb37-36" tabindex="-1"></a>       )</span>
<span id="cb37-37"><a aria-hidden="true" href="#cb37-37" tabindex="-1"></a>)</span>
<span id="cb37-38"><a aria-hidden="true" href="#cb37-38" tabindex="-1"></a>f.save(<span class="st">'../output/figures/top_10_genera_bac.svg'</span>)</span>
<span id="cb37-39"><a aria-hidden="true" href="#cb37-39" tabindex="-1"></a>f.draw()</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="50">
<div>
<figure class="figure">
<p><img class="figure-img" height="1520" src="microbial_diversity_analysis_files/figure-html/cell-39-output-1.png" width="1497"/></p>
</figure>
</div>
</div>
</div>
<div class="cell" data-execution_count="52" id="cell-63">
<div class="sourceCode cell-code" id="cb38"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb38-1"><a aria-hidden="true" href="#cb38-1" tabindex="-1"></a>blanks_abundance_df_16s <span class="op">=</span> (pd.read_csv(</span>
<span id="cb38-2"><a aria-hidden="true" href="#cb38-2" tabindex="-1"></a>    <span class="st">'../data/nov23/ALL_AND_BLANKS_WITH_POOLS/reduced_and_merged_ONLY_genus_16S.csv'</span></span>
<span id="cb38-3"><a aria-hidden="true" href="#cb38-3" tabindex="-1"></a>    )</span>
<span id="cb38-4"><a aria-hidden="true" href="#cb38-4" tabindex="-1"></a></span>
<span id="cb38-5"><a aria-hidden="true" href="#cb38-5" tabindex="-1"></a>.melt(<span class="st">'Genus'</span>, var_name<span class="op">=</span><span class="st">'sample_id'</span>, value_name<span class="op">=</span><span class="st">'counts'</span>)</span>
<span id="cb38-6"><a aria-hidden="true" href="#cb38-6" tabindex="-1"></a>.assign(sample_id<span class="op">=</span><span class="kw">lambda</span> dd: dd[<span class="st">'sample_id'</span>]</span>
<span id="cb38-7"><a aria-hidden="true" href="#cb38-7" tabindex="-1"></a>        .<span class="bu">str</span>.replace(<span class="st">'subsampled'</span>, <span class="st">''</span>)</span>
<span id="cb38-8"><a aria-hidden="true" href="#cb38-8" tabindex="-1"></a>        .<span class="bu">str</span>.replace(<span class="st">'.'</span>, <span class="st">''</span>)</span>
<span id="cb38-9"><a aria-hidden="true" href="#cb38-9" tabindex="-1"></a>        .<span class="bu">str</span>.replace(<span class="st">'new'</span>, <span class="st">'</span><span class="ch">\n</span><span class="st">(new)'</span>)</span>
<span id="cb38-10"><a aria-hidden="true" href="#cb38-10" tabindex="-1"></a>        )</span>
<span id="cb38-11"><a aria-hidden="true" href="#cb38-11" tabindex="-1"></a>.replace({<span class="st">'Blank-Flightpool'</span>: <span class="st">'B_Flight'</span>,</span>
<span id="cb38-12"><a aria-hidden="true" href="#cb38-12" tabindex="-1"></a>          <span class="st">'Blank-Groundpool'</span>: <span class="st">'B_Ground'</span>,</span>
<span id="cb38-13"><a aria-hidden="true" href="#cb38-13" tabindex="-1"></a>          })</span>
<span id="cb38-14"><a aria-hidden="true" href="#cb38-14" tabindex="-1"></a>.query(<span class="st">'sample_id.str.contains("B_")'</span>)</span>
<span id="cb38-15"><a aria-hidden="true" href="#cb38-15" tabindex="-1"></a>.assign(Kingdom<span class="op">=</span><span class="st">'Bacteria'</span>)</span>
<span id="cb38-16"><a aria-hidden="true" href="#cb38-16" tabindex="-1"></a>)</span>
<span id="cb38-17"><a aria-hidden="true" href="#cb38-17" tabindex="-1"></a></span>
<span id="cb38-18"><a aria-hidden="true" href="#cb38-18" tabindex="-1"></a>blanks_abundance_df_its <span class="op">=</span> (pd.read_csv(</span>
<span id="cb38-19"><a aria-hidden="true" href="#cb38-19" tabindex="-1"></a>    <span class="st">'../data/nov23/ALL_AND_BLANKS_WITH_POOLS/complete_table_ITS.csv'</span></span>
<span id="cb38-20"><a aria-hidden="true" href="#cb38-20" tabindex="-1"></a>    )</span>
<span id="cb38-21"><a aria-hidden="true" href="#cb38-21" tabindex="-1"></a></span>
<span id="cb38-22"><a aria-hidden="true" href="#cb38-22" tabindex="-1"></a>.melt(<span class="st">'Genus'</span>, var_name<span class="op">=</span><span class="st">'sample_id'</span>, value_name<span class="op">=</span><span class="st">'counts'</span>)</span>
<span id="cb38-23"><a aria-hidden="true" href="#cb38-23" tabindex="-1"></a>.assign(sample_id<span class="op">=</span><span class="kw">lambda</span> dd: dd[<span class="st">'sample_id'</span>]</span>
<span id="cb38-24"><a aria-hidden="true" href="#cb38-24" tabindex="-1"></a>        .<span class="bu">str</span>.replace(<span class="st">'subsampled'</span>, <span class="st">''</span>)</span>
<span id="cb38-25"><a aria-hidden="true" href="#cb38-25" tabindex="-1"></a>        .<span class="bu">str</span>.replace(<span class="st">'.'</span>, <span class="st">''</span>)</span>
<span id="cb38-26"><a aria-hidden="true" href="#cb38-26" tabindex="-1"></a>        .<span class="bu">str</span>.replace(<span class="st">'new'</span>, <span class="st">'</span><span class="ch">\n</span><span class="st">(new)'</span>)</span>
<span id="cb38-27"><a aria-hidden="true" href="#cb38-27" tabindex="-1"></a>        )</span>
<span id="cb38-28"><a aria-hidden="true" href="#cb38-28" tabindex="-1"></a>.replace({<span class="st">'Blank-Flightpool'</span>: <span class="st">'B_Flight'</span>,</span>
<span id="cb38-29"><a aria-hidden="true" href="#cb38-29" tabindex="-1"></a>          <span class="st">'Blank-Groundpool'</span>: <span class="st">'B_Ground'</span>,</span>
<span id="cb38-30"><a aria-hidden="true" href="#cb38-30" tabindex="-1"></a>          })</span>
<span id="cb38-31"><a aria-hidden="true" href="#cb38-31" tabindex="-1"></a>.query(<span class="st">'sample_id.str.contains("B_")'</span>)</span>
<span id="cb38-32"><a aria-hidden="true" href="#cb38-32" tabindex="-1"></a>.assign(Kingdom<span class="op">=</span><span class="st">'Fungi'</span>)</span>
<span id="cb38-33"><a aria-hidden="true" href="#cb38-33" tabindex="-1"></a>)</span>
<span id="cb38-34"><a aria-hidden="true" href="#cb38-34" tabindex="-1"></a></span>
<span id="cb38-35"><a aria-hidden="true" href="#cb38-35" tabindex="-1"></a>blanks_abundance_df <span class="op">=</span> pd.concat([blanks_abundance_df_16s, blanks_abundance_df_its])</span>
<span id="cb38-36"><a aria-hidden="true" href="#cb38-36" tabindex="-1"></a></span>
<span id="cb38-37"><a aria-hidden="true" href="#cb38-37" tabindex="-1"></a>full_genera_df <span class="op">=</span> (pd.concat([genera_long_df, blanks_abundance_df])</span>
<span id="cb38-38"><a aria-hidden="true" href="#cb38-38" tabindex="-1"></a>            .assign(Genus<span class="op">=</span><span class="kw">lambda</span> dd: dd.Genus.<span class="bu">str</span>.split(<span class="st">'__'</span>).<span class="bu">str</span>[<span class="op">-</span><span class="dv">1</span>])</span>
<span id="cb38-39"><a aria-hidden="true" href="#cb38-39" tabindex="-1"></a>            )   </span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="53" id="cell-64">
<div class="sourceCode cell-code" id="cb39"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb39-1"><a aria-hidden="true" href="#cb39-1" tabindex="-1"></a>f <span class="op">=</span> (full_genera_df</span>
<span id="cb39-2"><a aria-hidden="true" href="#cb39-2" tabindex="-1"></a> .query(<span class="st">'sample_id.str.startswith("S")'</span>)</span>
<span id="cb39-3"><a aria-hidden="true" href="#cb39-3" tabindex="-1"></a> .query(<span class="st">'Kingdom == "Fungi"'</span>)</span>
<span id="cb39-4"><a aria-hidden="true" href="#cb39-4" tabindex="-1"></a>  .assign(counts<span class="op">=</span><span class="kw">lambda</span> dd: dd.counts.astype(<span class="bu">int</span>))</span>
<span id="cb39-5"><a aria-hidden="true" href="#cb39-5" tabindex="-1"></a> .merge(samples_info)</span>
<span id="cb39-6"><a aria-hidden="true" href="#cb39-6" tabindex="-1"></a> .replace({<span class="st">'Genus'</span>: {<span class="st">'unknown'</span>: pd.NA}})</span>
<span id="cb39-7"><a aria-hidden="true" href="#cb39-7" tabindex="-1"></a> .assign(Genus<span class="op">=</span><span class="kw">lambda</span> dd: np.where(dd.Genus.isin(top_10_genera) <span class="op">|</span> dd.Genus.isna(), dd.Genus, <span class="st">'Other'</span>))</span>
<span id="cb39-8"><a aria-hidden="true" href="#cb39-8" tabindex="-1"></a> .assign(date_short<span class="op">=</span><span class="kw">lambda</span> dd: pd.to_datetime(dd.date).dt.strftime(<span class="st">'%m/</span><span class="sc">%d</span><span class="st">'</span>))</span>
<span id="cb39-9"><a aria-hidden="true" href="#cb39-9" tabindex="-1"></a> .assign(date_short<span class="op">=</span><span class="kw">lambda</span> dd: np.where(dd.sample_id<span class="op">==</span><span class="st">'S12'</span>, <span class="st">'02/28.5'</span>, dd.date_short))</span>
<span id="cb39-10"><a aria-hidden="true" href="#cb39-10" tabindex="-1"></a> .pipe(<span class="kw">lambda</span> dd: p9.ggplot(dd)</span>
<span id="cb39-11"><a aria-hidden="true" href="#cb39-11" tabindex="-1"></a>       <span class="op">+</span> p9.aes(<span class="st">'date_short'</span>, <span class="st">'counts'</span>, fill<span class="op">=</span><span class="st">'Genus'</span>)</span>
<span id="cb39-12"><a aria-hidden="true" href="#cb39-12" tabindex="-1"></a>       <span class="op">+</span> p9.geom_col(p9.aes(group<span class="op">=</span><span class="st">'sample_id'</span>))</span>
<span id="cb39-13"><a aria-hidden="true" href="#cb39-13" tabindex="-1"></a>       <span class="op">+</span> p9.facet_wrap(<span class="st">'Kingdom + " " + sample_mode'</span>, ncol<span class="op">=</span><span class="dv">1</span>)</span>
<span id="cb39-14"><a aria-hidden="true" href="#cb39-14" tabindex="-1"></a>       <span class="op">+</span> p9.labs(x<span class="op">=</span><span class="st">''</span>, y<span class="op">=</span><span class="st">'Read Counts'</span>, fill<span class="op">=</span><span class="st">''</span>)</span>
<span id="cb39-15"><a aria-hidden="true" href="#cb39-15" tabindex="-1"></a>       <span class="op">+</span> p9.theme(figure_size<span class="op">=</span>(<span class="dv">4</span>, <span class="dv">5</span>),</span>
<span id="cb39-16"><a aria-hidden="true" href="#cb39-16" tabindex="-1"></a>                  axis_text_x<span class="op">=</span>p9.element_text(angle<span class="op">=</span><span class="dv">45</span>),</span>
<span id="cb39-17"><a aria-hidden="true" href="#cb39-17" tabindex="-1"></a>                  legend_key_size<span class="op">=</span><span class="dv">7</span>,</span>
<span id="cb39-18"><a aria-hidden="true" href="#cb39-18" tabindex="-1"></a>                                    legend_text<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">7</span>, face<span class="op">=</span><span class="st">'italic'</span>),</span>
<span id="cb39-19"><a aria-hidden="true" href="#cb39-19" tabindex="-1"></a></span>
<span id="cb39-20"><a aria-hidden="true" href="#cb39-20" tabindex="-1"></a>                  legend_position<span class="op">=</span><span class="st">'bottom'</span></span>
<span id="cb39-21"><a aria-hidden="true" href="#cb39-21" tabindex="-1"></a>                  )</span>
<span id="cb39-22"><a aria-hidden="true" href="#cb39-22" tabindex="-1"></a>       )</span>
<span id="cb39-23"><a aria-hidden="true" href="#cb39-23" tabindex="-1"></a>)</span>
<span id="cb39-24"><a aria-hidden="true" href="#cb39-24" tabindex="-1"></a>f.save(<span class="st">'../output/figures/top_10_genera_fun.svg'</span>)</span>
<span id="cb39-25"><a aria-hidden="true" href="#cb39-25" tabindex="-1"></a>f.draw()</span>
<span id="cb39-26"><a aria-hidden="true" href="#cb39-26" tabindex="-1"></a><span class="co"># )</span></span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="53">
<div>
<figure class="figure">
<p><img class="figure-img" height="1520" src="microbial_diversity_analysis_files/figure-html/cell-41-output-1.png" width="1293"/></p>
</figure>
</div>
</div>
</div>
<div class="cell" data-execution_count="36" id="cell-65">
<div class="sourceCode cell-code" id="cb40"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb40-1"><a aria-hidden="true" href="#cb40-1" tabindex="-1"></a>(genera_long_df</span>
<span id="cb40-2"><a aria-hidden="true" href="#cb40-2" tabindex="-1"></a> .merge(samples_info)</span>
<span id="cb40-3"><a aria-hidden="true" href="#cb40-3" tabindex="-1"></a> .assign(Genus<span class="op">=</span><span class="kw">lambda</span> dd: np.where(dd.Genus.isin(top_10_genera), dd.Genus, <span class="st">'Other'</span>))</span>
<span id="cb40-4"><a aria-hidden="true" href="#cb40-4" tabindex="-1"></a> .query(<span class="st">'Kingdom=="Bacteria"'</span>)</span>
<span id="cb40-5"><a aria-hidden="true" href="#cb40-5" tabindex="-1"></a><span class="co">#  .groupby('Genus', as_index=False)</span></span>
<span id="cb40-6"><a aria-hidden="true" href="#cb40-6" tabindex="-1"></a><span class="co">#  .agg({'counts': 'sum'})</span></span>
<span id="cb40-7"><a aria-hidden="true" href="#cb40-7" tabindex="-1"></a> .assign(date_short<span class="op">=</span><span class="kw">lambda</span> dd: pd.to_datetime(dd.date).dt.strftime(<span class="st">'%m/</span><span class="sc">%d</span><span class="st">'</span>))</span>
<span id="cb40-8"><a aria-hidden="true" href="#cb40-8" tabindex="-1"></a> .pipe(<span class="kw">lambda</span> dd: p9.ggplot(dd)</span>
<span id="cb40-9"><a aria-hidden="true" href="#cb40-9" tabindex="-1"></a>       <span class="op">+</span> p9.aes(<span class="st">'date_short'</span>, <span class="st">'counts'</span>, fill<span class="op">=</span><span class="st">'Genus'</span>)</span>
<span id="cb40-10"><a aria-hidden="true" href="#cb40-10" tabindex="-1"></a>       <span class="op">+</span> p9.geom_col(p9.aes(group<span class="op">=</span><span class="st">'sample_id'</span>), position<span class="op">=</span><span class="st">'dodge'</span>)</span>
<span id="cb40-11"><a aria-hidden="true" href="#cb40-11" tabindex="-1"></a>       <span class="op">+</span> p9.facet_wrap(<span class="st">'Kingdom + " " + sample_mode'</span>, ncol<span class="op">=</span><span class="dv">1</span>)</span>
<span id="cb40-12"><a aria-hidden="true" href="#cb40-12" tabindex="-1"></a>       <span class="op">+</span> p9.labs(x<span class="op">=</span><span class="st">''</span>, y<span class="op">=</span><span class="st">'Read Counts'</span>, fill<span class="op">=</span><span class="st">''</span>)</span>
<span id="cb40-13"><a aria-hidden="true" href="#cb40-13" tabindex="-1"></a>       <span class="op">+</span> p9.theme(figure_size<span class="op">=</span>(<span class="fl">4.5</span>, <span class="dv">5</span>),</span>
<span id="cb40-14"><a aria-hidden="true" href="#cb40-14" tabindex="-1"></a>                  axis_text_x<span class="op">=</span>p9.element_text(angle<span class="op">=</span><span class="dv">45</span>),</span>
<span id="cb40-15"><a aria-hidden="true" href="#cb40-15" tabindex="-1"></a>                  legend_key_size<span class="op">=</span><span class="dv">7</span>,</span>
<span id="cb40-16"><a aria-hidden="true" href="#cb40-16" tabindex="-1"></a>                  legend_position<span class="op">=</span><span class="st">'bottom'</span></span>
<span id="cb40-17"><a aria-hidden="true" href="#cb40-17" tabindex="-1"></a>                  )</span>
<span id="cb40-18"><a aria-hidden="true" href="#cb40-18" tabindex="-1"></a>       )</span>
<span id="cb40-19"><a aria-hidden="true" href="#cb40-19" tabindex="-1"></a>)</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img class="figure-img" height="1500" src="microbial_diversity_analysis_files/figure-html/cell-42-output-1.png" width="1350"/></p>
</figure>
</div>
</div>
</div>
<div class="cell" data-execution_count="37" id="cell-66">
<div class="sourceCode cell-code" id="cb41"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb41-1"><a aria-hidden="true" href="#cb41-1" tabindex="-1"></a>f <span class="op">=</span> (presence_by_sample_type</span>
<span id="cb41-2"><a aria-hidden="true" href="#cb41-2" tabindex="-1"></a> .groupby([<span class="st">'area_presence'</span>, <span class="st">'Kingdom'</span>], as_index<span class="op">=</span><span class="va">False</span>).size()</span>
<span id="cb41-3"><a aria-hidden="true" href="#cb41-3" tabindex="-1"></a> .assign(area_presence<span class="op">=</span><span class="kw">lambda</span> dd: dd.area_presence.<span class="bu">str</span>.replace(<span class="st">' '</span>, <span class="st">'</span><span class="ch">\n</span><span class="st">'</span>))</span>
<span id="cb41-4"><a aria-hidden="true" href="#cb41-4" tabindex="-1"></a>    .pipe(<span class="kw">lambda</span> dd: p9.ggplot(dd)</span>
<span id="cb41-5"><a aria-hidden="true" href="#cb41-5" tabindex="-1"></a>       <span class="op">+</span> p9.aes(<span class="st">'area_presence'</span>, <span class="st">'size'</span>)</span>
<span id="cb41-6"><a aria-hidden="true" href="#cb41-6" tabindex="-1"></a>       <span class="op">+</span> p9.geom_col(p9.aes(fill<span class="op">=</span><span class="st">'Kingdom'</span>))</span>
<span id="cb41-7"><a aria-hidden="true" href="#cb41-7" tabindex="-1"></a>       <span class="op">+</span> p9.geom_text(p9.aes(label<span class="op">=</span><span class="st">'perc'</span>, y<span class="op">=</span><span class="st">'size + 10'</span>), ha<span class="op">=</span><span class="st">'center'</span>, color<span class="op">=</span>[<span class="st">'black'</span>, <span class="st">'black'</span>, <span class="st">'black'</span>], size<span class="op">=</span><span class="dv">8</span>,</span>
<span id="cb41-8"><a aria-hidden="true" href="#cb41-8" tabindex="-1"></a>                     data<span class="op">=</span>dd</span>
<span id="cb41-9"><a aria-hidden="true" href="#cb41-9" tabindex="-1"></a>                      .groupby(<span class="st">'area_presence'</span>, as_index<span class="op">=</span><span class="va">False</span>)</span>
<span id="cb41-10"><a aria-hidden="true" href="#cb41-10" tabindex="-1"></a>                      .<span class="bu">sum</span>()</span>
<span id="cb41-11"><a aria-hidden="true" href="#cb41-11" tabindex="-1"></a>                      .assign(perc<span class="op">=</span><span class="kw">lambda</span> ddd: ((ddd[<span class="st">'size'</span>] <span class="op">/</span> ddd[<span class="st">'size'</span>].<span class="bu">sum</span>()) <span class="op">*</span> <span class="dv">100</span>).<span class="bu">round</span>(<span class="dv">1</span>)</span>
<span id="cb41-12"><a aria-hidden="true" href="#cb41-12" tabindex="-1"></a>                              .astype(<span class="bu">str</span>)<span class="op">+</span> <span class="st">'%'</span>))</span>
<span id="cb41-13"><a aria-hidden="true" href="#cb41-13" tabindex="-1"></a>    <span class="co">#    + p9.coord_flip()</span></span>
<span id="cb41-14"><a aria-hidden="true" href="#cb41-14" tabindex="-1"></a>       <span class="op">+</span> p9.labs(x<span class="op">=</span><span class="st">''</span>, y<span class="op">=</span><span class="st">'Number of unique genera'</span>, fill<span class="op">=</span><span class="st">''</span>)</span>
<span id="cb41-15"><a aria-hidden="true" href="#cb41-15" tabindex="-1"></a>       <span class="op">+</span> p9.scale_fill_manual(values <span class="op">=</span> [<span class="st">'#D3894C'</span>, <span class="st">'#435B97'</span>] )</span>
<span id="cb41-16"><a aria-hidden="true" href="#cb41-16" tabindex="-1"></a>       <span class="op">+</span> p9.ylim(<span class="dv">0</span>, <span class="dv">300</span>)</span>
<span id="cb41-17"><a aria-hidden="true" href="#cb41-17" tabindex="-1"></a>       <span class="op">+</span> p9.annotate(geom<span class="op">=</span><span class="st">'text'</span>, x<span class="op">=</span><span class="fl">2.8</span>, y<span class="op">=</span><span class="dv">270</span>,</span>
<span id="cb41-18"><a aria-hidden="true" href="#cb41-18" tabindex="-1"></a>                     label<span class="op">=</span><span class="ss">f'n = </span><span class="sc">{</span>dd[<span class="st">"size"</span>]<span class="sc">.</span><span class="bu">sum</span>()<span class="sc">}</span><span class="ss"> genera'</span>, size<span class="op">=</span><span class="dv">7</span>)</span>
<span id="cb41-19"><a aria-hidden="true" href="#cb41-19" tabindex="-1"></a>       <span class="op">+</span> p9.theme(figure_size<span class="op">=</span>(<span class="dv">2</span>, <span class="fl">2.5</span>),</span>
<span id="cb41-20"><a aria-hidden="true" href="#cb41-20" tabindex="-1"></a>                  axis_title<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">8</span>),</span>
<span id="cb41-21"><a aria-hidden="true" href="#cb41-21" tabindex="-1"></a>                  legend_key_size<span class="op">=</span><span class="dv">6</span>,</span>
<span id="cb41-22"><a aria-hidden="true" href="#cb41-22" tabindex="-1"></a>                  legend_text<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">7</span>),</span>
<span id="cb41-23"><a aria-hidden="true" href="#cb41-23" tabindex="-1"></a>                  legend_position<span class="op">=</span>(<span class="fl">.735</span>, <span class="fl">.95</span>), legend_margin<span class="op">=</span><span class="dv">0</span>)</span>
<span id="cb41-24"><a aria-hidden="true" href="#cb41-24" tabindex="-1"></a>         )</span>
<span id="cb41-25"><a aria-hidden="true" href="#cb41-25" tabindex="-1"></a>)</span>
<span id="cb41-26"><a aria-hidden="true" href="#cb41-26" tabindex="-1"></a>f.save(<span class="st">'../output/figures/genera_presence_by_area_vert.svg'</span>)</span>
<span id="cb41-27"><a aria-hidden="true" href="#cb41-27" tabindex="-1"></a>f.draw()</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="37">
<div>
<figure class="figure">
<p><img class="figure-img" height="750" src="microbial_diversity_analysis_files/figure-html/cell-43-output-1.png" width="600"/></p>
</figure>
</div>
</div>
</div>
<div class="cell" data-execution_count="38" id="cell-67">
<div class="sourceCode cell-code" id="cb42"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb42-1"><a aria-hidden="true" href="#cb42-1" tabindex="-1"></a>f <span class="op">=</span> (presence_by_sample_type</span>
<span id="cb42-2"><a aria-hidden="true" href="#cb42-2" tabindex="-1"></a> .groupby([<span class="st">'month_presence'</span>, <span class="st">'Kingdom'</span>], as_index<span class="op">=</span><span class="va">False</span>).size()</span>
<span id="cb42-3"><a aria-hidden="true" href="#cb42-3" tabindex="-1"></a> .assign(month_presence<span class="op">=</span><span class="kw">lambda</span> dd: dd.month_presence.<span class="bu">str</span>.replace(<span class="st">' '</span>, <span class="st">'</span><span class="ch">\n</span><span class="st">'</span>))</span>
<span id="cb42-4"><a aria-hidden="true" href="#cb42-4" tabindex="-1"></a>    .pipe(<span class="kw">lambda</span> dd: p9.ggplot(dd)</span>
<span id="cb42-5"><a aria-hidden="true" href="#cb42-5" tabindex="-1"></a>       <span class="op">+</span> p9.aes(<span class="st">'month_presence'</span>, <span class="st">'size'</span>)</span>
<span id="cb42-6"><a aria-hidden="true" href="#cb42-6" tabindex="-1"></a>       <span class="op">+</span> p9.geom_col(p9.aes(fill<span class="op">=</span><span class="st">'Kingdom'</span>))</span>
<span id="cb42-7"><a aria-hidden="true" href="#cb42-7" tabindex="-1"></a>       <span class="op">+</span> p9.geom_text(p9.aes(label<span class="op">=</span><span class="st">'perc'</span>, y<span class="op">=</span><span class="st">'size + 10'</span>), ha<span class="op">=</span><span class="st">'center'</span>, color<span class="op">=</span>[<span class="st">'black'</span>, <span class="st">'black'</span>, <span class="st">'black'</span>], size<span class="op">=</span><span class="dv">8</span>,</span>
<span id="cb42-8"><a aria-hidden="true" href="#cb42-8" tabindex="-1"></a>                     data<span class="op">=</span>dd</span>
<span id="cb42-9"><a aria-hidden="true" href="#cb42-9" tabindex="-1"></a>                      .groupby(<span class="st">'month_presence'</span>, as_index<span class="op">=</span><span class="va">False</span>)</span>
<span id="cb42-10"><a aria-hidden="true" href="#cb42-10" tabindex="-1"></a>                      .<span class="bu">sum</span>()</span>
<span id="cb42-11"><a aria-hidden="true" href="#cb42-11" tabindex="-1"></a>                      .assign(perc<span class="op">=</span><span class="kw">lambda</span> ddd: ((ddd[<span class="st">'size'</span>] <span class="op">/</span> ddd[<span class="st">'size'</span>].<span class="bu">sum</span>()) <span class="op">*</span> <span class="dv">100</span>).<span class="bu">round</span>(<span class="dv">1</span>)</span>
<span id="cb42-12"><a aria-hidden="true" href="#cb42-12" tabindex="-1"></a>                              .astype(<span class="bu">str</span>)<span class="op">+</span> <span class="st">'%'</span>))</span>
<span id="cb42-13"><a aria-hidden="true" href="#cb42-13" tabindex="-1"></a>    <span class="co">#    + p9.coord_flip()</span></span>
<span id="cb42-14"><a aria-hidden="true" href="#cb42-14" tabindex="-1"></a>       <span class="op">+</span> p9.labs(x<span class="op">=</span><span class="st">''</span>, y<span class="op">=</span><span class="st">'Number of unique genera'</span>, fill<span class="op">=</span><span class="st">''</span>)</span>
<span id="cb42-15"><a aria-hidden="true" href="#cb42-15" tabindex="-1"></a>       <span class="op">+</span> p9.scale_fill_manual(values <span class="op">=</span> [<span class="st">'#D3894C'</span>, <span class="st">'#435B97'</span>] )</span>
<span id="cb42-16"><a aria-hidden="true" href="#cb42-16" tabindex="-1"></a>       <span class="op">+</span> p9.ylim(<span class="dv">0</span>, <span class="dv">300</span>)</span>
<span id="cb42-17"><a aria-hidden="true" href="#cb42-17" tabindex="-1"></a>       <span class="op">+</span> p9.annotate(geom<span class="op">=</span><span class="st">'text'</span>, x<span class="op">=</span><span class="fl">2.8</span>, y<span class="op">=</span><span class="dv">270</span>,</span>
<span id="cb42-18"><a aria-hidden="true" href="#cb42-18" tabindex="-1"></a>                     label<span class="op">=</span><span class="ss">f'n = </span><span class="sc">{</span>dd[<span class="st">"size"</span>]<span class="sc">.</span><span class="bu">sum</span>()<span class="sc">}</span><span class="ss"> genera'</span>, size<span class="op">=</span><span class="dv">7</span>)</span>
<span id="cb42-19"><a aria-hidden="true" href="#cb42-19" tabindex="-1"></a>       <span class="op">+</span> p9.theme(figure_size<span class="op">=</span>(<span class="dv">2</span>, <span class="fl">2.5</span>),</span>
<span id="cb42-20"><a aria-hidden="true" href="#cb42-20" tabindex="-1"></a>                  axis_title<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">8</span>),</span>
<span id="cb42-21"><a aria-hidden="true" href="#cb42-21" tabindex="-1"></a>                  legend_key_size<span class="op">=</span><span class="dv">6</span>,</span>
<span id="cb42-22"><a aria-hidden="true" href="#cb42-22" tabindex="-1"></a>                  legend_text<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">7</span>),</span>
<span id="cb42-23"><a aria-hidden="true" href="#cb42-23" tabindex="-1"></a>                  legend_position<span class="op">=</span>(<span class="fl">.735</span>, <span class="fl">.95</span>), legend_margin<span class="op">=</span><span class="dv">0</span>)</span>
<span id="cb42-24"><a aria-hidden="true" href="#cb42-24" tabindex="-1"></a>         )</span>
<span id="cb42-25"><a aria-hidden="true" href="#cb42-25" tabindex="-1"></a>)</span>
<span id="cb42-26"><a aria-hidden="true" href="#cb42-26" tabindex="-1"></a>f.save(<span class="st">'../output/figures/genera_presence_by_month_vert.svg'</span>)</span>
<span id="cb42-27"><a aria-hidden="true" href="#cb42-27" tabindex="-1"></a>f.draw()</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="38">
<div>
<figure class="figure">
<p><img class="figure-img" height="750" src="microbial_diversity_analysis_files/figure-html/cell-44-output-1.png" width="600"/></p>
</figure>
</div>
</div>
</div>
<div class="cell" data-execution_count="39" id="cell-68">
<details class="code-fold">
<summary>Show Code</summary>
<div class="sourceCode cell-code" id="cb43"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb43-1"><a aria-hidden="true" href="#cb43-1" tabindex="-1"></a>f <span class="op">=</span> (presence_by_sample_type</span>
<span id="cb43-2"><a aria-hidden="true" href="#cb43-2" tabindex="-1"></a> .groupby([<span class="st">'area_presence'</span>, <span class="st">'Kingdom'</span>], as_index<span class="op">=</span><span class="va">False</span>).size()</span>
<span id="cb43-3"><a aria-hidden="true" href="#cb43-3" tabindex="-1"></a>    .pipe(<span class="kw">lambda</span> dd: p9.ggplot(dd)</span>
<span id="cb43-4"><a aria-hidden="true" href="#cb43-4" tabindex="-1"></a>       <span class="op">+</span> p9.aes(<span class="st">'area_presence'</span>, <span class="st">'size'</span>)</span>
<span id="cb43-5"><a aria-hidden="true" href="#cb43-5" tabindex="-1"></a>       <span class="op">+</span> p9.geom_col(p9.aes(fill<span class="op">=</span><span class="st">'Kingdom'</span>))</span>
<span id="cb43-6"><a aria-hidden="true" href="#cb43-6" tabindex="-1"></a>       <span class="op">+</span> p9.geom_text(p9.aes(label<span class="op">=</span><span class="st">'perc'</span>, y<span class="op">=</span><span class="st">'size + 5'</span>), ha<span class="op">=</span><span class="st">'left'</span>, color<span class="op">=</span>[<span class="st">'black'</span>, <span class="st">'black'</span>, <span class="st">'black'</span>], size<span class="op">=</span><span class="dv">8</span>,</span>
<span id="cb43-7"><a aria-hidden="true" href="#cb43-7" tabindex="-1"></a>                     data<span class="op">=</span>dd</span>
<span id="cb43-8"><a aria-hidden="true" href="#cb43-8" tabindex="-1"></a>                      .groupby(<span class="st">'area_presence'</span>, as_index<span class="op">=</span><span class="va">False</span>)</span>
<span id="cb43-9"><a aria-hidden="true" href="#cb43-9" tabindex="-1"></a>                      .<span class="bu">sum</span>()</span>
<span id="cb43-10"><a aria-hidden="true" href="#cb43-10" tabindex="-1"></a>                      .assign(perc<span class="op">=</span><span class="kw">lambda</span> ddd: ((ddd[<span class="st">'size'</span>] <span class="op">/</span> ddd[<span class="st">'size'</span>].<span class="bu">sum</span>()) <span class="op">*</span> <span class="dv">100</span>).<span class="bu">round</span>(<span class="dv">1</span>)</span>
<span id="cb43-11"><a aria-hidden="true" href="#cb43-11" tabindex="-1"></a>                              .astype(<span class="bu">str</span>)<span class="op">+</span> <span class="st">'%'</span>))</span>
<span id="cb43-12"><a aria-hidden="true" href="#cb43-12" tabindex="-1"></a>       <span class="op">+</span> p9.coord_flip()</span>
<span id="cb43-13"><a aria-hidden="true" href="#cb43-13" tabindex="-1"></a>       <span class="op">+</span> p9.labs(x<span class="op">=</span><span class="st">''</span>, y<span class="op">=</span><span class="st">'Number of unique genera'</span>, fill<span class="op">=</span><span class="st">''</span>)</span>
<span id="cb43-14"><a aria-hidden="true" href="#cb43-14" tabindex="-1"></a>       <span class="op">+</span> p9.scale_fill_manual(values <span class="op">=</span> [<span class="st">'#D3894C'</span>, <span class="st">'#435B97'</span>] )</span>
<span id="cb43-15"><a aria-hidden="true" href="#cb43-15" tabindex="-1"></a>       <span class="op">+</span> p9.ylim(<span class="dv">0</span>, <span class="dv">330</span>)</span>
<span id="cb43-16"><a aria-hidden="true" href="#cb43-16" tabindex="-1"></a>       <span class="op">+</span> p9.annotate(geom<span class="op">=</span><span class="st">'text'</span>, x<span class="op">=</span><span class="fl">3.4</span>, y<span class="op">=</span><span class="dv">270</span>,</span>
<span id="cb43-17"><a aria-hidden="true" href="#cb43-17" tabindex="-1"></a>                     label<span class="op">=</span><span class="ss">f'n = </span><span class="sc">{</span>dd[<span class="st">"size"</span>]<span class="sc">.</span><span class="bu">sum</span>()<span class="sc">}</span><span class="ss"> genera'</span>, size<span class="op">=</span><span class="dv">8</span>)</span>
<span id="cb43-18"><a aria-hidden="true" href="#cb43-18" tabindex="-1"></a>       <span class="op">+</span> p9.theme(figure_size<span class="op">=</span>(<span class="fl">3.5</span>, <span class="dv">2</span>),</span>
<span id="cb43-19"><a aria-hidden="true" href="#cb43-19" tabindex="-1"></a>                  axis_title<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">8</span>),</span>
<span id="cb43-20"><a aria-hidden="true" href="#cb43-20" tabindex="-1"></a>                  legend_key_size<span class="op">=</span><span class="dv">6</span>,</span>
<span id="cb43-21"><a aria-hidden="true" href="#cb43-21" tabindex="-1"></a>                  legend_text<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">7</span>),</span>
<span id="cb43-22"><a aria-hidden="true" href="#cb43-22" tabindex="-1"></a>                  legend_position<span class="op">=</span>(<span class="fl">.835</span>, <span class="fl">.85</span>), legend_margin<span class="op">=</span><span class="dv">0</span>)</span>
<span id="cb43-23"><a aria-hidden="true" href="#cb43-23" tabindex="-1"></a>         )</span>
<span id="cb43-24"><a aria-hidden="true" href="#cb43-24" tabindex="-1"></a>)</span>
<span id="cb43-25"><a aria-hidden="true" href="#cb43-25" tabindex="-1"></a></span>
<span id="cb43-26"><a aria-hidden="true" href="#cb43-26" tabindex="-1"></a>f.save(<span class="st">'../output/figures/genera_presence_by_area.svg'</span>)</span>
<span id="cb43-27"><a aria-hidden="true" href="#cb43-27" tabindex="-1"></a>f.draw()</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-display" data-execution_count="39">
<div>
<figure class="figure">
<p><img class="figure-img" height="600" src="microbial_diversity_analysis_files/figure-html/cell-45-output-1.png" width="1050"/></p>
</figure>
</div>
</div>
</div>
<div class="cell" data-execution_count="40" id="cell-69">
<details class="code-fold">
<summary>Show Code</summary>
<div class="sourceCode cell-code" id="cb44"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb44-1"><a aria-hidden="true" href="#cb44-1" tabindex="-1"></a>f <span class="op">=</span> (presence_by_sample_type</span>
<span id="cb44-2"><a aria-hidden="true" href="#cb44-2" tabindex="-1"></a> .groupby([<span class="st">'month_presence'</span>, <span class="st">'Kingdom'</span>], as_index<span class="op">=</span><span class="va">False</span>).size()</span>
<span id="cb44-3"><a aria-hidden="true" href="#cb44-3" tabindex="-1"></a>    .pipe(<span class="kw">lambda</span> dd: p9.ggplot(dd)</span>
<span id="cb44-4"><a aria-hidden="true" href="#cb44-4" tabindex="-1"></a>       <span class="op">+</span> p9.aes(<span class="st">'month_presence'</span>, <span class="st">'size'</span>)</span>
<span id="cb44-5"><a aria-hidden="true" href="#cb44-5" tabindex="-1"></a>       <span class="op">+</span> p9.geom_col(p9.aes(fill<span class="op">=</span><span class="st">'Kingdom'</span>))</span>
<span id="cb44-6"><a aria-hidden="true" href="#cb44-6" tabindex="-1"></a>       <span class="op">+</span> p9.geom_text(p9.aes(label<span class="op">=</span><span class="st">'perc'</span>, y<span class="op">=</span><span class="st">'size + 5'</span>), ha<span class="op">=</span><span class="st">'left'</span>, color<span class="op">=</span>[<span class="st">'black'</span>, <span class="st">'black'</span>, <span class="st">'black'</span>], size<span class="op">=</span><span class="dv">8</span>,</span>
<span id="cb44-7"><a aria-hidden="true" href="#cb44-7" tabindex="-1"></a>                     data<span class="op">=</span>dd</span>
<span id="cb44-8"><a aria-hidden="true" href="#cb44-8" tabindex="-1"></a>                      .groupby(<span class="st">'month_presence'</span>, as_index<span class="op">=</span><span class="va">False</span>)</span>
<span id="cb44-9"><a aria-hidden="true" href="#cb44-9" tabindex="-1"></a>                      .<span class="bu">sum</span>()</span>
<span id="cb44-10"><a aria-hidden="true" href="#cb44-10" tabindex="-1"></a>                      .assign(perc<span class="op">=</span><span class="kw">lambda</span> ddd: ((ddd[<span class="st">'size'</span>] <span class="op">/</span> ddd[<span class="st">'size'</span>].<span class="bu">sum</span>()) <span class="op">*</span> <span class="dv">100</span>).<span class="bu">round</span>(<span class="dv">1</span>)</span>
<span id="cb44-11"><a aria-hidden="true" href="#cb44-11" tabindex="-1"></a>                              .astype(<span class="bu">str</span>)<span class="op">+</span> <span class="st">'%'</span>))</span>
<span id="cb44-12"><a aria-hidden="true" href="#cb44-12" tabindex="-1"></a>       <span class="op">+</span> p9.coord_flip()</span>
<span id="cb44-13"><a aria-hidden="true" href="#cb44-13" tabindex="-1"></a>       <span class="op">+</span> p9.labs(x<span class="op">=</span><span class="st">''</span>, y<span class="op">=</span><span class="st">'Number of unique genera'</span>, fill<span class="op">=</span><span class="st">''</span>)</span>
<span id="cb44-14"><a aria-hidden="true" href="#cb44-14" tabindex="-1"></a>       <span class="op">+</span> p9.scale_fill_manual(values <span class="op">=</span> [<span class="st">'#D3894C'</span>, <span class="st">'#435B97'</span>] )</span>
<span id="cb44-15"><a aria-hidden="true" href="#cb44-15" tabindex="-1"></a>       <span class="op">+</span> p9.ylim(<span class="dv">0</span>, <span class="dv">330</span>)</span>
<span id="cb44-16"><a aria-hidden="true" href="#cb44-16" tabindex="-1"></a>       <span class="op">+</span> p9.guides(fill<span class="op">=</span><span class="va">False</span>)</span>
<span id="cb44-17"><a aria-hidden="true" href="#cb44-17" tabindex="-1"></a>       <span class="op">+</span> p9.annotate(geom<span class="op">=</span><span class="st">'text'</span>, x<span class="op">=</span><span class="fl">.6</span>, y<span class="op">=</span><span class="dv">275</span>,</span>
<span id="cb44-18"><a aria-hidden="true" href="#cb44-18" tabindex="-1"></a>                     label<span class="op">=</span><span class="ss">f'n = </span><span class="sc">{</span>dd[<span class="st">"size"</span>]<span class="sc">.</span><span class="bu">sum</span>()<span class="sc">}</span><span class="ss"> genera'</span>, size<span class="op">=</span><span class="dv">8</span>)</span>
<span id="cb44-19"><a aria-hidden="true" href="#cb44-19" tabindex="-1"></a>       <span class="op">+</span> p9.theme(figure_size<span class="op">=</span>(<span class="fl">3.5</span>, <span class="dv">2</span>),</span>
<span id="cb44-20"><a aria-hidden="true" href="#cb44-20" tabindex="-1"></a>                  axis_title<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">8</span>),</span>
<span id="cb44-21"><a aria-hidden="true" href="#cb44-21" tabindex="-1"></a>                  legend_key_size<span class="op">=</span><span class="dv">7</span>,</span>
<span id="cb44-22"><a aria-hidden="true" href="#cb44-22" tabindex="-1"></a>                  legend_position<span class="op">=</span>(<span class="fl">.775</span>, <span class="fl">.75</span>), legend_margin<span class="op">=</span><span class="dv">0</span>)</span>
<span id="cb44-23"><a aria-hidden="true" href="#cb44-23" tabindex="-1"></a>         )</span>
<span id="cb44-24"><a aria-hidden="true" href="#cb44-24" tabindex="-1"></a>)</span>
<span id="cb44-25"><a aria-hidden="true" href="#cb44-25" tabindex="-1"></a></span>
<span id="cb44-26"><a aria-hidden="true" href="#cb44-26" tabindex="-1"></a>f.save(<span class="st">'../output/figures/genera_presence_by_month.svg'</span>)</span>
<span id="cb44-27"><a aria-hidden="true" href="#cb44-27" tabindex="-1"></a>f.draw()</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-display" data-execution_count="40">
<div>
<figure class="figure">
<p><img class="figure-img" height="600" src="microbial_diversity_analysis_files/figure-html/cell-46-output-1.png" width="1050"/></p>
</figure>
</div>
</div>
</div>
<div class="cell" data-execution_count="41" id="cell-70">
<div class="sourceCode cell-code" id="cb45"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb45-1"><a aria-hidden="true" href="#cb45-1" tabindex="-1"></a>presence_by_sample_type.presence_type.value_counts().rename(<span class="st">'Number of genera'</span>).to_frame()</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="41">
<div>
<table class="dataframe caption-top table table-sm table-striped small" data-border="1" data-quarto-postprocess="true">
<thead>
<tr class="header">
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th">Number of genera</th>
</tr>
<tr class="odd">
<th data-quarto-table-cell-role="th">presence_type</th>
<th data-quarto-table-cell-role="th"></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td data-quarto-table-cell-role="th">Only February-Flight</td>
<td>104</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">Only February-Ground</td>
<td>97</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">Only April</td>
<td>76</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">Only April-Flight</td>
<td>71</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">All except April-ground</td>
<td>63</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">All sample types</td>
<td>52</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">Only February</td>
<td>38</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">Only April-Ground</td>
<td>30</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">Only Ground</td>
<td>12</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">Only Flight</td>
<td>11</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">April-Flight and February-Ground</td>
<td>11</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">February-Flight and April-Ground</td>
<td>6</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</section>
<section class="level3" id="filtering-for-correlation-analysis">
<h3 class="anchored" data-anchor-id="filtering-for-correlation-analysis">Filtering for correlation analysis</h3>
<p>Firstly, if we look at how many unique OTUs we observe at each taxonomic level:</p>
<div class="cell" data-execution_count="42" id="cell-73">
<div class="sourceCode cell-code" id="cb46"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb46-1"><a aria-hidden="true" href="#cb46-1" tabindex="-1"></a>(abundance_long_genera</span>
<span id="cb46-2"><a aria-hidden="true" href="#cb46-2" tabindex="-1"></a>  .query(<span class="st">'Genus not in @inconclusive_list'</span>)</span>
<span id="cb46-3"><a aria-hidden="true" href="#cb46-3" tabindex="-1"></a>  .query(<span class="st">'Phylum != "unknown"'</span>)</span>
<span id="cb46-4"><a aria-hidden="true" href="#cb46-4" tabindex="-1"></a>  .query(<span class="st">'Phylum != "Fungi_phy_Incertae_sedis"'</span>)</span>
<span id="cb46-5"><a aria-hidden="true" href="#cb46-5" tabindex="-1"></a>)</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="42">
<div>
<table class="dataframe caption-top table table-sm table-striped small" data-border="1" data-quarto-postprocess="true">
<thead>
<tr class="header">
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th">Kingdom</th>
<th data-quarto-table-cell-role="th">Phylum</th>
<th data-quarto-table-cell-role="th">Class</th>
<th data-quarto-table-cell-role="th">Order</th>
<th data-quarto-table-cell-role="th">Family</th>
<th data-quarto-table-cell-role="th">Genus</th>
<th data-quarto-table-cell-role="th">sample_id</th>
<th data-quarto-table-cell-role="th">counts</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td data-quarto-table-cell-role="th">0</td>
<td>Bacteria</td>
<td>Acidobacteriota</td>
<td>Acidobacteriae</td>
<td>Acidobacteriales</td>
<td>Acidobacteriaceae (Subgroup 1)</td>
<td>Bryocella</td>
<td>S01</td>
<td>0.0</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">1</td>
<td>Bacteria</td>
<td>Acidobacteriota</td>
<td>Acidobacteriae</td>
<td>Acidobacteriales</td>
<td>Acidobacteriaceae (Subgroup 1)</td>
<td>Granulicella</td>
<td>S01</td>
<td>0.0</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">2</td>
<td>Bacteria</td>
<td>Acidobacteriota</td>
<td>Acidobacteriae</td>
<td>Acidobacteriales</td>
<td>Acidobacteriaceae (Subgroup 1)</td>
<td>Terriglobus</td>
<td>S01</td>
<td>0.0</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">4</td>
<td>Bacteria</td>
<td>Acidobacteriota</td>
<td>Blastocatellia</td>
<td>Blastocatellales</td>
<td>Blastocatellaceae</td>
<td>Blastocatella</td>
<td>S01</td>
<td>9.0</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">6</td>
<td>Bacteria</td>
<td>Acidobacteriota</td>
<td>Blastocatellia</td>
<td>Pyrinomonadales</td>
<td>Pyrinomonadaceae</td>
<td>RB41</td>
<td>S01</td>
<td>0.0</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">15427</td>
<td>Fungi</td>
<td>Basidiomycota</td>
<td>Tremellomycetes</td>
<td>Trichosporonales</td>
<td>Trichosporonaceae</td>
<td>Trichosporon</td>
<td>S22</td>
<td>0.0</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">15428</td>
<td>Fungi</td>
<td>Basidiomycota</td>
<td>Ustilaginomycetes</td>
<td>Ustilaginales</td>
<td>Ustilaginaceae</td>
<td>Anthracocystis</td>
<td>S22</td>
<td>0.0</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">15429</td>
<td>Fungi</td>
<td>Basidiomycota</td>
<td>Ustilaginomycetes</td>
<td>Ustilaginales</td>
<td>Ustilaginaceae</td>
<td>Moesziomyces</td>
<td>S22</td>
<td>0.0</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">15430</td>
<td>Fungi</td>
<td>Basidiomycota</td>
<td>Ustilaginomycetes</td>
<td>Ustilaginales</td>
<td>Ustilaginaceae</td>
<td>Ustilago</td>
<td>S22</td>
<td>0.0</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">15432</td>
<td>Fungi</td>
<td>Basidiomycota</td>
<td>Wallemiomycetes</td>
<td>Wallemiales</td>
<td>Wallemiaceae</td>
<td>Wallemia</td>
<td>S22</td>
<td>0.0</td>
</tr>
</tbody>
</table>
<p>11991 rows × 8 columns</p>
</div>
</div>
</div>
<div class="cell" data-execution_count="185" id="cell-74">
<div class="sourceCode cell-code" id="cb47"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb47-1"><a aria-hidden="true" href="#cb47-1" tabindex="-1"></a>(abundance_long_genera</span>
<span id="cb47-2"><a aria-hidden="true" href="#cb47-2" tabindex="-1"></a>  .query(<span class="st">'Genus not in @inconclusive_list'</span>)</span>
<span id="cb47-3"><a aria-hidden="true" href="#cb47-3" tabindex="-1"></a>  .query(<span class="st">'Phylum != "unknown"'</span>)</span>
<span id="cb47-4"><a aria-hidden="true" href="#cb47-4" tabindex="-1"></a>  .query(<span class="st">'Phylum != "Fungi_phy_Incertae_sedis"'</span>)</span>
<span id="cb47-5"><a aria-hidden="true" href="#cb47-5" tabindex="-1"></a>  .query(<span class="st">'counts &gt; 0'</span>)</span>
<span id="cb47-6"><a aria-hidden="true" href="#cb47-6" tabindex="-1"></a>  .merge(samples_info)</span>
<span id="cb47-7"><a aria-hidden="true" href="#cb47-7" tabindex="-1"></a>  .assign(month<span class="op">=</span><span class="kw">lambda</span> dd: pd.to_datetime(dd.date).dt.month_name())</span>
<span id="cb47-8"><a aria-hidden="true" href="#cb47-8" tabindex="-1"></a>  .Phylum.nunique()</span>
<span id="cb47-9"><a aria-hidden="true" href="#cb47-9" tabindex="-1"></a><span class="co">#   .groupby(['Kingdom']).nunique()</span></span>
<span id="cb47-10"><a aria-hidden="true" href="#cb47-10" tabindex="-1"></a>)</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="185">
<pre><code>21</code></pre>
</div>
</div>
<div class="cell" data-execution_count="145" id="cell-75">
<div class="sourceCode cell-code" id="cb49"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb49-1"><a aria-hidden="true" href="#cb49-1" tabindex="-1"></a>(abundance_long_genera</span>
<span id="cb49-2"><a aria-hidden="true" href="#cb49-2" tabindex="-1"></a>  .query(<span class="st">'Genus not in @inconclusive_list'</span>)</span>
<span id="cb49-3"><a aria-hidden="true" href="#cb49-3" tabindex="-1"></a>  .query(<span class="st">'Phylum != "unknown"'</span>)</span>
<span id="cb49-4"><a aria-hidden="true" href="#cb49-4" tabindex="-1"></a>  .query(<span class="st">'Phylum != "Fungi_phy_Incertae_sedis"'</span>)</span>
<span id="cb49-5"><a aria-hidden="true" href="#cb49-5" tabindex="-1"></a>  .query(<span class="st">'counts &gt; 0'</span>)</span>
<span id="cb49-6"><a aria-hidden="true" href="#cb49-6" tabindex="-1"></a>  .merge(samples_info)</span>
<span id="cb49-7"><a aria-hidden="true" href="#cb49-7" tabindex="-1"></a>  .assign(month<span class="op">=</span><span class="kw">lambda</span> dd: pd.to_datetime(dd.date).dt.month_name())</span>
<span id="cb49-8"><a aria-hidden="true" href="#cb49-8" tabindex="-1"></a>  <span class="co"># .query('month=="February"')</span></span>
<span id="cb49-9"><a aria-hidden="true" href="#cb49-9" tabindex="-1"></a>  .groupby([<span class="st">'Kingdom'</span>])</span>
<span id="cb49-10"><a aria-hidden="true" href="#cb49-10" tabindex="-1"></a>  [taxon_names[<span class="dv">1</span>:]]</span>
<span id="cb49-11"><a aria-hidden="true" href="#cb49-11" tabindex="-1"></a>  .nunique()</span>
<span id="cb49-12"><a aria-hidden="true" href="#cb49-12" tabindex="-1"></a>  .T</span>
<span id="cb49-13"><a aria-hidden="true" href="#cb49-13" tabindex="-1"></a>)</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="145">
<div>
<table class="dataframe caption-top table table-sm table-striped small" data-border="1" data-quarto-postprocess="true">
<thead>
<tr class="header">
<th data-quarto-table-cell-role="th">Kingdom</th>
<th data-quarto-table-cell-role="th">Bacteria</th>
<th data-quarto-table-cell-role="th">Fungi</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td data-quarto-table-cell-role="th">Phylum</td>
<td>15</td>
<td>2</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">Class</td>
<td>25</td>
<td>19</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">Order</td>
<td>68</td>
<td>68</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">Family</td>
<td>127</td>
<td>168</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">Genus</td>
<td>305</td>
<td>266</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="cell" data-execution_count="44" id="cell-76">
<details class="code-fold">
<summary>Show Code</summary>
<div class="sourceCode cell-code" id="cb50"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb50-1"><a aria-hidden="true" href="#cb50-1" tabindex="-1"></a>df <span class="op">=</span> (abundance_long_genera</span>
<span id="cb50-2"><a aria-hidden="true" href="#cb50-2" tabindex="-1"></a>  .query(<span class="st">'Genus not in @inconclusive_list'</span>)</span>
<span id="cb50-3"><a aria-hidden="true" href="#cb50-3" tabindex="-1"></a>  .query(<span class="st">'Phylum != "unknown"'</span>)</span>
<span id="cb50-4"><a aria-hidden="true" href="#cb50-4" tabindex="-1"></a>  .query(<span class="st">'Phylum != "Fungi_phy_Incertae_sedis"'</span>)</span>
<span id="cb50-5"><a aria-hidden="true" href="#cb50-5" tabindex="-1"></a>  .groupby([<span class="st">'Kingdom'</span>])</span>
<span id="cb50-6"><a aria-hidden="true" href="#cb50-6" tabindex="-1"></a>  [taxon_names[<span class="dv">1</span>:]]</span>
<span id="cb50-7"><a aria-hidden="true" href="#cb50-7" tabindex="-1"></a>  .nunique()</span>
<span id="cb50-8"><a aria-hidden="true" href="#cb50-8" tabindex="-1"></a>)</span>
<span id="cb50-9"><a aria-hidden="true" href="#cb50-9" tabindex="-1"></a></span>
<span id="cb50-10"><a aria-hidden="true" href="#cb50-10" tabindex="-1"></a>df.loc[<span class="st">'TOTAL'</span>] <span class="op">=</span> df.<span class="bu">sum</span>()</span>
<span id="cb50-11"><a aria-hidden="true" href="#cb50-11" tabindex="-1"></a>df.rename(columns<span class="op">=</span>{</span>
<span id="cb50-12"><a aria-hidden="true" href="#cb50-12" tabindex="-1"></a>    <span class="st">'Phylum'</span>: <span class="st">'Phyla'</span>,</span>
<span id="cb50-13"><a aria-hidden="true" href="#cb50-13" tabindex="-1"></a>    <span class="st">'Class'</span>: <span class="st">'Classes'</span>,</span>
<span id="cb50-14"><a aria-hidden="true" href="#cb50-14" tabindex="-1"></a>    <span class="st">'Order'</span>: <span class="st">'Orders'</span>,</span>
<span id="cb50-15"><a aria-hidden="true" href="#cb50-15" tabindex="-1"></a>    <span class="st">'Family'</span>: <span class="st">'Families'</span>,</span>
<span id="cb50-16"><a aria-hidden="true" href="#cb50-16" tabindex="-1"></a>    <span class="st">'Genus'</span>: <span class="st">'Genera'</span>})</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-display" data-execution_count="44">
<div>
<table class="dataframe caption-top table table-sm table-striped small" data-border="1" data-quarto-postprocess="true">
<thead>
<tr class="header">
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th">Phyla</th>
<th data-quarto-table-cell-role="th">Classes</th>
<th data-quarto-table-cell-role="th">Orders</th>
<th data-quarto-table-cell-role="th">Families</th>
<th data-quarto-table-cell-role="th">Genera</th>
</tr>
<tr class="odd">
<th data-quarto-table-cell-role="th">Kingdom</th>
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th"></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td data-quarto-table-cell-role="th">Bacteria</td>
<td>15</td>
<td>25</td>
<td>68</td>
<td>127</td>
<td>305</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">Fungi</td>
<td>2</td>
<td>19</td>
<td>68</td>
<td>168</td>
<td>266</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">TOTAL</td>
<td>17</td>
<td>44</td>
<td>136</td>
<td>295</td>
<td>571</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<p>Let’s make a full scan of how the total number of Genera and Phyla changes as we change over these parameters:</p>
<div class="cell" data-execution_count="167" id="cell-78">
<div class="sourceCode cell-code" id="cb51"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb51-1"><a aria-hidden="true" href="#cb51-1" tabindex="-1"></a>(abundance_long_phyla</span>
<span id="cb51-2"><a aria-hidden="true" href="#cb51-2" tabindex="-1"></a> .groupby([<span class="st">'Kingdom'</span>, <span class="st">'Phylum'</span>])</span>
<span id="cb51-3"><a aria-hidden="true" href="#cb51-3" tabindex="-1"></a> .counts.<span class="bu">sum</span>()</span>
<span id="cb51-4"><a aria-hidden="true" href="#cb51-4" tabindex="-1"></a> .sort_values()</span>
<span id="cb51-5"><a aria-hidden="true" href="#cb51-5" tabindex="-1"></a>)</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="167">
<pre><code>Kingdom   Phylum           
Bacteria  Dependentiae              10
          Gemmatimonadota           27
          Myxococcota               28
          Deinococcota              79
          Armatimonadota            98
          Patescibacteria          110
          Planctomycetota          119
          Chloroflexi              123
          Campylobacterota         245
          Cyanobacteria            273
          Desulfobacterota         476
          Fusobacteriota           565
          Acidobacteriota          581
          Verrucomicrobiota        835
          unknown                 2702
          Actinobacteriota        6048
Fungi     unknown                 9247
Bacteria  Bacteroidota           30860
          Firmicutes             66403
Fungi     Basidiomycota         222314
Bacteria  Proteobacteria       1562108
Fungi     Ascomycota           1964855
Name: counts, dtype: int64</code></pre>
</div>
</div>
<div class="cell" data-execution_count="45" id="cell-79">
<details class="code-fold">
<summary>Show Code</summary>
<div class="sourceCode cell-code" id="cb53"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb53-1"><a aria-hidden="true" href="#cb53-1" tabindex="-1"></a>n_df <span class="op">=</span> pd.concat([</span>
<span id="cb53-2"><a aria-hidden="true" href="#cb53-2" tabindex="-1"></a> abundance_long_genera</span>
<span id="cb53-3"><a aria-hidden="true" href="#cb53-3" tabindex="-1"></a>  .query(<span class="st">'Genus not in @inconclusive_list'</span>)</span>
<span id="cb53-4"><a aria-hidden="true" href="#cb53-4" tabindex="-1"></a>  .groupby([<span class="st">'Kingdom'</span>])</span>
<span id="cb53-5"><a aria-hidden="true" href="#cb53-5" tabindex="-1"></a>  [[<span class="st">'Genus'</span>]]</span>
<span id="cb53-6"><a aria-hidden="true" href="#cb53-6" tabindex="-1"></a>  .nunique(),</span>
<span id="cb53-7"><a aria-hidden="true" href="#cb53-7" tabindex="-1"></a>  (abundance_long_phyla</span>
<span id="cb53-8"><a aria-hidden="true" href="#cb53-8" tabindex="-1"></a>   .query(<span class="st">'Phylum not in ["unknown", "Fungi_phy_Incertae_sedis"]'</span>)</span>
<span id="cb53-9"><a aria-hidden="true" href="#cb53-9" tabindex="-1"></a>    .groupby([<span class="st">'Kingdom'</span>])</span>
<span id="cb53-10"><a aria-hidden="true" href="#cb53-10" tabindex="-1"></a>    [[<span class="st">'Phylum'</span>]]</span>
<span id="cb53-11"><a aria-hidden="true" href="#cb53-11" tabindex="-1"></a>    .nunique()</span>
<span id="cb53-12"><a aria-hidden="true" href="#cb53-12" tabindex="-1"></a>  )</span>
<span id="cb53-13"><a aria-hidden="true" href="#cb53-13" tabindex="-1"></a>]</span>
<span id="cb53-14"><a aria-hidden="true" href="#cb53-14" tabindex="-1"></a>).melt(ignore_index<span class="op">=</span><span class="va">False</span>, var_name<span class="op">=</span><span class="st">'level'</span>, value_name<span class="op">=</span><span class="st">'n'</span>).dropna().reset_index()</span>
<span id="cb53-15"><a aria-hidden="true" href="#cb53-15" tabindex="-1"></a></span>
<span id="cb53-16"><a aria-hidden="true" href="#cb53-16" tabindex="-1"></a>abundance_long_df <span class="op">=</span> pd.concat([</span>
<span id="cb53-17"><a aria-hidden="true" href="#cb53-17" tabindex="-1"></a>    (abundance_long_genera</span>
<span id="cb53-18"><a aria-hidden="true" href="#cb53-18" tabindex="-1"></a>    .assign(level<span class="op">=</span><span class="st">'Genera'</span>)</span>
<span id="cb53-19"><a aria-hidden="true" href="#cb53-19" tabindex="-1"></a>    .query(<span class="st">'Genus not in @inconclusive_list'</span>)),</span>
<span id="cb53-20"><a aria-hidden="true" href="#cb53-20" tabindex="-1"></a>    (abundance_long_phyla</span>
<span id="cb53-21"><a aria-hidden="true" href="#cb53-21" tabindex="-1"></a>     .assign(level<span class="op">=</span><span class="st">'Phyla'</span>)</span>
<span id="cb53-22"><a aria-hidden="true" href="#cb53-22" tabindex="-1"></a>     .query(<span class="st">'Phylum != "unknown"'</span>)),</span>
<span id="cb53-23"><a aria-hidden="true" href="#cb53-23" tabindex="-1"></a>])</span>
<span id="cb53-24"><a aria-hidden="true" href="#cb53-24" tabindex="-1"></a></span>
<span id="cb53-25"><a aria-hidden="true" href="#cb53-25" tabindex="-1"></a>scan_df_list <span class="op">=</span> []</span>
<span id="cb53-26"><a aria-hidden="true" href="#cb53-26" tabindex="-1"></a><span class="cf">for</span> min_counts, min_samples <span class="kw">in</span> product(<span class="bu">range</span>(<span class="dv">2</span>, <span class="dv">11</span>), <span class="bu">range</span>(<span class="dv">1</span>, <span class="dv">8</span>)):</span>
<span id="cb53-27"><a aria-hidden="true" href="#cb53-27" tabindex="-1"></a>    scan_df_list.append(</span>
<span id="cb53-28"><a aria-hidden="true" href="#cb53-28" tabindex="-1"></a>        (abundance_long_df</span>
<span id="cb53-29"><a aria-hidden="true" href="#cb53-29" tabindex="-1"></a>         .query(<span class="st">'Genus not in @inconclusive_list'</span>)</span>
<span id="cb53-30"><a aria-hidden="true" href="#cb53-30" tabindex="-1"></a>         .query(<span class="st">'counts &gt;= @min_counts'</span>)</span>
<span id="cb53-31"><a aria-hidden="true" href="#cb53-31" tabindex="-1"></a>         .groupby(taxon_names <span class="op">+</span> [<span class="st">'level'</span>], as_index<span class="op">=</span><span class="va">False</span>, dropna<span class="op">=</span><span class="va">False</span>)</span>
<span id="cb53-32"><a aria-hidden="true" href="#cb53-32" tabindex="-1"></a>         .<span class="bu">apply</span>(<span class="kw">lambda</span> dd: pd.Series(dd.loc[dd[<span class="st">'counts'</span>] <span class="op">&gt;=</span> min_counts].shape[<span class="dv">0</span>]))</span>
<span id="cb53-33"><a aria-hidden="true" href="#cb53-33" tabindex="-1"></a>         .assign(over_threshold<span class="op">=</span><span class="kw">lambda</span> dd: dd[<span class="dv">0</span>] <span class="op">&gt;=</span> min_samples)</span>
<span id="cb53-34"><a aria-hidden="true" href="#cb53-34" tabindex="-1"></a>         .groupby([<span class="st">'Kingdom'</span>, <span class="st">'level'</span>], as_index<span class="op">=</span><span class="va">False</span>)</span>
<span id="cb53-35"><a aria-hidden="true" href="#cb53-35" tabindex="-1"></a>         .over_threshold</span>
<span id="cb53-36"><a aria-hidden="true" href="#cb53-36" tabindex="-1"></a>         .agg({<span class="st">'over'</span>: <span class="st">'mean'</span>, <span class="st">'total_over'</span>: <span class="st">'sum'</span>})</span>
<span id="cb53-37"><a aria-hidden="true" href="#cb53-37" tabindex="-1"></a>         .assign(min_counts<span class="op">=</span>min_counts, min_samples<span class="op">=</span>min_samples)</span>
<span id="cb53-38"><a aria-hidden="true" href="#cb53-38" tabindex="-1"></a>        )</span>
<span id="cb53-39"><a aria-hidden="true" href="#cb53-39" tabindex="-1"></a>    )</span>
<span id="cb53-40"><a aria-hidden="true" href="#cb53-40" tabindex="-1"></a></span>
<span id="cb53-41"><a aria-hidden="true" href="#cb53-41" tabindex="-1"></a>scan_df <span class="op">=</span> (pd.concat(scan_df_list)</span>
<span id="cb53-42"><a aria-hidden="true" href="#cb53-42" tabindex="-1"></a>    .merge(n_df.replace({<span class="st">'level'</span>: {<span class="st">'Phylum'</span>: <span class="st">'Phyla'</span>, <span class="st">'Genus'</span>: <span class="st">'Genera'</span>}}))</span>
<span id="cb53-43"><a aria-hidden="true" href="#cb53-43" tabindex="-1"></a>    .assign(label<span class="op">=</span><span class="kw">lambda</span> dd: dd[<span class="st">'total_over'</span>].astype(<span class="bu">str</span>) <span class="op">+</span> <span class="st">'/'</span> <span class="op">+</span> dd[<span class="st">'n'</span>].astype(<span class="bu">int</span>).astype(<span class="bu">str</span>))</span>
<span id="cb53-44"><a aria-hidden="true" href="#cb53-44" tabindex="-1"></a>    .assign(percent_label<span class="op">=</span><span class="kw">lambda</span> dd: (dd[<span class="st">'over'</span>] <span class="op">*</span> <span class="dv">100</span>).<span class="bu">round</span>(<span class="dv">2</span>).astype(<span class="bu">str</span>) <span class="op">+</span> <span class="st">'%'</span>)</span>
<span id="cb53-45"><a aria-hidden="true" href="#cb53-45" tabindex="-1"></a>    .assign(criteria_label<span class="op">=</span><span class="kw">lambda</span> dd: </span>
<span id="cb53-46"><a aria-hidden="true" href="#cb53-46" tabindex="-1"></a>            <span class="st">'r='</span> <span class="op">+</span> dd[<span class="st">'min_counts'</span>].astype(<span class="bu">str</span>)  <span class="op">+</span> </span>
<span id="cb53-47"><a aria-hidden="true" href="#cb53-47" tabindex="-1"></a>            <span class="st">'; n='</span> <span class="op">+</span> dd[<span class="st">'min_samples'</span>].astype(<span class="bu">str</span>))</span>
<span id="cb53-48"><a aria-hidden="true" href="#cb53-48" tabindex="-1"></a>    .assign(criteria_label<span class="op">=</span><span class="kw">lambda</span> dd: pd.Categorical(dd[<span class="st">'criteria_label'</span>], </span>
<span id="cb53-49"><a aria-hidden="true" href="#cb53-49" tabindex="-1"></a>                                                     categories<span class="op">=</span>dd[<span class="st">'criteria_label'</span>].unique(),</span>
<span id="cb53-50"><a aria-hidden="true" href="#cb53-50" tabindex="-1"></a>                                                    ordered<span class="op">=</span><span class="va">True</span>))</span>
<span id="cb53-51"><a aria-hidden="true" href="#cb53-51" tabindex="-1"></a>          )</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell" data-execution_count="159" id="cell-80">
<div class="sourceCode cell-code" id="cb54"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb54-1"><a aria-hidden="true" href="#cb54-1" tabindex="-1"></a>scan_df</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="159">
<div>
<table class="dataframe caption-top table table-sm table-striped small" data-border="1" data-quarto-postprocess="true">
<thead>
<tr class="header">
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th">Kingdom</th>
<th data-quarto-table-cell-role="th">level</th>
<th data-quarto-table-cell-role="th">over</th>
<th data-quarto-table-cell-role="th">total_over</th>
<th data-quarto-table-cell-role="th">min_counts</th>
<th data-quarto-table-cell-role="th">min_samples</th>
<th data-quarto-table-cell-role="th">n</th>
<th data-quarto-table-cell-role="th">label</th>
<th data-quarto-table-cell-role="th">percent_label</th>
<th data-quarto-table-cell-role="th">criteria_label</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td data-quarto-table-cell-role="th">0</td>
<td>Bacteria</td>
<td>Genera</td>
<td>1.000000</td>
<td>305</td>
<td>2</td>
<td>1</td>
<td>305.0</td>
<td>305/305</td>
<td>100.0%</td>
<td>r=2; n=1</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">1</td>
<td>Bacteria</td>
<td>Genera</td>
<td>0.622951</td>
<td>190</td>
<td>2</td>
<td>2</td>
<td>305.0</td>
<td>190/305</td>
<td>62.3%</td>
<td>r=2; n=2</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">2</td>
<td>Bacteria</td>
<td>Genera</td>
<td>0.508197</td>
<td>155</td>
<td>2</td>
<td>3</td>
<td>305.0</td>
<td>155/305</td>
<td>50.82%</td>
<td>r=2; n=3</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">3</td>
<td>Bacteria</td>
<td>Genera</td>
<td>0.416393</td>
<td>127</td>
<td>2</td>
<td>4</td>
<td>305.0</td>
<td>127/305</td>
<td>41.64%</td>
<td>r=2; n=4</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">4</td>
<td>Bacteria</td>
<td>Genera</td>
<td>0.363934</td>
<td>111</td>
<td>2</td>
<td>5</td>
<td>305.0</td>
<td>111/305</td>
<td>36.39%</td>
<td>r=2; n=5</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">247</td>
<td>Fungi</td>
<td>Phyla</td>
<td>1.000000</td>
<td>2</td>
<td>10</td>
<td>3</td>
<td>2.0</td>
<td>2/2</td>
<td>100.0%</td>
<td>r=10; n=3</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">248</td>
<td>Fungi</td>
<td>Phyla</td>
<td>1.000000</td>
<td>2</td>
<td>10</td>
<td>4</td>
<td>2.0</td>
<td>2/2</td>
<td>100.0%</td>
<td>r=10; n=4</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">249</td>
<td>Fungi</td>
<td>Phyla</td>
<td>1.000000</td>
<td>2</td>
<td>10</td>
<td>5</td>
<td>2.0</td>
<td>2/2</td>
<td>100.0%</td>
<td>r=10; n=5</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">250</td>
<td>Fungi</td>
<td>Phyla</td>
<td>1.000000</td>
<td>2</td>
<td>10</td>
<td>6</td>
<td>2.0</td>
<td>2/2</td>
<td>100.0%</td>
<td>r=10; n=6</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">251</td>
<td>Fungi</td>
<td>Phyla</td>
<td>1.000000</td>
<td>2</td>
<td>10</td>
<td>7</td>
<td>2.0</td>
<td>2/2</td>
<td>100.0%</td>
<td>r=10; n=7</td>
</tr>
</tbody>
</table>
<p>252 rows × 10 columns</p>
</div>
</div>
</div>
<div class="cell" data-execution_count="46" id="cell-81">
<div class="sourceCode cell-code" id="cb55"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb55-1"><a aria-hidden="true" href="#cb55-1" tabindex="-1"></a>f <span class="op">=</span> (scan_df</span>
<span id="cb55-2"><a aria-hidden="true" href="#cb55-2" tabindex="-1"></a>.assign(source_level<span class="op">=</span><span class="kw">lambda</span> dd: dd[<span class="st">'level'</span>] <span class="op">+</span> <span class="st">'-'</span> <span class="op">+</span> dd[<span class="st">'Kingdom'</span>]) </span>
<span id="cb55-3"><a aria-hidden="true" href="#cb55-3" tabindex="-1"></a>.assign(pos<span class="op">=</span><span class="kw">lambda</span> dd: np.where(dd.over <span class="op">&gt;</span> <span class="fl">.9</span>, <span class="dv">0</span>, dd.over))</span>
<span id="cb55-4"><a aria-hidden="true" href="#cb55-4" tabindex="-1"></a>.assign(label_len<span class="op">=</span><span class="kw">lambda</span> dd: dd.label.<span class="bu">str</span>.<span class="bu">len</span>())</span>
<span id="cb55-5"><a aria-hidden="true" href="#cb55-5" tabindex="-1"></a>.assign(pos<span class="op">=</span><span class="kw">lambda</span> dd: np.where(dd.label_len <span class="op">&gt;</span> dd.over <span class="op">*</span> <span class="dv">10</span>, dd.over, <span class="dv">0</span>))</span>
<span id="cb55-6"><a aria-hidden="true" href="#cb55-6" tabindex="-1"></a>.pipe(<span class="kw">lambda</span> dd: p9.ggplot(dd)</span>
<span id="cb55-7"><a aria-hidden="true" href="#cb55-7" tabindex="-1"></a>        <span class="op">+</span> p9.aes(<span class="st">'source_level'</span>, <span class="st">'over'</span>)</span>
<span id="cb55-8"><a aria-hidden="true" href="#cb55-8" tabindex="-1"></a>        <span class="op">+</span> p9.facet_wrap(<span class="st">'criteria_label'</span>, ncol<span class="op">=</span><span class="dv">7</span>)</span>
<span id="cb55-9"><a aria-hidden="true" href="#cb55-9" tabindex="-1"></a>        <span class="op">+</span> p9.geom_col()</span>
<span id="cb55-10"><a aria-hidden="true" href="#cb55-10" tabindex="-1"></a>        <span class="op">+</span> p9.scale_y_continuous(</span>
<span id="cb55-11"><a aria-hidden="true" href="#cb55-11" tabindex="-1"></a>            labels<span class="op">=</span>percent_format(),</span>
<span id="cb55-12"><a aria-hidden="true" href="#cb55-12" tabindex="-1"></a>            limits<span class="op">=</span>(<span class="dv">0</span>, <span class="fl">1.35</span>),</span>
<span id="cb55-13"><a aria-hidden="true" href="#cb55-13" tabindex="-1"></a>            breaks<span class="op">=</span>[<span class="dv">0</span>, <span class="dv">1</span>])</span>
<span id="cb55-14"><a aria-hidden="true" href="#cb55-14" tabindex="-1"></a>        <span class="op">+</span> p9.geom_text(p9.aes(y<span class="op">=</span><span class="st">'pos'</span>, label<span class="op">=</span><span class="st">'label'</span>, color<span class="op">=</span><span class="st">'pos &lt; over'</span>),</span>
<span id="cb55-15"><a aria-hidden="true" href="#cb55-15" tabindex="-1"></a>                       nudge_y<span class="op">=</span><span class="fl">.02</span>, size<span class="op">=</span><span class="dv">8</span>, ha<span class="op">=</span><span class="st">'left'</span>)</span>
<span id="cb55-16"><a aria-hidden="true" href="#cb55-16" tabindex="-1"></a>        <span class="op">+</span> p9.scale_color_manual(values<span class="op">=</span>[<span class="st">'black'</span>, <span class="st">'white'</span>])</span>
<span id="cb55-17"><a aria-hidden="true" href="#cb55-17" tabindex="-1"></a>        <span class="op">+</span> p9.coord_flip()</span>
<span id="cb55-18"><a aria-hidden="true" href="#cb55-18" tabindex="-1"></a>        <span class="op">+</span> p9.guides(color<span class="op">=</span><span class="va">False</span>)</span>
<span id="cb55-19"><a aria-hidden="true" href="#cb55-19" tabindex="-1"></a>        <span class="op">+</span> p9.labs(x<span class="op">=</span><span class="st">''</span>, y<span class="op">=</span><span class="st">'</span><span class="sc">% o</span><span class="st">f taxa kept after filter'</span>,</span>
<span id="cb55-20"><a aria-hidden="true" href="#cb55-20" tabindex="-1"></a>                  caption<span class="op">=</span><span class="st">"*r = Min number of reads; n = Min number of different samples"</span>)</span>
<span id="cb55-21"><a aria-hidden="true" href="#cb55-21" tabindex="-1"></a>        <span class="op">+</span> p9.theme(figure_size<span class="op">=</span>(<span class="dv">8</span>, <span class="dv">9</span>),</span>
<span id="cb55-22"><a aria-hidden="true" href="#cb55-22" tabindex="-1"></a>                   axis_title_y<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">8</span>))</span>
<span id="cb55-23"><a aria-hidden="true" href="#cb55-23" tabindex="-1"></a>)</span>
<span id="cb55-24"><a aria-hidden="true" href="#cb55-24" tabindex="-1"></a>)</span>
<span id="cb55-25"><a aria-hidden="true" href="#cb55-25" tabindex="-1"></a><span class="co"># f.save('../output/figures/scan_filtering.svg')</span></span>
<span id="cb55-26"><a aria-hidden="true" href="#cb55-26" tabindex="-1"></a>f.draw()</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="46">
<div>
<figure class="figure">
<p><img class="figure-img" height="2700" src="microbial_diversity_analysis_files/figure-html/cell-55-output-1.png" width="2400"/></p>
</figure>
</div>
</div>
</div>
<div class="cell" data-execution_count="47" id="cell-82">
<details class="code-fold">
<summary>Show Code</summary>
<div class="sourceCode cell-code" id="cb56"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb56-1"><a aria-hidden="true" href="#cb56-1" tabindex="-1"></a>f <span class="op">=</span> (abundance_long_df</span>
<span id="cb56-2"><a aria-hidden="true" href="#cb56-2" tabindex="-1"></a> .groupby(taxon_names <span class="op">+</span> [<span class="st">'level'</span>], as_index<span class="op">=</span><span class="va">False</span>, dropna<span class="op">=</span><span class="va">False</span>)</span>
<span id="cb56-3"><a aria-hidden="true" href="#cb56-3" tabindex="-1"></a> .<span class="bu">apply</span>(<span class="kw">lambda</span> dd: pd.Series(dd.loc[dd[<span class="st">'counts'</span>] <span class="op">&gt;=</span> <span class="dv">2</span>].shape[<span class="dv">0</span>]))</span>
<span id="cb56-4"><a aria-hidden="true" href="#cb56-4" tabindex="-1"></a> .assign(over_threshold<span class="op">=</span><span class="kw">lambda</span> dd: dd[<span class="dv">0</span>] <span class="op">&gt;=</span> <span class="dv">2</span>)</span>
<span id="cb56-5"><a aria-hidden="true" href="#cb56-5" tabindex="-1"></a> .groupby([<span class="st">'Kingdom'</span>, <span class="st">'level'</span>], as_index<span class="op">=</span><span class="va">False</span>)</span>
<span id="cb56-6"><a aria-hidden="true" href="#cb56-6" tabindex="-1"></a> .over_threshold</span>
<span id="cb56-7"><a aria-hidden="true" href="#cb56-7" tabindex="-1"></a> .agg({<span class="st">'over'</span>: <span class="st">'mean'</span>, <span class="st">'total_over'</span>: <span class="st">'sum'</span>, <span class="st">'n'</span>: <span class="st">'size'</span>})</span>
<span id="cb56-8"><a aria-hidden="true" href="#cb56-8" tabindex="-1"></a> .reset_index()</span>
<span id="cb56-9"><a aria-hidden="true" href="#cb56-9" tabindex="-1"></a> .assign(label<span class="op">=</span><span class="kw">lambda</span> dd: dd[<span class="st">'total_over'</span>].astype(<span class="bu">str</span>) <span class="op">+</span> <span class="st">'/'</span> <span class="op">+</span> dd[<span class="st">'n'</span>].astype(<span class="bu">str</span>))</span>
<span id="cb56-10"><a aria-hidden="true" href="#cb56-10" tabindex="-1"></a> .assign(percent_label<span class="op">=</span><span class="kw">lambda</span> dd: (dd[<span class="st">'over'</span>] <span class="op">*</span> <span class="dv">100</span>).<span class="bu">round</span>(<span class="dv">2</span>).astype(<span class="bu">str</span>) <span class="op">+</span> <span class="st">'%'</span>)</span>
<span id="cb56-11"><a aria-hidden="true" href="#cb56-11" tabindex="-1"></a> .pipe(<span class="kw">lambda</span> dd: p9.ggplot(dd)</span>
<span id="cb56-12"><a aria-hidden="true" href="#cb56-12" tabindex="-1"></a>        <span class="op">+</span> p9.aes(<span class="st">'Kingdom'</span>, <span class="st">'over'</span>)</span>
<span id="cb56-13"><a aria-hidden="true" href="#cb56-13" tabindex="-1"></a>        <span class="op">+</span> p9.facet_wrap(<span class="st">'level'</span>)</span>
<span id="cb56-14"><a aria-hidden="true" href="#cb56-14" tabindex="-1"></a>        <span class="op">+</span> p9.geom_col()</span>
<span id="cb56-15"><a aria-hidden="true" href="#cb56-15" tabindex="-1"></a>        <span class="op">+</span> p9.scale_y_continuous(labels<span class="op">=</span>percent_format(), limits<span class="op">=</span>(<span class="dv">0</span>, <span class="fl">1.1</span>),</span>
<span id="cb56-16"><a aria-hidden="true" href="#cb56-16" tabindex="-1"></a>                                breaks<span class="op">=</span>[<span class="dv">0</span>, <span class="fl">.25</span>, <span class="fl">.5</span>, <span class="fl">.75</span>, <span class="dv">1</span>])</span>
<span id="cb56-17"><a aria-hidden="true" href="#cb56-17" tabindex="-1"></a>        <span class="op">+</span> p9.geom_text(p9.aes(y<span class="op">=</span><span class="st">'over + .04'</span>, label<span class="op">=</span><span class="st">'label'</span>), size<span class="op">=</span><span class="dv">8</span>)</span>
<span id="cb56-18"><a aria-hidden="true" href="#cb56-18" tabindex="-1"></a>        <span class="op">+</span> p9.geom_text(p9.aes(label<span class="op">=</span><span class="st">'percent_label'</span>), size<span class="op">=</span><span class="dv">9</span>, y<span class="op">=</span><span class="fl">.05</span>, color<span class="op">=</span><span class="st">'white'</span>)</span>
<span id="cb56-19"><a aria-hidden="true" href="#cb56-19" tabindex="-1"></a>        <span class="op">+</span> p9.labs(x<span class="op">=</span><span class="st">''</span>, y<span class="op">=</span><span class="st">'</span><span class="sc">% o</span><span class="st">f taxa kept after filter'</span>)</span>
<span id="cb56-20"><a aria-hidden="true" href="#cb56-20" tabindex="-1"></a>        <span class="op">+</span> p9.ggtitle(<span class="st">'Taxa with at least 2 reads across at least 2 samples'</span>)</span>
<span id="cb56-21"><a aria-hidden="true" href="#cb56-21" tabindex="-1"></a>        <span class="op">+</span> p9.theme(figure_size<span class="op">=</span>(<span class="dv">4</span>, <span class="dv">2</span>),</span>
<span id="cb56-22"><a aria-hidden="true" href="#cb56-22" tabindex="-1"></a>                   plot_title<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">8</span>, ha<span class="op">=</span><span class="st">'center'</span>),</span>
<span id="cb56-23"><a aria-hidden="true" href="#cb56-23" tabindex="-1"></a>                   axis_title_y<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">9</span>))</span>
<span id="cb56-24"><a aria-hidden="true" href="#cb56-24" tabindex="-1"></a>)</span>
<span id="cb56-25"><a aria-hidden="true" href="#cb56-25" tabindex="-1"></a>)</span>
<span id="cb56-26"><a aria-hidden="true" href="#cb56-26" tabindex="-1"></a></span>
<span id="cb56-27"><a aria-hidden="true" href="#cb56-27" tabindex="-1"></a>f.save(<span class="st">'../output/figures/remaining_after_filter.svg'</span>)</span>
<span id="cb56-28"><a aria-hidden="true" href="#cb56-28" tabindex="-1"></a>f.draw()</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-display" data-execution_count="47">
<div>
<figure class="figure">
<p><img class="figure-img" height="600" src="microbial_diversity_analysis_files/figure-html/cell-56-output-1.png" width="1200"/></p>
</figure>
</div>
</div>
</div>
</section>
<section class="level3" id="diversity-per-sample">
<h3 class="anchored" data-anchor-id="diversity-per-sample">Diversity per sample</h3>
<div class="cell" data-execution_count="48" id="cell-84">
<div class="sourceCode cell-code" id="cb57"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb57-1"><a aria-hidden="true" href="#cb57-1" tabindex="-1"></a><span class="im">import</span> statsmodels.api <span class="im">as</span> sm</span>
<span id="cb57-2"><a aria-hidden="true" href="#cb57-2" tabindex="-1"></a><span class="im">from</span> sklearn.manifold <span class="im">import</span> MDS</span>
<span id="cb57-3"><a aria-hidden="true" href="#cb57-3" tabindex="-1"></a><span class="im">from</span> scipy.stats <span class="im">import</span> ranksums</span>
<span id="cb57-4"><a aria-hidden="true" href="#cb57-4" tabindex="-1"></a><span class="im">from</span> sklearn.metrics <span class="im">import</span> pairwise_distances</span>
<span id="cb57-5"><a aria-hidden="true" href="#cb57-5" tabindex="-1"></a><span class="im">from</span> statsmodels.sandbox.stats.multicomp <span class="im">import</span> MultiComparison</span>
<span id="cb57-6"><a aria-hidden="true" href="#cb57-6" tabindex="-1"></a><span class="im">from</span> statsmodels.stats.multitest <span class="im">import</span> multipletests</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="49" id="cell-85">
<div class="sourceCode cell-code" id="cb58"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb58-1"><a aria-hidden="true" href="#cb58-1" tabindex="-1"></a>diversity_df <span class="op">=</span> (diversity_df</span>
<span id="cb58-2"><a aria-hidden="true" href="#cb58-2" tabindex="-1"></a>        .merge(samples_info)</span>
<span id="cb58-3"><a aria-hidden="true" href="#cb58-3" tabindex="-1"></a>        .assign(month<span class="op">=</span><span class="kw">lambda</span> dd: pd.to_datetime(dd.date).dt.month_name())</span>
<span id="cb58-4"><a aria-hidden="true" href="#cb58-4" tabindex="-1"></a>)</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="50" id="cell-86">
<div class="sourceCode cell-code" id="cb59"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb59-1"><a aria-hidden="true" href="#cb59-1" tabindex="-1"></a><span class="co"># Function to perform pairwise tests and return p-values</span></span>
<span id="cb59-2"><a aria-hidden="true" href="#cb59-2" tabindex="-1"></a><span class="kw">def</span> perform_pairwise_wilcoxon(data, group_col, value_col):</span>
<span id="cb59-3"><a aria-hidden="true" href="#cb59-3" tabindex="-1"></a>    groups <span class="op">=</span> data[group_col].unique()</span>
<span id="cb59-4"><a aria-hidden="true" href="#cb59-4" tabindex="-1"></a>    result <span class="op">=</span> ranksums(data.query(<span class="ss">f'</span><span class="sc">{</span>group_col<span class="sc">}</span><span class="ss"> == "</span><span class="sc">{</span>groups[<span class="dv">0</span>]<span class="sc">}</span><span class="ss">"'</span>)[value_col],</span>
<span id="cb59-5"><a aria-hidden="true" href="#cb59-5" tabindex="-1"></a>                        data.query(<span class="ss">f'</span><span class="sc">{</span>group_col<span class="sc">}</span><span class="ss"> == "</span><span class="sc">{</span>groups[<span class="dv">1</span>]<span class="sc">}</span><span class="ss">"'</span>)[value_col])</span>
<span id="cb59-6"><a aria-hidden="true" href="#cb59-6" tabindex="-1"></a>    <span class="cf">return</span> result.pvalue</span>
<span id="cb59-7"><a aria-hidden="true" href="#cb59-7" tabindex="-1"></a>perform_pairwise_wilcoxon(diversity_df, <span class="st">'month'</span>, <span class="st">'chao'</span>)</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="50">
<pre><code>0.5811694758577145</code></pre>
</div>
</div>
<div class="cell" data-execution_count="51" id="cell-87">
<div class="sourceCode cell-code" id="cb61"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb61-1"><a aria-hidden="true" href="#cb61-1" tabindex="-1"></a>libraries <span class="op">=</span> []</span>
<span id="cb61-2"><a aria-hidden="true" href="#cb61-2" tabindex="-1"></a>subgroups <span class="op">=</span> []</span>
<span id="cb61-3"><a aria-hidden="true" href="#cb61-3" tabindex="-1"></a>test_names <span class="op">=</span> []</span>
<span id="cb61-4"><a aria-hidden="true" href="#cb61-4" tabindex="-1"></a>chao1_p_values <span class="op">=</span> []</span>
<span id="cb61-5"><a aria-hidden="true" href="#cb61-5" tabindex="-1"></a>shannon_p_values <span class="op">=</span> []</span>
<span id="cb61-6"><a aria-hidden="true" href="#cb61-6" tabindex="-1"></a></span>
<span id="cb61-7"><a aria-hidden="true" href="#cb61-7" tabindex="-1"></a><span class="cf">for</span> s <span class="kw">in</span> [<span class="st">'ITS'</span>, <span class="st">'16S'</span>]:</span>
<span id="cb61-8"><a aria-hidden="true" href="#cb61-8" tabindex="-1"></a>    df_filtered <span class="op">=</span> diversity_df.query(<span class="st">'source == @s'</span>)</span>
<span id="cb61-9"><a aria-hidden="true" href="#cb61-9" tabindex="-1"></a>    <span class="cf">for</span> test_column <span class="kw">in</span> [<span class="st">'sample_mode'</span>, <span class="st">'month'</span>]:</span>
<span id="cb61-10"><a aria-hidden="true" href="#cb61-10" tabindex="-1"></a>        </span>
<span id="cb61-11"><a aria-hidden="true" href="#cb61-11" tabindex="-1"></a>        <span class="cf">if</span> test_column <span class="op">==</span> <span class="st">'sample_mode'</span>:</span>
<span id="cb61-12"><a aria-hidden="true" href="#cb61-12" tabindex="-1"></a>            test_name <span class="op">=</span> <span class="st">'Ground-Flight'</span></span>
<span id="cb61-13"><a aria-hidden="true" href="#cb61-13" tabindex="-1"></a>        <span class="cf">elif</span> test_column <span class="op">==</span> <span class="st">'month'</span>:</span>
<span id="cb61-14"><a aria-hidden="true" href="#cb61-14" tabindex="-1"></a>            test_name <span class="op">=</span> <span class="st">'February-April'</span></span>
<span id="cb61-15"><a aria-hidden="true" href="#cb61-15" tabindex="-1"></a>        m <span class="op">=</span> <span class="st">'All'</span></span>
<span id="cb61-16"><a aria-hidden="true" href="#cb61-16" tabindex="-1"></a>        <span class="co"># Compute p-values</span></span>
<span id="cb61-17"><a aria-hidden="true" href="#cb61-17" tabindex="-1"></a>        chao1_p <span class="op">=</span> perform_pairwise_wilcoxon(df_filtered, test_column, <span class="st">'chao'</span>)</span>
<span id="cb61-18"><a aria-hidden="true" href="#cb61-18" tabindex="-1"></a>        shannon_p <span class="op">=</span> perform_pairwise_wilcoxon(df_filtered, test_column, <span class="st">'shannon'</span>)</span>
<span id="cb61-19"><a aria-hidden="true" href="#cb61-19" tabindex="-1"></a></span>
<span id="cb61-20"><a aria-hidden="true" href="#cb61-20" tabindex="-1"></a>        <span class="co"># Append results</span></span>
<span id="cb61-21"><a aria-hidden="true" href="#cb61-21" tabindex="-1"></a>        libraries.append(s)</span>
<span id="cb61-22"><a aria-hidden="true" href="#cb61-22" tabindex="-1"></a>        subgroups.append(m)</span>
<span id="cb61-23"><a aria-hidden="true" href="#cb61-23" tabindex="-1"></a>        test_names.append(test_name)</span>
<span id="cb61-24"><a aria-hidden="true" href="#cb61-24" tabindex="-1"></a>        chao1_p_values.append(chao1_p)</span>
<span id="cb61-25"><a aria-hidden="true" href="#cb61-25" tabindex="-1"></a>        shannon_p_values.append(shannon_p)</span>
<span id="cb61-26"><a aria-hidden="true" href="#cb61-26" tabindex="-1"></a></span>
<span id="cb61-27"><a aria-hidden="true" href="#cb61-27" tabindex="-1"></a><span class="co"># Display the results</span></span>
<span id="cb61-28"><a aria-hidden="true" href="#cb61-28" tabindex="-1"></a>res_df <span class="op">=</span> pd.DataFrame({<span class="st">'Library'</span>: libraries,</span>
<span id="cb61-29"><a aria-hidden="true" href="#cb61-29" tabindex="-1"></a>                          <span class="st">'Group'</span>: subgroups,</span>
<span id="cb61-30"><a aria-hidden="true" href="#cb61-30" tabindex="-1"></a>                          <span class="st">'Test'</span>: test_names,</span>
<span id="cb61-31"><a aria-hidden="true" href="#cb61-31" tabindex="-1"></a>                          <span class="st">'Chao1_p'</span>: chao1_p_values,</span>
<span id="cb61-32"><a aria-hidden="true" href="#cb61-32" tabindex="-1"></a>                          <span class="st">'Shannon_p'</span>: shannon_p_values})</span>
<span id="cb61-33"><a aria-hidden="true" href="#cb61-33" tabindex="-1"></a></span>
<span id="cb61-34"><a aria-hidden="true" href="#cb61-34" tabindex="-1"></a>(res_df</span>
<span id="cb61-35"><a aria-hidden="true" href="#cb61-35" tabindex="-1"></a> .sort_values([<span class="st">'Test'</span>, <span class="st">'Group'</span>])</span>
<span id="cb61-36"><a aria-hidden="true" href="#cb61-36" tabindex="-1"></a> <span class="co"># apply multiple test correction</span></span>
<span id="cb61-37"><a aria-hidden="true" href="#cb61-37" tabindex="-1"></a> .assign(Chao1_p_adj<span class="op">=</span><span class="kw">lambda</span> dd: multipletests(dd.Chao1_p, method<span class="op">=</span><span class="st">'fdr_bh'</span>)[<span class="dv">1</span>],</span>
<span id="cb61-38"><a aria-hidden="true" href="#cb61-38" tabindex="-1"></a>         Shannon_p_adj<span class="op">=</span><span class="kw">lambda</span> dd: multipletests(dd.Shannon_p, method<span class="op">=</span><span class="st">'fdr_bh'</span>)[<span class="dv">1</span>])</span>
<span id="cb61-39"><a aria-hidden="true" href="#cb61-39" tabindex="-1"></a> .style</span>
<span id="cb61-40"><a aria-hidden="true" href="#cb61-40" tabindex="-1"></a> .hide(axis<span class="op">=</span><span class="st">'index'</span>)</span>
<span id="cb61-41"><a aria-hidden="true" href="#cb61-41" tabindex="-1"></a>.applymap(<span class="kw">lambda</span> x: <span class="st">'color: red'</span> <span class="cf">if</span> <span class="bu">isinstance</span>(x, <span class="bu">float</span>) <span class="kw">and</span> x <span class="op">&lt;</span> <span class="fl">.05</span> <span class="cf">else</span> <span class="st">''</span>)</span>
<span id="cb61-42"><a aria-hidden="true" href="#cb61-42" tabindex="-1"></a>)</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="51">
<style type="text/css">
#T_3a3b7_row0_col3, #T_3a3b7_row0_col4, #T_3a3b7_row1_col3, #T_3a3b7_row1_col4, #T_3a3b7_row1_col5, #T_3a3b7_row1_col6 {
  color: red;
}
</style>
<table class="caption-top table table-sm table-striped small" data-quarto-postprocess="true" id="T_3a3b7">
<thead>
<tr class="header">
<th class="col_heading level0 col0" data-quarto-table-cell-role="th" id="T_3a3b7_level0_col0">Library</th>
<th class="col_heading level0 col1" data-quarto-table-cell-role="th" id="T_3a3b7_level0_col1">Group</th>
<th class="col_heading level0 col2" data-quarto-table-cell-role="th" id="T_3a3b7_level0_col2">Test</th>
<th class="col_heading level0 col3" data-quarto-table-cell-role="th" id="T_3a3b7_level0_col3">Chao1_p</th>
<th class="col_heading level0 col4" data-quarto-table-cell-role="th" id="T_3a3b7_level0_col4">Shannon_p</th>
<th class="col_heading level0 col5" data-quarto-table-cell-role="th" id="T_3a3b7_level0_col5">Chao1_p_adj</th>
<th class="col_heading level0 col6" data-quarto-table-cell-role="th" id="T_3a3b7_level0_col6">Shannon_p_adj</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td class="data row0 col0" id="T_3a3b7_row0_col0">ITS</td>
<td class="data row0 col1" id="T_3a3b7_row0_col1">All</td>
<td class="data row0 col2" id="T_3a3b7_row0_col2">February-April</td>
<td class="data row0 col3" id="T_3a3b7_row0_col3">0.027578</td>
<td class="data row0 col4" id="T_3a3b7_row0_col4">0.025012</td>
<td class="data row0 col5" id="T_3a3b7_row0_col5">0.055156</td>
<td class="data row0 col6" id="T_3a3b7_row0_col6">0.050023</td>
</tr>
<tr class="even">
<td class="data row1 col0" id="T_3a3b7_row1_col0">16S</td>
<td class="data row1 col1" id="T_3a3b7_row1_col1">All</td>
<td class="data row1 col2" id="T_3a3b7_row1_col2">February-April</td>
<td class="data row1 col3" id="T_3a3b7_row1_col3">0.000252</td>
<td class="data row1 col4" id="T_3a3b7_row1_col4">0.000124</td>
<td class="data row1 col5" id="T_3a3b7_row1_col5">0.001009</td>
<td class="data row1 col6" id="T_3a3b7_row1_col6">0.000497</td>
</tr>
<tr class="odd">
<td class="data row2 col0" id="T_3a3b7_row2_col0">ITS</td>
<td class="data row2 col1" id="T_3a3b7_row2_col1">All</td>
<td class="data row2 col2" id="T_3a3b7_row2_col2">Ground-Flight</td>
<td class="data row2 col3" id="T_3a3b7_row2_col3">0.384673</td>
<td class="data row2 col4" id="T_3a3b7_row2_col4">0.198765</td>
<td class="data row2 col5" id="T_3a3b7_row2_col5">0.481321</td>
<td class="data row2 col6" id="T_3a3b7_row2_col6">0.204969</td>
</tr>
<tr class="even">
<td class="data row3 col0" id="T_3a3b7_row3_col0">16S</td>
<td class="data row3 col1" id="T_3a3b7_row3_col1">All</td>
<td class="data row3 col2" id="T_3a3b7_row3_col2">Ground-Flight</td>
<td class="data row3 col3" id="T_3a3b7_row3_col3">0.481321</td>
<td class="data row3 col4" id="T_3a3b7_row3_col4">0.204969</td>
<td class="data row3 col5" id="T_3a3b7_row3_col5">0.481321</td>
<td class="data row3 col6" id="T_3a3b7_row3_col6">0.204969</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="cell" data-execution_count="52" id="cell-88">
<div class="sourceCode cell-code" id="cb62"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb62-1"><a aria-hidden="true" href="#cb62-1" tabindex="-1"></a>sample_sizes <span class="op">=</span> (diversity_df</span>
<span id="cb62-2"><a aria-hidden="true" href="#cb62-2" tabindex="-1"></a>                .groupby([<span class="st">'source'</span>, <span class="st">'sample_mode'</span>])</span>
<span id="cb62-3"><a aria-hidden="true" href="#cb62-3" tabindex="-1"></a>                .size()</span>
<span id="cb62-4"><a aria-hidden="true" href="#cb62-4" tabindex="-1"></a>                .rename(<span class="st">'n'</span>)</span>
<span id="cb62-5"><a aria-hidden="true" href="#cb62-5" tabindex="-1"></a>                .<span class="bu">apply</span>(<span class="kw">lambda</span> x: <span class="ss">f'n = </span><span class="sc">{</span>x<span class="sc">}</span><span class="ss">'</span>)</span>
<span id="cb62-6"><a aria-hidden="true" href="#cb62-6" tabindex="-1"></a>                .reset_index()</span>
<span id="cb62-7"><a aria-hidden="true" href="#cb62-7" tabindex="-1"></a>                .assign(Kingdom<span class="op">=</span><span class="kw">lambda</span> dd: dd.source.<span class="bu">map</span>({<span class="st">'16S'</span>: <span class="st">'Bacteria'</span>,</span>
<span id="cb62-8"><a aria-hidden="true" href="#cb62-8" tabindex="-1"></a>                                                          <span class="st">'ITS'</span>: <span class="st">'Fungi'</span>}))</span>
<span id="cb62-9"><a aria-hidden="true" href="#cb62-9" tabindex="-1"></a>               )</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="53" id="cell-89">
<div class="sourceCode cell-code" id="cb63"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb63-1"><a aria-hidden="true" href="#cb63-1" tabindex="-1"></a>sample_sizes_month <span class="op">=</span> (diversity_df</span>
<span id="cb63-2"><a aria-hidden="true" href="#cb63-2" tabindex="-1"></a>                .groupby([<span class="st">'source'</span>, <span class="st">'month'</span>])</span>
<span id="cb63-3"><a aria-hidden="true" href="#cb63-3" tabindex="-1"></a>                .size()</span>
<span id="cb63-4"><a aria-hidden="true" href="#cb63-4" tabindex="-1"></a>                .rename(<span class="st">'n'</span>)</span>
<span id="cb63-5"><a aria-hidden="true" href="#cb63-5" tabindex="-1"></a>                .<span class="bu">apply</span>(<span class="kw">lambda</span> x: <span class="ss">f'n = </span><span class="sc">{</span>x<span class="sc">}</span><span class="ss">'</span>)</span>
<span id="cb63-6"><a aria-hidden="true" href="#cb63-6" tabindex="-1"></a>                .reset_index()</span>
<span id="cb63-7"><a aria-hidden="true" href="#cb63-7" tabindex="-1"></a>                            .assign(Kingdom<span class="op">=</span><span class="kw">lambda</span> dd: dd.source.<span class="bu">map</span>({<span class="st">'16S'</span>: <span class="st">'Bacteria'</span>,</span>
<span id="cb63-8"><a aria-hidden="true" href="#cb63-8" tabindex="-1"></a>                                                          <span class="st">'ITS'</span>: <span class="st">'Fungi'</span>}))</span>
<span id="cb63-9"><a aria-hidden="true" href="#cb63-9" tabindex="-1"></a>               )</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="54" id="cell-90">
<div class="sourceCode cell-code" id="cb64"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb64-1"><a aria-hidden="true" href="#cb64-1" tabindex="-1"></a>f <span class="op">=</span> (diversity_df</span>
<span id="cb64-2"><a aria-hidden="true" href="#cb64-2" tabindex="-1"></a> .assign(Kingdom<span class="op">=</span><span class="kw">lambda</span> dd: dd.source.<span class="bu">map</span>({<span class="st">'16S'</span>: <span class="st">'Bacteria'</span>,</span>
<span id="cb64-3"><a aria-hidden="true" href="#cb64-3" tabindex="-1"></a>                                           <span class="st">'ITS'</span>: <span class="st">'Fungi'</span>}))</span>
<span id="cb64-4"><a aria-hidden="true" href="#cb64-4" tabindex="-1"></a> .melt([<span class="st">'sample_id'</span>, <span class="st">'source'</span>, <span class="st">'date'</span>, <span class="st">'sample_mode'</span>, <span class="st">'Kingdom'</span>, <span class="st">'month'</span>])</span>
<span id="cb64-5"><a aria-hidden="true" href="#cb64-5" tabindex="-1"></a> .replace({<span class="st">'chao'</span>: <span class="st">'Chao1 Index'</span>, <span class="st">'shannon'</span>: <span class="st">'Shannon Index'</span>})</span>
<span id="cb64-6"><a aria-hidden="true" href="#cb64-6" tabindex="-1"></a>.pipe(<span class="kw">lambda</span> dd: p9.ggplot(dd)</span>
<span id="cb64-7"><a aria-hidden="true" href="#cb64-7" tabindex="-1"></a>      <span class="op">+</span> p9.aes(<span class="st">'sample_mode'</span>, <span class="st">'value'</span>, fill<span class="op">=</span><span class="st">'Kingdom'</span>)</span>
<span id="cb64-8"><a aria-hidden="true" href="#cb64-8" tabindex="-1"></a>      <span class="op">+</span> p9.geom_jitter(alpha<span class="op">=</span><span class="fl">.6</span>, width<span class="op">=</span><span class="dv">0</span>, height<span class="op">=</span><span class="dv">0</span>, fill<span class="op">=</span><span class="st">'black'</span>)</span>
<span id="cb64-9"><a aria-hidden="true" href="#cb64-9" tabindex="-1"></a>      <span class="op">+</span> p9.geom_boxplot(alpha<span class="op">=</span><span class="fl">.6</span>, outlier_alpha<span class="op">=</span><span class="dv">0</span>)</span>
<span id="cb64-10"><a aria-hidden="true" href="#cb64-10" tabindex="-1"></a>      <span class="op">+</span> p9.geom_blank(p9.aes(y<span class="op">=</span><span class="st">'value * 1.1'</span>))</span>
<span id="cb64-11"><a aria-hidden="true" href="#cb64-11" tabindex="-1"></a>      <span class="op">+</span> p9.facet_grid([<span class="st">'variable'</span>, <span class="st">'Kingdom'</span>], scales<span class="op">=</span><span class="st">'free_y'</span>)</span>
<span id="cb64-12"><a aria-hidden="true" href="#cb64-12" tabindex="-1"></a>      <span class="op">+</span> p9.scale_fill_manual(values <span class="op">=</span> [<span class="st">'#D3894C'</span>, <span class="st">'#435B97'</span>] )</span>
<span id="cb64-13"><a aria-hidden="true" href="#cb64-13" tabindex="-1"></a>      <span class="op">+</span> p9.geom_text(p9.aes(label<span class="op">=</span><span class="st">'n'</span>), y<span class="op">=-</span><span class="fl">.2</span>, size<span class="op">=</span><span class="dv">8</span>, data<span class="op">=</span>sample_sizes)</span>
<span id="cb64-14"><a aria-hidden="true" href="#cb64-14" tabindex="-1"></a>      <span class="op">+</span> p9.guides(fill<span class="op">=</span><span class="va">False</span>)</span>
<span id="cb64-15"><a aria-hidden="true" href="#cb64-15" tabindex="-1"></a>      <span class="op">+</span> p9.labs(x<span class="op">=</span><span class="st">''</span>, y<span class="op">=</span><span class="vs">r'$\alpha$-Diversity '</span>)</span>
<span id="cb64-16"><a aria-hidden="true" href="#cb64-16" tabindex="-1"></a>      <span class="op">+</span> p9.scale_y_continuous(expand<span class="op">=</span>(<span class="fl">.175</span>, <span class="fl">.1</span>))</span>
<span id="cb64-17"><a aria-hidden="true" href="#cb64-17" tabindex="-1"></a>      <span class="op">+</span> p9.theme(figure_size<span class="op">=</span>(<span class="dv">4</span>, <span class="dv">3</span>)</span>
<span id="cb64-18"><a aria-hidden="true" href="#cb64-18" tabindex="-1"></a>    )</span>
<span id="cb64-19"><a aria-hidden="true" href="#cb64-19" tabindex="-1"></a>)</span>
<span id="cb64-20"><a aria-hidden="true" href="#cb64-20" tabindex="-1"></a>)</span>
<span id="cb64-21"><a aria-hidden="true" href="#cb64-21" tabindex="-1"></a>f.save(<span class="st">'../output/figures/boxplots_diversity_by_sample_type.svg'</span>)</span>
<span id="cb64-22"><a aria-hidden="true" href="#cb64-22" tabindex="-1"></a>f.draw()</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="54">
<div>
<figure class="figure">
<p><img class="figure-img" height="900" src="microbial_diversity_analysis_files/figure-html/cell-63-output-1.png" width="1200"/></p>
</figure>
</div>
</div>
</div>
<div class="cell" data-execution_count="55" id="cell-91">
<div class="sourceCode cell-code" id="cb65"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb65-1"><a aria-hidden="true" href="#cb65-1" tabindex="-1"></a>sample_sizes_month <span class="op">=</span> sample_sizes_month.assign(Kingdom<span class="op">=</span><span class="kw">lambda</span> dd: dd.source.<span class="bu">map</span>({<span class="st">'16S'</span>: <span class="st">'Bacteria'</span>, <span class="st">'ITS'</span>: <span class="st">'Fungi'</span>}))</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="56" id="cell-92">
<div class="sourceCode cell-code" id="cb66"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb66-1"><a aria-hidden="true" href="#cb66-1" tabindex="-1"></a>f <span class="op">=</span> (diversity_df</span>
<span id="cb66-2"><a aria-hidden="true" href="#cb66-2" tabindex="-1"></a> .assign(Kingdom<span class="op">=</span><span class="kw">lambda</span> dd: dd.source.<span class="bu">map</span>({<span class="st">'16S'</span>: <span class="st">'Bacteria'</span>,</span>
<span id="cb66-3"><a aria-hidden="true" href="#cb66-3" tabindex="-1"></a>                                           <span class="st">'ITS'</span>: <span class="st">'Fungi'</span>}))</span>
<span id="cb66-4"><a aria-hidden="true" href="#cb66-4" tabindex="-1"></a> .melt([<span class="st">'sample_id'</span>, <span class="st">'source'</span>, <span class="st">'date'</span>, <span class="st">'sample_mode'</span>, <span class="st">'Kingdom'</span>, <span class="st">'month'</span>])</span>
<span id="cb66-5"><a aria-hidden="true" href="#cb66-5" tabindex="-1"></a> .replace({<span class="st">'chao'</span>: <span class="st">'Chao1 Index'</span>, <span class="st">'shannon'</span>: <span class="st">'Shannon Index'</span>})</span>
<span id="cb66-6"><a aria-hidden="true" href="#cb66-6" tabindex="-1"></a> .assign(month<span class="op">=</span><span class="kw">lambda</span> dd: pd.Categorical(dd.month, categories<span class="op">=</span>[<span class="st">'February'</span>, <span class="st">'April'</span>]))</span>
<span id="cb66-7"><a aria-hidden="true" href="#cb66-7" tabindex="-1"></a>.pipe(<span class="kw">lambda</span> dd: p9.ggplot(dd)</span>
<span id="cb66-8"><a aria-hidden="true" href="#cb66-8" tabindex="-1"></a>      <span class="op">+</span> p9.aes(<span class="st">'month'</span>, <span class="st">'value'</span>)</span>
<span id="cb66-9"><a aria-hidden="true" href="#cb66-9" tabindex="-1"></a>      <span class="op">+</span> p9.geom_jitter(p9.aes(fill<span class="op">=</span><span class="st">'sample_mode'</span>), alpha<span class="op">=</span><span class="fl">.6</span>, width<span class="op">=</span><span class="dv">0</span>, height<span class="op">=</span><span class="dv">0</span>, stroke<span class="op">=</span><span class="dv">0</span>, size<span class="op">=</span><span class="dv">2</span>)</span>
<span id="cb66-10"><a aria-hidden="true" href="#cb66-10" tabindex="-1"></a>      <span class="op">+</span> p9.geom_boxplot(alpha<span class="op">=</span><span class="fl">.6</span>, outlier_alpha<span class="op">=</span><span class="dv">0</span>)</span>
<span id="cb66-11"><a aria-hidden="true" href="#cb66-11" tabindex="-1"></a>      <span class="op">+</span> p9.geom_blank(p9.aes(y<span class="op">=</span><span class="st">'value * 1.1'</span>))</span>
<span id="cb66-12"><a aria-hidden="true" href="#cb66-12" tabindex="-1"></a>      <span class="op">+</span> p9.facet_grid([<span class="st">'variable'</span>, <span class="st">'Kingdom'</span>], scales<span class="op">=</span><span class="st">'free_y'</span>)</span>
<span id="cb66-13"><a aria-hidden="true" href="#cb66-13" tabindex="-1"></a>      <span class="op">+</span> p9.scale_fill_manual(values <span class="op">=</span> [<span class="st">'#D3894C'</span>, <span class="st">'#435B97'</span>] )</span>
<span id="cb66-14"><a aria-hidden="true" href="#cb66-14" tabindex="-1"></a>      <span class="op">+</span> p9.geom_text(p9.aes(label<span class="op">=</span><span class="st">'n'</span>), y<span class="op">=-</span><span class="fl">.2</span>, size<span class="op">=</span><span class="dv">8</span>, data<span class="op">=</span>sample_sizes_month)</span>
<span id="cb66-15"><a aria-hidden="true" href="#cb66-15" tabindex="-1"></a>      <span class="op">+</span> p9.guides(fill<span class="op">=</span><span class="va">False</span>)</span>
<span id="cb66-16"><a aria-hidden="true" href="#cb66-16" tabindex="-1"></a>      <span class="op">+</span> p9.labs(x<span class="op">=</span><span class="st">''</span>, y<span class="op">=</span><span class="vs">r'$\alpha$-Diversity '</span>)</span>
<span id="cb66-17"><a aria-hidden="true" href="#cb66-17" tabindex="-1"></a>      <span class="op">+</span> p9.scale_y_continuous(expand<span class="op">=</span>(<span class="fl">.175</span>, <span class="fl">.1</span>))</span>
<span id="cb66-18"><a aria-hidden="true" href="#cb66-18" tabindex="-1"></a>      <span class="op">+</span> p9.theme(figure_size<span class="op">=</span>(<span class="dv">4</span>, <span class="dv">3</span>)</span>
<span id="cb66-19"><a aria-hidden="true" href="#cb66-19" tabindex="-1"></a>    )</span>
<span id="cb66-20"><a aria-hidden="true" href="#cb66-20" tabindex="-1"></a>)</span>
<span id="cb66-21"><a aria-hidden="true" href="#cb66-21" tabindex="-1"></a>)</span>
<span id="cb66-22"><a aria-hidden="true" href="#cb66-22" tabindex="-1"></a>f.save(<span class="st">'../output/figures/boxplots_diversity_by_month.svg'</span>)</span>
<span id="cb66-23"><a aria-hidden="true" href="#cb66-23" tabindex="-1"></a>f.draw()</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="56">
<div>
<figure class="figure">
<p><img class="figure-img" height="900" src="microbial_diversity_analysis_files/figure-html/cell-65-output-1.png" width="1200"/></p>
</figure>
</div>
</div>
</div>
</section>
<section class="level3" id="distance-between-samples-nmds-and-permanova">
<h3 class="anchored" data-anchor-id="distance-between-samples-nmds-and-permanova">Distance between samples (nMDS and PERMANOVA)</h3>
<div class="cell" data-execution_count="57" id="cell-94">
<div class="sourceCode cell-code" id="cb67"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb67-1"><a aria-hidden="true" href="#cb67-1" tabindex="-1"></a>samples <span class="op">=</span> samples_info.query(<span class="st">'sample_id not in ["S06", "S14"]'</span>).sample_id</span>
<span id="cb67-2"><a aria-hidden="true" href="#cb67-2" tabindex="-1"></a>dm <span class="op">=</span> (abundance_df_genera</span>
<span id="cb67-3"><a aria-hidden="true" href="#cb67-3" tabindex="-1"></a> .query(<span class="st">'Genus!="unknown"'</span>)</span>
<span id="cb67-4"><a aria-hidden="true" href="#cb67-4" tabindex="-1"></a> .drop(columns<span class="op">=</span>[<span class="st">'S06'</span>] <span class="op">+</span> taxon_names[:<span class="op">-</span><span class="dv">1</span>])</span>
<span id="cb67-5"><a aria-hidden="true" href="#cb67-5" tabindex="-1"></a> .set_index(<span class="st">'Genus'</span>)</span>
<span id="cb67-6"><a aria-hidden="true" href="#cb67-6" tabindex="-1"></a> .T</span>
<span id="cb67-7"><a aria-hidden="true" href="#cb67-7" tabindex="-1"></a> .<span class="bu">apply</span>(<span class="kw">lambda</span> x: x <span class="op">/</span> x.<span class="bu">sum</span>(), axis<span class="op">=</span><span class="dv">1</span>)</span>
<span id="cb67-8"><a aria-hidden="true" href="#cb67-8" tabindex="-1"></a> .pipe(<span class="kw">lambda</span> dd: pairwise_distances(dd, metric<span class="op">=</span><span class="st">'braycurtis'</span>))</span>
<span id="cb67-9"><a aria-hidden="true" href="#cb67-9" tabindex="-1"></a>)</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
</div>
<section class="level4" id="non-metric-mds">
<h4 class="anchored" data-anchor-id="non-metric-mds">non-metric MDS</h4>
<div class="cell" data-execution_count="58" id="cell-96">
<div class="sourceCode cell-code" id="cb68"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb68-1"><a aria-hidden="true" href="#cb68-1" tabindex="-1"></a>np.random.seed(<span class="dv">105</span>)</span>
<span id="cb68-2"><a aria-hidden="true" href="#cb68-2" tabindex="-1"></a>best_stress <span class="op">=</span> np.inf</span>
<span id="cb68-3"><a aria-hidden="true" href="#cb68-3" tabindex="-1"></a>best_npos <span class="op">=</span> <span class="va">None</span></span>
<span id="cb68-4"><a aria-hidden="true" href="#cb68-4" tabindex="-1"></a>stresses <span class="op">=</span> []</span>
<span id="cb68-5"><a aria-hidden="true" href="#cb68-5" tabindex="-1"></a><span class="cf">for</span> _ <span class="kw">in</span> <span class="bu">range</span>(<span class="dv">1000</span>):</span>
<span id="cb68-6"><a aria-hidden="true" href="#cb68-6" tabindex="-1"></a>    nmds <span class="op">=</span> MDS(n_components<span class="op">=</span><span class="dv">2</span>,</span>
<span id="cb68-7"><a aria-hidden="true" href="#cb68-7" tabindex="-1"></a>            max_iter<span class="op">=</span><span class="dv">900</span>, </span>
<span id="cb68-8"><a aria-hidden="true" href="#cb68-8" tabindex="-1"></a>            eps<span class="op">=</span><span class="fl">1e-9</span>,</span>
<span id="cb68-9"><a aria-hidden="true" href="#cb68-9" tabindex="-1"></a>            dissimilarity<span class="op">=</span><span class="st">"precomputed"</span>,</span>
<span id="cb68-10"><a aria-hidden="true" href="#cb68-10" tabindex="-1"></a>            normalized_stress<span class="op">=</span><span class="va">True</span>,</span>
<span id="cb68-11"><a aria-hidden="true" href="#cb68-11" tabindex="-1"></a>            metric<span class="op">=</span><span class="va">False</span>)</span>
<span id="cb68-12"><a aria-hidden="true" href="#cb68-12" tabindex="-1"></a></span>
<span id="cb68-13"><a aria-hidden="true" href="#cb68-13" tabindex="-1"></a>    npos <span class="op">=</span> nmds.fit_transform(dm)</span>
<span id="cb68-14"><a aria-hidden="true" href="#cb68-14" tabindex="-1"></a></span>
<span id="cb68-15"><a aria-hidden="true" href="#cb68-15" tabindex="-1"></a>    stress <span class="op">=</span> nmds.stress_</span>
<span id="cb68-16"><a aria-hidden="true" href="#cb68-16" tabindex="-1"></a>    stresses.append(stress)</span>
<span id="cb68-17"><a aria-hidden="true" href="#cb68-17" tabindex="-1"></a>    <span class="cf">if</span> stress <span class="op">&lt;</span> best_stress:</span>
<span id="cb68-18"><a aria-hidden="true" href="#cb68-18" tabindex="-1"></a>        best_stress <span class="op">=</span> stress</span>
<span id="cb68-19"><a aria-hidden="true" href="#cb68-19" tabindex="-1"></a>        best_npos <span class="op">=</span> npos</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="59" id="cell-97">
<div class="sourceCode cell-code" id="cb69"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb69-1"><a aria-hidden="true" href="#cb69-1" tabindex="-1"></a>(p9.ggplot(pd.DataFrame(stresses, columns<span class="op">=</span>[<span class="st">'stress'</span>])) </span>
<span id="cb69-2"><a aria-hidden="true" href="#cb69-2" tabindex="-1"></a> <span class="op">+</span> p9.aes(<span class="st">'stress'</span>)</span>
<span id="cb69-3"><a aria-hidden="true" href="#cb69-3" tabindex="-1"></a> <span class="op">+</span> p9.geom_histogram(bins<span class="op">=</span><span class="dv">40</span>)</span>
<span id="cb69-4"><a aria-hidden="true" href="#cb69-4" tabindex="-1"></a> <span class="op">+</span> p9.theme(figure_size<span class="op">=</span>(<span class="dv">4</span>, <span class="dv">3</span>))</span>
<span id="cb69-5"><a aria-hidden="true" href="#cb69-5" tabindex="-1"></a>)</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img class="figure-img" height="900" src="microbial_diversity_analysis_files/figure-html/cell-68-output-1.png" width="1200"/></p>
</figure>
</div>
</div>
</div>
<div class="cell" data-execution_count="60" id="cell-98">
<div class="sourceCode cell-code" id="cb70"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb70-1"><a aria-hidden="true" href="#cb70-1" tabindex="-1"></a></span>
<span id="cb70-2"><a aria-hidden="true" href="#cb70-2" tabindex="-1"></a>f <span class="op">=</span> (pd.DataFrame(best_npos)</span>
<span id="cb70-3"><a aria-hidden="true" href="#cb70-3" tabindex="-1"></a> .assign(sample_id<span class="op">=</span>samples)</span>
<span id="cb70-4"><a aria-hidden="true" href="#cb70-4" tabindex="-1"></a> .merge(samples_info)</span>
<span id="cb70-5"><a aria-hidden="true" href="#cb70-5" tabindex="-1"></a> .assign(month<span class="op">=</span><span class="kw">lambda</span> dd: pd.to_datetime(dd.date).dt.month_name())</span>
<span id="cb70-6"><a aria-hidden="true" href="#cb70-6" tabindex="-1"></a> .rename(columns<span class="op">=</span>{<span class="dv">0</span>: <span class="st">'x'</span>, <span class="dv">1</span>: <span class="st">'y'</span>})</span>
<span id="cb70-7"><a aria-hidden="true" href="#cb70-7" tabindex="-1"></a> .pipe(<span class="kw">lambda</span> dd: p9.ggplot(dd)</span>
<span id="cb70-8"><a aria-hidden="true" href="#cb70-8" tabindex="-1"></a>       <span class="op">+</span> p9.aes(<span class="st">'x'</span>, <span class="st">'y'</span>, color<span class="op">=</span><span class="st">'sample_mode'</span>, shape<span class="op">=</span><span class="st">'month'</span>)</span>
<span id="cb70-9"><a aria-hidden="true" href="#cb70-9" tabindex="-1"></a>       <span class="op">+</span> p9.geom_point(size<span class="op">=</span><span class="fl">2.5</span>)</span>
<span id="cb70-10"><a aria-hidden="true" href="#cb70-10" tabindex="-1"></a>       <span class="op">+</span> p9.stat_ellipse(p9.aes(group<span class="op">=</span><span class="st">'sample_mode'</span>, fill<span class="op">=</span><span class="st">'sample_mode'</span>),</span>
<span id="cb70-11"><a aria-hidden="true" href="#cb70-11" tabindex="-1"></a>                         geom<span class="op">=</span><span class="st">'polygon'</span>, level<span class="op">=</span><span class="fl">.85</span>, alpha<span class="op">=</span><span class="fl">.25</span>)</span>
<span id="cb70-12"><a aria-hidden="true" href="#cb70-12" tabindex="-1"></a>       <span class="op">+</span> p9.scale_color_manual(values<span class="op">=</span>[<span class="st">'#D3894C'</span>, <span class="st">'#435B97'</span>])</span>
<span id="cb70-13"><a aria-hidden="true" href="#cb70-13" tabindex="-1"></a>       <span class="op">+</span> p9.scale_fill_manual(values<span class="op">=</span>[<span class="st">'#D3894C'</span>, <span class="st">'#435B97'</span>])</span>
<span id="cb70-14"><a aria-hidden="true" href="#cb70-14" tabindex="-1"></a>       <span class="op">+</span> p9.ylim(<span class="op">-</span><span class="fl">1.2</span>, <span class="dv">1</span>)</span>
<span id="cb70-15"><a aria-hidden="true" href="#cb70-15" tabindex="-1"></a>       <span class="op">+</span> p9.guides(fill<span class="op">=</span><span class="va">False</span>)</span>
<span id="cb70-16"><a aria-hidden="true" href="#cb70-16" tabindex="-1"></a>       <span class="op">+</span> p9.labs(x<span class="op">=</span><span class="st">'nMDS Dim 1'</span>, y<span class="op">=</span><span class="st">'nMDS Dim 2'</span>, color<span class="op">=</span><span class="st">''</span>, shape<span class="op">=</span><span class="st">''</span>)</span>
<span id="cb70-17"><a aria-hidden="true" href="#cb70-17" tabindex="-1"></a>       <span class="op">+</span> p9.annotate(geom<span class="op">=</span><span class="st">'text'</span>, x<span class="op">=-</span><span class="fl">.6</span>, y<span class="op">=</span><span class="fl">.95</span>, ha<span class="op">=</span><span class="st">'left'</span>,</span>
<span id="cb70-18"><a aria-hidden="true" href="#cb70-18" tabindex="-1"></a>                     label<span class="op">=</span><span class="st">'Stress-1 = '</span> <span class="op">+</span> <span class="bu">str</span>(<span class="bu">round</span>(best_stress, <span class="dv">2</span>)),</span>
<span id="cb70-19"><a aria-hidden="true" href="#cb70-19" tabindex="-1"></a>                     size<span class="op">=</span><span class="dv">9</span>)</span>
<span id="cb70-20"><a aria-hidden="true" href="#cb70-20" tabindex="-1"></a>       <span class="op">+</span> p9.theme(figure_size<span class="op">=</span>(<span class="dv">3</span>, <span class="dv">3</span>),</span>
<span id="cb70-21"><a aria-hidden="true" href="#cb70-21" tabindex="-1"></a>                  legend_position<span class="op">=</span><span class="st">'top'</span>,</span>
<span id="cb70-22"><a aria-hidden="true" href="#cb70-22" tabindex="-1"></a>                  legend_text<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">8</span>),</span>
<span id="cb70-23"><a aria-hidden="true" href="#cb70-23" tabindex="-1"></a>                  legend_key_size<span class="op">=</span><span class="dv">8</span>,</span>
<span id="cb70-24"><a aria-hidden="true" href="#cb70-24" tabindex="-1"></a>                  )</span>
<span id="cb70-25"><a aria-hidden="true" href="#cb70-25" tabindex="-1"></a>       )</span>
<span id="cb70-26"><a aria-hidden="true" href="#cb70-26" tabindex="-1"></a>)</span>
<span id="cb70-27"><a aria-hidden="true" href="#cb70-27" tabindex="-1"></a></span>
<span id="cb70-28"><a aria-hidden="true" href="#cb70-28" tabindex="-1"></a>f.save(<span class="st">'../output/figures/nmds_fg.svg'</span>)</span>
<span id="cb70-29"><a aria-hidden="true" href="#cb70-29" tabindex="-1"></a>f.draw()</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="60">
<div>
<figure class="figure">
<p><img class="figure-img" height="900" src="microbial_diversity_analysis_files/figure-html/cell-69-output-1.png" width="900"/></p>
</figure>
</div>
</div>
</div>
<div class="cell" data-execution_count="61" id="cell-99">
<div class="sourceCode cell-code" id="cb71"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb71-1"><a aria-hidden="true" href="#cb71-1" tabindex="-1"></a></span>
<span id="cb71-2"><a aria-hidden="true" href="#cb71-2" tabindex="-1"></a>f <span class="op">=</span> (pd.DataFrame(best_npos)</span>
<span id="cb71-3"><a aria-hidden="true" href="#cb71-3" tabindex="-1"></a> .assign(sample_id<span class="op">=</span>samples)</span>
<span id="cb71-4"><a aria-hidden="true" href="#cb71-4" tabindex="-1"></a> .merge(samples_info)</span>
<span id="cb71-5"><a aria-hidden="true" href="#cb71-5" tabindex="-1"></a> .assign(month<span class="op">=</span><span class="kw">lambda</span> dd: pd.to_datetime(dd.date).dt.month_name())</span>
<span id="cb71-6"><a aria-hidden="true" href="#cb71-6" tabindex="-1"></a> .rename(columns<span class="op">=</span>{<span class="dv">0</span>: <span class="st">'x'</span>, <span class="dv">1</span>: <span class="st">'y'</span>})</span>
<span id="cb71-7"><a aria-hidden="true" href="#cb71-7" tabindex="-1"></a> .pipe(<span class="kw">lambda</span> dd: p9.ggplot(dd)</span>
<span id="cb71-8"><a aria-hidden="true" href="#cb71-8" tabindex="-1"></a>       <span class="op">+</span> p9.aes(<span class="st">'x'</span>, <span class="st">'y'</span>, color<span class="op">=</span><span class="st">'sample_mode'</span>, shape<span class="op">=</span><span class="st">'month'</span>)</span>
<span id="cb71-9"><a aria-hidden="true" href="#cb71-9" tabindex="-1"></a>       <span class="op">+</span> p9.geom_point(size<span class="op">=</span><span class="fl">2.5</span>)</span>
<span id="cb71-10"><a aria-hidden="true" href="#cb71-10" tabindex="-1"></a>       <span class="op">+</span> p9.stat_ellipse(p9.aes(group<span class="op">=</span><span class="st">'month'</span>),</span>
<span id="cb71-11"><a aria-hidden="true" href="#cb71-11" tabindex="-1"></a>                         geom<span class="op">=</span><span class="st">'polygon'</span>, level<span class="op">=</span><span class="fl">.85</span>, alpha<span class="op">=</span><span class="fl">.25</span>)</span>
<span id="cb71-12"><a aria-hidden="true" href="#cb71-12" tabindex="-1"></a>       <span class="op">+</span> p9.scale_color_manual(values<span class="op">=</span>[<span class="st">'#D3894C'</span>, <span class="st">'#435B97'</span>])</span>
<span id="cb71-13"><a aria-hidden="true" href="#cb71-13" tabindex="-1"></a>       <span class="op">+</span> p9.scale_fill_manual(values<span class="op">=</span>[<span class="st">'#D3894C'</span>, <span class="st">'#435B97'</span>])</span>
<span id="cb71-14"><a aria-hidden="true" href="#cb71-14" tabindex="-1"></a>       <span class="op">+</span> p9.ylim(<span class="op">-</span><span class="fl">1.2</span>, <span class="dv">1</span>)</span>
<span id="cb71-15"><a aria-hidden="true" href="#cb71-15" tabindex="-1"></a>       <span class="op">+</span> p9.guides(fill<span class="op">=</span><span class="va">False</span>)</span>
<span id="cb71-16"><a aria-hidden="true" href="#cb71-16" tabindex="-1"></a>       <span class="op">+</span> p9.labs(x<span class="op">=</span><span class="st">'nMDS Dim 1'</span>, y<span class="op">=</span><span class="st">'nMDS Dim 2'</span>, color<span class="op">=</span><span class="st">''</span>, shape<span class="op">=</span><span class="st">''</span>)</span>
<span id="cb71-17"><a aria-hidden="true" href="#cb71-17" tabindex="-1"></a>       <span class="op">+</span> p9.annotate(geom<span class="op">=</span><span class="st">'text'</span>, x<span class="op">=-</span><span class="fl">.6</span>, y<span class="op">=</span><span class="fl">.95</span>, ha<span class="op">=</span><span class="st">'left'</span>,</span>
<span id="cb71-18"><a aria-hidden="true" href="#cb71-18" tabindex="-1"></a>                     label<span class="op">=</span><span class="st">'Stress-1 = '</span> <span class="op">+</span> <span class="bu">str</span>(<span class="bu">round</span>(best_stress, <span class="dv">2</span>)),</span>
<span id="cb71-19"><a aria-hidden="true" href="#cb71-19" tabindex="-1"></a>                     size<span class="op">=</span><span class="dv">9</span>)</span>
<span id="cb71-20"><a aria-hidden="true" href="#cb71-20" tabindex="-1"></a>       <span class="op">+</span> p9.theme(figure_size<span class="op">=</span>(<span class="dv">3</span>, <span class="dv">3</span>),</span>
<span id="cb71-21"><a aria-hidden="true" href="#cb71-21" tabindex="-1"></a>                  legend_position<span class="op">=</span><span class="st">'top'</span>,</span>
<span id="cb71-22"><a aria-hidden="true" href="#cb71-22" tabindex="-1"></a>                  legend_text<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">8</span>),</span>
<span id="cb71-23"><a aria-hidden="true" href="#cb71-23" tabindex="-1"></a>                  legend_key_size<span class="op">=</span><span class="dv">8</span>,</span>
<span id="cb71-24"><a aria-hidden="true" href="#cb71-24" tabindex="-1"></a>                  )</span>
<span id="cb71-25"><a aria-hidden="true" href="#cb71-25" tabindex="-1"></a>       )</span>
<span id="cb71-26"><a aria-hidden="true" href="#cb71-26" tabindex="-1"></a>)</span>
<span id="cb71-27"><a aria-hidden="true" href="#cb71-27" tabindex="-1"></a></span>
<span id="cb71-28"><a aria-hidden="true" href="#cb71-28" tabindex="-1"></a>f.save(<span class="st">'../output/figures/nmds_af.svg'</span>)</span>
<span id="cb71-29"><a aria-hidden="true" href="#cb71-29" tabindex="-1"></a>f.draw()</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="61">
<div>
<figure class="figure">
<p><img class="figure-img" height="900" src="microbial_diversity_analysis_files/figure-html/cell-70-output-1.png" width="900"/></p>
</figure>
</div>
</div>
</div>
<div class="cell" data-execution_count="62" id="cell-100">
<div class="sourceCode cell-code" id="cb72"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb72-1"><a aria-hidden="true" href="#cb72-1" tabindex="-1"></a></span>
<span id="cb72-2"><a aria-hidden="true" href="#cb72-2" tabindex="-1"></a>(pd.DataFrame(best_npos)</span>
<span id="cb72-3"><a aria-hidden="true" href="#cb72-3" tabindex="-1"></a> .assign(sample_id<span class="op">=</span>samples)</span>
<span id="cb72-4"><a aria-hidden="true" href="#cb72-4" tabindex="-1"></a> .merge(samples_info)</span>
<span id="cb72-5"><a aria-hidden="true" href="#cb72-5" tabindex="-1"></a> .assign(month<span class="op">=</span><span class="kw">lambda</span> dd: pd.to_datetime(dd.date).dt.month_name())</span>
<span id="cb72-6"><a aria-hidden="true" href="#cb72-6" tabindex="-1"></a> .rename(columns<span class="op">=</span>{<span class="dv">0</span>: <span class="st">'x'</span>, <span class="dv">1</span>: <span class="st">'y'</span>})</span>
<span id="cb72-7"><a aria-hidden="true" href="#cb72-7" tabindex="-1"></a> .pipe(<span class="kw">lambda</span> dd: p9.ggplot(dd)</span>
<span id="cb72-8"><a aria-hidden="true" href="#cb72-8" tabindex="-1"></a>       <span class="op">+</span> p9.aes(<span class="st">'x'</span>, <span class="st">'y'</span>, color<span class="op">=</span><span class="st">'sample_mode'</span>)</span>
<span id="cb72-9"><a aria-hidden="true" href="#cb72-9" tabindex="-1"></a>       <span class="op">+</span> p9.geom_point(size<span class="op">=</span><span class="fl">2.5</span>)</span>
<span id="cb72-10"><a aria-hidden="true" href="#cb72-10" tabindex="-1"></a>       <span class="op">+</span> p9.scale_color_manual(values<span class="op">=</span>[<span class="st">'#D3894C'</span>, <span class="st">'#435B97'</span>])</span>
<span id="cb72-11"><a aria-hidden="true" href="#cb72-11" tabindex="-1"></a>       <span class="op">+</span> p9.scale_fill_manual(values<span class="op">=</span>[<span class="st">'#D3894C'</span>, <span class="st">'#435B97'</span>])</span>
<span id="cb72-12"><a aria-hidden="true" href="#cb72-12" tabindex="-1"></a>       <span class="op">+</span> p9.stat_ellipse(p9.aes(fill<span class="op">=</span><span class="st">'sample_mode'</span>), geom<span class="op">=</span><span class="st">'polygon'</span>, level<span class="op">=</span><span class="fl">.95</span>, alpha<span class="op">=</span><span class="fl">.15</span>)</span>
<span id="cb72-13"><a aria-hidden="true" href="#cb72-13" tabindex="-1"></a>       <span class="op">+</span> p9.labs(x<span class="op">=</span><span class="st">'nMDS Dim 1'</span>, y<span class="op">=</span><span class="st">'nMDS Dim 2'</span>, color<span class="op">=</span><span class="st">''</span>, shape<span class="op">=</span><span class="st">''</span>, fill<span class="op">=</span><span class="st">''</span>)</span>
<span id="cb72-14"><a aria-hidden="true" href="#cb72-14" tabindex="-1"></a>       <span class="op">+</span> p9.annotate(geom<span class="op">=</span><span class="st">'text'</span>, x<span class="op">=-</span><span class="fl">.1</span>, y<span class="op">=</span><span class="fl">1.2</span>,</span>
<span id="cb72-15"><a aria-hidden="true" href="#cb72-15" tabindex="-1"></a>                     label<span class="op">=</span><span class="st">'Stress-1 = '</span> <span class="op">+</span> <span class="bu">str</span>(<span class="bu">round</span>(best_stress, <span class="dv">2</span>)),</span>
<span id="cb72-16"><a aria-hidden="true" href="#cb72-16" tabindex="-1"></a>                     size<span class="op">=</span><span class="dv">9</span>)</span>
<span id="cb72-17"><a aria-hidden="true" href="#cb72-17" tabindex="-1"></a>       <span class="op">+</span> p9.theme(figure_size<span class="op">=</span>(<span class="dv">5</span>, <span class="dv">4</span>),</span>
<span id="cb72-18"><a aria-hidden="true" href="#cb72-18" tabindex="-1"></a>                  legend_position<span class="op">=</span><span class="st">'top'</span>,</span>
<span id="cb72-19"><a aria-hidden="true" href="#cb72-19" tabindex="-1"></a>                  legend_text<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">8</span>),</span>
<span id="cb72-20"><a aria-hidden="true" href="#cb72-20" tabindex="-1"></a>                  legend_key_size<span class="op">=</span><span class="dv">8</span>,</span>
<span id="cb72-21"><a aria-hidden="true" href="#cb72-21" tabindex="-1"></a>                  )</span>
<span id="cb72-22"><a aria-hidden="true" href="#cb72-22" tabindex="-1"></a>       )</span>
<span id="cb72-23"><a aria-hidden="true" href="#cb72-23" tabindex="-1"></a>)</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img class="figure-img" height="1200" src="microbial_diversity_analysis_files/figure-html/cell-71-output-1.png" width="1500"/></p>
</figure>
</div>
</div>
</div>
<div class="cell" data-execution_count="63" id="cell-101">
<div class="sourceCode cell-code" id="cb73"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb73-1"><a aria-hidden="true" href="#cb73-1" tabindex="-1"></a>(pd.DataFrame(best_npos)</span>
<span id="cb73-2"><a aria-hidden="true" href="#cb73-2" tabindex="-1"></a> .assign(sample_id<span class="op">=</span>samples)</span>
<span id="cb73-3"><a aria-hidden="true" href="#cb73-3" tabindex="-1"></a> .merge(samples_info)</span>
<span id="cb73-4"><a aria-hidden="true" href="#cb73-4" tabindex="-1"></a> .assign(month<span class="op">=</span><span class="kw">lambda</span> dd: pd.to_datetime(dd.date).dt.month_name())</span>
<span id="cb73-5"><a aria-hidden="true" href="#cb73-5" tabindex="-1"></a> .rename(columns<span class="op">=</span>{<span class="dv">0</span>: <span class="st">'x'</span>, <span class="dv">1</span>: <span class="st">'y'</span>})</span>
<span id="cb73-6"><a aria-hidden="true" href="#cb73-6" tabindex="-1"></a> .pipe(<span class="kw">lambda</span> dd: p9.ggplot(dd)</span>
<span id="cb73-7"><a aria-hidden="true" href="#cb73-7" tabindex="-1"></a>       <span class="op">+</span> p9.aes(<span class="st">'x'</span>, <span class="st">'y'</span>, color<span class="op">=</span><span class="st">'sample_mode'</span>)</span>
<span id="cb73-8"><a aria-hidden="true" href="#cb73-8" tabindex="-1"></a>       <span class="op">+</span> p9.geom_point(size<span class="op">=</span><span class="fl">2.5</span>)</span>
<span id="cb73-9"><a aria-hidden="true" href="#cb73-9" tabindex="-1"></a>       <span class="op">+</span> p9.scale_color_manual(values<span class="op">=</span>[<span class="st">'#D3894C'</span>, <span class="st">'#435B97'</span>])</span>
<span id="cb73-10"><a aria-hidden="true" href="#cb73-10" tabindex="-1"></a>       <span class="op">+</span> p9.scale_fill_manual(values<span class="op">=</span>[<span class="st">'#D3894C'</span>, <span class="st">'#435B97'</span>])</span>
<span id="cb73-11"><a aria-hidden="true" href="#cb73-11" tabindex="-1"></a>       <span class="op">+</span> p9.stat_ellipse(p9.aes(fill<span class="op">=</span><span class="st">'sample_mode'</span>), geom<span class="op">=</span><span class="st">'polygon'</span>, level<span class="op">=</span><span class="fl">.95</span>, alpha<span class="op">=</span><span class="fl">.15</span>)</span>
<span id="cb73-12"><a aria-hidden="true" href="#cb73-12" tabindex="-1"></a>       <span class="op">+</span> p9.labs(x<span class="op">=</span><span class="st">'nMDS Dim 1'</span>, y<span class="op">=</span><span class="st">'nMDS Dim 2'</span>, color<span class="op">=</span><span class="st">''</span>, shape<span class="op">=</span><span class="st">''</span>, fill<span class="op">=</span><span class="st">''</span>)</span>
<span id="cb73-13"><a aria-hidden="true" href="#cb73-13" tabindex="-1"></a>       <span class="op">+</span> p9.annotate(geom<span class="op">=</span><span class="st">'text'</span>, x<span class="op">=-</span><span class="dv">0</span>, y<span class="op">=</span><span class="fl">1.5</span>,</span>
<span id="cb73-14"><a aria-hidden="true" href="#cb73-14" tabindex="-1"></a>                     label<span class="op">=</span><span class="st">'Stress-1 = '</span> <span class="op">+</span> <span class="bu">str</span>(<span class="bu">round</span>(best_stress, <span class="dv">2</span>)) <span class="op">+</span></span>
<span id="cb73-15"><a aria-hidden="true" href="#cb73-15" tabindex="-1"></a>                     <span class="st">'    PERMANOVA pseudo-$F$ = 0.68, $p$ = 0.59'</span>,</span>
<span id="cb73-16"><a aria-hidden="true" href="#cb73-16" tabindex="-1"></a>                     size<span class="op">=</span><span class="dv">9</span>)</span>
<span id="cb73-17"><a aria-hidden="true" href="#cb73-17" tabindex="-1"></a>       <span class="op">+</span> p9.scale_y_continuous(expand<span class="op">=</span>(<span class="fl">.02</span>, <span class="fl">.2</span>))</span>
<span id="cb73-18"><a aria-hidden="true" href="#cb73-18" tabindex="-1"></a>       <span class="op">+</span> p9.theme(figure_size<span class="op">=</span>(<span class="dv">5</span>, <span class="dv">4</span>),</span>
<span id="cb73-19"><a aria-hidden="true" href="#cb73-19" tabindex="-1"></a>                  legend_position<span class="op">=</span><span class="st">'top'</span>,</span>
<span id="cb73-20"><a aria-hidden="true" href="#cb73-20" tabindex="-1"></a>                  legend_text<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">8</span>),</span>
<span id="cb73-21"><a aria-hidden="true" href="#cb73-21" tabindex="-1"></a>                  legend_key_size<span class="op">=</span><span class="dv">8</span>,</span>
<span id="cb73-22"><a aria-hidden="true" href="#cb73-22" tabindex="-1"></a>                  )</span>
<span id="cb73-23"><a aria-hidden="true" href="#cb73-23" tabindex="-1"></a>       )</span>
<span id="cb73-24"><a aria-hidden="true" href="#cb73-24" tabindex="-1"></a>)</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img class="figure-img" height="1200" src="microbial_diversity_analysis_files/figure-html/cell-72-output-1.png" width="1500"/></p>
</figure>
</div>
</div>
</div>
<div class="cell" data-execution_count="64" id="cell-102">
<div class="sourceCode cell-code" id="cb74"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb74-1"><a aria-hidden="true" href="#cb74-1" tabindex="-1"></a></span>
<span id="cb74-2"><a aria-hidden="true" href="#cb74-2" tabindex="-1"></a>(pd.DataFrame(best_npos)</span>
<span id="cb74-3"><a aria-hidden="true" href="#cb74-3" tabindex="-1"></a> .assign(sample_id<span class="op">=</span>samples)</span>
<span id="cb74-4"><a aria-hidden="true" href="#cb74-4" tabindex="-1"></a> .merge(samples_info)</span>
<span id="cb74-5"><a aria-hidden="true" href="#cb74-5" tabindex="-1"></a> .assign(month<span class="op">=</span><span class="kw">lambda</span> dd: pd.to_datetime(dd.date).dt.month_name())</span>
<span id="cb74-6"><a aria-hidden="true" href="#cb74-6" tabindex="-1"></a> .rename(columns<span class="op">=</span>{<span class="dv">0</span>: <span class="st">'x'</span>, <span class="dv">1</span>: <span class="st">'y'</span>})</span>
<span id="cb74-7"><a aria-hidden="true" href="#cb74-7" tabindex="-1"></a> .pipe(<span class="kw">lambda</span> dd: p9.ggplot(dd)</span>
<span id="cb74-8"><a aria-hidden="true" href="#cb74-8" tabindex="-1"></a>       <span class="op">+</span> p9.aes(<span class="st">'x'</span>, <span class="st">'y'</span>, color<span class="op">=</span><span class="st">'month'</span>)</span>
<span id="cb74-9"><a aria-hidden="true" href="#cb74-9" tabindex="-1"></a>       <span class="op">+</span> p9.geom_point(size<span class="op">=</span><span class="fl">2.5</span>)</span>
<span id="cb74-10"><a aria-hidden="true" href="#cb74-10" tabindex="-1"></a>       <span class="op">+</span> p9.scale_color_manual(values<span class="op">=</span>[<span class="st">'#D3894C'</span>, <span class="st">'#435B97'</span>])</span>
<span id="cb74-11"><a aria-hidden="true" href="#cb74-11" tabindex="-1"></a>       <span class="op">+</span> p9.scale_fill_manual(values<span class="op">=</span>[<span class="st">'#D3894C'</span>, <span class="st">'#435B97'</span>])</span>
<span id="cb74-12"><a aria-hidden="true" href="#cb74-12" tabindex="-1"></a>       <span class="op">+</span> p9.stat_ellipse(p9.aes(fill<span class="op">=</span><span class="st">'month'</span>), geom<span class="op">=</span><span class="st">'polygon'</span>, level<span class="op">=</span><span class="fl">.95</span>, alpha<span class="op">=</span><span class="fl">.15</span>)</span>
<span id="cb74-13"><a aria-hidden="true" href="#cb74-13" tabindex="-1"></a>       <span class="op">+</span> p9.labs(x<span class="op">=</span><span class="st">'nMDS Dim 1'</span>, y<span class="op">=</span><span class="st">'nMDS Dim 2'</span>, color<span class="op">=</span><span class="st">''</span>, shape<span class="op">=</span><span class="st">''</span>, fill<span class="op">=</span><span class="st">''</span>)</span>
<span id="cb74-14"><a aria-hidden="true" href="#cb74-14" tabindex="-1"></a>       <span class="op">+</span> p9.annotate(geom<span class="op">=</span><span class="st">'text'</span>, x<span class="op">=-</span><span class="dv">0</span>, y<span class="op">=</span><span class="fl">1.5</span>,</span>
<span id="cb74-15"><a aria-hidden="true" href="#cb74-15" tabindex="-1"></a>                     label<span class="op">=</span><span class="st">'Stress-1 = '</span> <span class="op">+</span> <span class="bu">str</span>(<span class="bu">round</span>(best_stress, <span class="dv">2</span>)) <span class="op">+</span></span>
<span id="cb74-16"><a aria-hidden="true" href="#cb74-16" tabindex="-1"></a>                     <span class="st">'    PERMANOVA pseudo-$F$ = 4.0, $p$ = 0.017'</span>,</span>
<span id="cb74-17"><a aria-hidden="true" href="#cb74-17" tabindex="-1"></a>                     size<span class="op">=</span><span class="dv">9</span>)</span>
<span id="cb74-18"><a aria-hidden="true" href="#cb74-18" tabindex="-1"></a>       <span class="op">+</span> p9.scale_y_continuous(expand<span class="op">=</span>(<span class="fl">.02</span>, <span class="fl">.2</span>))</span>
<span id="cb74-19"><a aria-hidden="true" href="#cb74-19" tabindex="-1"></a>       <span class="op">+</span> p9.theme(figure_size<span class="op">=</span>(<span class="dv">5</span>, <span class="dv">4</span>),</span>
<span id="cb74-20"><a aria-hidden="true" href="#cb74-20" tabindex="-1"></a>                  legend_position<span class="op">=</span><span class="st">'top'</span>,</span>
<span id="cb74-21"><a aria-hidden="true" href="#cb74-21" tabindex="-1"></a>                  legend_text<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">8</span>),</span>
<span id="cb74-22"><a aria-hidden="true" href="#cb74-22" tabindex="-1"></a>                  legend_key_size<span class="op">=</span><span class="dv">8</span>,</span>
<span id="cb74-23"><a aria-hidden="true" href="#cb74-23" tabindex="-1"></a>                  )</span>
<span id="cb74-24"><a aria-hidden="true" href="#cb74-24" tabindex="-1"></a>       )</span>
<span id="cb74-25"><a aria-hidden="true" href="#cb74-25" tabindex="-1"></a>)</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img class="figure-img" height="1200" src="microbial_diversity_analysis_files/figure-html/cell-73-output-1.png" width="1500"/></p>
</figure>
</div>
</div>
</div>
</section>
<section class="level4" id="permanova">
<h4 class="anchored" data-anchor-id="permanova">PERMANOVA</h4>
<div class="cell" data-execution_count="65" id="cell-104">
<div class="sourceCode cell-code" id="cb75"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb75-1"><a aria-hidden="true" href="#cb75-1" tabindex="-1"></a>dm_pm <span class="op">=</span> DistanceMatrix(dm, ids<span class="op">=</span>samples)</span>
<span id="cb75-2"><a aria-hidden="true" href="#cb75-2" tabindex="-1"></a>pm_factors <span class="op">=</span> (samples_info</span>
<span id="cb75-3"><a aria-hidden="true" href="#cb75-3" tabindex="-1"></a> .query(<span class="st">'sample_id not in ["S06", "S14"]'</span>)</span>
<span id="cb75-4"><a aria-hidden="true" href="#cb75-4" tabindex="-1"></a> .assign(month<span class="op">=</span><span class="kw">lambda</span> dd: pd.to_datetime(dd.date).dt.month_name())</span>
<span id="cb75-5"><a aria-hidden="true" href="#cb75-5" tabindex="-1"></a> .drop(columns<span class="op">=</span>[<span class="st">'date'</span>])</span>
<span id="cb75-6"><a aria-hidden="true" href="#cb75-6" tabindex="-1"></a>)</span>
<span id="cb75-7"><a aria-hidden="true" href="#cb75-7" tabindex="-1"></a>permanova_sample_mode <span class="op">=</span> permanova(dm_pm, pm_factors[<span class="st">'sample_mode'</span>])</span>
<span id="cb75-8"><a aria-hidden="true" href="#cb75-8" tabindex="-1"></a><span class="bu">print</span>(<span class="st">'PERMANOVA for sample_mode:</span><span class="ch">\n</span><span class="st">'</span>, permanova_sample_mode)</span>
<span id="cb75-9"><a aria-hidden="true" href="#cb75-9" tabindex="-1"></a>permanova_month <span class="op">=</span> permanova(dm_pm, pm_factors[<span class="st">'month'</span>])</span>
<span id="cb75-10"><a aria-hidden="true" href="#cb75-10" tabindex="-1"></a><span class="bu">print</span>(<span class="st">'</span><span class="ch">\n</span><span class="st">PERMANOVA for month:</span><span class="ch">\n</span><span class="st">'</span>, permanova_month)</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>PERMANOVA for sample_mode:
 method name               PERMANOVA
test statistic name        pseudo-F
sample size                      20
number of groups                  2
test statistic             0.676167
p-value                       0.558
number of permutations          999
Name: PERMANOVA results, dtype: object

PERMANOVA for month:
 method name               PERMANOVA
test statistic name        pseudo-F
sample size                      20
number of groups                  2
test statistic             3.999877
p-value                       0.013
number of permutations          999
Name: PERMANOVA results, dtype: object</code></pre>
</div>
</div>
</section>
</section>
<section class="level3" id="barplots">
<h3 class="anchored" data-anchor-id="barplots">Barplots</h3>
<div class="cell" data-execution_count="66" id="cell-106">
<div class="sourceCode cell-code" id="cb77"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb77-1"><a aria-hidden="true" href="#cb77-1" tabindex="-1"></a>f <span class="op">=</span> (abundance_df_phyla</span>
<span id="cb77-2"><a aria-hidden="true" href="#cb77-2" tabindex="-1"></a> .melt([<span class="st">'Kingdom'</span>, <span class="st">'Phylum'</span>], var_name<span class="op">=</span><span class="st">'sample_id'</span>, value_name<span class="op">=</span><span class="st">'counts'</span>)</span>
<span id="cb77-3"><a aria-hidden="true" href="#cb77-3" tabindex="-1"></a> .groupby([<span class="st">'Kingdom'</span>, <span class="st">'sample_id'</span>], as_index<span class="op">=</span><span class="va">False</span>)</span>
<span id="cb77-4"><a aria-hidden="true" href="#cb77-4" tabindex="-1"></a> .<span class="bu">apply</span>(<span class="kw">lambda</span> dd: dd.assign(relative_abundance<span class="op">=</span>dd.counts <span class="op">/</span> dd.counts.<span class="bu">sum</span>()))</span>
<span id="cb77-5"><a aria-hidden="true" href="#cb77-5" tabindex="-1"></a> .reset_index(drop<span class="op">=</span><span class="va">True</span>)</span>
<span id="cb77-6"><a aria-hidden="true" href="#cb77-6" tabindex="-1"></a> .melt([<span class="st">'Kingdom'</span>, <span class="st">'Phylum'</span>, <span class="st">'sample_id'</span>])</span>
<span id="cb77-7"><a aria-hidden="true" href="#cb77-7" tabindex="-1"></a> .assign(value<span class="op">=</span><span class="kw">lambda</span> dd: dd[<span class="st">'value'</span>].astype(<span class="bu">float</span>))</span>
<span id="cb77-8"><a aria-hidden="true" href="#cb77-8" tabindex="-1"></a> .replace(<span class="st">'unknown'</span>, pd.NA)</span>
<span id="cb77-9"><a aria-hidden="true" href="#cb77-9" tabindex="-1"></a>  .pipe(<span class="kw">lambda</span> dd: p9.ggplot(dd)</span>
<span id="cb77-10"><a aria-hidden="true" href="#cb77-10" tabindex="-1"></a>       <span class="op">+</span> p9.aes(<span class="st">'sample_id'</span>, <span class="st">'value'</span>, fill<span class="op">=</span><span class="st">'Phylum'</span>)</span>
<span id="cb77-11"><a aria-hidden="true" href="#cb77-11" tabindex="-1"></a>       <span class="op">+</span> p9.geom_col()</span>
<span id="cb77-12"><a aria-hidden="true" href="#cb77-12" tabindex="-1"></a>       <span class="op">+</span> p9.labs(x<span class="op">=</span><span class="st">''</span>, y<span class="op">=</span><span class="st">''</span>)</span>
<span id="cb77-13"><a aria-hidden="true" href="#cb77-13" tabindex="-1"></a>       <span class="op">+</span> p9.guides(fill<span class="op">=</span>p9.guide_legend(ncol<span class="op">=</span><span class="dv">1</span>))</span>
<span id="cb77-14"><a aria-hidden="true" href="#cb77-14" tabindex="-1"></a>       <span class="op">+</span> p9.facet_grid([<span class="st">'variable'</span>, <span class="st">'Kingdom'</span>], scales<span class="op">=</span><span class="st">'free_y'</span>)</span>
<span id="cb77-15"><a aria-hidden="true" href="#cb77-15" tabindex="-1"></a>       <span class="op">+</span> p9.theme(</span>
<span id="cb77-16"><a aria-hidden="true" href="#cb77-16" tabindex="-1"></a>           legend_key_size<span class="op">=</span><span class="dv">7</span>,</span>
<span id="cb77-17"><a aria-hidden="true" href="#cb77-17" tabindex="-1"></a>           axis_text_x<span class="op">=</span>p9.element_text(angle<span class="op">=</span><span class="dv">90</span>, size<span class="op">=</span><span class="dv">8</span>),</span>
<span id="cb77-18"><a aria-hidden="true" href="#cb77-18" tabindex="-1"></a>           figure_size<span class="op">=</span>(<span class="dv">6</span>, <span class="dv">3</span>),</span>
<span id="cb77-19"><a aria-hidden="true" href="#cb77-19" tabindex="-1"></a>           legend_text<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">7</span>),</span>
<span id="cb77-20"><a aria-hidden="true" href="#cb77-20" tabindex="-1"></a></span>
<span id="cb77-21"><a aria-hidden="true" href="#cb77-21" tabindex="-1"></a>       )</span>
<span id="cb77-22"><a aria-hidden="true" href="#cb77-22" tabindex="-1"></a>       )</span>
<span id="cb77-23"><a aria-hidden="true" href="#cb77-23" tabindex="-1"></a> ).save(<span class="st">'../output/figures/abundance_by_sample_type_phyla.svg'</span>)</span>
<span id="cb77-24"><a aria-hidden="true" href="#cb77-24" tabindex="-1"></a>Image(<span class="st">'../output/figures/abundance_by_sample_type_phyla.png'</span>)</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="66">
<div>
<figure class="figure">
<p><img class="img-fluid figure-img" src="microbial_diversity_analysis_files/figure-html/cell-75-output-1.png"/></p>
</figure>
</div>
</div>
</div>
<section class="level4" id="bubble-charts">
<h4 class="anchored" data-anchor-id="bubble-charts">Bubble Charts</h4>
<section class="level5" id="phyla">
<h5 class="anchored" data-anchor-id="phyla">Phyla</h5>
<div class="cell" data-execution_count="68" id="cell-109">
<details class="code-fold">
<summary>Show Code</summary>
<div class="sourceCode cell-code" id="cb78"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb78-1"><a aria-hidden="true" href="#cb78-1" tabindex="-1"></a>(abundance_df_phyla</span>
<span id="cb78-2"><a aria-hidden="true" href="#cb78-2" tabindex="-1"></a> .melt([<span class="st">'Kingdom'</span>, <span class="st">'Phylum'</span>], var_name<span class="op">=</span><span class="st">'sample_id'</span>)</span>
<span id="cb78-3"><a aria-hidden="true" href="#cb78-3" tabindex="-1"></a> .groupby([<span class="st">'Kingdom'</span>, <span class="st">'sample_id'</span>], as_index<span class="op">=</span><span class="va">False</span>)</span>
<span id="cb78-4"><a aria-hidden="true" href="#cb78-4" tabindex="-1"></a> .<span class="bu">apply</span>(<span class="kw">lambda</span> dd: dd.assign(relative_abundance<span class="op">=</span><span class="kw">lambda</span> dd: dd.value <span class="op">/</span> dd.value.<span class="bu">sum</span>()))</span>
<span id="cb78-5"><a aria-hidden="true" href="#cb78-5" tabindex="-1"></a> .query(<span class="st">'Phylum != "unknown"'</span>)</span>
<span id="cb78-6"><a aria-hidden="true" href="#cb78-6" tabindex="-1"></a> .query(<span class="st">'Kingdom == "Fungi"'</span>)</span>
<span id="cb78-7"><a aria-hidden="true" href="#cb78-7" tabindex="-1"></a> .merge(samples_info, on<span class="op">=</span><span class="st">'sample_id'</span>)</span>
<span id="cb78-8"><a aria-hidden="true" href="#cb78-8" tabindex="-1"></a> .assign(sample_id<span class="op">=</span><span class="kw">lambda</span> dd: <span class="st">'S'</span> <span class="op">+</span> dd.sample_id.<span class="bu">str</span>[<span class="op">-</span><span class="dv">2</span>:])</span>
<span id="cb78-9"><a aria-hidden="true" href="#cb78-9" tabindex="-1"></a>  .query(<span class="st">'relative_abundance &gt; 0'</span>)</span>
<span id="cb78-10"><a aria-hidden="true" href="#cb78-10" tabindex="-1"></a> .pipe(<span class="kw">lambda</span> dd: p9.ggplot(dd)</span>
<span id="cb78-11"><a aria-hidden="true" href="#cb78-11" tabindex="-1"></a>       <span class="op">+</span> p9.aes(<span class="st">'sample_id'</span>, <span class="st">'reorder(Phylum, relative_abundance)'</span>,</span>
<span id="cb78-12"><a aria-hidden="true" href="#cb78-12" tabindex="-1"></a>                 size<span class="op">=</span><span class="st">'relative_abundance'</span>, color<span class="op">=</span><span class="st">'Kingdom'</span>, fill<span class="op">=</span><span class="st">'Kingdom'</span>)</span>
<span id="cb78-13"><a aria-hidden="true" href="#cb78-13" tabindex="-1"></a>       <span class="op">+</span> p9.geom_point(alpha<span class="op">=</span><span class="fl">.8</span>)</span>
<span id="cb78-14"><a aria-hidden="true" href="#cb78-14" tabindex="-1"></a>       <span class="op">+</span> p9.facet_grid([<span class="st">'Kingdom'</span>, <span class="st">'sample_mode'</span>], scales<span class="op">=</span><span class="st">'free'</span>, space<span class="op">=</span><span class="st">'free'</span>)</span>
<span id="cb78-15"><a aria-hidden="true" href="#cb78-15" tabindex="-1"></a>       <span class="op">+</span> p9.scale_size_continuous(trans<span class="op">=</span><span class="st">'log2'</span>, <span class="bu">range</span><span class="op">=</span>(<span class="fl">.2</span>, <span class="dv">9</span>), labels<span class="op">=</span>percent_format(), </span>
<span id="cb78-16"><a aria-hidden="true" href="#cb78-16" tabindex="-1"></a>                                    limits<span class="op">=</span>(<span class="fl">.0005</span>, <span class="fl">.9</span>), breaks<span class="op">=</span>[<span class="fl">.001</span>, <span class="fl">.01</span>, <span class="fl">.05</span>, <span class="fl">.5</span>])</span>
<span id="cb78-17"><a aria-hidden="true" href="#cb78-17" tabindex="-1"></a>       <span class="op">+</span> p9.scale_y_discrete(expand<span class="op">=</span>(<span class="fl">.35</span>, <span class="fl">.35</span>))</span>
<span id="cb78-18"><a aria-hidden="true" href="#cb78-18" tabindex="-1"></a>       <span class="op">+</span> p9.guides(size<span class="op">=</span>p9.guide_legend(ncol<span class="op">=</span><span class="dv">1</span>))</span>
<span id="cb78-19"><a aria-hidden="true" href="#cb78-19" tabindex="-1"></a>       <span class="op">+</span> p9.scale_fill_manual(values<span class="op">=</span>[<span class="st">'#435B97'</span>])</span>
<span id="cb78-20"><a aria-hidden="true" href="#cb78-20" tabindex="-1"></a>       <span class="op">+</span> p9.scale_color_manual(values<span class="op">=</span>[<span class="st">'#435B97'</span>])</span>
<span id="cb78-21"><a aria-hidden="true" href="#cb78-21" tabindex="-1"></a>       <span class="op">+</span> p9.labs(x<span class="op">=</span><span class="st">''</span>, y<span class="op">=</span><span class="st">''</span>, size<span class="op">=</span><span class="st">'Relative Abundance'</span>, color<span class="op">=</span><span class="st">'Kingdom'</span>,</span>
<span id="cb78-22"><a aria-hidden="true" href="#cb78-22" tabindex="-1"></a>                  )</span>
<span id="cb78-23"><a aria-hidden="true" href="#cb78-23" tabindex="-1"></a>       <span class="op">+</span> p9.theme(figure_size<span class="op">=</span>(<span class="dv">9</span>, <span class="dv">2</span>),</span>
<span id="cb78-24"><a aria-hidden="true" href="#cb78-24" tabindex="-1"></a>                  strip_text<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">12</span>),</span>
<span id="cb78-25"><a aria-hidden="true" href="#cb78-25" tabindex="-1"></a>                  axis_text_x<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">8</span>),</span>
<span id="cb78-26"><a aria-hidden="true" href="#cb78-26" tabindex="-1"></a>       </span>
<span id="cb78-27"><a aria-hidden="true" href="#cb78-27" tabindex="-1"></a>       )</span>
<span id="cb78-28"><a aria-hidden="true" href="#cb78-28" tabindex="-1"></a> )</span>
<span id="cb78-29"><a aria-hidden="true" href="#cb78-29" tabindex="-1"></a>).save(<span class="st">'../output/figures/abundance_phyla_fungi.svg'</span>, bbox_inches<span class="op">=</span><span class="st">'tight'</span>)</span>
<span id="cb78-30"><a aria-hidden="true" href="#cb78-30" tabindex="-1"></a></span>
<span id="cb78-31"><a aria-hidden="true" href="#cb78-31" tabindex="-1"></a></span>
<span id="cb78-32"><a aria-hidden="true" href="#cb78-32" tabindex="-1"></a>(abundance_df_phyla</span>
<span id="cb78-33"><a aria-hidden="true" href="#cb78-33" tabindex="-1"></a> .melt([<span class="st">'Kingdom'</span>, <span class="st">'Phylum'</span>], var_name<span class="op">=</span><span class="st">'sample_id'</span>)</span>
<span id="cb78-34"><a aria-hidden="true" href="#cb78-34" tabindex="-1"></a> .groupby([<span class="st">'Kingdom'</span>, <span class="st">'sample_id'</span>], as_index<span class="op">=</span><span class="va">False</span>)</span>
<span id="cb78-35"><a aria-hidden="true" href="#cb78-35" tabindex="-1"></a> .<span class="bu">apply</span>(<span class="kw">lambda</span> dd: dd.assign(relative_abundance<span class="op">=</span><span class="kw">lambda</span> dd: dd.value <span class="op">/</span> dd.value.<span class="bu">sum</span>()))</span>
<span id="cb78-36"><a aria-hidden="true" href="#cb78-36" tabindex="-1"></a> .query(<span class="st">'Phylum != "unknown"'</span>)</span>
<span id="cb78-37"><a aria-hidden="true" href="#cb78-37" tabindex="-1"></a> .query(<span class="st">'Kingdom == "Bacteria"'</span>)</span>
<span id="cb78-38"><a aria-hidden="true" href="#cb78-38" tabindex="-1"></a> .query(<span class="st">'relative_abundance &gt; 0'</span>)</span>
<span id="cb78-39"><a aria-hidden="true" href="#cb78-39" tabindex="-1"></a> .merge(samples_info, on<span class="op">=</span><span class="st">'sample_id'</span>)</span>
<span id="cb78-40"><a aria-hidden="true" href="#cb78-40" tabindex="-1"></a> .assign(sample_id<span class="op">=</span><span class="kw">lambda</span> dd: <span class="st">'S'</span> <span class="op">+</span> dd.sample_id.<span class="bu">str</span>[<span class="op">-</span><span class="dv">2</span>:])</span>
<span id="cb78-41"><a aria-hidden="true" href="#cb78-41" tabindex="-1"></a> .pipe(<span class="kw">lambda</span> dd: p9.ggplot(dd)</span>
<span id="cb78-42"><a aria-hidden="true" href="#cb78-42" tabindex="-1"></a>       <span class="op">+</span> p9.aes(<span class="st">'sample_id'</span>, <span class="st">'reorder(Phylum, relative_abundance)'</span>,</span>
<span id="cb78-43"><a aria-hidden="true" href="#cb78-43" tabindex="-1"></a>                 size<span class="op">=</span><span class="st">'relative_abundance'</span>, color<span class="op">=</span><span class="st">'Kingdom'</span>, fill<span class="op">=</span><span class="st">'Kingdom'</span>)</span>
<span id="cb78-44"><a aria-hidden="true" href="#cb78-44" tabindex="-1"></a>       <span class="op">+</span> p9.geom_point(alpha<span class="op">=</span><span class="fl">.8</span>)</span>
<span id="cb78-45"><a aria-hidden="true" href="#cb78-45" tabindex="-1"></a>       <span class="op">+</span> p9.facet_grid([<span class="st">'Kingdom'</span>, <span class="st">'sample_mode'</span>], scales<span class="op">=</span><span class="st">'free'</span>, space<span class="op">=</span><span class="st">'free'</span>)</span>
<span id="cb78-46"><a aria-hidden="true" href="#cb78-46" tabindex="-1"></a>       <span class="op">+</span> p9.scale_size_continuous(trans<span class="op">=</span><span class="st">'log2'</span>, <span class="bu">range</span><span class="op">=</span>(<span class="fl">.2</span>, <span class="dv">9</span>), labels<span class="op">=</span>percent_format(), </span>
<span id="cb78-47"><a aria-hidden="true" href="#cb78-47" tabindex="-1"></a>                                    limits<span class="op">=</span>(<span class="fl">.0005</span>, <span class="fl">.9</span>), breaks<span class="op">=</span>[<span class="fl">.001</span>, <span class="fl">.01</span>, <span class="fl">.05</span>, <span class="fl">.5</span>])</span>
<span id="cb78-48"><a aria-hidden="true" href="#cb78-48" tabindex="-1"></a>       <span class="op">+</span> p9.scale_y_discrete(expand<span class="op">=</span>(<span class="fl">.05</span>, <span class="fl">.05</span>))</span>
<span id="cb78-49"><a aria-hidden="true" href="#cb78-49" tabindex="-1"></a>       <span class="op">+</span> p9.guides(size<span class="op">=</span>p9.guide_legend(ncol<span class="op">=</span><span class="dv">1</span>))</span>
<span id="cb78-50"><a aria-hidden="true" href="#cb78-50" tabindex="-1"></a>       <span class="op">+</span> p9.scale_fill_manual(values<span class="op">=</span>[<span class="st">'#D3894C'</span>])</span>
<span id="cb78-51"><a aria-hidden="true" href="#cb78-51" tabindex="-1"></a>       <span class="op">+</span> p9.scale_color_manual(values<span class="op">=</span>[<span class="st">'#D3894C'</span>])</span>
<span id="cb78-52"><a aria-hidden="true" href="#cb78-52" tabindex="-1"></a>       <span class="op">+</span> p9.labs(x<span class="op">=</span><span class="st">''</span>, y<span class="op">=</span><span class="st">''</span>, size<span class="op">=</span><span class="st">'Relative Abundance'</span>, color<span class="op">=</span><span class="st">'Kingdom'</span>,</span>
<span id="cb78-53"><a aria-hidden="true" href="#cb78-53" tabindex="-1"></a>                  )</span>
<span id="cb78-54"><a aria-hidden="true" href="#cb78-54" tabindex="-1"></a>       <span class="op">+</span> p9.theme(figure_size<span class="op">=</span>(<span class="dv">9</span>, <span class="dv">5</span>),</span>
<span id="cb78-55"><a aria-hidden="true" href="#cb78-55" tabindex="-1"></a>                  strip_text<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">12</span>),</span>
<span id="cb78-56"><a aria-hidden="true" href="#cb78-56" tabindex="-1"></a>                  axis_text_x<span class="op">=</span>p9.element_blank(),</span>
<span id="cb78-57"><a aria-hidden="true" href="#cb78-57" tabindex="-1"></a>                  axis_ticks_major_x<span class="op">=</span>p9.element_blank(),</span>
<span id="cb78-58"><a aria-hidden="true" href="#cb78-58" tabindex="-1"></a>       </span>
<span id="cb78-59"><a aria-hidden="true" href="#cb78-59" tabindex="-1"></a>       )</span>
<span id="cb78-60"><a aria-hidden="true" href="#cb78-60" tabindex="-1"></a> )</span>
<span id="cb78-61"><a aria-hidden="true" href="#cb78-61" tabindex="-1"></a>).save(<span class="st">'../output/figures/abundance_phyla_bacteria.svg'</span>)</span>
<span id="cb78-62"><a aria-hidden="true" href="#cb78-62" tabindex="-1"></a></span>
<span id="cb78-63"><a aria-hidden="true" href="#cb78-63" tabindex="-1"></a><span class="co"># The actual figure is made in Inkscape since plotnine doesn't support</span></span>
<span id="cb78-64"><a aria-hidden="true" href="#cb78-64" tabindex="-1"></a><span class="co"># facets of different sizes and the difference between the number of bacterial</span></span>
<span id="cb78-65"><a aria-hidden="true" href="#cb78-65" tabindex="-1"></a><span class="co"># and fungal phyla is too large to make a single figure with the same size</span></span>
<span id="cb78-66"><a aria-hidden="true" href="#cb78-66" tabindex="-1"></a></span>
<span id="cb78-67"><a aria-hidden="true" href="#cb78-67" tabindex="-1"></a>Image(<span class="st">'../output/figures/abundance_phyla.png'</span>)</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-display" data-execution_count="68">
<div>
<figure class="figure">
<p><img class="img-fluid figure-img" src="microbial_diversity_analysis_files/figure-html/cell-76-output-1.png"/></p>
</figure>
</div>
</div>
</div>
</section>
<section class="level5" id="genera">
<h5 class="anchored" data-anchor-id="genera">Genera</h5>
<p>To represent the diversity and fluctuation of genera across all samples, what we will do is select the top 30 most common (by total number of reads across all 22 samples)</p>
<div class="cell" data-execution_count="22" id="cell-112">
<div class="sourceCode cell-code" id="cb79"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb79-1"><a aria-hidden="true" href="#cb79-1" tabindex="-1"></a>taxon_names <span class="op">=</span> [<span class="st">'Kingdom'</span>, <span class="st">'Phylum'</span>, <span class="st">'Class'</span>, <span class="st">'Order'</span>, <span class="st">'Family'</span>, <span class="st">'Genus'</span>]</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="87" id="cell-113">
<div class="sourceCode cell-code" id="cb80"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb80-1"><a aria-hidden="true" href="#cb80-1" tabindex="-1"></a>n <span class="op">=</span> <span class="dv">30</span></span>
<span id="cb80-2"><a aria-hidden="true" href="#cb80-2" tabindex="-1"></a>l <span class="op">=</span> (<span class="dv">8</span> <span class="op">/</span> <span class="dv">20</span>) <span class="op">*</span> n</span>
<span id="cb80-3"><a aria-hidden="true" href="#cb80-3" tabindex="-1"></a></span>
<span id="cb80-4"><a aria-hidden="true" href="#cb80-4" tabindex="-1"></a>over_3_samples_genera <span class="op">=</span> samples_per_otu.query(<span class="st">'n_samples &gt;= 3'</span>).Genus</span>
<span id="cb80-5"><a aria-hidden="true" href="#cb80-5" tabindex="-1"></a>selected_bacterial_genera <span class="op">=</span> (abundance_long_genera</span>
<span id="cb80-6"><a aria-hidden="true" href="#cb80-6" tabindex="-1"></a>    .query(<span class="st">'Kingdom=="Bacteria"'</span>)</span>
<span id="cb80-7"><a aria-hidden="true" href="#cb80-7" tabindex="-1"></a>    .query(<span class="st">'Genus not in @inconclusive_list'</span>)</span>
<span id="cb80-8"><a aria-hidden="true" href="#cb80-8" tabindex="-1"></a>    .query(<span class="st">'Genus in @over_3_samples_genera'</span>)</span>
<span id="cb80-9"><a aria-hidden="true" href="#cb80-9" tabindex="-1"></a>    .groupby(taxon_names)</span>
<span id="cb80-10"><a aria-hidden="true" href="#cb80-10" tabindex="-1"></a>    .counts</span>
<span id="cb80-11"><a aria-hidden="true" href="#cb80-11" tabindex="-1"></a>    .<span class="bu">sum</span>()</span>
<span id="cb80-12"><a aria-hidden="true" href="#cb80-12" tabindex="-1"></a>    .sort_values(ascending<span class="op">=</span><span class="va">False</span>)</span>
<span id="cb80-13"><a aria-hidden="true" href="#cb80-13" tabindex="-1"></a>    .iloc[:n]</span>
<span id="cb80-14"><a aria-hidden="true" href="#cb80-14" tabindex="-1"></a>    .reset_index()</span>
<span id="cb80-15"><a aria-hidden="true" href="#cb80-15" tabindex="-1"></a>    [<span class="st">'Genus'</span>].unique()</span>
<span id="cb80-16"><a aria-hidden="true" href="#cb80-16" tabindex="-1"></a>)</span>
<span id="cb80-17"><a aria-hidden="true" href="#cb80-17" tabindex="-1"></a></span>
<span id="cb80-18"><a aria-hidden="true" href="#cb80-18" tabindex="-1"></a>selected_fungal_genera <span class="op">=</span> (abundance_long_genera</span>
<span id="cb80-19"><a aria-hidden="true" href="#cb80-19" tabindex="-1"></a>    .query(<span class="st">'Genus not in @inconclusive_list'</span>)</span>
<span id="cb80-20"><a aria-hidden="true" href="#cb80-20" tabindex="-1"></a>    .query(<span class="st">'Kingdom=="Fungi"'</span>)</span>
<span id="cb80-21"><a aria-hidden="true" href="#cb80-21" tabindex="-1"></a>    .query(<span class="st">'Genus in @over_3_samples_genera'</span>)</span>
<span id="cb80-22"><a aria-hidden="true" href="#cb80-22" tabindex="-1"></a>    .groupby(taxon_names)</span>
<span id="cb80-23"><a aria-hidden="true" href="#cb80-23" tabindex="-1"></a>    .counts</span>
<span id="cb80-24"><a aria-hidden="true" href="#cb80-24" tabindex="-1"></a>    .<span class="bu">sum</span>()</span>
<span id="cb80-25"><a aria-hidden="true" href="#cb80-25" tabindex="-1"></a>    .sort_values(ascending<span class="op">=</span><span class="va">False</span>)</span>
<span id="cb80-26"><a aria-hidden="true" href="#cb80-26" tabindex="-1"></a>    .iloc[:n]</span>
<span id="cb80-27"><a aria-hidden="true" href="#cb80-27" tabindex="-1"></a>    .reset_index()</span>
<span id="cb80-28"><a aria-hidden="true" href="#cb80-28" tabindex="-1"></a>    [<span class="st">'Genus'</span>].unique()</span>
<span id="cb80-29"><a aria-hidden="true" href="#cb80-29" tabindex="-1"></a>)</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="151" id="cell-114">
<details class="code-fold">
<summary>Show Code</summary>
<div class="sourceCode cell-code" id="cb81"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb81-1"><a aria-hidden="true" href="#cb81-1" tabindex="-1"></a>n <span class="op">=</span> <span class="dv">30</span></span>
<span id="cb81-2"><a aria-hidden="true" href="#cb81-2" tabindex="-1"></a>l <span class="op">=</span> (<span class="dv">8</span> <span class="op">/</span> <span class="dv">20</span>) <span class="op">*</span> n</span>
<span id="cb81-3"><a aria-hidden="true" href="#cb81-3" tabindex="-1"></a></span>
<span id="cb81-4"><a aria-hidden="true" href="#cb81-4" tabindex="-1"></a>over_3_samples_genera <span class="op">=</span> samples_per_otu.query(<span class="st">'n_samples &gt;= 3'</span>).Genus</span>
<span id="cb81-5"><a aria-hidden="true" href="#cb81-5" tabindex="-1"></a>selected_bacterial_genera <span class="op">=</span> (abundance_long_genera</span>
<span id="cb81-6"><a aria-hidden="true" href="#cb81-6" tabindex="-1"></a>    .query(<span class="st">'Kingdom=="Bacteria"'</span>)</span>
<span id="cb81-7"><a aria-hidden="true" href="#cb81-7" tabindex="-1"></a>    .query(<span class="st">'Genus not in @inconclusive_list'</span>)</span>
<span id="cb81-8"><a aria-hidden="true" href="#cb81-8" tabindex="-1"></a>    .query(<span class="st">'Genus in @over_3_samples_genera'</span>)</span>
<span id="cb81-9"><a aria-hidden="true" href="#cb81-9" tabindex="-1"></a>    .groupby(taxon_names)</span>
<span id="cb81-10"><a aria-hidden="true" href="#cb81-10" tabindex="-1"></a>    .counts</span>
<span id="cb81-11"><a aria-hidden="true" href="#cb81-11" tabindex="-1"></a>    .<span class="bu">sum</span>()</span>
<span id="cb81-12"><a aria-hidden="true" href="#cb81-12" tabindex="-1"></a>    .sort_values(ascending<span class="op">=</span><span class="va">False</span>)</span>
<span id="cb81-13"><a aria-hidden="true" href="#cb81-13" tabindex="-1"></a>    .iloc[:n]</span>
<span id="cb81-14"><a aria-hidden="true" href="#cb81-14" tabindex="-1"></a>    .reset_index()</span>
<span id="cb81-15"><a aria-hidden="true" href="#cb81-15" tabindex="-1"></a>    [<span class="st">'Genus'</span>].unique()</span>
<span id="cb81-16"><a aria-hidden="true" href="#cb81-16" tabindex="-1"></a>)</span>
<span id="cb81-17"><a aria-hidden="true" href="#cb81-17" tabindex="-1"></a></span>
<span id="cb81-18"><a aria-hidden="true" href="#cb81-18" tabindex="-1"></a>selected_fungal_genera <span class="op">=</span> (abundance_long_genera</span>
<span id="cb81-19"><a aria-hidden="true" href="#cb81-19" tabindex="-1"></a>    .query(<span class="st">'Genus not in @inconclusive_list'</span>)</span>
<span id="cb81-20"><a aria-hidden="true" href="#cb81-20" tabindex="-1"></a>    .query(<span class="st">'Kingdom=="Fungi"'</span>)</span>
<span id="cb81-21"><a aria-hidden="true" href="#cb81-21" tabindex="-1"></a>    .query(<span class="st">'Genus in @over_3_samples_genera'</span>)</span>
<span id="cb81-22"><a aria-hidden="true" href="#cb81-22" tabindex="-1"></a>    .groupby(taxon_names)</span>
<span id="cb81-23"><a aria-hidden="true" href="#cb81-23" tabindex="-1"></a>    .counts</span>
<span id="cb81-24"><a aria-hidden="true" href="#cb81-24" tabindex="-1"></a>    .<span class="bu">sum</span>()</span>
<span id="cb81-25"><a aria-hidden="true" href="#cb81-25" tabindex="-1"></a>    .sort_values(ascending<span class="op">=</span><span class="va">False</span>)</span>
<span id="cb81-26"><a aria-hidden="true" href="#cb81-26" tabindex="-1"></a>    .iloc[:n]</span>
<span id="cb81-27"><a aria-hidden="true" href="#cb81-27" tabindex="-1"></a>    .reset_index()</span>
<span id="cb81-28"><a aria-hidden="true" href="#cb81-28" tabindex="-1"></a>    [<span class="st">'Genus'</span>].unique()</span>
<span id="cb81-29"><a aria-hidden="true" href="#cb81-29" tabindex="-1"></a>)</span>
<span id="cb81-30"><a aria-hidden="true" href="#cb81-30" tabindex="-1"></a></span>
<span id="cb81-31"><a aria-hidden="true" href="#cb81-31" tabindex="-1"></a>ordered_genera <span class="op">=</span> selected_bacterial_genera.tolist() <span class="op">+</span> selected_fungal_genera.tolist()</span>
<span id="cb81-32"><a aria-hidden="true" href="#cb81-32" tabindex="-1"></a>ordered_genera.reverse()</span>
<span id="cb81-33"><a aria-hidden="true" href="#cb81-33" tabindex="-1"></a></span>
<span id="cb81-34"><a aria-hidden="true" href="#cb81-34" tabindex="-1"></a>f <span class="op">=</span> (abundance_long_genera</span>
<span id="cb81-35"><a aria-hidden="true" href="#cb81-35" tabindex="-1"></a> .groupby([<span class="st">'Kingdom'</span>, <span class="st">'sample_id'</span>], as_index<span class="op">=</span><span class="va">False</span>)</span>
<span id="cb81-36"><a aria-hidden="true" href="#cb81-36" tabindex="-1"></a> .<span class="bu">apply</span>(<span class="kw">lambda</span> dd: dd.assign(relative_abundance<span class="op">=</span><span class="kw">lambda</span> dd: dd.counts <span class="op">/</span> dd.counts.<span class="bu">sum</span>()))</span>
<span id="cb81-37"><a aria-hidden="true" href="#cb81-37" tabindex="-1"></a> .query(<span class="st">'Genus in @selected_bacterial_genera or Genus in @selected_fungal_genera'</span>)</span>
<span id="cb81-38"><a aria-hidden="true" href="#cb81-38" tabindex="-1"></a> .merge(samples_info, on<span class="op">=</span><span class="st">'sample_id'</span>)</span>
<span id="cb81-39"><a aria-hidden="true" href="#cb81-39" tabindex="-1"></a> .query(<span class="st">'relative_abundance &gt; 0'</span>)</span>
<span id="cb81-40"><a aria-hidden="true" href="#cb81-40" tabindex="-1"></a> .assign(Genus<span class="op">=</span><span class="kw">lambda</span> dd: pd.Categorical(dd.Genus, categories<span class="op">=</span>ordered_genera, ordered<span class="op">=</span><span class="va">True</span>))</span>
<span id="cb81-41"><a aria-hidden="true" href="#cb81-41" tabindex="-1"></a> .pipe(<span class="kw">lambda</span> dd: p9.ggplot(dd)</span>
<span id="cb81-42"><a aria-hidden="true" href="#cb81-42" tabindex="-1"></a>       <span class="op">+</span> p9.aes(<span class="st">'sample_id'</span>, <span class="st">'Genus'</span>, size<span class="op">=</span><span class="st">'relative_abundance'</span>, color<span class="op">=</span><span class="st">'Class'</span>, fill<span class="op">=</span><span class="st">'Class'</span>)</span>
<span id="cb81-43"><a aria-hidden="true" href="#cb81-43" tabindex="-1"></a>       <span class="op">+</span> p9.geom_point(alpha<span class="op">=</span><span class="fl">.8</span>)</span>
<span id="cb81-44"><a aria-hidden="true" href="#cb81-44" tabindex="-1"></a>       <span class="op">+</span> p9.facet_grid([<span class="st">'Kingdom'</span>, <span class="st">'sample_mode'</span>], scales<span class="op">=</span><span class="st">'free'</span>)</span>
<span id="cb81-45"><a aria-hidden="true" href="#cb81-45" tabindex="-1"></a>       <span class="op">+</span> p9.scale_size_continuous(trans<span class="op">=</span><span class="st">'log10'</span>, <span class="bu">range</span><span class="op">=</span>(<span class="fl">.2</span>, <span class="dv">9</span>), labels<span class="op">=</span>percent_format(), </span>
<span id="cb81-46"><a aria-hidden="true" href="#cb81-46" tabindex="-1"></a>                                    limits<span class="op">=</span>(<span class="fl">.0005</span>, <span class="fl">.9</span>), breaks<span class="op">=</span>[<span class="fl">.001</span>, <span class="fl">.01</span>, <span class="fl">.05</span>, <span class="fl">.5</span>])</span>
<span id="cb81-47"><a aria-hidden="true" href="#cb81-47" tabindex="-1"></a>       <span class="op">+</span> p9.scale_y_discrete(expand<span class="op">=</span>(<span class="fl">.035</span>, <span class="fl">.035</span>))</span>
<span id="cb81-48"><a aria-hidden="true" href="#cb81-48" tabindex="-1"></a>       <span class="op">+</span> p9.guides(size<span class="op">=</span>p9.guide_legend(ncol<span class="op">=</span><span class="dv">1</span>))</span>
<span id="cb81-49"><a aria-hidden="true" href="#cb81-49" tabindex="-1"></a>       <span class="op">+</span> p9.labs(x<span class="op">=</span><span class="st">''</span>, y<span class="op">=</span><span class="st">''</span>, size<span class="op">=</span><span class="st">'Relative Abundance'</span>, color<span class="op">=</span><span class="st">'Class'</span>,</span>
<span id="cb81-50"><a aria-hidden="true" href="#cb81-50" tabindex="-1"></a>                  )</span>
<span id="cb81-51"><a aria-hidden="true" href="#cb81-51" tabindex="-1"></a>       <span class="op">+</span> p9.theme(figure_size<span class="op">=</span>(<span class="dv">12</span>, l),</span>
<span id="cb81-52"><a aria-hidden="true" href="#cb81-52" tabindex="-1"></a>                  strip_text<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">12</span>),</span>
<span id="cb81-53"><a aria-hidden="true" href="#cb81-53" tabindex="-1"></a>                  axis_text<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">10</span>),</span>
<span id="cb81-54"><a aria-hidden="true" href="#cb81-54" tabindex="-1"></a>       </span>
<span id="cb81-55"><a aria-hidden="true" href="#cb81-55" tabindex="-1"></a>       )</span>
<span id="cb81-56"><a aria-hidden="true" href="#cb81-56" tabindex="-1"></a> )</span>
<span id="cb81-57"><a aria-hidden="true" href="#cb81-57" tabindex="-1"></a> )</span>
<span id="cb81-58"><a aria-hidden="true" href="#cb81-58" tabindex="-1"></a></span>
<span id="cb81-59"><a aria-hidden="true" href="#cb81-59" tabindex="-1"></a>f.save(<span class="st">'../output/figures/top30_genera_bubbles.svg'</span>)</span>
<span id="cb81-60"><a aria-hidden="true" href="#cb81-60" tabindex="-1"></a>f.draw()</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-error">
<div class="ansi-escaped-output">
<pre><span class="ansi-red-fg">---------------------------------------------------------------------------</span>
<span class="ansi-red-fg">NameError</span>                                 Traceback (most recent call last)
Cell <span class="ansi-green-fg">In[151], line 5</span>
<span class="ansi-green-fg ansi-bold">      2</span> n <span style="color:rgb(98,98,98)">=</span> <span style="color:rgb(98,98,98)">30</span>
<span class="ansi-green-fg ansi-bold">      3</span> l <span style="color:rgb(98,98,98)">=</span> (<span style="color:rgb(98,98,98)">8</span> <span style="color:rgb(98,98,98)">/</span> <span style="color:rgb(98,98,98)">20</span>) <span style="color:rgb(98,98,98)">*</span> n
<span class="ansi-green-fg">----&gt; 5</span> over_3_samples_genera <span style="color:rgb(98,98,98)">=</span> <span class="ansi-yellow-bg">samples_per_otu</span><span style="color:rgb(98,98,98)">.</span>query(<span style="color:rgb(175,0,0)">'</span><span style="color:rgb(175,0,0)">n_samples &gt;= 3</span><span style="color:rgb(175,0,0)">'</span>)<span style="color:rgb(98,98,98)">.</span>Genus
<span class="ansi-green-fg ansi-bold">      6</span> selected_bacterial_genera <span style="color:rgb(98,98,98)">=</span> (abundance_long_genera
<span class="ansi-green-fg ansi-bold">      7</span>     <span style="color:rgb(98,98,98)">.</span>query(<span style="color:rgb(175,0,0)">'</span><span style="color:rgb(175,0,0)">Kingdom==</span><span style="color:rgb(175,0,0)">"</span><span style="color:rgb(175,0,0)">Bacteria</span><span style="color:rgb(175,0,0)">"</span><span style="color:rgb(175,0,0)">'</span>)
<span class="ansi-green-fg ansi-bold">      8</span>     <span style="color:rgb(98,98,98)">.</span>query(<span style="color:rgb(175,0,0)">'</span><span style="color:rgb(175,0,0)">Genus not in @inconclusive_list</span><span style="color:rgb(175,0,0)">'</span>)
<span class="ansi-green-fg">   (...)</span>
<span class="ansi-green-fg ansi-bold">     16</span>     [<span style="color:rgb(175,0,0)">'</span><span style="color:rgb(175,0,0)">Genus</span><span style="color:rgb(175,0,0)">'</span>]<span style="color:rgb(98,98,98)">.</span>unique()
<span class="ansi-green-fg ansi-bold">     17</span> )
<span class="ansi-green-fg ansi-bold">     19</span> selected_fungal_genera <span style="color:rgb(98,98,98)">=</span> (abundance_long_genera
<span class="ansi-green-fg ansi-bold">     20</span>     <span style="color:rgb(98,98,98)">.</span>query(<span style="color:rgb(175,0,0)">'</span><span style="color:rgb(175,0,0)">Genus not in @inconclusive_list</span><span style="color:rgb(175,0,0)">'</span>)
<span class="ansi-green-fg ansi-bold">     21</span>     <span style="color:rgb(98,98,98)">.</span>query(<span style="color:rgb(175,0,0)">'</span><span style="color:rgb(175,0,0)">Kingdom==</span><span style="color:rgb(175,0,0)">"</span><span style="color:rgb(175,0,0)">Fungi</span><span style="color:rgb(175,0,0)">"</span><span style="color:rgb(175,0,0)">'</span>)
<span class="ansi-green-fg">   (...)</span>
<span class="ansi-green-fg ansi-bold">     29</span>     [<span style="color:rgb(175,0,0)">'</span><span style="color:rgb(175,0,0)">Genus</span><span style="color:rgb(175,0,0)">'</span>]<span style="color:rgb(98,98,98)">.</span>unique()
<span class="ansi-green-fg ansi-bold">     30</span> )

<span class="ansi-red-fg">NameError</span>: name 'samples_per_otu' is not defined</pre>
</div>
</div>
</div>
<div class="cell" data-execution_count="72" id="cell-115">
<div class="sourceCode cell-code" id="cb82"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb82-1"><a aria-hidden="true" href="#cb82-1" tabindex="-1"></a>dm_16s <span class="op">=</span> (abundance_df_genera</span>
<span id="cb82-2"><a aria-hidden="true" href="#cb82-2" tabindex="-1"></a> .query(<span class="st">'Genus!="unknown"'</span>)</span>
<span id="cb82-3"><a aria-hidden="true" href="#cb82-3" tabindex="-1"></a> .query(<span class="st">'Kingdom == "Bacteria"'</span>)</span>
<span id="cb82-4"><a aria-hidden="true" href="#cb82-4" tabindex="-1"></a> .drop(columns<span class="op">=</span>taxon_names[:<span class="op">-</span><span class="dv">1</span>])</span>
<span id="cb82-5"><a aria-hidden="true" href="#cb82-5" tabindex="-1"></a> .set_index(<span class="st">'Genus'</span>)</span>
<span id="cb82-6"><a aria-hidden="true" href="#cb82-6" tabindex="-1"></a> .T</span>
<span id="cb82-7"><a aria-hidden="true" href="#cb82-7" tabindex="-1"></a> .<span class="bu">apply</span>(<span class="kw">lambda</span> x: x <span class="op">/</span> x.<span class="bu">sum</span>(), axis<span class="op">=</span><span class="dv">1</span>)</span>
<span id="cb82-8"><a aria-hidden="true" href="#cb82-8" tabindex="-1"></a> .pipe(<span class="kw">lambda</span> dd: pairwise_distances(dd, metric<span class="op">=</span><span class="st">'braycurtis'</span>))</span>
<span id="cb82-9"><a aria-hidden="true" href="#cb82-9" tabindex="-1"></a>)</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="73" id="cell-116">
<div class="sourceCode cell-code" id="cb83"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb83-1"><a aria-hidden="true" href="#cb83-1" tabindex="-1"></a>dm_its <span class="op">=</span> (abundance_df_genera</span>
<span id="cb83-2"><a aria-hidden="true" href="#cb83-2" tabindex="-1"></a> .query(<span class="st">'Genus!="unknown"'</span>)</span>
<span id="cb83-3"><a aria-hidden="true" href="#cb83-3" tabindex="-1"></a> .query(<span class="st">'Kingdom == "Fungi"'</span>)</span>
<span id="cb83-4"><a aria-hidden="true" href="#cb83-4" tabindex="-1"></a> .drop(columns<span class="op">=</span>[<span class="st">'S06'</span>] <span class="op">+</span> taxon_names[:<span class="op">-</span><span class="dv">1</span>])</span>
<span id="cb83-5"><a aria-hidden="true" href="#cb83-5" tabindex="-1"></a> .set_index(<span class="st">'Genus'</span>)</span>
<span id="cb83-6"><a aria-hidden="true" href="#cb83-6" tabindex="-1"></a> .T</span>
<span id="cb83-7"><a aria-hidden="true" href="#cb83-7" tabindex="-1"></a> .<span class="bu">apply</span>(<span class="kw">lambda</span> x: x <span class="op">/</span> x.<span class="bu">sum</span>(), axis<span class="op">=</span><span class="dv">1</span>)</span>
<span id="cb83-8"><a aria-hidden="true" href="#cb83-8" tabindex="-1"></a> .pipe(<span class="kw">lambda</span> dd: pairwise_distances(dd, metric<span class="op">=</span><span class="st">'braycurtis'</span>))</span>
<span id="cb83-9"><a aria-hidden="true" href="#cb83-9" tabindex="-1"></a>)</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="74" id="cell-117">
<div class="sourceCode cell-code" id="cb84"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb84-1"><a aria-hidden="true" href="#cb84-1" tabindex="-1"></a>sample_columns_s06 <span class="op">=</span> samples_info.query(<span class="st">'sample_id not in ["S06", "S14"]'</span>).sample_id</span>
<span id="cb84-2"><a aria-hidden="true" href="#cb84-2" tabindex="-1"></a></span>
<span id="cb84-3"><a aria-hidden="true" href="#cb84-3" tabindex="-1"></a>bc_df <span class="op">=</span> (pd.DataFrame(dm, index<span class="op">=</span>sample_columns_s06, columns<span class="op">=</span>sample_columns_s06)</span>
<span id="cb84-4"><a aria-hidden="true" href="#cb84-4" tabindex="-1"></a> .reset_index()</span>
<span id="cb84-5"><a aria-hidden="true" href="#cb84-5" tabindex="-1"></a> .rename(columns<span class="op">=</span>{<span class="st">'index'</span>: <span class="st">'sample_id'</span>})</span>
<span id="cb84-6"><a aria-hidden="true" href="#cb84-6" tabindex="-1"></a> .melt(<span class="st">'sample_id'</span>, var_name<span class="op">=</span><span class="st">'sample_id2'</span>, value_name<span class="op">=</span><span class="st">'distance'</span>)</span>
<span id="cb84-7"><a aria-hidden="true" href="#cb84-7" tabindex="-1"></a> .sort_values([<span class="st">'sample_id'</span>, <span class="st">'sample_id2'</span>])</span>
<span id="cb84-8"><a aria-hidden="true" href="#cb84-8" tabindex="-1"></a> .merge(samples_info, on<span class="op">=</span><span class="st">'sample_id'</span>)</span>
<span id="cb84-9"><a aria-hidden="true" href="#cb84-9" tabindex="-1"></a> .merge(samples_info, left_on<span class="op">=</span><span class="st">'sample_id2'</span>, right_on<span class="op">=</span><span class="st">'sample_id'</span>, suffixes<span class="op">=</span>[<span class="st">''</span>, <span class="st">'_2'</span>])</span>
<span id="cb84-10"><a aria-hidden="true" href="#cb84-10" tabindex="-1"></a> .assign(timedelta<span class="op">=</span><span class="kw">lambda</span> dd: pd.to_datetime(dd.date_2) <span class="op">-</span> pd.to_datetime(dd.date))</span>
<span id="cb84-11"><a aria-hidden="true" href="#cb84-11" tabindex="-1"></a> .assign(combi<span class="op">=</span><span class="kw">lambda</span> dd: dd[<span class="st">'sample_mode'</span>] <span class="op">+</span> <span class="vs">r'$\rightarrow$'</span> <span class="op">+</span> dd[<span class="st">'sample_mode_2'</span>])</span>
<span id="cb84-12"><a aria-hidden="true" href="#cb84-12" tabindex="-1"></a>)</span>
<span id="cb84-13"><a aria-hidden="true" href="#cb84-13" tabindex="-1"></a></span>
<span id="cb84-14"><a aria-hidden="true" href="#cb84-14" tabindex="-1"></a>bc_df_16s <span class="op">=</span> (pd.DataFrame(dm_16s, index<span class="op">=</span>sample_columns, columns<span class="op">=</span>sample_columns)</span>
<span id="cb84-15"><a aria-hidden="true" href="#cb84-15" tabindex="-1"></a> .reset_index()</span>
<span id="cb84-16"><a aria-hidden="true" href="#cb84-16" tabindex="-1"></a> .rename(columns<span class="op">=</span>{<span class="st">'index'</span>: <span class="st">'sample_id'</span>})</span>
<span id="cb84-17"><a aria-hidden="true" href="#cb84-17" tabindex="-1"></a> .melt(<span class="st">'sample_id'</span>, var_name<span class="op">=</span><span class="st">'sample_id2'</span>, value_name<span class="op">=</span><span class="st">'distance'</span>)</span>
<span id="cb84-18"><a aria-hidden="true" href="#cb84-18" tabindex="-1"></a> .sort_values([<span class="st">'sample_id'</span>, <span class="st">'sample_id2'</span>])</span>
<span id="cb84-19"><a aria-hidden="true" href="#cb84-19" tabindex="-1"></a> .merge(samples_info, on<span class="op">=</span><span class="st">'sample_id'</span>)</span>
<span id="cb84-20"><a aria-hidden="true" href="#cb84-20" tabindex="-1"></a> .merge(samples_info, left_on<span class="op">=</span><span class="st">'sample_id2'</span>, right_on<span class="op">=</span><span class="st">'sample_id'</span>, suffixes<span class="op">=</span>[<span class="st">''</span>, <span class="st">'_2'</span>])</span>
<span id="cb84-21"><a aria-hidden="true" href="#cb84-21" tabindex="-1"></a> .assign(timedelta<span class="op">=</span><span class="kw">lambda</span> dd: pd.to_datetime(dd.date_2) <span class="op">-</span> pd.to_datetime(dd.date))</span>
<span id="cb84-22"><a aria-hidden="true" href="#cb84-22" tabindex="-1"></a> .assign(combi<span class="op">=</span><span class="kw">lambda</span> dd: dd[<span class="st">'sample_mode'</span>] <span class="op">+</span> <span class="vs">r'$\rightarrow$'</span> <span class="op">+</span> dd[<span class="st">'sample_mode_2'</span>])</span>
<span id="cb84-23"><a aria-hidden="true" href="#cb84-23" tabindex="-1"></a>)</span>
<span id="cb84-24"><a aria-hidden="true" href="#cb84-24" tabindex="-1"></a></span>
<span id="cb84-25"><a aria-hidden="true" href="#cb84-25" tabindex="-1"></a>bc_df_its <span class="op">=</span> (pd.DataFrame(dm_its, index<span class="op">=</span>sample_columns_s06, columns<span class="op">=</span>sample_columns_s06)</span>
<span id="cb84-26"><a aria-hidden="true" href="#cb84-26" tabindex="-1"></a> .reset_index()</span>
<span id="cb84-27"><a aria-hidden="true" href="#cb84-27" tabindex="-1"></a> .rename(columns<span class="op">=</span>{<span class="st">'index'</span>: <span class="st">'sample_id'</span>})</span>
<span id="cb84-28"><a aria-hidden="true" href="#cb84-28" tabindex="-1"></a> .melt(<span class="st">'sample_id'</span>, var_name<span class="op">=</span><span class="st">'sample_id2'</span>, value_name<span class="op">=</span><span class="st">'distance'</span>)</span>
<span id="cb84-29"><a aria-hidden="true" href="#cb84-29" tabindex="-1"></a> .sort_values([<span class="st">'sample_id'</span>, <span class="st">'sample_id2'</span>])</span>
<span id="cb84-30"><a aria-hidden="true" href="#cb84-30" tabindex="-1"></a> .merge(samples_info, on<span class="op">=</span><span class="st">'sample_id'</span>)</span>
<span id="cb84-31"><a aria-hidden="true" href="#cb84-31" tabindex="-1"></a> .merge(samples_info, left_on<span class="op">=</span><span class="st">'sample_id2'</span>, right_on<span class="op">=</span><span class="st">'sample_id'</span>, suffixes<span class="op">=</span>[<span class="st">''</span>, <span class="st">'_2'</span>])</span>
<span id="cb84-32"><a aria-hidden="true" href="#cb84-32" tabindex="-1"></a> .assign(timedelta<span class="op">=</span><span class="kw">lambda</span> dd: pd.to_datetime(dd.date_2) <span class="op">-</span> pd.to_datetime(dd.date))</span>
<span id="cb84-33"><a aria-hidden="true" href="#cb84-33" tabindex="-1"></a> .assign(combi<span class="op">=</span><span class="kw">lambda</span> dd: dd[<span class="st">'sample_mode'</span>] <span class="op">+</span> <span class="vs">r'$\rightarrow$'</span> <span class="op">+</span> dd[<span class="st">'sample_mode_2'</span>])</span>
<span id="cb84-34"><a aria-hidden="true" href="#cb84-34" tabindex="-1"></a>)</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="75" id="cell-118">
<div class="sourceCode cell-code" id="cb85"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb85-1"><a aria-hidden="true" href="#cb85-1" tabindex="-1"></a>(bc_df</span>
<span id="cb85-2"><a aria-hidden="true" href="#cb85-2" tabindex="-1"></a> .query(<span class="st">'sample_id &lt; sample_id2'</span>)</span>
<span id="cb85-3"><a aria-hidden="true" href="#cb85-3" tabindex="-1"></a> .assign(label<span class="op">=</span><span class="kw">lambda</span> dd: dd.distance.astype(<span class="bu">str</span>).<span class="bu">str</span>[<span class="dv">1</span>:<span class="dv">4</span>])</span>
<span id="cb85-4"><a aria-hidden="true" href="#cb85-4" tabindex="-1"></a> .pipe(<span class="kw">lambda</span> dd: p9.ggplot(dd)</span>
<span id="cb85-5"><a aria-hidden="true" href="#cb85-5" tabindex="-1"></a>        <span class="op">+</span> p9.aes(<span class="st">'sample_id'</span>, <span class="st">'sample_id2'</span>, fill<span class="op">=</span><span class="st">'distance'</span>)</span>
<span id="cb85-6"><a aria-hidden="true" href="#cb85-6" tabindex="-1"></a>        <span class="op">+</span> p9.geom_tile()</span>
<span id="cb85-7"><a aria-hidden="true" href="#cb85-7" tabindex="-1"></a>        <span class="op">+</span> p9.scale_fill_continuous(<span class="st">'inferno'</span>, na_value<span class="op">=</span><span class="st">'white'</span>)</span>
<span id="cb85-8"><a aria-hidden="true" href="#cb85-8" tabindex="-1"></a>        <span class="op">+</span> p9.theme_classic()</span>
<span id="cb85-9"><a aria-hidden="true" href="#cb85-9" tabindex="-1"></a>        <span class="op">+</span> p9.annotate(geom<span class="op">=</span><span class="st">'hline'</span>, yintercept<span class="op">=</span><span class="fl">10.5</span>, linetype<span class="op">=</span><span class="st">'dashed'</span>)</span>
<span id="cb85-10"><a aria-hidden="true" href="#cb85-10" tabindex="-1"></a>        <span class="op">+</span> p9.annotate(geom<span class="op">=</span><span class="st">'vline'</span>, xintercept<span class="op">=</span><span class="fl">11.5</span>, linetype<span class="op">=</span><span class="st">'dashed'</span>)</span>
<span id="cb85-11"><a aria-hidden="true" href="#cb85-11" tabindex="-1"></a>        <span class="op">+</span> p9.geom_text(p9.aes(label<span class="op">=</span><span class="st">'label'</span>, color<span class="op">=</span><span class="st">'distance &gt;= .35'</span>), size<span class="op">=</span><span class="dv">8</span>)</span>
<span id="cb85-12"><a aria-hidden="true" href="#cb85-12" tabindex="-1"></a>        <span class="op">+</span> p9.scale_color_manual([<span class="st">'white'</span>, <span class="st">'black'</span>])</span>
<span id="cb85-13"><a aria-hidden="true" href="#cb85-13" tabindex="-1"></a>        <span class="op">+</span> p9.guides(fill<span class="op">=</span><span class="va">False</span>, color<span class="op">=</span><span class="va">False</span>)</span>
<span id="cb85-14"><a aria-hidden="true" href="#cb85-14" tabindex="-1"></a>        <span class="op">+</span> p9.labs(x<span class="op">=</span><span class="st">''</span>, y<span class="op">=</span><span class="st">''</span>, title<span class="op">=</span><span class="st">'Bray-Curtis dissimilarity (ITS + 16S)'</span>)</span>
<span id="cb85-15"><a aria-hidden="true" href="#cb85-15" tabindex="-1"></a>        <span class="op">+</span> p9.theme(</span>
<span id="cb85-16"><a aria-hidden="true" href="#cb85-16" tabindex="-1"></a>            axis_text_x<span class="op">=</span>p9.element_text(angle<span class="op">=</span><span class="dv">90</span>),</span>
<span id="cb85-17"><a aria-hidden="true" href="#cb85-17" tabindex="-1"></a>            figure_size<span class="op">=</span>(<span class="dv">5</span>, <span class="dv">5</span>),</span>
<span id="cb85-18"><a aria-hidden="true" href="#cb85-18" tabindex="-1"></a>            dpi<span class="op">=</span><span class="dv">400</span>,</span>
<span id="cb85-19"><a aria-hidden="true" href="#cb85-19" tabindex="-1"></a>            title<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">12</span>, ha<span class="op">=</span><span class="st">'center'</span>),</span>
<span id="cb85-20"><a aria-hidden="true" href="#cb85-20" tabindex="-1"></a>            )</span>
<span id="cb85-21"><a aria-hidden="true" href="#cb85-21" tabindex="-1"></a>      )</span>
<span id="cb85-22"><a aria-hidden="true" href="#cb85-22" tabindex="-1"></a>)</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img class="figure-img" height="2000" src="microbial_diversity_analysis_files/figure-html/cell-83-output-1.png" width="2000"/></p>
</figure>
</div>
</div>
</div>
<div class="cell" data-execution_count="76" id="cell-119">
<div class="sourceCode cell-code" id="cb86"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb86-1"><a aria-hidden="true" href="#cb86-1" tabindex="-1"></a>(bc_df_16s</span>
<span id="cb86-2"><a aria-hidden="true" href="#cb86-2" tabindex="-1"></a> .query(<span class="st">'sample_id &lt; sample_id2'</span>)</span>
<span id="cb86-3"><a aria-hidden="true" href="#cb86-3" tabindex="-1"></a> .assign(label<span class="op">=</span><span class="kw">lambda</span> dd: dd.distance.astype(<span class="bu">str</span>).<span class="bu">str</span>[<span class="dv">1</span>:<span class="dv">4</span>])</span>
<span id="cb86-4"><a aria-hidden="true" href="#cb86-4" tabindex="-1"></a> .pipe(<span class="kw">lambda</span> dd: p9.ggplot(dd)</span>
<span id="cb86-5"><a aria-hidden="true" href="#cb86-5" tabindex="-1"></a>        <span class="op">+</span> p9.aes(<span class="st">'sample_id'</span>, <span class="st">'sample_id2'</span>, fill<span class="op">=</span><span class="st">'distance'</span>)</span>
<span id="cb86-6"><a aria-hidden="true" href="#cb86-6" tabindex="-1"></a>        <span class="op">+</span> p9.geom_tile()</span>
<span id="cb86-7"><a aria-hidden="true" href="#cb86-7" tabindex="-1"></a>        <span class="op">+</span> p9.scale_fill_continuous(<span class="st">'inferno'</span>, na_value<span class="op">=</span><span class="st">'white'</span>)</span>
<span id="cb86-8"><a aria-hidden="true" href="#cb86-8" tabindex="-1"></a>        <span class="op">+</span> p9.theme_classic()</span>
<span id="cb86-9"><a aria-hidden="true" href="#cb86-9" tabindex="-1"></a>        <span class="op">+</span> p9.annotate(geom<span class="op">=</span><span class="st">'hline'</span>, yintercept<span class="op">=</span><span class="fl">10.5</span>, linetype<span class="op">=</span><span class="st">'dashed'</span>)</span>
<span id="cb86-10"><a aria-hidden="true" href="#cb86-10" tabindex="-1"></a>        <span class="op">+</span> p9.annotate(geom<span class="op">=</span><span class="st">'vline'</span>, xintercept<span class="op">=</span><span class="fl">11.5</span>, linetype<span class="op">=</span><span class="st">'dashed'</span>)</span>
<span id="cb86-11"><a aria-hidden="true" href="#cb86-11" tabindex="-1"></a>        <span class="op">+</span> p9.geom_text(p9.aes(label<span class="op">=</span><span class="st">'label'</span>, color<span class="op">=</span><span class="st">'distance &gt;= .35'</span>), size<span class="op">=</span><span class="dv">8</span>)</span>
<span id="cb86-12"><a aria-hidden="true" href="#cb86-12" tabindex="-1"></a>        <span class="op">+</span> p9.scale_color_manual([<span class="st">'white'</span>, <span class="st">'black'</span>])</span>
<span id="cb86-13"><a aria-hidden="true" href="#cb86-13" tabindex="-1"></a>        <span class="op">+</span> p9.guides(fill<span class="op">=</span><span class="va">False</span>, color<span class="op">=</span><span class="va">False</span>)</span>
<span id="cb86-14"><a aria-hidden="true" href="#cb86-14" tabindex="-1"></a>        <span class="op">+</span> p9.labs(x<span class="op">=</span><span class="st">''</span>, y<span class="op">=</span><span class="st">''</span>, title<span class="op">=</span><span class="st">'Bray-Curtis dissimilarity (16S-only)'</span>)</span>
<span id="cb86-15"><a aria-hidden="true" href="#cb86-15" tabindex="-1"></a>        <span class="op">+</span> p9.theme(</span>
<span id="cb86-16"><a aria-hidden="true" href="#cb86-16" tabindex="-1"></a>            axis_text_x<span class="op">=</span>p9.element_text(angle<span class="op">=</span><span class="dv">90</span>),</span>
<span id="cb86-17"><a aria-hidden="true" href="#cb86-17" tabindex="-1"></a>            figure_size<span class="op">=</span>(<span class="dv">5</span>, <span class="dv">5</span>),</span>
<span id="cb86-18"><a aria-hidden="true" href="#cb86-18" tabindex="-1"></a>            dpi<span class="op">=</span><span class="dv">400</span>,</span>
<span id="cb86-19"><a aria-hidden="true" href="#cb86-19" tabindex="-1"></a>            title<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">12</span>, ha<span class="op">=</span><span class="st">'center'</span>),</span>
<span id="cb86-20"><a aria-hidden="true" href="#cb86-20" tabindex="-1"></a>            )</span>
<span id="cb86-21"><a aria-hidden="true" href="#cb86-21" tabindex="-1"></a>      )</span>
<span id="cb86-22"><a aria-hidden="true" href="#cb86-22" tabindex="-1"></a>)</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img class="figure-img" height="2000" src="microbial_diversity_analysis_files/figure-html/cell-84-output-1.png" width="2000"/></p>
</figure>
</div>
</div>
</div>
<div class="cell" data-execution_count="77" id="cell-120">
<div class="sourceCode cell-code" id="cb87"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb87-1"><a aria-hidden="true" href="#cb87-1" tabindex="-1"></a>(bc_df_its</span>
<span id="cb87-2"><a aria-hidden="true" href="#cb87-2" tabindex="-1"></a> .query(<span class="st">'sample_id &lt; sample_id2'</span>)</span>
<span id="cb87-3"><a aria-hidden="true" href="#cb87-3" tabindex="-1"></a> .assign(label<span class="op">=</span><span class="kw">lambda</span> dd: dd.distance.astype(<span class="bu">str</span>).<span class="bu">str</span>[<span class="dv">1</span>:<span class="dv">4</span>])</span>
<span id="cb87-4"><a aria-hidden="true" href="#cb87-4" tabindex="-1"></a> .pipe(<span class="kw">lambda</span> dd: p9.ggplot(dd)</span>
<span id="cb87-5"><a aria-hidden="true" href="#cb87-5" tabindex="-1"></a>        <span class="op">+</span> p9.aes(<span class="st">'sample_id'</span>, <span class="st">'sample_id2'</span>, fill<span class="op">=</span><span class="st">'distance'</span>)</span>
<span id="cb87-6"><a aria-hidden="true" href="#cb87-6" tabindex="-1"></a>        <span class="op">+</span> p9.geom_tile()</span>
<span id="cb87-7"><a aria-hidden="true" href="#cb87-7" tabindex="-1"></a>        <span class="op">+</span> p9.scale_fill_continuous(<span class="st">'inferno'</span>, na_value<span class="op">=</span><span class="st">'white'</span>)</span>
<span id="cb87-8"><a aria-hidden="true" href="#cb87-8" tabindex="-1"></a>        <span class="op">+</span> p9.theme_classic()</span>
<span id="cb87-9"><a aria-hidden="true" href="#cb87-9" tabindex="-1"></a>        <span class="op">+</span> p9.annotate(geom<span class="op">=</span><span class="st">'hline'</span>, yintercept<span class="op">=</span><span class="fl">10.5</span>, linetype<span class="op">=</span><span class="st">'dashed'</span>)</span>
<span id="cb87-10"><a aria-hidden="true" href="#cb87-10" tabindex="-1"></a>        <span class="op">+</span> p9.annotate(geom<span class="op">=</span><span class="st">'vline'</span>, xintercept<span class="op">=</span><span class="fl">11.5</span>, linetype<span class="op">=</span><span class="st">'dashed'</span>)</span>
<span id="cb87-11"><a aria-hidden="true" href="#cb87-11" tabindex="-1"></a>        <span class="op">+</span> p9.geom_text(p9.aes(label<span class="op">=</span><span class="st">'label'</span>, color<span class="op">=</span><span class="st">'distance &gt;= .35'</span>), size<span class="op">=</span><span class="dv">8</span>)</span>
<span id="cb87-12"><a aria-hidden="true" href="#cb87-12" tabindex="-1"></a>        <span class="op">+</span> p9.scale_color_manual([<span class="st">'white'</span>, <span class="st">'black'</span>])</span>
<span id="cb87-13"><a aria-hidden="true" href="#cb87-13" tabindex="-1"></a>        <span class="op">+</span> p9.guides(fill<span class="op">=</span><span class="va">False</span>, color<span class="op">=</span><span class="va">False</span>)</span>
<span id="cb87-14"><a aria-hidden="true" href="#cb87-14" tabindex="-1"></a>        <span class="op">+</span> p9.labs(x<span class="op">=</span><span class="st">''</span>, y<span class="op">=</span><span class="st">''</span>, title<span class="op">=</span><span class="st">'Bray-Curtis dissimilarity (ITS-only)'</span>)</span>
<span id="cb87-15"><a aria-hidden="true" href="#cb87-15" tabindex="-1"></a>        <span class="op">+</span> p9.theme(</span>
<span id="cb87-16"><a aria-hidden="true" href="#cb87-16" tabindex="-1"></a>            axis_text_x<span class="op">=</span>p9.element_text(angle<span class="op">=</span><span class="dv">90</span>),</span>
<span id="cb87-17"><a aria-hidden="true" href="#cb87-17" tabindex="-1"></a>            figure_size<span class="op">=</span>(<span class="dv">5</span>, <span class="dv">5</span>),</span>
<span id="cb87-18"><a aria-hidden="true" href="#cb87-18" tabindex="-1"></a>            dpi<span class="op">=</span><span class="dv">400</span>,</span>
<span id="cb87-19"><a aria-hidden="true" href="#cb87-19" tabindex="-1"></a>            title<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">12</span>, ha<span class="op">=</span><span class="st">'center'</span>),</span>
<span id="cb87-20"><a aria-hidden="true" href="#cb87-20" tabindex="-1"></a>            )</span>
<span id="cb87-21"><a aria-hidden="true" href="#cb87-21" tabindex="-1"></a>      )</span>
<span id="cb87-22"><a aria-hidden="true" href="#cb87-22" tabindex="-1"></a>)</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img class="figure-img" height="2000" src="microbial_diversity_analysis_files/figure-html/cell-85-output-1.png" width="2000"/></p>
</figure>
</div>
</div>
</div>
<div class="cell" data-execution_count="78" id="cell-121">
<div class="sourceCode cell-code" id="cb88"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb88-1"><a aria-hidden="true" href="#cb88-1" tabindex="-1"></a>(bc_df</span>
<span id="cb88-2"><a aria-hidden="true" href="#cb88-2" tabindex="-1"></a> .loc[<span class="kw">lambda</span> dd: dd[<span class="st">'timedelta'</span>] <span class="op">&gt;=</span> pd.Timedelta(<span class="dv">0</span>)]</span>
<span id="cb88-3"><a aria-hidden="true" href="#cb88-3" tabindex="-1"></a> .query(<span class="st">'sample_id &lt; sample_id2'</span>)</span>
<span id="cb88-4"><a aria-hidden="true" href="#cb88-4" tabindex="-1"></a> .assign(label<span class="op">=</span><span class="kw">lambda</span> dd: dd[<span class="st">'sample_id'</span>] <span class="op">+</span> <span class="vs">r'$\rightarrow$'</span> <span class="op">+</span> dd[<span class="st">'sample_id2'</span>])</span>
<span id="cb88-5"><a aria-hidden="true" href="#cb88-5" tabindex="-1"></a> .sort_values(<span class="st">'distance'</span>)</span>
<span id="cb88-6"><a aria-hidden="true" href="#cb88-6" tabindex="-1"></a>  .iloc[[<span class="op">-</span>i <span class="cf">for</span> i <span class="kw">in</span> <span class="bu">range</span>(<span class="dv">1</span>, <span class="dv">21</span>)] <span class="op">+</span> [i <span class="cf">for</span> i <span class="kw">in</span> <span class="bu">reversed</span>(<span class="bu">range</span>(<span class="dv">22</span>))]]</span>
<span id="cb88-7"><a aria-hidden="true" href="#cb88-7" tabindex="-1"></a> .assign(label<span class="op">=</span><span class="kw">lambda</span> dd: pd.Categorical(dd.label, categories<span class="op">=</span>dd.label, ordered<span class="op">=</span><span class="va">True</span>))</span>
<span id="cb88-8"><a aria-hidden="true" href="#cb88-8" tabindex="-1"></a>  .assign(time_dist<span class="op">=</span><span class="kw">lambda</span> dd: dd.timedelta.<span class="bu">abs</span>().astype(<span class="bu">str</span>).<span class="bu">str</span>.split(<span class="st">' '</span>).<span class="bu">str</span>[:<span class="dv">2</span>].<span class="bu">str</span>.join(<span class="st">' '</span>))</span>
<span id="cb88-9"><a aria-hidden="true" href="#cb88-9" tabindex="-1"></a>   .assign(group<span class="op">=</span><span class="kw">lambda</span> dd: dd.distance.<span class="bu">apply</span>(<span class="kw">lambda</span> x: <span class="st">'(1) Most similar'</span> <span class="cf">if</span> x <span class="op">&lt;</span> <span class="fl">.3</span></span>
<span id="cb88-10"><a aria-hidden="true" href="#cb88-10" tabindex="-1"></a>                                         <span class="cf">else</span> <span class="st">'(2) Least Similar'</span>))</span>
<span id="cb88-11"><a aria-hidden="true" href="#cb88-11" tabindex="-1"></a> .pipe(<span class="kw">lambda</span> dd: p9.ggplot(dd)</span>
<span id="cb88-12"><a aria-hidden="true" href="#cb88-12" tabindex="-1"></a>       <span class="op">+</span> p9.aes(<span class="st">'label'</span>, <span class="st">'distance'</span>, fill<span class="op">=</span><span class="st">'combi'</span>)</span>
<span id="cb88-13"><a aria-hidden="true" href="#cb88-13" tabindex="-1"></a>       <span class="op">+</span> p9.coord_flip()</span>
<span id="cb88-14"><a aria-hidden="true" href="#cb88-14" tabindex="-1"></a>       <span class="op">+</span> p9.geom_col()</span>
<span id="cb88-15"><a aria-hidden="true" href="#cb88-15" tabindex="-1"></a>       <span class="op">+</span> p9.scale_y_continuous(labels<span class="op">=</span>percent_format(), limits<span class="op">=</span>(<span class="dv">0</span>, <span class="fl">1.1</span>), breaks<span class="op">=</span>[<span class="dv">0</span>, <span class="dv">1</span><span class="op">/</span><span class="dv">3</span>, <span class="dv">2</span><span class="op">/</span><span class="dv">3</span>, <span class="dv">1</span>])</span>
<span id="cb88-16"><a aria-hidden="true" href="#cb88-16" tabindex="-1"></a>       <span class="op">+</span> p9.facet_wrap(<span class="st">'group'</span>, ncol<span class="op">=</span><span class="dv">1</span>, scales<span class="op">=</span><span class="st">'free_y'</span>)</span>
<span id="cb88-17"><a aria-hidden="true" href="#cb88-17" tabindex="-1"></a>       <span class="op">+</span> p9.scale_fill_manual(values<span class="op">=</span>[<span class="st">'#D3894C'</span>, <span class="st">'#435B97'</span>, <span class="st">'#ab2422'</span>, <span class="st">'#32a86d'</span>])</span>
<span id="cb88-18"><a aria-hidden="true" href="#cb88-18" tabindex="-1"></a>       <span class="op">+</span> p9.geom_text(p9.aes(label<span class="op">=</span><span class="st">'time_dist'</span>, y<span class="op">=</span><span class="st">'distance + .01'</span>), size<span class="op">=</span><span class="dv">7</span>, ha<span class="op">=</span><span class="st">'left'</span>)</span>
<span id="cb88-19"><a aria-hidden="true" href="#cb88-19" tabindex="-1"></a>       <span class="op">+</span> p9.labs(x<span class="op">=</span><span class="st">''</span>, y<span class="op">=</span><span class="st">'Bray Curtis dissimilarity'</span>, fill<span class="op">=</span><span class="st">''</span>)</span>
<span id="cb88-20"><a aria-hidden="true" href="#cb88-20" tabindex="-1"></a>       <span class="op">+</span> p9.ggtitle(<span class="st">'ITS + 16S'</span>)</span>
<span id="cb88-21"><a aria-hidden="true" href="#cb88-21" tabindex="-1"></a>       <span class="op">+</span> p9.theme(figure_size<span class="op">=</span>(<span class="dv">4</span>, <span class="dv">5</span>),</span>
<span id="cb88-22"><a aria-hidden="true" href="#cb88-22" tabindex="-1"></a>                  axis_text_y<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">8</span>),</span>
<span id="cb88-23"><a aria-hidden="true" href="#cb88-23" tabindex="-1"></a>                  axis_title_x<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">10</span>),</span>
<span id="cb88-24"><a aria-hidden="true" href="#cb88-24" tabindex="-1"></a>                  legend_title_align<span class="op">=</span><span class="st">'center'</span>,</span>
<span id="cb88-25"><a aria-hidden="true" href="#cb88-25" tabindex="-1"></a>                  legend_key_size<span class="op">=</span><span class="dv">6</span>,</span>
<span id="cb88-26"><a aria-hidden="true" href="#cb88-26" tabindex="-1"></a>                  legend_text<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">8</span>),</span>
<span id="cb88-27"><a aria-hidden="true" href="#cb88-27" tabindex="-1"></a>                  title<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">11</span>, ha<span class="op">=</span><span class="st">'center'</span>),</span>
<span id="cb88-28"><a aria-hidden="true" href="#cb88-28" tabindex="-1"></a>       )</span>
<span id="cb88-29"><a aria-hidden="true" href="#cb88-29" tabindex="-1"></a>)</span>
<span id="cb88-30"><a aria-hidden="true" href="#cb88-30" tabindex="-1"></a>)</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img class="figure-img" height="1500" src="microbial_diversity_analysis_files/figure-html/cell-86-output-1.png" width="1200"/></p>
</figure>
</div>
</div>
</div>
<div class="cell" id="cell-122">
<div class="sourceCode cell-code" id="cb89"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb89-1"><a aria-hidden="true" href="#cb89-1" tabindex="-1"></a><span class="bu">range</span>(<span class="dv">21</span>)</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="1279">
<pre><code>range(0, 21)</code></pre>
</div>
</div>
<div class="cell" id="cell-123">
<div class="sourceCode cell-code" id="cb91"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb91-1"><a aria-hidden="true" href="#cb91-1" tabindex="-1"></a>(bc_df_16s</span>
<span id="cb91-2"><a aria-hidden="true" href="#cb91-2" tabindex="-1"></a> .loc[<span class="kw">lambda</span> dd: dd[<span class="st">'timedelta'</span>] <span class="op">&gt;=</span> pd.Timedelta(<span class="dv">0</span>)]</span>
<span id="cb91-3"><a aria-hidden="true" href="#cb91-3" tabindex="-1"></a> .query(<span class="st">'sample_id &lt; sample_id2'</span>)</span>
<span id="cb91-4"><a aria-hidden="true" href="#cb91-4" tabindex="-1"></a> .assign(label<span class="op">=</span><span class="kw">lambda</span> dd: dd[<span class="st">'sample_id'</span>] <span class="op">+</span> <span class="vs">r'$\rightarrow$'</span> <span class="op">+</span> dd[<span class="st">'sample_id2'</span>])</span>
<span id="cb91-5"><a aria-hidden="true" href="#cb91-5" tabindex="-1"></a> .sort_values(<span class="st">'distance'</span>)</span>
<span id="cb91-6"><a aria-hidden="true" href="#cb91-6" tabindex="-1"></a>  .iloc[[<span class="op">-</span>i <span class="cf">for</span> i <span class="kw">in</span> <span class="bu">range</span>(<span class="dv">1</span>, <span class="dv">21</span>)] <span class="op">+</span> [i <span class="cf">for</span> i <span class="kw">in</span> <span class="bu">reversed</span>(<span class="bu">range</span>(<span class="dv">21</span>))]]</span>
<span id="cb91-7"><a aria-hidden="true" href="#cb91-7" tabindex="-1"></a> .assign(label<span class="op">=</span><span class="kw">lambda</span> dd: pd.Categorical(dd.label, categories<span class="op">=</span>dd.label, ordered<span class="op">=</span><span class="va">True</span>))</span>
<span id="cb91-8"><a aria-hidden="true" href="#cb91-8" tabindex="-1"></a>  .assign(time_dist<span class="op">=</span><span class="kw">lambda</span> dd: dd.timedelta.<span class="bu">abs</span>().astype(<span class="bu">str</span>).<span class="bu">str</span>.split(<span class="st">' '</span>).<span class="bu">str</span>[:<span class="dv">2</span>].<span class="bu">str</span>.join(<span class="st">' '</span>))</span>
<span id="cb91-9"><a aria-hidden="true" href="#cb91-9" tabindex="-1"></a>   .assign(group<span class="op">=</span><span class="kw">lambda</span> dd: dd.distance.<span class="bu">apply</span>(<span class="kw">lambda</span> x: <span class="st">'(1) Most similar'</span> <span class="cf">if</span> x <span class="op">&lt;</span> <span class="fl">.29</span></span>
<span id="cb91-10"><a aria-hidden="true" href="#cb91-10" tabindex="-1"></a>                                         <span class="cf">else</span> <span class="st">'(2) Least Similar'</span>))</span>
<span id="cb91-11"><a aria-hidden="true" href="#cb91-11" tabindex="-1"></a> .pipe(<span class="kw">lambda</span> dd: p9.ggplot(dd)</span>
<span id="cb91-12"><a aria-hidden="true" href="#cb91-12" tabindex="-1"></a>       <span class="op">+</span> p9.aes(<span class="st">'label'</span>, <span class="st">'distance'</span>, fill<span class="op">=</span><span class="st">'combi'</span>)</span>
<span id="cb91-13"><a aria-hidden="true" href="#cb91-13" tabindex="-1"></a>       <span class="op">+</span> p9.coord_flip()</span>
<span id="cb91-14"><a aria-hidden="true" href="#cb91-14" tabindex="-1"></a>       <span class="op">+</span> p9.geom_col()</span>
<span id="cb91-15"><a aria-hidden="true" href="#cb91-15" tabindex="-1"></a>       <span class="op">+</span> p9.scale_y_continuous(labels<span class="op">=</span>percent_format(), limits<span class="op">=</span>(<span class="dv">0</span>, <span class="fl">1.1</span>), breaks<span class="op">=</span>[<span class="dv">0</span>, <span class="dv">1</span><span class="op">/</span><span class="dv">3</span>, <span class="dv">2</span><span class="op">/</span><span class="dv">3</span>, <span class="dv">1</span>])</span>
<span id="cb91-16"><a aria-hidden="true" href="#cb91-16" tabindex="-1"></a>       <span class="op">+</span> p9.facet_wrap(<span class="st">'group'</span>, ncol<span class="op">=</span><span class="dv">1</span>, scales<span class="op">=</span><span class="st">'free_y'</span>)</span>
<span id="cb91-17"><a aria-hidden="true" href="#cb91-17" tabindex="-1"></a>       <span class="op">+</span> p9.scale_fill_manual(values<span class="op">=</span>[<span class="st">'#D3894C'</span>, <span class="st">'#435B97'</span>, <span class="st">'#ab2422'</span>, <span class="st">'#32a86d'</span>])</span>
<span id="cb91-18"><a aria-hidden="true" href="#cb91-18" tabindex="-1"></a>       <span class="op">+</span> p9.geom_text(p9.aes(label<span class="op">=</span><span class="st">'time_dist'</span>, y<span class="op">=</span><span class="st">'distance + .01'</span>), size<span class="op">=</span><span class="dv">7</span>, ha<span class="op">=</span><span class="st">'left'</span>)</span>
<span id="cb91-19"><a aria-hidden="true" href="#cb91-19" tabindex="-1"></a>       <span class="op">+</span> p9.labs(x<span class="op">=</span><span class="st">''</span>, y<span class="op">=</span><span class="st">'Bray Curtis dissimilarity'</span>, fill<span class="op">=</span><span class="st">''</span>)</span>
<span id="cb91-20"><a aria-hidden="true" href="#cb91-20" tabindex="-1"></a>       <span class="op">+</span> p9.ggtitle(<span class="st">'16S-only'</span>)</span>
<span id="cb91-21"><a aria-hidden="true" href="#cb91-21" tabindex="-1"></a>       <span class="op">+</span> p9.theme(figure_size<span class="op">=</span>(<span class="dv">4</span>, <span class="dv">5</span>),</span>
<span id="cb91-22"><a aria-hidden="true" href="#cb91-22" tabindex="-1"></a>                  axis_text_y<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">8</span>),</span>
<span id="cb91-23"><a aria-hidden="true" href="#cb91-23" tabindex="-1"></a>                  axis_title_x<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">10</span>),</span>
<span id="cb91-24"><a aria-hidden="true" href="#cb91-24" tabindex="-1"></a>                  legend_title_align<span class="op">=</span><span class="st">'center'</span>,</span>
<span id="cb91-25"><a aria-hidden="true" href="#cb91-25" tabindex="-1"></a>                  legend_key_size<span class="op">=</span><span class="dv">6</span>,</span>
<span id="cb91-26"><a aria-hidden="true" href="#cb91-26" tabindex="-1"></a>                  legend_text<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">8</span>),</span>
<span id="cb91-27"><a aria-hidden="true" href="#cb91-27" tabindex="-1"></a>                  title<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">11</span>, ha<span class="op">=</span><span class="st">'center'</span>),</span>
<span id="cb91-28"><a aria-hidden="true" href="#cb91-28" tabindex="-1"></a>       )</span>
<span id="cb91-29"><a aria-hidden="true" href="#cb91-29" tabindex="-1"></a>)</span>
<span id="cb91-30"><a aria-hidden="true" href="#cb91-30" tabindex="-1"></a>)</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img class="figure-img" height="1500" src="microbial_diversity_analysis_files/figure-html/cell-88-output-1.png" width="1200"/></p>
</figure>
</div>
</div>
</div>
<div class="cell" id="cell-124">
<div class="sourceCode cell-code" id="cb92"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb92-1"><a aria-hidden="true" href="#cb92-1" tabindex="-1"></a>(bc_df_its</span>
<span id="cb92-2"><a aria-hidden="true" href="#cb92-2" tabindex="-1"></a> .loc[<span class="kw">lambda</span> dd: dd[<span class="st">'timedelta'</span>] <span class="op">&gt;=</span> pd.Timedelta(<span class="dv">0</span>)]</span>
<span id="cb92-3"><a aria-hidden="true" href="#cb92-3" tabindex="-1"></a> .query(<span class="st">'sample_id &lt; sample_id2'</span>)</span>
<span id="cb92-4"><a aria-hidden="true" href="#cb92-4" tabindex="-1"></a> .assign(label<span class="op">=</span><span class="kw">lambda</span> dd: dd[<span class="st">'sample_id'</span>] <span class="op">+</span> <span class="vs">r'$\rightarrow$'</span> <span class="op">+</span> dd[<span class="st">'sample_id2'</span>])</span>
<span id="cb92-5"><a aria-hidden="true" href="#cb92-5" tabindex="-1"></a> .sort_values(<span class="st">'distance'</span>)</span>
<span id="cb92-6"><a aria-hidden="true" href="#cb92-6" tabindex="-1"></a>  .iloc[[<span class="op">-</span>i <span class="cf">for</span> i <span class="kw">in</span> <span class="bu">range</span>(<span class="dv">1</span>, <span class="dv">21</span>)] <span class="op">+</span> [i <span class="cf">for</span> i <span class="kw">in</span> <span class="bu">reversed</span>(<span class="bu">range</span>(<span class="dv">20</span>))]]</span>
<span id="cb92-7"><a aria-hidden="true" href="#cb92-7" tabindex="-1"></a> .assign(label<span class="op">=</span><span class="kw">lambda</span> dd: pd.Categorical(dd.label, categories<span class="op">=</span>dd.label, ordered<span class="op">=</span><span class="va">True</span>))</span>
<span id="cb92-8"><a aria-hidden="true" href="#cb92-8" tabindex="-1"></a>  .assign(time_dist<span class="op">=</span><span class="kw">lambda</span> dd: dd.timedelta.<span class="bu">abs</span>().astype(<span class="bu">str</span>).<span class="bu">str</span>.split(<span class="st">' '</span>).<span class="bu">str</span>[:<span class="dv">2</span>].<span class="bu">str</span>.join(<span class="st">' '</span>))</span>
<span id="cb92-9"><a aria-hidden="true" href="#cb92-9" tabindex="-1"></a>   .assign(group<span class="op">=</span><span class="kw">lambda</span> dd: dd.distance.<span class="bu">apply</span>(<span class="kw">lambda</span> x: <span class="st">'(1) Most similar'</span> <span class="cf">if</span> x <span class="op">&lt;</span> <span class="fl">.3</span></span>
<span id="cb92-10"><a aria-hidden="true" href="#cb92-10" tabindex="-1"></a>                                         <span class="cf">else</span> <span class="st">'(2) Least Similar'</span>))</span>
<span id="cb92-11"><a aria-hidden="true" href="#cb92-11" tabindex="-1"></a> .pipe(<span class="kw">lambda</span> dd: p9.ggplot(dd)</span>
<span id="cb92-12"><a aria-hidden="true" href="#cb92-12" tabindex="-1"></a>       <span class="op">+</span> p9.aes(<span class="st">'label'</span>, <span class="st">'distance'</span>, fill<span class="op">=</span><span class="st">'combi'</span>)</span>
<span id="cb92-13"><a aria-hidden="true" href="#cb92-13" tabindex="-1"></a>       <span class="op">+</span> p9.coord_flip()</span>
<span id="cb92-14"><a aria-hidden="true" href="#cb92-14" tabindex="-1"></a>       <span class="op">+</span> p9.geom_col()</span>
<span id="cb92-15"><a aria-hidden="true" href="#cb92-15" tabindex="-1"></a>       <span class="op">+</span> p9.scale_y_continuous(labels<span class="op">=</span>percent_format(), limits<span class="op">=</span>(<span class="dv">0</span>, <span class="fl">1.1</span>), breaks<span class="op">=</span>[<span class="dv">0</span>, <span class="dv">1</span><span class="op">/</span><span class="dv">3</span>, <span class="dv">2</span><span class="op">/</span><span class="dv">3</span>, <span class="dv">1</span>])</span>
<span id="cb92-16"><a aria-hidden="true" href="#cb92-16" tabindex="-1"></a>       <span class="op">+</span> p9.facet_wrap(<span class="st">'group'</span>, ncol<span class="op">=</span><span class="dv">1</span>, scales<span class="op">=</span><span class="st">'free_y'</span>)</span>
<span id="cb92-17"><a aria-hidden="true" href="#cb92-17" tabindex="-1"></a>       <span class="op">+</span> p9.scale_fill_manual(values<span class="op">=</span>[<span class="st">'#D3894C'</span>, <span class="st">'#435B97'</span>, <span class="st">'#ab2422'</span>, <span class="st">'#32a86d'</span>])</span>
<span id="cb92-18"><a aria-hidden="true" href="#cb92-18" tabindex="-1"></a>       <span class="op">+</span> p9.geom_text(p9.aes(label<span class="op">=</span><span class="st">'time_dist'</span>, y<span class="op">=</span><span class="st">'distance + .01'</span>), size<span class="op">=</span><span class="dv">7</span>, ha<span class="op">=</span><span class="st">'left'</span>)</span>
<span id="cb92-19"><a aria-hidden="true" href="#cb92-19" tabindex="-1"></a>       <span class="op">+</span> p9.labs(x<span class="op">=</span><span class="st">''</span>, y<span class="op">=</span><span class="st">'Bray Curtis dissimilarity'</span>, fill<span class="op">=</span><span class="st">''</span>)</span>
<span id="cb92-20"><a aria-hidden="true" href="#cb92-20" tabindex="-1"></a>       <span class="op">+</span> p9.ggtitle(<span class="st">'ITS-only'</span>)</span>
<span id="cb92-21"><a aria-hidden="true" href="#cb92-21" tabindex="-1"></a>       <span class="op">+</span> p9.theme(figure_size<span class="op">=</span>(<span class="dv">4</span>, <span class="dv">5</span>),</span>
<span id="cb92-22"><a aria-hidden="true" href="#cb92-22" tabindex="-1"></a>                  axis_text_y<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">8</span>),</span>
<span id="cb92-23"><a aria-hidden="true" href="#cb92-23" tabindex="-1"></a>                  axis_title_x<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">10</span>),</span>
<span id="cb92-24"><a aria-hidden="true" href="#cb92-24" tabindex="-1"></a>                  legend_title_align<span class="op">=</span><span class="st">'center'</span>,</span>
<span id="cb92-25"><a aria-hidden="true" href="#cb92-25" tabindex="-1"></a>                  legend_key_size<span class="op">=</span><span class="dv">6</span>,</span>
<span id="cb92-26"><a aria-hidden="true" href="#cb92-26" tabindex="-1"></a>                  legend_text<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">8</span>),</span>
<span id="cb92-27"><a aria-hidden="true" href="#cb92-27" tabindex="-1"></a>                  title<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">11</span>, ha<span class="op">=</span><span class="st">'center'</span>),</span>
<span id="cb92-28"><a aria-hidden="true" href="#cb92-28" tabindex="-1"></a>       )</span>
<span id="cb92-29"><a aria-hidden="true" href="#cb92-29" tabindex="-1"></a>)</span>
<span id="cb92-30"><a aria-hidden="true" href="#cb92-30" tabindex="-1"></a>)</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img class="figure-img" height="1500" src="microbial_diversity_analysis_files/figure-html/cell-89-output-1.png" width="1200"/></p>
</figure>
</div>
</div>
</div>
</section>
</section>
</section>
<section class="level3" id="correlation-with-heavy-metals">
<h3 class="anchored" data-anchor-id="correlation-with-heavy-metals">Correlation with heavy metals</h3>
<div class="cell" data-execution_count="54" id="cell-126">
<div class="sourceCode cell-code" id="cb93"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb93-1"><a aria-hidden="true" href="#cb93-1" tabindex="-1"></a><span class="im">from</span> functools <span class="im">import</span> partial</span>
<span id="cb93-2"><a aria-hidden="true" href="#cb93-2" tabindex="-1"></a><span class="im">from</span> scipy.stats <span class="im">import</span> spearmanr</span>
<span id="cb93-3"><a aria-hidden="true" href="#cb93-3" tabindex="-1"></a><span class="im">from</span> statsmodels.stats.multitest <span class="im">import</span> fdrcorrection</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="55" id="cell-127">
<div class="sourceCode cell-code" id="cb94"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb94-1"><a aria-hidden="true" href="#cb94-1" tabindex="-1"></a>metals_matrix <span class="op">=</span> pd.read_csv(<span class="st">'../data/metals.csv'</span>, index_col<span class="op">=</span><span class="dv">0</span>)</span>
<span id="cb94-2"><a aria-hidden="true" href="#cb94-2" tabindex="-1"></a>metals_matrix.columns <span class="op">=</span> metals_matrix.columns.<span class="bu">str</span>.replace(<span class="st">'S0'</span>, <span class="st">'S'</span>, regex<span class="op">=</span><span class="va">False</span>)</span>
<span id="cb94-3"><a aria-hidden="true" href="#cb94-3" tabindex="-1"></a>metals_matrix <span class="op">=</span> metals_matrix.reset_index().replace(<span class="st">'Al2O3'</span>, <span class="st">'Al'</span>).set_index(<span class="st">'species'</span>)</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="56" id="cell-128">
<div class="sourceCode cell-code" id="cb95"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb95-1"><a aria-hidden="true" href="#cb95-1" tabindex="-1"></a>n_samples_per_metal <span class="op">=</span> (metals_matrix</span>
<span id="cb95-2"><a aria-hidden="true" href="#cb95-2" tabindex="-1"></a>                        .applymap(<span class="kw">lambda</span> x: x <span class="op">&gt;</span> <span class="dv">0</span>)</span>
<span id="cb95-3"><a aria-hidden="true" href="#cb95-3" tabindex="-1"></a>                        .<span class="bu">sum</span>(axis<span class="op">=</span><span class="dv">1</span>)</span>
<span id="cb95-4"><a aria-hidden="true" href="#cb95-4" tabindex="-1"></a>                        .rename(<span class="st">'n_samples'</span>)</span>
<span id="cb95-5"><a aria-hidden="true" href="#cb95-5" tabindex="-1"></a>                        .reset_index()</span>
<span id="cb95-6"><a aria-hidden="true" href="#cb95-6" tabindex="-1"></a>                        .assign(keep<span class="op">=</span><span class="kw">lambda</span> dd: dd.n_samples <span class="op">&gt;</span> <span class="dv">10</span>)</span>
<span id="cb95-7"><a aria-hidden="true" href="#cb95-7" tabindex="-1"></a>)</span>
<span id="cb95-8"><a aria-hidden="true" href="#cb95-8" tabindex="-1"></a></span>
<span id="cb95-9"><a aria-hidden="true" href="#cb95-9" tabindex="-1"></a>(n_samples_per_metal.pipe(<span class="kw">lambda</span> dd: p9.ggplot(dd) </span>
<span id="cb95-10"><a aria-hidden="true" href="#cb95-10" tabindex="-1"></a>                <span class="op">+</span> p9.aes(<span class="st">'reorder(species, n_samples)'</span>, <span class="st">'n_samples'</span>) </span>
<span id="cb95-11"><a aria-hidden="true" href="#cb95-11" tabindex="-1"></a>                <span class="op">+</span> p9.geom_col(p9.aes(alpha<span class="op">=</span><span class="st">'keep'</span>))</span>
<span id="cb95-12"><a aria-hidden="true" href="#cb95-12" tabindex="-1"></a>                <span class="op">+</span> p9.scale_alpha_manual(values<span class="op">=</span>[<span class="fl">.5</span>, <span class="dv">1</span>])</span>
<span id="cb95-13"><a aria-hidden="true" href="#cb95-13" tabindex="-1"></a>                <span class="op">+</span> p9.coord_flip()</span>
<span id="cb95-14"><a aria-hidden="true" href="#cb95-14" tabindex="-1"></a>                <span class="op">+</span> p9.guides(alpha<span class="op">=</span><span class="va">False</span>)</span>
<span id="cb95-15"><a aria-hidden="true" href="#cb95-15" tabindex="-1"></a>                <span class="op">+</span> p9.labs(x<span class="op">=</span><span class="st">''</span>, y<span class="op">=</span><span class="st">'Number of Samples over detection limit'</span>)</span>
<span id="cb95-16"><a aria-hidden="true" href="#cb95-16" tabindex="-1"></a>                <span class="op">+</span> p9.theme(</span>
<span id="cb95-17"><a aria-hidden="true" href="#cb95-17" tabindex="-1"></a>                    figure_size<span class="op">=</span>(<span class="dv">4</span>, <span class="dv">6</span>),</span>
<span id="cb95-18"><a aria-hidden="true" href="#cb95-18" tabindex="-1"></a>                    axis_text_y<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">8</span>),</span>
<span id="cb95-19"><a aria-hidden="true" href="#cb95-19" tabindex="-1"></a>                )</span>
<span id="cb95-20"><a aria-hidden="true" href="#cb95-20" tabindex="-1"></a>)</span>
<span id="cb95-21"><a aria-hidden="true" href="#cb95-21" tabindex="-1"></a>)</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img class="figure-img" height="1800" src="microbial_diversity_analysis_files/figure-html/cell-92-output-1.png" width="1200"/></p>
</figure>
</div>
</div>
</div>
<div class="cell" data-execution_count="57" id="cell-129">
<div class="sourceCode cell-code" id="cb96"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb96-1"><a aria-hidden="true" href="#cb96-1" tabindex="-1"></a>get_pvals <span class="op">=</span> partial(<span class="kw">lambda</span> x, y: spearmanr(x, y).pvalue)</span>
<span id="cb96-2"><a aria-hidden="true" href="#cb96-2" tabindex="-1"></a>get_spearmanr <span class="op">=</span> partial(<span class="kw">lambda</span> x, y: spearmanr(x, y).correlation)</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="58" id="cell-130">
<div class="sourceCode cell-code" id="cb97"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb97-1"><a aria-hidden="true" href="#cb97-1" tabindex="-1"></a>micro_matrix <span class="op">=</span> (abundance_df_genera</span>
<span id="cb97-2"><a aria-hidden="true" href="#cb97-2" tabindex="-1"></a> .drop(columns<span class="op">=</span><span class="st">'S06'</span>)</span>
<span id="cb97-3"><a aria-hidden="true" href="#cb97-3" tabindex="-1"></a> .query(<span class="st">'Genus != "unknown"'</span>)</span>
<span id="cb97-4"><a aria-hidden="true" href="#cb97-4" tabindex="-1"></a> .drop(columns<span class="op">=</span>taxon_names[:<span class="op">-</span><span class="dv">1</span>])</span>
<span id="cb97-5"><a aria-hidden="true" href="#cb97-5" tabindex="-1"></a> .set_index(<span class="st">'Genus'</span>)</span>
<span id="cb97-6"><a aria-hidden="true" href="#cb97-6" tabindex="-1"></a> .T</span>
<span id="cb97-7"><a aria-hidden="true" href="#cb97-7" tabindex="-1"></a>)</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="59" id="cell-131">
<div class="sourceCode cell-code" id="cb98"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb98-1"><a aria-hidden="true" href="#cb98-1" tabindex="-1"></a>micro_matrix</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="59">
<div>
<table class="dataframe caption-top table table-sm table-striped small" data-border="1" data-quarto-postprocess="true">
<thead>
<tr class="header">
<th data-quarto-table-cell-role="th">Genus</th>
<th data-quarto-table-cell-role="th">Bryocella</th>
<th data-quarto-table-cell-role="th">Granulicella</th>
<th data-quarto-table-cell-role="th">Terriglobus</th>
<th data-quarto-table-cell-role="th">Blastocatella</th>
<th data-quarto-table-cell-role="th">RB41</th>
<th data-quarto-table-cell-role="th">Actinomyces</th>
<th data-quarto-table-cell-role="th">Mobiluncus</th>
<th data-quarto-table-cell-role="th">Bifidobacterium</th>
<th data-quarto-table-cell-role="th">Gardnerella</th>
<th data-quarto-table-cell-role="th">Scardovia</th>
<th data-quarto-table-cell-role="th">...</th>
<th data-quarto-table-cell-role="th">Hannaella</th>
<th data-quarto-table-cell-role="th">Vishniacozyma</th>
<th data-quarto-table-cell-role="th">Cryptococcus</th>
<th data-quarto-table-cell-role="th">Papiliotrema</th>
<th data-quarto-table-cell-role="th">Cutaneotrichosporon</th>
<th data-quarto-table-cell-role="th">Trichosporon</th>
<th data-quarto-table-cell-role="th">Anthracocystis</th>
<th data-quarto-table-cell-role="th">Moesziomyces</th>
<th data-quarto-table-cell-role="th">Ustilago</th>
<th data-quarto-table-cell-role="th">Wallemia</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td data-quarto-table-cell-role="th">S01</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>9</td>
<td>0</td>
<td>57</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>...</td>
<td>0</td>
<td>37</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>53</td>
<td>0</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">S02</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>...</td>
<td>0</td>
<td>15</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>28</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">S03</td>
<td>37</td>
<td>200</td>
<td>166</td>
<td>0</td>
<td>0</td>
<td>3</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>...</td>
<td>0</td>
<td>2</td>
<td>0</td>
<td>46</td>
<td>0</td>
<td>0</td>
<td>172</td>
<td>52</td>
<td>90</td>
<td>0</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">S04</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>...</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">S05</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>...</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">S07</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>...</td>
<td>1312</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">S08</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>...</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">S09</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>7</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>...</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">S10</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>6</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>...</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>568</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>174</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">S11</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>4</td>
<td>0</td>
<td>...</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>89</td>
<td>0</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">S12</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>...</td>
<td>130</td>
<td>0</td>
<td>0</td>
<td>23</td>
<td>31</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">S13</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>27</td>
<td>0</td>
<td>22</td>
<td>0</td>
<td>0</td>
<td>...</td>
<td>0</td>
<td>43</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">S15</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>...</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">S16</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>51</td>
<td>0</td>
<td>9</td>
<td>0</td>
<td>0</td>
<td>...</td>
<td>0</td>
<td>24</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">S17</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>...</td>
<td>0</td>
<td>13</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>65</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">S18</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>...</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">S19</td>
<td>10</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>26</td>
<td>48</td>
<td>0</td>
<td>68</td>
<td>0</td>
<td>0</td>
<td>...</td>
<td>7</td>
<td>4</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">S20</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>12</td>
<td>0</td>
<td>93</td>
<td>0</td>
<td>0</td>
<td>...</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">S21</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>117</td>
<td>0</td>
<td>131</td>
<td>0</td>
<td>26</td>
<td>...</td>
<td>0</td>
<td>0</td>
<td>325</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>785</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">S22</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>57</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>...</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
</tr>
</tbody>
</table>
<p>20 rows × 572 columns</p>
</div>
</div>
</div>
<div class="cell" data-execution_count="60" id="cell-132">
<div class="sourceCode cell-code" id="cb99"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb99-1"><a aria-hidden="true" href="#cb99-1" tabindex="-1"></a>flight_samples <span class="op">=</span> [s <span class="cf">for</span> s, i <span class="kw">in</span> sample_map.items() <span class="cf">if</span> i <span class="op">==</span> <span class="st">"Flight"</span>]</span>
<span id="cb99-2"><a aria-hidden="true" href="#cb99-2" tabindex="-1"></a>ground_samples <span class="op">=</span> [s <span class="cf">for</span> s, i <span class="kw">in</span> sample_map.items() <span class="cf">if</span> i <span class="op">==</span> <span class="st">"Ground"</span>]</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="61" id="cell-133">
<div class="sourceCode cell-code" id="cb100"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb100-1"><a aria-hidden="true" href="#cb100-1" tabindex="-1"></a>corrs_matrix <span class="op">=</span> (metals_matrix</span>
<span id="cb100-2"><a aria-hidden="true" href="#cb100-2" tabindex="-1"></a>                .reset_index()</span>
<span id="cb100-3"><a aria-hidden="true" href="#cb100-3" tabindex="-1"></a>                .drop(columns<span class="op">=</span><span class="st">'S14'</span>)</span>
<span id="cb100-4"><a aria-hidden="true" href="#cb100-4" tabindex="-1"></a>                .loc[<span class="kw">lambda</span> dd: dd.species.isin(n_samples_per_metal.query(<span class="st">'keep'</span>).species)]</span>
<span id="cb100-5"><a aria-hidden="true" href="#cb100-5" tabindex="-1"></a>                .set_index(<span class="st">'species'</span>)</span>
<span id="cb100-6"><a aria-hidden="true" href="#cb100-6" tabindex="-1"></a>                .T</span>
<span id="cb100-7"><a aria-hidden="true" href="#cb100-7" tabindex="-1"></a>                .<span class="bu">apply</span>(<span class="kw">lambda</span> dd: micro_matrix.corrwith(dd, axis<span class="op">=</span><span class="dv">0</span>, method<span class="op">=</span>get_spearmanr))</span>
<span id="cb100-8"><a aria-hidden="true" href="#cb100-8" tabindex="-1"></a>                .reset_index()</span>
<span id="cb100-9"><a aria-hidden="true" href="#cb100-9" tabindex="-1"></a>)</span>
<span id="cb100-10"><a aria-hidden="true" href="#cb100-10" tabindex="-1"></a></span>
<span id="cb100-11"><a aria-hidden="true" href="#cb100-11" tabindex="-1"></a>p_values_matrix <span class="op">=</span> (metals_matrix</span>
<span id="cb100-12"><a aria-hidden="true" href="#cb100-12" tabindex="-1"></a>                .reset_index()</span>
<span id="cb100-13"><a aria-hidden="true" href="#cb100-13" tabindex="-1"></a>                .drop(columns<span class="op">=</span><span class="st">'S14'</span>)</span>
<span id="cb100-14"><a aria-hidden="true" href="#cb100-14" tabindex="-1"></a>                .loc[<span class="kw">lambda</span> dd: dd.species.isin(n_samples_per_metal.query(<span class="st">'keep'</span>).species)]</span>
<span id="cb100-15"><a aria-hidden="true" href="#cb100-15" tabindex="-1"></a>                .set_index(<span class="st">'species'</span>)</span>
<span id="cb100-16"><a aria-hidden="true" href="#cb100-16" tabindex="-1"></a>                .T</span>
<span id="cb100-17"><a aria-hidden="true" href="#cb100-17" tabindex="-1"></a>                .<span class="bu">apply</span>(<span class="kw">lambda</span> dd: micro_matrix.corrwith(dd, axis<span class="op">=</span><span class="dv">0</span>, method<span class="op">=</span>get_pvals))</span>
<span id="cb100-18"><a aria-hidden="true" href="#cb100-18" tabindex="-1"></a>                .reset_index()</span>
<span id="cb100-19"><a aria-hidden="true" href="#cb100-19" tabindex="-1"></a>)</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="62" id="cell-134">
<div class="sourceCode cell-code" id="cb101"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb101-1"><a aria-hidden="true" href="#cb101-1" tabindex="-1"></a>flight_present_genera <span class="op">=</span> micro_matrix.loc[flight_samples].<span class="bu">map</span>(<span class="kw">lambda</span> x: x <span class="op">&gt;</span> <span class="dv">0</span>).T.<span class="bu">sum</span>(axis<span class="op">=</span><span class="dv">1</span>).loc[<span class="kw">lambda</span> x: x <span class="op">&gt;=</span> <span class="dv">2</span>].index</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="63" id="cell-135">
<div class="sourceCode cell-code" id="cb102"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb102-1"><a aria-hidden="true" href="#cb102-1" tabindex="-1"></a>corrs_matrix_flight <span class="op">=</span> (metals_matrix</span>
<span id="cb102-2"><a aria-hidden="true" href="#cb102-2" tabindex="-1"></a>                [flight_samples]</span>
<span id="cb102-3"><a aria-hidden="true" href="#cb102-3" tabindex="-1"></a>                .reset_index()</span>
<span id="cb102-4"><a aria-hidden="true" href="#cb102-4" tabindex="-1"></a>                .loc[<span class="kw">lambda</span> dd: dd.species.isin(n_samples_per_metal.query(<span class="st">'keep'</span>).species)]</span>
<span id="cb102-5"><a aria-hidden="true" href="#cb102-5" tabindex="-1"></a>                .set_index(<span class="st">'species'</span>)</span>
<span id="cb102-6"><a aria-hidden="true" href="#cb102-6" tabindex="-1"></a>                .T</span>
<span id="cb102-7"><a aria-hidden="true" href="#cb102-7" tabindex="-1"></a>                .<span class="bu">apply</span>(<span class="kw">lambda</span> dd: micro_matrix[flight_present_genera].loc[flight_samples].corrwith(dd, axis<span class="op">=</span><span class="dv">0</span>, method<span class="op">=</span>get_spearmanr))</span>
<span id="cb102-8"><a aria-hidden="true" href="#cb102-8" tabindex="-1"></a>                .reset_index()</span>
<span id="cb102-9"><a aria-hidden="true" href="#cb102-9" tabindex="-1"></a>)</span>
<span id="cb102-10"><a aria-hidden="true" href="#cb102-10" tabindex="-1"></a></span>
<span id="cb102-11"><a aria-hidden="true" href="#cb102-11" tabindex="-1"></a>p_values_matrix_flight <span class="op">=</span> (metals_matrix</span>
<span id="cb102-12"><a aria-hidden="true" href="#cb102-12" tabindex="-1"></a>                [flight_samples]</span>
<span id="cb102-13"><a aria-hidden="true" href="#cb102-13" tabindex="-1"></a>                .reset_index()</span>
<span id="cb102-14"><a aria-hidden="true" href="#cb102-14" tabindex="-1"></a>                .loc[<span class="kw">lambda</span> dd: dd.species.isin(n_samples_per_metal.query(<span class="st">'keep'</span>).species)]</span>
<span id="cb102-15"><a aria-hidden="true" href="#cb102-15" tabindex="-1"></a>                .set_index(<span class="st">'species'</span>)</span>
<span id="cb102-16"><a aria-hidden="true" href="#cb102-16" tabindex="-1"></a>                .T</span>
<span id="cb102-17"><a aria-hidden="true" href="#cb102-17" tabindex="-1"></a>                .<span class="bu">apply</span>(<span class="kw">lambda</span> dd: micro_matrix[flight_present_genera].loc[flight_samples].corrwith(dd, axis<span class="op">=</span><span class="dv">0</span>, method<span class="op">=</span>get_pvals))</span>
<span id="cb102-18"><a aria-hidden="true" href="#cb102-18" tabindex="-1"></a>                .reset_index()</span>
<span id="cb102-19"><a aria-hidden="true" href="#cb102-19" tabindex="-1"></a>)</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="64" id="cell-136">
<div class="sourceCode cell-code" id="cb103"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb103-1"><a aria-hidden="true" href="#cb103-1" tabindex="-1"></a>ground_samples.pop(ground_samples.index(<span class="st">'S06'</span>))</span>
<span id="cb103-2"><a aria-hidden="true" href="#cb103-2" tabindex="-1"></a>ground_samples.pop(ground_samples.index(<span class="st">'S14'</span>))</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="64">
<pre><code>'S14'</code></pre>
</div>
</div>
<div class="cell" data-execution_count="65" id="cell-137">
<div class="sourceCode cell-code" id="cb105"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb105-1"><a aria-hidden="true" href="#cb105-1" tabindex="-1"></a>corrs_matrix_ground <span class="op">=</span> (metals_matrix</span>
<span id="cb105-2"><a aria-hidden="true" href="#cb105-2" tabindex="-1"></a>                [ground_samples]</span>
<span id="cb105-3"><a aria-hidden="true" href="#cb105-3" tabindex="-1"></a>                .reset_index()</span>
<span id="cb105-4"><a aria-hidden="true" href="#cb105-4" tabindex="-1"></a>                .loc[<span class="kw">lambda</span> dd: dd.species.isin(n_samples_per_metal.query(<span class="st">'keep'</span>).species)]</span>
<span id="cb105-5"><a aria-hidden="true" href="#cb105-5" tabindex="-1"></a>                .set_index(<span class="st">'species'</span>)</span>
<span id="cb105-6"><a aria-hidden="true" href="#cb105-6" tabindex="-1"></a>                .T</span>
<span id="cb105-7"><a aria-hidden="true" href="#cb105-7" tabindex="-1"></a>                .<span class="bu">apply</span>(<span class="kw">lambda</span> dd: micro_matrix.loc[ground_samples].corrwith(dd, axis<span class="op">=</span><span class="dv">0</span>, method<span class="op">=</span>get_spearmanr))</span>
<span id="cb105-8"><a aria-hidden="true" href="#cb105-8" tabindex="-1"></a>                .reset_index()</span>
<span id="cb105-9"><a aria-hidden="true" href="#cb105-9" tabindex="-1"></a>)</span>
<span id="cb105-10"><a aria-hidden="true" href="#cb105-10" tabindex="-1"></a></span>
<span id="cb105-11"><a aria-hidden="true" href="#cb105-11" tabindex="-1"></a>p_values_matrix_ground <span class="op">=</span> (metals_matrix</span>
<span id="cb105-12"><a aria-hidden="true" href="#cb105-12" tabindex="-1"></a>                [ground_samples]</span>
<span id="cb105-13"><a aria-hidden="true" href="#cb105-13" tabindex="-1"></a>                .reset_index()</span>
<span id="cb105-14"><a aria-hidden="true" href="#cb105-14" tabindex="-1"></a>                .loc[<span class="kw">lambda</span> dd: dd.species.isin(n_samples_per_metal.query(<span class="st">'keep'</span>).species)]</span>
<span id="cb105-15"><a aria-hidden="true" href="#cb105-15" tabindex="-1"></a>                .set_index(<span class="st">'species'</span>)</span>
<span id="cb105-16"><a aria-hidden="true" href="#cb105-16" tabindex="-1"></a>                .T</span>
<span id="cb105-17"><a aria-hidden="true" href="#cb105-17" tabindex="-1"></a>                .<span class="bu">apply</span>(<span class="kw">lambda</span> dd: micro_matrix.loc[ground_samples].corrwith(dd, axis<span class="op">=</span><span class="dv">0</span>, method<span class="op">=</span>get_pvals))</span>
<span id="cb105-18"><a aria-hidden="true" href="#cb105-18" tabindex="-1"></a>                .reset_index()</span>
<span id="cb105-19"><a aria-hidden="true" href="#cb105-19" tabindex="-1"></a>)</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="66" id="cell-138">
<div class="sourceCode cell-code" id="cb106"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb106-1"><a aria-hidden="true" href="#cb106-1" tabindex="-1"></a>corrs_list <span class="op">=</span> (corrs_matrix</span>
<span id="cb106-2"><a aria-hidden="true" href="#cb106-2" tabindex="-1"></a>              .dropna()</span>
<span id="cb106-3"><a aria-hidden="true" href="#cb106-3" tabindex="-1"></a>                .melt(<span class="st">'Genus'</span>, value_name<span class="op">=</span><span class="st">'spearman_r'</span>)</span>
<span id="cb106-4"><a aria-hidden="true" href="#cb106-4" tabindex="-1"></a>                .merge(p_values_matrix.melt(<span class="st">'Genus'</span>, value_name<span class="op">=</span><span class="st">'spearman_p_value'</span>))</span>
<span id="cb106-5"><a aria-hidden="true" href="#cb106-5" tabindex="-1"></a>                .merge(corrs_matrix.melt(<span class="st">'Genus'</span>, value_name<span class="op">=</span><span class="st">'spearman_r'</span>))</span>
<span id="cb106-6"><a aria-hidden="true" href="#cb106-6" tabindex="-1"></a>                .assign(spearman_fdr<span class="op">=</span><span class="kw">lambda</span> dd: fdrcorrection(dd.spearman_p_value)[<span class="dv">1</span>])</span>
<span id="cb106-7"><a aria-hidden="true" href="#cb106-7" tabindex="-1"></a>)</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="67" id="cell-139">
<div class="sourceCode cell-code" id="cb107"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb107-1"><a aria-hidden="true" href="#cb107-1" tabindex="-1"></a>corrs_list_flights <span class="op">=</span> (corrs_matrix_flight</span>
<span id="cb107-2"><a aria-hidden="true" href="#cb107-2" tabindex="-1"></a>              .dropna()</span>
<span id="cb107-3"><a aria-hidden="true" href="#cb107-3" tabindex="-1"></a>              .melt(<span class="st">'Genus'</span>, value_name<span class="op">=</span><span class="st">'spearman_r'</span>)</span>
<span id="cb107-4"><a aria-hidden="true" href="#cb107-4" tabindex="-1"></a>              .merge(p_values_matrix_flight.melt(<span class="st">'Genus'</span>, value_name<span class="op">=</span><span class="st">'spearman_p_value'</span>))</span>
<span id="cb107-5"><a aria-hidden="true" href="#cb107-5" tabindex="-1"></a>              .merge(corrs_matrix_flight.melt(<span class="st">'Genus'</span>, value_name<span class="op">=</span><span class="st">'spearman_r'</span>))</span>
<span id="cb107-6"><a aria-hidden="true" href="#cb107-6" tabindex="-1"></a>              .assign(spearman_fdr<span class="op">=</span><span class="kw">lambda</span> dd: fdrcorrection(dd.spearman_p_value)[<span class="dv">1</span>])</span>
<span id="cb107-7"><a aria-hidden="true" href="#cb107-7" tabindex="-1"></a>)</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="68" id="cell-140">
<div class="sourceCode cell-code" id="cb108"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb108-1"><a aria-hidden="true" href="#cb108-1" tabindex="-1"></a>corrs_list_ground <span class="op">=</span> (corrs_matrix_ground</span>
<span id="cb108-2"><a aria-hidden="true" href="#cb108-2" tabindex="-1"></a>                .dropna()</span>
<span id="cb108-3"><a aria-hidden="true" href="#cb108-3" tabindex="-1"></a>                .melt(<span class="st">'Genus'</span>, value_name<span class="op">=</span><span class="st">'spearman_r'</span>)</span>
<span id="cb108-4"><a aria-hidden="true" href="#cb108-4" tabindex="-1"></a>                .merge(p_values_matrix_ground.melt(<span class="st">'Genus'</span>, value_name<span class="op">=</span><span class="st">'spearman_p_value'</span>))</span>
<span id="cb108-5"><a aria-hidden="true" href="#cb108-5" tabindex="-1"></a>                .merge(corrs_matrix_ground.melt(<span class="st">'Genus'</span>, value_name<span class="op">=</span><span class="st">'spearman_r'</span>))</span>
<span id="cb108-6"><a aria-hidden="true" href="#cb108-6" tabindex="-1"></a>                .assign(spearman_fdr<span class="op">=</span><span class="kw">lambda</span> dd: fdrcorrection(dd.spearman_p_value)[<span class="dv">1</span>])</span>
<span id="cb108-7"><a aria-hidden="true" href="#cb108-7" tabindex="-1"></a>    )</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="69" id="cell-141">
<div class="sourceCode cell-code" id="cb109"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb109-1"><a aria-hidden="true" href="#cb109-1" tabindex="-1"></a>sig_genera <span class="op">=</span> (corrs_matrix</span>
<span id="cb109-2"><a aria-hidden="true" href="#cb109-2" tabindex="-1"></a>                      .set_index(<span class="st">'Genus'</span>)</span>
<span id="cb109-3"><a aria-hidden="true" href="#cb109-3" tabindex="-1"></a>                      .<span class="bu">apply</span>(<span class="kw">lambda</span> x: x.<span class="bu">max</span>(), axis<span class="op">=</span><span class="dv">1</span>)</span>
<span id="cb109-4"><a aria-hidden="true" href="#cb109-4" tabindex="-1"></a>                      .loc[<span class="kw">lambda</span> x: x <span class="op">&gt;</span> <span class="fl">.544</span>]</span>
<span id="cb109-5"><a aria-hidden="true" href="#cb109-5" tabindex="-1"></a>                      .index.values</span>
<span id="cb109-6"><a aria-hidden="true" href="#cb109-6" tabindex="-1"></a>)</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" id="cell-142">
<div class="sourceCode cell-code" id="cb110"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb110-1"><a aria-hidden="true" href="#cb110-1" tabindex="-1"></a>kingdom_genera <span class="op">=</span> abundance_df_genera[[<span class="st">'Kingdom'</span>, <span class="st">'Genus'</span>]].drop_duplicates()</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="77" id="cell-143">
<div class="sourceCode cell-code" id="cb111"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb111-1"><a aria-hidden="true" href="#cb111-1" tabindex="-1"></a> (corrs_matrix_flight</span>
<span id="cb111-2"><a aria-hidden="true" href="#cb111-2" tabindex="-1"></a> .set_index(<span class="st">'Genus'</span>)</span>
<span id="cb111-3"><a aria-hidden="true" href="#cb111-3" tabindex="-1"></a> .<span class="bu">apply</span>(<span class="kw">lambda</span> x: x.<span class="bu">max</span>(), axis<span class="op">=</span><span class="dv">1</span>)</span>
<span id="cb111-4"><a aria-hidden="true" href="#cb111-4" tabindex="-1"></a> .reset_index()</span>
<span id="cb111-5"><a aria-hidden="true" href="#cb111-5" tabindex="-1"></a> .merge(kingdom_genera)</span>
<span id="cb111-6"><a aria-hidden="true" href="#cb111-6" tabindex="-1"></a><span class="co">#  .groupby(['Kingdom'])</span></span>
<span id="cb111-7"><a aria-hidden="true" href="#cb111-7" tabindex="-1"></a> )</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="77">
<div>
<table class="dataframe caption-top table table-sm table-striped small" data-border="1" data-quarto-postprocess="true">
<thead>
<tr class="header">
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th">Genus</th>
<th data-quarto-table-cell-role="th">0</th>
<th data-quarto-table-cell-role="th">Kingdom</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td data-quarto-table-cell-role="th">0</td>
<td>Bryocella</td>
<td>0.789352</td>
<td>Bacteria</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">1</td>
<td>Actinomyces</td>
<td>0.859856</td>
<td>Bacteria</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">2</td>
<td>Bifidobacterium</td>
<td>0.812777</td>
<td>Bacteria</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">3</td>
<td>Corynebacterium</td>
<td>0.700389</td>
<td>Bacteria</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">4</td>
<td>Lawsonella</td>
<td>0.499327</td>
<td>Bacteria</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">...</td>
<td>...</td>
<td>...</td>
<td>...</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">203</td>
<td>Filobasidium</td>
<td>0.671837</td>
<td>Fungi</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">204</td>
<td>Dioszegia</td>
<td>0.613251</td>
<td>Fungi</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">205</td>
<td>Hannaella</td>
<td>0.622799</td>
<td>Fungi</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">206</td>
<td>Vishniacozyma</td>
<td>0.536602</td>
<td>Fungi</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">207</td>
<td>Ustilago</td>
<td>0.315244</td>
<td>Fungi</td>
</tr>
</tbody>
</table>
<p>208 rows × 3 columns</p>
</div>
</div>
</div>
<div class="cell" data-execution_count="84" id="cell-144">
<div class="sourceCode cell-code" id="cb112"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb112-1"><a aria-hidden="true" href="#cb112-1" tabindex="-1"></a>sig_genera_flights <span class="op">=</span> (corrs_matrix_flight</span>
<span id="cb112-2"><a aria-hidden="true" href="#cb112-2" tabindex="-1"></a> .set_index(<span class="st">'Genus'</span>)</span>
<span id="cb112-3"><a aria-hidden="true" href="#cb112-3" tabindex="-1"></a> .<span class="bu">apply</span>(<span class="kw">lambda</span> x: x.<span class="bu">max</span>(), axis<span class="op">=</span><span class="dv">1</span>)</span>
<span id="cb112-4"><a aria-hidden="true" href="#cb112-4" tabindex="-1"></a> .reset_index()</span>
<span id="cb112-5"><a aria-hidden="true" href="#cb112-5" tabindex="-1"></a> .merge(kingdom_genera)</span>
<span id="cb112-6"><a aria-hidden="true" href="#cb112-6" tabindex="-1"></a> .groupby([<span class="st">'Kingdom'</span>])</span>
<span id="cb112-7"><a aria-hidden="true" href="#cb112-7" tabindex="-1"></a> .<span class="bu">apply</span>(<span class="kw">lambda</span> dd: dd.sort_values(<span class="dv">0</span>, ascending<span class="op">=</span><span class="va">False</span>).iloc[:<span class="dv">40</span>])</span>
<span id="cb112-8"><a aria-hidden="true" href="#cb112-8" tabindex="-1"></a> .Genus</span>
<span id="cb112-9"><a aria-hidden="true" href="#cb112-9" tabindex="-1"></a> .values</span>
<span id="cb112-10"><a aria-hidden="true" href="#cb112-10" tabindex="-1"></a>)</span>
<span id="cb112-11"><a aria-hidden="true" href="#cb112-11" tabindex="-1"></a>sig_genera_flights</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="84">
<pre><code>array(['Staphylococcus', 'Dialister', 'Actinomyces', 'Rothia',
       'Campylobacter', 'Megasphaera', 'Eggerthella', 'Subdoligranulum',
       'Holdemanella', '[Ruminococcus] gnavus group', 'Barnesiella',
       'Ruminococcus', 'Pseudomonas', '[Ruminococcus] torques group',
       'Pygmaiobacter', 'UBA1819', 'Lachnospiraceae UCG-007', 'Megamonas',
       'Klebsiella', 'Dorea', 'Lachnospiraceae UCG-004', 'Akkermansia',
       'Bifidobacterium', 'Atopobium', 'Collinsella', 'Asteroleplasma',
       'Anaerostipes', 'Butyricicoccus', 'Parasutterella',
       'Flavonifractor', 'Lachnospira', 'Hungatella', 'Epulopiscium',
       'Christensenellaceae R-7 group', 'Bryocella', 'Faecalitalea',
       'Acinetobacter', 'UCG-005', 'Streptococcus', 'Lactobacillus',
       'Penicillium', 'Rhodotorula', 'Malassezia', 'Daedaleopsis',
       'Lentinula', 'Hypoxylon', 'Filobasidium', 'Stereum', 'Hannaella',
       'Dioszegia', 'Amyloxenasma', 'Catenulostroma', 'Trametes',
       'Vuilleminia', 'Stemphylium', 'Sarocladium', 'Vishniacozyma',
       'Phaeosclera', 'Symmetrospora', 'Coprinopsis', 'Kondoa',
       'Exophiala', 'Gymnopus', 'Lophiostoma', 'Coniosporium',
       'Verrucocladosporium', 'Paecilomyces', 'Chaetomium', 'Radulomyces',
       'Xylodon', 'Incrucipulum', 'Phlebia', 'Acremonium', 'Stanjemonium',
       'Knufia', 'Torula', 'Cladosporium', 'Phaeosphaeria',
       'Malasseziaceae_gen_Incertae_sedis', 'Petrophila'], dtype=object)</code></pre>
</div>
</div>
<div class="cell" data-execution_count="85" id="cell-145">
<div class="sourceCode cell-code" id="cb114"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb114-1"><a aria-hidden="true" href="#cb114-1" tabindex="-1"></a>sig_genera_ground <span class="op">=</span> (corrs_matrix_ground</span>
<span id="cb114-2"><a aria-hidden="true" href="#cb114-2" tabindex="-1"></a>  .set_index(<span class="st">'Genus'</span>)</span>
<span id="cb114-3"><a aria-hidden="true" href="#cb114-3" tabindex="-1"></a> .<span class="bu">apply</span>(<span class="kw">lambda</span> x: x.<span class="bu">max</span>(), axis<span class="op">=</span><span class="dv">1</span>)</span>
<span id="cb114-4"><a aria-hidden="true" href="#cb114-4" tabindex="-1"></a> .reset_index()</span>
<span id="cb114-5"><a aria-hidden="true" href="#cb114-5" tabindex="-1"></a> .merge(kingdom_genera)</span>
<span id="cb114-6"><a aria-hidden="true" href="#cb114-6" tabindex="-1"></a> .groupby([<span class="st">'Kingdom'</span>])</span>
<span id="cb114-7"><a aria-hidden="true" href="#cb114-7" tabindex="-1"></a> .<span class="bu">apply</span>(<span class="kw">lambda</span> dd: dd.sort_values(<span class="dv">0</span>, ascending<span class="op">=</span><span class="va">False</span>).iloc[:<span class="dv">40</span>])</span>
<span id="cb114-8"><a aria-hidden="true" href="#cb114-8" tabindex="-1"></a> .Genus</span>
<span id="cb114-9"><a aria-hidden="true" href="#cb114-9" tabindex="-1"></a> .values</span>
<span id="cb114-10"><a aria-hidden="true" href="#cb114-10" tabindex="-1"></a>)</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="55" id="cell-146">
<div class="sourceCode cell-code" id="cb115"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb115-1"><a aria-hidden="true" href="#cb115-1" tabindex="-1"></a>(corrs_list</span>
<span id="cb115-2"><a aria-hidden="true" href="#cb115-2" tabindex="-1"></a> .loc[<span class="kw">lambda</span> dd: dd.Genus.isin(sig_genera)]</span>
<span id="cb115-3"><a aria-hidden="true" href="#cb115-3" tabindex="-1"></a> .assign(Genus<span class="op">=</span><span class="kw">lambda</span> dd: dd.Genus.<span class="bu">str</span>.split(<span class="st">'-'</span>).<span class="bu">str</span>[<span class="op">-</span><span class="dv">1</span>])</span>
<span id="cb115-4"><a aria-hidden="true" href="#cb115-4" tabindex="-1"></a> .pipe(<span class="kw">lambda</span> dd: p9.ggplot(dd) </span>
<span id="cb115-5"><a aria-hidden="true" href="#cb115-5" tabindex="-1"></a>                <span class="op">+</span> p9.aes(<span class="st">'species'</span>, <span class="st">'Genus'</span>, fill<span class="op">=</span><span class="st">'spearman_r'</span>)</span>
<span id="cb115-6"><a aria-hidden="true" href="#cb115-6" tabindex="-1"></a>                <span class="op">+</span> p9.geom_tile()</span>
<span id="cb115-7"><a aria-hidden="true" href="#cb115-7" tabindex="-1"></a>                <span class="op">+</span> p9.labs(x<span class="op">=</span><span class="st">''</span>, y<span class="op">=</span><span class="st">''</span>, fill<span class="op">=</span><span class="st">"Spearman's $r$"</span>)</span>
<span id="cb115-8"><a aria-hidden="true" href="#cb115-8" tabindex="-1"></a>                <span class="op">+</span> p9.geom_text(label<span class="op">=</span><span class="st">'+'</span>, data<span class="op">=</span>dd.query(<span class="st">'.01 &lt; spearman_p_value &lt; .05'</span>),</span>
<span id="cb115-9"><a aria-hidden="true" href="#cb115-9" tabindex="-1"></a>                     size<span class="op">=</span><span class="dv">5</span>, va<span class="op">=</span><span class="st">'center_baseline'</span>, color<span class="op">=</span><span class="st">'white'</span>)</span>
<span id="cb115-10"><a aria-hidden="true" href="#cb115-10" tabindex="-1"></a>                <span class="op">+</span> p9.geom_text(label<span class="op">=</span><span class="st">'$*$'</span>, data<span class="op">=</span>dd.query(<span class="st">'spearman_p_value &lt; .01'</span>),</span>
<span id="cb115-11"><a aria-hidden="true" href="#cb115-11" tabindex="-1"></a>                     size<span class="op">=</span><span class="dv">5</span>, va<span class="op">=</span><span class="st">'center_baseline'</span>, color<span class="op">=</span><span class="st">'white'</span>)</span>
<span id="cb115-12"><a aria-hidden="true" href="#cb115-12" tabindex="-1"></a>                <span class="op">+</span> p9.scale_fill_continuous(<span class="st">'RdBu_r'</span>, limits<span class="op">=</span>(<span class="op">-</span><span class="fl">.9</span>, <span class="fl">.9</span>))</span>
<span id="cb115-13"><a aria-hidden="true" href="#cb115-13" tabindex="-1"></a>                <span class="op">+</span> p9.theme(figure_size<span class="op">=</span>(<span class="dv">5</span>, <span class="dv">4</span>),</span>
<span id="cb115-14"><a aria-hidden="true" href="#cb115-14" tabindex="-1"></a>                           legend_key_size<span class="op">=</span><span class="dv">8</span>,</span>
<span id="cb115-15"><a aria-hidden="true" href="#cb115-15" tabindex="-1"></a>                           axis_text_x<span class="op">=</span>p9.element_text(rotation<span class="op">=</span><span class="dv">90</span>, size<span class="op">=</span><span class="dv">7</span>),</span>
<span id="cb115-16"><a aria-hidden="true" href="#cb115-16" tabindex="-1"></a>                           axis_text_y<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">7</span>),</span>
<span id="cb115-17"><a aria-hidden="true" href="#cb115-17" tabindex="-1"></a>                           legend_text<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">7</span>),</span>
<span id="cb115-18"><a aria-hidden="true" href="#cb115-18" tabindex="-1"></a>                           legend_title<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">8</span>),</span>
<span id="cb115-19"><a aria-hidden="true" href="#cb115-19" tabindex="-1"></a>                )</span>
<span id="cb115-20"><a aria-hidden="true" href="#cb115-20" tabindex="-1"></a>))</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img class="figure-img" height="1200" src="microbial_diversity_analysis_files/figure-html/cell-110-output-1.png" width="1500"/></p>
</figure>
</div>
</div>
</div>
<div class="cell" data-execution_count="143" id="cell-147">
<div class="sourceCode cell-code" id="cb116"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb116-1"><a aria-hidden="true" href="#cb116-1" tabindex="-1"></a>(micro_matrix</span>
<span id="cb116-2"><a aria-hidden="true" href="#cb116-2" tabindex="-1"></a> .merge(metals_matrix.T, left_index<span class="op">=</span><span class="va">True</span>, right_index<span class="op">=</span><span class="va">True</span>)</span>
<span id="cb116-3"><a aria-hidden="true" href="#cb116-3" tabindex="-1"></a> .reset_index()</span>
<span id="cb116-4"><a aria-hidden="true" href="#cb116-4" tabindex="-1"></a> .melt(<span class="st">'index'</span>, var_name<span class="op">=</span><span class="st">'species'</span>, value_name<span class="op">=</span><span class="st">'abundance'</span>)</span>
<span id="cb116-5"><a aria-hidden="true" href="#cb116-5" tabindex="-1"></a> .assign(sample_mode<span class="op">=</span><span class="kw">lambda</span> dd: dd[<span class="st">'index'</span>].<span class="bu">map</span>(sample_map))</span>
<span id="cb116-6"><a aria-hidden="true" href="#cb116-6" tabindex="-1"></a> .query(<span class="st">'species.isin(["Sphingomonas", "Zn"])'</span>)</span>
<span id="cb116-7"><a aria-hidden="true" href="#cb116-7" tabindex="-1"></a> .pipe(<span class="kw">lambda</span> dd: p9.ggplot(dd) <span class="op">+</span> p9.aes(<span class="st">'index'</span>, <span class="st">'abundance'</span>)</span>
<span id="cb116-8"><a aria-hidden="true" href="#cb116-8" tabindex="-1"></a>       <span class="op">+</span> p9.facet_grid([<span class="st">'species'</span>, <span class="st">'sample_mode'</span>], scales<span class="op">=</span><span class="st">'free'</span>)</span>
<span id="cb116-9"><a aria-hidden="true" href="#cb116-9" tabindex="-1"></a>       <span class="op">+</span> p9.geom_line(group<span class="op">=</span><span class="dv">1</span>)</span>
<span id="cb116-10"><a aria-hidden="true" href="#cb116-10" tabindex="-1"></a>       <span class="op">+</span> p9.labs(x<span class="op">=</span><span class="st">''</span>, y<span class="op">=</span><span class="st">''</span>)</span>
<span id="cb116-11"><a aria-hidden="true" href="#cb116-11" tabindex="-1"></a>       <span class="op">+</span> p9.theme(figure_size<span class="op">=</span>(<span class="fl">5.5</span>, <span class="dv">3</span>),</span>
<span id="cb116-12"><a aria-hidden="true" href="#cb116-12" tabindex="-1"></a>                  axis_text_x<span class="op">=</span>p9.element_text(rotation<span class="op">=</span><span class="dv">45</span>, size<span class="op">=</span><span class="dv">8</span>),</span>
<span id="cb116-13"><a aria-hidden="true" href="#cb116-13" tabindex="-1"></a>                  )</span>
<span id="cb116-14"><a aria-hidden="true" href="#cb116-14" tabindex="-1"></a>       )</span>
<span id="cb116-15"><a aria-hidden="true" href="#cb116-15" tabindex="-1"></a>)</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img class="figure-img" height="900" src="microbial_diversity_analysis_files/figure-html/cell-111-output-1.png" width="1650"/></p>
</figure>
</div>
</div>
</div>
<div class="cell" data-execution_count="89" id="cell-148">
<div class="sourceCode cell-code" id="cb117"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb117-1"><a aria-hidden="true" href="#cb117-1" tabindex="-1"></a>f <span class="op">=</span> (corrs_list</span>
<span id="cb117-2"><a aria-hidden="true" href="#cb117-2" tabindex="-1"></a>.query(<span class="st">'Genus in @selected_bacterial_genera'</span>) </span>
<span id="cb117-3"><a aria-hidden="true" href="#cb117-3" tabindex="-1"></a>.assign(Genus<span class="op">=</span><span class="kw">lambda</span> dd: dd.Genus.<span class="bu">str</span>.split(<span class="st">'-'</span>).<span class="bu">str</span>[<span class="op">-</span><span class="dv">1</span>])</span>
<span id="cb117-4"><a aria-hidden="true" href="#cb117-4" tabindex="-1"></a> .pipe(<span class="kw">lambda</span> dd: p9.ggplot(dd) </span>
<span id="cb117-5"><a aria-hidden="true" href="#cb117-5" tabindex="-1"></a>                <span class="op">+</span> p9.aes(<span class="st">'species'</span>, <span class="st">'Genus'</span>, fill<span class="op">=</span><span class="st">'spearman_r'</span>)</span>
<span id="cb117-6"><a aria-hidden="true" href="#cb117-6" tabindex="-1"></a>                <span class="op">+</span> p9.geom_tile()</span>
<span id="cb117-7"><a aria-hidden="true" href="#cb117-7" tabindex="-1"></a>                <span class="op">+</span> p9.labs(x<span class="op">=</span><span class="st">''</span>, y<span class="op">=</span><span class="st">''</span>, fill<span class="op">=</span><span class="st">"Spearman's $r$"</span>)</span>
<span id="cb117-8"><a aria-hidden="true" href="#cb117-8" tabindex="-1"></a>                <span class="op">+</span> p9.geom_text(label<span class="op">=</span><span class="st">'+'</span>, data<span class="op">=</span>dd.query(<span class="st">'.01 &lt; spearman_p_value &lt; .05'</span>),</span>
<span id="cb117-9"><a aria-hidden="true" href="#cb117-9" tabindex="-1"></a>                     size<span class="op">=</span><span class="dv">5</span>, va<span class="op">=</span><span class="st">'center_baseline'</span>, color<span class="op">=</span><span class="st">'white'</span>)</span>
<span id="cb117-10"><a aria-hidden="true" href="#cb117-10" tabindex="-1"></a>                <span class="op">+</span> p9.geom_text(label<span class="op">=</span><span class="st">'$*$'</span>, data<span class="op">=</span>dd.query(<span class="st">'spearman_p_value &lt; .01'</span>),</span>
<span id="cb117-11"><a aria-hidden="true" href="#cb117-11" tabindex="-1"></a>                     size<span class="op">=</span><span class="dv">5</span>, va<span class="op">=</span><span class="st">'center_baseline'</span>, color<span class="op">=</span><span class="st">'white'</span>)</span>
<span id="cb117-12"><a aria-hidden="true" href="#cb117-12" tabindex="-1"></a>                <span class="op">+</span> p9.scale_fill_continuous(<span class="st">'RdBu_r'</span>, limits<span class="op">=</span>(<span class="op">-</span><span class="fl">.9</span>, <span class="fl">.9</span>))</span>
<span id="cb117-13"><a aria-hidden="true" href="#cb117-13" tabindex="-1"></a>                <span class="op">+</span> p9.theme(figure_size<span class="op">=</span>(<span class="fl">4.5</span>, <span class="dv">4</span>),</span>
<span id="cb117-14"><a aria-hidden="true" href="#cb117-14" tabindex="-1"></a>                           legend_position<span class="op">=</span><span class="st">'top'</span>,</span>
<span id="cb117-15"><a aria-hidden="true" href="#cb117-15" tabindex="-1"></a>                           legend_key_size<span class="op">=</span><span class="dv">8</span>,</span>
<span id="cb117-16"><a aria-hidden="true" href="#cb117-16" tabindex="-1"></a>                           legend_text<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">7</span>, y<span class="op">=</span><span class="dv">3</span>),</span>
<span id="cb117-17"><a aria-hidden="true" href="#cb117-17" tabindex="-1"></a>                           axis_text_x<span class="op">=</span>p9.element_text(rotation<span class="op">=</span><span class="dv">90</span>, size<span class="op">=</span><span class="dv">7</span>),</span>
<span id="cb117-18"><a aria-hidden="true" href="#cb117-18" tabindex="-1"></a>                           axis_text_y<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">7</span>, face<span class="op">=</span><span class="st">'italic'</span>),</span>
<span id="cb117-19"><a aria-hidden="true" href="#cb117-19" tabindex="-1"></a>                           legend_title<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">8</span>),</span>
<span id="cb117-20"><a aria-hidden="true" href="#cb117-20" tabindex="-1"></a>                )</span>
<span id="cb117-21"><a aria-hidden="true" href="#cb117-21" tabindex="-1"></a>))</span>
<span id="cb117-22"><a aria-hidden="true" href="#cb117-22" tabindex="-1"></a>f.save(<span class="st">'../output/figures/correlations_bacteria.svg'</span>, bbox_inches<span class="op">=</span><span class="st">'tight'</span>)</span>
<span id="cb117-23"><a aria-hidden="true" href="#cb117-23" tabindex="-1"></a>f.draw()</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="89">
<div>
<figure class="figure">
<p><img class="figure-img" height="1233" src="microbial_diversity_analysis_files/figure-html/cell-112-output-1.png" width="1383"/></p>
</figure>
</div>
</div>
</div>
<div class="cell" data-execution_count="63" id="cell-149">
<div class="sourceCode cell-code" id="cb118"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb118-1"><a aria-hidden="true" href="#cb118-1" tabindex="-1"></a>f <span class="op">=</span> (corrs_list_flights</span>
<span id="cb118-2"><a aria-hidden="true" href="#cb118-2" tabindex="-1"></a>.query(<span class="st">'Genus in @selected_bacterial_genera'</span>) </span>
<span id="cb118-3"><a aria-hidden="true" href="#cb118-3" tabindex="-1"></a>.assign(Genus<span class="op">=</span><span class="kw">lambda</span> dd: dd.Genus.<span class="bu">str</span>.split(<span class="st">'-'</span>).<span class="bu">str</span>[<span class="op">-</span><span class="dv">1</span>])</span>
<span id="cb118-4"><a aria-hidden="true" href="#cb118-4" tabindex="-1"></a> .pipe(<span class="kw">lambda</span> dd: p9.ggplot(dd) </span>
<span id="cb118-5"><a aria-hidden="true" href="#cb118-5" tabindex="-1"></a>                <span class="op">+</span> p9.aes(<span class="st">'species'</span>, <span class="st">'Genus'</span>, fill<span class="op">=</span><span class="st">'spearman_r'</span>)</span>
<span id="cb118-6"><a aria-hidden="true" href="#cb118-6" tabindex="-1"></a>                <span class="op">+</span> p9.geom_tile()</span>
<span id="cb118-7"><a aria-hidden="true" href="#cb118-7" tabindex="-1"></a>                <span class="op">+</span> p9.labs(x<span class="op">=</span><span class="st">''</span>, y<span class="op">=</span><span class="st">''</span>, fill<span class="op">=</span><span class="st">"Spearman's $r$"</span>)</span>
<span id="cb118-8"><a aria-hidden="true" href="#cb118-8" tabindex="-1"></a>                <span class="op">+</span> p9.geom_text(label<span class="op">=</span><span class="st">'+'</span>, data<span class="op">=</span>dd.query(<span class="st">'.01 &lt; spearman_p_value &lt; .05'</span>),</span>
<span id="cb118-9"><a aria-hidden="true" href="#cb118-9" tabindex="-1"></a>                     size<span class="op">=</span><span class="dv">5</span>, va<span class="op">=</span><span class="st">'center_baseline'</span>, color<span class="op">=</span><span class="st">'white'</span>)</span>
<span id="cb118-10"><a aria-hidden="true" href="#cb118-10" tabindex="-1"></a>                <span class="op">+</span> p9.geom_text(label<span class="op">=</span><span class="st">'$*$'</span>, data<span class="op">=</span>dd.query(<span class="st">'spearman_p_value &lt; .01'</span>),</span>
<span id="cb118-11"><a aria-hidden="true" href="#cb118-11" tabindex="-1"></a>                     size<span class="op">=</span><span class="dv">5</span>, va<span class="op">=</span><span class="st">'center_baseline'</span>, color<span class="op">=</span><span class="st">'white'</span>)</span>
<span id="cb118-12"><a aria-hidden="true" href="#cb118-12" tabindex="-1"></a>                <span class="op">+</span> p9.scale_fill_continuous(<span class="st">'RdBu_r'</span>, limits<span class="op">=</span>(<span class="op">-</span><span class="fl">.9</span>, <span class="fl">.9</span>))</span>
<span id="cb118-13"><a aria-hidden="true" href="#cb118-13" tabindex="-1"></a>                <span class="op">+</span> p9.theme(figure_size<span class="op">=</span>(<span class="fl">4.5</span>, <span class="dv">4</span>),</span>
<span id="cb118-14"><a aria-hidden="true" href="#cb118-14" tabindex="-1"></a>                           legend_position<span class="op">=</span><span class="st">'top'</span>,</span>
<span id="cb118-15"><a aria-hidden="true" href="#cb118-15" tabindex="-1"></a>                           legend_key_size<span class="op">=</span><span class="dv">8</span>,</span>
<span id="cb118-16"><a aria-hidden="true" href="#cb118-16" tabindex="-1"></a>                           legend_text<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">7</span>, y<span class="op">=</span><span class="dv">3</span>),</span>
<span id="cb118-17"><a aria-hidden="true" href="#cb118-17" tabindex="-1"></a>                           axis_text_x<span class="op">=</span>p9.element_text(rotation<span class="op">=</span><span class="dv">90</span>, size<span class="op">=</span><span class="dv">7</span>),</span>
<span id="cb118-18"><a aria-hidden="true" href="#cb118-18" tabindex="-1"></a>                           axis_text_y<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">7</span>),</span>
<span id="cb118-19"><a aria-hidden="true" href="#cb118-19" tabindex="-1"></a>                           legend_title<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">8</span>),</span>
<span id="cb118-20"><a aria-hidden="true" href="#cb118-20" tabindex="-1"></a>                )</span>
<span id="cb118-21"><a aria-hidden="true" href="#cb118-21" tabindex="-1"></a>))</span>
<span id="cb118-22"><a aria-hidden="true" href="#cb118-22" tabindex="-1"></a>f.save(<span class="st">'../output/figures/corr_bac_flights.svg'</span>, bbox_inches<span class="op">=</span><span class="st">'tight'</span>)</span>
<span id="cb118-23"><a aria-hidden="true" href="#cb118-23" tabindex="-1"></a>f.draw()</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="63">
<div>
<figure class="figure">
<p><img class="figure-img" height="1233" src="microbial_diversity_analysis_files/figure-html/cell-113-output-1.png" width="1383"/></p>
</figure>
</div>
</div>
</div>
<div class="cell" data-execution_count="64" id="cell-150">
<div class="sourceCode cell-code" id="cb119"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb119-1"><a aria-hidden="true" href="#cb119-1" tabindex="-1"></a>f <span class="op">=</span> (corrs_list_ground</span>
<span id="cb119-2"><a aria-hidden="true" href="#cb119-2" tabindex="-1"></a> .query(<span class="st">'Genus in @selected_bacterial_genera'</span>) </span>
<span id="cb119-3"><a aria-hidden="true" href="#cb119-3" tabindex="-1"></a>.assign(Genus<span class="op">=</span><span class="kw">lambda</span> dd: dd.Genus.<span class="bu">str</span>.split(<span class="st">'-'</span>).<span class="bu">str</span>[<span class="op">-</span><span class="dv">1</span>])</span>
<span id="cb119-4"><a aria-hidden="true" href="#cb119-4" tabindex="-1"></a> .pipe(<span class="kw">lambda</span> dd: p9.ggplot(dd) </span>
<span id="cb119-5"><a aria-hidden="true" href="#cb119-5" tabindex="-1"></a>                <span class="op">+</span> p9.aes(<span class="st">'species'</span>, <span class="st">'Genus'</span>, fill<span class="op">=</span><span class="st">'spearman_r'</span>)</span>
<span id="cb119-6"><a aria-hidden="true" href="#cb119-6" tabindex="-1"></a>                <span class="op">+</span> p9.geom_tile()</span>
<span id="cb119-7"><a aria-hidden="true" href="#cb119-7" tabindex="-1"></a>                <span class="op">+</span> p9.labs(x<span class="op">=</span><span class="st">''</span>, y<span class="op">=</span><span class="st">''</span>, fill<span class="op">=</span><span class="st">"Spearman's $r$"</span>)</span>
<span id="cb119-8"><a aria-hidden="true" href="#cb119-8" tabindex="-1"></a>                <span class="op">+</span> p9.geom_text(label<span class="op">=</span><span class="st">'+'</span>, data<span class="op">=</span>dd.query(<span class="st">'.01 &lt; spearman_p_value &lt; .05'</span>),</span>
<span id="cb119-9"><a aria-hidden="true" href="#cb119-9" tabindex="-1"></a>                     size<span class="op">=</span><span class="dv">5</span>, va<span class="op">=</span><span class="st">'center_baseline'</span>, color<span class="op">=</span><span class="st">'white'</span>)</span>
<span id="cb119-10"><a aria-hidden="true" href="#cb119-10" tabindex="-1"></a>                <span class="op">+</span> p9.geom_text(label<span class="op">=</span><span class="st">'$*$'</span>, data<span class="op">=</span>dd.query(<span class="st">'spearman_p_value &lt; .01'</span>),</span>
<span id="cb119-11"><a aria-hidden="true" href="#cb119-11" tabindex="-1"></a>                     size<span class="op">=</span><span class="dv">5</span>, va<span class="op">=</span><span class="st">'center_baseline'</span>, color<span class="op">=</span><span class="st">'white'</span>)</span>
<span id="cb119-12"><a aria-hidden="true" href="#cb119-12" tabindex="-1"></a>                <span class="op">+</span> p9.scale_fill_continuous(<span class="st">'RdBu_r'</span>, limits<span class="op">=</span>(<span class="op">-</span><span class="fl">.9</span>, <span class="fl">.9</span>))</span>
<span id="cb119-13"><a aria-hidden="true" href="#cb119-13" tabindex="-1"></a>                <span class="op">+</span> p9.theme(figure_size<span class="op">=</span>(<span class="fl">4.5</span>, <span class="dv">4</span>),</span>
<span id="cb119-14"><a aria-hidden="true" href="#cb119-14" tabindex="-1"></a>                           legend_position<span class="op">=</span><span class="st">'top'</span>,</span>
<span id="cb119-15"><a aria-hidden="true" href="#cb119-15" tabindex="-1"></a>                           legend_key_size<span class="op">=</span><span class="dv">8</span>,</span>
<span id="cb119-16"><a aria-hidden="true" href="#cb119-16" tabindex="-1"></a>                           legend_text<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">7</span>, y<span class="op">=</span><span class="dv">3</span>),</span>
<span id="cb119-17"><a aria-hidden="true" href="#cb119-17" tabindex="-1"></a>                           axis_text_x<span class="op">=</span>p9.element_text(rotation<span class="op">=</span><span class="dv">90</span>, size<span class="op">=</span><span class="dv">7</span>),</span>
<span id="cb119-18"><a aria-hidden="true" href="#cb119-18" tabindex="-1"></a>                           axis_text_y<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">7</span>),</span>
<span id="cb119-19"><a aria-hidden="true" href="#cb119-19" tabindex="-1"></a>                           legend_title<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">8</span>),</span>
<span id="cb119-20"><a aria-hidden="true" href="#cb119-20" tabindex="-1"></a>                )</span>
<span id="cb119-21"><a aria-hidden="true" href="#cb119-21" tabindex="-1"></a>))</span>
<span id="cb119-22"><a aria-hidden="true" href="#cb119-22" tabindex="-1"></a>f.save(<span class="st">'../output/figures/corr_bac_ground.svg'</span>, bbox_inches<span class="op">=</span><span class="st">'tight'</span>)</span>
<span id="cb119-23"><a aria-hidden="true" href="#cb119-23" tabindex="-1"></a>f.draw()</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="64">
<div>
<figure class="figure">
<p><img class="figure-img" height="1233" src="microbial_diversity_analysis_files/figure-html/cell-114-output-1.png" width="1383"/></p>
</figure>
</div>
</div>
</div>
<div class="cell" data-execution_count="92" id="cell-151">
<div class="sourceCode cell-code" id="cb120"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb120-1"><a aria-hidden="true" href="#cb120-1" tabindex="-1"></a>abundance_df_genera.query(<span class="st">'Phylum.str.startswith("Deino")'</span>)</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="92">
<div>
<table class="dataframe caption-top table table-sm table-striped small" data-border="1" data-quarto-postprocess="true">
<thead>
<tr class="header">
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th">Kingdom</th>
<th data-quarto-table-cell-role="th">Phylum</th>
<th data-quarto-table-cell-role="th">Class</th>
<th data-quarto-table-cell-role="th">Order</th>
<th data-quarto-table-cell-role="th">Family</th>
<th data-quarto-table-cell-role="th">Genus</th>
<th data-quarto-table-cell-role="th">S01</th>
<th data-quarto-table-cell-role="th">S02</th>
<th data-quarto-table-cell-role="th">S03</th>
<th data-quarto-table-cell-role="th">S04</th>
<th data-quarto-table-cell-role="th">...</th>
<th data-quarto-table-cell-role="th">S12</th>
<th data-quarto-table-cell-role="th">S13</th>
<th data-quarto-table-cell-role="th">S15</th>
<th data-quarto-table-cell-role="th">S16</th>
<th data-quarto-table-cell-role="th">S17</th>
<th data-quarto-table-cell-role="th">S18</th>
<th data-quarto-table-cell-role="th">S19</th>
<th data-quarto-table-cell-role="th">S20</th>
<th data-quarto-table-cell-role="th">S21</th>
<th data-quarto-table-cell-role="th">S22</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td data-quarto-table-cell-role="th">129</td>
<td>Bacteria</td>
<td>Deinococcota</td>
<td>Deinococci</td>
<td>Deinococcales</td>
<td>Deinococcaceae</td>
<td>Deinococcus</td>
<td>6</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>...</td>
<td>0</td>
<td>19</td>
<td>12</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">130</td>
<td>Bacteria</td>
<td>Deinococcota</td>
<td>Deinococci</td>
<td>Deinococcales</td>
<td>Trueperaceae</td>
<td>Truepera</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>...</td>
<td>38</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
</tr>
</tbody>
</table>
<p>2 rows × 27 columns</p>
</div>
</div>
</div>
<div class="cell" data-execution_count="255" id="cell-152">
<div class="sourceCode cell-code" id="cb121"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb121-1"><a aria-hidden="true" href="#cb121-1" tabindex="-1"></a>f <span class="op">=</span> (corrs_list_flights</span>
<span id="cb121-2"><a aria-hidden="true" href="#cb121-2" tabindex="-1"></a>.query(<span class="st">'Genus in @selected_fungal_genera'</span>) </span>
<span id="cb121-3"><a aria-hidden="true" href="#cb121-3" tabindex="-1"></a>.assign(Genus<span class="op">=</span><span class="kw">lambda</span> dd: dd.Genus.<span class="bu">str</span>.split(<span class="st">'-'</span>).<span class="bu">str</span>[<span class="op">-</span><span class="dv">1</span>])</span>
<span id="cb121-4"><a aria-hidden="true" href="#cb121-4" tabindex="-1"></a> .pipe(<span class="kw">lambda</span> dd: p9.ggplot(dd) </span>
<span id="cb121-5"><a aria-hidden="true" href="#cb121-5" tabindex="-1"></a>                <span class="op">+</span> p9.aes(<span class="st">'species'</span>, <span class="st">'Genus'</span>, fill<span class="op">=</span><span class="st">'spearman_r'</span>)</span>
<span id="cb121-6"><a aria-hidden="true" href="#cb121-6" tabindex="-1"></a>                <span class="op">+</span> p9.geom_tile()</span>
<span id="cb121-7"><a aria-hidden="true" href="#cb121-7" tabindex="-1"></a>                <span class="op">+</span> p9.labs(x<span class="op">=</span><span class="st">''</span>, y<span class="op">=</span><span class="st">''</span>, fill<span class="op">=</span><span class="st">"Spearman's $r$"</span>)</span>
<span id="cb121-8"><a aria-hidden="true" href="#cb121-8" tabindex="-1"></a>                <span class="op">+</span> p9.geom_text(label<span class="op">=</span><span class="st">'+'</span>, data<span class="op">=</span>dd.query(<span class="st">'.01 &lt; spearman_p_value &lt; .05'</span>),</span>
<span id="cb121-9"><a aria-hidden="true" href="#cb121-9" tabindex="-1"></a>                     size<span class="op">=</span><span class="dv">5</span>, va<span class="op">=</span><span class="st">'center_baseline'</span>, color<span class="op">=</span><span class="st">'white'</span>)</span>
<span id="cb121-10"><a aria-hidden="true" href="#cb121-10" tabindex="-1"></a>                <span class="op">+</span> p9.geom_text(label<span class="op">=</span><span class="st">'$*$'</span>, data<span class="op">=</span>dd.query(<span class="st">'spearman_p_value &lt; .01'</span>),</span>
<span id="cb121-11"><a aria-hidden="true" href="#cb121-11" tabindex="-1"></a>                     size<span class="op">=</span><span class="dv">5</span>, va<span class="op">=</span><span class="st">'center_baseline'</span>, color<span class="op">=</span><span class="st">'white'</span>)</span>
<span id="cb121-12"><a aria-hidden="true" href="#cb121-12" tabindex="-1"></a>                <span class="op">+</span> p9.scale_fill_continuous(<span class="st">'RdBu_r'</span>, limits<span class="op">=</span>(<span class="op">-</span><span class="fl">.9</span>, <span class="fl">.9</span>))</span>
<span id="cb121-13"><a aria-hidden="true" href="#cb121-13" tabindex="-1"></a>                <span class="op">+</span> p9.theme(figure_size<span class="op">=</span>(<span class="dv">4</span>, <span class="dv">4</span>),</span>
<span id="cb121-14"><a aria-hidden="true" href="#cb121-14" tabindex="-1"></a>                           legend_position<span class="op">=</span><span class="st">'top'</span>,</span>
<span id="cb121-15"><a aria-hidden="true" href="#cb121-15" tabindex="-1"></a>                           legend_key_size<span class="op">=</span><span class="dv">8</span>,</span>
<span id="cb121-16"><a aria-hidden="true" href="#cb121-16" tabindex="-1"></a>                           legend_text<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">7</span>, y<span class="op">=</span><span class="dv">3</span>),</span>
<span id="cb121-17"><a aria-hidden="true" href="#cb121-17" tabindex="-1"></a>                           axis_text_x<span class="op">=</span>p9.element_text(rotation<span class="op">=</span><span class="dv">90</span>, size<span class="op">=</span><span class="dv">7</span>),</span>
<span id="cb121-18"><a aria-hidden="true" href="#cb121-18" tabindex="-1"></a>                           axis_text_y<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">7</span>),</span>
<span id="cb121-19"><a aria-hidden="true" href="#cb121-19" tabindex="-1"></a>                           legend_title<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">8</span>),</span>
<span id="cb121-20"><a aria-hidden="true" href="#cb121-20" tabindex="-1"></a>                )</span>
<span id="cb121-21"><a aria-hidden="true" href="#cb121-21" tabindex="-1"></a>))</span>
<span id="cb121-22"><a aria-hidden="true" href="#cb121-22" tabindex="-1"></a>f.save(<span class="st">'../output/figures/corr_fun_flights.svg'</span>, bbox_inches<span class="op">=</span><span class="st">'tight'</span>)</span>
<span id="cb121-23"><a aria-hidden="true" href="#cb121-23" tabindex="-1"></a>f.draw()</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="255">
<div>
<figure class="figure">
<p><img class="figure-img" height="1200" src="microbial_diversity_analysis_files/figure-html/cell-116-output-1.png" width="1200"/></p>
</figure>
</div>
</div>
</div>
<div class="cell" data-execution_count="258" id="cell-153">
<div class="sourceCode cell-code" id="cb122"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb122-1"><a aria-hidden="true" href="#cb122-1" tabindex="-1"></a>f <span class="op">=</span> (corrs_list_ground</span>
<span id="cb122-2"><a aria-hidden="true" href="#cb122-2" tabindex="-1"></a> .query(<span class="st">'Genus in @selected_fungal_genera'</span>) </span>
<span id="cb122-3"><a aria-hidden="true" href="#cb122-3" tabindex="-1"></a>.assign(Genus<span class="op">=</span><span class="kw">lambda</span> dd: dd.Genus.<span class="bu">str</span>.split(<span class="st">'-'</span>).<span class="bu">str</span>[<span class="op">-</span><span class="dv">1</span>])</span>
<span id="cb122-4"><a aria-hidden="true" href="#cb122-4" tabindex="-1"></a> .pipe(<span class="kw">lambda</span> dd: p9.ggplot(dd) </span>
<span id="cb122-5"><a aria-hidden="true" href="#cb122-5" tabindex="-1"></a>                <span class="op">+</span> p9.aes(<span class="st">'species'</span>, <span class="st">'Genus'</span>, fill<span class="op">=</span><span class="st">'spearman_r'</span>)</span>
<span id="cb122-6"><a aria-hidden="true" href="#cb122-6" tabindex="-1"></a>                <span class="op">+</span> p9.geom_tile()</span>
<span id="cb122-7"><a aria-hidden="true" href="#cb122-7" tabindex="-1"></a>                <span class="op">+</span> p9.labs(x<span class="op">=</span><span class="st">''</span>, y<span class="op">=</span><span class="st">''</span>, fill<span class="op">=</span><span class="st">"Spearman's $r$"</span>)</span>
<span id="cb122-8"><a aria-hidden="true" href="#cb122-8" tabindex="-1"></a>                <span class="op">+</span> p9.geom_text(label<span class="op">=</span><span class="st">'+'</span>, data<span class="op">=</span>dd.query(<span class="st">'.01 &lt; spearman_p_value &lt; .05'</span>),</span>
<span id="cb122-9"><a aria-hidden="true" href="#cb122-9" tabindex="-1"></a>                     size<span class="op">=</span><span class="dv">5</span>, va<span class="op">=</span><span class="st">'center_baseline'</span>, color<span class="op">=</span><span class="st">'white'</span>)</span>
<span id="cb122-10"><a aria-hidden="true" href="#cb122-10" tabindex="-1"></a>                <span class="op">+</span> p9.geom_text(label<span class="op">=</span><span class="st">'$*$'</span>, data<span class="op">=</span>dd.query(<span class="st">'spearman_p_value &lt; .01'</span>),</span>
<span id="cb122-11"><a aria-hidden="true" href="#cb122-11" tabindex="-1"></a>                     size<span class="op">=</span><span class="dv">5</span>, va<span class="op">=</span><span class="st">'center_baseline'</span>, color<span class="op">=</span><span class="st">'white'</span>)</span>
<span id="cb122-12"><a aria-hidden="true" href="#cb122-12" tabindex="-1"></a>                <span class="op">+</span> p9.scale_fill_continuous(<span class="st">'RdBu_r'</span>, limits<span class="op">=</span>(<span class="op">-</span><span class="fl">.9</span>, <span class="fl">.9</span>))</span>
<span id="cb122-13"><a aria-hidden="true" href="#cb122-13" tabindex="-1"></a>                <span class="op">+</span> p9.theme(figure_size<span class="op">=</span>(<span class="dv">4</span>, <span class="dv">4</span>),</span>
<span id="cb122-14"><a aria-hidden="true" href="#cb122-14" tabindex="-1"></a>                           legend_position<span class="op">=</span><span class="st">'top'</span>,</span>
<span id="cb122-15"><a aria-hidden="true" href="#cb122-15" tabindex="-1"></a>                           legend_key_size<span class="op">=</span><span class="dv">8</span>,</span>
<span id="cb122-16"><a aria-hidden="true" href="#cb122-16" tabindex="-1"></a>                           legend_text<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">7</span>, y<span class="op">=</span><span class="dv">3</span>),</span>
<span id="cb122-17"><a aria-hidden="true" href="#cb122-17" tabindex="-1"></a>                           axis_text_x<span class="op">=</span>p9.element_text(rotation<span class="op">=</span><span class="dv">90</span>, size<span class="op">=</span><span class="dv">7</span>),</span>
<span id="cb122-18"><a aria-hidden="true" href="#cb122-18" tabindex="-1"></a>                           axis_text_y<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">7</span>),</span>
<span id="cb122-19"><a aria-hidden="true" href="#cb122-19" tabindex="-1"></a>                           legend_title<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">8</span>),</span>
<span id="cb122-20"><a aria-hidden="true" href="#cb122-20" tabindex="-1"></a>                )</span>
<span id="cb122-21"><a aria-hidden="true" href="#cb122-21" tabindex="-1"></a>))</span>
<span id="cb122-22"><a aria-hidden="true" href="#cb122-22" tabindex="-1"></a></span>
<span id="cb122-23"><a aria-hidden="true" href="#cb122-23" tabindex="-1"></a>f.save(<span class="st">'../output/figures/corr_fun_ground.svg'</span>, bbox_inches<span class="op">=</span><span class="st">'tight'</span>)</span>
<span id="cb122-24"><a aria-hidden="true" href="#cb122-24" tabindex="-1"></a>f.draw()</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="258">
<div>
<figure class="figure">
<p><img class="figure-img" height="1200" src="microbial_diversity_analysis_files/figure-html/cell-117-output-1.png" width="1200"/></p>
</figure>
</div>
</div>
</div>
<div class="cell" data-execution_count="112" id="cell-154">
<div class="sourceCode cell-code" id="cb123"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb123-1"><a aria-hidden="true" href="#cb123-1" tabindex="-1"></a>top_correlated_genera <span class="op">=</span> (corrs_matrix_flight</span>
<span id="cb123-2"><a aria-hidden="true" href="#cb123-2" tabindex="-1"></a> .set_index(<span class="st">'Genus'</span>)</span>
<span id="cb123-3"><a aria-hidden="true" href="#cb123-3" tabindex="-1"></a> .<span class="bu">apply</span>(<span class="kw">lambda</span> x: x.<span class="bu">abs</span>().<span class="bu">max</span>(), axis<span class="op">=</span><span class="dv">1</span>)</span>
<span id="cb123-4"><a aria-hidden="true" href="#cb123-4" tabindex="-1"></a> .rename(<span class="st">'correlation'</span>)</span>
<span id="cb123-5"><a aria-hidden="true" href="#cb123-5" tabindex="-1"></a> .reset_index()</span>
<span id="cb123-6"><a aria-hidden="true" href="#cb123-6" tabindex="-1"></a> .merge(kingdom_genera)</span>
<span id="cb123-7"><a aria-hidden="true" href="#cb123-7" tabindex="-1"></a> .groupby([<span class="st">'Kingdom'</span>], as_index<span class="op">=</span><span class="va">False</span>)</span>
<span id="cb123-8"><a aria-hidden="true" href="#cb123-8" tabindex="-1"></a> .<span class="bu">apply</span>(<span class="kw">lambda</span> dd: dd.sort_values(by<span class="op">=</span><span class="st">'correlation'</span>, ascending<span class="op">=</span><span class="va">False</span>).iloc[:<span class="dv">30</span>])</span>
<span id="cb123-9"><a aria-hidden="true" href="#cb123-9" tabindex="-1"></a>)</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="173" id="cell-155">
<div class="sourceCode cell-code" id="cb124"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb124-1"><a aria-hidden="true" href="#cb124-1" tabindex="-1"></a>top_correlated_genera <span class="op">=</span> (corrs_matrix_flight</span>
<span id="cb124-2"><a aria-hidden="true" href="#cb124-2" tabindex="-1"></a> .set_index(<span class="st">'Genus'</span>)</span>
<span id="cb124-3"><a aria-hidden="true" href="#cb124-3" tabindex="-1"></a> .<span class="bu">apply</span>(<span class="kw">lambda</span> x: x.<span class="bu">abs</span>().<span class="bu">max</span>(), axis<span class="op">=</span><span class="dv">1</span>)</span>
<span id="cb124-4"><a aria-hidden="true" href="#cb124-4" tabindex="-1"></a> .rename(<span class="st">'correlation'</span>)</span>
<span id="cb124-5"><a aria-hidden="true" href="#cb124-5" tabindex="-1"></a> .reset_index()</span>
<span id="cb124-6"><a aria-hidden="true" href="#cb124-6" tabindex="-1"></a> .merge(kingdom_genera)</span>
<span id="cb124-7"><a aria-hidden="true" href="#cb124-7" tabindex="-1"></a> .groupby([<span class="st">'Kingdom'</span>], as_index<span class="op">=</span><span class="va">False</span>)</span>
<span id="cb124-8"><a aria-hidden="true" href="#cb124-8" tabindex="-1"></a> .<span class="bu">apply</span>(<span class="kw">lambda</span> dd: dd.sort_values(by<span class="op">=</span><span class="st">'correlation'</span>, ascending<span class="op">=</span><span class="va">False</span>).iloc[:<span class="dv">40</span>])</span>
<span id="cb124-9"><a aria-hidden="true" href="#cb124-9" tabindex="-1"></a>)</span>
<span id="cb124-10"><a aria-hidden="true" href="#cb124-10" tabindex="-1"></a></span>
<span id="cb124-11"><a aria-hidden="true" href="#cb124-11" tabindex="-1"></a>f <span class="op">=</span> (corrs_list_flights</span>
<span id="cb124-12"><a aria-hidden="true" href="#cb124-12" tabindex="-1"></a> .merge(kingdom_genera)</span>
<span id="cb124-13"><a aria-hidden="true" href="#cb124-13" tabindex="-1"></a> .merge(top_correlated_genera.sort_values(<span class="st">'correlation'</span>, ascending<span class="op">=</span><span class="va">False</span>).iloc[:<span class="dv">40</span>][[<span class="st">'Genus'</span>]], on<span class="op">=</span>[<span class="st">'Genus'</span>])</span>
<span id="cb124-14"><a aria-hidden="true" href="#cb124-14" tabindex="-1"></a> .assign(Genus<span class="op">=</span><span class="kw">lambda</span> dd: pd.Categorical(dd.Genus, categories<span class="op">=</span>top_correlated_genera.Genus, ordered<span class="op">=</span><span class="va">True</span>))</span>
<span id="cb124-15"><a aria-hidden="true" href="#cb124-15" tabindex="-1"></a> .pipe(<span class="kw">lambda</span> dd: p9.ggplot(dd) </span>
<span id="cb124-16"><a aria-hidden="true" href="#cb124-16" tabindex="-1"></a>                <span class="op">+</span> p9.aes(<span class="st">'Genus'</span>, <span class="st">'species'</span>, fill<span class="op">=</span><span class="st">'spearman_r'</span>)</span>
<span id="cb124-17"><a aria-hidden="true" href="#cb124-17" tabindex="-1"></a>                <span class="op">+</span> p9.geom_tile()</span>
<span id="cb124-18"><a aria-hidden="true" href="#cb124-18" tabindex="-1"></a>                <span class="op">+</span> p9.labs(x<span class="op">=</span><span class="st">''</span>, y<span class="op">=</span><span class="st">''</span>, fill<span class="op">=</span><span class="st">"Spearman's $r$"</span>)</span>
<span id="cb124-19"><a aria-hidden="true" href="#cb124-19" tabindex="-1"></a>                <span class="op">+</span> p9.geom_text(label<span class="op">=</span><span class="st">'+'</span>, data<span class="op">=</span>dd.query(<span class="st">'.01 &lt; spearman_p_value &lt; .05'</span>),</span>
<span id="cb124-20"><a aria-hidden="true" href="#cb124-20" tabindex="-1"></a>                     size<span class="op">=</span><span class="dv">5</span>, va<span class="op">=</span><span class="st">'center_baseline'</span>, color<span class="op">=</span><span class="st">'white'</span>)</span>
<span id="cb124-21"><a aria-hidden="true" href="#cb124-21" tabindex="-1"></a>                <span class="op">+</span> p9.geom_text(label<span class="op">=</span><span class="st">'$*$'</span>, data<span class="op">=</span>dd.query(<span class="st">'spearman_p_value &lt; .01'</span>),</span>
<span id="cb124-22"><a aria-hidden="true" href="#cb124-22" tabindex="-1"></a>                     size<span class="op">=</span><span class="dv">5</span>, va<span class="op">=</span><span class="st">'center_baseline'</span>, color<span class="op">=</span><span class="st">'white'</span>)</span>
<span id="cb124-23"><a aria-hidden="true" href="#cb124-23" tabindex="-1"></a>                <span class="op">+</span> p9.scale_fill_continuous(<span class="st">'RdBu_r'</span>, limits<span class="op">=</span>(<span class="op">-</span><span class="fl">.9</span>, <span class="fl">.9</span>))</span>
<span id="cb124-24"><a aria-hidden="true" href="#cb124-24" tabindex="-1"></a>                <span class="op">+</span> p9.theme(figure_size<span class="op">=</span>(<span class="dv">4</span>, <span class="fl">4.5</span>),</span>
<span id="cb124-25"><a aria-hidden="true" href="#cb124-25" tabindex="-1"></a>                           legend_key_size<span class="op">=</span><span class="dv">8</span>,</span>
<span id="cb124-26"><a aria-hidden="true" href="#cb124-26" tabindex="-1"></a>                           axis_text_x<span class="op">=</span>p9.element_text(rotation<span class="op">=</span><span class="dv">90</span>, size<span class="op">=</span><span class="dv">7</span>),</span>
<span id="cb124-27"><a aria-hidden="true" href="#cb124-27" tabindex="-1"></a>                           axis_text_y<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">7</span>),</span>
<span id="cb124-28"><a aria-hidden="true" href="#cb124-28" tabindex="-1"></a>                           legend_text<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">7</span>),</span>
<span id="cb124-29"><a aria-hidden="true" href="#cb124-29" tabindex="-1"></a>                           legend_title<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">8</span>),</span>
<span id="cb124-30"><a aria-hidden="true" href="#cb124-30" tabindex="-1"></a>                           legend_position<span class="op">=</span><span class="st">'bottom'</span></span>
<span id="cb124-31"><a aria-hidden="true" href="#cb124-31" tabindex="-1"></a>                )</span>
<span id="cb124-32"><a aria-hidden="true" href="#cb124-32" tabindex="-1"></a>)</span>
<span id="cb124-33"><a aria-hidden="true" href="#cb124-33" tabindex="-1"></a>)</span>
<span id="cb124-34"><a aria-hidden="true" href="#cb124-34" tabindex="-1"></a>f.save(<span class="st">'../output/figures/top40_corr.svg'</span>)</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="167" id="cell-156">
<div class="sourceCode cell-code" id="cb125"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb125-1"><a aria-hidden="true" href="#cb125-1" tabindex="-1"></a>(corrs_list_flights</span>
<span id="cb125-2"><a aria-hidden="true" href="#cb125-2" tabindex="-1"></a> .merge(kingdom_genera)</span>
<span id="cb125-3"><a aria-hidden="true" href="#cb125-3" tabindex="-1"></a> .merge(top_correlated_genera[[<span class="st">'Genus'</span>]], on<span class="op">=</span>[<span class="st">'Genus'</span>])</span>
<span id="cb125-4"><a aria-hidden="true" href="#cb125-4" tabindex="-1"></a> .assign(Genus<span class="op">=</span><span class="kw">lambda</span> dd: pd.Categorical(dd.Genus, categories<span class="op">=</span>top_correlated_genera.Genus, ordered<span class="op">=</span><span class="va">True</span>))</span>
<span id="cb125-5"><a aria-hidden="true" href="#cb125-5" tabindex="-1"></a> .pipe(<span class="kw">lambda</span> dd: p9.ggplot(dd) </span>
<span id="cb125-6"><a aria-hidden="true" href="#cb125-6" tabindex="-1"></a>                <span class="op">+</span> p9.aes(<span class="st">'species'</span>, <span class="st">'Genus'</span>, fill<span class="op">=</span><span class="st">'spearman_r'</span>)</span>
<span id="cb125-7"><a aria-hidden="true" href="#cb125-7" tabindex="-1"></a>                <span class="op">+</span> p9.geom_tile(</span>
<span id="cb125-8"><a aria-hidden="true" href="#cb125-8" tabindex="-1"></a>                    <span class="co"># color='black'</span></span>
<span id="cb125-9"><a aria-hidden="true" href="#cb125-9" tabindex="-1"></a>                    )</span>
<span id="cb125-10"><a aria-hidden="true" href="#cb125-10" tabindex="-1"></a>                <span class="op">+</span> p9.labs(x<span class="op">=</span><span class="st">''</span>, y<span class="op">=</span><span class="st">''</span>, fill<span class="op">=</span><span class="st">"Spearman's $r$"</span>)</span>
<span id="cb125-11"><a aria-hidden="true" href="#cb125-11" tabindex="-1"></a>                <span class="op">+</span> p9.geom_text(label<span class="op">=</span><span class="st">'+'</span>, data<span class="op">=</span>dd.query(<span class="st">'.01 &lt; spearman_p_value &lt; .05'</span>),</span>
<span id="cb125-12"><a aria-hidden="true" href="#cb125-12" tabindex="-1"></a>                     size<span class="op">=</span><span class="dv">5</span>, va<span class="op">=</span><span class="st">'center_baseline'</span>, color<span class="op">=</span><span class="st">'white'</span>)</span>
<span id="cb125-13"><a aria-hidden="true" href="#cb125-13" tabindex="-1"></a>                <span class="op">+</span> p9.geom_text(label<span class="op">=</span><span class="st">'$*$'</span>, data<span class="op">=</span>dd.query(<span class="st">'spearman_p_value &lt; .01'</span>),</span>
<span id="cb125-14"><a aria-hidden="true" href="#cb125-14" tabindex="-1"></a>                     size<span class="op">=</span><span class="dv">5</span>, va<span class="op">=</span><span class="st">'center_baseline'</span>, color<span class="op">=</span><span class="st">'white'</span>)</span>
<span id="cb125-15"><a aria-hidden="true" href="#cb125-15" tabindex="-1"></a>                <span class="op">+</span> p9.scale_fill_continuous(<span class="st">'RdBu_r'</span>, limits<span class="op">=</span>(<span class="op">-</span><span class="fl">.9</span>, <span class="fl">.9</span>))</span>
<span id="cb125-16"><a aria-hidden="true" href="#cb125-16" tabindex="-1"></a>                <span class="op">+</span> p9.facet_wrap(<span class="st">'Kingdom'</span>, scales<span class="op">=</span><span class="st">'free_y'</span>)</span>
<span id="cb125-17"><a aria-hidden="true" href="#cb125-17" tabindex="-1"></a>                <span class="op">+</span> p9.theme(figure_size<span class="op">=</span>(<span class="dv">9</span>, <span class="dv">4</span>),</span>
<span id="cb125-18"><a aria-hidden="true" href="#cb125-18" tabindex="-1"></a>                           legend_key_size<span class="op">=</span><span class="dv">8</span>,</span>
<span id="cb125-19"><a aria-hidden="true" href="#cb125-19" tabindex="-1"></a>                           axis_text_x<span class="op">=</span>p9.element_text(rotation<span class="op">=</span><span class="dv">90</span>, size<span class="op">=</span><span class="dv">7</span>),</span>
<span id="cb125-20"><a aria-hidden="true" href="#cb125-20" tabindex="-1"></a>                           axis_text_y<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">7</span>),</span>
<span id="cb125-21"><a aria-hidden="true" href="#cb125-21" tabindex="-1"></a>                           legend_text<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">7</span>),</span>
<span id="cb125-22"><a aria-hidden="true" href="#cb125-22" tabindex="-1"></a>                           legend_title<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">8</span>),</span>
<span id="cb125-23"><a aria-hidden="true" href="#cb125-23" tabindex="-1"></a>                )</span>
<span id="cb125-24"><a aria-hidden="true" href="#cb125-24" tabindex="-1"></a>)</span>
<span id="cb125-25"><a aria-hidden="true" href="#cb125-25" tabindex="-1"></a>)</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img class="figure-img" height="1200" src="microbial_diversity_analysis_files/figure-html/cell-120-output-1.png" width="2700"/></p>
</figure>
</div>
</div>
</div>
<div class="cell" id="cell-157">
<div class="sourceCode cell-code" id="cb126"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb126-1"><a aria-hidden="true" href="#cb126-1" tabindex="-1"></a></span>
<span id="cb126-2"><a aria-hidden="true" href="#cb126-2" tabindex="-1"></a>f <span class="op">=</span> (corrs_list</span>
<span id="cb126-3"><a aria-hidden="true" href="#cb126-3" tabindex="-1"></a>.query(<span class="st">'Genus in @selected_bacterial_genera'</span>) </span>
<span id="cb126-4"><a aria-hidden="true" href="#cb126-4" tabindex="-1"></a>.assign(Genus<span class="op">=</span><span class="kw">lambda</span> dd: dd.Genus.<span class="bu">str</span>.split(<span class="st">'-'</span>).<span class="bu">str</span>[<span class="op">-</span><span class="dv">1</span>])</span>
<span id="cb126-5"><a aria-hidden="true" href="#cb126-5" tabindex="-1"></a> .pipe(<span class="kw">lambda</span> dd: p9.ggplot(dd) </span>
<span id="cb126-6"><a aria-hidden="true" href="#cb126-6" tabindex="-1"></a>                <span class="op">+</span> p9.aes(<span class="st">'species'</span>, <span class="st">'Genus'</span>, fill<span class="op">=</span><span class="st">'spearman_r'</span>)</span>
<span id="cb126-7"><a aria-hidden="true" href="#cb126-7" tabindex="-1"></a>                <span class="op">+</span> p9.geom_tile()</span>
<span id="cb126-8"><a aria-hidden="true" href="#cb126-8" tabindex="-1"></a>                <span class="op">+</span> p9.labs(x<span class="op">=</span><span class="st">''</span>, y<span class="op">=</span><span class="st">''</span>, fill<span class="op">=</span><span class="st">"Spearman's $r$"</span>)</span>
<span id="cb126-9"><a aria-hidden="true" href="#cb126-9" tabindex="-1"></a>                <span class="op">+</span> p9.geom_text(label<span class="op">=</span><span class="st">'+'</span>, data<span class="op">=</span>dd.query(<span class="st">'.01 &lt; spearman_p_value &lt; .05'</span>),</span>
<span id="cb126-10"><a aria-hidden="true" href="#cb126-10" tabindex="-1"></a>                     size<span class="op">=</span><span class="dv">5</span>, va<span class="op">=</span><span class="st">'center_baseline'</span>, color<span class="op">=</span><span class="st">'white'</span>)</span>
<span id="cb126-11"><a aria-hidden="true" href="#cb126-11" tabindex="-1"></a>                <span class="op">+</span> p9.geom_text(label<span class="op">=</span><span class="st">'$*$'</span>, data<span class="op">=</span>dd.query(<span class="st">'spearman_p_value &lt; .01'</span>),</span>
<span id="cb126-12"><a aria-hidden="true" href="#cb126-12" tabindex="-1"></a>                     size<span class="op">=</span><span class="dv">5</span>, va<span class="op">=</span><span class="st">'center_baseline'</span>, color<span class="op">=</span><span class="st">'white'</span>)</span>
<span id="cb126-13"><a aria-hidden="true" href="#cb126-13" tabindex="-1"></a>                <span class="op">+</span> p9.scale_fill_continuous(<span class="st">'RdBu_r'</span>, limits<span class="op">=</span>(<span class="op">-</span><span class="fl">.9</span>, <span class="fl">.9</span>))</span>
<span id="cb126-14"><a aria-hidden="true" href="#cb126-14" tabindex="-1"></a>                <span class="op">+</span> p9.theme(figure_size<span class="op">=</span>(<span class="dv">4</span>, <span class="dv">4</span>),</span>
<span id="cb126-15"><a aria-hidden="true" href="#cb126-15" tabindex="-1"></a>                           legend_position<span class="op">=</span><span class="st">'top'</span>,</span>
<span id="cb126-16"><a aria-hidden="true" href="#cb126-16" tabindex="-1"></a>                           legend_key_size<span class="op">=</span><span class="dv">8</span>,</span>
<span id="cb126-17"><a aria-hidden="true" href="#cb126-17" tabindex="-1"></a>                           legend_text<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">7</span>, y<span class="op">=</span><span class="dv">3</span>),</span>
<span id="cb126-18"><a aria-hidden="true" href="#cb126-18" tabindex="-1"></a>                           axis_text_x<span class="op">=</span>p9.element_text(rotation<span class="op">=</span><span class="dv">90</span>, size<span class="op">=</span><span class="dv">7</span>),</span>
<span id="cb126-19"><a aria-hidden="true" href="#cb126-19" tabindex="-1"></a>                           axis_text_y<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">7</span>),</span>
<span id="cb126-20"><a aria-hidden="true" href="#cb126-20" tabindex="-1"></a>                           legend_title<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">8</span>),</span>
<span id="cb126-21"><a aria-hidden="true" href="#cb126-21" tabindex="-1"></a>                )</span>
<span id="cb126-22"><a aria-hidden="true" href="#cb126-22" tabindex="-1"></a>))</span>
<span id="cb126-23"><a aria-hidden="true" href="#cb126-23" tabindex="-1"></a>f.save(<span class="st">'../output/figures/correlations_bacteria.svg'</span>, bbox_inches<span class="op">=</span><span class="st">'tight'</span>)</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" id="cell-158">
<div class="sourceCode cell-code" id="cb127"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb127-1"><a aria-hidden="true" href="#cb127-1" tabindex="-1"></a>f <span class="op">=</span> (corrs_list</span>
<span id="cb127-2"><a aria-hidden="true" href="#cb127-2" tabindex="-1"></a>.query(<span class="st">'Genus in @selected_fungal_genera'</span>) </span>
<span id="cb127-3"><a aria-hidden="true" href="#cb127-3" tabindex="-1"></a>.assign(Genus<span class="op">=</span><span class="kw">lambda</span> dd: dd.Genus.<span class="bu">str</span>.split(<span class="st">'-'</span>).<span class="bu">str</span>[<span class="op">-</span><span class="dv">1</span>])</span>
<span id="cb127-4"><a aria-hidden="true" href="#cb127-4" tabindex="-1"></a> .pipe(<span class="kw">lambda</span> dd: p9.ggplot(dd) </span>
<span id="cb127-5"><a aria-hidden="true" href="#cb127-5" tabindex="-1"></a>                <span class="op">+</span> p9.aes(<span class="st">'species'</span>, <span class="st">'Genus'</span>, fill<span class="op">=</span><span class="st">'spearman_r'</span>)</span>
<span id="cb127-6"><a aria-hidden="true" href="#cb127-6" tabindex="-1"></a>                <span class="op">+</span> p9.geom_tile()</span>
<span id="cb127-7"><a aria-hidden="true" href="#cb127-7" tabindex="-1"></a>                <span class="op">+</span> p9.labs(x<span class="op">=</span><span class="st">''</span>, y<span class="op">=</span><span class="st">''</span>, fill<span class="op">=</span><span class="st">"Spearman's $r$"</span>)</span>
<span id="cb127-8"><a aria-hidden="true" href="#cb127-8" tabindex="-1"></a>                <span class="op">+</span> p9.geom_text(label<span class="op">=</span><span class="st">'+'</span>, data<span class="op">=</span>dd.query(<span class="st">'.01 &lt; spearman_p_value &lt; .05'</span>),</span>
<span id="cb127-9"><a aria-hidden="true" href="#cb127-9" tabindex="-1"></a>                     size<span class="op">=</span><span class="dv">5</span>, va<span class="op">=</span><span class="st">'center_baseline'</span>, color<span class="op">=</span><span class="st">'white'</span>)</span>
<span id="cb127-10"><a aria-hidden="true" href="#cb127-10" tabindex="-1"></a>                <span class="op">+</span> p9.geom_text(label<span class="op">=</span><span class="st">'$*$'</span>, data<span class="op">=</span>dd.query(<span class="st">'spearman_p_value &lt; .01'</span>),</span>
<span id="cb127-11"><a aria-hidden="true" href="#cb127-11" tabindex="-1"></a>                     size<span class="op">=</span><span class="dv">5</span>, va<span class="op">=</span><span class="st">'center_baseline'</span>, color<span class="op">=</span><span class="st">'white'</span>)</span>
<span id="cb127-12"><a aria-hidden="true" href="#cb127-12" tabindex="-1"></a>                <span class="op">+</span> p9.scale_fill_continuous(<span class="st">'RdBu_r'</span>, limits<span class="op">=</span>(<span class="op">-</span><span class="fl">.9</span>, <span class="fl">.9</span>))</span>
<span id="cb127-13"><a aria-hidden="true" href="#cb127-13" tabindex="-1"></a>                <span class="op">+</span> p9.theme(figure_size<span class="op">=</span>(<span class="dv">4</span>, <span class="dv">4</span>),</span>
<span id="cb127-14"><a aria-hidden="true" href="#cb127-14" tabindex="-1"></a>                           legend_position<span class="op">=</span><span class="st">'top'</span>,</span>
<span id="cb127-15"><a aria-hidden="true" href="#cb127-15" tabindex="-1"></a>                           legend_key_size<span class="op">=</span><span class="dv">8</span>,</span>
<span id="cb127-16"><a aria-hidden="true" href="#cb127-16" tabindex="-1"></a>                           legend_text<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">7</span>, y<span class="op">=</span><span class="dv">3</span>),</span>
<span id="cb127-17"><a aria-hidden="true" href="#cb127-17" tabindex="-1"></a>                           axis_text_x<span class="op">=</span>p9.element_text(rotation<span class="op">=</span><span class="dv">90</span>, size<span class="op">=</span><span class="dv">7</span>),</span>
<span id="cb127-18"><a aria-hidden="true" href="#cb127-18" tabindex="-1"></a>                           axis_text_y<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">7</span>),</span>
<span id="cb127-19"><a aria-hidden="true" href="#cb127-19" tabindex="-1"></a>                           legend_title<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">8</span>),</span>
<span id="cb127-20"><a aria-hidden="true" href="#cb127-20" tabindex="-1"></a>                )</span>
<span id="cb127-21"><a aria-hidden="true" href="#cb127-21" tabindex="-1"></a>))</span>
<span id="cb127-22"><a aria-hidden="true" href="#cb127-22" tabindex="-1"></a>f.save(<span class="st">'../output/figures/correlations_fungi.svg'</span>, bbox_inches<span class="op">=</span><span class="st">'tight'</span>)</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" id="cell-159">
<div class="sourceCode cell-code" id="cb128"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb128-1"><a aria-hidden="true" href="#cb128-1" tabindex="-1"></a>sig_genera <span class="op">=</span> (corrs_matrix</span>
<span id="cb128-2"><a aria-hidden="true" href="#cb128-2" tabindex="-1"></a>                      .set_index(<span class="st">'Genus'</span>)</span>
<span id="cb128-3"><a aria-hidden="true" href="#cb128-3" tabindex="-1"></a>                      .<span class="bu">apply</span>(<span class="kw">lambda</span> x: x.<span class="bu">max</span>(), axis<span class="op">=</span><span class="dv">1</span>)</span>
<span id="cb128-4"><a aria-hidden="true" href="#cb128-4" tabindex="-1"></a>                      .loc[<span class="kw">lambda</span> x: x <span class="op">&gt;</span> <span class="fl">.544</span>]</span>
<span id="cb128-5"><a aria-hidden="true" href="#cb128-5" tabindex="-1"></a>                      .index.values</span>
<span id="cb128-6"><a aria-hidden="true" href="#cb128-6" tabindex="-1"></a>)</span>
<span id="cb128-7"><a aria-hidden="true" href="#cb128-7" tabindex="-1"></a></span>
<span id="cb128-8"><a aria-hidden="true" href="#cb128-8" tabindex="-1"></a>(corrs_list</span>
<span id="cb128-9"><a aria-hidden="true" href="#cb128-9" tabindex="-1"></a> .loc[<span class="kw">lambda</span> dd: dd.Genus.isin(sig_genera)]</span>
<span id="cb128-10"><a aria-hidden="true" href="#cb128-10" tabindex="-1"></a> .assign(Genus<span class="op">=</span><span class="kw">lambda</span> dd: dd.Genus.<span class="bu">str</span>.split(<span class="st">'-'</span>).<span class="bu">str</span>[<span class="op">-</span><span class="dv">1</span>])</span>
<span id="cb128-11"><a aria-hidden="true" href="#cb128-11" tabindex="-1"></a> .pipe(<span class="kw">lambda</span> dd: p9.ggplot(dd) </span>
<span id="cb128-12"><a aria-hidden="true" href="#cb128-12" tabindex="-1"></a>                <span class="op">+</span> p9.aes(<span class="st">'species'</span>, <span class="st">'Genus'</span>, fill<span class="op">=</span><span class="st">'spearman_r'</span>)</span>
<span id="cb128-13"><a aria-hidden="true" href="#cb128-13" tabindex="-1"></a>                <span class="op">+</span> p9.geom_tile()</span>
<span id="cb128-14"><a aria-hidden="true" href="#cb128-14" tabindex="-1"></a>                <span class="op">+</span> p9.labs(x<span class="op">=</span><span class="st">''</span>, y<span class="op">=</span><span class="st">''</span>, fill<span class="op">=</span><span class="st">"Spearman's $r$"</span>)</span>
<span id="cb128-15"><a aria-hidden="true" href="#cb128-15" tabindex="-1"></a>                <span class="op">+</span> p9.geom_text(label<span class="op">=</span><span class="st">'+'</span>, data<span class="op">=</span>dd.query(<span class="st">'.01 &lt; spearman_p_value &lt; .05'</span>),</span>
<span id="cb128-16"><a aria-hidden="true" href="#cb128-16" tabindex="-1"></a>                     size<span class="op">=</span><span class="dv">5</span>, va<span class="op">=</span><span class="st">'center_baseline'</span>, color<span class="op">=</span><span class="st">'white'</span>)</span>
<span id="cb128-17"><a aria-hidden="true" href="#cb128-17" tabindex="-1"></a>                <span class="op">+</span> p9.geom_text(label<span class="op">=</span><span class="st">'$*$'</span>, data<span class="op">=</span>dd.query(<span class="st">'spearman_p_value &lt; .01'</span>),</span>
<span id="cb128-18"><a aria-hidden="true" href="#cb128-18" tabindex="-1"></a>                     size<span class="op">=</span><span class="dv">5</span>, va<span class="op">=</span><span class="st">'center_baseline'</span>, color<span class="op">=</span><span class="st">'white'</span>)</span>
<span id="cb128-19"><a aria-hidden="true" href="#cb128-19" tabindex="-1"></a>                <span class="op">+</span> p9.scale_fill_continuous(<span class="st">'RdBu_r'</span>, limits<span class="op">=</span>(<span class="op">-</span><span class="fl">.9</span>, <span class="fl">.9</span>))</span>
<span id="cb128-20"><a aria-hidden="true" href="#cb128-20" tabindex="-1"></a>                <span class="op">+</span> p9.theme(figure_size<span class="op">=</span>(<span class="dv">5</span>, <span class="dv">4</span>),</span>
<span id="cb128-21"><a aria-hidden="true" href="#cb128-21" tabindex="-1"></a>                           legend_key_size<span class="op">=</span><span class="dv">8</span>,</span>
<span id="cb128-22"><a aria-hidden="true" href="#cb128-22" tabindex="-1"></a>                           axis_text_x<span class="op">=</span>p9.element_text(rotation<span class="op">=</span><span class="dv">90</span>, size<span class="op">=</span><span class="dv">7</span>),</span>
<span id="cb128-23"><a aria-hidden="true" href="#cb128-23" tabindex="-1"></a>                           axis_text_y<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">7</span>),</span>
<span id="cb128-24"><a aria-hidden="true" href="#cb128-24" tabindex="-1"></a>                           legend_text<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">7</span>),</span>
<span id="cb128-25"><a aria-hidden="true" href="#cb128-25" tabindex="-1"></a>                           legend_title<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">8</span>),</span>
<span id="cb128-26"><a aria-hidden="true" href="#cb128-26" tabindex="-1"></a>                )</span>
<span id="cb128-27"><a aria-hidden="true" href="#cb128-27" tabindex="-1"></a>))</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img class="figure-img" height="1200" src="microbial_diversity_analysis_files/figure-html/cell-123-output-1.png" width="1500"/></p>
</figure>
</div>
</div>
</div>
</section>
<section class="level3" id="pathogen-bubble-plots">
<h3 class="anchored" data-anchor-id="pathogen-bubble-plots">Pathogen bubble plots</h3>
<div class="cell" data-execution_count="79" id="cell-161">
<div class="sourceCode cell-code" id="cb129"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb129-1"><a aria-hidden="true" href="#cb129-1" tabindex="-1"></a>bac_pathogens_df <span class="op">=</span> (</span>
<span id="cb129-2"><a aria-hidden="true" href="#cb129-2" tabindex="-1"></a>    pd.read_excel(<span class="st">'../data/Species 2023_final.xlsx'</span>,</span>
<span id="cb129-3"><a aria-hidden="true" href="#cb129-3" tabindex="-1"></a>                  sheet_name<span class="op">=</span><span class="st">'JV analysis_Bacteria'</span>)</span>
<span id="cb129-4"><a aria-hidden="true" href="#cb129-4" tabindex="-1"></a>                  [[<span class="st">'Genus'</span>, <span class="st">'Species'</span>]]</span>
<span id="cb129-5"><a aria-hidden="true" href="#cb129-5" tabindex="-1"></a>    .assign(label<span class="op">=</span><span class="st">'Bacterial Potential Human Pathogens'</span>)</span>
<span id="cb129-6"><a aria-hidden="true" href="#cb129-6" tabindex="-1"></a>)</span>
<span id="cb129-7"><a aria-hidden="true" href="#cb129-7" tabindex="-1"></a></span>
<span id="cb129-8"><a aria-hidden="true" href="#cb129-8" tabindex="-1"></a>fungi_pathogens_df <span class="op">=</span> (pd.read_excel(<span class="st">'../data/Species 2023_final.xlsx'</span>,</span>
<span id="cb129-9"><a aria-hidden="true" href="#cb129-9" tabindex="-1"></a>                                     sheet_name<span class="op">=</span><span class="st">'JV analysis_Fungi'</span>, </span>
<span id="cb129-10"><a aria-hidden="true" href="#cb129-10" tabindex="-1"></a>                                     skiprows<span class="op">=</span><span class="dv">1</span>)</span>
<span id="cb129-11"><a aria-hidden="true" href="#cb129-11" tabindex="-1"></a>                       [[<span class="st">'Genus'</span>, <span class="st">'Species'</span>]]</span>
<span id="cb129-12"><a aria-hidden="true" href="#cb129-12" tabindex="-1"></a>                       .dropna()</span>
<span id="cb129-13"><a aria-hidden="true" href="#cb129-13" tabindex="-1"></a>    .assign(label<span class="op">=</span><span class="st">'Fungal Potential Human Pathogens'</span>) </span>
<span id="cb129-14"><a aria-hidden="true" href="#cb129-14" tabindex="-1"></a>)</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="24" id="cell-162">
<div class="sourceCode cell-code" id="cb130"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb130-1"><a aria-hidden="true" href="#cb130-1" tabindex="-1"></a>taxon_names <span class="op">=</span> [<span class="st">'Kingdom'</span>, <span class="st">'Phylum'</span>, <span class="st">'Class'</span>, <span class="st">'Order'</span>, <span class="st">'Family'</span>, <span class="st">'Genus'</span>, <span class="st">'Species'</span>]</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="80" id="cell-163">
<div class="sourceCode cell-code" id="cb131"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb131-1"><a aria-hidden="true" href="#cb131-1" tabindex="-1"></a>fun_df_sp <span class="op">=</span> (pd.read_csv(<span class="st">'../data/nov23/ITS/bootstrap90/reduced_ITS_species_(ASV+taxonomy+S1-S22)_90_NB.csv'</span>)</span>
<span id="cb131-2"><a aria-hidden="true" href="#cb131-2" tabindex="-1"></a>          .query(<span class="st">'Kingdom == "k__Fungi"'</span>)</span>
<span id="cb131-3"><a aria-hidden="true" href="#cb131-3" tabindex="-1"></a>)</span>
<span id="cb131-4"><a aria-hidden="true" href="#cb131-4" tabindex="-1"></a>fun_df_sp[taxon_names] <span class="op">=</span> fun_df_sp[taxon_names].<span class="bu">apply</span>(<span class="kw">lambda</span> x: x.<span class="bu">str</span>.split(<span class="st">'__'</span>).<span class="bu">str</span>[<span class="op">-</span><span class="dv">1</span>])</span>
<span id="cb131-5"><a aria-hidden="true" href="#cb131-5" tabindex="-1"></a>fun_df_sp <span class="op">=</span> fun_df_sp.rename(columns<span class="op">=</span>zfill_map).fillna(<span class="st">'unknown'</span>)</span>
<span id="cb131-6"><a aria-hidden="true" href="#cb131-6" tabindex="-1"></a></span>
<span id="cb131-7"><a aria-hidden="true" href="#cb131-7" tabindex="-1"></a>bac_df_sp <span class="op">=</span> (pd.read_csv(<span class="st">'../data/nov23/16S/Boostraping RDP_NBC_90/reduced_16S_species_(ASV+taxonomy+S1-S22)_90_NB.csv'</span>)</span>
<span id="cb131-8"><a aria-hidden="true" href="#cb131-8" tabindex="-1"></a>            .rename(columns<span class="op">=</span>zfill_map)</span>
<span id="cb131-9"><a aria-hidden="true" href="#cb131-9" tabindex="-1"></a>            .assign(Genus<span class="op">=</span><span class="kw">lambda</span> dd: dd.Genus.<span class="bu">str</span>.replace(<span class="st">' '</span>, <span class="st">'_'</span>))</span>
<span id="cb131-10"><a aria-hidden="true" href="#cb131-10" tabindex="-1"></a>            .fillna(<span class="st">'unknown'</span>)</span>
<span id="cb131-11"><a aria-hidden="true" href="#cb131-11" tabindex="-1"></a>)</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="107" id="cell-164">
<div class="sourceCode cell-code" id="cb132"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb132-1"><a aria-hidden="true" href="#cb132-1" tabindex="-1"></a>(bac_df_sp.query(<span class="st">'Genus=="Serratia"'</span>).set_index(taxon_names).T.<span class="bu">sum</span>().reset_index()</span>
<span id="cb132-2"><a aria-hidden="true" href="#cb132-2" tabindex="-1"></a> .assign(freq<span class="op">=</span><span class="kw">lambda</span> dd: dd[<span class="dv">0</span>] <span class="op">/</span> dd[<span class="dv">0</span>].<span class="bu">sum</span>())</span>
<span id="cb132-3"><a aria-hidden="true" href="#cb132-3" tabindex="-1"></a>)</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="107">
<div>
<table class="dataframe caption-top table table-sm table-striped small" data-border="1" data-quarto-postprocess="true">
<thead>
<tr class="header">
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th">Kingdom</th>
<th data-quarto-table-cell-role="th">Phylum</th>
<th data-quarto-table-cell-role="th">Class</th>
<th data-quarto-table-cell-role="th">Order</th>
<th data-quarto-table-cell-role="th">Family</th>
<th data-quarto-table-cell-role="th">Genus</th>
<th data-quarto-table-cell-role="th">Species</th>
<th data-quarto-table-cell-role="th">0</th>
<th data-quarto-table-cell-role="th">freq</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td data-quarto-table-cell-role="th">0</td>
<td>Bacteria</td>
<td>Proteobacteria</td>
<td>Gammaproteobacteria</td>
<td>Enterobacterales</td>
<td>Yersiniaceae</td>
<td>Serratia</td>
<td>marcescens</td>
<td>131</td>
<td>0.0298</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">1</td>
<td>Bacteria</td>
<td>Proteobacteria</td>
<td>Gammaproteobacteria</td>
<td>Enterobacterales</td>
<td>Yersiniaceae</td>
<td>Serratia</td>
<td>unknown</td>
<td>4265</td>
<td>0.9702</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="cell" data-execution_count="82" id="cell-165">
<div class="sourceCode cell-code" id="cb133"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb133-1"><a aria-hidden="true" href="#cb133-1" tabindex="-1"></a>f <span class="op">=</span> (pd.concat([bac_df_sp, fun_df_sp])</span>
<span id="cb133-2"><a aria-hidden="true" href="#cb133-2" tabindex="-1"></a>.merge(pd.concat([bac_pathogens_df, fungi_pathogens_df]),</span>
<span id="cb133-3"><a aria-hidden="true" href="#cb133-3" tabindex="-1"></a>       on<span class="op">=</span>[<span class="st">'Genus'</span>, <span class="st">'Species'</span>], how<span class="op">=</span><span class="st">'right'</span>)</span>
<span id="cb133-4"><a aria-hidden="true" href="#cb133-4" tabindex="-1"></a>.drop(columns<span class="op">=</span>taxon_names[:<span class="op">-</span><span class="dv">2</span>])</span>
<span id="cb133-5"><a aria-hidden="true" href="#cb133-5" tabindex="-1"></a>.assign(label<span class="op">=</span><span class="kw">lambda</span> dd: dd.label.fillna(<span class="st">'Fungal Human Pathogens'</span>))</span>
<span id="cb133-6"><a aria-hidden="true" href="#cb133-6" tabindex="-1"></a>.melt([<span class="st">'Genus'</span>, <span class="st">'Species'</span>, <span class="st">'label'</span>], var_name<span class="op">=</span><span class="st">'sample_id'</span>, value_name<span class="op">=</span><span class="st">'counts'</span>)</span>
<span id="cb133-7"><a aria-hidden="true" href="#cb133-7" tabindex="-1"></a>.merge(samples_info, on<span class="op">=</span><span class="st">'sample_id'</span>)</span>
<span id="cb133-8"><a aria-hidden="true" href="#cb133-8" tabindex="-1"></a>.groupby([<span class="st">'label'</span>, <span class="st">'sample_mode'</span>, <span class="st">'sample_id'</span>])</span>
<span id="cb133-9"><a aria-hidden="true" href="#cb133-9" tabindex="-1"></a>.aggregate({<span class="st">'counts'</span>: <span class="st">'sum'</span>})</span>
<span id="cb133-10"><a aria-hidden="true" href="#cb133-10" tabindex="-1"></a>.reset_index()</span>
<span id="cb133-11"><a aria-hidden="true" href="#cb133-11" tabindex="-1"></a>.groupby([<span class="st">'label'</span>, <span class="st">'sample_mode'</span>])</span>
<span id="cb133-12"><a aria-hidden="true" href="#cb133-12" tabindex="-1"></a>.aggregate({<span class="st">'counts'</span>: <span class="st">'mean'</span>})</span>
<span id="cb133-13"><a aria-hidden="true" href="#cb133-13" tabindex="-1"></a>.reset_index()</span>
<span id="cb133-14"><a aria-hidden="true" href="#cb133-14" tabindex="-1"></a>.assign(Kingdom<span class="op">=</span><span class="kw">lambda</span> dd: dd.label.<span class="bu">str</span>.split().<span class="bu">str</span>[<span class="dv">0</span>])</span>
<span id="cb133-15"><a aria-hidden="true" href="#cb133-15" tabindex="-1"></a>.replace({<span class="st">'Kingdom'</span>: {<span class="st">'Bacterial'</span>: <span class="st">'Bacteria'</span>, <span class="st">'Fungal'</span>: <span class="st">'Fungi'</span>}})</span>
<span id="cb133-16"><a aria-hidden="true" href="#cb133-16" tabindex="-1"></a>.assign(label<span class="op">=</span><span class="kw">lambda</span> dd: dd.label.<span class="bu">str</span>.split().<span class="bu">str</span>[<span class="dv">1</span>:].<span class="bu">str</span>.join(<span class="st">' '</span>))</span>
<span id="cb133-17"><a aria-hidden="true" href="#cb133-17" tabindex="-1"></a>.pipe(<span class="kw">lambda</span> dd: p9.ggplot(dd) <span class="op">+</span> p9.aes(<span class="st">'label'</span>, <span class="st">'counts'</span>, fill<span class="op">=</span><span class="st">'sample_mode'</span>)</span>
<span id="cb133-18"><a aria-hidden="true" href="#cb133-18" tabindex="-1"></a><span class="op">+</span> p9.geom_col(position<span class="op">=</span><span class="st">'dodge'</span>)</span>
<span id="cb133-19"><a aria-hidden="true" href="#cb133-19" tabindex="-1"></a><span class="op">+</span> p9.scale_fill_manual(values<span class="op">=</span>[<span class="st">'#D3894C'</span>, <span class="st">'#435B97'</span>])</span>
<span id="cb133-20"><a aria-hidden="true" href="#cb133-20" tabindex="-1"></a><span class="op">+</span> p9.facet_wrap(<span class="st">'Kingdom'</span>, scales<span class="op">=</span><span class="st">'free_x'</span>, ncol<span class="op">=</span><span class="dv">1</span>)</span>
<span id="cb133-21"><a aria-hidden="true" href="#cb133-21" tabindex="-1"></a><span class="op">+</span> p9.labs(x<span class="op">=</span><span class="st">''</span>, y<span class="op">=</span><span class="st">'Read counts'</span>, fill<span class="op">=</span><span class="st">''</span>)</span>
<span id="cb133-22"><a aria-hidden="true" href="#cb133-22" tabindex="-1"></a><span class="op">+</span> p9.coord_flip()</span>
<span id="cb133-23"><a aria-hidden="true" href="#cb133-23" tabindex="-1"></a><span class="op">+</span> p9.theme(legend_position<span class="op">=</span><span class="st">'top'</span>,</span>
<span id="cb133-24"><a aria-hidden="true" href="#cb133-24" tabindex="-1"></a>           figure_size<span class="op">=</span>(<span class="dv">5</span>, <span class="dv">5</span>),</span>
<span id="cb133-25"><a aria-hidden="true" href="#cb133-25" tabindex="-1"></a>           strip_text<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">12</span>),</span>
<span id="cb133-26"><a aria-hidden="true" href="#cb133-26" tabindex="-1"></a>           )</span>
<span id="cb133-27"><a aria-hidden="true" href="#cb133-27" tabindex="-1"></a>)</span>
<span id="cb133-28"><a aria-hidden="true" href="#cb133-28" tabindex="-1"></a>)</span>
<span id="cb133-29"><a aria-hidden="true" href="#cb133-29" tabindex="-1"></a>f.save(<span class="st">'../output/figures/pathogens_bars.svg'</span>)</span>
<span id="cb133-30"><a aria-hidden="true" href="#cb133-30" tabindex="-1"></a>f.draw()</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="82">
<div>
<figure class="figure">
<p><img class="figure-img" height="1530" src="microbial_diversity_analysis_files/figure-html/cell-128-output-1.png" width="1537"/></p>
</figure>
</div>
</div>
</div>
<div class="cell" data-execution_count="83" id="cell-166">
<div class="sourceCode cell-code" id="cb134"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb134-1"><a aria-hidden="true" href="#cb134-1" tabindex="-1"></a>all_species_df <span class="op">=</span> (pd.read_csv(<span class="st">'../data/nov23/ALL_AND_BLANKS_WITH_POOLS/complete_table_naivebayes.csv'</span>)</span>
<span id="cb134-2"><a aria-hidden="true" href="#cb134-2" tabindex="-1"></a>                  .assign(Genus<span class="op">=</span><span class="kw">lambda</span> dd: dd.Genus.<span class="bu">str</span>.replace(<span class="st">' '</span>, <span class="st">'_'</span>))</span>
<span id="cb134-3"><a aria-hidden="true" href="#cb134-3" tabindex="-1"></a>                  )</span>
<span id="cb134-4"><a aria-hidden="true" href="#cb134-4" tabindex="-1"></a>blank_species_df <span class="op">=</span> (pd.read_csv(<span class="st">'../data/nov23/ALL_AND_BLANKS_WITH_POOLS/complete_table_naivebayes.csv'</span>)</span>
<span id="cb134-5"><a aria-hidden="true" href="#cb134-5" tabindex="-1"></a> .query(<span class="st">'Species.notna()'</span>)</span>
<span id="cb134-6"><a aria-hidden="true" href="#cb134-6" tabindex="-1"></a> .assign(Genus<span class="op">=</span><span class="kw">lambda</span> dd: dd.Genus.<span class="bu">str</span>.replace(<span class="st">' '</span>, <span class="st">'_'</span>))</span>
<span id="cb134-7"><a aria-hidden="true" href="#cb134-7" tabindex="-1"></a> [[ <span class="st">'Genus'</span>, <span class="st">'Species'</span>, <span class="st">'Blank-Flightpoolsubsampled'</span>, <span class="st">'Blank-Groundpoolsubsampled'</span>]]</span>
<span id="cb134-8"><a aria-hidden="true" href="#cb134-8" tabindex="-1"></a> .melt([<span class="st">'Genus'</span>, <span class="st">'Species'</span>], var_name<span class="op">=</span><span class="st">'sample_id'</span>, value_name<span class="op">=</span><span class="st">'counts'</span>)</span>
<span id="cb134-9"><a aria-hidden="true" href="#cb134-9" tabindex="-1"></a> .assign(sample_label<span class="op">=</span><span class="kw">lambda</span> dd: </span>
<span id="cb134-10"><a aria-hidden="true" href="#cb134-10" tabindex="-1"></a>         np.where(dd.sample_id.<span class="bu">str</span>.contains(<span class="st">'Flight'</span>), <span class="st">'Flight Blanks'</span>, <span class="st">'Ground Blanks'</span>))</span>
<span id="cb134-11"><a aria-hidden="true" href="#cb134-11" tabindex="-1"></a> .drop(columns<span class="op">=</span><span class="st">'sample_id'</span>)</span>
<span id="cb134-12"><a aria-hidden="true" href="#cb134-12" tabindex="-1"></a> .groupby([<span class="st">'Genus'</span>, <span class="st">'Species'</span>, <span class="st">'sample_label'</span>])</span>
<span id="cb134-13"><a aria-hidden="true" href="#cb134-13" tabindex="-1"></a>        .aggregate({<span class="st">'counts'</span>: <span class="st">'sum'</span>})</span>
<span id="cb134-14"><a aria-hidden="true" href="#cb134-14" tabindex="-1"></a> .reset_index()</span>
<span id="cb134-15"><a aria-hidden="true" href="#cb134-15" tabindex="-1"></a> .groupby(<span class="st">'sample_label'</span>, as_index<span class="op">=</span><span class="va">False</span>)</span>
<span id="cb134-16"><a aria-hidden="true" href="#cb134-16" tabindex="-1"></a> .<span class="bu">apply</span>(<span class="kw">lambda</span> dd: dd.assign(relative_abundance<span class="op">=</span><span class="kw">lambda</span> dd: dd.counts <span class="op">/</span> dd.counts.<span class="bu">sum</span>()))</span>
<span id="cb134-17"><a aria-hidden="true" href="#cb134-17" tabindex="-1"></a>)</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="84" id="cell-167">
<div class="sourceCode cell-code" id="cb135"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb135-1"><a aria-hidden="true" href="#cb135-1" tabindex="-1"></a>taxon_names <span class="op">=</span> [<span class="st">'Kingdom'</span>, <span class="st">'Phylum'</span>, <span class="st">'Class'</span>, <span class="st">'Order'</span>, <span class="st">'Family'</span>, <span class="st">'Genus'</span>, <span class="st">'Species'</span>]</span>
<span id="cb135-2"><a aria-hidden="true" href="#cb135-2" tabindex="-1"></a>sample_fun_path <span class="op">=</span> (fun_df_sp</span>
<span id="cb135-3"><a aria-hidden="true" href="#cb135-3" tabindex="-1"></a>    .assign(Species<span class="op">=</span><span class="kw">lambda</span> dd: dd.Species.<span class="bu">str</span>.split(<span class="st">'__'</span>).<span class="bu">str</span>[<span class="op">-</span><span class="dv">1</span>])</span>
<span id="cb135-4"><a aria-hidden="true" href="#cb135-4" tabindex="-1"></a>    .melt(taxon_names, var_name<span class="op">=</span><span class="st">'sample_id'</span>, value_name<span class="op">=</span><span class="st">'counts'</span>)</span>
<span id="cb135-5"><a aria-hidden="true" href="#cb135-5" tabindex="-1"></a>    .assign(sample_mode<span class="op">=</span><span class="kw">lambda</span> dd: dd.sample_id.<span class="bu">map</span>(sample_map))</span>
<span id="cb135-6"><a aria-hidden="true" href="#cb135-6" tabindex="-1"></a>    .assign(sample_label<span class="op">=</span><span class="kw">lambda</span> dd: dd.sample_mode <span class="op">+</span> <span class="st">' samples'</span>)</span>
<span id="cb135-7"><a aria-hidden="true" href="#cb135-7" tabindex="-1"></a>    .groupby([<span class="st">'Genus'</span>, <span class="st">'Species'</span>, <span class="st">'sample_label'</span>])</span>
<span id="cb135-8"><a aria-hidden="true" href="#cb135-8" tabindex="-1"></a>    .aggregate({<span class="st">'counts'</span>: <span class="st">'sum'</span>})</span>
<span id="cb135-9"><a aria-hidden="true" href="#cb135-9" tabindex="-1"></a>    .groupby(<span class="st">'sample_label'</span>, as_index<span class="op">=</span><span class="va">False</span>)</span>
<span id="cb135-10"><a aria-hidden="true" href="#cb135-10" tabindex="-1"></a>    .<span class="bu">apply</span>(<span class="kw">lambda</span> dd: dd.assign(relative_abundance<span class="op">=</span><span class="kw">lambda</span> dd: dd.counts <span class="op">/</span> dd.counts.<span class="bu">sum</span>()))</span>
<span id="cb135-11"><a aria-hidden="true" href="#cb135-11" tabindex="-1"></a>    .reset_index()</span>
<span id="cb135-12"><a aria-hidden="true" href="#cb135-12" tabindex="-1"></a>    .merge(fungi_pathogens_df)</span>
<span id="cb135-13"><a aria-hidden="true" href="#cb135-13" tabindex="-1"></a>)</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="85" id="cell-168">
<div class="sourceCode cell-code" id="cb136"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb136-1"><a aria-hidden="true" href="#cb136-1" tabindex="-1"></a>blank_bac_pathogens <span class="op">=</span> (all_species_df</span>
<span id="cb136-2"><a aria-hidden="true" href="#cb136-2" tabindex="-1"></a> .fillna(<span class="st">'unknown'</span>)</span>
<span id="cb136-3"><a aria-hidden="true" href="#cb136-3" tabindex="-1"></a> .groupby([<span class="st">'Genus'</span>, <span class="st">'Species'</span>])</span>
<span id="cb136-4"><a aria-hidden="true" href="#cb136-4" tabindex="-1"></a> .<span class="bu">sum</span>(numeric_only<span class="op">=</span><span class="va">True</span>)</span>
<span id="cb136-5"><a aria-hidden="true" href="#cb136-5" tabindex="-1"></a> [[<span class="st">'Blank-Flightpoolsubsampled'</span>, <span class="st">'Blank-Groundpoolsubsampled'</span>]]</span>
<span id="cb136-6"><a aria-hidden="true" href="#cb136-6" tabindex="-1"></a> .reset_index()</span>
<span id="cb136-7"><a aria-hidden="true" href="#cb136-7" tabindex="-1"></a> .melt([<span class="st">'Genus'</span>, <span class="st">'Species'</span>])</span>
<span id="cb136-8"><a aria-hidden="true" href="#cb136-8" tabindex="-1"></a> .groupby(<span class="st">'variable'</span>)</span>
<span id="cb136-9"><a aria-hidden="true" href="#cb136-9" tabindex="-1"></a> .<span class="bu">apply</span>(<span class="kw">lambda</span> dd: dd.assign(relative_abundance<span class="op">=</span><span class="kw">lambda</span> dd: dd.value <span class="op">/</span> dd.value.<span class="bu">sum</span>()))</span>
<span id="cb136-10"><a aria-hidden="true" href="#cb136-10" tabindex="-1"></a> .merge(bac_pathogens_df, on<span class="op">=</span>[<span class="st">'Genus'</span>, <span class="st">'Species'</span>], how<span class="op">=</span><span class="st">'right'</span>)</span>
<span id="cb136-11"><a aria-hidden="true" href="#cb136-11" tabindex="-1"></a> .assign(value<span class="op">=</span><span class="kw">lambda</span> dd: dd.value.fillna(<span class="dv">0</span>))</span>
<span id="cb136-12"><a aria-hidden="true" href="#cb136-12" tabindex="-1"></a> .query(<span class="st">'variable.notna()'</span>)</span>
<span id="cb136-13"><a aria-hidden="true" href="#cb136-13" tabindex="-1"></a> .replace({</span>
<span id="cb136-14"><a aria-hidden="true" href="#cb136-14" tabindex="-1"></a>     <span class="st">'Blank-Flightpoolsubsampled'</span>: <span class="st">'Flight Blanks'</span>,</span>
<span id="cb136-15"><a aria-hidden="true" href="#cb136-15" tabindex="-1"></a>     <span class="st">'Blank-Groundpoolsubsampled'</span>: <span class="st">'Ground Blanks'</span></span>
<span id="cb136-16"><a aria-hidden="true" href="#cb136-16" tabindex="-1"></a> })</span>
<span id="cb136-17"><a aria-hidden="true" href="#cb136-17" tabindex="-1"></a> .rename(columns<span class="op">=</span>{<span class="st">'variable'</span>: <span class="st">'sample_label'</span>})</span>
<span id="cb136-18"><a aria-hidden="true" href="#cb136-18" tabindex="-1"></a>)</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="86" id="cell-169">
<div class="sourceCode cell-code" id="cb137"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb137-1"><a aria-hidden="true" href="#cb137-1" tabindex="-1"></a>samples_pathogens_bac_df <span class="op">=</span> (bac_df_sp</span>
<span id="cb137-2"><a aria-hidden="true" href="#cb137-2" tabindex="-1"></a> .fillna(<span class="st">'unknown'</span>)</span>
<span id="cb137-3"><a aria-hidden="true" href="#cb137-3" tabindex="-1"></a> .groupby([<span class="st">'Genus'</span>, <span class="st">'Species'</span>])</span>
<span id="cb137-4"><a aria-hidden="true" href="#cb137-4" tabindex="-1"></a> .<span class="bu">sum</span>(numeric_only<span class="op">=</span><span class="va">True</span>)</span>
<span id="cb137-5"><a aria-hidden="true" href="#cb137-5" tabindex="-1"></a> .reset_index()</span>
<span id="cb137-6"><a aria-hidden="true" href="#cb137-6" tabindex="-1"></a> .melt([<span class="st">'Genus'</span>, <span class="st">'Species'</span>])</span>
<span id="cb137-7"><a aria-hidden="true" href="#cb137-7" tabindex="-1"></a> .assign(value<span class="op">=</span><span class="kw">lambda</span> dd: dd.value.fillna(<span class="dv">0</span>))</span>
<span id="cb137-8"><a aria-hidden="true" href="#cb137-8" tabindex="-1"></a> .query(<span class="st">'variable.notna()'</span>)</span>
<span id="cb137-9"><a aria-hidden="true" href="#cb137-9" tabindex="-1"></a> .assign(sample_mode<span class="op">=</span><span class="kw">lambda</span> dd: dd.variable.<span class="bu">map</span>(sample_map))</span>
<span id="cb137-10"><a aria-hidden="true" href="#cb137-10" tabindex="-1"></a> .groupby([<span class="st">'Genus'</span>, <span class="st">'Species'</span>, <span class="st">'sample_mode'</span>], as_index<span class="op">=</span><span class="va">False</span>)</span>
<span id="cb137-11"><a aria-hidden="true" href="#cb137-11" tabindex="-1"></a> .value</span>
<span id="cb137-12"><a aria-hidden="true" href="#cb137-12" tabindex="-1"></a> .mean()</span>
<span id="cb137-13"><a aria-hidden="true" href="#cb137-13" tabindex="-1"></a> .<span class="bu">round</span>(<span class="dv">1</span>)</span>
<span id="cb137-14"><a aria-hidden="true" href="#cb137-14" tabindex="-1"></a> .assign(sample_label<span class="op">=</span><span class="kw">lambda</span> dd: dd.sample_mode <span class="op">+</span> <span class="st">' samples'</span>)</span>
<span id="cb137-15"><a aria-hidden="true" href="#cb137-15" tabindex="-1"></a> .groupby(<span class="st">'sample_label'</span>, as_index<span class="op">=</span><span class="va">False</span>)</span>
<span id="cb137-16"><a aria-hidden="true" href="#cb137-16" tabindex="-1"></a> .<span class="bu">apply</span>(<span class="kw">lambda</span> dd: dd.assign(relative_abundance<span class="op">=</span><span class="kw">lambda</span> dd: dd.value <span class="op">/</span> dd.value.<span class="bu">sum</span>()))</span>
<span id="cb137-17"><a aria-hidden="true" href="#cb137-17" tabindex="-1"></a> .merge(bac_pathogens_df, on<span class="op">=</span>[<span class="st">'Genus'</span>, <span class="st">'Species'</span>], how<span class="op">=</span><span class="st">'right'</span>)</span>
<span id="cb137-18"><a aria-hidden="true" href="#cb137-18" tabindex="-1"></a> .drop(columns<span class="op">=</span><span class="st">'sample_mode'</span>)</span>
<span id="cb137-19"><a aria-hidden="true" href="#cb137-19" tabindex="-1"></a>)</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="128" id="cell-170">
<div class="sourceCode cell-code" id="cb138"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb138-1"><a aria-hidden="true" href="#cb138-1" tabindex="-1"></a>bac_names <span class="op">=</span> (pd.concat([blank_bac_pathogens, samples_pathogens_bac_df])</span>
<span id="cb138-2"><a aria-hidden="true" href="#cb138-2" tabindex="-1"></a> .query(<span class="st">'sample_label.notna()'</span>)</span>
<span id="cb138-3"><a aria-hidden="true" href="#cb138-3" tabindex="-1"></a> .pivot(index<span class="op">=</span>[<span class="st">'Genus'</span>, <span class="st">'Species'</span>], columns<span class="op">=</span><span class="st">'sample_label'</span>, values<span class="op">=</span><span class="st">'relative_abundance'</span>)</span>
<span id="cb138-4"><a aria-hidden="true" href="#cb138-4" tabindex="-1"></a> .fillna(<span class="dv">0</span>)</span>
<span id="cb138-5"><a aria-hidden="true" href="#cb138-5" tabindex="-1"></a> .loc[<span class="kw">lambda</span> dd: dd.<span class="bu">sum</span>(axis<span class="op">=</span><span class="dv">1</span>) <span class="op">&gt;</span> <span class="dv">0</span>]</span>
<span id="cb138-6"><a aria-hidden="true" href="#cb138-6" tabindex="-1"></a> .loc[<span class="kw">lambda</span> dd: dd.<span class="bu">max</span>(axis<span class="op">=</span><span class="dv">1</span>).<span class="bu">round</span>(<span class="dv">5</span>) <span class="op">&gt;=</span> <span class="fl">.0001</span>]</span>
<span id="cb138-7"><a aria-hidden="true" href="#cb138-7" tabindex="-1"></a> .reset_index()</span>
<span id="cb138-8"><a aria-hidden="true" href="#cb138-8" tabindex="-1"></a> .assign(total<span class="op">=</span><span class="kw">lambda</span> dd: dd[<span class="st">'Flight samples'</span>] <span class="op">+</span> dd[<span class="st">'Ground samples'</span>])</span>
<span id="cb138-9"><a aria-hidden="true" href="#cb138-9" tabindex="-1"></a> .sort_values(<span class="st">'total'</span>)</span>
<span id="cb138-10"><a aria-hidden="true" href="#cb138-10" tabindex="-1"></a> .assign(name<span class="op">=</span><span class="kw">lambda</span> dd: dd.Genus.<span class="bu">str</span>[<span class="dv">0</span>] <span class="op">+</span> <span class="st">'. '</span> <span class="op">+</span> dd.Species)</span>
<span id="cb138-11"><a aria-hidden="true" href="#cb138-11" tabindex="-1"></a> [<span class="st">'name'</span>]</span>
<span id="cb138-12"><a aria-hidden="true" href="#cb138-12" tabindex="-1"></a>)</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="133" id="cell-171">
<div class="sourceCode cell-code" id="cb139"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb139-1"><a aria-hidden="true" href="#cb139-1" tabindex="-1"></a>f <span class="op">=</span> (pd.concat([blank_bac_pathogens, samples_pathogens_bac_df])</span>
<span id="cb139-2"><a aria-hidden="true" href="#cb139-2" tabindex="-1"></a> .query(<span class="st">'sample_label.notna()'</span>)</span>
<span id="cb139-3"><a aria-hidden="true" href="#cb139-3" tabindex="-1"></a> .pivot(index<span class="op">=</span>[<span class="st">'Genus'</span>, <span class="st">'Species'</span>], columns<span class="op">=</span><span class="st">'sample_label'</span>, values<span class="op">=</span><span class="st">'relative_abundance'</span>)</span>
<span id="cb139-4"><a aria-hidden="true" href="#cb139-4" tabindex="-1"></a> .fillna(<span class="dv">0</span>)</span>
<span id="cb139-5"><a aria-hidden="true" href="#cb139-5" tabindex="-1"></a> .loc[<span class="kw">lambda</span> dd: dd.<span class="bu">sum</span>(axis<span class="op">=</span><span class="dv">1</span>) <span class="op">&gt;</span> <span class="dv">0</span>]</span>
<span id="cb139-6"><a aria-hidden="true" href="#cb139-6" tabindex="-1"></a> .loc[<span class="kw">lambda</span> dd: dd.<span class="bu">max</span>(axis<span class="op">=</span><span class="dv">1</span>).<span class="bu">round</span>(<span class="dv">5</span>) <span class="op">&gt;=</span> <span class="fl">.0001</span>]</span>
<span id="cb139-7"><a aria-hidden="true" href="#cb139-7" tabindex="-1"></a> .reset_index()</span>
<span id="cb139-8"><a aria-hidden="true" href="#cb139-8" tabindex="-1"></a> .melt([<span class="st">'Genus'</span>, <span class="st">'Species'</span>], value_name<span class="op">=</span><span class="st">'relative_abundance'</span>)</span>
<span id="cb139-9"><a aria-hidden="true" href="#cb139-9" tabindex="-1"></a> .assign(Genus_short<span class="op">=</span><span class="kw">lambda</span> dd: dd.Genus.<span class="bu">str</span>[<span class="dv">0</span>] <span class="op">+</span> <span class="st">'. '</span>)</span>
<span id="cb139-10"><a aria-hidden="true" href="#cb139-10" tabindex="-1"></a> .assign(fullname<span class="op">=</span><span class="kw">lambda</span> dd: dd.Genus_short <span class="op">+</span> dd.Species)</span>
<span id="cb139-11"><a aria-hidden="true" href="#cb139-11" tabindex="-1"></a> .query(<span class="st">'fullname != "C. gingivalis"'</span>)</span>
<span id="cb139-12"><a aria-hidden="true" href="#cb139-12" tabindex="-1"></a> .assign(sample_label<span class="op">=</span><span class="kw">lambda</span> dd: dd.sample_label.<span class="bu">str</span>.replace(<span class="st">' '</span>, <span class="st">'</span><span class="ch">\n</span><span class="st">'</span>))</span>
<span id="cb139-13"><a aria-hidden="true" href="#cb139-13" tabindex="-1"></a> .assign(sample_label<span class="op">=</span><span class="kw">lambda</span> dd: </span>
<span id="cb139-14"><a aria-hidden="true" href="#cb139-14" tabindex="-1"></a>         pd.Categorical(dd.sample_label,</span>
<span id="cb139-15"><a aria-hidden="true" href="#cb139-15" tabindex="-1"></a>    categories<span class="op">=</span>[<span class="st">'Flight</span><span class="ch">\n</span><span class="st">samples'</span>, <span class="st">'Ground</span><span class="ch">\n</span><span class="st">samples'</span>,</span>
<span id="cb139-16"><a aria-hidden="true" href="#cb139-16" tabindex="-1"></a>                <span class="st">'Flight</span><span class="ch">\n</span><span class="st">Blanks'</span>, <span class="st">'Ground</span><span class="ch">\n</span><span class="st">Blanks'</span>])</span>
<span id="cb139-17"><a aria-hidden="true" href="#cb139-17" tabindex="-1"></a>    )</span>
<span id="cb139-18"><a aria-hidden="true" href="#cb139-18" tabindex="-1"></a> .merge(bac_pathogens_df)</span>
<span id="cb139-19"><a aria-hidden="true" href="#cb139-19" tabindex="-1"></a> .assign(fullname<span class="op">=</span><span class="kw">lambda</span> dd: pd.Categorical(dd.fullname, categories<span class="op">=</span>bac_names, ordered<span class="op">=</span><span class="va">True</span>))</span>
<span id="cb139-20"><a aria-hidden="true" href="#cb139-20" tabindex="-1"></a> .pipe(<span class="kw">lambda</span> dd: p9.ggplot(dd)</span>
<span id="cb139-21"><a aria-hidden="true" href="#cb139-21" tabindex="-1"></a>    <span class="op">+</span> p9.aes(<span class="st">'fullname'</span>, <span class="st">'sample_label'</span>)</span>
<span id="cb139-22"><a aria-hidden="true" href="#cb139-22" tabindex="-1"></a>    <span class="op">+</span> p9.geom_tile(p9.aes(fill<span class="op">=</span><span class="st">'relative_abundance &gt; 0'</span>), color<span class="op">=</span><span class="st">'black'</span>)</span>
<span id="cb139-23"><a aria-hidden="true" href="#cb139-23" tabindex="-1"></a>    <span class="op">+</span> p9.coord_flip()</span>
<span id="cb139-24"><a aria-hidden="true" href="#cb139-24" tabindex="-1"></a>    <span class="op">+</span> p9.geom_text(p9.aes(label<span class="op">=</span><span class="st">'round(relative_abundance * 100, 3).astype(str) + "%"'</span>), size<span class="op">=</span><span class="dv">8</span>)</span>
<span id="cb139-25"><a aria-hidden="true" href="#cb139-25" tabindex="-1"></a>    <span class="op">+</span> p9.labs(x<span class="op">=</span><span class="st">''</span>, y<span class="op">=</span><span class="st">''</span>)</span>
<span id="cb139-26"><a aria-hidden="true" href="#cb139-26" tabindex="-1"></a><span class="co">#     + p9.facet_wrap('label', scales='free_y', ncol=1)</span></span>
<span id="cb139-27"><a aria-hidden="true" href="#cb139-27" tabindex="-1"></a>    <span class="op">+</span> p9.scale_x_discrete(expand<span class="op">=</span>(<span class="dv">0</span>, <span class="dv">0</span>), trans<span class="op">=</span><span class="st">'reverse'</span>)</span>
<span id="cb139-28"><a aria-hidden="true" href="#cb139-28" tabindex="-1"></a>    <span class="op">+</span> p9.scale_y_discrete(expand<span class="op">=</span>(<span class="dv">0</span>, <span class="dv">0</span>))</span>
<span id="cb139-29"><a aria-hidden="true" href="#cb139-29" tabindex="-1"></a>    <span class="op">+</span> p9.guides(fill<span class="op">=</span><span class="va">False</span>)</span>
<span id="cb139-30"><a aria-hidden="true" href="#cb139-30" tabindex="-1"></a>    <span class="op">+</span> p9.theme(figure_size<span class="op">=</span>(<span class="fl">3.5</span>, <span class="dv">8</span>),</span>
<span id="cb139-31"><a aria-hidden="true" href="#cb139-31" tabindex="-1"></a>               axis_text_x<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">8</span>),</span>
<span id="cb139-32"><a aria-hidden="true" href="#cb139-32" tabindex="-1"></a>               axis_text_y<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">8</span>, face<span class="op">=</span><span class="st">'italic'</span>),</span>
<span id="cb139-33"><a aria-hidden="true" href="#cb139-33" tabindex="-1"></a>               strip_text<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">8</span>),</span>
<span id="cb139-34"><a aria-hidden="true" href="#cb139-34" tabindex="-1"></a>               )</span>
<span id="cb139-35"><a aria-hidden="true" href="#cb139-35" tabindex="-1"></a>       )</span>
<span id="cb139-36"><a aria-hidden="true" href="#cb139-36" tabindex="-1"></a>)</span>
<span id="cb139-37"><a aria-hidden="true" href="#cb139-37" tabindex="-1"></a>f.save(<span class="st">'../output/figures/bac_pathogens_table.svg'</span>)</span>
<span id="cb139-38"><a aria-hidden="true" href="#cb139-38" tabindex="-1"></a>f.draw()</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="133">
<div>
<figure class="figure">
<p><img class="figure-img" height="2439" src="microbial_diversity_analysis_files/figure-html/cell-134-output-1.png" width="1089"/></p>
</figure>
</div>
</div>
</div>
<div class="cell" id="cell-172">
<div class="sourceCode cell-code" id="cb140"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb140-1"><a aria-hidden="true" href="#cb140-1" tabindex="-1"></a>fun_names <span class="op">=</span> (pd.concat([sample_fun_path, blank_species_df])</span>
<span id="cb140-2"><a aria-hidden="true" href="#cb140-2" tabindex="-1"></a> .pivot(index<span class="op">=</span>[<span class="st">'Genus'</span>, <span class="st">'Species'</span>], columns<span class="op">=</span><span class="st">'sample_label'</span>, values<span class="op">=</span><span class="st">'relative_abundance'</span>)</span>
<span id="cb140-3"><a aria-hidden="true" href="#cb140-3" tabindex="-1"></a> .fillna(<span class="dv">0</span>)</span>
<span id="cb140-4"><a aria-hidden="true" href="#cb140-4" tabindex="-1"></a> .loc[<span class="kw">lambda</span> dd: dd.<span class="bu">sum</span>(axis<span class="op">=</span><span class="dv">1</span>) <span class="op">&gt;</span> <span class="dv">0</span>]</span>
<span id="cb140-5"><a aria-hidden="true" href="#cb140-5" tabindex="-1"></a> .loc[<span class="kw">lambda</span> dd: dd.<span class="bu">max</span>(axis<span class="op">=</span><span class="dv">1</span>) <span class="op">&gt;</span> <span class="fl">.0001</span>]</span>
<span id="cb140-6"><a aria-hidden="true" href="#cb140-6" tabindex="-1"></a> .reset_index()</span>
<span id="cb140-7"><a aria-hidden="true" href="#cb140-7" tabindex="-1"></a> .merge(pathogenic_fungi_df)</span>
<span id="cb140-8"><a aria-hidden="true" href="#cb140-8" tabindex="-1"></a> .assign(total<span class="op">=</span><span class="kw">lambda</span> dd: dd[<span class="st">'Flight samples'</span>] <span class="op">+</span> dd[<span class="st">'Ground samples'</span>])</span>
<span id="cb140-9"><a aria-hidden="true" href="#cb140-9" tabindex="-1"></a> .sort_values(<span class="st">'total'</span>)</span>
<span id="cb140-10"><a aria-hidden="true" href="#cb140-10" tabindex="-1"></a> .assign(name<span class="op">=</span><span class="kw">lambda</span> dd: dd.Genus.<span class="bu">str</span>[<span class="dv">0</span>] <span class="op">+</span> <span class="st">'. '</span> <span class="op">+</span> dd.Species)</span>
<span id="cb140-11"><a aria-hidden="true" href="#cb140-11" tabindex="-1"></a> [<span class="st">'name'</span>]</span>
<span id="cb140-12"><a aria-hidden="true" href="#cb140-12" tabindex="-1"></a></span>
<span id="cb140-13"><a aria-hidden="true" href="#cb140-13" tabindex="-1"></a>)</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" id="cell-173">
<div class="sourceCode cell-code" id="cb141"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb141-1"><a aria-hidden="true" href="#cb141-1" tabindex="-1"></a>f <span class="op">=</span> (pd.concat([sample_fun_path, blank_species_df])</span>
<span id="cb141-2"><a aria-hidden="true" href="#cb141-2" tabindex="-1"></a> .pivot(index<span class="op">=</span>[<span class="st">'Genus'</span>, <span class="st">'Species'</span>], columns<span class="op">=</span><span class="st">'sample_label'</span>, values<span class="op">=</span><span class="st">'relative_abundance'</span>)</span>
<span id="cb141-3"><a aria-hidden="true" href="#cb141-3" tabindex="-1"></a> .fillna(<span class="dv">0</span>)</span>
<span id="cb141-4"><a aria-hidden="true" href="#cb141-4" tabindex="-1"></a> .reset_index()</span>
<span id="cb141-5"><a aria-hidden="true" href="#cb141-5" tabindex="-1"></a> .melt([<span class="st">'Genus'</span>, <span class="st">'Species'</span>], var_name<span class="op">=</span><span class="st">'sample_label'</span>, value_name<span class="op">=</span><span class="st">'relative_abundance'</span>)</span>
<span id="cb141-6"><a aria-hidden="true" href="#cb141-6" tabindex="-1"></a> .merge(pathogenic_fungi_df)</span>
<span id="cb141-7"><a aria-hidden="true" href="#cb141-7" tabindex="-1"></a> .assign(Genus_short<span class="op">=</span><span class="kw">lambda</span> dd: dd.Genus.<span class="bu">str</span>[<span class="dv">0</span>] <span class="op">+</span> <span class="st">'. '</span>)</span>
<span id="cb141-8"><a aria-hidden="true" href="#cb141-8" tabindex="-1"></a> .assign(fullname<span class="op">=</span><span class="kw">lambda</span> dd: dd.Genus_short <span class="op">+</span> dd.Species)</span>
<span id="cb141-9"><a aria-hidden="true" href="#cb141-9" tabindex="-1"></a> .assign(sample_label<span class="op">=</span><span class="kw">lambda</span> dd: dd.sample_label.<span class="bu">str</span>.replace(<span class="st">' '</span>, <span class="st">'</span><span class="ch">\n</span><span class="st">'</span>))</span>
<span id="cb141-10"><a aria-hidden="true" href="#cb141-10" tabindex="-1"></a> .assign(sample_label<span class="op">=</span><span class="kw">lambda</span> dd: dd.sample_label.<span class="bu">str</span>.replace(<span class="st">'Samples'</span>, <span class="st">'samples'</span>))</span>
<span id="cb141-11"><a aria-hidden="true" href="#cb141-11" tabindex="-1"></a> .query(<span class="st">'fullname.isin(@fun_names)'</span>)</span>
<span id="cb141-12"><a aria-hidden="true" href="#cb141-12" tabindex="-1"></a> .assign(sample_label<span class="op">=</span><span class="kw">lambda</span> dd: pd.Categorical(dd.sample_label,</span>
<span id="cb141-13"><a aria-hidden="true" href="#cb141-13" tabindex="-1"></a>                      categories<span class="op">=</span>[<span class="st">'Flight</span><span class="ch">\n</span><span class="st">samples'</span>, <span class="st">'Ground</span><span class="ch">\n</span><span class="st">samples'</span>, <span class="st">'Flight</span><span class="ch">\n</span><span class="st">Blanks'</span>, <span class="st">'Ground</span><span class="ch">\n</span><span class="st">Blanks'</span>]))</span>
<span id="cb141-14"><a aria-hidden="true" href="#cb141-14" tabindex="-1"></a> .assign(fullname<span class="op">=</span><span class="kw">lambda</span> dd: pd.Categorical(dd.fullname, categories<span class="op">=</span>fun_names, ordered<span class="op">=</span><span class="va">True</span>))</span>
<span id="cb141-15"><a aria-hidden="true" href="#cb141-15" tabindex="-1"></a> .pipe(<span class="kw">lambda</span> dd: p9.ggplot(dd)</span>
<span id="cb141-16"><a aria-hidden="true" href="#cb141-16" tabindex="-1"></a>    <span class="op">+</span> p9.aes(<span class="st">'fullname'</span>, <span class="st">'sample_label'</span>)</span>
<span id="cb141-17"><a aria-hidden="true" href="#cb141-17" tabindex="-1"></a>    <span class="op">+</span> p9.geom_tile(p9.aes(fill<span class="op">=</span><span class="st">'relative_abundance &gt; 0'</span>), color<span class="op">=</span><span class="st">'black'</span>)</span>
<span id="cb141-18"><a aria-hidden="true" href="#cb141-18" tabindex="-1"></a>    <span class="op">+</span> p9.coord_flip()</span>
<span id="cb141-19"><a aria-hidden="true" href="#cb141-19" tabindex="-1"></a>    <span class="op">+</span> p9.geom_text(p9.aes(label<span class="op">=</span><span class="st">'round(relative_abundance * 100, 3).astype(str) + "%"'</span>), size<span class="op">=</span><span class="dv">8</span>)</span>
<span id="cb141-20"><a aria-hidden="true" href="#cb141-20" tabindex="-1"></a>    <span class="op">+</span> p9.labs(x<span class="op">=</span><span class="st">''</span>, y<span class="op">=</span><span class="st">''</span>)</span>
<span id="cb141-21"><a aria-hidden="true" href="#cb141-21" tabindex="-1"></a>    <span class="op">+</span> p9.facet_wrap(<span class="st">'label'</span>, scales<span class="op">=</span><span class="st">'free_y'</span>, ncol<span class="op">=</span><span class="dv">1</span>)</span>
<span id="cb141-22"><a aria-hidden="true" href="#cb141-22" tabindex="-1"></a>    <span class="op">+</span> p9.scale_x_discrete(expand<span class="op">=</span>(<span class="dv">0</span>, <span class="dv">0</span>), trans<span class="op">=</span><span class="st">'reverse'</span>)</span>
<span id="cb141-23"><a aria-hidden="true" href="#cb141-23" tabindex="-1"></a>    <span class="op">+</span> p9.scale_y_discrete(expand<span class="op">=</span>(<span class="dv">0</span>, <span class="dv">0</span>))</span>
<span id="cb141-24"><a aria-hidden="true" href="#cb141-24" tabindex="-1"></a>    <span class="op">+</span> p9.guides(fill<span class="op">=</span><span class="va">False</span>)</span>
<span id="cb141-25"><a aria-hidden="true" href="#cb141-25" tabindex="-1"></a>    <span class="op">+</span> p9.theme(figure_size<span class="op">=</span>(<span class="fl">3.5</span>, <span class="fl">4.5</span>),</span>
<span id="cb141-26"><a aria-hidden="true" href="#cb141-26" tabindex="-1"></a>               axis_text_x<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">8</span>),</span>
<span id="cb141-27"><a aria-hidden="true" href="#cb141-27" tabindex="-1"></a>               strip_text<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">8</span>),</span>
<span id="cb141-28"><a aria-hidden="true" href="#cb141-28" tabindex="-1"></a>       )</span>
<span id="cb141-29"><a aria-hidden="true" href="#cb141-29" tabindex="-1"></a>)</span>
<span id="cb141-30"><a aria-hidden="true" href="#cb141-30" tabindex="-1"></a>)</span>
<span id="cb141-31"><a aria-hidden="true" href="#cb141-31" tabindex="-1"></a>f.save(<span class="st">'../output/figures/fun_pathogens_table.svg'</span>)</span>
<span id="cb141-32"><a aria-hidden="true" href="#cb141-32" tabindex="-1"></a>f.draw()</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="1558">
<div>
<figure class="figure">
<p><img class="figure-img" height="1350" src="microbial_diversity_analysis_files/figure-html/cell-136-output-1.png" width="1050"/></p>
</figure>
</div>
</div>
</div>
<div class="cell" data-execution_count="137" id="cell-174">
<div class="sourceCode cell-code" id="cb142"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb142-1"><a aria-hidden="true" href="#cb142-1" tabindex="-1"></a>(bac_df_sp</span>
<span id="cb142-2"><a aria-hidden="true" href="#cb142-2" tabindex="-1"></a> .drop(columns<span class="op">=</span>taxon_names[:<span class="op">-</span><span class="dv">2</span>])</span>
<span id="cb142-3"><a aria-hidden="true" href="#cb142-3" tabindex="-1"></a> .merge(bac_pathogens_df, on<span class="op">=</span>[<span class="st">'Genus'</span>, <span class="st">'Species'</span>], how<span class="op">=</span><span class="st">'right'</span>)</span>
<span id="cb142-4"><a aria-hidden="true" href="#cb142-4" tabindex="-1"></a> .melt([<span class="st">'Genus'</span>, <span class="st">'Species'</span>, <span class="st">'label'</span>], var_name<span class="op">=</span><span class="st">'sample_id'</span>, value_name<span class="op">=</span><span class="st">'counts'</span>)</span>
<span id="cb142-5"><a aria-hidden="true" href="#cb142-5" tabindex="-1"></a> .assign(sample_mode<span class="op">=</span><span class="kw">lambda</span> dd: dd.sample_id.<span class="bu">map</span>(sample_map))</span>
<span id="cb142-6"><a aria-hidden="true" href="#cb142-6" tabindex="-1"></a> .assign(name<span class="op">=</span><span class="kw">lambda</span> dd: dd.Genus.<span class="bu">str</span>[<span class="dv">0</span>] <span class="op">+</span> <span class="st">'. '</span> <span class="op">+</span> dd.Species)</span>
<span id="cb142-7"><a aria-hidden="true" href="#cb142-7" tabindex="-1"></a> .assign(name<span class="op">=</span><span class="kw">lambda</span> dd: pd.Categorical(dd.name, categories<span class="op">=</span>bac_names, ordered<span class="op">=</span><span class="va">True</span>))</span>
<span id="cb142-8"><a aria-hidden="true" href="#cb142-8" tabindex="-1"></a> .dropna()</span>
<span id="cb142-9"><a aria-hidden="true" href="#cb142-9" tabindex="-1"></a> .query(<span class="st">'counts &gt; 0'</span>)</span>
<span id="cb142-10"><a aria-hidden="true" href="#cb142-10" tabindex="-1"></a> .name.nunique()</span>
<span id="cb142-11"><a aria-hidden="true" href="#cb142-11" tabindex="-1"></a>)</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="137">
<pre><code>46</code></pre>
</div>
</div>
<div class="cell" data-execution_count="150" id="cell-175">
<div class="sourceCode cell-code" id="cb144"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb144-1"><a aria-hidden="true" href="#cb144-1" tabindex="-1"></a>f <span class="op">=</span> (bac_df_sp</span>
<span id="cb144-2"><a aria-hidden="true" href="#cb144-2" tabindex="-1"></a> .drop(columns<span class="op">=</span>taxon_names[:<span class="op">-</span><span class="dv">2</span>])</span>
<span id="cb144-3"><a aria-hidden="true" href="#cb144-3" tabindex="-1"></a> .merge(bac_pathogens_df, on<span class="op">=</span>[<span class="st">'Genus'</span>, <span class="st">'Species'</span>], how<span class="op">=</span><span class="st">'right'</span>)</span>
<span id="cb144-4"><a aria-hidden="true" href="#cb144-4" tabindex="-1"></a> .melt([<span class="st">'Genus'</span>, <span class="st">'Species'</span>, <span class="st">'label'</span>], var_name<span class="op">=</span><span class="st">'sample_id'</span>, value_name<span class="op">=</span><span class="st">'counts'</span>)</span>
<span id="cb144-5"><a aria-hidden="true" href="#cb144-5" tabindex="-1"></a> .assign(sample_mode<span class="op">=</span><span class="kw">lambda</span> dd: dd.sample_id.<span class="bu">map</span>(sample_map))</span>
<span id="cb144-6"><a aria-hidden="true" href="#cb144-6" tabindex="-1"></a> .assign(name<span class="op">=</span><span class="kw">lambda</span> dd: dd.Genus.<span class="bu">str</span>[<span class="dv">0</span>] <span class="op">+</span> <span class="st">'. '</span> <span class="op">+</span> dd.Species)</span>
<span id="cb144-7"><a aria-hidden="true" href="#cb144-7" tabindex="-1"></a> .assign(name<span class="op">=</span><span class="kw">lambda</span> dd: pd.Categorical(dd.name, categories<span class="op">=</span>bac_names, ordered<span class="op">=</span><span class="va">True</span>))</span>
<span id="cb144-8"><a aria-hidden="true" href="#cb144-8" tabindex="-1"></a> .dropna()</span>
<span id="cb144-9"><a aria-hidden="true" href="#cb144-9" tabindex="-1"></a> .query(<span class="st">'counts &gt; 0'</span>)</span>
<span id="cb144-10"><a aria-hidden="true" href="#cb144-10" tabindex="-1"></a> .pipe(<span class="kw">lambda</span> dd: p9.ggplot(dd)</span>
<span id="cb144-11"><a aria-hidden="true" href="#cb144-11" tabindex="-1"></a>      <span class="op">+</span> p9.aes(<span class="st">'sample_id'</span>,</span>
<span id="cb144-12"><a aria-hidden="true" href="#cb144-12" tabindex="-1"></a>               <span class="st">'name'</span>,</span>
<span id="cb144-13"><a aria-hidden="true" href="#cb144-13" tabindex="-1"></a>               size<span class="op">=</span><span class="st">'counts'</span>,</span>
<span id="cb144-14"><a aria-hidden="true" href="#cb144-14" tabindex="-1"></a>               color<span class="op">=</span><span class="st">'sample_mode'</span>,</span>
<span id="cb144-15"><a aria-hidden="true" href="#cb144-15" tabindex="-1"></a>               fill<span class="op">=</span><span class="st">'sample_mode'</span>)</span>
<span id="cb144-16"><a aria-hidden="true" href="#cb144-16" tabindex="-1"></a>       <span class="op">+</span> p9.geom_point(alpha<span class="op">=</span><span class="fl">.8</span>)</span>
<span id="cb144-17"><a aria-hidden="true" href="#cb144-17" tabindex="-1"></a>       <span class="op">+</span> p9.scale_y_discrete(expand<span class="op">=</span>(<span class="fl">.015</span>, <span class="fl">.015</span>))</span>
<span id="cb144-18"><a aria-hidden="true" href="#cb144-18" tabindex="-1"></a>       <span class="op">+</span> p9.scale_fill_manual(values<span class="op">=</span>[<span class="st">'#D3894C'</span>, <span class="st">'#435B97'</span>])</span>
<span id="cb144-19"><a aria-hidden="true" href="#cb144-19" tabindex="-1"></a>       <span class="op">+</span> p9.scale_color_manual(values<span class="op">=</span>[<span class="st">'#D3894C'</span>, <span class="st">'#435B97'</span>])</span>
<span id="cb144-20"><a aria-hidden="true" href="#cb144-20" tabindex="-1"></a>       <span class="op">+</span> p9.guides(size<span class="op">=</span>p9.guide_legend(ncol<span class="op">=</span><span class="dv">1</span>))</span>
<span id="cb144-21"><a aria-hidden="true" href="#cb144-21" tabindex="-1"></a>       <span class="op">+</span> p9.labs(x<span class="op">=</span><span class="st">''</span>, y<span class="op">=</span><span class="st">''</span>, size<span class="op">=</span><span class="st">'Total Reads'</span>, color<span class="op">=</span><span class="st">''</span>, fill<span class="op">=</span><span class="st">''</span>,</span>
<span id="cb144-22"><a aria-hidden="true" href="#cb144-22" tabindex="-1"></a>                  )</span>
<span id="cb144-23"><a aria-hidden="true" href="#cb144-23" tabindex="-1"></a>      <span class="co">#  + p9.facet_wrap('label', ncol=1, scales='free_y')</span></span>
<span id="cb144-24"><a aria-hidden="true" href="#cb144-24" tabindex="-1"></a>       <span class="op">+</span> p9.theme(figure_size<span class="op">=</span>(<span class="fl">3.5</span>, <span class="fl">9.5</span>),</span>
<span id="cb144-25"><a aria-hidden="true" href="#cb144-25" tabindex="-1"></a>                  strip_text<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">12</span>),</span>
<span id="cb144-26"><a aria-hidden="true" href="#cb144-26" tabindex="-1"></a>                  axis_text_y<span class="op">=</span>p9.element_blank(),</span>
<span id="cb144-27"><a aria-hidden="true" href="#cb144-27" tabindex="-1"></a>                  axis_ticks_y<span class="op">=</span>p9.element_blank(),</span>
<span id="cb144-28"><a aria-hidden="true" href="#cb144-28" tabindex="-1"></a>                  axis_text_x<span class="op">=</span>p9.element_text(angle<span class="op">=</span><span class="dv">90</span>),</span>
<span id="cb144-29"><a aria-hidden="true" href="#cb144-29" tabindex="-1"></a>                  legend_position<span class="op">=</span><span class="st">'bottom'</span>,</span>
<span id="cb144-30"><a aria-hidden="true" href="#cb144-30" tabindex="-1"></a>                  legend_direction<span class="op">=</span><span class="st">'horizontal'</span>,</span>
<span id="cb144-31"><a aria-hidden="true" href="#cb144-31" tabindex="-1"></a>       </span>
<span id="cb144-32"><a aria-hidden="true" href="#cb144-32" tabindex="-1"></a>       )</span>
<span id="cb144-33"><a aria-hidden="true" href="#cb144-33" tabindex="-1"></a> )</span>
<span id="cb144-34"><a aria-hidden="true" href="#cb144-34" tabindex="-1"></a>)</span>
<span id="cb144-35"><a aria-hidden="true" href="#cb144-35" tabindex="-1"></a>f.save(<span class="st">'../output/figures/bac_pathogens_samples.svg'</span>)</span>
<span id="cb144-36"><a aria-hidden="true" href="#cb144-36" tabindex="-1"></a>f.draw()</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="150">
<div>
<figure class="figure">
<p><img class="figure-img" height="2888" src="microbial_diversity_analysis_files/figure-html/cell-138-output-1.png" width="1112"/></p>
</figure>
</div>
</div>
</div>
<div class="cell" id="cell-176">
<div class="sourceCode cell-code" id="cb145"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb145-1"><a aria-hidden="true" href="#cb145-1" tabindex="-1"></a>f <span class="op">=</span> (fun_df_sp</span>
<span id="cb145-2"><a aria-hidden="true" href="#cb145-2" tabindex="-1"></a> .drop(columns<span class="op">=</span>taxon_names[:<span class="op">-</span><span class="dv">2</span>])</span>
<span id="cb145-3"><a aria-hidden="true" href="#cb145-3" tabindex="-1"></a> .merge(pathogenic_fungi_df, on<span class="op">=</span>[<span class="st">'Genus'</span>, <span class="st">'Species'</span>], how<span class="op">=</span><span class="st">'right'</span>)</span>
<span id="cb145-4"><a aria-hidden="true" href="#cb145-4" tabindex="-1"></a> .melt([<span class="st">'Genus'</span>, <span class="st">'Species'</span>, <span class="st">'label'</span>], var_name<span class="op">=</span><span class="st">'sample_id'</span>, value_name<span class="op">=</span><span class="st">'counts'</span>)</span>
<span id="cb145-5"><a aria-hidden="true" href="#cb145-5" tabindex="-1"></a> .assign(sample_mode<span class="op">=</span><span class="kw">lambda</span> dd: dd.sample_id.<span class="bu">map</span>(sample_map))</span>
<span id="cb145-6"><a aria-hidden="true" href="#cb145-6" tabindex="-1"></a> .assign(name<span class="op">=</span><span class="kw">lambda</span> dd: dd.Genus.<span class="bu">str</span>[<span class="dv">0</span>] <span class="op">+</span> <span class="st">'. '</span> <span class="op">+</span> dd.Species)</span>
<span id="cb145-7"><a aria-hidden="true" href="#cb145-7" tabindex="-1"></a> .assign(name<span class="op">=</span><span class="kw">lambda</span> dd: pd.Categorical(dd.name, categories<span class="op">=</span>fun_names, ordered<span class="op">=</span><span class="va">True</span>))</span>
<span id="cb145-8"><a aria-hidden="true" href="#cb145-8" tabindex="-1"></a>  .dropna()</span>
<span id="cb145-9"><a aria-hidden="true" href="#cb145-9" tabindex="-1"></a> .query(<span class="st">'counts &gt; 0'</span>)</span>
<span id="cb145-10"><a aria-hidden="true" href="#cb145-10" tabindex="-1"></a> .pipe(<span class="kw">lambda</span> dd: p9.ggplot(dd)</span>
<span id="cb145-11"><a aria-hidden="true" href="#cb145-11" tabindex="-1"></a>      <span class="op">+</span> p9.aes(<span class="st">'sample_id'</span>,</span>
<span id="cb145-12"><a aria-hidden="true" href="#cb145-12" tabindex="-1"></a>               <span class="st">'name'</span>,</span>
<span id="cb145-13"><a aria-hidden="true" href="#cb145-13" tabindex="-1"></a>               size<span class="op">=</span><span class="st">'counts'</span>,</span>
<span id="cb145-14"><a aria-hidden="true" href="#cb145-14" tabindex="-1"></a>               color<span class="op">=</span><span class="st">'sample_mode'</span>,</span>
<span id="cb145-15"><a aria-hidden="true" href="#cb145-15" tabindex="-1"></a>               fill<span class="op">=</span><span class="st">'sample_mode'</span>)</span>
<span id="cb145-16"><a aria-hidden="true" href="#cb145-16" tabindex="-1"></a>       <span class="op">+</span> p9.geom_point(alpha<span class="op">=</span><span class="fl">.8</span>)</span>
<span id="cb145-17"><a aria-hidden="true" href="#cb145-17" tabindex="-1"></a>       <span class="op">+</span> p9.scale_y_discrete(expand<span class="op">=</span>(<span class="fl">.035</span>, <span class="fl">.035</span>))</span>
<span id="cb145-18"><a aria-hidden="true" href="#cb145-18" tabindex="-1"></a>       <span class="op">+</span> p9.scale_fill_manual(values<span class="op">=</span>[<span class="st">'#D3894C'</span>, <span class="st">'#435B97'</span>])</span>
<span id="cb145-19"><a aria-hidden="true" href="#cb145-19" tabindex="-1"></a>       <span class="op">+</span> p9.scale_color_manual(values<span class="op">=</span>[<span class="st">'#D3894C'</span>, <span class="st">'#435B97'</span>])</span>
<span id="cb145-20"><a aria-hidden="true" href="#cb145-20" tabindex="-1"></a>       <span class="op">+</span> p9.guides(size<span class="op">=</span>p9.guide_legend(ncol<span class="op">=</span><span class="dv">1</span>))</span>
<span id="cb145-21"><a aria-hidden="true" href="#cb145-21" tabindex="-1"></a>       <span class="op">+</span> p9.labs(x<span class="op">=</span><span class="st">''</span>, y<span class="op">=</span><span class="st">''</span>, size<span class="op">=</span><span class="st">'Total Reads'</span>, color<span class="op">=</span><span class="st">''</span>, fill<span class="op">=</span><span class="st">''</span>,</span>
<span id="cb145-22"><a aria-hidden="true" href="#cb145-22" tabindex="-1"></a>                  )</span>
<span id="cb145-23"><a aria-hidden="true" href="#cb145-23" tabindex="-1"></a>       <span class="op">+</span> p9.facet_wrap(<span class="st">'label'</span>, ncol<span class="op">=</span><span class="dv">1</span>, scales<span class="op">=</span><span class="st">'free_y'</span>)</span>
<span id="cb145-24"><a aria-hidden="true" href="#cb145-24" tabindex="-1"></a>       <span class="op">+</span> p9.theme(figure_size<span class="op">=</span>(<span class="dv">6</span>, <span class="fl">4.5</span>),</span>
<span id="cb145-25"><a aria-hidden="true" href="#cb145-25" tabindex="-1"></a>                  strip_text<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">12</span>),</span>
<span id="cb145-26"><a aria-hidden="true" href="#cb145-26" tabindex="-1"></a>                  axis_text<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">10</span>),</span>
<span id="cb145-27"><a aria-hidden="true" href="#cb145-27" tabindex="-1"></a>                  axis_text_x<span class="op">=</span>p9.element_text(angle<span class="op">=</span><span class="dv">90</span>),</span>
<span id="cb145-28"><a aria-hidden="true" href="#cb145-28" tabindex="-1"></a>       </span>
<span id="cb145-29"><a aria-hidden="true" href="#cb145-29" tabindex="-1"></a>       )</span>
<span id="cb145-30"><a aria-hidden="true" href="#cb145-30" tabindex="-1"></a> )</span>
<span id="cb145-31"><a aria-hidden="true" href="#cb145-31" tabindex="-1"></a>)</span>
<span id="cb145-32"><a aria-hidden="true" href="#cb145-32" tabindex="-1"></a></span>
<span id="cb145-33"><a aria-hidden="true" href="#cb145-33" tabindex="-1"></a>f.save(<span class="st">'../output/figures/fun_pathogens_samples.svg'</span>)</span>
<span id="cb145-34"><a aria-hidden="true" href="#cb145-34" tabindex="-1"></a>f.draw()</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="1569">
<div>
<figure class="figure">
<p><img class="figure-img" height="1350" src="microbial_diversity_analysis_files/figure-html/cell-139-output-1.png" width="1800"/></p>
</figure>
</div>
</div>
</div>
<div class="cell" id="cell-177">
<div class="sourceCode cell-code" id="cb146"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb146-1"><a aria-hidden="true" href="#cb146-1" tabindex="-1"></a>(pd.concat([blank_bac_pathogens, samples_pathogens_bac_df])</span>
<span id="cb146-2"><a aria-hidden="true" href="#cb146-2" tabindex="-1"></a> .query(<span class="st">'sample_label.notna()'</span>)</span>
<span id="cb146-3"><a aria-hidden="true" href="#cb146-3" tabindex="-1"></a> .pivot(index<span class="op">=</span>[<span class="st">'Genus'</span>, <span class="st">'Species'</span>], columns<span class="op">=</span><span class="st">'sample_label'</span>, values<span class="op">=</span><span class="st">'relative_abundance'</span>)</span>
<span id="cb146-4"><a aria-hidden="true" href="#cb146-4" tabindex="-1"></a> .fillna(<span class="dv">0</span>)</span>
<span id="cb146-5"><a aria-hidden="true" href="#cb146-5" tabindex="-1"></a> .loc[<span class="kw">lambda</span> dd: dd.<span class="bu">sum</span>(axis<span class="op">=</span><span class="dv">1</span>) <span class="op">&gt;</span> <span class="dv">0</span>]</span>
<span id="cb146-6"><a aria-hidden="true" href="#cb146-6" tabindex="-1"></a> .shape</span>
<span id="cb146-7"><a aria-hidden="true" href="#cb146-7" tabindex="-1"></a>)</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="1063">
<pre><code>(41, 4)</code></pre>
</div>
</div>
<div class="cell" id="cell-178">
<div class="sourceCode cell-code" id="cb148"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb148-1"><a aria-hidden="true" href="#cb148-1" tabindex="-1"></a>sample_fun_path</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="1401">
<div>
<table class="dataframe caption-top table table-sm table-striped small" data-border="1" data-quarto-postprocess="true">
<thead>
<tr class="header">
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th">level_0</th>
<th data-quarto-table-cell-role="th">Genus</th>
<th data-quarto-table-cell-role="th">Species</th>
<th data-quarto-table-cell-role="th">sample_label</th>
<th data-quarto-table-cell-role="th">counts</th>
<th data-quarto-table-cell-role="th">relative_abundance</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td data-quarto-table-cell-role="th">0</td>
<td>0</td>
<td>Aureobasidium</td>
<td>pullulans</td>
<td>Flight samples</td>
<td>3234</td>
<td>0.004145</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">1</td>
<td>0</td>
<td>Candida</td>
<td>parapsilosis</td>
<td>Flight samples</td>
<td>37</td>
<td>0.000047</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">2</td>
<td>0</td>
<td>Candida</td>
<td>zeylanoides</td>
<td>Flight samples</td>
<td>0</td>
<td>0.000000</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">3</td>
<td>0</td>
<td>Cladosporium</td>
<td>halotolerans</td>
<td>Flight samples</td>
<td>0</td>
<td>0.000000</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">4</td>
<td>0</td>
<td>Cladosporium</td>
<td>herbarum</td>
<td>Flight samples</td>
<td>18053</td>
<td>0.023141</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">5</td>
<td>0</td>
<td>Cutaneotrichosporon</td>
<td>debeurmannianum</td>
<td>Flight samples</td>
<td>0</td>
<td>0.000000</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">6</td>
<td>0</td>
<td>Cystobasidium</td>
<td>slooffiae</td>
<td>Flight samples</td>
<td>712</td>
<td>0.000913</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">7</td>
<td>0</td>
<td>Exophiala</td>
<td>oligosperma</td>
<td>Flight samples</td>
<td>724033</td>
<td>0.928078</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">8</td>
<td>0</td>
<td>Malassezia</td>
<td>globosa</td>
<td>Flight samples</td>
<td>7634</td>
<td>0.009785</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">9</td>
<td>0</td>
<td>Malassezia</td>
<td>restricta</td>
<td>Flight samples</td>
<td>20547</td>
<td>0.026338</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">10</td>
<td>0</td>
<td>Malassezia</td>
<td>sympodialis</td>
<td>Flight samples</td>
<td>433</td>
<td>0.000555</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">11</td>
<td>0</td>
<td>Naganishia</td>
<td>diffluens</td>
<td>Flight samples</td>
<td>276</td>
<td>0.000354</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">12</td>
<td>0</td>
<td>Sarocladium</td>
<td>kiliense</td>
<td>Flight samples</td>
<td>5062</td>
<td>0.006489</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">13</td>
<td>0</td>
<td>Sterigmatomyces</td>
<td>halophilus</td>
<td>Flight samples</td>
<td>56</td>
<td>0.000072</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">14</td>
<td>0</td>
<td>Trichosporon</td>
<td>asahii</td>
<td>Flight samples</td>
<td>65</td>
<td>0.000083</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">15</td>
<td>1</td>
<td>Aureobasidium</td>
<td>pullulans</td>
<td>Ground samples</td>
<td>3611</td>
<td>0.005121</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">16</td>
<td>1</td>
<td>Candida</td>
<td>parapsilosis</td>
<td>Ground samples</td>
<td>0</td>
<td>0.000000</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">17</td>
<td>1</td>
<td>Candida</td>
<td>zeylanoides</td>
<td>Ground samples</td>
<td>162</td>
<td>0.000230</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">18</td>
<td>1</td>
<td>Cladosporium</td>
<td>halotolerans</td>
<td>Ground samples</td>
<td>1498</td>
<td>0.002124</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">19</td>
<td>1</td>
<td>Cladosporium</td>
<td>herbarum</td>
<td>Ground samples</td>
<td>21110</td>
<td>0.029936</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">20</td>
<td>1</td>
<td>Cutaneotrichosporon</td>
<td>debeurmannianum</td>
<td>Ground samples</td>
<td>31</td>
<td>0.000044</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">21</td>
<td>1</td>
<td>Cystobasidium</td>
<td>slooffiae</td>
<td>Ground samples</td>
<td>0</td>
<td>0.000000</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">22</td>
<td>1</td>
<td>Exophiala</td>
<td>oligosperma</td>
<td>Ground samples</td>
<td>620463</td>
<td>0.879886</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">23</td>
<td>1</td>
<td>Malassezia</td>
<td>globosa</td>
<td>Ground samples</td>
<td>14631</td>
<td>0.020748</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">24</td>
<td>1</td>
<td>Malassezia</td>
<td>restricta</td>
<td>Ground samples</td>
<td>25462</td>
<td>0.036108</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">25</td>
<td>1</td>
<td>Malassezia</td>
<td>sympodialis</td>
<td>Ground samples</td>
<td>28</td>
<td>0.000040</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">26</td>
<td>1</td>
<td>Naganishia</td>
<td>diffluens</td>
<td>Ground samples</td>
<td>1534</td>
<td>0.002175</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">27</td>
<td>1</td>
<td>Sarocladium</td>
<td>kiliense</td>
<td>Ground samples</td>
<td>7180</td>
<td>0.010182</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">28</td>
<td>1</td>
<td>Sterigmatomyces</td>
<td>halophilus</td>
<td>Ground samples</td>
<td>9453</td>
<td>0.013405</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">29</td>
<td>1</td>
<td>Trichosporon</td>
<td>asahii</td>
<td>Ground samples</td>
<td>0</td>
<td>0.000000</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="cell" id="cell-179">
<div class="sourceCode cell-code" id="cb149"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb149-1"><a aria-hidden="true" href="#cb149-1" tabindex="-1"></a>sample_fun_path.groupby([<span class="st">'sample_label'</span>])[<span class="st">'relative_abundance'</span>].<span class="bu">sum</span>() <span class="op">*</span> <span class="dv">100</span></span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="1412">
<pre><code>sample_label
Flight samples    64.838768
Ground samples    70.998165
Name: relative_abundance, dtype: float64</code></pre>
</div>
</div>
<div class="cell" id="cell-180">
<div class="sourceCode cell-code" id="cb151"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb151-1"><a aria-hidden="true" href="#cb151-1" tabindex="-1"></a>samples_pathogens_bac_df.groupby([<span class="st">'sample_label'</span>])[<span class="st">'relative_abundance'</span>].<span class="bu">sum</span>() <span class="op">*</span> <span class="dv">100</span></span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="1411">
<pre><code>sample_label
Flight samples    0.683429
Ground samples    0.634687
Name: relative_abundance, dtype: float64</code></pre>
</div>
</div>
<div class="cell" id="cell-181">
<div class="sourceCode cell-code" id="cb153"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb153-1"><a aria-hidden="true" href="#cb153-1" tabindex="-1"></a>f <span class="op">=</span> (pd.concat([blank_bac_pathogens, samples_pathogens_bac_df])</span>
<span id="cb153-2"><a aria-hidden="true" href="#cb153-2" tabindex="-1"></a> .query(<span class="st">'sample_label.notna()'</span>)</span>
<span id="cb153-3"><a aria-hidden="true" href="#cb153-3" tabindex="-1"></a> .pivot(index<span class="op">=</span>[<span class="st">'Genus'</span>, <span class="st">'Species'</span>], columns<span class="op">=</span><span class="st">'sample_label'</span>, values<span class="op">=</span><span class="st">'relative_abundance'</span>)</span>
<span id="cb153-4"><a aria-hidden="true" href="#cb153-4" tabindex="-1"></a> .fillna(<span class="dv">0</span>)</span>
<span id="cb153-5"><a aria-hidden="true" href="#cb153-5" tabindex="-1"></a> .loc[<span class="kw">lambda</span> dd: dd.<span class="bu">sum</span>(axis<span class="op">=</span><span class="dv">1</span>) <span class="op">&gt;</span> <span class="dv">0</span>]</span>
<span id="cb153-6"><a aria-hidden="true" href="#cb153-6" tabindex="-1"></a> .reset_index()</span>
<span id="cb153-7"><a aria-hidden="true" href="#cb153-7" tabindex="-1"></a> .melt([<span class="st">'Genus'</span>, <span class="st">'Species'</span>], value_name<span class="op">=</span><span class="st">'relative_abundance'</span>)</span>
<span id="cb153-8"><a aria-hidden="true" href="#cb153-8" tabindex="-1"></a> .assign(Genus_short<span class="op">=</span><span class="kw">lambda</span> dd: dd.Genus.<span class="bu">str</span>[<span class="dv">0</span>] <span class="op">+</span> <span class="st">'. '</span>)</span>
<span id="cb153-9"><a aria-hidden="true" href="#cb153-9" tabindex="-1"></a> .assign(fullname<span class="op">=</span><span class="kw">lambda</span> dd: dd.Genus_short <span class="op">+</span> dd.Species)</span>
<span id="cb153-10"><a aria-hidden="true" href="#cb153-10" tabindex="-1"></a> .assign(sample_label<span class="op">=</span><span class="kw">lambda</span> dd: dd.sample_label.<span class="bu">str</span>.replace(<span class="st">' '</span>, <span class="st">'</span><span class="ch">\n</span><span class="st">'</span>))</span>
<span id="cb153-11"><a aria-hidden="true" href="#cb153-11" tabindex="-1"></a> .assign(sample_label<span class="op">=</span><span class="kw">lambda</span> dd: </span>
<span id="cb153-12"><a aria-hidden="true" href="#cb153-12" tabindex="-1"></a>         pd.Categorical(dd.sample_label,</span>
<span id="cb153-13"><a aria-hidden="true" href="#cb153-13" tabindex="-1"></a>    categories<span class="op">=</span>[<span class="st">'Flight</span><span class="ch">\n</span><span class="st">samples'</span>, <span class="st">'Ground</span><span class="ch">\n</span><span class="st">samples'</span>,</span>
<span id="cb153-14"><a aria-hidden="true" href="#cb153-14" tabindex="-1"></a>                <span class="st">'Flight</span><span class="ch">\n</span><span class="st">Blanks'</span>, <span class="st">'Ground</span><span class="ch">\n</span><span class="st">Blanks'</span>])</span>
<span id="cb153-15"><a aria-hidden="true" href="#cb153-15" tabindex="-1"></a>    )</span>
<span id="cb153-16"><a aria-hidden="true" href="#cb153-16" tabindex="-1"></a> .merge(bac_pathogens_df)</span>
<span id="cb153-17"><a aria-hidden="true" href="#cb153-17" tabindex="-1"></a> .query(<span class="st">'label.str.contains("Opportunistic")'</span>)</span>
<span id="cb153-18"><a aria-hidden="true" href="#cb153-18" tabindex="-1"></a> .pipe(<span class="kw">lambda</span> dd: p9.ggplot(dd)</span>
<span id="cb153-19"><a aria-hidden="true" href="#cb153-19" tabindex="-1"></a>    <span class="op">+</span> p9.aes(<span class="st">'fullname'</span>, <span class="st">'sample_label'</span>)</span>
<span id="cb153-20"><a aria-hidden="true" href="#cb153-20" tabindex="-1"></a>    <span class="op">+</span> p9.geom_tile(p9.aes(fill<span class="op">=</span><span class="st">'relative_abundance &gt; 0'</span>), color<span class="op">=</span><span class="st">'black'</span>)</span>
<span id="cb153-21"><a aria-hidden="true" href="#cb153-21" tabindex="-1"></a>    <span class="op">+</span> p9.coord_flip()</span>
<span id="cb153-22"><a aria-hidden="true" href="#cb153-22" tabindex="-1"></a>    <span class="op">+</span> p9.geom_text(p9.aes(label<span class="op">=</span><span class="st">'round(relative_abundance * 100, 3).astype(str) + "%"'</span>), size<span class="op">=</span><span class="dv">8</span>)</span>
<span id="cb153-23"><a aria-hidden="true" href="#cb153-23" tabindex="-1"></a>    <span class="op">+</span> p9.labs(x<span class="op">=</span><span class="st">''</span>, y<span class="op">=</span><span class="st">''</span>)</span>
<span id="cb153-24"><a aria-hidden="true" href="#cb153-24" tabindex="-1"></a>    <span class="op">+</span> p9.facet_wrap(<span class="st">'label'</span>, scales<span class="op">=</span><span class="st">'free_y'</span>, ncol<span class="op">=</span><span class="dv">1</span>)</span>
<span id="cb153-25"><a aria-hidden="true" href="#cb153-25" tabindex="-1"></a>    <span class="op">+</span> p9.scale_x_discrete(expand<span class="op">=</span>(<span class="dv">0</span>, <span class="dv">0</span>), trans<span class="op">=</span><span class="st">'reverse'</span>)</span>
<span id="cb153-26"><a aria-hidden="true" href="#cb153-26" tabindex="-1"></a>    <span class="op">+</span> p9.scale_y_discrete(expand<span class="op">=</span>(<span class="dv">0</span>, <span class="dv">0</span>))</span>
<span id="cb153-27"><a aria-hidden="true" href="#cb153-27" tabindex="-1"></a>    <span class="op">+</span> p9.theme(figure_size<span class="op">=</span>(<span class="fl">3.5</span>, <span class="dv">5</span>), strip_text<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="fl">7.5</span>))</span>
<span id="cb153-28"><a aria-hidden="true" href="#cb153-28" tabindex="-1"></a>    <span class="op">+</span> p9.guides(fill<span class="op">=</span><span class="va">False</span>)</span>
<span id="cb153-29"><a aria-hidden="true" href="#cb153-29" tabindex="-1"></a>       )</span>
<span id="cb153-30"><a aria-hidden="true" href="#cb153-30" tabindex="-1"></a>)</span>
<span id="cb153-31"><a aria-hidden="true" href="#cb153-31" tabindex="-1"></a>f.save(<span class="st">'../output/figures/bac_opp_pathogens_table.svg'</span>)</span>
<span id="cb153-32"><a aria-hidden="true" href="#cb153-32" tabindex="-1"></a>f.draw()</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="1068">
<div>
<figure class="figure">
<p><img class="figure-img" height="1500" src="microbial_diversity_analysis_files/figure-html/cell-144-output-1.png" width="1050"/></p>
</figure>
</div>
</div>
</div>
<div class="cell" id="cell-182">
<div class="sourceCode cell-code" id="cb154"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb154-1"><a aria-hidden="true" href="#cb154-1" tabindex="-1"></a>fun_names <span class="op">=</span> (fun_df_sp</span>
<span id="cb154-2"><a aria-hidden="true" href="#cb154-2" tabindex="-1"></a> .drop(columns<span class="op">=</span>taxon_names[:<span class="op">-</span><span class="dv">2</span>])</span>
<span id="cb154-3"><a aria-hidden="true" href="#cb154-3" tabindex="-1"></a> .merge(fungi_pathogens_df, on<span class="op">=</span>[<span class="st">'Genus'</span>, <span class="st">'Species'</span>], how<span class="op">=</span><span class="st">'right'</span>)</span>
<span id="cb154-4"><a aria-hidden="true" href="#cb154-4" tabindex="-1"></a> .melt([<span class="st">'Genus'</span>, <span class="st">'Species'</span>], var_name<span class="op">=</span><span class="st">'sample_id'</span>, value_name<span class="op">=</span><span class="st">'counts'</span>)</span>
<span id="cb154-5"><a aria-hidden="true" href="#cb154-5" tabindex="-1"></a> .assign(sample_mode<span class="op">=</span><span class="kw">lambda</span> dd: dd.sample_id.<span class="bu">map</span>(sample_map))</span>
<span id="cb154-6"><a aria-hidden="true" href="#cb154-6" tabindex="-1"></a> .assign(name<span class="op">=</span><span class="kw">lambda</span> dd: dd.Genus.<span class="bu">str</span>[<span class="dv">0</span>] <span class="op">+</span> <span class="st">'. '</span> <span class="op">+</span> dd.Species)</span>
<span id="cb154-7"><a aria-hidden="true" href="#cb154-7" tabindex="-1"></a> .assign(label<span class="op">=</span><span class="st">'Fungal Human Pathogens'</span>)</span>
<span id="cb154-8"><a aria-hidden="true" href="#cb154-8" tabindex="-1"></a> .query(<span class="st">'counts &gt; 0'</span>)</span>
<span id="cb154-9"><a aria-hidden="true" href="#cb154-9" tabindex="-1"></a> .query(<span class="st">'sample_mode == "Flight"'</span>)</span>
<span id="cb154-10"><a aria-hidden="true" href="#cb154-10" tabindex="-1"></a> .groupby(<span class="st">'name'</span>)</span>
<span id="cb154-11"><a aria-hidden="true" href="#cb154-11" tabindex="-1"></a> [<span class="st">'counts'</span>]</span>
<span id="cb154-12"><a aria-hidden="true" href="#cb154-12" tabindex="-1"></a> .<span class="bu">sum</span>()</span>
<span id="cb154-13"><a aria-hidden="true" href="#cb154-13" tabindex="-1"></a> .sort_values()</span>
<span id="cb154-14"><a aria-hidden="true" href="#cb154-14" tabindex="-1"></a> .index</span>
<span id="cb154-15"><a aria-hidden="true" href="#cb154-15" tabindex="-1"></a> .values</span>
<span id="cb154-16"><a aria-hidden="true" href="#cb154-16" tabindex="-1"></a>)</span>
<span id="cb154-17"><a aria-hidden="true" href="#cb154-17" tabindex="-1"></a></span>
<span id="cb154-18"><a aria-hidden="true" href="#cb154-18" tabindex="-1"></a>bac_path_names <span class="op">=</span> (bac_df_sp</span>
<span id="cb154-19"><a aria-hidden="true" href="#cb154-19" tabindex="-1"></a>    .merge(bac_pathogens_df</span>
<span id="cb154-20"><a aria-hidden="true" href="#cb154-20" tabindex="-1"></a>           .query(<span class="st">'label == "Bacterial Human Pathogens"'</span>),</span>
<span id="cb154-21"><a aria-hidden="true" href="#cb154-21" tabindex="-1"></a>           on<span class="op">=</span>[<span class="st">'Genus'</span>, <span class="st">'Species'</span>], how<span class="op">=</span><span class="st">'right'</span>)</span>
<span id="cb154-22"><a aria-hidden="true" href="#cb154-22" tabindex="-1"></a>    .assign(Genus<span class="op">=</span><span class="kw">lambda</span> dd: dd.Genus.<span class="bu">str</span>[<span class="dv">0</span>] <span class="op">+</span> <span class="st">'.'</span>)</span>
<span id="cb154-23"><a aria-hidden="true" href="#cb154-23" tabindex="-1"></a>    .drop(columns<span class="op">=</span>taxon_names[:<span class="op">-</span><span class="dv">2</span>] <span class="op">+</span> [<span class="st">'label'</span>])</span>
<span id="cb154-24"><a aria-hidden="true" href="#cb154-24" tabindex="-1"></a>    .melt([<span class="st">'Genus'</span>, <span class="st">'Species'</span>], var_name<span class="op">=</span><span class="st">'sample_id'</span>, value_name<span class="op">=</span><span class="st">'counts'</span>)</span>
<span id="cb154-25"><a aria-hidden="true" href="#cb154-25" tabindex="-1"></a>    .assign(sample_mode<span class="op">=</span><span class="kw">lambda</span> dd: dd.sample_id.<span class="bu">map</span>(sample_map))</span>
<span id="cb154-26"><a aria-hidden="true" href="#cb154-26" tabindex="-1"></a>    .assign(name<span class="op">=</span><span class="kw">lambda</span> dd: dd.Genus <span class="op">+</span> <span class="st">' '</span> <span class="op">+</span> dd.Species)</span>
<span id="cb154-27"><a aria-hidden="true" href="#cb154-27" tabindex="-1"></a>    .query(<span class="st">'counts &gt; 0'</span>)</span>
<span id="cb154-28"><a aria-hidden="true" href="#cb154-28" tabindex="-1"></a>    .query(<span class="st">'sample_mode == "Flight"'</span>)</span>
<span id="cb154-29"><a aria-hidden="true" href="#cb154-29" tabindex="-1"></a>    .groupby(<span class="st">'name'</span>)</span>
<span id="cb154-30"><a aria-hidden="true" href="#cb154-30" tabindex="-1"></a>    .aggregate({<span class="st">'counts'</span>: <span class="st">'sum'</span>})</span>
<span id="cb154-31"><a aria-hidden="true" href="#cb154-31" tabindex="-1"></a>    .sort_values(<span class="st">'counts'</span>)</span>
<span id="cb154-32"><a aria-hidden="true" href="#cb154-32" tabindex="-1"></a>    .index.values</span>
<span id="cb154-33"><a aria-hidden="true" href="#cb154-33" tabindex="-1"></a>)</span>
<span id="cb154-34"><a aria-hidden="true" href="#cb154-34" tabindex="-1"></a></span>
<span id="cb154-35"><a aria-hidden="true" href="#cb154-35" tabindex="-1"></a>bac_opp_path_names <span class="op">=</span> (bac_df_sp</span>
<span id="cb154-36"><a aria-hidden="true" href="#cb154-36" tabindex="-1"></a>    .merge(bac_pathogens_df</span>
<span id="cb154-37"><a aria-hidden="true" href="#cb154-37" tabindex="-1"></a>           .query(<span class="st">'label == "Bacterial Opportunistic Human Pathogens"'</span>),</span>
<span id="cb154-38"><a aria-hidden="true" href="#cb154-38" tabindex="-1"></a>           on<span class="op">=</span>[<span class="st">'Genus'</span>, <span class="st">'Species'</span>], how<span class="op">=</span><span class="st">'right'</span>)</span>
<span id="cb154-39"><a aria-hidden="true" href="#cb154-39" tabindex="-1"></a>    .assign(Genus<span class="op">=</span><span class="kw">lambda</span> dd: dd.Genus.<span class="bu">str</span>[<span class="dv">0</span>] <span class="op">+</span> <span class="st">'.'</span>)</span>
<span id="cb154-40"><a aria-hidden="true" href="#cb154-40" tabindex="-1"></a>    .drop(columns<span class="op">=</span>taxon_names[:<span class="op">-</span><span class="dv">2</span>] <span class="op">+</span> [<span class="st">'label'</span>])</span>
<span id="cb154-41"><a aria-hidden="true" href="#cb154-41" tabindex="-1"></a>    .melt([<span class="st">'Genus'</span>, <span class="st">'Species'</span>], var_name<span class="op">=</span><span class="st">'sample_id'</span>, value_name<span class="op">=</span><span class="st">'counts'</span>)</span>
<span id="cb154-42"><a aria-hidden="true" href="#cb154-42" tabindex="-1"></a>    .assign(sample_mode<span class="op">=</span><span class="kw">lambda</span> dd: dd.sample_id.<span class="bu">map</span>(sample_map))</span>
<span id="cb154-43"><a aria-hidden="true" href="#cb154-43" tabindex="-1"></a>    .assign(name<span class="op">=</span><span class="kw">lambda</span> dd: dd.Genus <span class="op">+</span> <span class="st">' '</span> <span class="op">+</span> dd.Species)</span>
<span id="cb154-44"><a aria-hidden="true" href="#cb154-44" tabindex="-1"></a>    .query(<span class="st">'counts &gt; 0'</span>)</span>
<span id="cb154-45"><a aria-hidden="true" href="#cb154-45" tabindex="-1"></a>    .query(<span class="st">'sample_mode == "Flight"'</span>)</span>
<span id="cb154-46"><a aria-hidden="true" href="#cb154-46" tabindex="-1"></a>    .groupby(<span class="st">'name'</span>)</span>
<span id="cb154-47"><a aria-hidden="true" href="#cb154-47" tabindex="-1"></a>    .aggregate({<span class="st">'counts'</span>: <span class="st">'sum'</span>})</span>
<span id="cb154-48"><a aria-hidden="true" href="#cb154-48" tabindex="-1"></a>    .sort_values(<span class="st">'counts'</span>)</span>
<span id="cb154-49"><a aria-hidden="true" href="#cb154-49" tabindex="-1"></a>    .index.values</span>
<span id="cb154-50"><a aria-hidden="true" href="#cb154-50" tabindex="-1"></a>)</span>
<span id="cb154-51"><a aria-hidden="true" href="#cb154-51" tabindex="-1"></a></span>
<span id="cb154-52"><a aria-hidden="true" href="#cb154-52" tabindex="-1"></a>extra_opp <span class="op">=</span> (bac_pathogens_df</span>
<span id="cb154-53"><a aria-hidden="true" href="#cb154-53" tabindex="-1"></a>.query(<span class="st">'label == "Bacterial Opportunistic Human Pathogens"'</span>)</span>
<span id="cb154-54"><a aria-hidden="true" href="#cb154-54" tabindex="-1"></a> .assign(name<span class="op">=</span><span class="kw">lambda</span> dd: dd.Genus <span class="op">+</span> <span class="st">' '</span> <span class="op">+</span> dd.Species)</span>
<span id="cb154-55"><a aria-hidden="true" href="#cb154-55" tabindex="-1"></a>.query(<span class="st">'name not in @bac_opp_path_names'</span>)</span>
<span id="cb154-56"><a aria-hidden="true" href="#cb154-56" tabindex="-1"></a>.merge(bac_df_sp, on<span class="op">=</span>[<span class="st">'Genus'</span>, <span class="st">'Species'</span>], how<span class="op">=</span><span class="st">'left'</span>)</span>
<span id="cb154-57"><a aria-hidden="true" href="#cb154-57" tabindex="-1"></a>.drop(columns<span class="op">=</span>taxon_names <span class="op">+</span> [<span class="st">'label'</span>])</span>
<span id="cb154-58"><a aria-hidden="true" href="#cb154-58" tabindex="-1"></a>.melt([<span class="st">'name'</span>], var_name<span class="op">=</span><span class="st">'sample_id'</span>, value_name<span class="op">=</span><span class="st">'counts'</span>)</span>
<span id="cb154-59"><a aria-hidden="true" href="#cb154-59" tabindex="-1"></a>.groupby(<span class="st">'name'</span>)</span>
<span id="cb154-60"><a aria-hidden="true" href="#cb154-60" tabindex="-1"></a>.aggregate({<span class="st">'counts'</span>: <span class="st">'sum'</span>})</span>
<span id="cb154-61"><a aria-hidden="true" href="#cb154-61" tabindex="-1"></a>.sort_values(<span class="st">'counts'</span>)</span>
<span id="cb154-62"><a aria-hidden="true" href="#cb154-62" tabindex="-1"></a>.index.values</span>
<span id="cb154-63"><a aria-hidden="true" href="#cb154-63" tabindex="-1"></a>)</span>
<span id="cb154-64"><a aria-hidden="true" href="#cb154-64" tabindex="-1"></a></span>
<span id="cb154-65"><a aria-hidden="true" href="#cb154-65" tabindex="-1"></a>extra_fun <span class="op">=</span> (fungi_pathogens_df</span>
<span id="cb154-66"><a aria-hidden="true" href="#cb154-66" tabindex="-1"></a>.assign(name<span class="op">=</span><span class="kw">lambda</span> dd: dd.Genus.<span class="bu">str</span>[<span class="dv">0</span>] <span class="op">+</span> <span class="st">'. '</span> <span class="op">+</span> dd.Species)</span>
<span id="cb154-67"><a aria-hidden="true" href="#cb154-67" tabindex="-1"></a>.query(<span class="st">'name not in @fun_names'</span>)</span>
<span id="cb154-68"><a aria-hidden="true" href="#cb154-68" tabindex="-1"></a>.merge(fun_df_sp, on<span class="op">=</span>[<span class="st">'Genus'</span>, <span class="st">'Species'</span>], how<span class="op">=</span><span class="st">'left'</span>)</span>
<span id="cb154-69"><a aria-hidden="true" href="#cb154-69" tabindex="-1"></a>.drop(columns<span class="op">=</span>taxon_names)</span>
<span id="cb154-70"><a aria-hidden="true" href="#cb154-70" tabindex="-1"></a>.melt([<span class="st">'name'</span>], var_name<span class="op">=</span><span class="st">'sample_id'</span>, value_name<span class="op">=</span><span class="st">'counts'</span>)</span>
<span id="cb154-71"><a aria-hidden="true" href="#cb154-71" tabindex="-1"></a>.groupby(<span class="st">'name'</span>)</span>
<span id="cb154-72"><a aria-hidden="true" href="#cb154-72" tabindex="-1"></a>.aggregate({<span class="st">'counts'</span>: <span class="st">'sum'</span>})</span>
<span id="cb154-73"><a aria-hidden="true" href="#cb154-73" tabindex="-1"></a>.sort_values(<span class="st">'counts'</span>)</span>
<span id="cb154-74"><a aria-hidden="true" href="#cb154-74" tabindex="-1"></a>.index.values</span>
<span id="cb154-75"><a aria-hidden="true" href="#cb154-75" tabindex="-1"></a>)</span>
<span id="cb154-76"><a aria-hidden="true" href="#cb154-76" tabindex="-1"></a></span>
<span id="cb154-77"><a aria-hidden="true" href="#cb154-77" tabindex="-1"></a>bac_opp_path_names <span class="op">=</span> np.concatenate([extra_opp, bac_opp_path_names])</span>
<span id="cb154-78"><a aria-hidden="true" href="#cb154-78" tabindex="-1"></a>fun_names <span class="op">=</span> np.concatenate([extra_fun, fun_names])</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" id="cell-183">
<div class="sourceCode cell-code" id="cb155"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb155-1"><a aria-hidden="true" href="#cb155-1" tabindex="-1"></a>bac_opp_path_names</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="1070">
<pre><code>array(['Porphyromonas gingivalis', 'Cardiobacterium valvarum',
       'Fusobacterium gonidiaformans', 'Aggregatibacter aphrophilus',
       'Campylobacter ureolyticus', 'Corynebacterium durum',
       'Capnocytophaga leadbetteri', 'Prevotella disiens',
       'Acinetobacter baumannii', 'Parabacteroides goldsteinii',
       'Chryseobacterium hominis', 'Capnocytophaga gingivalis',
       'Campylobacter gracilis', 'Ralstonia pickettii',
       'Ralstonia insidiosa', 'Alloiococcus otitis',
       'Bilophila wadsworthia', 'Rothia dentocariosa',
       'Campylobacter concisus', 'Eggerthella lenta',
       'Staphylococcus equorum', 'Bacteroides eggerthii',
       'Ruminococcus bicirculans', 'Bacteroides plebeius',
       'Acinetobacter johnsonii', 'Cutibacterium acnes', 'C. ureolyticus',
       'F. gonidiaformans', 'C. leadbetteri', 'A. aphrophilus',
       'R. pickettii', 'P. goldsteinii', 'R. dentocariosa', 'C. gracilis',
       'S. equorum', 'C. gingivalis', 'E. lenta', 'B. wadsworthia',
       'C. concisus', 'B. eggerthii', 'R. bicirculans', 'A. johnsonii',
       'B. plebeius', 'C. acnes'], dtype=object)</code></pre>
</div>
</div>
<div class="cell" id="cell-184">
<div class="sourceCode cell-code" id="cb157"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb157-1"><a aria-hidden="true" href="#cb157-1" tabindex="-1"></a>pathogenic_fungi_df.dtypes</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="1430">
<pre><code>Genus      object
Species    object
label      object
dtype: object</code></pre>
</div>
</div>
<div class="cell" id="cell-185">
<div class="sourceCode cell-code" id="cb159"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb159-1"><a aria-hidden="true" href="#cb159-1" tabindex="-1"></a>(</span>
<span id="cb159-2"><a aria-hidden="true" href="#cb159-2" tabindex="-1"></a>    fun_df_sp</span>
<span id="cb159-3"><a aria-hidden="true" href="#cb159-3" tabindex="-1"></a> .drop(columns<span class="op">=</span>taxon_names[:<span class="op">-</span><span class="dv">2</span>])</span>
<span id="cb159-4"><a aria-hidden="true" href="#cb159-4" tabindex="-1"></a> .merge(pathogenic_fungi_df, on<span class="op">=</span>[<span class="st">'Genus'</span>, <span class="st">'Species'</span>], how<span class="op">=</span><span class="st">'right'</span>)</span>
<span id="cb159-5"><a aria-hidden="true" href="#cb159-5" tabindex="-1"></a> .melt([<span class="st">'Genus'</span>, <span class="st">'Species'</span>, <span class="st">'label'</span>], var_name<span class="op">=</span><span class="st">'sample_id'</span>, value_name<span class="op">=</span><span class="st">'counts'</span>)</span>
<span id="cb159-6"><a aria-hidden="true" href="#cb159-6" tabindex="-1"></a> .assign(sample_mode<span class="op">=</span><span class="kw">lambda</span> dd: dd.sample_id.<span class="bu">map</span>(sample_map))</span>
<span id="cb159-7"><a aria-hidden="true" href="#cb159-7" tabindex="-1"></a> .assign(name<span class="op">=</span><span class="kw">lambda</span> dd: dd.Genus.<span class="bu">str</span>[<span class="dv">0</span>] <span class="op">+</span> <span class="st">'. '</span> <span class="op">+</span> dd.Species)</span>
<span id="cb159-8"><a aria-hidden="true" href="#cb159-8" tabindex="-1"></a> .assign(label<span class="op">=</span><span class="st">'Fungal Human Pathogens'</span>)</span>
<span id="cb159-9"><a aria-hidden="true" href="#cb159-9" tabindex="-1"></a>)</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="1433">
<div>
<table class="dataframe caption-top table table-sm table-striped small" data-border="1" data-quarto-postprocess="true">
<thead>
<tr class="header">
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th">Genus</th>
<th data-quarto-table-cell-role="th">Species</th>
<th data-quarto-table-cell-role="th">label</th>
<th data-quarto-table-cell-role="th">sample_id</th>
<th data-quarto-table-cell-role="th">counts</th>
<th data-quarto-table-cell-role="th">sample_mode</th>
<th data-quarto-table-cell-role="th">name</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td data-quarto-table-cell-role="th">0</td>
<td>Acremonium</td>
<td>cavaraeanum</td>
<td>Fungal Human Pathogens</td>
<td>S01</td>
<td>0</td>
<td>Flight</td>
<td>A. cavaraeanum</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">1</td>
<td>Acremonium</td>
<td>fusidioides</td>
<td>Fungal Human Pathogens</td>
<td>S01</td>
<td>0</td>
<td>Flight</td>
<td>A. fusidioides</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">2</td>
<td>Acremonium</td>
<td>hyalinulum</td>
<td>Fungal Human Pathogens</td>
<td>S01</td>
<td>0</td>
<td>Flight</td>
<td>A. hyalinulum</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">3</td>
<td>Alternaria</td>
<td>metachromatica</td>
<td>Fungal Human Pathogens</td>
<td>S01</td>
<td>4384</td>
<td>Flight</td>
<td>A. metachromatica</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">4</td>
<td>Alternaria</td>
<td>rosae</td>
<td>Fungal Human Pathogens</td>
<td>S01</td>
<td>44</td>
<td>Flight</td>
<td>A. rosae</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">595</td>
<td>Saccharomyces</td>
<td>cerevisiae</td>
<td>Fungal Human Pathogens</td>
<td>S22</td>
<td>0</td>
<td>Ground</td>
<td>S. cerevisiae</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">596</td>
<td>Sarocladium</td>
<td>kiliense</td>
<td>Fungal Human Pathogens</td>
<td>S22</td>
<td>4109</td>
<td>Ground</td>
<td>S. kiliense</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">597</td>
<td>Sarocladium</td>
<td>subulatum</td>
<td>Fungal Human Pathogens</td>
<td>S22</td>
<td>0</td>
<td>Ground</td>
<td>S. subulatum</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">598</td>
<td>Sarocladium</td>
<td>zeae</td>
<td>Fungal Human Pathogens</td>
<td>S22</td>
<td>0</td>
<td>Ground</td>
<td>S. zeae</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">599</td>
<td>Trichosporon</td>
<td>asahii</td>
<td>Fungal Human Pathogens</td>
<td>S22</td>
<td>0</td>
<td>Ground</td>
<td>T. asahii</td>
</tr>
</tbody>
</table>
<p>600 rows × 7 columns</p>
</div>
</div>
</div>
<div class="cell" id="cell-186">
<div class="sourceCode cell-code" id="cb160"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb160-1"><a aria-hidden="true" href="#cb160-1" tabindex="-1"></a>fun_names <span class="op">=</span> (fun_df_sp</span>
<span id="cb160-2"><a aria-hidden="true" href="#cb160-2" tabindex="-1"></a> .drop(columns<span class="op">=</span>taxon_names[:<span class="op">-</span><span class="dv">2</span>])</span>
<span id="cb160-3"><a aria-hidden="true" href="#cb160-3" tabindex="-1"></a> .merge(pathogenic_fungi_df, on<span class="op">=</span>[<span class="st">'Genus'</span>, <span class="st">'Species'</span>], how<span class="op">=</span><span class="st">'right'</span>)</span>
<span id="cb160-4"><a aria-hidden="true" href="#cb160-4" tabindex="-1"></a> .melt([<span class="st">'Genus'</span>, <span class="st">'Species'</span>, <span class="st">'label'</span>], var_name<span class="op">=</span><span class="st">'sample_id'</span>, value_name<span class="op">=</span><span class="st">'counts'</span>)</span>
<span id="cb160-5"><a aria-hidden="true" href="#cb160-5" tabindex="-1"></a> .groupby([<span class="st">'Genus'</span>, <span class="st">'Species'</span>])</span>
<span id="cb160-6"><a aria-hidden="true" href="#cb160-6" tabindex="-1"></a> .counts</span>
<span id="cb160-7"><a aria-hidden="true" href="#cb160-7" tabindex="-1"></a> .<span class="bu">sum</span>()</span>
<span id="cb160-8"><a aria-hidden="true" href="#cb160-8" tabindex="-1"></a> .sort_values()</span>
<span id="cb160-9"><a aria-hidden="true" href="#cb160-9" tabindex="-1"></a> .reset_index()</span>
<span id="cb160-10"><a aria-hidden="true" href="#cb160-10" tabindex="-1"></a> .assign(name<span class="op">=</span><span class="kw">lambda</span> dd: dd.Genus.<span class="bu">str</span>[<span class="dv">0</span>] <span class="op">+</span> <span class="st">'. '</span> <span class="op">+</span> dd.Species)</span>
<span id="cb160-11"><a aria-hidden="true" href="#cb160-11" tabindex="-1"></a> .name</span>
<span id="cb160-12"><a aria-hidden="true" href="#cb160-12" tabindex="-1"></a>)</span>
<span id="cb160-13"><a aria-hidden="true" href="#cb160-13" tabindex="-1"></a>(fun_df_sp</span>
<span id="cb160-14"><a aria-hidden="true" href="#cb160-14" tabindex="-1"></a> .drop(columns<span class="op">=</span>taxon_names[:<span class="op">-</span><span class="dv">2</span>])</span>
<span id="cb160-15"><a aria-hidden="true" href="#cb160-15" tabindex="-1"></a> .merge(pathogenic_fungi_df, on<span class="op">=</span>[<span class="st">'Genus'</span>, <span class="st">'Species'</span>], how<span class="op">=</span><span class="st">'right'</span>)</span>
<span id="cb160-16"><a aria-hidden="true" href="#cb160-16" tabindex="-1"></a> .melt([<span class="st">'Genus'</span>, <span class="st">'Species'</span>, <span class="st">'label'</span>], var_name<span class="op">=</span><span class="st">'sample_id'</span>, value_name<span class="op">=</span><span class="st">'counts'</span>)</span>
<span id="cb160-17"><a aria-hidden="true" href="#cb160-17" tabindex="-1"></a> .assign(sample_mode<span class="op">=</span><span class="kw">lambda</span> dd: dd.sample_id.<span class="bu">map</span>(sample_map))</span>
<span id="cb160-18"><a aria-hidden="true" href="#cb160-18" tabindex="-1"></a> .assign(name<span class="op">=</span><span class="kw">lambda</span> dd: dd.Genus.<span class="bu">str</span>[<span class="dv">0</span>] <span class="op">+</span> <span class="st">'. '</span> <span class="op">+</span> dd.Species)</span>
<span id="cb160-19"><a aria-hidden="true" href="#cb160-19" tabindex="-1"></a> .assign(name<span class="op">=</span><span class="kw">lambda</span> dd: pd.Categorical(dd.name, categories<span class="op">=</span>fun_names, ordered<span class="op">=</span><span class="va">True</span>))</span>
<span id="cb160-20"><a aria-hidden="true" href="#cb160-20" tabindex="-1"></a> .query(<span class="st">'counts &gt; 0'</span>)</span>
<span id="cb160-21"><a aria-hidden="true" href="#cb160-21" tabindex="-1"></a> .pipe(<span class="kw">lambda</span> dd: p9.ggplot(dd)</span>
<span id="cb160-22"><a aria-hidden="true" href="#cb160-22" tabindex="-1"></a>      <span class="op">+</span> p9.aes(<span class="st">'sample_id'</span>,</span>
<span id="cb160-23"><a aria-hidden="true" href="#cb160-23" tabindex="-1"></a>               <span class="st">'name'</span>,</span>
<span id="cb160-24"><a aria-hidden="true" href="#cb160-24" tabindex="-1"></a>               size<span class="op">=</span><span class="st">'counts'</span>,</span>
<span id="cb160-25"><a aria-hidden="true" href="#cb160-25" tabindex="-1"></a>               color<span class="op">=</span><span class="st">'sample_mode'</span>,</span>
<span id="cb160-26"><a aria-hidden="true" href="#cb160-26" tabindex="-1"></a>               fill<span class="op">=</span><span class="st">'sample_mode'</span>)</span>
<span id="cb160-27"><a aria-hidden="true" href="#cb160-27" tabindex="-1"></a>       <span class="op">+</span> p9.geom_point(alpha<span class="op">=</span><span class="fl">.8</span>)</span>
<span id="cb160-28"><a aria-hidden="true" href="#cb160-28" tabindex="-1"></a>       <span class="op">+</span> p9.scale_y_discrete(expand<span class="op">=</span>(<span class="fl">.035</span>, <span class="fl">.035</span>))</span>
<span id="cb160-29"><a aria-hidden="true" href="#cb160-29" tabindex="-1"></a>       <span class="op">+</span> p9.scale_fill_manual(values<span class="op">=</span>[<span class="st">'#D3894C'</span>, <span class="st">'#435B97'</span>])</span>
<span id="cb160-30"><a aria-hidden="true" href="#cb160-30" tabindex="-1"></a>       <span class="op">+</span> p9.scale_color_manual(values<span class="op">=</span>[<span class="st">'#D3894C'</span>, <span class="st">'#435B97'</span>])</span>
<span id="cb160-31"><a aria-hidden="true" href="#cb160-31" tabindex="-1"></a>       <span class="op">+</span> p9.guides(size<span class="op">=</span>p9.guide_legend(ncol<span class="op">=</span><span class="dv">1</span>))</span>
<span id="cb160-32"><a aria-hidden="true" href="#cb160-32" tabindex="-1"></a>       <span class="op">+</span> p9.labs(x<span class="op">=</span><span class="st">''</span>, y<span class="op">=</span><span class="st">''</span>, size<span class="op">=</span><span class="st">'Total Reads'</span>, color<span class="op">=</span><span class="st">''</span>, fill<span class="op">=</span><span class="st">''</span>,</span>
<span id="cb160-33"><a aria-hidden="true" href="#cb160-33" tabindex="-1"></a>                  )</span>
<span id="cb160-34"><a aria-hidden="true" href="#cb160-34" tabindex="-1"></a>       <span class="op">+</span> p9.facet_wrap(<span class="st">'label'</span>, ncol<span class="op">=</span><span class="dv">1</span>, scales<span class="op">=</span><span class="st">'free_y'</span>)</span>
<span id="cb160-35"><a aria-hidden="true" href="#cb160-35" tabindex="-1"></a>       <span class="op">+</span> p9.theme(figure_size<span class="op">=</span>(<span class="dv">6</span>, <span class="fl">4.5</span>),</span>
<span id="cb160-36"><a aria-hidden="true" href="#cb160-36" tabindex="-1"></a>                  strip_text<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">12</span>),</span>
<span id="cb160-37"><a aria-hidden="true" href="#cb160-37" tabindex="-1"></a>                  axis_text<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">10</span>),</span>
<span id="cb160-38"><a aria-hidden="true" href="#cb160-38" tabindex="-1"></a>                  axis_text_x<span class="op">=</span>p9.element_text(angle<span class="op">=</span><span class="dv">90</span>),</span>
<span id="cb160-39"><a aria-hidden="true" href="#cb160-39" tabindex="-1"></a>       </span>
<span id="cb160-40"><a aria-hidden="true" href="#cb160-40" tabindex="-1"></a>       )</span>
<span id="cb160-41"><a aria-hidden="true" href="#cb160-41" tabindex="-1"></a> )</span>
<span id="cb160-42"><a aria-hidden="true" href="#cb160-42" tabindex="-1"></a>)</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img class="figure-img" height="1350" src="microbial_diversity_analysis_files/figure-html/cell-149-output-1.png" width="1800"/></p>
</figure>
</div>
</div>
</div>
<div class="cell" id="cell-187">
<div class="sourceCode cell-code" id="cb161"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb161-1"><a aria-hidden="true" href="#cb161-1" tabindex="-1"></a>bac_names <span class="op">=</span> (bac_df_sp</span>
<span id="cb161-2"><a aria-hidden="true" href="#cb161-2" tabindex="-1"></a> .drop(columns<span class="op">=</span>taxon_names[:<span class="op">-</span><span class="dv">2</span>])</span>
<span id="cb161-3"><a aria-hidden="true" href="#cb161-3" tabindex="-1"></a> .merge(pathogenic_bacteria_df, on<span class="op">=</span>[<span class="st">'Genus'</span>, <span class="st">'Species'</span>], how<span class="op">=</span><span class="st">'right'</span>)</span>
<span id="cb161-4"><a aria-hidden="true" href="#cb161-4" tabindex="-1"></a> .melt([<span class="st">'Genus'</span>, <span class="st">'Species'</span>, <span class="st">'label'</span>], var_name<span class="op">=</span><span class="st">'sample_id'</span>, value_name<span class="op">=</span><span class="st">'counts'</span>)</span>
<span id="cb161-5"><a aria-hidden="true" href="#cb161-5" tabindex="-1"></a> .groupby([<span class="st">'Genus'</span>, <span class="st">'Species'</span>])</span>
<span id="cb161-6"><a aria-hidden="true" href="#cb161-6" tabindex="-1"></a> .counts</span>
<span id="cb161-7"><a aria-hidden="true" href="#cb161-7" tabindex="-1"></a> .<span class="bu">sum</span>()</span>
<span id="cb161-8"><a aria-hidden="true" href="#cb161-8" tabindex="-1"></a> .sort_values()</span>
<span id="cb161-9"><a aria-hidden="true" href="#cb161-9" tabindex="-1"></a> .reset_index()</span>
<span id="cb161-10"><a aria-hidden="true" href="#cb161-10" tabindex="-1"></a> .assign(name<span class="op">=</span><span class="kw">lambda</span> dd: dd.Genus.<span class="bu">str</span>[<span class="dv">0</span>] <span class="op">+</span> <span class="st">'. '</span> <span class="op">+</span> dd.Species)</span>
<span id="cb161-11"><a aria-hidden="true" href="#cb161-11" tabindex="-1"></a> .name</span>
<span id="cb161-12"><a aria-hidden="true" href="#cb161-12" tabindex="-1"></a>)</span>
<span id="cb161-13"><a aria-hidden="true" href="#cb161-13" tabindex="-1"></a></span>
<span id="cb161-14"><a aria-hidden="true" href="#cb161-14" tabindex="-1"></a>(bac_df_sp</span>
<span id="cb161-15"><a aria-hidden="true" href="#cb161-15" tabindex="-1"></a> .drop(columns<span class="op">=</span>taxon_names[:<span class="op">-</span><span class="dv">2</span>])</span>
<span id="cb161-16"><a aria-hidden="true" href="#cb161-16" tabindex="-1"></a> .merge(pathogenic_bacteria_df, on<span class="op">=</span>[<span class="st">'Genus'</span>, <span class="st">'Species'</span>], how<span class="op">=</span><span class="st">'right'</span>)</span>
<span id="cb161-17"><a aria-hidden="true" href="#cb161-17" tabindex="-1"></a> .melt([<span class="st">'Genus'</span>, <span class="st">'Species'</span>, <span class="st">'label'</span>], var_name<span class="op">=</span><span class="st">'sample_id'</span>, value_name<span class="op">=</span><span class="st">'counts'</span>)</span>
<span id="cb161-18"><a aria-hidden="true" href="#cb161-18" tabindex="-1"></a> .assign(sample_mode<span class="op">=</span><span class="kw">lambda</span> dd: dd.sample_id.<span class="bu">map</span>(sample_map))</span>
<span id="cb161-19"><a aria-hidden="true" href="#cb161-19" tabindex="-1"></a> .assign(name<span class="op">=</span><span class="kw">lambda</span> dd: dd.Genus.<span class="bu">str</span>[<span class="dv">0</span>] <span class="op">+</span> <span class="st">'. '</span> <span class="op">+</span> dd.Species)</span>
<span id="cb161-20"><a aria-hidden="true" href="#cb161-20" tabindex="-1"></a> .assign(name<span class="op">=</span><span class="kw">lambda</span> dd: pd.Categorical(dd.name, categories<span class="op">=</span>bac_names, ordered<span class="op">=</span><span class="va">True</span>))</span>
<span id="cb161-21"><a aria-hidden="true" href="#cb161-21" tabindex="-1"></a> .query(<span class="st">'counts &gt; 0'</span>)</span>
<span id="cb161-22"><a aria-hidden="true" href="#cb161-22" tabindex="-1"></a> .pipe(<span class="kw">lambda</span> dd: p9.ggplot(dd)</span>
<span id="cb161-23"><a aria-hidden="true" href="#cb161-23" tabindex="-1"></a>      <span class="op">+</span> p9.aes(<span class="st">'sample_id'</span>,</span>
<span id="cb161-24"><a aria-hidden="true" href="#cb161-24" tabindex="-1"></a>               <span class="st">'name'</span>,</span>
<span id="cb161-25"><a aria-hidden="true" href="#cb161-25" tabindex="-1"></a>               size<span class="op">=</span><span class="st">'counts'</span>,</span>
<span id="cb161-26"><a aria-hidden="true" href="#cb161-26" tabindex="-1"></a>               color<span class="op">=</span><span class="st">'sample_mode'</span>,</span>
<span id="cb161-27"><a aria-hidden="true" href="#cb161-27" tabindex="-1"></a>               fill<span class="op">=</span><span class="st">'sample_mode'</span>)</span>
<span id="cb161-28"><a aria-hidden="true" href="#cb161-28" tabindex="-1"></a>       <span class="op">+</span> p9.geom_point(alpha<span class="op">=</span><span class="fl">.8</span>)</span>
<span id="cb161-29"><a aria-hidden="true" href="#cb161-29" tabindex="-1"></a>       <span class="op">+</span> p9.scale_y_discrete(expand<span class="op">=</span>(<span class="fl">.01</span>, <span class="fl">.01</span>))</span>
<span id="cb161-30"><a aria-hidden="true" href="#cb161-30" tabindex="-1"></a>       <span class="op">+</span> p9.scale_fill_manual(values<span class="op">=</span>[<span class="st">'#D3894C'</span>, <span class="st">'#435B97'</span>])</span>
<span id="cb161-31"><a aria-hidden="true" href="#cb161-31" tabindex="-1"></a>       <span class="op">+</span> p9.scale_color_manual(values<span class="op">=</span>[<span class="st">'#D3894C'</span>, <span class="st">'#435B97'</span>])</span>
<span id="cb161-32"><a aria-hidden="true" href="#cb161-32" tabindex="-1"></a>       <span class="op">+</span> p9.guides(size<span class="op">=</span>p9.guide_legend(ncol<span class="op">=</span><span class="dv">1</span>))</span>
<span id="cb161-33"><a aria-hidden="true" href="#cb161-33" tabindex="-1"></a>       <span class="op">+</span> p9.labs(x<span class="op">=</span><span class="st">''</span>, y<span class="op">=</span><span class="st">''</span>, size<span class="op">=</span><span class="st">'Total Reads'</span>, color<span class="op">=</span><span class="st">''</span>, fill<span class="op">=</span><span class="st">''</span>,</span>
<span id="cb161-34"><a aria-hidden="true" href="#cb161-34" tabindex="-1"></a>                  )</span>
<span id="cb161-35"><a aria-hidden="true" href="#cb161-35" tabindex="-1"></a>       <span class="op">+</span> p9.facet_wrap(<span class="st">'label'</span>, ncol<span class="op">=</span><span class="dv">1</span>, scales<span class="op">=</span><span class="st">'free_y'</span>)</span>
<span id="cb161-36"><a aria-hidden="true" href="#cb161-36" tabindex="-1"></a>       <span class="op">+</span> p9.theme(figure_size<span class="op">=</span>(<span class="dv">6</span>, <span class="dv">9</span>),</span>
<span id="cb161-37"><a aria-hidden="true" href="#cb161-37" tabindex="-1"></a>                  strip_text<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">12</span>),</span>
<span id="cb161-38"><a aria-hidden="true" href="#cb161-38" tabindex="-1"></a>                  axis_text_y<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">8</span>),</span>
<span id="cb161-39"><a aria-hidden="true" href="#cb161-39" tabindex="-1"></a>                  axis_text_x<span class="op">=</span>p9.element_text(angle<span class="op">=</span><span class="dv">90</span>),</span>
<span id="cb161-40"><a aria-hidden="true" href="#cb161-40" tabindex="-1"></a>       </span>
<span id="cb161-41"><a aria-hidden="true" href="#cb161-41" tabindex="-1"></a>       )</span>
<span id="cb161-42"><a aria-hidden="true" href="#cb161-42" tabindex="-1"></a> )</span>
<span id="cb161-43"><a aria-hidden="true" href="#cb161-43" tabindex="-1"></a>)</span>
<span id="cb161-44"><a aria-hidden="true" href="#cb161-44" tabindex="-1"></a></span>
<span id="cb161-45"><a aria-hidden="true" href="#cb161-45" tabindex="-1"></a>(fun_df_sp</span>
<span id="cb161-46"><a aria-hidden="true" href="#cb161-46" tabindex="-1"></a> .drop(columns<span class="op">=</span>taxon_names[:<span class="op">-</span><span class="dv">2</span>])</span>
<span id="cb161-47"><a aria-hidden="true" href="#cb161-47" tabindex="-1"></a> .merge(pathogenic_fungi_df, on<span class="op">=</span>[<span class="st">'Genus'</span>, <span class="st">'Species'</span>], how<span class="op">=</span><span class="st">'right'</span>)</span>
<span id="cb161-48"><a aria-hidden="true" href="#cb161-48" tabindex="-1"></a> .melt([<span class="st">'Genus'</span>, <span class="st">'Species'</span>, <span class="st">'label'</span>], var_name<span class="op">=</span><span class="st">'sample_id'</span>, value_name<span class="op">=</span><span class="st">'counts'</span>)</span>
<span id="cb161-49"><a aria-hidden="true" href="#cb161-49" tabindex="-1"></a> .assign(sample_mode<span class="op">=</span><span class="kw">lambda</span> dd: dd.sample_id.<span class="bu">map</span>(sample_map))</span>
<span id="cb161-50"><a aria-hidden="true" href="#cb161-50" tabindex="-1"></a> .assign(name<span class="op">=</span><span class="kw">lambda</span> dd: dd.Genus.<span class="bu">str</span>[<span class="dv">0</span>] <span class="op">+</span> <span class="st">'. '</span> <span class="op">+</span> dd.Species)</span>
<span id="cb161-51"><a aria-hidden="true" href="#cb161-51" tabindex="-1"></a> .assign(name<span class="op">=</span><span class="kw">lambda</span> dd: pd.Categorical(dd.name, categories<span class="op">=</span>fun_names, ordered<span class="op">=</span><span class="va">True</span>))</span>
<span id="cb161-52"><a aria-hidden="true" href="#cb161-52" tabindex="-1"></a> .query(<span class="st">'counts &gt; 0'</span>)</span>
<span id="cb161-53"><a aria-hidden="true" href="#cb161-53" tabindex="-1"></a> .pipe(<span class="kw">lambda</span> dd: p9.ggplot(dd)</span>
<span id="cb161-54"><a aria-hidden="true" href="#cb161-54" tabindex="-1"></a>      <span class="op">+</span> p9.aes(<span class="st">'sample_id'</span>,</span>
<span id="cb161-55"><a aria-hidden="true" href="#cb161-55" tabindex="-1"></a>               <span class="st">'name'</span>,</span>
<span id="cb161-56"><a aria-hidden="true" href="#cb161-56" tabindex="-1"></a>               size<span class="op">=</span><span class="st">'counts'</span>,</span>
<span id="cb161-57"><a aria-hidden="true" href="#cb161-57" tabindex="-1"></a>               color<span class="op">=</span><span class="st">'sample_mode'</span>,</span>
<span id="cb161-58"><a aria-hidden="true" href="#cb161-58" tabindex="-1"></a>               fill<span class="op">=</span><span class="st">'sample_mode'</span>)</span>
<span id="cb161-59"><a aria-hidden="true" href="#cb161-59" tabindex="-1"></a>       <span class="op">+</span> p9.geom_point(alpha<span class="op">=</span><span class="fl">.8</span>)</span>
<span id="cb161-60"><a aria-hidden="true" href="#cb161-60" tabindex="-1"></a>       <span class="op">+</span> p9.scale_y_discrete(expand<span class="op">=</span>(<span class="fl">.035</span>, <span class="fl">.035</span>))</span>
<span id="cb161-61"><a aria-hidden="true" href="#cb161-61" tabindex="-1"></a>       <span class="op">+</span> p9.scale_fill_manual(values<span class="op">=</span>[<span class="st">'#D3894C'</span>, <span class="st">'#435B97'</span>])</span>
<span id="cb161-62"><a aria-hidden="true" href="#cb161-62" tabindex="-1"></a>       <span class="op">+</span> p9.scale_color_manual(values<span class="op">=</span>[<span class="st">'#D3894C'</span>, <span class="st">'#435B97'</span>])</span>
<span id="cb161-63"><a aria-hidden="true" href="#cb161-63" tabindex="-1"></a>       <span class="op">+</span> p9.guides(size<span class="op">=</span>p9.guide_legend(ncol<span class="op">=</span><span class="dv">1</span>))</span>
<span id="cb161-64"><a aria-hidden="true" href="#cb161-64" tabindex="-1"></a>       <span class="op">+</span> p9.labs(x<span class="op">=</span><span class="st">''</span>, y<span class="op">=</span><span class="st">''</span>, size<span class="op">=</span><span class="st">'Total Reads'</span>, color<span class="op">=</span><span class="st">''</span>, fill<span class="op">=</span><span class="st">''</span>,</span>
<span id="cb161-65"><a aria-hidden="true" href="#cb161-65" tabindex="-1"></a>                  )</span>
<span id="cb161-66"><a aria-hidden="true" href="#cb161-66" tabindex="-1"></a>       <span class="op">+</span> p9.facet_wrap(<span class="st">'label'</span>, ncol<span class="op">=</span><span class="dv">1</span>, scales<span class="op">=</span><span class="st">'free_y'</span>)</span>
<span id="cb161-67"><a aria-hidden="true" href="#cb161-67" tabindex="-1"></a>       <span class="op">+</span> p9.theme(figure_size<span class="op">=</span>(<span class="dv">6</span>, <span class="fl">4.5</span>),</span>
<span id="cb161-68"><a aria-hidden="true" href="#cb161-68" tabindex="-1"></a>                  strip_text<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">12</span>),</span>
<span id="cb161-69"><a aria-hidden="true" href="#cb161-69" tabindex="-1"></a>                  axis_text<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">10</span>),</span>
<span id="cb161-70"><a aria-hidden="true" href="#cb161-70" tabindex="-1"></a>                  axis_text_x<span class="op">=</span>p9.element_text(angle<span class="op">=</span><span class="dv">90</span>),</span>
<span id="cb161-71"><a aria-hidden="true" href="#cb161-71" tabindex="-1"></a>       </span>
<span id="cb161-72"><a aria-hidden="true" href="#cb161-72" tabindex="-1"></a>       )</span>
<span id="cb161-73"><a aria-hidden="true" href="#cb161-73" tabindex="-1"></a> )</span>
<span id="cb161-74"><a aria-hidden="true" href="#cb161-74" tabindex="-1"></a>)</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img class="figure-img" height="2700" src="microbial_diversity_analysis_files/figure-html/cell-150-output-1.png" width="1800"/></p>
</figure>
</div>
</div>
</div>
<div class="cell" id="cell-188">
<div class="sourceCode cell-code" id="cb162"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb162-1"><a aria-hidden="true" href="#cb162-1" tabindex="-1"></a>f <span class="op">=</span> (fun_df_sp</span>
<span id="cb162-2"><a aria-hidden="true" href="#cb162-2" tabindex="-1"></a> .drop(columns<span class="op">=</span>taxon_names[:<span class="op">-</span><span class="dv">2</span>])</span>
<span id="cb162-3"><a aria-hidden="true" href="#cb162-3" tabindex="-1"></a> .merge(fungi_pathogens_df, on<span class="op">=</span>[<span class="st">'Genus'</span>, <span class="st">'Species'</span>], how<span class="op">=</span><span class="st">'right'</span>)</span>
<span id="cb162-4"><a aria-hidden="true" href="#cb162-4" tabindex="-1"></a> .melt([<span class="st">'Genus'</span>, <span class="st">'Species'</span>], var_name<span class="op">=</span><span class="st">'sample_id'</span>, value_name<span class="op">=</span><span class="st">'counts'</span>)</span>
<span id="cb162-5"><a aria-hidden="true" href="#cb162-5" tabindex="-1"></a> .assign(sample_mode<span class="op">=</span><span class="kw">lambda</span> dd: dd.sample_id.<span class="bu">map</span>(sample_map))</span>
<span id="cb162-6"><a aria-hidden="true" href="#cb162-6" tabindex="-1"></a> .assign(name<span class="op">=</span><span class="kw">lambda</span> dd: dd.Genus.<span class="bu">str</span>[<span class="dv">0</span>] <span class="op">+</span> <span class="st">'. '</span> <span class="op">+</span> dd.Species)</span>
<span id="cb162-7"><a aria-hidden="true" href="#cb162-7" tabindex="-1"></a> .assign(label<span class="op">=</span><span class="st">'Fungal Human Pathogens'</span>)</span>
<span id="cb162-8"><a aria-hidden="true" href="#cb162-8" tabindex="-1"></a> .assign(name<span class="op">=</span><span class="kw">lambda</span> dd: pd.Categorical(dd.name, categories<span class="op">=</span>fun_names, ordered<span class="op">=</span><span class="va">True</span>))</span>
<span id="cb162-9"><a aria-hidden="true" href="#cb162-9" tabindex="-1"></a> .query(<span class="st">'counts &gt; 0'</span>)</span>
<span id="cb162-10"><a aria-hidden="true" href="#cb162-10" tabindex="-1"></a> .pipe(<span class="kw">lambda</span> dd: p9.ggplot(dd)</span>
<span id="cb162-11"><a aria-hidden="true" href="#cb162-11" tabindex="-1"></a>      <span class="op">+</span> p9.aes(<span class="st">'sample_id'</span>,</span>
<span id="cb162-12"><a aria-hidden="true" href="#cb162-12" tabindex="-1"></a>               <span class="st">'name'</span>,</span>
<span id="cb162-13"><a aria-hidden="true" href="#cb162-13" tabindex="-1"></a>               size<span class="op">=</span><span class="st">'counts'</span>,</span>
<span id="cb162-14"><a aria-hidden="true" href="#cb162-14" tabindex="-1"></a>               color<span class="op">=</span><span class="st">'sample_mode'</span>,</span>
<span id="cb162-15"><a aria-hidden="true" href="#cb162-15" tabindex="-1"></a>               fill<span class="op">=</span><span class="st">'sample_mode'</span>)</span>
<span id="cb162-16"><a aria-hidden="true" href="#cb162-16" tabindex="-1"></a>       <span class="op">+</span> p9.geom_point(alpha<span class="op">=</span><span class="fl">.8</span>)</span>
<span id="cb162-17"><a aria-hidden="true" href="#cb162-17" tabindex="-1"></a>       <span class="op">+</span> p9.scale_y_discrete(expand<span class="op">=</span>(<span class="fl">.035</span>, <span class="fl">.035</span>))</span>
<span id="cb162-18"><a aria-hidden="true" href="#cb162-18" tabindex="-1"></a>       <span class="op">+</span> p9.scale_fill_manual(values<span class="op">=</span>[<span class="st">'#D3894C'</span>, <span class="st">'#435B97'</span>])</span>
<span id="cb162-19"><a aria-hidden="true" href="#cb162-19" tabindex="-1"></a>       <span class="op">+</span> p9.scale_color_manual(values<span class="op">=</span>[<span class="st">'#D3894C'</span>, <span class="st">'#435B97'</span>])</span>
<span id="cb162-20"><a aria-hidden="true" href="#cb162-20" tabindex="-1"></a>       <span class="op">+</span> p9.guides(size<span class="op">=</span>p9.guide_legend(ncol<span class="op">=</span><span class="dv">1</span>))</span>
<span id="cb162-21"><a aria-hidden="true" href="#cb162-21" tabindex="-1"></a>       <span class="op">+</span> p9.labs(x<span class="op">=</span><span class="st">''</span>, y<span class="op">=</span><span class="st">''</span>, size<span class="op">=</span><span class="st">'Total Reads'</span>, color<span class="op">=</span><span class="st">''</span>, fill<span class="op">=</span><span class="st">''</span>,</span>
<span id="cb162-22"><a aria-hidden="true" href="#cb162-22" tabindex="-1"></a>                  )</span>
<span id="cb162-23"><a aria-hidden="true" href="#cb162-23" tabindex="-1"></a>       <span class="op">+</span> p9.facet_wrap(<span class="st">'label'</span>, ncol<span class="op">=</span><span class="dv">1</span>, scales<span class="op">=</span><span class="st">'free_y'</span>)</span>
<span id="cb162-24"><a aria-hidden="true" href="#cb162-24" tabindex="-1"></a>       <span class="op">+</span> p9.theme(figure_size<span class="op">=</span>(<span class="dv">6</span>, <span class="fl">3.5</span>),</span>
<span id="cb162-25"><a aria-hidden="true" href="#cb162-25" tabindex="-1"></a>                  strip_text<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">12</span>),</span>
<span id="cb162-26"><a aria-hidden="true" href="#cb162-26" tabindex="-1"></a>                  axis_text<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">10</span>),</span>
<span id="cb162-27"><a aria-hidden="true" href="#cb162-27" tabindex="-1"></a>                  axis_text_x<span class="op">=</span>p9.element_text(angle<span class="op">=</span><span class="dv">90</span>),</span>
<span id="cb162-28"><a aria-hidden="true" href="#cb162-28" tabindex="-1"></a>       </span>
<span id="cb162-29"><a aria-hidden="true" href="#cb162-29" tabindex="-1"></a>       )</span>
<span id="cb162-30"><a aria-hidden="true" href="#cb162-30" tabindex="-1"></a> )</span>
<span id="cb162-31"><a aria-hidden="true" href="#cb162-31" tabindex="-1"></a>       )</span>
<span id="cb162-32"><a aria-hidden="true" href="#cb162-32" tabindex="-1"></a>f.save(<span class="st">'../output/figures/fungal_pathogens.svg'</span>)</span>
<span id="cb162-33"><a aria-hidden="true" href="#cb162-33" tabindex="-1"></a>f.draw()</span>
<span id="cb162-34"><a aria-hidden="true" href="#cb162-34" tabindex="-1"></a></span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="1316">
<div>
<figure class="figure">
<p><img class="figure-img" height="1050" src="microbial_diversity_analysis_files/figure-html/cell-151-output-1.png" width="1800"/></p>
</figure>
</div>
</div>
</div>
<div class="cell" id="cell-189">
<div class="sourceCode cell-code" id="cb163"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb163-1"><a aria-hidden="true" href="#cb163-1" tabindex="-1"></a>f <span class="op">=</span> (bac_df_sp</span>
<span id="cb163-2"><a aria-hidden="true" href="#cb163-2" tabindex="-1"></a> .assign(Genus<span class="op">=</span><span class="kw">lambda</span> dd: dd.Genus.<span class="bu">str</span>[<span class="dv">0</span>] <span class="op">+</span> <span class="st">'.'</span>)</span>
<span id="cb163-3"><a aria-hidden="true" href="#cb163-3" tabindex="-1"></a> .drop(columns<span class="op">=</span>taxon_names[:<span class="op">-</span><span class="dv">2</span>])</span>
<span id="cb163-4"><a aria-hidden="true" href="#cb163-4" tabindex="-1"></a> .merge(bac_pathogens_df, on<span class="op">=</span>[<span class="st">'Genus'</span>, <span class="st">'Species'</span>], how<span class="op">=</span><span class="st">'right'</span>)</span>
<span id="cb163-5"><a aria-hidden="true" href="#cb163-5" tabindex="-1"></a> .melt([<span class="st">'Genus'</span>, <span class="st">'Species'</span>, <span class="st">'label'</span>], var_name<span class="op">=</span><span class="st">'sample_id'</span>, value_name<span class="op">=</span><span class="st">'counts'</span>)</span>
<span id="cb163-6"><a aria-hidden="true" href="#cb163-6" tabindex="-1"></a> .assign(sample_mode<span class="op">=</span><span class="kw">lambda</span> dd: dd.sample_id.<span class="bu">map</span>(sample_map))</span>
<span id="cb163-7"><a aria-hidden="true" href="#cb163-7" tabindex="-1"></a> .assign(name<span class="op">=</span><span class="kw">lambda</span> dd: dd.Genus <span class="op">+</span> <span class="st">' '</span> <span class="op">+</span> dd.Species)</span>
<span id="cb163-8"><a aria-hidden="true" href="#cb163-8" tabindex="-1"></a> .query(<span class="st">'label.str.contains("Opportunistic")'</span>)</span>
<span id="cb163-9"><a aria-hidden="true" href="#cb163-9" tabindex="-1"></a> .query(<span class="st">'counts &gt; 0'</span>)</span>
<span id="cb163-10"><a aria-hidden="true" href="#cb163-10" tabindex="-1"></a> .assign(name<span class="op">=</span><span class="kw">lambda</span> dd: </span>
<span id="cb163-11"><a aria-hidden="true" href="#cb163-11" tabindex="-1"></a>         pd.Categorical(</span>
<span id="cb163-12"><a aria-hidden="true" href="#cb163-12" tabindex="-1"></a>            dd.name,</span>
<span id="cb163-13"><a aria-hidden="true" href="#cb163-13" tabindex="-1"></a>            categories<span class="op">=</span>bac_opp_path_names,</span>
<span id="cb163-14"><a aria-hidden="true" href="#cb163-14" tabindex="-1"></a>            ordered<span class="op">=</span><span class="va">True</span>))</span>
<span id="cb163-15"><a aria-hidden="true" href="#cb163-15" tabindex="-1"></a> .pipe(<span class="kw">lambda</span> dd: p9.ggplot(dd)</span>
<span id="cb163-16"><a aria-hidden="true" href="#cb163-16" tabindex="-1"></a>      <span class="op">+</span> p9.aes(<span class="st">'sample_id'</span>,</span>
<span id="cb163-17"><a aria-hidden="true" href="#cb163-17" tabindex="-1"></a>               <span class="st">'name'</span>,</span>
<span id="cb163-18"><a aria-hidden="true" href="#cb163-18" tabindex="-1"></a>               size<span class="op">=</span><span class="st">'counts'</span>,</span>
<span id="cb163-19"><a aria-hidden="true" href="#cb163-19" tabindex="-1"></a>               color<span class="op">=</span><span class="st">'sample_mode'</span>,</span>
<span id="cb163-20"><a aria-hidden="true" href="#cb163-20" tabindex="-1"></a>               fill<span class="op">=</span><span class="st">'sample_mode'</span>)</span>
<span id="cb163-21"><a aria-hidden="true" href="#cb163-21" tabindex="-1"></a>       <span class="op">+</span> p9.geom_point(alpha<span class="op">=</span><span class="fl">.8</span>)</span>
<span id="cb163-22"><a aria-hidden="true" href="#cb163-22" tabindex="-1"></a>       <span class="op">+</span> p9.scale_y_discrete(expand<span class="op">=</span>(<span class="fl">.035</span>, <span class="fl">.035</span>))</span>
<span id="cb163-23"><a aria-hidden="true" href="#cb163-23" tabindex="-1"></a>       <span class="op">+</span> p9.scale_size_continuous(limits<span class="op">=</span>(<span class="dv">1</span>, <span class="dv">1000</span>))</span>
<span id="cb163-24"><a aria-hidden="true" href="#cb163-24" tabindex="-1"></a>       <span class="op">+</span> p9.scale_fill_manual(values<span class="op">=</span>[<span class="st">'#D3894C'</span>, <span class="st">'#435B97'</span>])</span>
<span id="cb163-25"><a aria-hidden="true" href="#cb163-25" tabindex="-1"></a>       <span class="op">+</span> p9.scale_color_manual(values<span class="op">=</span>[<span class="st">'#D3894C'</span>, <span class="st">'#435B97'</span>])</span>
<span id="cb163-26"><a aria-hidden="true" href="#cb163-26" tabindex="-1"></a>       <span class="op">+</span> p9.guides(size<span class="op">=</span>p9.guide_legend(ncol<span class="op">=</span><span class="dv">1</span>))</span>
<span id="cb163-27"><a aria-hidden="true" href="#cb163-27" tabindex="-1"></a>       <span class="op">+</span> p9.facet_wrap(<span class="st">'label'</span>, ncol<span class="op">=</span><span class="dv">1</span>)</span>
<span id="cb163-28"><a aria-hidden="true" href="#cb163-28" tabindex="-1"></a>       <span class="op">+</span> p9.labs(x<span class="op">=</span><span class="st">''</span>, y<span class="op">=</span><span class="st">''</span>, size<span class="op">=</span><span class="st">'Total Reads'</span>, color<span class="op">=</span><span class="st">''</span>, fill<span class="op">=</span><span class="st">''</span>,</span>
<span id="cb163-29"><a aria-hidden="true" href="#cb163-29" tabindex="-1"></a>                  )</span>
<span id="cb163-30"><a aria-hidden="true" href="#cb163-30" tabindex="-1"></a>       <span class="op">+</span> p9.facet_wrap(<span class="st">'label'</span>, ncol<span class="op">=</span><span class="dv">1</span>, scales<span class="op">=</span><span class="st">'free_y'</span>)</span>
<span id="cb163-31"><a aria-hidden="true" href="#cb163-31" tabindex="-1"></a>       <span class="op">+</span> p9.theme(figure_size<span class="op">=</span>(<span class="fl">5.5</span>, <span class="dv">5</span>),</span>
<span id="cb163-32"><a aria-hidden="true" href="#cb163-32" tabindex="-1"></a>                  strip_text<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="fl">10.4</span>),</span>
<span id="cb163-33"><a aria-hidden="true" href="#cb163-33" tabindex="-1"></a>                  axis_text<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">10</span>),</span>
<span id="cb163-34"><a aria-hidden="true" href="#cb163-34" tabindex="-1"></a>                  axis_text_x<span class="op">=</span>p9.element_text(angle<span class="op">=</span><span class="dv">90</span>),</span>
<span id="cb163-35"><a aria-hidden="true" href="#cb163-35" tabindex="-1"></a>       </span>
<span id="cb163-36"><a aria-hidden="true" href="#cb163-36" tabindex="-1"></a>       )</span>
<span id="cb163-37"><a aria-hidden="true" href="#cb163-37" tabindex="-1"></a> )</span>
<span id="cb163-38"><a aria-hidden="true" href="#cb163-38" tabindex="-1"></a></span>
<span id="cb163-39"><a aria-hidden="true" href="#cb163-39" tabindex="-1"></a>       )</span>
<span id="cb163-40"><a aria-hidden="true" href="#cb163-40" tabindex="-1"></a>f.save(<span class="st">'../output/figures/bacterial_opportunistic_pathogens.svg'</span>)</span>
<span id="cb163-41"><a aria-hidden="true" href="#cb163-41" tabindex="-1"></a>f.draw()</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="1346">
<div>
<figure class="figure">
<p><img class="figure-img" height="1500" src="microbial_diversity_analysis_files/figure-html/cell-152-output-1.png" width="1650"/></p>
</figure>
</div>
</div>
</div>
<div class="cell" id="cell-190">
<div class="sourceCode cell-code" id="cb164"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb164-1"><a aria-hidden="true" href="#cb164-1" tabindex="-1"></a>f <span class="op">=</span> (bac_df_sp</span>
<span id="cb164-2"><a aria-hidden="true" href="#cb164-2" tabindex="-1"></a> .drop(columns<span class="op">=</span>taxon_names[:<span class="op">-</span><span class="dv">2</span>])</span>
<span id="cb164-3"><a aria-hidden="true" href="#cb164-3" tabindex="-1"></a> .merge(bac_pathogens_df, on<span class="op">=</span>[<span class="st">'Genus'</span>, <span class="st">'Species'</span>], how<span class="op">=</span><span class="st">'right'</span>)</span>
<span id="cb164-4"><a aria-hidden="true" href="#cb164-4" tabindex="-1"></a>     .assign(Genus<span class="op">=</span><span class="kw">lambda</span> dd: dd.Genus.<span class="bu">str</span>[<span class="dv">0</span>] <span class="op">+</span> <span class="st">'.'</span>)</span>
<span id="cb164-5"><a aria-hidden="true" href="#cb164-5" tabindex="-1"></a> .melt([<span class="st">'Genus'</span>, <span class="st">'Species'</span>, <span class="st">'label'</span>], var_name<span class="op">=</span><span class="st">'sample_id'</span>, value_name<span class="op">=</span><span class="st">'counts'</span>)</span>
<span id="cb164-6"><a aria-hidden="true" href="#cb164-6" tabindex="-1"></a> .assign(sample_mode<span class="op">=</span><span class="kw">lambda</span> dd: dd.sample_id.<span class="bu">map</span>(sample_map))</span>
<span id="cb164-7"><a aria-hidden="true" href="#cb164-7" tabindex="-1"></a> .assign(name<span class="op">=</span><span class="kw">lambda</span> dd: dd.Genus <span class="op">+</span> <span class="st">' '</span> <span class="op">+</span> dd.Species)</span>
<span id="cb164-8"><a aria-hidden="true" href="#cb164-8" tabindex="-1"></a> .query(<span class="st">'counts &gt; 0'</span>)</span>
<span id="cb164-9"><a aria-hidden="true" href="#cb164-9" tabindex="-1"></a> .query(<span class="st">'not label.str.contains("Opportunistic")'</span>)</span>
<span id="cb164-10"><a aria-hidden="true" href="#cb164-10" tabindex="-1"></a> .assign(name<span class="op">=</span><span class="kw">lambda</span> dd: pd.Categorical(dd.name, categories<span class="op">=</span>bac_path_names, ordered<span class="op">=</span><span class="va">True</span>))</span>
<span id="cb164-11"><a aria-hidden="true" href="#cb164-11" tabindex="-1"></a> .pipe(<span class="kw">lambda</span> dd: p9.ggplot(dd)</span>
<span id="cb164-12"><a aria-hidden="true" href="#cb164-12" tabindex="-1"></a>      <span class="op">+</span> p9.aes(<span class="st">'sample_id'</span>,</span>
<span id="cb164-13"><a aria-hidden="true" href="#cb164-13" tabindex="-1"></a>               <span class="st">'name'</span>,</span>
<span id="cb164-14"><a aria-hidden="true" href="#cb164-14" tabindex="-1"></a>               size<span class="op">=</span><span class="st">'counts'</span>,</span>
<span id="cb164-15"><a aria-hidden="true" href="#cb164-15" tabindex="-1"></a>               color<span class="op">=</span><span class="st">'sample_mode'</span>,</span>
<span id="cb164-16"><a aria-hidden="true" href="#cb164-16" tabindex="-1"></a>               fill<span class="op">=</span><span class="st">'sample_mode'</span>)</span>
<span id="cb164-17"><a aria-hidden="true" href="#cb164-17" tabindex="-1"></a>       <span class="op">+</span> p9.geom_point(alpha<span class="op">=</span><span class="fl">.8</span>)</span>
<span id="cb164-18"><a aria-hidden="true" href="#cb164-18" tabindex="-1"></a>       <span class="op">+</span> p9.scale_y_discrete(expand<span class="op">=</span>(<span class="fl">.05</span>, <span class="fl">.05</span>))</span>
<span id="cb164-19"><a aria-hidden="true" href="#cb164-19" tabindex="-1"></a>       <span class="op">+</span> p9.scale_size_continuous(limits<span class="op">=</span>(<span class="dv">1</span>, <span class="dv">1000</span>))</span>
<span id="cb164-20"><a aria-hidden="true" href="#cb164-20" tabindex="-1"></a>       <span class="op">+</span> p9.scale_fill_manual(values<span class="op">=</span>[<span class="st">'#D3894C'</span>, <span class="st">'#435B97'</span>])</span>
<span id="cb164-21"><a aria-hidden="true" href="#cb164-21" tabindex="-1"></a>       <span class="op">+</span> p9.scale_color_manual(values<span class="op">=</span>[<span class="st">'#D3894C'</span>, <span class="st">'#435B97'</span>])</span>
<span id="cb164-22"><a aria-hidden="true" href="#cb164-22" tabindex="-1"></a>       <span class="op">+</span> p9.guides(size<span class="op">=</span>p9.guide_legend(ncol<span class="op">=</span><span class="dv">1</span>))</span>
<span id="cb164-23"><a aria-hidden="true" href="#cb164-23" tabindex="-1"></a>       <span class="op">+</span> p9.facet_wrap(<span class="st">'label'</span>, ncol<span class="op">=</span><span class="dv">1</span>)</span>
<span id="cb164-24"><a aria-hidden="true" href="#cb164-24" tabindex="-1"></a>       <span class="op">+</span> p9.labs(x<span class="op">=</span><span class="st">''</span>, y<span class="op">=</span><span class="st">''</span>, size<span class="op">=</span><span class="st">'Total Reads'</span>, color<span class="op">=</span><span class="st">''</span>, fill<span class="op">=</span><span class="st">''</span>,</span>
<span id="cb164-25"><a aria-hidden="true" href="#cb164-25" tabindex="-1"></a>                  )</span>
<span id="cb164-26"><a aria-hidden="true" href="#cb164-26" tabindex="-1"></a>       <span class="op">+</span> p9.facet_wrap(<span class="st">'label'</span>, ncol<span class="op">=</span><span class="dv">1</span>, scales<span class="op">=</span><span class="st">'free_y'</span>)</span>
<span id="cb164-27"><a aria-hidden="true" href="#cb164-27" tabindex="-1"></a>       <span class="op">+</span> p9.theme(figure_size<span class="op">=</span>(<span class="fl">5.75</span>, <span class="dv">3</span>),</span>
<span id="cb164-28"><a aria-hidden="true" href="#cb164-28" tabindex="-1"></a>                  strip_text<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">12</span>),</span>
<span id="cb164-29"><a aria-hidden="true" href="#cb164-29" tabindex="-1"></a>                  axis_text<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">10</span>),</span>
<span id="cb164-30"><a aria-hidden="true" href="#cb164-30" tabindex="-1"></a>                  axis_text_x<span class="op">=</span>p9.element_text(angle<span class="op">=</span><span class="dv">90</span>),</span>
<span id="cb164-31"><a aria-hidden="true" href="#cb164-31" tabindex="-1"></a>       </span>
<span id="cb164-32"><a aria-hidden="true" href="#cb164-32" tabindex="-1"></a>       )</span>
<span id="cb164-33"><a aria-hidden="true" href="#cb164-33" tabindex="-1"></a> )</span>
<span id="cb164-34"><a aria-hidden="true" href="#cb164-34" tabindex="-1"></a></span>
<span id="cb164-35"><a aria-hidden="true" href="#cb164-35" tabindex="-1"></a>       )</span>
<span id="cb164-36"><a aria-hidden="true" href="#cb164-36" tabindex="-1"></a>f.save(<span class="st">'../output/figures/bacterial_pathogens.svg'</span>)</span>
<span id="cb164-37"><a aria-hidden="true" href="#cb164-37" tabindex="-1"></a>f.draw()</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="1076">
<div>
<figure class="figure">
<p><img class="figure-img" height="900" src="microbial_diversity_analysis_files/figure-html/cell-153-output-1.png" width="1725"/></p>
</figure>
</div>
</div>
</div>
</section>
<section class="level3" id="re-sequencing-nov23">
<h3 class="anchored" data-anchor-id="re-sequencing-nov23">Re-sequencing Nov’23</h3>
<div class="cell" id="cell-192">
<div class="sourceCode cell-code" id="cb165"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb165-1"><a aria-hidden="true" href="#cb165-1" tabindex="-1"></a>individual_blanks <span class="op">=</span> [<span class="ss">f'B</span><span class="sc">{</span>i<span class="sc">}</span><span class="ss">'</span> <span class="cf">for</span> i <span class="kw">in</span> <span class="bu">range</span>(<span class="dv">1</span>, <span class="dv">22</span>)]</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" id="cell-193">
<div class="sourceCode cell-code" id="cb166"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb166-1"><a aria-hidden="true" href="#cb166-1" tabindex="-1"></a>top_genera_bac <span class="op">=</span> (full_genera_df</span>
<span id="cb166-2"><a aria-hidden="true" href="#cb166-2" tabindex="-1"></a>                .query(<span class="st">'Kingdom == "Bacteria"'</span>)</span>
<span id="cb166-3"><a aria-hidden="true" href="#cb166-3" tabindex="-1"></a>                .query(<span class="st">'Genus != "unknown"'</span>)</span>
<span id="cb166-4"><a aria-hidden="true" href="#cb166-4" tabindex="-1"></a>                .groupby(<span class="st">'Genus'</span>)</span>
<span id="cb166-5"><a aria-hidden="true" href="#cb166-5" tabindex="-1"></a>                .aggregate({<span class="st">'counts'</span>: <span class="st">'sum'</span>})</span>
<span id="cb166-6"><a aria-hidden="true" href="#cb166-6" tabindex="-1"></a>                .sort_values(<span class="st">'counts'</span>)</span>
<span id="cb166-7"><a aria-hidden="true" href="#cb166-7" tabindex="-1"></a>                .tail(<span class="dv">20</span>)</span>
<span id="cb166-8"><a aria-hidden="true" href="#cb166-8" tabindex="-1"></a>                .index</span>
<span id="cb166-9"><a aria-hidden="true" href="#cb166-9" tabindex="-1"></a>                .values</span>
<span id="cb166-10"><a aria-hidden="true" href="#cb166-10" tabindex="-1"></a>)</span>
<span id="cb166-11"><a aria-hidden="true" href="#cb166-11" tabindex="-1"></a></span>
<span id="cb166-12"><a aria-hidden="true" href="#cb166-12" tabindex="-1"></a>top_genera_fun <span class="op">=</span> (full_genera_df</span>
<span id="cb166-13"><a aria-hidden="true" href="#cb166-13" tabindex="-1"></a>              .query(<span class="st">'Kingdom == "Fungi"'</span>)</span>
<span id="cb166-14"><a aria-hidden="true" href="#cb166-14" tabindex="-1"></a>              .query(<span class="st">'Genus != "unknown"'</span>)</span>
<span id="cb166-15"><a aria-hidden="true" href="#cb166-15" tabindex="-1"></a>              .groupby(<span class="st">'Genus'</span>)</span>
<span id="cb166-16"><a aria-hidden="true" href="#cb166-16" tabindex="-1"></a>              .aggregate({<span class="st">'counts'</span>: <span class="st">'sum'</span>})</span>
<span id="cb166-17"><a aria-hidden="true" href="#cb166-17" tabindex="-1"></a>              .sort_values(<span class="st">'counts'</span>)</span>
<span id="cb166-18"><a aria-hidden="true" href="#cb166-18" tabindex="-1"></a>              .tail(<span class="dv">20</span>)</span>
<span id="cb166-19"><a aria-hidden="true" href="#cb166-19" tabindex="-1"></a>              .index</span>
<span id="cb166-20"><a aria-hidden="true" href="#cb166-20" tabindex="-1"></a>              .values</span>
<span id="cb166-21"><a aria-hidden="true" href="#cb166-21" tabindex="-1"></a>)</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" id="cell-194">
<div class="sourceCode cell-code" id="cb167"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb167-1"><a aria-hidden="true" href="#cb167-1" tabindex="-1"></a>new_labels_bac <span class="op">=</span> np.concatenate([</span>
<span id="cb167-2"><a aria-hidden="true" href="#cb167-2" tabindex="-1"></a>    np.sort(np.concatenate([top_genera_bac, [<span class="st">'Other'</span>]])),</span>
<span id="cb167-3"><a aria-hidden="true" href="#cb167-3" tabindex="-1"></a>    [<span class="st">'Unknown'</span>]</span>
<span id="cb167-4"><a aria-hidden="true" href="#cb167-4" tabindex="-1"></a>    ])</span>
<span id="cb167-5"><a aria-hidden="true" href="#cb167-5" tabindex="-1"></a></span>
<span id="cb167-6"><a aria-hidden="true" href="#cb167-6" tabindex="-1"></a>new_labels_fun <span class="op">=</span> np.concatenate([</span>
<span id="cb167-7"><a aria-hidden="true" href="#cb167-7" tabindex="-1"></a>    np.sort(np.concatenate([top_genera_fun, [<span class="st">'Other'</span>]])),</span>
<span id="cb167-8"><a aria-hidden="true" href="#cb167-8" tabindex="-1"></a>    [<span class="st">'Unknown'</span>]</span>
<span id="cb167-9"><a aria-hidden="true" href="#cb167-9" tabindex="-1"></a>    ])</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" id="cell-195">
<div class="sourceCode cell-code" id="cb168"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb168-1"><a aria-hidden="true" href="#cb168-1" tabindex="-1"></a>sample_map[<span class="st">'B_Flight'</span>] <span class="op">=</span> <span class="st">'Flight'</span></span>
<span id="cb168-2"><a aria-hidden="true" href="#cb168-2" tabindex="-1"></a>sample_map[<span class="st">'B_Ground'</span>] <span class="op">=</span> <span class="st">'Ground'</span></span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" id="cell-196">
<div class="sourceCode cell-code" id="cb169"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb169-1"><a aria-hidden="true" href="#cb169-1" tabindex="-1"></a>genera_in_blanks <span class="op">=</span> (full_genera_df</span>
<span id="cb169-2"><a aria-hidden="true" href="#cb169-2" tabindex="-1"></a>                    .query(<span class="st">'sample_id.str.startswith("B")'</span>)</span>
<span id="cb169-3"><a aria-hidden="true" href="#cb169-3" tabindex="-1"></a>                    .query(<span class="st">'counts &gt; 0'</span>)</span>
<span id="cb169-4"><a aria-hidden="true" href="#cb169-4" tabindex="-1"></a>                    .Genus.unique()</span>
<span id="cb169-5"><a aria-hidden="true" href="#cb169-5" tabindex="-1"></a>)</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" id="cell-197">
<div class="sourceCode cell-code" id="cb170"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb170-1"><a aria-hidden="true" href="#cb170-1" tabindex="-1"></a>total_counts_per_sample <span class="op">=</span> (full_genera_df</span>
<span id="cb170-2"><a aria-hidden="true" href="#cb170-2" tabindex="-1"></a> .query(<span class="st">'sample_id.str.startswith("S")'</span>)</span>
<span id="cb170-3"><a aria-hidden="true" href="#cb170-3" tabindex="-1"></a> .groupby([<span class="st">'sample_id'</span>, <span class="st">'Kingdom'</span>], as_index<span class="op">=</span><span class="va">False</span>)</span>
<span id="cb170-4"><a aria-hidden="true" href="#cb170-4" tabindex="-1"></a> .counts.<span class="bu">sum</span>()</span>
<span id="cb170-5"><a aria-hidden="true" href="#cb170-5" tabindex="-1"></a>)</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" id="cell-198">
<div class="sourceCode cell-code" id="cb171"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb171-1"><a aria-hidden="true" href="#cb171-1" tabindex="-1"></a>total_counts_blanks_removed <span class="op">=</span> (full_genera_df</span>
<span id="cb171-2"><a aria-hidden="true" href="#cb171-2" tabindex="-1"></a> .query(<span class="st">'sample_id.str.startswith("S")'</span>)</span>
<span id="cb171-3"><a aria-hidden="true" href="#cb171-3" tabindex="-1"></a> .query(<span class="st">'Genus not in @genera_in_blanks'</span>)</span>
<span id="cb171-4"><a aria-hidden="true" href="#cb171-4" tabindex="-1"></a> .groupby([<span class="st">'sample_id'</span>, <span class="st">'Kingdom'</span>], as_index<span class="op">=</span><span class="va">False</span>)</span>
<span id="cb171-5"><a aria-hidden="true" href="#cb171-5" tabindex="-1"></a> .counts.<span class="bu">sum</span>()</span>
<span id="cb171-6"><a aria-hidden="true" href="#cb171-6" tabindex="-1"></a>)</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" id="cell-199">
<div class="sourceCode cell-code" id="cb172"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb172-1"><a aria-hidden="true" href="#cb172-1" tabindex="-1"></a>(total_counts_per_sample</span>
<span id="cb172-2"><a aria-hidden="true" href="#cb172-2" tabindex="-1"></a> .merge(total_counts_blanks_removed, on<span class="op">=</span>[<span class="st">'sample_id'</span>, <span class="st">'Kingdom'</span>], how<span class="op">=</span><span class="st">'left'</span>,</span>
<span id="cb172-3"><a aria-hidden="true" href="#cb172-3" tabindex="-1"></a>        suffixes<span class="op">=</span>(<span class="st">'_total'</span>, <span class="st">'_blanks_removed'</span>))</span>
<span id="cb172-4"><a aria-hidden="true" href="#cb172-4" tabindex="-1"></a> .query(<span class="st">'counts_total &gt; 0'</span>)</span>
<span id="cb172-5"><a aria-hidden="true" href="#cb172-5" tabindex="-1"></a> .groupby(<span class="st">'Kingdom'</span>)</span>
<span id="cb172-6"><a aria-hidden="true" href="#cb172-6" tabindex="-1"></a>    .aggregate({<span class="st">'counts_total'</span>: <span class="st">'sum'</span>, <span class="st">'counts_blanks_removed'</span>: <span class="st">'sum'</span>})</span>
<span id="cb172-7"><a aria-hidden="true" href="#cb172-7" tabindex="-1"></a>    .assign(percentage<span class="op">=</span><span class="kw">lambda</span> dd: dd.counts_blanks_removed <span class="op">/</span> dd.counts_total)</span>
<span id="cb172-8"><a aria-hidden="true" href="#cb172-8" tabindex="-1"></a><span class="co">#  .assign(percentage=lambda dd: dd.counts_blanks_removed / dd.counts_total)</span></span>
<span id="cb172-9"><a aria-hidden="true" href="#cb172-9" tabindex="-1"></a><span class="co">#  .melt(['sample_id', 'Kingdom'], var_name='counts_type', value_name='counts')</span></span>
<span id="cb172-10"><a aria-hidden="true" href="#cb172-10" tabindex="-1"></a><span class="co">#  .assign(counts=lambda dd: dd.counts.astype(int))</span></span>
<span id="cb172-11"><a aria-hidden="true" href="#cb172-11" tabindex="-1"></a></span>
<span id="cb172-12"><a aria-hidden="true" href="#cb172-12" tabindex="-1"></a><span class="co">#  .query('variable != "percentage"')</span></span>
<span id="cb172-13"><a aria-hidden="true" href="#cb172-13" tabindex="-1"></a>)</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="783">
<div>
<table class="dataframe caption-top table table-sm table-striped small" data-border="1" data-quarto-postprocess="true">
<thead>
<tr class="header">
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th">counts_total</th>
<th data-quarto-table-cell-role="th">counts_blanks_removed</th>
<th data-quarto-table-cell-role="th">percentage</th>
</tr>
<tr class="odd">
<th data-quarto-table-cell-role="th">Kingdom</th>
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th"></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td data-quarto-table-cell-role="th">Bacteria</td>
<td>1671690.0</td>
<td>85696.0</td>
<td>0.051263</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">Fungi</td>
<td>2196416.0</td>
<td>1816849.0</td>
<td>0.827188</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="cell" id="cell-200">
<div class="sourceCode cell-code" id="cb173"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb173-1"><a aria-hidden="true" href="#cb173-1" tabindex="-1"></a>(total_counts_per_sample</span>
<span id="cb173-2"><a aria-hidden="true" href="#cb173-2" tabindex="-1"></a> .merge(total_counts_blanks_removed, on<span class="op">=</span>[<span class="st">'sample_id'</span>, <span class="st">'Kingdom'</span>], how<span class="op">=</span><span class="st">'left'</span>,</span>
<span id="cb173-3"><a aria-hidden="true" href="#cb173-3" tabindex="-1"></a>        suffixes<span class="op">=</span>(<span class="st">'_total'</span>, <span class="st">'_blanks_removed'</span>))</span>
<span id="cb173-4"><a aria-hidden="true" href="#cb173-4" tabindex="-1"></a> .query(<span class="st">'counts_total &gt; 0'</span>)</span>
<span id="cb173-5"><a aria-hidden="true" href="#cb173-5" tabindex="-1"></a><span class="co">#  .assign(percentage=lambda dd: dd.counts_blanks_removed / dd.counts_total)</span></span>
<span id="cb173-6"><a aria-hidden="true" href="#cb173-6" tabindex="-1"></a> .melt([<span class="st">'sample_id'</span>, <span class="st">'Kingdom'</span>], var_name<span class="op">=</span><span class="st">'counts_type'</span>, value_name<span class="op">=</span><span class="st">'counts'</span>)</span>
<span id="cb173-7"><a aria-hidden="true" href="#cb173-7" tabindex="-1"></a> .assign(counts<span class="op">=</span><span class="kw">lambda</span> dd: dd.counts.astype(<span class="bu">int</span>))</span>
<span id="cb173-8"><a aria-hidden="true" href="#cb173-8" tabindex="-1"></a> .replace({<span class="st">'counts_type'</span>: </span>
<span id="cb173-9"><a aria-hidden="true" href="#cb173-9" tabindex="-1"></a>           {<span class="st">'counts_total'</span>: <span class="st">'(1) Total Reads'</span>, </span>
<span id="cb173-10"><a aria-hidden="true" href="#cb173-10" tabindex="-1"></a>            <span class="st">'counts_blanks_removed'</span>: <span class="st">'(2) Reads after removing Genera in Blanks'</span>}})</span>
<span id="cb173-11"><a aria-hidden="true" href="#cb173-11" tabindex="-1"></a> .pipe(<span class="kw">lambda</span> dd: p9.ggplot(dd)</span>
<span id="cb173-12"><a aria-hidden="true" href="#cb173-12" tabindex="-1"></a>       <span class="op">+</span> p9.aes(<span class="st">'sample_id'</span>, <span class="st">'counts'</span>)</span>
<span id="cb173-13"><a aria-hidden="true" href="#cb173-13" tabindex="-1"></a>       <span class="op">+</span> p9.geom_col(p9.aes(fill<span class="op">=</span><span class="st">'counts_type'</span>), position<span class="op">=</span><span class="st">'dodge'</span>)</span>
<span id="cb173-14"><a aria-hidden="true" href="#cb173-14" tabindex="-1"></a>       <span class="op">+</span> p9.facet_wrap(<span class="st">'Kingdom'</span>, ncol<span class="op">=</span><span class="dv">1</span>, scales<span class="op">=</span><span class="st">'free_y'</span>)</span>
<span id="cb173-15"><a aria-hidden="true" href="#cb173-15" tabindex="-1"></a>       <span class="op">+</span> p9.labs(x<span class="op">=</span><span class="st">''</span>, y<span class="op">=</span><span class="st">'Read Counts'</span>, fill<span class="op">=</span><span class="st">''</span>)</span>
<span id="cb173-16"><a aria-hidden="true" href="#cb173-16" tabindex="-1"></a>       <span class="op">+</span> p9.theme(legend_position<span class="op">=</span><span class="st">'top'</span>)</span>
<span id="cb173-17"><a aria-hidden="true" href="#cb173-17" tabindex="-1"></a>       )</span>
<span id="cb173-18"><a aria-hidden="true" href="#cb173-18" tabindex="-1"></a>)</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img class="figure-img" height="1440" src="microbial_diversity_analysis_files/figure-html/cell-162-output-1.png" width="1920"/></p>
</figure>
</div>
</div>
</div>
<div class="cell" id="cell-202">
<div class="sourceCode cell-code" id="cb174"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb174-1"><a aria-hidden="true" href="#cb174-1" tabindex="-1"></a>(full_genera_df</span>
<span id="cb174-2"><a aria-hidden="true" href="#cb174-2" tabindex="-1"></a> .query(<span class="st">'Kingdom == "Bacteria"'</span>)</span>
<span id="cb174-3"><a aria-hidden="true" href="#cb174-3" tabindex="-1"></a> .replace({<span class="st">'unknown'</span>, np.nan})</span>
<span id="cb174-4"><a aria-hidden="true" href="#cb174-4" tabindex="-1"></a> .assign(Genus<span class="op">=</span><span class="kw">lambda</span> dd: np.where(dd.Genus.isin(top_genera_bac) <span class="op">|</span> dd.Genus.isna(), dd.Genus, <span class="st">'Other'</span>))</span>
<span id="cb174-5"><a aria-hidden="true" href="#cb174-5" tabindex="-1"></a> .groupby([<span class="st">'Genus'</span>, <span class="st">'sample_id'</span>], as_index<span class="op">=</span><span class="va">False</span>, dropna<span class="op">=</span><span class="va">False</span>)</span>
<span id="cb174-6"><a aria-hidden="true" href="#cb174-6" tabindex="-1"></a>    .aggregate({<span class="st">'counts'</span>: <span class="st">'sum'</span>})</span>
<span id="cb174-7"><a aria-hidden="true" href="#cb174-7" tabindex="-1"></a> .groupby(<span class="st">'sample_id'</span>, as_index<span class="op">=</span><span class="va">False</span>)</span>
<span id="cb174-8"><a aria-hidden="true" href="#cb174-8" tabindex="-1"></a> .<span class="bu">apply</span>(<span class="kw">lambda</span> dd: dd.assign(relative_abundance<span class="op">=</span><span class="kw">lambda</span> dd: dd.counts <span class="op">/</span> dd.counts.<span class="bu">sum</span>()))</span>
<span id="cb174-9"><a aria-hidden="true" href="#cb174-9" tabindex="-1"></a> .reset_index(drop<span class="op">=</span><span class="va">True</span>)</span>
<span id="cb174-10"><a aria-hidden="true" href="#cb174-10" tabindex="-1"></a> .melt([<span class="st">'sample_id'</span>, <span class="st">'Genus'</span>], var_name<span class="op">=</span><span class="st">'metric'</span>, value_name<span class="op">=</span><span class="st">'value'</span>)</span>
<span id="cb174-11"><a aria-hidden="true" href="#cb174-11" tabindex="-1"></a> .replace({<span class="st">'counts'</span>: <span class="st">'Number of Reads'</span>,</span>
<span id="cb174-12"><a aria-hidden="true" href="#cb174-12" tabindex="-1"></a>           <span class="st">'relative_abundance'</span>: <span class="st">'Relative Abundance'</span>})</span>
<span id="cb174-13"><a aria-hidden="true" href="#cb174-13" tabindex="-1"></a> .query(<span class="st">'sample_id != "S06"'</span>)</span>
<span id="cb174-14"><a aria-hidden="true" href="#cb174-14" tabindex="-1"></a> .assign(sample_mode<span class="op">=</span><span class="kw">lambda</span> dd: dd.sample_id.<span class="bu">map</span>(sample_map))</span>
<span id="cb174-15"><a aria-hidden="true" href="#cb174-15" tabindex="-1"></a> .replace({<span class="st">'B_Flight'</span>: <span class="st">'Blanks Pool'</span>,</span>
<span id="cb174-16"><a aria-hidden="true" href="#cb174-16" tabindex="-1"></a>           <span class="st">'B_Ground'</span>: <span class="st">'Blanks Pool'</span>})</span>
<span id="cb174-17"><a aria-hidden="true" href="#cb174-17" tabindex="-1"></a> .assign(sample_label<span class="op">=</span><span class="kw">lambda</span> dd: dd.sample_id <span class="op">+</span> <span class="st">'</span><span class="ch">\n</span><span class="st">('</span> <span class="op">+</span> dd.sample_mode <span class="op">+</span> <span class="st">')'</span>)</span>
<span id="cb174-18"><a aria-hidden="true" href="#cb174-18" tabindex="-1"></a> </span>
<span id="cb174-19"><a aria-hidden="true" href="#cb174-19" tabindex="-1"></a> .pipe(<span class="kw">lambda</span> dd: </span>
<span id="cb174-20"><a aria-hidden="true" href="#cb174-20" tabindex="-1"></a>       p9.ggplot(dd) </span>
<span id="cb174-21"><a aria-hidden="true" href="#cb174-21" tabindex="-1"></a>     <span class="op">+</span> p9.aes(<span class="st">'sample_label'</span>, <span class="st">'value'</span>, fill<span class="op">=</span><span class="st">'Genus'</span>)</span>
<span id="cb174-22"><a aria-hidden="true" href="#cb174-22" tabindex="-1"></a>     <span class="op">+</span> p9.geom_col()</span>
<span id="cb174-23"><a aria-hidden="true" href="#cb174-23" tabindex="-1"></a>     <span class="op">+</span> p9.coord_flip()</span>
<span id="cb174-24"><a aria-hidden="true" href="#cb174-24" tabindex="-1"></a>     <span class="op">+</span> p9.facet_wrap(<span class="st">'metric'</span>, scales<span class="op">=</span><span class="st">'free_x'</span>)</span>
<span id="cb174-25"><a aria-hidden="true" href="#cb174-25" tabindex="-1"></a>     <span class="op">+</span> p9.guides(fill<span class="op">=</span>p9.guide_legend(ncol<span class="op">=</span><span class="dv">4</span>))</span>
<span id="cb174-26"><a aria-hidden="true" href="#cb174-26" tabindex="-1"></a>     <span class="op">+</span> p9.labs(x<span class="op">=</span><span class="st">''</span>, y<span class="op">=</span><span class="st">''</span>, fill<span class="op">=</span><span class="st">''</span>)</span>
<span id="cb174-27"><a aria-hidden="true" href="#cb174-27" tabindex="-1"></a>     <span class="op">+</span> p9.scale_y_continuous(expand<span class="op">=</span>(<span class="dv">0</span>, <span class="dv">0</span>))</span>
<span id="cb174-28"><a aria-hidden="true" href="#cb174-28" tabindex="-1"></a>     <span class="op">+</span> p9.scale_fill_discrete(labels<span class="op">=</span><span class="bu">list</span>(new_labels_bac))</span>
<span id="cb174-29"><a aria-hidden="true" href="#cb174-29" tabindex="-1"></a>     <span class="op">+</span> p9.theme(</span>
<span id="cb174-30"><a aria-hidden="true" href="#cb174-30" tabindex="-1"></a>         figure_size<span class="op">=</span>(<span class="dv">7</span>, <span class="dv">8</span>),</span>
<span id="cb174-31"><a aria-hidden="true" href="#cb174-31" tabindex="-1"></a>         axis_text_y<span class="op">=</span>p9.element_text(ha<span class="op">=</span><span class="st">'center'</span>, x<span class="op">=-</span><span class="fl">.1</span>),</span>
<span id="cb174-32"><a aria-hidden="true" href="#cb174-32" tabindex="-1"></a>         legend_key_size<span class="op">=</span><span class="dv">8</span>,</span>
<span id="cb174-33"><a aria-hidden="true" href="#cb174-33" tabindex="-1"></a>         legend_text<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">8</span>),</span>
<span id="cb174-34"><a aria-hidden="true" href="#cb174-34" tabindex="-1"></a>         legend_position<span class="op">=</span><span class="st">'bottom'</span></span>
<span id="cb174-35"><a aria-hidden="true" href="#cb174-35" tabindex="-1"></a>         )</span>
<span id="cb174-36"><a aria-hidden="true" href="#cb174-36" tabindex="-1"></a>     )</span>
<span id="cb174-37"><a aria-hidden="true" href="#cb174-37" tabindex="-1"></a>)</span>
<span id="cb174-38"><a aria-hidden="true" href="#cb174-38" tabindex="-1"></a></span>
<span id="cb174-39"><a aria-hidden="true" href="#cb174-39" tabindex="-1"></a> </span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img class="figure-img" height="2400" src="microbial_diversity_analysis_files/figure-html/cell-163-output-1.png" width="2100"/></p>
</figure>
</div>
</div>
</div>
<div class="cell" id="cell-203">
<div class="sourceCode cell-code" id="cb175"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb175-1"><a aria-hidden="true" href="#cb175-1" tabindex="-1"></a>(full_genera_df</span>
<span id="cb175-2"><a aria-hidden="true" href="#cb175-2" tabindex="-1"></a> .query(<span class="st">'Kingdom == "Fungi"'</span>)</span>
<span id="cb175-3"><a aria-hidden="true" href="#cb175-3" tabindex="-1"></a> .query(<span class="st">'sample_id!="S06"'</span>)</span>
<span id="cb175-4"><a aria-hidden="true" href="#cb175-4" tabindex="-1"></a> .replace({<span class="st">'Genus'</span>: {<span class="st">'unknown'</span>: np.nan}})</span>
<span id="cb175-5"><a aria-hidden="true" href="#cb175-5" tabindex="-1"></a> .assign(Genus<span class="op">=</span><span class="kw">lambda</span> dd: np.where(dd.Genus.isin(top_genera_fun) <span class="op">|</span> dd.Genus.isna(), dd.Genus, <span class="st">'Other'</span>))</span>
<span id="cb175-6"><a aria-hidden="true" href="#cb175-6" tabindex="-1"></a> .groupby([<span class="st">'Genus'</span>, <span class="st">'sample_id'</span>], as_index<span class="op">=</span><span class="va">False</span>, dropna<span class="op">=</span><span class="va">False</span>)</span>
<span id="cb175-7"><a aria-hidden="true" href="#cb175-7" tabindex="-1"></a> .aggregate({<span class="st">'counts'</span>: <span class="st">'sum'</span>})</span>
<span id="cb175-8"><a aria-hidden="true" href="#cb175-8" tabindex="-1"></a> .groupby(<span class="st">'sample_id'</span>, as_index<span class="op">=</span><span class="va">False</span>)</span>
<span id="cb175-9"><a aria-hidden="true" href="#cb175-9" tabindex="-1"></a> .<span class="bu">apply</span>(<span class="kw">lambda</span> dd: dd.assign(relative_abundance<span class="op">=</span><span class="kw">lambda</span> dd: dd.counts <span class="op">/</span> dd.counts.<span class="bu">sum</span>()))</span>
<span id="cb175-10"><a aria-hidden="true" href="#cb175-10" tabindex="-1"></a> .reset_index(drop<span class="op">=</span><span class="va">True</span>)</span>
<span id="cb175-11"><a aria-hidden="true" href="#cb175-11" tabindex="-1"></a> .melt([<span class="st">'sample_id'</span>, <span class="st">'Genus'</span>], var_name<span class="op">=</span><span class="st">'metric'</span>, value_name<span class="op">=</span><span class="st">'value'</span>)</span>
<span id="cb175-12"><a aria-hidden="true" href="#cb175-12" tabindex="-1"></a> .replace({<span class="st">'counts'</span>: <span class="st">'Number of Reads'</span>,</span>
<span id="cb175-13"><a aria-hidden="true" href="#cb175-13" tabindex="-1"></a>           <span class="st">'relative_abundance'</span>: <span class="st">'Relative Abundance'</span>})</span>
<span id="cb175-14"><a aria-hidden="true" href="#cb175-14" tabindex="-1"></a> .query(<span class="st">'sample_id != "S06"'</span>)</span>
<span id="cb175-15"><a aria-hidden="true" href="#cb175-15" tabindex="-1"></a> .assign(sample_mode<span class="op">=</span><span class="kw">lambda</span> dd: dd.sample_id.<span class="bu">map</span>(sample_map))</span>
<span id="cb175-16"><a aria-hidden="true" href="#cb175-16" tabindex="-1"></a> .replace({<span class="st">'B_Flight'</span>: <span class="st">'Blanks Pool'</span>,</span>
<span id="cb175-17"><a aria-hidden="true" href="#cb175-17" tabindex="-1"></a>           <span class="st">'B_Ground'</span>: <span class="st">'Blanks Pool'</span>})</span>
<span id="cb175-18"><a aria-hidden="true" href="#cb175-18" tabindex="-1"></a> .assign(sample_label<span class="op">=</span><span class="kw">lambda</span> dd: dd.sample_id <span class="op">+</span> <span class="st">'</span><span class="ch">\n</span><span class="st">('</span> <span class="op">+</span> dd.sample_mode <span class="op">+</span> <span class="st">')'</span>)</span>
<span id="cb175-19"><a aria-hidden="true" href="#cb175-19" tabindex="-1"></a> </span>
<span id="cb175-20"><a aria-hidden="true" href="#cb175-20" tabindex="-1"></a> .pipe(<span class="kw">lambda</span> dd: </span>
<span id="cb175-21"><a aria-hidden="true" href="#cb175-21" tabindex="-1"></a>       p9.ggplot(dd) </span>
<span id="cb175-22"><a aria-hidden="true" href="#cb175-22" tabindex="-1"></a>     <span class="op">+</span> p9.aes(<span class="st">'sample_label'</span>, <span class="st">'value'</span>, fill<span class="op">=</span><span class="st">'Genus'</span>)</span>
<span id="cb175-23"><a aria-hidden="true" href="#cb175-23" tabindex="-1"></a>     <span class="op">+</span> p9.geom_col()</span>
<span id="cb175-24"><a aria-hidden="true" href="#cb175-24" tabindex="-1"></a>     <span class="op">+</span> p9.coord_flip()</span>
<span id="cb175-25"><a aria-hidden="true" href="#cb175-25" tabindex="-1"></a>     <span class="op">+</span> p9.facet_wrap(<span class="st">'metric'</span>, scales<span class="op">=</span><span class="st">'free_x'</span>)</span>
<span id="cb175-26"><a aria-hidden="true" href="#cb175-26" tabindex="-1"></a>     <span class="op">+</span> p9.guides(fill<span class="op">=</span>p9.guide_legend(ncol<span class="op">=</span><span class="dv">4</span>))</span>
<span id="cb175-27"><a aria-hidden="true" href="#cb175-27" tabindex="-1"></a>     <span class="op">+</span> p9.labs(x<span class="op">=</span><span class="st">''</span>, y<span class="op">=</span><span class="st">''</span>, fill<span class="op">=</span><span class="st">''</span>)</span>
<span id="cb175-28"><a aria-hidden="true" href="#cb175-28" tabindex="-1"></a>     <span class="op">+</span> p9.scale_y_continuous(expand<span class="op">=</span>(<span class="dv">0</span>, <span class="dv">0</span>))</span>
<span id="cb175-29"><a aria-hidden="true" href="#cb175-29" tabindex="-1"></a>     <span class="op">+</span> p9.scale_fill_discrete(labels<span class="op">=</span><span class="bu">list</span>(new_labels_fun))</span>
<span id="cb175-30"><a aria-hidden="true" href="#cb175-30" tabindex="-1"></a>     <span class="op">+</span> p9.theme(</span>
<span id="cb175-31"><a aria-hidden="true" href="#cb175-31" tabindex="-1"></a>         figure_size<span class="op">=</span>(<span class="dv">7</span>, <span class="dv">8</span>),</span>
<span id="cb175-32"><a aria-hidden="true" href="#cb175-32" tabindex="-1"></a>         axis_text_y<span class="op">=</span>p9.element_text(ha<span class="op">=</span><span class="st">'center'</span>, x<span class="op">=-</span><span class="fl">.1</span>),</span>
<span id="cb175-33"><a aria-hidden="true" href="#cb175-33" tabindex="-1"></a>         legend_key_size<span class="op">=</span><span class="dv">8</span>,</span>
<span id="cb175-34"><a aria-hidden="true" href="#cb175-34" tabindex="-1"></a>         legend_text<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">8</span>),</span>
<span id="cb175-35"><a aria-hidden="true" href="#cb175-35" tabindex="-1"></a>         legend_position<span class="op">=</span><span class="st">'bottom'</span></span>
<span id="cb175-36"><a aria-hidden="true" href="#cb175-36" tabindex="-1"></a>         )</span>
<span id="cb175-37"><a aria-hidden="true" href="#cb175-37" tabindex="-1"></a>     )</span>
<span id="cb175-38"><a aria-hidden="true" href="#cb175-38" tabindex="-1"></a>)</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img class="figure-img" height="2400" src="microbial_diversity_analysis_files/figure-html/cell-164-output-1.png" width="2100"/></p>
</figure>
</div>
</div>
</div>
<div class="cell" id="cell-204">
<div class="sourceCode cell-code" id="cb176"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb176-1"><a aria-hidden="true" href="#cb176-1" tabindex="-1"></a>(pd.concat([genera_long_df, blanks_abundance_df])</span>
<span id="cb176-2"><a aria-hidden="true" href="#cb176-2" tabindex="-1"></a> .assign(Genus<span class="op">=</span><span class="kw">lambda</span> dd: np.where(dd.Genus.isin(top_genera) <span class="op">|</span> dd.Genus.isna(), dd.Genus, <span class="st">'Other'</span>))</span>
<span id="cb176-3"><a aria-hidden="true" href="#cb176-3" tabindex="-1"></a> .groupby([<span class="st">'Genus'</span>, <span class="st">'sample_id'</span>], as_index<span class="op">=</span><span class="va">False</span>, dropna<span class="op">=</span><span class="va">False</span>)</span>
<span id="cb176-4"><a aria-hidden="true" href="#cb176-4" tabindex="-1"></a>    .aggregate({<span class="st">'counts'</span>: <span class="st">'sum'</span>})</span>
<span id="cb176-5"><a aria-hidden="true" href="#cb176-5" tabindex="-1"></a> .groupby(<span class="st">'sample_id'</span>, as_index<span class="op">=</span><span class="va">False</span>)</span>
<span id="cb176-6"><a aria-hidden="true" href="#cb176-6" tabindex="-1"></a> .<span class="bu">apply</span>(<span class="kw">lambda</span> dd: dd.assign(relative_abundance<span class="op">=</span><span class="kw">lambda</span> dd: dd.counts <span class="op">/</span> dd.counts.<span class="bu">sum</span>()))</span>
<span id="cb176-7"><a aria-hidden="true" href="#cb176-7" tabindex="-1"></a> .reset_index(drop<span class="op">=</span><span class="va">True</span>)</span>
<span id="cb176-8"><a aria-hidden="true" href="#cb176-8" tabindex="-1"></a> .melt([<span class="st">'sample_id'</span>, <span class="st">'Genus'</span>], var_name<span class="op">=</span><span class="st">'metric'</span>, value_name<span class="op">=</span><span class="st">'value'</span>)</span>
<span id="cb176-9"><a aria-hidden="true" href="#cb176-9" tabindex="-1"></a> .replace({<span class="st">'counts'</span>: <span class="st">'Number of Reads'</span>,</span>
<span id="cb176-10"><a aria-hidden="true" href="#cb176-10" tabindex="-1"></a>           <span class="st">'relative_abundance'</span>: <span class="st">'Relative Abundance'</span>})</span>
<span id="cb176-11"><a aria-hidden="true" href="#cb176-11" tabindex="-1"></a> .query(<span class="st">'sample_id != "S06"'</span>)</span>
<span id="cb176-12"><a aria-hidden="true" href="#cb176-12" tabindex="-1"></a> .assign(sample_mode<span class="op">=</span><span class="kw">lambda</span> dd: dd.sample_id.<span class="bu">map</span>(sample_map))</span>
<span id="cb176-13"><a aria-hidden="true" href="#cb176-13" tabindex="-1"></a> .replace({<span class="st">'B_Flight'</span>: <span class="st">'Blanks Pool'</span>,</span>
<span id="cb176-14"><a aria-hidden="true" href="#cb176-14" tabindex="-1"></a>           <span class="st">'B_Ground'</span>: <span class="st">'Blanks Pool'</span>})</span>
<span id="cb176-15"><a aria-hidden="true" href="#cb176-15" tabindex="-1"></a> .assign(sample_label<span class="op">=</span><span class="kw">lambda</span> dd: dd.sample_id <span class="op">+</span> <span class="st">'</span><span class="ch">\n</span><span class="st">('</span> <span class="op">+</span> dd.sample_mode <span class="op">+</span> <span class="st">')'</span>)</span>
<span id="cb176-16"><a aria-hidden="true" href="#cb176-16" tabindex="-1"></a> .query(<span class="st">'Genus=="Sphingomonas"'</span>)</span>
<span id="cb176-17"><a aria-hidden="true" href="#cb176-17" tabindex="-1"></a> .query(<span class="st">'metric=="Relative Abundance"'</span>)</span>
<span id="cb176-18"><a aria-hidden="true" href="#cb176-18" tabindex="-1"></a> .pipe(<span class="kw">lambda</span> dd: p9.ggplot(dd) </span>
<span id="cb176-19"><a aria-hidden="true" href="#cb176-19" tabindex="-1"></a>       <span class="op">+</span> p9.aes(<span class="st">'sample_label'</span>, <span class="st">'value'</span>)</span>
<span id="cb176-20"><a aria-hidden="true" href="#cb176-20" tabindex="-1"></a>       <span class="op">+</span> p9.geom_col()</span>
<span id="cb176-21"><a aria-hidden="true" href="#cb176-21" tabindex="-1"></a>       <span class="op">+</span> p9.geom_text(p9.aes(label<span class="op">=</span><span class="st">'(value * 100).round(1).astype(str) + "%"'</span>), ha<span class="op">=</span><span class="st">'left'</span>, size<span class="op">=</span><span class="dv">9</span>, nudge_y<span class="op">=</span><span class="fl">.01</span>)</span>
<span id="cb176-22"><a aria-hidden="true" href="#cb176-22" tabindex="-1"></a>       <span class="op">+</span> p9.scale_y_continuous(labels<span class="op">=</span>percent_format(), expand<span class="op">=</span>(<span class="dv">0</span>, <span class="dv">0</span>), limits<span class="op">=</span>(<span class="op">-</span><span class="fl">.01</span>, <span class="fl">1.07</span>))</span>
<span id="cb176-23"><a aria-hidden="true" href="#cb176-23" tabindex="-1"></a>       <span class="op">+</span> p9.coord_flip()</span>
<span id="cb176-24"><a aria-hidden="true" href="#cb176-24" tabindex="-1"></a>       <span class="op">+</span> p9.labs(x<span class="op">=</span><span class="st">''</span>, y<span class="op">=</span><span class="st">'Relative abundance of Sphingomonas'</span>)</span>
<span id="cb176-25"><a aria-hidden="true" href="#cb176-25" tabindex="-1"></a>       <span class="op">+</span> p9.theme(figure_size<span class="op">=</span>(<span class="dv">5</span>, <span class="dv">6</span>),</span>
<span id="cb176-26"><a aria-hidden="true" href="#cb176-26" tabindex="-1"></a>                           axis_text_y<span class="op">=</span>p9.element_text(ha<span class="op">=</span><span class="st">'center'</span>, x<span class="op">=-</span><span class="fl">.06</span>, size<span class="op">=</span><span class="dv">8</span>),</span>
<span id="cb176-27"><a aria-hidden="true" href="#cb176-27" tabindex="-1"></a>         legend_key_size<span class="op">=</span><span class="dv">8</span>,</span>
<span id="cb176-28"><a aria-hidden="true" href="#cb176-28" tabindex="-1"></a>                  )</span>
<span id="cb176-29"><a aria-hidden="true" href="#cb176-29" tabindex="-1"></a>       </span>
<span id="cb176-30"><a aria-hidden="true" href="#cb176-30" tabindex="-1"></a>       )</span>
<span id="cb176-31"><a aria-hidden="true" href="#cb176-31" tabindex="-1"></a>)</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img class="figure-img" height="1800" src="microbial_diversity_analysis_files/figure-html/cell-165-output-1.png" width="1500"/></p>
</figure>
</div>
</div>
</div>
<div class="cell" id="cell-205">
<div class="sourceCode cell-code" id="cb177"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb177-1"><a aria-hidden="true" href="#cb177-1" tabindex="-1"></a>(pd.concat([genera_long_df, blanks_abundance_df.fillna(<span class="st">'unknown'</span>)])</span>
<span id="cb177-2"><a aria-hidden="true" href="#cb177-2" tabindex="-1"></a> .groupby([<span class="st">'Kingdom'</span>, <span class="st">'Genus'</span>, <span class="st">'sample_id'</span>], as_index<span class="op">=</span><span class="va">False</span>, dropna<span class="op">=</span><span class="va">False</span>)</span>
<span id="cb177-3"><a aria-hidden="true" href="#cb177-3" tabindex="-1"></a> .aggregate({<span class="st">'counts'</span>: <span class="st">'sum'</span>})</span>
<span id="cb177-4"><a aria-hidden="true" href="#cb177-4" tabindex="-1"></a> .groupby([<span class="st">'sample_id'</span>, <span class="st">'Kingdom'</span>], as_index<span class="op">=</span><span class="va">False</span>)</span>
<span id="cb177-5"><a aria-hidden="true" href="#cb177-5" tabindex="-1"></a> .<span class="bu">apply</span>(<span class="kw">lambda</span> dd: dd.assign(relative_abundance<span class="op">=</span><span class="kw">lambda</span> dd: dd.counts <span class="op">/</span> dd.counts.<span class="bu">sum</span>()))</span>
<span id="cb177-6"><a aria-hidden="true" href="#cb177-6" tabindex="-1"></a> .reset_index(drop<span class="op">=</span><span class="va">True</span>)</span>
<span id="cb177-7"><a aria-hidden="true" href="#cb177-7" tabindex="-1"></a> .melt([<span class="st">'sample_id'</span>, <span class="st">'Genus'</span>, <span class="st">'Kingdom'</span>], var_name<span class="op">=</span><span class="st">'metric'</span>, value_name<span class="op">=</span><span class="st">'value'</span>)</span>
<span id="cb177-8"><a aria-hidden="true" href="#cb177-8" tabindex="-1"></a> .replace({<span class="st">'counts'</span>: <span class="st">'Number of Reads'</span>,</span>
<span id="cb177-9"><a aria-hidden="true" href="#cb177-9" tabindex="-1"></a>           <span class="st">'relative_abundance'</span>: <span class="st">'Relative Abundance'</span>})</span>
<span id="cb177-10"><a aria-hidden="true" href="#cb177-10" tabindex="-1"></a> .query(<span class="st">'sample_id != "S06"'</span>)</span>
<span id="cb177-11"><a aria-hidden="true" href="#cb177-11" tabindex="-1"></a> .assign(sample_mode<span class="op">=</span><span class="kw">lambda</span> dd: dd.sample_id.<span class="bu">map</span>(sample_map))</span>
<span id="cb177-12"><a aria-hidden="true" href="#cb177-12" tabindex="-1"></a> .replace({<span class="st">'B_Flight'</span>: <span class="st">'Blanks Pool'</span>,</span>
<span id="cb177-13"><a aria-hidden="true" href="#cb177-13" tabindex="-1"></a>           <span class="st">'B_Ground'</span>: <span class="st">'Blanks Pool'</span>})</span>
<span id="cb177-14"><a aria-hidden="true" href="#cb177-14" tabindex="-1"></a> .assign(sample_label<span class="op">=</span><span class="kw">lambda</span> dd: dd.sample_id <span class="op">+</span> <span class="st">'</span><span class="ch">\n</span><span class="st">('</span> <span class="op">+</span> dd.sample_mode <span class="op">+</span> <span class="st">')'</span>)</span>
<span id="cb177-15"><a aria-hidden="true" href="#cb177-15" tabindex="-1"></a> .query(<span class="st">'Genus=="unknown"'</span>)</span>
<span id="cb177-16"><a aria-hidden="true" href="#cb177-16" tabindex="-1"></a> .query(<span class="st">'metric=="Relative Abundance"'</span>)</span>
<span id="cb177-17"><a aria-hidden="true" href="#cb177-17" tabindex="-1"></a> .pipe(<span class="kw">lambda</span> dd: p9.ggplot(dd) </span>
<span id="cb177-18"><a aria-hidden="true" href="#cb177-18" tabindex="-1"></a>       <span class="op">+</span> p9.aes(<span class="st">'sample_label'</span>, <span class="st">'value'</span>)</span>
<span id="cb177-19"><a aria-hidden="true" href="#cb177-19" tabindex="-1"></a>       <span class="op">+</span> p9.geom_col()</span>
<span id="cb177-20"><a aria-hidden="true" href="#cb177-20" tabindex="-1"></a>       <span class="op">+</span> p9.geom_text(p9.aes(label<span class="op">=</span><span class="st">'(value * 100).round(1).astype(str) + "%"'</span>), ha<span class="op">=</span><span class="st">'left'</span>, size<span class="op">=</span><span class="dv">9</span>, nudge_y<span class="op">=</span><span class="fl">.01</span>)</span>
<span id="cb177-21"><a aria-hidden="true" href="#cb177-21" tabindex="-1"></a>       <span class="op">+</span> p9.scale_y_continuous(labels<span class="op">=</span>percent_format(),</span>
<span id="cb177-22"><a aria-hidden="true" href="#cb177-22" tabindex="-1"></a>                               expand<span class="op">=</span>(<span class="dv">0</span>, <span class="dv">0</span>), </span>
<span id="cb177-23"><a aria-hidden="true" href="#cb177-23" tabindex="-1"></a>                               limits<span class="op">=</span>(<span class="op">-</span><span class="fl">.01</span>, <span class="fl">.35</span>))</span>
<span id="cb177-24"><a aria-hidden="true" href="#cb177-24" tabindex="-1"></a>       <span class="op">+</span> p9.coord_flip()</span>
<span id="cb177-25"><a aria-hidden="true" href="#cb177-25" tabindex="-1"></a>       <span class="op">+</span> p9.facet_wrap(<span class="st">'Kingdom'</span>)</span>
<span id="cb177-26"><a aria-hidden="true" href="#cb177-26" tabindex="-1"></a>       <span class="op">+</span> p9.labs(x<span class="op">=</span><span class="st">''</span>, y<span class="op">=</span><span class="st">'Relative abundance of unresolved reads at Genus level'</span>)</span>
<span id="cb177-27"><a aria-hidden="true" href="#cb177-27" tabindex="-1"></a>       <span class="op">+</span> p9.theme(</span>
<span id="cb177-28"><a aria-hidden="true" href="#cb177-28" tabindex="-1"></a>           figure_size<span class="op">=</span>(<span class="dv">6</span>, <span class="dv">7</span>),</span>
<span id="cb177-29"><a aria-hidden="true" href="#cb177-29" tabindex="-1"></a>           axis_text_y<span class="op">=</span>p9.element_text(ha<span class="op">=</span><span class="st">'center'</span>, x<span class="op">=-</span><span class="fl">.11</span>, size<span class="op">=</span><span class="dv">8</span>),</span>
<span id="cb177-30"><a aria-hidden="true" href="#cb177-30" tabindex="-1"></a>           legend_key_size<span class="op">=</span><span class="dv">8</span>,</span>
<span id="cb177-31"><a aria-hidden="true" href="#cb177-31" tabindex="-1"></a>                  )</span>
<span id="cb177-32"><a aria-hidden="true" href="#cb177-32" tabindex="-1"></a>       </span>
<span id="cb177-33"><a aria-hidden="true" href="#cb177-33" tabindex="-1"></a>       )</span>
<span id="cb177-34"><a aria-hidden="true" href="#cb177-34" tabindex="-1"></a>)</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img class="figure-img" height="2100" src="microbial_diversity_analysis_files/figure-html/cell-166-output-1.png" width="1800"/></p>
</figure>
</div>
</div>
</div>
<div class="cell" id="cell-206">
<div class="sourceCode cell-code" id="cb178"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb178-1"><a aria-hidden="true" href="#cb178-1" tabindex="-1"></a>full_matrix_bac <span class="op">=</span> (full_genera_df</span>
<span id="cb178-2"><a aria-hidden="true" href="#cb178-2" tabindex="-1"></a> .fillna(<span class="st">'unknown'</span>)</span>
<span id="cb178-3"><a aria-hidden="true" href="#cb178-3" tabindex="-1"></a> .query(<span class="st">'Kingdom == "Bacteria"'</span>)</span>
<span id="cb178-4"><a aria-hidden="true" href="#cb178-4" tabindex="-1"></a> .groupby([<span class="st">'Genus'</span>, <span class="st">'sample_id'</span>], as_index<span class="op">=</span><span class="va">False</span>, dropna<span class="op">=</span><span class="va">False</span>)</span>
<span id="cb178-5"><a aria-hidden="true" href="#cb178-5" tabindex="-1"></a> .aggregate({<span class="st">'counts'</span>: <span class="st">'sum'</span>})</span>
<span id="cb178-6"><a aria-hidden="true" href="#cb178-6" tabindex="-1"></a> .groupby(<span class="st">'sample_id'</span>, as_index<span class="op">=</span><span class="va">False</span>)</span>
<span id="cb178-7"><a aria-hidden="true" href="#cb178-7" tabindex="-1"></a> .<span class="bu">apply</span>(<span class="kw">lambda</span> dd: dd.assign(relative_abundance<span class="op">=</span><span class="kw">lambda</span> dd: dd.counts <span class="op">/</span> dd.counts.<span class="bu">sum</span>()))</span>
<span id="cb178-8"><a aria-hidden="true" href="#cb178-8" tabindex="-1"></a> .reset_index(drop<span class="op">=</span><span class="va">True</span>)</span>
<span id="cb178-9"><a aria-hidden="true" href="#cb178-9" tabindex="-1"></a> .melt([<span class="st">'sample_id'</span>, <span class="st">'Genus'</span>], var_name<span class="op">=</span><span class="st">'metric'</span>, value_name<span class="op">=</span><span class="st">'value'</span>)</span>
<span id="cb178-10"><a aria-hidden="true" href="#cb178-10" tabindex="-1"></a> .replace({<span class="st">'counts'</span>: <span class="st">'Number of Reads'</span>,</span>
<span id="cb178-11"><a aria-hidden="true" href="#cb178-11" tabindex="-1"></a>           <span class="st">'relative_abundance'</span>: <span class="st">'Relative Abundance'</span>})</span>
<span id="cb178-12"><a aria-hidden="true" href="#cb178-12" tabindex="-1"></a> .query(<span class="st">'sample_id != "S06"'</span>)</span>
<span id="cb178-13"><a aria-hidden="true" href="#cb178-13" tabindex="-1"></a> .assign(sample_mode<span class="op">=</span><span class="kw">lambda</span> dd: dd.sample_id.<span class="bu">map</span>(sample_map))</span>
<span id="cb178-14"><a aria-hidden="true" href="#cb178-14" tabindex="-1"></a> .replace({<span class="st">'B_Flight'</span>: <span class="st">'Blanks Pool'</span>,</span>
<span id="cb178-15"><a aria-hidden="true" href="#cb178-15" tabindex="-1"></a>           <span class="st">'B_Ground'</span>: <span class="st">'Blanks Pool'</span>})</span>
<span id="cb178-16"><a aria-hidden="true" href="#cb178-16" tabindex="-1"></a> .assign(sample_label<span class="op">=</span><span class="kw">lambda</span> dd: dd.sample_id <span class="op">+</span> <span class="st">'</span><span class="ch">\n</span><span class="st">('</span> <span class="op">+</span> dd.sample_mode <span class="op">+</span> <span class="st">')'</span>)</span>
<span id="cb178-17"><a aria-hidden="true" href="#cb178-17" tabindex="-1"></a> .query(<span class="st">'metric=="Relative Abundance"'</span>)</span>
<span id="cb178-18"><a aria-hidden="true" href="#cb178-18" tabindex="-1"></a> .pivot(index<span class="op">=</span><span class="st">'sample_label'</span>, columns<span class="op">=</span><span class="st">'Genus'</span>, values<span class="op">=</span><span class="st">'value'</span>)</span>
<span id="cb178-19"><a aria-hidden="true" href="#cb178-19" tabindex="-1"></a> .fillna(<span class="dv">0</span>)</span>
<span id="cb178-20"><a aria-hidden="true" href="#cb178-20" tabindex="-1"></a>)</span>
<span id="cb178-21"><a aria-hidden="true" href="#cb178-21" tabindex="-1"></a>samples <span class="op">=</span> full_matrix_bac.index.values</span>
<span id="cb178-22"><a aria-hidden="true" href="#cb178-22" tabindex="-1"></a>dm <span class="op">=</span> (full_matrix_bac</span>
<span id="cb178-23"><a aria-hidden="true" href="#cb178-23" tabindex="-1"></a> .pipe(<span class="kw">lambda</span> dd: pairwise_distances(dd, metric<span class="op">=</span><span class="st">'braycurtis'</span>))</span>
<span id="cb178-24"><a aria-hidden="true" href="#cb178-24" tabindex="-1"></a>)</span>
<span id="cb178-25"><a aria-hidden="true" href="#cb178-25" tabindex="-1"></a>(pd.DataFrame(dm, columns<span class="op">=</span>samples)</span>
<span id="cb178-26"><a aria-hidden="true" href="#cb178-26" tabindex="-1"></a> .assign(sample_id<span class="op">=</span>samples)</span>
<span id="cb178-27"><a aria-hidden="true" href="#cb178-27" tabindex="-1"></a> .melt(<span class="st">'sample_id'</span>, var_name<span class="op">=</span><span class="st">'sample_id2'</span>, value_name<span class="op">=</span><span class="st">'distance'</span>)</span>
<span id="cb178-28"><a aria-hidden="true" href="#cb178-28" tabindex="-1"></a> .assign(label<span class="op">=</span><span class="kw">lambda</span> dd: dd.distance.astype(<span class="bu">str</span>).<span class="bu">str</span>[<span class="dv">1</span>:<span class="dv">4</span>])</span>
<span id="cb178-29"><a aria-hidden="true" href="#cb178-29" tabindex="-1"></a> .assign(sample_id<span class="op">=</span><span class="kw">lambda</span> dd: </span>
<span id="cb178-30"><a aria-hidden="true" href="#cb178-30" tabindex="-1"></a>         pd.Categorical(dd.sample_id, categories<span class="op">=</span>samples, ordered<span class="op">=</span><span class="va">True</span>))</span>
<span id="cb178-31"><a aria-hidden="true" href="#cb178-31" tabindex="-1"></a> .assign(sample_id2<span class="op">=</span><span class="kw">lambda</span> dd: </span>
<span id="cb178-32"><a aria-hidden="true" href="#cb178-32" tabindex="-1"></a>         pd.Categorical(dd.sample_id2, categories<span class="op">=</span>samples, ordered<span class="op">=</span><span class="va">True</span>))</span>
<span id="cb178-33"><a aria-hidden="true" href="#cb178-33" tabindex="-1"></a> .assign(sample_id<span class="op">=</span><span class="kw">lambda</span> dd: dd.sample_id.<span class="bu">str</span>.replace(<span class="st">'</span><span class="ch">\n</span><span class="st">'</span>, <span class="st">' '</span>))</span>
<span id="cb178-34"><a aria-hidden="true" href="#cb178-34" tabindex="-1"></a> .assign(sample_id2<span class="op">=</span><span class="kw">lambda</span> dd: dd.sample_id2.<span class="bu">str</span>.replace(<span class="st">'</span><span class="ch">\n</span><span class="st">'</span>, <span class="st">' '</span>))</span>
<span id="cb178-35"><a aria-hidden="true" href="#cb178-35" tabindex="-1"></a> .pipe(<span class="kw">lambda</span> dd: p9.ggplot(dd)</span>
<span id="cb178-36"><a aria-hidden="true" href="#cb178-36" tabindex="-1"></a>        <span class="op">+</span> p9.aes(<span class="st">'sample_id'</span>, <span class="st">'sample_id2'</span>, fill<span class="op">=</span><span class="st">'distance'</span>)</span>
<span id="cb178-37"><a aria-hidden="true" href="#cb178-37" tabindex="-1"></a>        <span class="op">+</span> p9.geom_tile()</span>
<span id="cb178-38"><a aria-hidden="true" href="#cb178-38" tabindex="-1"></a>        <span class="op">+</span> p9.scale_fill_continuous(<span class="st">'inferno'</span>, na_value<span class="op">=</span><span class="st">'white'</span>)</span>
<span id="cb178-39"><a aria-hidden="true" href="#cb178-39" tabindex="-1"></a>        <span class="op">+</span> p9.theme_classic()</span>
<span id="cb178-40"><a aria-hidden="true" href="#cb178-40" tabindex="-1"></a>        <span class="op">+</span> p9.geom_text(p9.aes(label<span class="op">=</span><span class="st">'label'</span>, color<span class="op">=</span><span class="st">'distance &gt;= .35'</span>), size<span class="op">=</span><span class="dv">7</span>)</span>
<span id="cb178-41"><a aria-hidden="true" href="#cb178-41" tabindex="-1"></a>        <span class="op">+</span> p9.scale_color_manual([<span class="st">'white'</span>, <span class="st">'black'</span>])</span>
<span id="cb178-42"><a aria-hidden="true" href="#cb178-42" tabindex="-1"></a>        <span class="op">+</span> p9.guides(fill<span class="op">=</span><span class="va">False</span>, color<span class="op">=</span><span class="va">False</span>)</span>
<span id="cb178-43"><a aria-hidden="true" href="#cb178-43" tabindex="-1"></a>        <span class="op">+</span> p9.labs(x<span class="op">=</span><span class="st">''</span>, y<span class="op">=</span><span class="st">''</span>, title<span class="op">=</span><span class="st">'Bray-Curtis dissimilarity (16S)'</span>)</span>
<span id="cb178-44"><a aria-hidden="true" href="#cb178-44" tabindex="-1"></a>        <span class="op">+</span> p9.theme(</span>
<span id="cb178-45"><a aria-hidden="true" href="#cb178-45" tabindex="-1"></a>            axis_text_x<span class="op">=</span>p9.element_text(angle<span class="op">=</span><span class="dv">90</span>),</span>
<span id="cb178-46"><a aria-hidden="true" href="#cb178-46" tabindex="-1"></a>            axis_text<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">7</span>),</span>
<span id="cb178-47"><a aria-hidden="true" href="#cb178-47" tabindex="-1"></a>            figure_size<span class="op">=</span>(<span class="dv">5</span>, <span class="dv">5</span>),</span>
<span id="cb178-48"><a aria-hidden="true" href="#cb178-48" tabindex="-1"></a>            dpi<span class="op">=</span><span class="dv">400</span>,</span>
<span id="cb178-49"><a aria-hidden="true" href="#cb178-49" tabindex="-1"></a>            title<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">12</span>, ha<span class="op">=</span><span class="st">'center'</span>),</span>
<span id="cb178-50"><a aria-hidden="true" href="#cb178-50" tabindex="-1"></a>            )</span>
<span id="cb178-51"><a aria-hidden="true" href="#cb178-51" tabindex="-1"></a>      )</span>
<span id="cb178-52"><a aria-hidden="true" href="#cb178-52" tabindex="-1"></a></span>
<span id="cb178-53"><a aria-hidden="true" href="#cb178-53" tabindex="-1"></a>)</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img class="figure-img" height="2000" src="microbial_diversity_analysis_files/figure-html/cell-167-output-1.png" width="2000"/></p>
</figure>
</div>
</div>
</div>
<div class="cell" id="cell-207">
<div class="sourceCode cell-code" id="cb179"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb179-1"><a aria-hidden="true" href="#cb179-1" tabindex="-1"></a>full_matrix_fun <span class="op">=</span> (full_genera_df</span>
<span id="cb179-2"><a aria-hidden="true" href="#cb179-2" tabindex="-1"></a> .fillna(<span class="st">'unknown'</span>)</span>
<span id="cb179-3"><a aria-hidden="true" href="#cb179-3" tabindex="-1"></a> .query(<span class="st">'Kingdom == "Fungi"'</span>)</span>
<span id="cb179-4"><a aria-hidden="true" href="#cb179-4" tabindex="-1"></a> .groupby([<span class="st">'Genus'</span>, <span class="st">'sample_id'</span>], as_index<span class="op">=</span><span class="va">False</span>, dropna<span class="op">=</span><span class="va">False</span>)</span>
<span id="cb179-5"><a aria-hidden="true" href="#cb179-5" tabindex="-1"></a> .aggregate({<span class="st">'counts'</span>: <span class="st">'sum'</span>})</span>
<span id="cb179-6"><a aria-hidden="true" href="#cb179-6" tabindex="-1"></a> .groupby(<span class="st">'sample_id'</span>, as_index<span class="op">=</span><span class="va">False</span>)</span>
<span id="cb179-7"><a aria-hidden="true" href="#cb179-7" tabindex="-1"></a> .<span class="bu">apply</span>(<span class="kw">lambda</span> dd: dd.assign(relative_abundance<span class="op">=</span><span class="kw">lambda</span> dd: dd.counts <span class="op">/</span> dd.counts.<span class="bu">sum</span>()))</span>
<span id="cb179-8"><a aria-hidden="true" href="#cb179-8" tabindex="-1"></a> .reset_index(drop<span class="op">=</span><span class="va">True</span>)</span>
<span id="cb179-9"><a aria-hidden="true" href="#cb179-9" tabindex="-1"></a> .melt([<span class="st">'sample_id'</span>, <span class="st">'Genus'</span>], var_name<span class="op">=</span><span class="st">'metric'</span>, value_name<span class="op">=</span><span class="st">'value'</span>)</span>
<span id="cb179-10"><a aria-hidden="true" href="#cb179-10" tabindex="-1"></a> .replace({<span class="st">'counts'</span>: <span class="st">'Number of Reads'</span>,</span>
<span id="cb179-11"><a aria-hidden="true" href="#cb179-11" tabindex="-1"></a>           <span class="st">'relative_abundance'</span>: <span class="st">'Relative Abundance'</span>})</span>
<span id="cb179-12"><a aria-hidden="true" href="#cb179-12" tabindex="-1"></a> .query(<span class="st">'sample_id != "S06"'</span>)</span>
<span id="cb179-13"><a aria-hidden="true" href="#cb179-13" tabindex="-1"></a> .assign(sample_mode<span class="op">=</span><span class="kw">lambda</span> dd: dd.sample_id.<span class="bu">map</span>(sample_map))</span>
<span id="cb179-14"><a aria-hidden="true" href="#cb179-14" tabindex="-1"></a> .replace({<span class="st">'B_Flight'</span>: <span class="st">'Blanks Pool'</span>,</span>
<span id="cb179-15"><a aria-hidden="true" href="#cb179-15" tabindex="-1"></a>           <span class="st">'B_Ground'</span>: <span class="st">'Blanks Pool'</span>})</span>
<span id="cb179-16"><a aria-hidden="true" href="#cb179-16" tabindex="-1"></a> .assign(sample_label<span class="op">=</span><span class="kw">lambda</span> dd: dd.sample_id <span class="op">+</span> <span class="st">'</span><span class="ch">\n</span><span class="st">('</span> <span class="op">+</span> dd.sample_mode <span class="op">+</span> <span class="st">')'</span>)</span>
<span id="cb179-17"><a aria-hidden="true" href="#cb179-17" tabindex="-1"></a> .query(<span class="st">'metric=="Relative Abundance"'</span>)</span>
<span id="cb179-18"><a aria-hidden="true" href="#cb179-18" tabindex="-1"></a> .pivot(index<span class="op">=</span><span class="st">'sample_label'</span>, columns<span class="op">=</span><span class="st">'Genus'</span>, values<span class="op">=</span><span class="st">'value'</span>)</span>
<span id="cb179-19"><a aria-hidden="true" href="#cb179-19" tabindex="-1"></a> .fillna(<span class="dv">0</span>)</span>
<span id="cb179-20"><a aria-hidden="true" href="#cb179-20" tabindex="-1"></a>)</span>
<span id="cb179-21"><a aria-hidden="true" href="#cb179-21" tabindex="-1"></a>samples <span class="op">=</span> full_matrix_fun.index.values</span>
<span id="cb179-22"><a aria-hidden="true" href="#cb179-22" tabindex="-1"></a>dm <span class="op">=</span> (full_matrix_fun</span>
<span id="cb179-23"><a aria-hidden="true" href="#cb179-23" tabindex="-1"></a> .pipe(<span class="kw">lambda</span> dd: pairwise_distances(dd, metric<span class="op">=</span><span class="st">'braycurtis'</span>))</span>
<span id="cb179-24"><a aria-hidden="true" href="#cb179-24" tabindex="-1"></a>)</span>
<span id="cb179-25"><a aria-hidden="true" href="#cb179-25" tabindex="-1"></a>(pd.DataFrame(dm, columns<span class="op">=</span>samples)</span>
<span id="cb179-26"><a aria-hidden="true" href="#cb179-26" tabindex="-1"></a> .assign(sample_id<span class="op">=</span>samples)</span>
<span id="cb179-27"><a aria-hidden="true" href="#cb179-27" tabindex="-1"></a> .melt(<span class="st">'sample_id'</span>, var_name<span class="op">=</span><span class="st">'sample_id2'</span>, value_name<span class="op">=</span><span class="st">'distance'</span>)</span>
<span id="cb179-28"><a aria-hidden="true" href="#cb179-28" tabindex="-1"></a> .assign(label<span class="op">=</span><span class="kw">lambda</span> dd: dd.distance.astype(<span class="bu">str</span>).<span class="bu">str</span>[<span class="dv">1</span>:<span class="dv">4</span>])</span>
<span id="cb179-29"><a aria-hidden="true" href="#cb179-29" tabindex="-1"></a> .assign(sample_id<span class="op">=</span><span class="kw">lambda</span> dd: </span>
<span id="cb179-30"><a aria-hidden="true" href="#cb179-30" tabindex="-1"></a>         pd.Categorical(dd.sample_id, categories<span class="op">=</span>samples, ordered<span class="op">=</span><span class="va">True</span>))</span>
<span id="cb179-31"><a aria-hidden="true" href="#cb179-31" tabindex="-1"></a> .assign(sample_id2<span class="op">=</span><span class="kw">lambda</span> dd: </span>
<span id="cb179-32"><a aria-hidden="true" href="#cb179-32" tabindex="-1"></a>         pd.Categorical(dd.sample_id2, categories<span class="op">=</span>samples, ordered<span class="op">=</span><span class="va">True</span>))</span>
<span id="cb179-33"><a aria-hidden="true" href="#cb179-33" tabindex="-1"></a> .assign(sample_id<span class="op">=</span><span class="kw">lambda</span> dd: dd.sample_id.<span class="bu">str</span>.replace(<span class="st">'</span><span class="ch">\n</span><span class="st">'</span>, <span class="st">' '</span>))</span>
<span id="cb179-34"><a aria-hidden="true" href="#cb179-34" tabindex="-1"></a> .assign(sample_id2<span class="op">=</span><span class="kw">lambda</span> dd: dd.sample_id2.<span class="bu">str</span>.replace(<span class="st">'</span><span class="ch">\n</span><span class="st">'</span>, <span class="st">' '</span>))</span>
<span id="cb179-35"><a aria-hidden="true" href="#cb179-35" tabindex="-1"></a> .pipe(<span class="kw">lambda</span> dd: p9.ggplot(dd)</span>
<span id="cb179-36"><a aria-hidden="true" href="#cb179-36" tabindex="-1"></a>        <span class="op">+</span> p9.aes(<span class="st">'sample_id'</span>, <span class="st">'sample_id2'</span>, fill<span class="op">=</span><span class="st">'distance'</span>)</span>
<span id="cb179-37"><a aria-hidden="true" href="#cb179-37" tabindex="-1"></a>        <span class="op">+</span> p9.geom_tile()</span>
<span id="cb179-38"><a aria-hidden="true" href="#cb179-38" tabindex="-1"></a>        <span class="op">+</span> p9.scale_fill_continuous(<span class="st">'inferno'</span>, na_value<span class="op">=</span><span class="st">'white'</span>)</span>
<span id="cb179-39"><a aria-hidden="true" href="#cb179-39" tabindex="-1"></a>        <span class="op">+</span> p9.theme_classic()</span>
<span id="cb179-40"><a aria-hidden="true" href="#cb179-40" tabindex="-1"></a>        <span class="op">+</span> p9.geom_text(p9.aes(label<span class="op">=</span><span class="st">'label'</span>, color<span class="op">=</span><span class="st">'distance &gt;= .35'</span>), size<span class="op">=</span><span class="dv">7</span>)</span>
<span id="cb179-41"><a aria-hidden="true" href="#cb179-41" tabindex="-1"></a>        <span class="op">+</span> p9.scale_color_manual([<span class="st">'white'</span>, <span class="st">'black'</span>])</span>
<span id="cb179-42"><a aria-hidden="true" href="#cb179-42" tabindex="-1"></a>        <span class="op">+</span> p9.guides(fill<span class="op">=</span><span class="va">False</span>, color<span class="op">=</span><span class="va">False</span>)</span>
<span id="cb179-43"><a aria-hidden="true" href="#cb179-43" tabindex="-1"></a>        <span class="op">+</span> p9.labs(x<span class="op">=</span><span class="st">''</span>, y<span class="op">=</span><span class="st">''</span>, title<span class="op">=</span><span class="st">'Bray-Curtis dissimilarity (ITS)'</span>)</span>
<span id="cb179-44"><a aria-hidden="true" href="#cb179-44" tabindex="-1"></a>        <span class="op">+</span> p9.theme(</span>
<span id="cb179-45"><a aria-hidden="true" href="#cb179-45" tabindex="-1"></a>            axis_text_x<span class="op">=</span>p9.element_text(angle<span class="op">=</span><span class="dv">90</span>),</span>
<span id="cb179-46"><a aria-hidden="true" href="#cb179-46" tabindex="-1"></a>            axis_text<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">7</span>),</span>
<span id="cb179-47"><a aria-hidden="true" href="#cb179-47" tabindex="-1"></a>            figure_size<span class="op">=</span>(<span class="dv">5</span>, <span class="dv">5</span>),</span>
<span id="cb179-48"><a aria-hidden="true" href="#cb179-48" tabindex="-1"></a>            dpi<span class="op">=</span><span class="dv">400</span>,</span>
<span id="cb179-49"><a aria-hidden="true" href="#cb179-49" tabindex="-1"></a>            title<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">12</span>, ha<span class="op">=</span><span class="st">'center'</span>),</span>
<span id="cb179-50"><a aria-hidden="true" href="#cb179-50" tabindex="-1"></a>            )</span>
<span id="cb179-51"><a aria-hidden="true" href="#cb179-51" tabindex="-1"></a>      )</span>
<span id="cb179-52"><a aria-hidden="true" href="#cb179-52" tabindex="-1"></a></span>
<span id="cb179-53"><a aria-hidden="true" href="#cb179-53" tabindex="-1"></a>)</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img class="figure-img" height="2000" src="microbial_diversity_analysis_files/figure-html/cell-168-output-1.png" width="2000"/></p>
</figure>
</div>
</div>
</div>
<div class="cell" id="cell-208">
<div class="sourceCode cell-code" id="cb180"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb180-1"><a aria-hidden="true" href="#cb180-1" tabindex="-1"></a>(full_long_df </span>
<span id="cb180-2"><a aria-hidden="true" href="#cb180-2" tabindex="-1"></a> .query(<span class="st">'Genus.notna()'</span>)</span>
<span id="cb180-3"><a aria-hidden="true" href="#cb180-3" tabindex="-1"></a>.assign(Genus<span class="op">=</span><span class="kw">lambda</span> dd: </span>
<span id="cb180-4"><a aria-hidden="true" href="#cb180-4" tabindex="-1"></a>        pd.Categorical(dd.Genus, categories<span class="op">=</span>top_30_genera_samples, ordered<span class="op">=</span><span class="va">True</span>))</span>
<span id="cb180-5"><a aria-hidden="true" href="#cb180-5" tabindex="-1"></a>.dropna()</span>
<span id="cb180-6"><a aria-hidden="true" href="#cb180-6" tabindex="-1"></a>.assign(sample_id<span class="op">=</span><span class="kw">lambda</span> dd: dd.sample_label.<span class="bu">str</span>.split(<span class="st">'</span><span class="ch">\n</span><span class="st">'</span>).<span class="bu">str</span>[<span class="dv">0</span>])</span>
<span id="cb180-7"><a aria-hidden="true" href="#cb180-7" tabindex="-1"></a>.merge(samples_info, how<span class="op">=</span><span class="st">'left'</span>)</span>
<span id="cb180-8"><a aria-hidden="true" href="#cb180-8" tabindex="-1"></a>.assign(month<span class="op">=</span><span class="kw">lambda</span> dd: pd.to_datetime(dd.date).dt.month_name().fillna(<span class="st">'Blanks'</span>))</span>
<span id="cb180-9"><a aria-hidden="true" href="#cb180-9" tabindex="-1"></a>.assign(sample_mode<span class="op">=</span><span class="kw">lambda</span> dd: </span>
<span id="cb180-10"><a aria-hidden="true" href="#cb180-10" tabindex="-1"></a>        np.where(dd.sample_mode.isna, </span>
<span id="cb180-11"><a aria-hidden="true" href="#cb180-11" tabindex="-1"></a>                 dd.sample_label.<span class="bu">str</span>.split(<span class="st">'('</span>).<span class="bu">str</span>[<span class="op">-</span><span class="dv">1</span>].<span class="bu">str</span>[:<span class="op">-</span><span class="dv">1</span>], </span>
<span id="cb180-12"><a aria-hidden="true" href="#cb180-12" tabindex="-1"></a>                 dd.sample_mode))</span>
<span id="cb180-13"><a aria-hidden="true" href="#cb180-13" tabindex="-1"></a>.assign(sample_group_label<span class="op">=</span><span class="kw">lambda</span> dd: dd.sample_mode <span class="op">+</span> <span class="st">' '</span> <span class="op">+</span> dd.month)</span>
<span id="cb180-14"><a aria-hidden="true" href="#cb180-14" tabindex="-1"></a>.groupby([<span class="st">'sample_group_label'</span>, <span class="st">'sample_mode'</span>, <span class="st">'Genus'</span>])</span>
<span id="cb180-15"><a aria-hidden="true" href="#cb180-15" tabindex="-1"></a>.aggregate({<span class="st">'value'</span>: <span class="st">'mean'</span>})</span>
<span id="cb180-16"><a aria-hidden="true" href="#cb180-16" tabindex="-1"></a>.reset_index()</span>
<span id="cb180-17"><a aria-hidden="true" href="#cb180-17" tabindex="-1"></a>.query(<span class="st">'value &gt; 0'</span>)</span>
<span id="cb180-18"><a aria-hidden="true" href="#cb180-18" tabindex="-1"></a>.assign(value<span class="op">=</span><span class="kw">lambda</span> dd: np.where(dd.value <span class="op">&lt;</span> treshold, treshold, dd.value))</span>
<span id="cb180-19"><a aria-hidden="true" href="#cb180-19" tabindex="-1"></a>.assign(sample_group_label<span class="op">=</span><span class="kw">lambda</span> dd: pd.Categorical(dd.sample_group_label,</span>
<span id="cb180-20"><a aria-hidden="true" href="#cb180-20" tabindex="-1"></a>        categories<span class="op">=</span>[<span class="st">'Flight Blanks'</span>, <span class="st">'Ground Blanks'</span>, <span class="st">'Flight February'</span>, <span class="st">'Ground February'</span>,</span>
<span id="cb180-21"><a aria-hidden="true" href="#cb180-21" tabindex="-1"></a>                    <span class="st">'Flight April'</span>, <span class="st">'Ground April'</span>], ordered<span class="op">=</span><span class="va">True</span>))</span>
<span id="cb180-22"><a aria-hidden="true" href="#cb180-22" tabindex="-1"></a> .pipe(<span class="kw">lambda</span> dd: p9.ggplot(dd)</span>
<span id="cb180-23"><a aria-hidden="true" href="#cb180-23" tabindex="-1"></a>       <span class="op">+</span> p9.aes(<span class="st">'sample_group_label'</span>, <span class="st">'Genus'</span>, size<span class="op">=</span><span class="st">'value'</span>, fill<span class="op">=</span><span class="st">'sample_mode'</span>)</span>
<span id="cb180-24"><a aria-hidden="true" href="#cb180-24" tabindex="-1"></a>       <span class="op">+</span> p9.geom_point(alpha<span class="op">=</span><span class="fl">.8</span>, stroke<span class="op">=</span><span class="dv">0</span>)</span>
<span id="cb180-25"><a aria-hidden="true" href="#cb180-25" tabindex="-1"></a>       <span class="op">+</span> p9.scale_size_continuous(trans<span class="op">=</span><span class="st">'log10'</span>,</span>
<span id="cb180-26"><a aria-hidden="true" href="#cb180-26" tabindex="-1"></a>                                  <span class="bu">range</span><span class="op">=</span>(<span class="dv">1</span>, <span class="dv">9</span>),</span>
<span id="cb180-27"><a aria-hidden="true" href="#cb180-27" tabindex="-1"></a>                                  labels<span class="op">=</span>percent_format(), </span>
<span id="cb180-28"><a aria-hidden="true" href="#cb180-28" tabindex="-1"></a>                                  limits<span class="op">=</span>(treshold <span class="op">-</span> <span class="fl">1e-5</span>, <span class="fl">.9</span>),</span>
<span id="cb180-29"><a aria-hidden="true" href="#cb180-29" tabindex="-1"></a>                                  breaks<span class="op">=</span>[<span class="fl">.001</span>, <span class="fl">.01</span>, <span class="fl">.05</span>, <span class="fl">.5</span>])</span>
<span id="cb180-30"><a aria-hidden="true" href="#cb180-30" tabindex="-1"></a>       <span class="op">+</span> p9.scale_y_discrete(expand<span class="op">=</span>(<span class="fl">.035</span>, <span class="fl">.035</span>))</span>
<span id="cb180-31"><a aria-hidden="true" href="#cb180-31" tabindex="-1"></a>       <span class="op">+</span> p9.scale_fill_manual(values<span class="op">=</span>[<span class="st">'#D3894C'</span>, <span class="st">'#435B97'</span>])</span>
<span id="cb180-32"><a aria-hidden="true" href="#cb180-32" tabindex="-1"></a>       <span class="op">+</span> p9.guides(size<span class="op">=</span>p9.guide_legend(ncol<span class="op">=</span><span class="dv">4</span>))</span>
<span id="cb180-33"><a aria-hidden="true" href="#cb180-33" tabindex="-1"></a>       <span class="op">+</span> p9.labs(x<span class="op">=</span><span class="st">''</span>, y<span class="op">=</span><span class="st">''</span>, size<span class="op">=</span><span class="st">'Relative Abundance'</span>, color<span class="op">=</span><span class="st">'Class'</span>,</span>
<span id="cb180-34"><a aria-hidden="true" href="#cb180-34" tabindex="-1"></a>                 title<span class="op">=</span><span class="st">'Relative Abundance of Top 30 Genera in Samples'</span>,</span>
<span id="cb180-35"><a aria-hidden="true" href="#cb180-35" tabindex="-1"></a>                 fill<span class="op">=</span><span class="st">''</span></span>
<span id="cb180-36"><a aria-hidden="true" href="#cb180-36" tabindex="-1"></a>                 )</span>
<span id="cb180-37"><a aria-hidden="true" href="#cb180-37" tabindex="-1"></a>                  )</span>
<span id="cb180-38"><a aria-hidden="true" href="#cb180-38" tabindex="-1"></a>       <span class="op">+</span> p9.annotate(geom<span class="op">=</span><span class="st">'vline'</span>, linetype<span class="op">=</span><span class="st">'dashed'</span>, xintercept<span class="op">=</span><span class="fl">2.5</span>, color<span class="op">=</span><span class="st">'grey'</span>)</span>
<span id="cb180-39"><a aria-hidden="true" href="#cb180-39" tabindex="-1"></a>       <span class="op">+</span> p9.theme(figure_size<span class="op">=</span>(<span class="fl">3.25</span>, <span class="fl">7.5</span>),</span>
<span id="cb180-40"><a aria-hidden="true" href="#cb180-40" tabindex="-1"></a>                  strip_text<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">12</span>),</span>
<span id="cb180-41"><a aria-hidden="true" href="#cb180-41" tabindex="-1"></a>                  axis_text<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">10</span>),</span>
<span id="cb180-42"><a aria-hidden="true" href="#cb180-42" tabindex="-1"></a>                  legend_position<span class="op">=</span><span class="st">'bottom'</span>,</span>
<span id="cb180-43"><a aria-hidden="true" href="#cb180-43" tabindex="-1"></a>                  title<span class="op">=</span>p9.element_text(ha<span class="op">=</span><span class="st">'center'</span>),</span>
<span id="cb180-44"><a aria-hidden="true" href="#cb180-44" tabindex="-1"></a>                  legend_title<span class="op">=</span>p9.element_text(x<span class="op">=</span><span class="dv">300</span>),</span>
<span id="cb180-45"><a aria-hidden="true" href="#cb180-45" tabindex="-1"></a>                  axis_text_x<span class="op">=</span>p9.element_text(angle<span class="op">=</span><span class="dv">90</span>),</span>
<span id="cb180-46"><a aria-hidden="true" href="#cb180-46" tabindex="-1"></a>       )</span>
<span id="cb180-47"><a aria-hidden="true" href="#cb180-47" tabindex="-1"></a> )</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img class="figure-img" height="2250" src="microbial_diversity_analysis_files/figure-html/cell-169-output-1.png" width="975"/></p>
</figure>
</div>
</div>
</div>
<div class="cell" id="cell-209">
<div class="sourceCode cell-code" id="cb181"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb181-1"><a aria-hidden="true" href="#cb181-1" tabindex="-1"></a>(full_long_df </span>
<span id="cb181-2"><a aria-hidden="true" href="#cb181-2" tabindex="-1"></a> .query(<span class="st">'Genus.notna()'</span>)</span>
<span id="cb181-3"><a aria-hidden="true" href="#cb181-3" tabindex="-1"></a>.assign(Genus<span class="op">=</span><span class="kw">lambda</span> dd: </span>
<span id="cb181-4"><a aria-hidden="true" href="#cb181-4" tabindex="-1"></a>        pd.Categorical(dd.Genus, categories<span class="op">=</span>top_30_genera_blanks, ordered<span class="op">=</span><span class="va">True</span>))</span>
<span id="cb181-5"><a aria-hidden="true" href="#cb181-5" tabindex="-1"></a>.dropna()</span>
<span id="cb181-6"><a aria-hidden="true" href="#cb181-6" tabindex="-1"></a>.replace({<span class="st">'Genus'</span>: {<span class="st">'Malasseziaceae_gen_Incertae_sedis'</span>: <span class="st">'Malasseziaceae'</span>}})</span>
<span id="cb181-7"><a aria-hidden="true" href="#cb181-7" tabindex="-1"></a>.assign(sample_id<span class="op">=</span><span class="kw">lambda</span> dd: dd.sample_label.<span class="bu">str</span>.split(<span class="st">'</span><span class="ch">\n</span><span class="st">'</span>).<span class="bu">str</span>[<span class="dv">0</span>])</span>
<span id="cb181-8"><a aria-hidden="true" href="#cb181-8" tabindex="-1"></a>.merge(samples_info, how<span class="op">=</span><span class="st">'left'</span>)</span>
<span id="cb181-9"><a aria-hidden="true" href="#cb181-9" tabindex="-1"></a>.assign(month<span class="op">=</span><span class="kw">lambda</span> dd: pd.to_datetime(dd.date).dt.month_name().fillna(<span class="st">'Blanks'</span>))</span>
<span id="cb181-10"><a aria-hidden="true" href="#cb181-10" tabindex="-1"></a>.assign(sample_mode<span class="op">=</span><span class="kw">lambda</span> dd: </span>
<span id="cb181-11"><a aria-hidden="true" href="#cb181-11" tabindex="-1"></a>        np.where(dd.sample_mode.isna, </span>
<span id="cb181-12"><a aria-hidden="true" href="#cb181-12" tabindex="-1"></a>                 dd.sample_label.<span class="bu">str</span>.split(<span class="st">'('</span>).<span class="bu">str</span>[<span class="op">-</span><span class="dv">1</span>].<span class="bu">str</span>[:<span class="op">-</span><span class="dv">1</span>], </span>
<span id="cb181-13"><a aria-hidden="true" href="#cb181-13" tabindex="-1"></a>                 dd.sample_mode))</span>
<span id="cb181-14"><a aria-hidden="true" href="#cb181-14" tabindex="-1"></a>.assign(sample_group_label<span class="op">=</span><span class="kw">lambda</span> dd: dd.sample_mode <span class="op">+</span> <span class="st">' '</span> <span class="op">+</span> dd.month)</span>
<span id="cb181-15"><a aria-hidden="true" href="#cb181-15" tabindex="-1"></a>.groupby([<span class="st">'sample_group_label'</span>, <span class="st">'sample_mode'</span>, <span class="st">'Genus'</span>])</span>
<span id="cb181-16"><a aria-hidden="true" href="#cb181-16" tabindex="-1"></a>.aggregate({<span class="st">'value'</span>: <span class="st">'mean'</span>})</span>
<span id="cb181-17"><a aria-hidden="true" href="#cb181-17" tabindex="-1"></a>.reset_index()</span>
<span id="cb181-18"><a aria-hidden="true" href="#cb181-18" tabindex="-1"></a>.query(<span class="st">'value &gt; 0'</span>)</span>
<span id="cb181-19"><a aria-hidden="true" href="#cb181-19" tabindex="-1"></a>.assign(value<span class="op">=</span><span class="kw">lambda</span> dd: np.where(dd.value <span class="op">&lt;</span> treshold, treshold, dd.value))</span>
<span id="cb181-20"><a aria-hidden="true" href="#cb181-20" tabindex="-1"></a>.assign(sample_group_label<span class="op">=</span><span class="kw">lambda</span> dd: pd.Categorical(dd.sample_group_label,</span>
<span id="cb181-21"><a aria-hidden="true" href="#cb181-21" tabindex="-1"></a>        categories<span class="op">=</span>[<span class="st">'Flight Blanks'</span>, <span class="st">'Ground Blanks'</span>, <span class="st">'Flight February'</span>, <span class="st">'Ground February'</span>,</span>
<span id="cb181-22"><a aria-hidden="true" href="#cb181-22" tabindex="-1"></a>                    <span class="st">'Flight April'</span>, <span class="st">'Ground April'</span>], ordered<span class="op">=</span><span class="va">True</span>))</span>
<span id="cb181-23"><a aria-hidden="true" href="#cb181-23" tabindex="-1"></a> .pipe(<span class="kw">lambda</span> dd: p9.ggplot(dd)</span>
<span id="cb181-24"><a aria-hidden="true" href="#cb181-24" tabindex="-1"></a>       <span class="op">+</span> p9.aes(<span class="st">'sample_group_label'</span>, <span class="st">'Genus'</span>, size<span class="op">=</span><span class="st">'value'</span>, fill<span class="op">=</span><span class="st">'sample_mode'</span>)</span>
<span id="cb181-25"><a aria-hidden="true" href="#cb181-25" tabindex="-1"></a>       <span class="op">+</span> p9.geom_point(alpha<span class="op">=</span><span class="fl">.8</span>, stroke<span class="op">=</span><span class="dv">0</span>)</span>
<span id="cb181-26"><a aria-hidden="true" href="#cb181-26" tabindex="-1"></a>       <span class="op">+</span> p9.scale_size_continuous(trans<span class="op">=</span><span class="st">'log10'</span>,</span>
<span id="cb181-27"><a aria-hidden="true" href="#cb181-27" tabindex="-1"></a>                                  <span class="bu">range</span><span class="op">=</span>(<span class="dv">1</span>, <span class="dv">9</span>),</span>
<span id="cb181-28"><a aria-hidden="true" href="#cb181-28" tabindex="-1"></a>                                  labels<span class="op">=</span>percent_format(), </span>
<span id="cb181-29"><a aria-hidden="true" href="#cb181-29" tabindex="-1"></a>                                  limits<span class="op">=</span>(treshold <span class="op">-</span> <span class="fl">1e-5</span>, <span class="fl">.9</span>),</span>
<span id="cb181-30"><a aria-hidden="true" href="#cb181-30" tabindex="-1"></a>                                  breaks<span class="op">=</span>[<span class="fl">.001</span>, <span class="fl">.01</span>, <span class="fl">.05</span>, <span class="fl">.5</span>])</span>
<span id="cb181-31"><a aria-hidden="true" href="#cb181-31" tabindex="-1"></a>       <span class="op">+</span> p9.scale_y_discrete(expand<span class="op">=</span>(<span class="fl">.035</span>, <span class="fl">.035</span>))</span>
<span id="cb181-32"><a aria-hidden="true" href="#cb181-32" tabindex="-1"></a>       <span class="op">+</span> p9.scale_fill_manual(values<span class="op">=</span>[<span class="st">'#D3894C'</span>, <span class="st">'#435B97'</span>])</span>
<span id="cb181-33"><a aria-hidden="true" href="#cb181-33" tabindex="-1"></a>       <span class="op">+</span> p9.guides(size<span class="op">=</span>p9.guide_legend(ncol<span class="op">=</span><span class="dv">4</span>))</span>
<span id="cb181-34"><a aria-hidden="true" href="#cb181-34" tabindex="-1"></a>       <span class="op">+</span> p9.labs(x<span class="op">=</span><span class="st">''</span>, y<span class="op">=</span><span class="st">''</span>, size<span class="op">=</span><span class="st">'Relative Abundance'</span>, color<span class="op">=</span><span class="st">'Class'</span>,</span>
<span id="cb181-35"><a aria-hidden="true" href="#cb181-35" tabindex="-1"></a>                 title<span class="op">=</span><span class="st">'Relative Abundance of Top 30 Genera in Samples'</span>,</span>
<span id="cb181-36"><a aria-hidden="true" href="#cb181-36" tabindex="-1"></a>                 fill<span class="op">=</span><span class="st">''</span></span>
<span id="cb181-37"><a aria-hidden="true" href="#cb181-37" tabindex="-1"></a>                 )</span>
<span id="cb181-38"><a aria-hidden="true" href="#cb181-38" tabindex="-1"></a>                  )</span>
<span id="cb181-39"><a aria-hidden="true" href="#cb181-39" tabindex="-1"></a>       <span class="op">+</span> p9.annotate(geom<span class="op">=</span><span class="st">'vline'</span>, linetype<span class="op">=</span><span class="st">'dashed'</span>, xintercept<span class="op">=</span><span class="fl">2.5</span>, color<span class="op">=</span><span class="st">'grey'</span>)</span>
<span id="cb181-40"><a aria-hidden="true" href="#cb181-40" tabindex="-1"></a>       <span class="op">+</span> p9.theme(figure_size<span class="op">=</span>(<span class="dv">3</span>, <span class="fl">7.5</span>),</span>
<span id="cb181-41"><a aria-hidden="true" href="#cb181-41" tabindex="-1"></a>                  strip_text<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">12</span>),</span>
<span id="cb181-42"><a aria-hidden="true" href="#cb181-42" tabindex="-1"></a>                  axis_text<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">10</span>),</span>
<span id="cb181-43"><a aria-hidden="true" href="#cb181-43" tabindex="-1"></a>                  legend_position<span class="op">=</span><span class="st">'bottom'</span>,</span>
<span id="cb181-44"><a aria-hidden="true" href="#cb181-44" tabindex="-1"></a>                  title<span class="op">=</span>p9.element_text(ha<span class="op">=</span><span class="st">'center'</span>),</span>
<span id="cb181-45"><a aria-hidden="true" href="#cb181-45" tabindex="-1"></a>                  legend_title<span class="op">=</span>p9.element_text(x<span class="op">=</span><span class="dv">300</span>),</span>
<span id="cb181-46"><a aria-hidden="true" href="#cb181-46" tabindex="-1"></a>                  axis_text_x<span class="op">=</span>p9.element_text(angle<span class="op">=</span><span class="dv">90</span>),</span>
<span id="cb181-47"><a aria-hidden="true" href="#cb181-47" tabindex="-1"></a>       )</span>
<span id="cb181-48"><a aria-hidden="true" href="#cb181-48" tabindex="-1"></a> )</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img class="figure-img" height="2250" src="microbial_diversity_analysis_files/figure-html/cell-170-output-1.png" width="900"/></p>
</figure>
</div>
</div>
</div>
<div class="cell" id="cell-210">
<div class="sourceCode cell-code" id="cb182"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb182-1"><a aria-hidden="true" href="#cb182-1" tabindex="-1"></a>full_long_df <span class="op">=</span> (full_matrix_bac</span>
<span id="cb182-2"><a aria-hidden="true" href="#cb182-2" tabindex="-1"></a>                .reset_index()</span>
<span id="cb182-3"><a aria-hidden="true" href="#cb182-3" tabindex="-1"></a>                .melt(<span class="st">'sample_label'</span>, var_name<span class="op">=</span><span class="st">'Genus'</span>, value_name<span class="op">=</span><span class="st">'value'</span>)</span>
<span id="cb182-4"><a aria-hidden="true" href="#cb182-4" tabindex="-1"></a>)</span>
<span id="cb182-5"><a aria-hidden="true" href="#cb182-5" tabindex="-1"></a></span>
<span id="cb182-6"><a aria-hidden="true" href="#cb182-6" tabindex="-1"></a></span>
<span id="cb182-7"><a aria-hidden="true" href="#cb182-7" tabindex="-1"></a>top_30_genera_blanks <span class="op">=</span> (</span>
<span id="cb182-8"><a aria-hidden="true" href="#cb182-8" tabindex="-1"></a>    full_long_df</span>
<span id="cb182-9"><a aria-hidden="true" href="#cb182-9" tabindex="-1"></a>    .query(<span class="st">'Genus!="unknown"'</span>)</span>
<span id="cb182-10"><a aria-hidden="true" href="#cb182-10" tabindex="-1"></a>    .query(<span class="st">'sample_label.str.contains("Pool")'</span>)</span>
<span id="cb182-11"><a aria-hidden="true" href="#cb182-11" tabindex="-1"></a>    .groupby(<span class="st">'Genus'</span>)</span>
<span id="cb182-12"><a aria-hidden="true" href="#cb182-12" tabindex="-1"></a>    [<span class="st">'value'</span>]</span>
<span id="cb182-13"><a aria-hidden="true" href="#cb182-13" tabindex="-1"></a>    .<span class="bu">sum</span>()</span>
<span id="cb182-14"><a aria-hidden="true" href="#cb182-14" tabindex="-1"></a>    .sort_values(ascending<span class="op">=</span><span class="va">True</span>)</span>
<span id="cb182-15"><a aria-hidden="true" href="#cb182-15" tabindex="-1"></a>    .tail(<span class="dv">30</span>)</span>
<span id="cb182-16"><a aria-hidden="true" href="#cb182-16" tabindex="-1"></a>    .index</span>
<span id="cb182-17"><a aria-hidden="true" href="#cb182-17" tabindex="-1"></a>    .values</span>
<span id="cb182-18"><a aria-hidden="true" href="#cb182-18" tabindex="-1"></a>)</span>
<span id="cb182-19"><a aria-hidden="true" href="#cb182-19" tabindex="-1"></a></span>
<span id="cb182-20"><a aria-hidden="true" href="#cb182-20" tabindex="-1"></a>top_30_genera_samples_april <span class="op">=</span> (</span>
<span id="cb182-21"><a aria-hidden="true" href="#cb182-21" tabindex="-1"></a>    full_long_df</span>
<span id="cb182-22"><a aria-hidden="true" href="#cb182-22" tabindex="-1"></a>    .query(<span class="st">'Genus!="unknown"'</span>)</span>
<span id="cb182-23"><a aria-hidden="true" href="#cb182-23" tabindex="-1"></a>    .query(<span class="st">'sample_label.str.startswith("S")'</span>)</span>
<span id="cb182-24"><a aria-hidden="true" href="#cb182-24" tabindex="-1"></a>    .assign(sample_id<span class="op">=</span><span class="kw">lambda</span> dd: dd.sample_label.<span class="bu">str</span>.split(<span class="st">'</span><span class="ch">\n</span><span class="st">'</span>).<span class="bu">str</span>[<span class="dv">0</span>])</span>
<span id="cb182-25"><a aria-hidden="true" href="#cb182-25" tabindex="-1"></a>    .merge(samples_info, how<span class="op">=</span><span class="st">'left'</span>)</span>
<span id="cb182-26"><a aria-hidden="true" href="#cb182-26" tabindex="-1"></a>    .assign(month<span class="op">=</span><span class="kw">lambda</span> dd: pd.to_datetime(dd.date).dt.month_name().fillna(<span class="st">'Blanks'</span>))</span>
<span id="cb182-27"><a aria-hidden="true" href="#cb182-27" tabindex="-1"></a>    .query(<span class="st">'month=="April"'</span>)</span>
<span id="cb182-28"><a aria-hidden="true" href="#cb182-28" tabindex="-1"></a>    .groupby(<span class="st">'Genus'</span>)</span>
<span id="cb182-29"><a aria-hidden="true" href="#cb182-29" tabindex="-1"></a>    [<span class="st">'value'</span>]</span>
<span id="cb182-30"><a aria-hidden="true" href="#cb182-30" tabindex="-1"></a>    .<span class="bu">sum</span>()</span>
<span id="cb182-31"><a aria-hidden="true" href="#cb182-31" tabindex="-1"></a>    .sort_values(ascending<span class="op">=</span><span class="va">True</span>)</span>
<span id="cb182-32"><a aria-hidden="true" href="#cb182-32" tabindex="-1"></a>    .tail(<span class="dv">30</span>)</span>
<span id="cb182-33"><a aria-hidden="true" href="#cb182-33" tabindex="-1"></a>    .index</span>
<span id="cb182-34"><a aria-hidden="true" href="#cb182-34" tabindex="-1"></a>    .values</span>
<span id="cb182-35"><a aria-hidden="true" href="#cb182-35" tabindex="-1"></a>)</span>
<span id="cb182-36"><a aria-hidden="true" href="#cb182-36" tabindex="-1"></a></span>
<span id="cb182-37"><a aria-hidden="true" href="#cb182-37" tabindex="-1"></a>top_30_genera_samples_february <span class="op">=</span> (</span>
<span id="cb182-38"><a aria-hidden="true" href="#cb182-38" tabindex="-1"></a>    full_long_df</span>
<span id="cb182-39"><a aria-hidden="true" href="#cb182-39" tabindex="-1"></a>    .query(<span class="st">'Genus!="unknown"'</span>)</span>
<span id="cb182-40"><a aria-hidden="true" href="#cb182-40" tabindex="-1"></a>    .query(<span class="st">'sample_label.str.startswith("S")'</span>)</span>
<span id="cb182-41"><a aria-hidden="true" href="#cb182-41" tabindex="-1"></a>    .assign(sample_id<span class="op">=</span><span class="kw">lambda</span> dd: dd.sample_label.<span class="bu">str</span>.split(<span class="st">'</span><span class="ch">\n</span><span class="st">'</span>).<span class="bu">str</span>[<span class="dv">0</span>])</span>
<span id="cb182-42"><a aria-hidden="true" href="#cb182-42" tabindex="-1"></a>    .merge(samples_info, how<span class="op">=</span><span class="st">'left'</span>)</span>
<span id="cb182-43"><a aria-hidden="true" href="#cb182-43" tabindex="-1"></a>    .assign(month<span class="op">=</span><span class="kw">lambda</span> dd: pd.to_datetime(dd.date).dt.month_name().fillna(<span class="st">'Blanks'</span>))</span>
<span id="cb182-44"><a aria-hidden="true" href="#cb182-44" tabindex="-1"></a>    .query(<span class="st">'month=="February"'</span>)</span>
<span id="cb182-45"><a aria-hidden="true" href="#cb182-45" tabindex="-1"></a>    .groupby(<span class="st">'Genus'</span>)</span>
<span id="cb182-46"><a aria-hidden="true" href="#cb182-46" tabindex="-1"></a>    [<span class="st">'value'</span>]</span>
<span id="cb182-47"><a aria-hidden="true" href="#cb182-47" tabindex="-1"></a>    .<span class="bu">sum</span>()</span>
<span id="cb182-48"><a aria-hidden="true" href="#cb182-48" tabindex="-1"></a>    .sort_values(ascending<span class="op">=</span><span class="va">True</span>)</span>
<span id="cb182-49"><a aria-hidden="true" href="#cb182-49" tabindex="-1"></a>    .tail(<span class="dv">30</span>)</span>
<span id="cb182-50"><a aria-hidden="true" href="#cb182-50" tabindex="-1"></a>    .index</span>
<span id="cb182-51"><a aria-hidden="true" href="#cb182-51" tabindex="-1"></a>    .values</span>
<span id="cb182-52"><a aria-hidden="true" href="#cb182-52" tabindex="-1"></a>)</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" id="cell-211">
<div class="sourceCode cell-code" id="cb183"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb183-1"><a aria-hidden="true" href="#cb183-1" tabindex="-1"></a>f <span class="op">=</span> (full_long_df </span>
<span id="cb183-2"><a aria-hidden="true" href="#cb183-2" tabindex="-1"></a> .query(<span class="st">'Genus.notna()'</span>)</span>
<span id="cb183-3"><a aria-hidden="true" href="#cb183-3" tabindex="-1"></a>.assign(Genus<span class="op">=</span><span class="kw">lambda</span> dd: </span>
<span id="cb183-4"><a aria-hidden="true" href="#cb183-4" tabindex="-1"></a>        pd.Categorical(dd.Genus, categories<span class="op">=</span>top_30_genera_samples_february, ordered<span class="op">=</span><span class="va">True</span>))</span>
<span id="cb183-5"><a aria-hidden="true" href="#cb183-5" tabindex="-1"></a>.dropna()</span>
<span id="cb183-6"><a aria-hidden="true" href="#cb183-6" tabindex="-1"></a>.assign(sample_id<span class="op">=</span><span class="kw">lambda</span> dd: dd.sample_label.<span class="bu">str</span>.split(<span class="st">'</span><span class="ch">\n</span><span class="st">'</span>).<span class="bu">str</span>[<span class="dv">0</span>])</span>
<span id="cb183-7"><a aria-hidden="true" href="#cb183-7" tabindex="-1"></a>.merge(samples_info, how<span class="op">=</span><span class="st">'left'</span>)</span>
<span id="cb183-8"><a aria-hidden="true" href="#cb183-8" tabindex="-1"></a>.assign(month<span class="op">=</span><span class="kw">lambda</span> dd: pd.to_datetime(dd.date).dt.month_name().fillna(<span class="st">'Blanks'</span>))</span>
<span id="cb183-9"><a aria-hidden="true" href="#cb183-9" tabindex="-1"></a>.assign(sample_mode<span class="op">=</span><span class="kw">lambda</span> dd: </span>
<span id="cb183-10"><a aria-hidden="true" href="#cb183-10" tabindex="-1"></a>        np.where(dd.sample_mode.isna, </span>
<span id="cb183-11"><a aria-hidden="true" href="#cb183-11" tabindex="-1"></a>                 dd.sample_label.<span class="bu">str</span>.split(<span class="st">'('</span>).<span class="bu">str</span>[<span class="op">-</span><span class="dv">1</span>].<span class="bu">str</span>[:<span class="op">-</span><span class="dv">1</span>], </span>
<span id="cb183-12"><a aria-hidden="true" href="#cb183-12" tabindex="-1"></a>                 dd.sample_mode))</span>
<span id="cb183-13"><a aria-hidden="true" href="#cb183-13" tabindex="-1"></a>.assign(sample_group_label<span class="op">=</span><span class="kw">lambda</span> dd: dd.sample_mode <span class="op">+</span> <span class="st">' '</span> <span class="op">+</span> dd.month)</span>
<span id="cb183-14"><a aria-hidden="true" href="#cb183-14" tabindex="-1"></a>.groupby([<span class="st">'sample_group_label'</span>, <span class="st">'sample_mode'</span>, <span class="st">'Genus'</span>])</span>
<span id="cb183-15"><a aria-hidden="true" href="#cb183-15" tabindex="-1"></a>.aggregate({<span class="st">'value'</span>: <span class="st">'mean'</span>})</span>
<span id="cb183-16"><a aria-hidden="true" href="#cb183-16" tabindex="-1"></a>.reset_index()</span>
<span id="cb183-17"><a aria-hidden="true" href="#cb183-17" tabindex="-1"></a>.query(<span class="st">'value &gt; 0'</span>)</span>
<span id="cb183-18"><a aria-hidden="true" href="#cb183-18" tabindex="-1"></a>.assign(value<span class="op">=</span><span class="kw">lambda</span> dd: np.where(dd.value <span class="op">&lt;</span> treshold, treshold, dd.value))</span>
<span id="cb183-19"><a aria-hidden="true" href="#cb183-19" tabindex="-1"></a>.assign(sample_group_label<span class="op">=</span><span class="kw">lambda</span> dd: pd.Categorical(dd.sample_group_label,</span>
<span id="cb183-20"><a aria-hidden="true" href="#cb183-20" tabindex="-1"></a>        categories<span class="op">=</span>[<span class="st">'Flight Blanks'</span>, <span class="st">'Ground Blanks'</span>, <span class="st">'Flight February'</span>, <span class="st">'Ground February'</span>,</span>
<span id="cb183-21"><a aria-hidden="true" href="#cb183-21" tabindex="-1"></a>                    <span class="st">'Flight April'</span>, <span class="st">'Ground April'</span>], ordered<span class="op">=</span><span class="va">True</span>))</span>
<span id="cb183-22"><a aria-hidden="true" href="#cb183-22" tabindex="-1"></a> .pipe(<span class="kw">lambda</span> dd: p9.ggplot(dd)</span>
<span id="cb183-23"><a aria-hidden="true" href="#cb183-23" tabindex="-1"></a>       <span class="op">+</span> p9.aes(<span class="st">'sample_group_label'</span>, <span class="st">'Genus'</span>, size<span class="op">=</span><span class="st">'value'</span>, fill<span class="op">=</span><span class="st">'sample_mode'</span>)</span>
<span id="cb183-24"><a aria-hidden="true" href="#cb183-24" tabindex="-1"></a>       <span class="op">+</span> p9.geom_point(alpha<span class="op">=</span><span class="fl">.8</span>, stroke<span class="op">=</span><span class="dv">0</span>)</span>
<span id="cb183-25"><a aria-hidden="true" href="#cb183-25" tabindex="-1"></a>       <span class="op">+</span> p9.scale_size_continuous(trans<span class="op">=</span><span class="st">'log10'</span>,</span>
<span id="cb183-26"><a aria-hidden="true" href="#cb183-26" tabindex="-1"></a>                                  <span class="bu">range</span><span class="op">=</span>(<span class="dv">1</span>, <span class="dv">9</span>),</span>
<span id="cb183-27"><a aria-hidden="true" href="#cb183-27" tabindex="-1"></a>                                  labels<span class="op">=</span>percent_format(), </span>
<span id="cb183-28"><a aria-hidden="true" href="#cb183-28" tabindex="-1"></a>                                  limits<span class="op">=</span>(treshold <span class="op">-</span> <span class="fl">1e-5</span>, <span class="fl">.9</span>),</span>
<span id="cb183-29"><a aria-hidden="true" href="#cb183-29" tabindex="-1"></a>                                  breaks<span class="op">=</span>[<span class="fl">.001</span>, <span class="fl">.01</span>, <span class="fl">.05</span>, <span class="fl">.5</span>])</span>
<span id="cb183-30"><a aria-hidden="true" href="#cb183-30" tabindex="-1"></a>       <span class="op">+</span> p9.scale_y_discrete(expand<span class="op">=</span>(<span class="fl">.035</span>, <span class="fl">.035</span>))</span>
<span id="cb183-31"><a aria-hidden="true" href="#cb183-31" tabindex="-1"></a>       <span class="op">+</span> p9.scale_fill_manual(values<span class="op">=</span>[<span class="st">'#D3894C'</span>, <span class="st">'#435B97'</span>])</span>
<span id="cb183-32"><a aria-hidden="true" href="#cb183-32" tabindex="-1"></a>       <span class="op">+</span> p9.guides(size<span class="op">=</span><span class="va">False</span>)</span>
<span id="cb183-33"><a aria-hidden="true" href="#cb183-33" tabindex="-1"></a>       <span class="op">+</span> p9.labs(x<span class="op">=</span><span class="st">''</span>, y<span class="op">=</span><span class="st">''</span>, size<span class="op">=</span><span class="st">'Relative Abundance'</span>, color<span class="op">=</span><span class="st">'Class'</span>,</span>
<span id="cb183-34"><a aria-hidden="true" href="#cb183-34" tabindex="-1"></a>                 title<span class="op">=</span><span class="st">''</span>,</span>
<span id="cb183-35"><a aria-hidden="true" href="#cb183-35" tabindex="-1"></a>                 fill<span class="op">=</span><span class="st">''</span></span>
<span id="cb183-36"><a aria-hidden="true" href="#cb183-36" tabindex="-1"></a>                 )</span>
<span id="cb183-37"><a aria-hidden="true" href="#cb183-37" tabindex="-1"></a>                  )</span>
<span id="cb183-38"><a aria-hidden="true" href="#cb183-38" tabindex="-1"></a>       <span class="op">+</span> p9.annotate(geom<span class="op">=</span><span class="st">'vline'</span>, linetype<span class="op">=</span><span class="st">'dashed'</span>, xintercept<span class="op">=</span><span class="fl">2.5</span>, color<span class="op">=</span><span class="st">'grey'</span>)</span>
<span id="cb183-39"><a aria-hidden="true" href="#cb183-39" tabindex="-1"></a>       <span class="op">+</span> p9.theme(figure_size<span class="op">=</span>(<span class="dv">3</span>, <span class="fl">7.5</span>),</span>
<span id="cb183-40"><a aria-hidden="true" href="#cb183-40" tabindex="-1"></a>                  strip_text<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">12</span>),</span>
<span id="cb183-41"><a aria-hidden="true" href="#cb183-41" tabindex="-1"></a>                  axis_text<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">10</span>),</span>
<span id="cb183-42"><a aria-hidden="true" href="#cb183-42" tabindex="-1"></a>                  legend_position<span class="op">=</span><span class="st">'bottom'</span>,</span>
<span id="cb183-43"><a aria-hidden="true" href="#cb183-43" tabindex="-1"></a>                  title<span class="op">=</span>p9.element_text(ha<span class="op">=</span><span class="st">'center'</span>),</span>
<span id="cb183-44"><a aria-hidden="true" href="#cb183-44" tabindex="-1"></a>                  legend_title<span class="op">=</span>p9.element_text(x<span class="op">=</span><span class="dv">300</span>),</span>
<span id="cb183-45"><a aria-hidden="true" href="#cb183-45" tabindex="-1"></a>                  axis_text_x<span class="op">=</span>p9.element_text(angle<span class="op">=</span><span class="dv">90</span>),</span>
<span id="cb183-46"><a aria-hidden="true" href="#cb183-46" tabindex="-1"></a>       )</span>
<span id="cb183-47"><a aria-hidden="true" href="#cb183-47" tabindex="-1"></a> )</span>
<span id="cb183-48"><a aria-hidden="true" href="#cb183-48" tabindex="-1"></a></span>
<span id="cb183-49"><a aria-hidden="true" href="#cb183-49" tabindex="-1"></a>f.save(<span class="st">'../output/figures/feb_top_30_genera.svg'</span>)</span>
<span id="cb183-50"><a aria-hidden="true" href="#cb183-50" tabindex="-1"></a>f.draw()</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="852">
<div>
<figure class="figure">
<p><img class="figure-img" height="2250" src="microbial_diversity_analysis_files/figure-html/cell-172-output-1.png" width="900"/></p>
</figure>
</div>
</div>
</div>
<div class="cell" id="cell-212">
<div class="sourceCode cell-code" id="cb184"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb184-1"><a aria-hidden="true" href="#cb184-1" tabindex="-1"></a>f <span class="op">=</span> (full_long_df </span>
<span id="cb184-2"><a aria-hidden="true" href="#cb184-2" tabindex="-1"></a> .query(<span class="st">'Genus.notna()'</span>)</span>
<span id="cb184-3"><a aria-hidden="true" href="#cb184-3" tabindex="-1"></a>.assign(Genus<span class="op">=</span><span class="kw">lambda</span> dd: </span>
<span id="cb184-4"><a aria-hidden="true" href="#cb184-4" tabindex="-1"></a>        pd.Categorical(dd.Genus, categories<span class="op">=</span>top_30_genera_samples_april, ordered<span class="op">=</span><span class="va">True</span>))</span>
<span id="cb184-5"><a aria-hidden="true" href="#cb184-5" tabindex="-1"></a>.dropna()</span>
<span id="cb184-6"><a aria-hidden="true" href="#cb184-6" tabindex="-1"></a>.assign(sample_id<span class="op">=</span><span class="kw">lambda</span> dd: dd.sample_label.<span class="bu">str</span>.split(<span class="st">'</span><span class="ch">\n</span><span class="st">'</span>).<span class="bu">str</span>[<span class="dv">0</span>])</span>
<span id="cb184-7"><a aria-hidden="true" href="#cb184-7" tabindex="-1"></a>.merge(samples_info, how<span class="op">=</span><span class="st">'left'</span>)</span>
<span id="cb184-8"><a aria-hidden="true" href="#cb184-8" tabindex="-1"></a>.assign(month<span class="op">=</span><span class="kw">lambda</span> dd: pd.to_datetime(dd.date).dt.month_name().fillna(<span class="st">'Blanks'</span>))</span>
<span id="cb184-9"><a aria-hidden="true" href="#cb184-9" tabindex="-1"></a>.assign(sample_mode<span class="op">=</span><span class="kw">lambda</span> dd: </span>
<span id="cb184-10"><a aria-hidden="true" href="#cb184-10" tabindex="-1"></a>        np.where(dd.sample_mode.isna, </span>
<span id="cb184-11"><a aria-hidden="true" href="#cb184-11" tabindex="-1"></a>                 dd.sample_label.<span class="bu">str</span>.split(<span class="st">'('</span>).<span class="bu">str</span>[<span class="op">-</span><span class="dv">1</span>].<span class="bu">str</span>[:<span class="op">-</span><span class="dv">1</span>], </span>
<span id="cb184-12"><a aria-hidden="true" href="#cb184-12" tabindex="-1"></a>                 dd.sample_mode))</span>
<span id="cb184-13"><a aria-hidden="true" href="#cb184-13" tabindex="-1"></a>.assign(sample_group_label<span class="op">=</span><span class="kw">lambda</span> dd: dd.sample_mode <span class="op">+</span> <span class="st">' '</span> <span class="op">+</span> dd.month)</span>
<span id="cb184-14"><a aria-hidden="true" href="#cb184-14" tabindex="-1"></a>.groupby([<span class="st">'sample_group_label'</span>, <span class="st">'sample_mode'</span>, <span class="st">'Genus'</span>])</span>
<span id="cb184-15"><a aria-hidden="true" href="#cb184-15" tabindex="-1"></a>.aggregate({<span class="st">'value'</span>: <span class="st">'mean'</span>})</span>
<span id="cb184-16"><a aria-hidden="true" href="#cb184-16" tabindex="-1"></a>.reset_index()</span>
<span id="cb184-17"><a aria-hidden="true" href="#cb184-17" tabindex="-1"></a>.query(<span class="st">'value &gt; 0'</span>)</span>
<span id="cb184-18"><a aria-hidden="true" href="#cb184-18" tabindex="-1"></a>.assign(value<span class="op">=</span><span class="kw">lambda</span> dd: np.where(dd.value <span class="op">&lt;</span> treshold, treshold, dd.value))</span>
<span id="cb184-19"><a aria-hidden="true" href="#cb184-19" tabindex="-1"></a>.assign(sample_group_label<span class="op">=</span><span class="kw">lambda</span> dd: pd.Categorical(dd.sample_group_label,</span>
<span id="cb184-20"><a aria-hidden="true" href="#cb184-20" tabindex="-1"></a>        categories<span class="op">=</span>[<span class="st">'Flight Blanks'</span>, <span class="st">'Ground Blanks'</span>, <span class="st">'Flight February'</span>, <span class="st">'Ground February'</span>,</span>
<span id="cb184-21"><a aria-hidden="true" href="#cb184-21" tabindex="-1"></a>                    <span class="st">'Flight April'</span>, <span class="st">'Ground April'</span>], ordered<span class="op">=</span><span class="va">True</span>))</span>
<span id="cb184-22"><a aria-hidden="true" href="#cb184-22" tabindex="-1"></a> .pipe(<span class="kw">lambda</span> dd: p9.ggplot(dd)</span>
<span id="cb184-23"><a aria-hidden="true" href="#cb184-23" tabindex="-1"></a>       <span class="op">+</span> p9.aes(<span class="st">'sample_group_label'</span>, <span class="st">'Genus'</span>, size<span class="op">=</span><span class="st">'value'</span>, fill<span class="op">=</span><span class="st">'sample_mode'</span>)</span>
<span id="cb184-24"><a aria-hidden="true" href="#cb184-24" tabindex="-1"></a>       <span class="op">+</span> p9.geom_point(alpha<span class="op">=</span><span class="fl">.8</span>, stroke<span class="op">=</span><span class="dv">0</span>)</span>
<span id="cb184-25"><a aria-hidden="true" href="#cb184-25" tabindex="-1"></a>       <span class="op">+</span> p9.scale_size_continuous(trans<span class="op">=</span><span class="st">'log10'</span>,</span>
<span id="cb184-26"><a aria-hidden="true" href="#cb184-26" tabindex="-1"></a>                                  <span class="bu">range</span><span class="op">=</span>(<span class="dv">1</span>, <span class="dv">9</span>),</span>
<span id="cb184-27"><a aria-hidden="true" href="#cb184-27" tabindex="-1"></a>                                  labels<span class="op">=</span>percent_format(), </span>
<span id="cb184-28"><a aria-hidden="true" href="#cb184-28" tabindex="-1"></a>                                  limits<span class="op">=</span>(treshold <span class="op">-</span> <span class="fl">1e-5</span>, <span class="fl">.9</span>),</span>
<span id="cb184-29"><a aria-hidden="true" href="#cb184-29" tabindex="-1"></a>                                  breaks<span class="op">=</span>[<span class="fl">.001</span>, <span class="fl">.01</span>, <span class="fl">.05</span>, <span class="fl">.5</span>])</span>
<span id="cb184-30"><a aria-hidden="true" href="#cb184-30" tabindex="-1"></a>       <span class="op">+</span> p9.scale_y_discrete(expand<span class="op">=</span>(<span class="fl">.035</span>, <span class="fl">.035</span>))</span>
<span id="cb184-31"><a aria-hidden="true" href="#cb184-31" tabindex="-1"></a>       <span class="op">+</span> p9.scale_fill_manual(values<span class="op">=</span>[<span class="st">'#D3894C'</span>, <span class="st">'#435B97'</span>])</span>
<span id="cb184-32"><a aria-hidden="true" href="#cb184-32" tabindex="-1"></a>       <span class="op">+</span> p9.guides(size<span class="op">=</span><span class="va">False</span>)</span>
<span id="cb184-33"><a aria-hidden="true" href="#cb184-33" tabindex="-1"></a>       <span class="op">+</span> p9.labs(x<span class="op">=</span><span class="st">''</span>, y<span class="op">=</span><span class="st">''</span>, size<span class="op">=</span><span class="st">'Relative Abundance'</span>, color<span class="op">=</span><span class="st">'Class'</span>,</span>
<span id="cb184-34"><a aria-hidden="true" href="#cb184-34" tabindex="-1"></a>                 title<span class="op">=</span><span class="st">''</span>,</span>
<span id="cb184-35"><a aria-hidden="true" href="#cb184-35" tabindex="-1"></a>                 fill<span class="op">=</span><span class="st">''</span></span>
<span id="cb184-36"><a aria-hidden="true" href="#cb184-36" tabindex="-1"></a>                 )</span>
<span id="cb184-37"><a aria-hidden="true" href="#cb184-37" tabindex="-1"></a>                  )</span>
<span id="cb184-38"><a aria-hidden="true" href="#cb184-38" tabindex="-1"></a>       <span class="op">+</span> p9.annotate(geom<span class="op">=</span><span class="st">'vline'</span>, linetype<span class="op">=</span><span class="st">'dashed'</span>, xintercept<span class="op">=</span><span class="fl">2.5</span>, color<span class="op">=</span><span class="st">'grey'</span>)</span>
<span id="cb184-39"><a aria-hidden="true" href="#cb184-39" tabindex="-1"></a>       <span class="op">+</span> p9.theme(figure_size<span class="op">=</span>(<span class="dv">5</span>, <span class="fl">7.5</span>),</span>
<span id="cb184-40"><a aria-hidden="true" href="#cb184-40" tabindex="-1"></a>                  strip_text<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">12</span>),</span>
<span id="cb184-41"><a aria-hidden="true" href="#cb184-41" tabindex="-1"></a>                  axis_text<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">10</span>),</span>
<span id="cb184-42"><a aria-hidden="true" href="#cb184-42" tabindex="-1"></a>                  legend_position<span class="op">=</span><span class="st">'bottom'</span>,</span>
<span id="cb184-43"><a aria-hidden="true" href="#cb184-43" tabindex="-1"></a>                  title<span class="op">=</span>p9.element_text(ha<span class="op">=</span><span class="st">'center'</span>),</span>
<span id="cb184-44"><a aria-hidden="true" href="#cb184-44" tabindex="-1"></a>                  legend_title<span class="op">=</span>p9.element_text(x<span class="op">=</span><span class="dv">300</span>),</span>
<span id="cb184-45"><a aria-hidden="true" href="#cb184-45" tabindex="-1"></a>                  axis_text_x<span class="op">=</span>p9.element_text(angle<span class="op">=</span><span class="dv">90</span>),</span>
<span id="cb184-46"><a aria-hidden="true" href="#cb184-46" tabindex="-1"></a>       )</span>
<span id="cb184-47"><a aria-hidden="true" href="#cb184-47" tabindex="-1"></a> )</span>
<span id="cb184-48"><a aria-hidden="true" href="#cb184-48" tabindex="-1"></a>f.save(<span class="st">'../output/figures/apr_top_30_genera.svg'</span>)</span>
<span id="cb184-49"><a aria-hidden="true" href="#cb184-49" tabindex="-1"></a>f.draw()</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="856">
<div>
<figure class="figure">
<p><img class="figure-img" height="2250" src="microbial_diversity_analysis_files/figure-html/cell-173-output-1.png" width="1500"/></p>
</figure>
</div>
</div>
</div>
<div class="cell" id="cell-213">
<div class="sourceCode cell-code" id="cb185"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb185-1"><a aria-hidden="true" href="#cb185-1" tabindex="-1"></a>f <span class="op">=</span> (full_long_df </span>
<span id="cb185-2"><a aria-hidden="true" href="#cb185-2" tabindex="-1"></a> .query(<span class="st">'Genus.notna()'</span>)</span>
<span id="cb185-3"><a aria-hidden="true" href="#cb185-3" tabindex="-1"></a>.assign(Genus<span class="op">=</span><span class="kw">lambda</span> dd: </span>
<span id="cb185-4"><a aria-hidden="true" href="#cb185-4" tabindex="-1"></a>        pd.Categorical(dd.Genus, categories<span class="op">=</span>top_30_genera_blanks, ordered<span class="op">=</span><span class="va">True</span>))</span>
<span id="cb185-5"><a aria-hidden="true" href="#cb185-5" tabindex="-1"></a>.dropna()</span>
<span id="cb185-6"><a aria-hidden="true" href="#cb185-6" tabindex="-1"></a>.assign(sample_id<span class="op">=</span><span class="kw">lambda</span> dd: dd.sample_label.<span class="bu">str</span>.split(<span class="st">'</span><span class="ch">\n</span><span class="st">'</span>).<span class="bu">str</span>[<span class="dv">0</span>])</span>
<span id="cb185-7"><a aria-hidden="true" href="#cb185-7" tabindex="-1"></a>.merge(samples_info, how<span class="op">=</span><span class="st">'left'</span>)</span>
<span id="cb185-8"><a aria-hidden="true" href="#cb185-8" tabindex="-1"></a>.assign(month<span class="op">=</span><span class="kw">lambda</span> dd: pd.to_datetime(dd.date).dt.month_name().fillna(<span class="st">'Blanks'</span>))</span>
<span id="cb185-9"><a aria-hidden="true" href="#cb185-9" tabindex="-1"></a>.assign(sample_mode<span class="op">=</span><span class="kw">lambda</span> dd: </span>
<span id="cb185-10"><a aria-hidden="true" href="#cb185-10" tabindex="-1"></a>        np.where(dd.sample_mode.isna, </span>
<span id="cb185-11"><a aria-hidden="true" href="#cb185-11" tabindex="-1"></a>                 dd.sample_label.<span class="bu">str</span>.split(<span class="st">'('</span>).<span class="bu">str</span>[<span class="op">-</span><span class="dv">1</span>].<span class="bu">str</span>[:<span class="op">-</span><span class="dv">1</span>], </span>
<span id="cb185-12"><a aria-hidden="true" href="#cb185-12" tabindex="-1"></a>                 dd.sample_mode))</span>
<span id="cb185-13"><a aria-hidden="true" href="#cb185-13" tabindex="-1"></a>.assign(sample_group_label<span class="op">=</span><span class="kw">lambda</span> dd: dd.sample_mode <span class="op">+</span> <span class="st">' '</span> <span class="op">+</span> dd.month)</span>
<span id="cb185-14"><a aria-hidden="true" href="#cb185-14" tabindex="-1"></a>.groupby([<span class="st">'sample_group_label'</span>, <span class="st">'sample_mode'</span>, <span class="st">'Genus'</span>])</span>
<span id="cb185-15"><a aria-hidden="true" href="#cb185-15" tabindex="-1"></a>.aggregate({<span class="st">'value'</span>: <span class="st">'mean'</span>})</span>
<span id="cb185-16"><a aria-hidden="true" href="#cb185-16" tabindex="-1"></a>.reset_index()</span>
<span id="cb185-17"><a aria-hidden="true" href="#cb185-17" tabindex="-1"></a>.query(<span class="st">'value &gt; 0'</span>)</span>
<span id="cb185-18"><a aria-hidden="true" href="#cb185-18" tabindex="-1"></a>.assign(value<span class="op">=</span><span class="kw">lambda</span> dd: np.where(dd.value <span class="op">&lt;</span> treshold, treshold, dd.value))</span>
<span id="cb185-19"><a aria-hidden="true" href="#cb185-19" tabindex="-1"></a>.assign(sample_group_label<span class="op">=</span><span class="kw">lambda</span> dd: pd.Categorical(dd.sample_group_label,</span>
<span id="cb185-20"><a aria-hidden="true" href="#cb185-20" tabindex="-1"></a>        categories<span class="op">=</span>[<span class="st">'Flight Blanks'</span>, <span class="st">'Ground Blanks'</span>, <span class="st">'Flight February'</span>, <span class="st">'Ground February'</span>,</span>
<span id="cb185-21"><a aria-hidden="true" href="#cb185-21" tabindex="-1"></a>                    <span class="st">'Flight April'</span>, <span class="st">'Ground April'</span>], ordered<span class="op">=</span><span class="va">True</span>))</span>
<span id="cb185-22"><a aria-hidden="true" href="#cb185-22" tabindex="-1"></a> .pipe(<span class="kw">lambda</span> dd: p9.ggplot(dd)</span>
<span id="cb185-23"><a aria-hidden="true" href="#cb185-23" tabindex="-1"></a>       <span class="op">+</span> p9.aes(<span class="st">'sample_group_label'</span>, <span class="st">'Genus'</span>, size<span class="op">=</span><span class="st">'value'</span>, fill<span class="op">=</span><span class="st">'sample_mode'</span>)</span>
<span id="cb185-24"><a aria-hidden="true" href="#cb185-24" tabindex="-1"></a>       <span class="op">+</span> p9.geom_point(alpha<span class="op">=</span><span class="fl">.8</span>, stroke<span class="op">=</span><span class="dv">0</span>)</span>
<span id="cb185-25"><a aria-hidden="true" href="#cb185-25" tabindex="-1"></a>       <span class="op">+</span> p9.scale_size_continuous(trans<span class="op">=</span><span class="st">'log10'</span>,</span>
<span id="cb185-26"><a aria-hidden="true" href="#cb185-26" tabindex="-1"></a>                                  <span class="bu">range</span><span class="op">=</span>(<span class="dv">1</span>, <span class="dv">9</span>),</span>
<span id="cb185-27"><a aria-hidden="true" href="#cb185-27" tabindex="-1"></a>                                  labels<span class="op">=</span>percent_format(), </span>
<span id="cb185-28"><a aria-hidden="true" href="#cb185-28" tabindex="-1"></a>                                  limits<span class="op">=</span>(treshold <span class="op">-</span> <span class="fl">1e-5</span>, <span class="fl">.9</span>),</span>
<span id="cb185-29"><a aria-hidden="true" href="#cb185-29" tabindex="-1"></a>                                  breaks<span class="op">=</span>[<span class="fl">.001</span>, <span class="fl">.01</span>, <span class="fl">.05</span>, <span class="fl">.5</span>])</span>
<span id="cb185-30"><a aria-hidden="true" href="#cb185-30" tabindex="-1"></a>       <span class="op">+</span> p9.scale_y_discrete(expand<span class="op">=</span>(<span class="fl">.035</span>, <span class="fl">.035</span>))</span>
<span id="cb185-31"><a aria-hidden="true" href="#cb185-31" tabindex="-1"></a>       <span class="op">+</span> p9.scale_fill_manual(values<span class="op">=</span>[<span class="st">'#D3894C'</span>, <span class="st">'#435B97'</span>])</span>
<span id="cb185-32"><a aria-hidden="true" href="#cb185-32" tabindex="-1"></a>       <span class="op">+</span> p9.guides(size<span class="op">=</span><span class="va">False</span>)</span>
<span id="cb185-33"><a aria-hidden="true" href="#cb185-33" tabindex="-1"></a>       <span class="op">+</span> p9.labs(x<span class="op">=</span><span class="st">''</span>, y<span class="op">=</span><span class="st">''</span>, size<span class="op">=</span><span class="st">'Relative Abundance'</span>, color<span class="op">=</span><span class="st">'Class'</span>,</span>
<span id="cb185-34"><a aria-hidden="true" href="#cb185-34" tabindex="-1"></a>                 title<span class="op">=</span><span class="st">''</span>,</span>
<span id="cb185-35"><a aria-hidden="true" href="#cb185-35" tabindex="-1"></a>                 fill<span class="op">=</span><span class="st">''</span></span>
<span id="cb185-36"><a aria-hidden="true" href="#cb185-36" tabindex="-1"></a>                 )</span>
<span id="cb185-37"><a aria-hidden="true" href="#cb185-37" tabindex="-1"></a>                  )</span>
<span id="cb185-38"><a aria-hidden="true" href="#cb185-38" tabindex="-1"></a>       <span class="op">+</span> p9.annotate(geom<span class="op">=</span><span class="st">'vline'</span>, linetype<span class="op">=</span><span class="st">'dashed'</span>, xintercept<span class="op">=</span><span class="fl">2.5</span>, color<span class="op">=</span><span class="st">'grey'</span>)</span>
<span id="cb185-39"><a aria-hidden="true" href="#cb185-39" tabindex="-1"></a>       <span class="op">+</span> p9.theme(figure_size<span class="op">=</span>(<span class="fl">5.5</span>, <span class="fl">7.5</span>),</span>
<span id="cb185-40"><a aria-hidden="true" href="#cb185-40" tabindex="-1"></a>                  strip_text<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">12</span>),</span>
<span id="cb185-41"><a aria-hidden="true" href="#cb185-41" tabindex="-1"></a>                  axis_text<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">10</span>),</span>
<span id="cb185-42"><a aria-hidden="true" href="#cb185-42" tabindex="-1"></a>                  legend_position<span class="op">=</span><span class="st">'bottom'</span>,</span>
<span id="cb185-43"><a aria-hidden="true" href="#cb185-43" tabindex="-1"></a>                  title<span class="op">=</span>p9.element_text(ha<span class="op">=</span><span class="st">'center'</span>),</span>
<span id="cb185-44"><a aria-hidden="true" href="#cb185-44" tabindex="-1"></a>                  legend_title<span class="op">=</span>p9.element_text(x<span class="op">=</span><span class="dv">300</span>),</span>
<span id="cb185-45"><a aria-hidden="true" href="#cb185-45" tabindex="-1"></a>                  axis_text_x<span class="op">=</span>p9.element_text(angle<span class="op">=</span><span class="dv">90</span>),</span>
<span id="cb185-46"><a aria-hidden="true" href="#cb185-46" tabindex="-1"></a>       )</span>
<span id="cb185-47"><a aria-hidden="true" href="#cb185-47" tabindex="-1"></a> )</span>
<span id="cb185-48"><a aria-hidden="true" href="#cb185-48" tabindex="-1"></a>f.save(<span class="st">'../output/figures/blanks_top_30_genera.svg'</span>)</span>
<span id="cb185-49"><a aria-hidden="true" href="#cb185-49" tabindex="-1"></a>f.draw()</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="858">
<div>
<figure class="figure">
<p><img class="figure-img" height="2250" src="microbial_diversity_analysis_files/figure-html/cell-174-output-1.png" width="1650"/></p>
</figure>
</div>
</div>
</div>
<div class="cell" id="cell-214">
<div class="sourceCode cell-code" id="cb186"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb186-1"><a aria-hidden="true" href="#cb186-1" tabindex="-1"></a>full_long_df <span class="op">=</span> (full_matrix_fun</span>
<span id="cb186-2"><a aria-hidden="true" href="#cb186-2" tabindex="-1"></a>                .reset_index()</span>
<span id="cb186-3"><a aria-hidden="true" href="#cb186-3" tabindex="-1"></a>                .melt(<span class="st">'sample_label'</span>, var_name<span class="op">=</span><span class="st">'Genus'</span>, value_name<span class="op">=</span><span class="st">'value'</span>)</span>
<span id="cb186-4"><a aria-hidden="true" href="#cb186-4" tabindex="-1"></a>)</span>
<span id="cb186-5"><a aria-hidden="true" href="#cb186-5" tabindex="-1"></a></span>
<span id="cb186-6"><a aria-hidden="true" href="#cb186-6" tabindex="-1"></a></span>
<span id="cb186-7"><a aria-hidden="true" href="#cb186-7" tabindex="-1"></a>top_30_genera_blanks <span class="op">=</span> (</span>
<span id="cb186-8"><a aria-hidden="true" href="#cb186-8" tabindex="-1"></a>    full_long_df</span>
<span id="cb186-9"><a aria-hidden="true" href="#cb186-9" tabindex="-1"></a>    .query(<span class="st">'Genus!="unknown"'</span>)</span>
<span id="cb186-10"><a aria-hidden="true" href="#cb186-10" tabindex="-1"></a>    .query(<span class="st">'sample_label.str.contains("Pool")'</span>)</span>
<span id="cb186-11"><a aria-hidden="true" href="#cb186-11" tabindex="-1"></a>    .groupby(<span class="st">'Genus'</span>)</span>
<span id="cb186-12"><a aria-hidden="true" href="#cb186-12" tabindex="-1"></a>    [<span class="st">'value'</span>]</span>
<span id="cb186-13"><a aria-hidden="true" href="#cb186-13" tabindex="-1"></a>    .<span class="bu">sum</span>()</span>
<span id="cb186-14"><a aria-hidden="true" href="#cb186-14" tabindex="-1"></a>    .sort_values(ascending<span class="op">=</span><span class="va">True</span>)</span>
<span id="cb186-15"><a aria-hidden="true" href="#cb186-15" tabindex="-1"></a>    .tail(<span class="dv">30</span>)</span>
<span id="cb186-16"><a aria-hidden="true" href="#cb186-16" tabindex="-1"></a>    .index</span>
<span id="cb186-17"><a aria-hidden="true" href="#cb186-17" tabindex="-1"></a>    .values</span>
<span id="cb186-18"><a aria-hidden="true" href="#cb186-18" tabindex="-1"></a>)</span>
<span id="cb186-19"><a aria-hidden="true" href="#cb186-19" tabindex="-1"></a></span>
<span id="cb186-20"><a aria-hidden="true" href="#cb186-20" tabindex="-1"></a>top_30_genera_samples_april <span class="op">=</span> (</span>
<span id="cb186-21"><a aria-hidden="true" href="#cb186-21" tabindex="-1"></a>    full_long_df</span>
<span id="cb186-22"><a aria-hidden="true" href="#cb186-22" tabindex="-1"></a>    .query(<span class="st">'Genus!="unknown"'</span>)</span>
<span id="cb186-23"><a aria-hidden="true" href="#cb186-23" tabindex="-1"></a>    .query(<span class="st">'sample_label.str.startswith("S")'</span>)</span>
<span id="cb186-24"><a aria-hidden="true" href="#cb186-24" tabindex="-1"></a>    .assign(sample_id<span class="op">=</span><span class="kw">lambda</span> dd: dd.sample_label.<span class="bu">str</span>.split(<span class="st">'</span><span class="ch">\n</span><span class="st">'</span>).<span class="bu">str</span>[<span class="dv">0</span>])</span>
<span id="cb186-25"><a aria-hidden="true" href="#cb186-25" tabindex="-1"></a>    .merge(samples_info, how<span class="op">=</span><span class="st">'left'</span>)</span>
<span id="cb186-26"><a aria-hidden="true" href="#cb186-26" tabindex="-1"></a>    .assign(month<span class="op">=</span><span class="kw">lambda</span> dd: pd.to_datetime(dd.date).dt.month_name().fillna(<span class="st">'Blanks'</span>))</span>
<span id="cb186-27"><a aria-hidden="true" href="#cb186-27" tabindex="-1"></a>    .query(<span class="st">'month=="April"'</span>)</span>
<span id="cb186-28"><a aria-hidden="true" href="#cb186-28" tabindex="-1"></a>    .groupby(<span class="st">'Genus'</span>)</span>
<span id="cb186-29"><a aria-hidden="true" href="#cb186-29" tabindex="-1"></a>    [<span class="st">'value'</span>]</span>
<span id="cb186-30"><a aria-hidden="true" href="#cb186-30" tabindex="-1"></a>    .<span class="bu">sum</span>()</span>
<span id="cb186-31"><a aria-hidden="true" href="#cb186-31" tabindex="-1"></a>    .sort_values(ascending<span class="op">=</span><span class="va">True</span>)</span>
<span id="cb186-32"><a aria-hidden="true" href="#cb186-32" tabindex="-1"></a>    .tail(<span class="dv">30</span>)</span>
<span id="cb186-33"><a aria-hidden="true" href="#cb186-33" tabindex="-1"></a>    .index</span>
<span id="cb186-34"><a aria-hidden="true" href="#cb186-34" tabindex="-1"></a>    .values</span>
<span id="cb186-35"><a aria-hidden="true" href="#cb186-35" tabindex="-1"></a>)</span>
<span id="cb186-36"><a aria-hidden="true" href="#cb186-36" tabindex="-1"></a></span>
<span id="cb186-37"><a aria-hidden="true" href="#cb186-37" tabindex="-1"></a>top_30_genera_samples_february <span class="op">=</span> (</span>
<span id="cb186-38"><a aria-hidden="true" href="#cb186-38" tabindex="-1"></a>    full_long_df</span>
<span id="cb186-39"><a aria-hidden="true" href="#cb186-39" tabindex="-1"></a>    .query(<span class="st">'Genus!="unknown"'</span>)</span>
<span id="cb186-40"><a aria-hidden="true" href="#cb186-40" tabindex="-1"></a>    .query(<span class="st">'sample_label.str.startswith("S")'</span>)</span>
<span id="cb186-41"><a aria-hidden="true" href="#cb186-41" tabindex="-1"></a>    .assign(sample_id<span class="op">=</span><span class="kw">lambda</span> dd: dd.sample_label.<span class="bu">str</span>.split(<span class="st">'</span><span class="ch">\n</span><span class="st">'</span>).<span class="bu">str</span>[<span class="dv">0</span>])</span>
<span id="cb186-42"><a aria-hidden="true" href="#cb186-42" tabindex="-1"></a>    .merge(samples_info, how<span class="op">=</span><span class="st">'left'</span>)</span>
<span id="cb186-43"><a aria-hidden="true" href="#cb186-43" tabindex="-1"></a>    .assign(month<span class="op">=</span><span class="kw">lambda</span> dd: pd.to_datetime(dd.date).dt.month_name().fillna(<span class="st">'Blanks'</span>))</span>
<span id="cb186-44"><a aria-hidden="true" href="#cb186-44" tabindex="-1"></a>    .query(<span class="st">'month=="February"'</span>)</span>
<span id="cb186-45"><a aria-hidden="true" href="#cb186-45" tabindex="-1"></a>    .groupby(<span class="st">'Genus'</span>)</span>
<span id="cb186-46"><a aria-hidden="true" href="#cb186-46" tabindex="-1"></a>    [<span class="st">'value'</span>]</span>
<span id="cb186-47"><a aria-hidden="true" href="#cb186-47" tabindex="-1"></a>    .<span class="bu">sum</span>()</span>
<span id="cb186-48"><a aria-hidden="true" href="#cb186-48" tabindex="-1"></a>    .sort_values(ascending<span class="op">=</span><span class="va">True</span>)</span>
<span id="cb186-49"><a aria-hidden="true" href="#cb186-49" tabindex="-1"></a>    .tail(<span class="dv">30</span>)</span>
<span id="cb186-50"><a aria-hidden="true" href="#cb186-50" tabindex="-1"></a>    .index</span>
<span id="cb186-51"><a aria-hidden="true" href="#cb186-51" tabindex="-1"></a>    .values</span>
<span id="cb186-52"><a aria-hidden="true" href="#cb186-52" tabindex="-1"></a>)</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" id="cell-215">
<div class="sourceCode cell-code" id="cb187"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb187-1"><a aria-hidden="true" href="#cb187-1" tabindex="-1"></a>f <span class="op">=</span> (full_long_df </span>
<span id="cb187-2"><a aria-hidden="true" href="#cb187-2" tabindex="-1"></a> .query(<span class="st">'Genus.notna()'</span>)</span>
<span id="cb187-3"><a aria-hidden="true" href="#cb187-3" tabindex="-1"></a>.assign(Genus<span class="op">=</span><span class="kw">lambda</span> dd: </span>
<span id="cb187-4"><a aria-hidden="true" href="#cb187-4" tabindex="-1"></a>        pd.Categorical(dd.Genus, categories<span class="op">=</span>top_30_genera_samples_february, ordered<span class="op">=</span><span class="va">True</span>))</span>
<span id="cb187-5"><a aria-hidden="true" href="#cb187-5" tabindex="-1"></a>.dropna()</span>
<span id="cb187-6"><a aria-hidden="true" href="#cb187-6" tabindex="-1"></a>.assign(sample_id<span class="op">=</span><span class="kw">lambda</span> dd: dd.sample_label.<span class="bu">str</span>.split(<span class="st">'</span><span class="ch">\n</span><span class="st">'</span>).<span class="bu">str</span>[<span class="dv">0</span>])</span>
<span id="cb187-7"><a aria-hidden="true" href="#cb187-7" tabindex="-1"></a>.merge(samples_info, how<span class="op">=</span><span class="st">'left'</span>)</span>
<span id="cb187-8"><a aria-hidden="true" href="#cb187-8" tabindex="-1"></a>.assign(month<span class="op">=</span><span class="kw">lambda</span> dd: pd.to_datetime(dd.date).dt.month_name().fillna(<span class="st">'Blanks'</span>))</span>
<span id="cb187-9"><a aria-hidden="true" href="#cb187-9" tabindex="-1"></a>.assign(sample_mode<span class="op">=</span><span class="kw">lambda</span> dd: </span>
<span id="cb187-10"><a aria-hidden="true" href="#cb187-10" tabindex="-1"></a>        np.where(dd.sample_mode.isna, </span>
<span id="cb187-11"><a aria-hidden="true" href="#cb187-11" tabindex="-1"></a>                 dd.sample_label.<span class="bu">str</span>.split(<span class="st">'('</span>).<span class="bu">str</span>[<span class="op">-</span><span class="dv">1</span>].<span class="bu">str</span>[:<span class="op">-</span><span class="dv">1</span>], </span>
<span id="cb187-12"><a aria-hidden="true" href="#cb187-12" tabindex="-1"></a>                 dd.sample_mode))</span>
<span id="cb187-13"><a aria-hidden="true" href="#cb187-13" tabindex="-1"></a>.assign(sample_group_label<span class="op">=</span><span class="kw">lambda</span> dd: dd.sample_mode <span class="op">+</span> <span class="st">' '</span> <span class="op">+</span> dd.month)</span>
<span id="cb187-14"><a aria-hidden="true" href="#cb187-14" tabindex="-1"></a>.groupby([<span class="st">'sample_group_label'</span>, <span class="st">'sample_mode'</span>, <span class="st">'Genus'</span>])</span>
<span id="cb187-15"><a aria-hidden="true" href="#cb187-15" tabindex="-1"></a>.aggregate({<span class="st">'value'</span>: <span class="st">'mean'</span>})</span>
<span id="cb187-16"><a aria-hidden="true" href="#cb187-16" tabindex="-1"></a>.reset_index()</span>
<span id="cb187-17"><a aria-hidden="true" href="#cb187-17" tabindex="-1"></a>.query(<span class="st">'value &gt; 0'</span>)</span>
<span id="cb187-18"><a aria-hidden="true" href="#cb187-18" tabindex="-1"></a>.assign(value<span class="op">=</span><span class="kw">lambda</span> dd: np.where(dd.value <span class="op">&lt;</span> treshold, treshold, dd.value))</span>
<span id="cb187-19"><a aria-hidden="true" href="#cb187-19" tabindex="-1"></a>.assign(sample_group_label<span class="op">=</span><span class="kw">lambda</span> dd: pd.Categorical(dd.sample_group_label,</span>
<span id="cb187-20"><a aria-hidden="true" href="#cb187-20" tabindex="-1"></a>        categories<span class="op">=</span>[<span class="st">'Flight Blanks'</span>, <span class="st">'Ground Blanks'</span>, <span class="st">'Flight February'</span>, <span class="st">'Ground February'</span>,</span>
<span id="cb187-21"><a aria-hidden="true" href="#cb187-21" tabindex="-1"></a>                    <span class="st">'Flight April'</span>, <span class="st">'Ground April'</span>], ordered<span class="op">=</span><span class="va">True</span>))</span>
<span id="cb187-22"><a aria-hidden="true" href="#cb187-22" tabindex="-1"></a> .pipe(<span class="kw">lambda</span> dd: p9.ggplot(dd)</span>
<span id="cb187-23"><a aria-hidden="true" href="#cb187-23" tabindex="-1"></a>       <span class="op">+</span> p9.aes(<span class="st">'sample_group_label'</span>, <span class="st">'Genus'</span>, size<span class="op">=</span><span class="st">'value'</span>, fill<span class="op">=</span><span class="st">'sample_mode'</span>)</span>
<span id="cb187-24"><a aria-hidden="true" href="#cb187-24" tabindex="-1"></a>       <span class="op">+</span> p9.geom_point(alpha<span class="op">=</span><span class="fl">.8</span>, stroke<span class="op">=</span><span class="dv">0</span>)</span>
<span id="cb187-25"><a aria-hidden="true" href="#cb187-25" tabindex="-1"></a>       <span class="op">+</span> p9.scale_size_continuous(trans<span class="op">=</span><span class="st">'log10'</span>,</span>
<span id="cb187-26"><a aria-hidden="true" href="#cb187-26" tabindex="-1"></a>                                  <span class="bu">range</span><span class="op">=</span>(<span class="dv">1</span>, <span class="dv">9</span>),</span>
<span id="cb187-27"><a aria-hidden="true" href="#cb187-27" tabindex="-1"></a>                                  labels<span class="op">=</span>percent_format(), </span>
<span id="cb187-28"><a aria-hidden="true" href="#cb187-28" tabindex="-1"></a>                                  limits<span class="op">=</span>(treshold <span class="op">-</span> <span class="fl">1e-5</span>, <span class="fl">.9</span>),</span>
<span id="cb187-29"><a aria-hidden="true" href="#cb187-29" tabindex="-1"></a>                                  breaks<span class="op">=</span>[<span class="fl">.001</span>, <span class="fl">.01</span>, <span class="fl">.05</span>, <span class="fl">.5</span>])</span>
<span id="cb187-30"><a aria-hidden="true" href="#cb187-30" tabindex="-1"></a>       <span class="op">+</span> p9.scale_y_discrete(expand<span class="op">=</span>(<span class="fl">.035</span>, <span class="fl">.035</span>))</span>
<span id="cb187-31"><a aria-hidden="true" href="#cb187-31" tabindex="-1"></a>       <span class="op">+</span> p9.scale_fill_manual(values<span class="op">=</span>[<span class="st">'#D3894C'</span>, <span class="st">'#435B97'</span>])</span>
<span id="cb187-32"><a aria-hidden="true" href="#cb187-32" tabindex="-1"></a>       <span class="op">+</span> p9.guides(size<span class="op">=</span><span class="va">False</span>)</span>
<span id="cb187-33"><a aria-hidden="true" href="#cb187-33" tabindex="-1"></a>       <span class="op">+</span> p9.labs(x<span class="op">=</span><span class="st">''</span>, y<span class="op">=</span><span class="st">''</span>, size<span class="op">=</span><span class="st">'Relative Abundance'</span>, color<span class="op">=</span><span class="st">'Class'</span>,</span>
<span id="cb187-34"><a aria-hidden="true" href="#cb187-34" tabindex="-1"></a>                 title<span class="op">=</span><span class="st">''</span>,</span>
<span id="cb187-35"><a aria-hidden="true" href="#cb187-35" tabindex="-1"></a>                 fill<span class="op">=</span><span class="st">''</span></span>
<span id="cb187-36"><a aria-hidden="true" href="#cb187-36" tabindex="-1"></a>                 )</span>
<span id="cb187-37"><a aria-hidden="true" href="#cb187-37" tabindex="-1"></a>                  )</span>
<span id="cb187-38"><a aria-hidden="true" href="#cb187-38" tabindex="-1"></a>       <span class="op">+</span> p9.annotate(geom<span class="op">=</span><span class="st">'vline'</span>, linetype<span class="op">=</span><span class="st">'dashed'</span>, xintercept<span class="op">=</span><span class="fl">2.5</span>, color<span class="op">=</span><span class="st">'grey'</span>)</span>
<span id="cb187-39"><a aria-hidden="true" href="#cb187-39" tabindex="-1"></a>       <span class="op">+</span> p9.theme(figure_size<span class="op">=</span>(<span class="fl">3.75</span>, <span class="fl">7.5</span>),</span>
<span id="cb187-40"><a aria-hidden="true" href="#cb187-40" tabindex="-1"></a>                  strip_text<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">12</span>),</span>
<span id="cb187-41"><a aria-hidden="true" href="#cb187-41" tabindex="-1"></a>                  axis_text<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">10</span>),</span>
<span id="cb187-42"><a aria-hidden="true" href="#cb187-42" tabindex="-1"></a>                  legend_position<span class="op">=</span><span class="st">'bottom'</span>,</span>
<span id="cb187-43"><a aria-hidden="true" href="#cb187-43" tabindex="-1"></a>                  title<span class="op">=</span>p9.element_text(ha<span class="op">=</span><span class="st">'center'</span>),</span>
<span id="cb187-44"><a aria-hidden="true" href="#cb187-44" tabindex="-1"></a>                  legend_title<span class="op">=</span>p9.element_text(x<span class="op">=</span><span class="dv">300</span>),</span>
<span id="cb187-45"><a aria-hidden="true" href="#cb187-45" tabindex="-1"></a>                  axis_text_x<span class="op">=</span>p9.element_text(angle<span class="op">=</span><span class="dv">90</span>),</span>
<span id="cb187-46"><a aria-hidden="true" href="#cb187-46" tabindex="-1"></a>       )</span>
<span id="cb187-47"><a aria-hidden="true" href="#cb187-47" tabindex="-1"></a> )</span>
<span id="cb187-48"><a aria-hidden="true" href="#cb187-48" tabindex="-1"></a></span>
<span id="cb187-49"><a aria-hidden="true" href="#cb187-49" tabindex="-1"></a>f.save(<span class="st">'../output/figures/feb_top_30_genera_fun.svg'</span>)</span>
<span id="cb187-50"><a aria-hidden="true" href="#cb187-50" tabindex="-1"></a>f.draw()</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="863">
<div>
<figure class="figure">
<p><img class="figure-img" height="2250" src="microbial_diversity_analysis_files/figure-html/cell-176-output-1.png" width="1125"/></p>
</figure>
</div>
</div>
</div>
<div class="cell" id="cell-216">
<div class="sourceCode cell-code" id="cb188"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb188-1"><a aria-hidden="true" href="#cb188-1" tabindex="-1"></a>f <span class="op">=</span> (full_long_df </span>
<span id="cb188-2"><a aria-hidden="true" href="#cb188-2" tabindex="-1"></a> .query(<span class="st">'Genus.notna()'</span>)</span>
<span id="cb188-3"><a aria-hidden="true" href="#cb188-3" tabindex="-1"></a>.assign(Genus<span class="op">=</span><span class="kw">lambda</span> dd: </span>
<span id="cb188-4"><a aria-hidden="true" href="#cb188-4" tabindex="-1"></a>        pd.Categorical(dd.Genus, categories<span class="op">=</span>top_30_genera_samples_april, ordered<span class="op">=</span><span class="va">True</span>))</span>
<span id="cb188-5"><a aria-hidden="true" href="#cb188-5" tabindex="-1"></a>.dropna()</span>
<span id="cb188-6"><a aria-hidden="true" href="#cb188-6" tabindex="-1"></a>.assign(sample_id<span class="op">=</span><span class="kw">lambda</span> dd: dd.sample_label.<span class="bu">str</span>.split(<span class="st">'</span><span class="ch">\n</span><span class="st">'</span>).<span class="bu">str</span>[<span class="dv">0</span>])</span>
<span id="cb188-7"><a aria-hidden="true" href="#cb188-7" tabindex="-1"></a>.merge(samples_info, how<span class="op">=</span><span class="st">'left'</span>)</span>
<span id="cb188-8"><a aria-hidden="true" href="#cb188-8" tabindex="-1"></a>.assign(month<span class="op">=</span><span class="kw">lambda</span> dd: pd.to_datetime(dd.date).dt.month_name().fillna(<span class="st">'Blanks'</span>))</span>
<span id="cb188-9"><a aria-hidden="true" href="#cb188-9" tabindex="-1"></a>.assign(sample_mode<span class="op">=</span><span class="kw">lambda</span> dd: </span>
<span id="cb188-10"><a aria-hidden="true" href="#cb188-10" tabindex="-1"></a>        np.where(dd.sample_mode.isna, </span>
<span id="cb188-11"><a aria-hidden="true" href="#cb188-11" tabindex="-1"></a>                 dd.sample_label.<span class="bu">str</span>.split(<span class="st">'('</span>).<span class="bu">str</span>[<span class="op">-</span><span class="dv">1</span>].<span class="bu">str</span>[:<span class="op">-</span><span class="dv">1</span>], </span>
<span id="cb188-12"><a aria-hidden="true" href="#cb188-12" tabindex="-1"></a>                 dd.sample_mode))</span>
<span id="cb188-13"><a aria-hidden="true" href="#cb188-13" tabindex="-1"></a>.assign(sample_group_label<span class="op">=</span><span class="kw">lambda</span> dd: dd.sample_mode <span class="op">+</span> <span class="st">' '</span> <span class="op">+</span> dd.month)</span>
<span id="cb188-14"><a aria-hidden="true" href="#cb188-14" tabindex="-1"></a>.groupby([<span class="st">'sample_group_label'</span>, <span class="st">'sample_mode'</span>, <span class="st">'Genus'</span>])</span>
<span id="cb188-15"><a aria-hidden="true" href="#cb188-15" tabindex="-1"></a>.aggregate({<span class="st">'value'</span>: <span class="st">'mean'</span>})</span>
<span id="cb188-16"><a aria-hidden="true" href="#cb188-16" tabindex="-1"></a>.reset_index()</span>
<span id="cb188-17"><a aria-hidden="true" href="#cb188-17" tabindex="-1"></a>.query(<span class="st">'value &gt; 0'</span>)</span>
<span id="cb188-18"><a aria-hidden="true" href="#cb188-18" tabindex="-1"></a>.assign(value<span class="op">=</span><span class="kw">lambda</span> dd: np.where(dd.value <span class="op">&lt;</span> treshold, treshold, dd.value))</span>
<span id="cb188-19"><a aria-hidden="true" href="#cb188-19" tabindex="-1"></a>.assign(sample_group_label<span class="op">=</span><span class="kw">lambda</span> dd: pd.Categorical(dd.sample_group_label,</span>
<span id="cb188-20"><a aria-hidden="true" href="#cb188-20" tabindex="-1"></a>        categories<span class="op">=</span>[<span class="st">'Flight Blanks'</span>, <span class="st">'Ground Blanks'</span>, <span class="st">'Flight February'</span>, <span class="st">'Ground February'</span>,</span>
<span id="cb188-21"><a aria-hidden="true" href="#cb188-21" tabindex="-1"></a>                    <span class="st">'Flight April'</span>, <span class="st">'Ground April'</span>], ordered<span class="op">=</span><span class="va">True</span>))</span>
<span id="cb188-22"><a aria-hidden="true" href="#cb188-22" tabindex="-1"></a> .pipe(<span class="kw">lambda</span> dd: p9.ggplot(dd)</span>
<span id="cb188-23"><a aria-hidden="true" href="#cb188-23" tabindex="-1"></a>       <span class="op">+</span> p9.aes(<span class="st">'sample_group_label'</span>, <span class="st">'Genus'</span>, size<span class="op">=</span><span class="st">'value'</span>, fill<span class="op">=</span><span class="st">'sample_mode'</span>)</span>
<span id="cb188-24"><a aria-hidden="true" href="#cb188-24" tabindex="-1"></a>       <span class="op">+</span> p9.geom_point(alpha<span class="op">=</span><span class="fl">.8</span>, stroke<span class="op">=</span><span class="dv">0</span>)</span>
<span id="cb188-25"><a aria-hidden="true" href="#cb188-25" tabindex="-1"></a>       <span class="op">+</span> p9.scale_size_continuous(trans<span class="op">=</span><span class="st">'log10'</span>,</span>
<span id="cb188-26"><a aria-hidden="true" href="#cb188-26" tabindex="-1"></a>                                  <span class="bu">range</span><span class="op">=</span>(<span class="dv">1</span>, <span class="dv">9</span>),</span>
<span id="cb188-27"><a aria-hidden="true" href="#cb188-27" tabindex="-1"></a>                                  labels<span class="op">=</span>percent_format(), </span>
<span id="cb188-28"><a aria-hidden="true" href="#cb188-28" tabindex="-1"></a>                                  limits<span class="op">=</span>(treshold <span class="op">-</span> <span class="fl">1e-5</span>, <span class="fl">.9</span>),</span>
<span id="cb188-29"><a aria-hidden="true" href="#cb188-29" tabindex="-1"></a>                                  breaks<span class="op">=</span>[<span class="fl">.001</span>, <span class="fl">.01</span>, <span class="fl">.05</span>, <span class="fl">.5</span>])</span>
<span id="cb188-30"><a aria-hidden="true" href="#cb188-30" tabindex="-1"></a>       <span class="op">+</span> p9.scale_y_discrete(expand<span class="op">=</span>(<span class="fl">.035</span>, <span class="fl">.035</span>))</span>
<span id="cb188-31"><a aria-hidden="true" href="#cb188-31" tabindex="-1"></a>       <span class="op">+</span> p9.scale_fill_manual(values<span class="op">=</span>[<span class="st">'#D3894C'</span>, <span class="st">'#435B97'</span>])</span>
<span id="cb188-32"><a aria-hidden="true" href="#cb188-32" tabindex="-1"></a>       <span class="op">+</span> p9.guides(size<span class="op">=</span><span class="va">False</span>)</span>
<span id="cb188-33"><a aria-hidden="true" href="#cb188-33" tabindex="-1"></a>       <span class="op">+</span> p9.labs(x<span class="op">=</span><span class="st">''</span>, y<span class="op">=</span><span class="st">''</span>, size<span class="op">=</span><span class="st">'Relative Abundance'</span>, color<span class="op">=</span><span class="st">'Class'</span>,</span>
<span id="cb188-34"><a aria-hidden="true" href="#cb188-34" tabindex="-1"></a>                 title<span class="op">=</span><span class="st">''</span>,</span>
<span id="cb188-35"><a aria-hidden="true" href="#cb188-35" tabindex="-1"></a>                 fill<span class="op">=</span><span class="st">''</span></span>
<span id="cb188-36"><a aria-hidden="true" href="#cb188-36" tabindex="-1"></a>                 )</span>
<span id="cb188-37"><a aria-hidden="true" href="#cb188-37" tabindex="-1"></a>                  )</span>
<span id="cb188-38"><a aria-hidden="true" href="#cb188-38" tabindex="-1"></a>       <span class="op">+</span> p9.annotate(geom<span class="op">=</span><span class="st">'vline'</span>, linetype<span class="op">=</span><span class="st">'dashed'</span>, xintercept<span class="op">=</span><span class="fl">2.5</span>, color<span class="op">=</span><span class="st">'grey'</span>)</span>
<span id="cb188-39"><a aria-hidden="true" href="#cb188-39" tabindex="-1"></a>       <span class="op">+</span> p9.theme(figure_size<span class="op">=</span>(<span class="fl">3.75</span>, <span class="fl">7.5</span>),</span>
<span id="cb188-40"><a aria-hidden="true" href="#cb188-40" tabindex="-1"></a>                  strip_text<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">12</span>),</span>
<span id="cb188-41"><a aria-hidden="true" href="#cb188-41" tabindex="-1"></a>                  axis_text<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">10</span>),</span>
<span id="cb188-42"><a aria-hidden="true" href="#cb188-42" tabindex="-1"></a>                  legend_position<span class="op">=</span><span class="st">'bottom'</span>,</span>
<span id="cb188-43"><a aria-hidden="true" href="#cb188-43" tabindex="-1"></a>                  title<span class="op">=</span>p9.element_text(ha<span class="op">=</span><span class="st">'center'</span>),</span>
<span id="cb188-44"><a aria-hidden="true" href="#cb188-44" tabindex="-1"></a>                  legend_title<span class="op">=</span>p9.element_text(x<span class="op">=</span><span class="dv">300</span>),</span>
<span id="cb188-45"><a aria-hidden="true" href="#cb188-45" tabindex="-1"></a>                  axis_text_x<span class="op">=</span>p9.element_text(angle<span class="op">=</span><span class="dv">90</span>),</span>
<span id="cb188-46"><a aria-hidden="true" href="#cb188-46" tabindex="-1"></a>       )</span>
<span id="cb188-47"><a aria-hidden="true" href="#cb188-47" tabindex="-1"></a> )</span>
<span id="cb188-48"><a aria-hidden="true" href="#cb188-48" tabindex="-1"></a>f.save(<span class="st">'../output/figures/apr_top_30_genera_fun.svg'</span>)</span>
<span id="cb188-49"><a aria-hidden="true" href="#cb188-49" tabindex="-1"></a>f.draw()</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="865">
<div>
<figure class="figure">
<p><img class="figure-img" height="2250" src="microbial_diversity_analysis_files/figure-html/cell-177-output-1.png" width="1125"/></p>
</figure>
</div>
</div>
</div>
<div class="cell" data-execution_count="69" id="cell-217">
<div class="sourceCode cell-code" id="cb189"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb189-1"><a aria-hidden="true" href="#cb189-1" tabindex="-1"></a>pathogenic_bacteria_df <span class="op">=</span> (</span>
<span id="cb189-2"><a aria-hidden="true" href="#cb189-2" tabindex="-1"></a>    pd.read_excel(<span class="st">'../data/Species 2023_final.xlsx'</span>,</span>
<span id="cb189-3"><a aria-hidden="true" href="#cb189-3" tabindex="-1"></a>                  sheet_name<span class="op">=</span><span class="st">'JV analysis_Bacteria'</span>)</span>
<span id="cb189-4"><a aria-hidden="true" href="#cb189-4" tabindex="-1"></a>                  [[<span class="st">'Genus'</span>, <span class="st">'Species'</span>]]</span>
<span id="cb189-5"><a aria-hidden="true" href="#cb189-5" tabindex="-1"></a>    .assign(label<span class="op">=</span><span class="st">'Bacterial Potential Human Pathogens'</span>)</span>
<span id="cb189-6"><a aria-hidden="true" href="#cb189-6" tabindex="-1"></a>)</span>
<span id="cb189-7"><a aria-hidden="true" href="#cb189-7" tabindex="-1"></a></span>
<span id="cb189-8"><a aria-hidden="true" href="#cb189-8" tabindex="-1"></a>pathogenic_fungi_df <span class="op">=</span> (pd.read_excel(<span class="st">'../data/Species 2023_final.xlsx'</span>,</span>
<span id="cb189-9"><a aria-hidden="true" href="#cb189-9" tabindex="-1"></a>                                     sheet_name<span class="op">=</span><span class="st">'JV analysis_Fungi'</span>, </span>
<span id="cb189-10"><a aria-hidden="true" href="#cb189-10" tabindex="-1"></a>                                     skiprows<span class="op">=</span><span class="dv">1</span>)</span>
<span id="cb189-11"><a aria-hidden="true" href="#cb189-11" tabindex="-1"></a>                       [[<span class="st">'Genus'</span>, <span class="st">'Species'</span>]]</span>
<span id="cb189-12"><a aria-hidden="true" href="#cb189-12" tabindex="-1"></a>                       .dropna()</span>
<span id="cb189-13"><a aria-hidden="true" href="#cb189-13" tabindex="-1"></a>    .assign(label<span class="op">=</span><span class="st">'Fungal Potential Human Pathogens'</span>) </span>
<span id="cb189-14"><a aria-hidden="true" href="#cb189-14" tabindex="-1"></a>)</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="76" id="cell-218">
<div class="sourceCode cell-code" id="cb190"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb190-1"><a aria-hidden="true" href="#cb190-1" tabindex="-1"></a>pathogenic_fungi_df</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="76">
<div>
<table class="dataframe caption-top table table-sm table-striped small" data-border="1" data-quarto-postprocess="true">
<thead>
<tr class="header">
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th">Genus</th>
<th data-quarto-table-cell-role="th">Species</th>
<th data-quarto-table-cell-role="th">label</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td data-quarto-table-cell-role="th">0</td>
<td>Acremonium</td>
<td>cavaraeanum</td>
<td>Fungal Potential Human Pathogens</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">1</td>
<td>Acremonium</td>
<td>fusidioides</td>
<td>Fungal Potential Human Pathogens</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">2</td>
<td>Acremonium</td>
<td>hyalinulum</td>
<td>Fungal Potential Human Pathogens</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">3</td>
<td>Alternaria</td>
<td>metachromatica</td>
<td>Fungal Potential Human Pathogens</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">4</td>
<td>Alternaria</td>
<td>rosae</td>
<td>Fungal Potential Human Pathogens</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">5</td>
<td>Alternaria</td>
<td>sonchi</td>
<td>Fungal Potential Human Pathogens</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">6</td>
<td>Aspergillus</td>
<td>intermedius</td>
<td>Fungal Potential Human Pathogens</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">7</td>
<td>Aspergillus</td>
<td>penicillioides</td>
<td>Fungal Potential Human Pathogens</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">8</td>
<td>Aspergillus</td>
<td>unguis</td>
<td>Fungal Potential Human Pathogens</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">9</td>
<td>Aureobasidium</td>
<td>pullulans</td>
<td>Fungal Potential Human Pathogens</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">10</td>
<td>Beauveria</td>
<td>bassiana</td>
<td>Fungal Potential Human Pathogens</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">11</td>
<td>Candida</td>
<td>parapsilosis</td>
<td>Fungal Potential Human Pathogens</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">12</td>
<td>Candida</td>
<td>zeylanoides</td>
<td>Fungal Potential Human Pathogens</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">13</td>
<td>Cladosporium</td>
<td>aphidis</td>
<td>Fungal Potential Human Pathogens</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">14</td>
<td>Cladosporium</td>
<td>domesticum</td>
<td>Fungal Potential Human Pathogens</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">15</td>
<td>Cladosporium</td>
<td>halotolerans</td>
<td>Fungal Potential Human Pathogens</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">16</td>
<td>Cladosporium</td>
<td>herbarum</td>
<td>Fungal Potential Human Pathogens</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">17</td>
<td>Cladosporium</td>
<td>sphaerospermum</td>
<td>Fungal Potential Human Pathogens</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">18</td>
<td>Cladosporium</td>
<td>velox</td>
<td>Fungal Potential Human Pathogens</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">19</td>
<td>Cutaneotrichosporon</td>
<td>debeurmannianum</td>
<td>Fungal Potential Human Pathogens</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">20</td>
<td>Epicoccum</td>
<td>italicum</td>
<td>Fungal Potential Human Pathogens</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">21</td>
<td>Exophiala</td>
<td>oligosperma</td>
<td>Fungal Potential Human Pathogens</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">22</td>
<td>Malassezia</td>
<td>globosa</td>
<td>Fungal Potential Human Pathogens</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">23</td>
<td>Malassezia</td>
<td>restricta</td>
<td>Fungal Potential Human Pathogens</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">24</td>
<td>Malassezia</td>
<td>sympodialis</td>
<td>Fungal Potential Human Pathogens</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">25</td>
<td>Saccharomyces</td>
<td>cerevisiae</td>
<td>Fungal Potential Human Pathogens</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">26</td>
<td>Sarocladium</td>
<td>kiliense</td>
<td>Fungal Potential Human Pathogens</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">27</td>
<td>Sarocladium</td>
<td>subulatum</td>
<td>Fungal Potential Human Pathogens</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">28</td>
<td>Sarocladium</td>
<td>zeae</td>
<td>Fungal Potential Human Pathogens</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">29</td>
<td>Trichosporon</td>
<td>asahii</td>
<td>Fungal Potential Human Pathogens</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="cell" data-execution_count="78" id="cell-219">
<div class="sourceCode cell-code" id="cb191"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb191-1"><a aria-hidden="true" href="#cb191-1" tabindex="-1"></a>bac_pathogens_df <span class="op">=</span> pathogenic_bacteria_df</span>
<span id="cb191-2"><a aria-hidden="true" href="#cb191-2" tabindex="-1"></a>fungi_pathogens_df <span class="op">=</span> pathogenic_fungi_df</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" id="cell-220">
<div class="sourceCode cell-code" id="cb192"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb192-1"><a aria-hidden="true" href="#cb192-1" tabindex="-1"></a>f <span class="op">=</span> (full_long_df </span>
<span id="cb192-2"><a aria-hidden="true" href="#cb192-2" tabindex="-1"></a> .query(<span class="st">'Genus.notna()'</span>)</span>
<span id="cb192-3"><a aria-hidden="true" href="#cb192-3" tabindex="-1"></a>.assign(Genus<span class="op">=</span><span class="kw">lambda</span> dd: </span>
<span id="cb192-4"><a aria-hidden="true" href="#cb192-4" tabindex="-1"></a>        pd.Categorical(dd.Genus, categories<span class="op">=</span>top_30_genera_blanks, ordered<span class="op">=</span><span class="va">True</span>))</span>
<span id="cb192-5"><a aria-hidden="true" href="#cb192-5" tabindex="-1"></a>.dropna()</span>
<span id="cb192-6"><a aria-hidden="true" href="#cb192-6" tabindex="-1"></a>.assign(sample_id<span class="op">=</span><span class="kw">lambda</span> dd: dd.sample_label.<span class="bu">str</span>.split(<span class="st">'</span><span class="ch">\n</span><span class="st">'</span>).<span class="bu">str</span>[<span class="dv">0</span>])</span>
<span id="cb192-7"><a aria-hidden="true" href="#cb192-7" tabindex="-1"></a>.merge(samples_info, how<span class="op">=</span><span class="st">'left'</span>)</span>
<span id="cb192-8"><a aria-hidden="true" href="#cb192-8" tabindex="-1"></a> .replace({<span class="st">'Genus'</span>: {<span class="st">'Malasseziaceae_gen_Incertae_sedis'</span>: <span class="st">'Malasseziaceae'</span>}})</span>
<span id="cb192-9"><a aria-hidden="true" href="#cb192-9" tabindex="-1"></a>.assign(month<span class="op">=</span><span class="kw">lambda</span> dd: pd.to_datetime(dd.date).dt.month_name().fillna(<span class="st">'Blanks'</span>))</span>
<span id="cb192-10"><a aria-hidden="true" href="#cb192-10" tabindex="-1"></a>.assign(sample_mode<span class="op">=</span><span class="kw">lambda</span> dd: </span>
<span id="cb192-11"><a aria-hidden="true" href="#cb192-11" tabindex="-1"></a>        np.where(dd.sample_mode.isna, </span>
<span id="cb192-12"><a aria-hidden="true" href="#cb192-12" tabindex="-1"></a>                 dd.sample_label.<span class="bu">str</span>.split(<span class="st">'('</span>).<span class="bu">str</span>[<span class="op">-</span><span class="dv">1</span>].<span class="bu">str</span>[:<span class="op">-</span><span class="dv">1</span>], </span>
<span id="cb192-13"><a aria-hidden="true" href="#cb192-13" tabindex="-1"></a>                 dd.sample_mode))</span>
<span id="cb192-14"><a aria-hidden="true" href="#cb192-14" tabindex="-1"></a>.assign(sample_group_label<span class="op">=</span><span class="kw">lambda</span> dd: dd.sample_mode <span class="op">+</span> <span class="st">' '</span> <span class="op">+</span> dd.month)</span>
<span id="cb192-15"><a aria-hidden="true" href="#cb192-15" tabindex="-1"></a>.groupby([<span class="st">'sample_group_label'</span>, <span class="st">'sample_mode'</span>, <span class="st">'Genus'</span>])</span>
<span id="cb192-16"><a aria-hidden="true" href="#cb192-16" tabindex="-1"></a>.aggregate({<span class="st">'value'</span>: <span class="st">'mean'</span>})</span>
<span id="cb192-17"><a aria-hidden="true" href="#cb192-17" tabindex="-1"></a>.reset_index()</span>
<span id="cb192-18"><a aria-hidden="true" href="#cb192-18" tabindex="-1"></a>.query(<span class="st">'value &gt; 0'</span>)</span>
<span id="cb192-19"><a aria-hidden="true" href="#cb192-19" tabindex="-1"></a>.assign(value<span class="op">=</span><span class="kw">lambda</span> dd: np.where(dd.value <span class="op">&lt;</span> treshold, treshold, dd.value))</span>
<span id="cb192-20"><a aria-hidden="true" href="#cb192-20" tabindex="-1"></a>.assign(sample_group_label<span class="op">=</span><span class="kw">lambda</span> dd: pd.Categorical(dd.sample_group_label,</span>
<span id="cb192-21"><a aria-hidden="true" href="#cb192-21" tabindex="-1"></a>        categories<span class="op">=</span>[<span class="st">'Flight Blanks'</span>, <span class="st">'Ground Blanks'</span>, <span class="st">'Flight February'</span>, <span class="st">'Ground February'</span>,</span>
<span id="cb192-22"><a aria-hidden="true" href="#cb192-22" tabindex="-1"></a>                    <span class="st">'Flight April'</span>, <span class="st">'Ground April'</span>], ordered<span class="op">=</span><span class="va">True</span>))</span>
<span id="cb192-23"><a aria-hidden="true" href="#cb192-23" tabindex="-1"></a> .pipe(<span class="kw">lambda</span> dd: p9.ggplot(dd)</span>
<span id="cb192-24"><a aria-hidden="true" href="#cb192-24" tabindex="-1"></a>       <span class="op">+</span> p9.aes(<span class="st">'sample_group_label'</span>, <span class="st">'Genus'</span>, size<span class="op">=</span><span class="st">'value'</span>, fill<span class="op">=</span><span class="st">'sample_mode'</span>)</span>
<span id="cb192-25"><a aria-hidden="true" href="#cb192-25" tabindex="-1"></a>       <span class="op">+</span> p9.geom_point(alpha<span class="op">=</span><span class="fl">.8</span>, stroke<span class="op">=</span><span class="dv">0</span>)</span>
<span id="cb192-26"><a aria-hidden="true" href="#cb192-26" tabindex="-1"></a>       <span class="op">+</span> p9.scale_size_continuous(trans<span class="op">=</span><span class="st">'log10'</span>,</span>
<span id="cb192-27"><a aria-hidden="true" href="#cb192-27" tabindex="-1"></a>                                  <span class="bu">range</span><span class="op">=</span>(<span class="dv">1</span>, <span class="dv">9</span>),</span>
<span id="cb192-28"><a aria-hidden="true" href="#cb192-28" tabindex="-1"></a>                                  labels<span class="op">=</span>percent_format(), </span>
<span id="cb192-29"><a aria-hidden="true" href="#cb192-29" tabindex="-1"></a>                                  limits<span class="op">=</span>(treshold <span class="op">-</span> <span class="fl">1e-5</span>, <span class="fl">.9</span>),</span>
<span id="cb192-30"><a aria-hidden="true" href="#cb192-30" tabindex="-1"></a>                                  breaks<span class="op">=</span>[<span class="fl">.001</span>, <span class="fl">.01</span>, <span class="fl">.05</span>, <span class="fl">.5</span>])</span>
<span id="cb192-31"><a aria-hidden="true" href="#cb192-31" tabindex="-1"></a>       <span class="op">+</span> p9.scale_y_discrete(expand<span class="op">=</span>(<span class="fl">.035</span>, <span class="fl">.035</span>))</span>
<span id="cb192-32"><a aria-hidden="true" href="#cb192-32" tabindex="-1"></a>       <span class="op">+</span> p9.scale_fill_manual(values<span class="op">=</span>[<span class="st">'#D3894C'</span>, <span class="st">'#435B97'</span>])</span>
<span id="cb192-33"><a aria-hidden="true" href="#cb192-33" tabindex="-1"></a>       <span class="op">+</span> p9.labs(x<span class="op">=</span><span class="st">''</span>, y<span class="op">=</span><span class="st">''</span>, size<span class="op">=</span><span class="st">'Relative Abundance'</span>, color<span class="op">=</span><span class="st">'Class'</span>,</span>
<span id="cb192-34"><a aria-hidden="true" href="#cb192-34" tabindex="-1"></a>                 title<span class="op">=</span><span class="st">''</span>,</span>
<span id="cb192-35"><a aria-hidden="true" href="#cb192-35" tabindex="-1"></a>                 fill<span class="op">=</span><span class="st">''</span></span>
<span id="cb192-36"><a aria-hidden="true" href="#cb192-36" tabindex="-1"></a>                 )</span>
<span id="cb192-37"><a aria-hidden="true" href="#cb192-37" tabindex="-1"></a>                  )</span>
<span id="cb192-38"><a aria-hidden="true" href="#cb192-38" tabindex="-1"></a>       <span class="op">+</span> p9.annotate(geom<span class="op">=</span><span class="st">'vline'</span>, linetype<span class="op">=</span><span class="st">'dashed'</span>, xintercept<span class="op">=</span><span class="fl">2.5</span>, color<span class="op">=</span><span class="st">'grey'</span>)</span>
<span id="cb192-39"><a aria-hidden="true" href="#cb192-39" tabindex="-1"></a>       <span class="op">+</span> p9.theme(figure_size<span class="op">=</span>(<span class="fl">3.75</span>, <span class="fl">7.5</span>),</span>
<span id="cb192-40"><a aria-hidden="true" href="#cb192-40" tabindex="-1"></a>                  strip_text<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">12</span>),</span>
<span id="cb192-41"><a aria-hidden="true" href="#cb192-41" tabindex="-1"></a>                  axis_text<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">10</span>),</span>
<span id="cb192-42"><a aria-hidden="true" href="#cb192-42" tabindex="-1"></a>                  legend_position<span class="op">=</span><span class="st">'bottom'</span>,</span>
<span id="cb192-43"><a aria-hidden="true" href="#cb192-43" tabindex="-1"></a>                  title<span class="op">=</span>p9.element_text(ha<span class="op">=</span><span class="st">'center'</span>),</span>
<span id="cb192-44"><a aria-hidden="true" href="#cb192-44" tabindex="-1"></a>                  legend_title<span class="op">=</span>p9.element_text(x<span class="op">=</span><span class="dv">300</span>),</span>
<span id="cb192-45"><a aria-hidden="true" href="#cb192-45" tabindex="-1"></a>                  axis_text_x<span class="op">=</span>p9.element_text(angle<span class="op">=</span><span class="dv">90</span>),</span>
<span id="cb192-46"><a aria-hidden="true" href="#cb192-46" tabindex="-1"></a>       )</span>
<span id="cb192-47"><a aria-hidden="true" href="#cb192-47" tabindex="-1"></a> )</span>
<span id="cb192-48"><a aria-hidden="true" href="#cb192-48" tabindex="-1"></a>f.save(<span class="st">'../output/figures/blanks_top_30_genera_fun.svg'</span>)</span>
<span id="cb192-49"><a aria-hidden="true" href="#cb192-49" tabindex="-1"></a>f.draw()</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="867">
<div>
<figure class="figure">
<p><img class="figure-img" height="2250" src="microbial_diversity_analysis_files/figure-html/cell-181-output-1.png" width="1125"/></p>
</figure>
</div>
</div>
</div>
<div class="cell" id="cell-221">
<div class="sourceCode cell-code" id="cb193"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb193-1"><a aria-hidden="true" href="#cb193-1" tabindex="-1"></a>full_long_df <span class="op">=</span> (full_matrix_bac</span>
<span id="cb193-2"><a aria-hidden="true" href="#cb193-2" tabindex="-1"></a>                .reset_index()</span>
<span id="cb193-3"><a aria-hidden="true" href="#cb193-3" tabindex="-1"></a>                .melt(<span class="st">'sample_label'</span>, var_name<span class="op">=</span><span class="st">'Genus'</span>, value_name<span class="op">=</span><span class="st">'value'</span>)</span>
<span id="cb193-4"><a aria-hidden="true" href="#cb193-4" tabindex="-1"></a>)</span>
<span id="cb193-5"><a aria-hidden="true" href="#cb193-5" tabindex="-1"></a></span>
<span id="cb193-6"><a aria-hidden="true" href="#cb193-6" tabindex="-1"></a></span>
<span id="cb193-7"><a aria-hidden="true" href="#cb193-7" tabindex="-1"></a>top_30_genera_blanks <span class="op">=</span> (</span>
<span id="cb193-8"><a aria-hidden="true" href="#cb193-8" tabindex="-1"></a>    full_long_df</span>
<span id="cb193-9"><a aria-hidden="true" href="#cb193-9" tabindex="-1"></a>    .query(<span class="st">'Genus!="unknown"'</span>)</span>
<span id="cb193-10"><a aria-hidden="true" href="#cb193-10" tabindex="-1"></a>    .query(<span class="st">'sample_label.str.contains("Pool")'</span>)</span>
<span id="cb193-11"><a aria-hidden="true" href="#cb193-11" tabindex="-1"></a>    .groupby(<span class="st">'Genus'</span>)</span>
<span id="cb193-12"><a aria-hidden="true" href="#cb193-12" tabindex="-1"></a>    [<span class="st">'value'</span>]</span>
<span id="cb193-13"><a aria-hidden="true" href="#cb193-13" tabindex="-1"></a>    .<span class="bu">sum</span>()</span>
<span id="cb193-14"><a aria-hidden="true" href="#cb193-14" tabindex="-1"></a>    .sort_values(ascending<span class="op">=</span><span class="va">True</span>)</span>
<span id="cb193-15"><a aria-hidden="true" href="#cb193-15" tabindex="-1"></a>    .tail(<span class="dv">30</span>)</span>
<span id="cb193-16"><a aria-hidden="true" href="#cb193-16" tabindex="-1"></a>    .index</span>
<span id="cb193-17"><a aria-hidden="true" href="#cb193-17" tabindex="-1"></a>    .values</span>
<span id="cb193-18"><a aria-hidden="true" href="#cb193-18" tabindex="-1"></a>)</span>
<span id="cb193-19"><a aria-hidden="true" href="#cb193-19" tabindex="-1"></a></span>
<span id="cb193-20"><a aria-hidden="true" href="#cb193-20" tabindex="-1"></a>top_30_genera_samples <span class="op">=</span> (</span>
<span id="cb193-21"><a aria-hidden="true" href="#cb193-21" tabindex="-1"></a>    full_long_df</span>
<span id="cb193-22"><a aria-hidden="true" href="#cb193-22" tabindex="-1"></a>    .query(<span class="st">'Genus!="unknown"'</span>)</span>
<span id="cb193-23"><a aria-hidden="true" href="#cb193-23" tabindex="-1"></a>    .query(<span class="st">'sample_label.str.startswith("S")'</span>)</span>
<span id="cb193-24"><a aria-hidden="true" href="#cb193-24" tabindex="-1"></a>    .groupby(<span class="st">'Genus'</span>)</span>
<span id="cb193-25"><a aria-hidden="true" href="#cb193-25" tabindex="-1"></a>    [<span class="st">'value'</span>]</span>
<span id="cb193-26"><a aria-hidden="true" href="#cb193-26" tabindex="-1"></a>    .<span class="bu">sum</span>()</span>
<span id="cb193-27"><a aria-hidden="true" href="#cb193-27" tabindex="-1"></a>    .sort_values(ascending<span class="op">=</span><span class="va">True</span>)</span>
<span id="cb193-28"><a aria-hidden="true" href="#cb193-28" tabindex="-1"></a>    .tail(<span class="dv">30</span>)</span>
<span id="cb193-29"><a aria-hidden="true" href="#cb193-29" tabindex="-1"></a>    .index</span>
<span id="cb193-30"><a aria-hidden="true" href="#cb193-30" tabindex="-1"></a>    .values</span>
<span id="cb193-31"><a aria-hidden="true" href="#cb193-31" tabindex="-1"></a>)</span>
<span id="cb193-32"><a aria-hidden="true" href="#cb193-32" tabindex="-1"></a>treshold <span class="op">=</span> <span class="fl">.0005</span></span>
<span id="cb193-33"><a aria-hidden="true" href="#cb193-33" tabindex="-1"></a>f <span class="op">=</span> (full_long_df </span>
<span id="cb193-34"><a aria-hidden="true" href="#cb193-34" tabindex="-1"></a> .query(<span class="st">'Genus.notna()'</span>)</span>
<span id="cb193-35"><a aria-hidden="true" href="#cb193-35" tabindex="-1"></a>.assign(Genus<span class="op">=</span><span class="kw">lambda</span> dd: </span>
<span id="cb193-36"><a aria-hidden="true" href="#cb193-36" tabindex="-1"></a>        pd.Categorical(dd.Genus, categories<span class="op">=</span>top_30_genera_samples, ordered<span class="op">=</span><span class="va">True</span>))</span>
<span id="cb193-37"><a aria-hidden="true" href="#cb193-37" tabindex="-1"></a>.dropna()</span>
<span id="cb193-38"><a aria-hidden="true" href="#cb193-38" tabindex="-1"></a>.replace({<span class="st">'Allorhizobium-Neorhizobium-Pararhizobium-Rhizobium'</span>: <span class="st">'Rhizobium'</span>})</span>
<span id="cb193-39"><a aria-hidden="true" href="#cb193-39" tabindex="-1"></a>.assign(sample_mode<span class="op">=</span><span class="kw">lambda</span> dd: np.where(dd.sample_label.<span class="bu">str</span>.contains(<span class="st">'Flight'</span>), <span class="st">'Flight'</span>, <span class="st">'Ground'</span>))</span>
<span id="cb193-40"><a aria-hidden="true" href="#cb193-40" tabindex="-1"></a>.assign(sample_label<span class="op">=</span><span class="kw">lambda</span> dd: dd.sample_label.<span class="bu">str</span>.replace(<span class="st">'</span><span class="ch">\n</span><span class="st">'</span>, <span class="st">' '</span>))</span>
<span id="cb193-41"><a aria-hidden="true" href="#cb193-41" tabindex="-1"></a>.query(<span class="st">'value &gt; 0'</span>)</span>
<span id="cb193-42"><a aria-hidden="true" href="#cb193-42" tabindex="-1"></a>.assign(value<span class="op">=</span><span class="kw">lambda</span> dd: np.where(dd.value <span class="op">&lt;</span> treshold, treshold, dd.value))</span>
<span id="cb193-43"><a aria-hidden="true" href="#cb193-43" tabindex="-1"></a> .pipe(<span class="kw">lambda</span> dd: p9.ggplot(dd)</span>
<span id="cb193-44"><a aria-hidden="true" href="#cb193-44" tabindex="-1"></a>       <span class="op">+</span> p9.aes(<span class="st">'sample_label'</span>, <span class="st">'Genus'</span>, size<span class="op">=</span><span class="st">'value'</span>, fill<span class="op">=</span><span class="st">'sample_mode'</span>)</span>
<span id="cb193-45"><a aria-hidden="true" href="#cb193-45" tabindex="-1"></a>       <span class="op">+</span> p9.geom_point(alpha<span class="op">=</span><span class="fl">.8</span>, stroke<span class="op">=</span><span class="dv">0</span>)</span>
<span id="cb193-46"><a aria-hidden="true" href="#cb193-46" tabindex="-1"></a>       <span class="op">+</span> p9.scale_size_continuous(trans<span class="op">=</span><span class="st">'log10'</span>,</span>
<span id="cb193-47"><a aria-hidden="true" href="#cb193-47" tabindex="-1"></a>                                  <span class="bu">range</span><span class="op">=</span>(<span class="dv">1</span>, <span class="dv">9</span>),</span>
<span id="cb193-48"><a aria-hidden="true" href="#cb193-48" tabindex="-1"></a>                                  labels<span class="op">=</span>percent_format(), </span>
<span id="cb193-49"><a aria-hidden="true" href="#cb193-49" tabindex="-1"></a>                                  limits<span class="op">=</span>(treshold <span class="op">-</span> <span class="fl">1e-5</span>, <span class="fl">.9</span>),</span>
<span id="cb193-50"><a aria-hidden="true" href="#cb193-50" tabindex="-1"></a>                                  breaks<span class="op">=</span>[<span class="fl">.001</span>, <span class="fl">.01</span>, <span class="fl">.05</span>, <span class="fl">.5</span>])</span>
<span id="cb193-51"><a aria-hidden="true" href="#cb193-51" tabindex="-1"></a>       <span class="op">+</span> p9.scale_y_discrete(expand<span class="op">=</span>(<span class="fl">.035</span>, <span class="fl">.035</span>))</span>
<span id="cb193-52"><a aria-hidden="true" href="#cb193-52" tabindex="-1"></a>       <span class="op">+</span> p9.scale_fill_manual(values<span class="op">=</span>[<span class="st">'#D3894C'</span>, <span class="st">'#435B97'</span>])</span>
<span id="cb193-53"><a aria-hidden="true" href="#cb193-53" tabindex="-1"></a>       <span class="op">+</span> p9.guides(size<span class="op">=</span>p9.guide_legend(ncol<span class="op">=</span><span class="dv">4</span>))</span>
<span id="cb193-54"><a aria-hidden="true" href="#cb193-54" tabindex="-1"></a>       <span class="op">+</span> p9.labs(x<span class="op">=</span><span class="st">''</span>, y<span class="op">=</span><span class="st">''</span>, size<span class="op">=</span><span class="st">'Relative Abundance'</span>, color<span class="op">=</span><span class="st">'Class'</span>,</span>
<span id="cb193-55"><a aria-hidden="true" href="#cb193-55" tabindex="-1"></a>                 title<span class="op">=</span><span class="st">'Relative Abundance of Top 30 Genera in Samples'</span>,</span>
<span id="cb193-56"><a aria-hidden="true" href="#cb193-56" tabindex="-1"></a>                 fill<span class="op">=</span><span class="st">''</span></span>
<span id="cb193-57"><a aria-hidden="true" href="#cb193-57" tabindex="-1"></a>                 )</span>
<span id="cb193-58"><a aria-hidden="true" href="#cb193-58" tabindex="-1"></a>                  )</span>
<span id="cb193-59"><a aria-hidden="true" href="#cb193-59" tabindex="-1"></a>       <span class="op">+</span> p9.annotate(geom<span class="op">=</span><span class="st">'vline'</span>, linetype<span class="op">=</span><span class="st">'dashed'</span>, xintercept<span class="op">=</span><span class="fl">2.5</span>, color<span class="op">=</span><span class="st">'grey'</span>)</span>
<span id="cb193-60"><a aria-hidden="true" href="#cb193-60" tabindex="-1"></a>       <span class="op">+</span> p9.theme(figure_size<span class="op">=</span>(<span class="fl">7.5</span>, <span class="fl">7.5</span>),</span>
<span id="cb193-61"><a aria-hidden="true" href="#cb193-61" tabindex="-1"></a>                  strip_text<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">12</span>),</span>
<span id="cb193-62"><a aria-hidden="true" href="#cb193-62" tabindex="-1"></a>                  axis_text<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">10</span>),</span>
<span id="cb193-63"><a aria-hidden="true" href="#cb193-63" tabindex="-1"></a>                  legend_position<span class="op">=</span><span class="st">'bottom'</span>,</span>
<span id="cb193-64"><a aria-hidden="true" href="#cb193-64" tabindex="-1"></a>                  title<span class="op">=</span>p9.element_text(ha<span class="op">=</span><span class="st">'center'</span>),</span>
<span id="cb193-65"><a aria-hidden="true" href="#cb193-65" tabindex="-1"></a>                  legend_title<span class="op">=</span>p9.element_text(x<span class="op">=</span><span class="dv">300</span>),</span>
<span id="cb193-66"><a aria-hidden="true" href="#cb193-66" tabindex="-1"></a>                  axis_text_x<span class="op">=</span>p9.element_text(angle<span class="op">=</span><span class="dv">90</span>),</span>
<span id="cb193-67"><a aria-hidden="true" href="#cb193-67" tabindex="-1"></a>       )</span>
<span id="cb193-68"><a aria-hidden="true" href="#cb193-68" tabindex="-1"></a> )</span>
<span id="cb193-69"><a aria-hidden="true" href="#cb193-69" tabindex="-1"></a>f.save(<span class="st">'../output/figures/top_genera_samples.svg'</span>)</span>
<span id="cb193-70"><a aria-hidden="true" href="#cb193-70" tabindex="-1"></a>f.draw()</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="520">
<div>
<figure class="figure">
<p><img class="figure-img" height="2250" src="microbial_diversity_analysis_files/figure-html/cell-182-output-1.png" width="2250"/></p>
</figure>
</div>
</div>
</div>
<div class="cell" id="cell-222">
<div class="sourceCode cell-code" id="cb194"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb194-1"><a aria-hidden="true" href="#cb194-1" tabindex="-1"></a></span>
<span id="cb194-2"><a aria-hidden="true" href="#cb194-2" tabindex="-1"></a>g <span class="op">=</span> (full_long_df </span>
<span id="cb194-3"><a aria-hidden="true" href="#cb194-3" tabindex="-1"></a> .query(<span class="st">'Genus.notna()'</span>)</span>
<span id="cb194-4"><a aria-hidden="true" href="#cb194-4" tabindex="-1"></a>.assign(Genus<span class="op">=</span><span class="kw">lambda</span> dd: </span>
<span id="cb194-5"><a aria-hidden="true" href="#cb194-5" tabindex="-1"></a>        pd.Categorical(dd.Genus, categories<span class="op">=</span>top_30_genera_blanks, ordered<span class="op">=</span><span class="va">True</span>))</span>
<span id="cb194-6"><a aria-hidden="true" href="#cb194-6" tabindex="-1"></a>.dropna()</span>
<span id="cb194-7"><a aria-hidden="true" href="#cb194-7" tabindex="-1"></a>.replace({<span class="st">'Burkholderia-Caballeronia-Paraburkholderia'</span>: <span class="st">'Burkholderia-Caballeronia'</span>})</span>
<span id="cb194-8"><a aria-hidden="true" href="#cb194-8" tabindex="-1"></a>.assign(sample_mode<span class="op">=</span><span class="kw">lambda</span> dd: np.where(dd.sample_label.<span class="bu">str</span>.contains(<span class="st">'Flight'</span>), <span class="st">'Flight'</span>, <span class="st">'Ground'</span>))</span>
<span id="cb194-9"><a aria-hidden="true" href="#cb194-9" tabindex="-1"></a>.assign(sample_label<span class="op">=</span><span class="kw">lambda</span> dd: dd.sample_label.<span class="bu">str</span>.replace(<span class="st">'</span><span class="ch">\n</span><span class="st">'</span>, <span class="st">' '</span>))</span>
<span id="cb194-10"><a aria-hidden="true" href="#cb194-10" tabindex="-1"></a>.assign(value<span class="op">=</span><span class="kw">lambda</span> dd: np.where(dd.value <span class="op">&lt;</span> treshold, treshold, dd.value))</span>
<span id="cb194-11"><a aria-hidden="true" href="#cb194-11" tabindex="-1"></a></span>
<span id="cb194-12"><a aria-hidden="true" href="#cb194-12" tabindex="-1"></a> .pipe(<span class="kw">lambda</span> dd: p9.ggplot(dd)</span>
<span id="cb194-13"><a aria-hidden="true" href="#cb194-13" tabindex="-1"></a>       <span class="op">+</span> p9.aes(<span class="st">'sample_label'</span>, <span class="st">'Genus'</span>, size<span class="op">=</span><span class="st">'value'</span>, fill<span class="op">=</span><span class="st">'sample_mode'</span>)</span>
<span id="cb194-14"><a aria-hidden="true" href="#cb194-14" tabindex="-1"></a>       <span class="op">+</span> p9.geom_point(alpha<span class="op">=</span><span class="fl">.8</span>, stroke<span class="op">=</span><span class="dv">0</span>)</span>
<span id="cb194-15"><a aria-hidden="true" href="#cb194-15" tabindex="-1"></a>       <span class="op">+</span> p9.scale_size_continuous(trans<span class="op">=</span><span class="st">'log10'</span>, <span class="bu">range</span><span class="op">=</span>(<span class="dv">1</span>, <span class="dv">9</span>),</span>
<span id="cb194-16"><a aria-hidden="true" href="#cb194-16" tabindex="-1"></a>                                  labels<span class="op">=</span>percent_format(), </span>
<span id="cb194-17"><a aria-hidden="true" href="#cb194-17" tabindex="-1"></a>                                  limits<span class="op">=</span>(treshold <span class="op">-</span> <span class="fl">1e-5</span>, <span class="fl">.9</span>),</span>
<span id="cb194-18"><a aria-hidden="true" href="#cb194-18" tabindex="-1"></a>                                  breaks<span class="op">=</span>[<span class="fl">.001</span>, <span class="fl">.01</span>, <span class="fl">.05</span>, <span class="fl">.5</span>])</span>
<span id="cb194-19"><a aria-hidden="true" href="#cb194-19" tabindex="-1"></a>       <span class="op">+</span> p9.scale_y_discrete(expand<span class="op">=</span>(<span class="fl">.035</span>, <span class="fl">.035</span>))</span>
<span id="cb194-20"><a aria-hidden="true" href="#cb194-20" tabindex="-1"></a>       <span class="op">+</span> p9.scale_fill_manual(values<span class="op">=</span>[<span class="st">'#D3894C'</span>, <span class="st">'#435B97'</span>])</span>
<span id="cb194-21"><a aria-hidden="true" href="#cb194-21" tabindex="-1"></a>       <span class="op">+</span> p9.guides(size<span class="op">=</span>p9.guide_legend(ncol<span class="op">=</span><span class="dv">4</span>))</span>
<span id="cb194-22"><a aria-hidden="true" href="#cb194-22" tabindex="-1"></a>       <span class="op">+</span> p9.labs(x<span class="op">=</span><span class="st">''</span>, y<span class="op">=</span><span class="st">''</span>, size<span class="op">=</span><span class="st">'Relative Abundance'</span>, color<span class="op">=</span><span class="st">'Class'</span>,</span>
<span id="cb194-23"><a aria-hidden="true" href="#cb194-23" tabindex="-1"></a>                 title<span class="op">=</span><span class="st">'Relative Abundance of Top 30 Genera in Blanks'</span>,</span>
<span id="cb194-24"><a aria-hidden="true" href="#cb194-24" tabindex="-1"></a>                 fill<span class="op">=</span><span class="st">''</span></span>
<span id="cb194-25"><a aria-hidden="true" href="#cb194-25" tabindex="-1"></a>                 )</span>
<span id="cb194-26"><a aria-hidden="true" href="#cb194-26" tabindex="-1"></a>                  )</span>
<span id="cb194-27"><a aria-hidden="true" href="#cb194-27" tabindex="-1"></a>       <span class="op">+</span> p9.annotate(geom<span class="op">=</span><span class="st">'vline'</span>, linetype<span class="op">=</span><span class="st">'dashed'</span>, xintercept<span class="op">=</span><span class="fl">2.5</span>, color<span class="op">=</span><span class="st">'grey'</span>)</span>
<span id="cb194-28"><a aria-hidden="true" href="#cb194-28" tabindex="-1"></a>       <span class="op">+</span> p9.theme(figure_size<span class="op">=</span>(<span class="fl">6.75</span>, <span class="fl">7.5</span>),</span>
<span id="cb194-29"><a aria-hidden="true" href="#cb194-29" tabindex="-1"></a>                  strip_text<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">12</span>),</span>
<span id="cb194-30"><a aria-hidden="true" href="#cb194-30" tabindex="-1"></a>                  axis_text<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">10</span>),</span>
<span id="cb194-31"><a aria-hidden="true" href="#cb194-31" tabindex="-1"></a>                  legend_position<span class="op">=</span><span class="st">'bottom'</span>,</span>
<span id="cb194-32"><a aria-hidden="true" href="#cb194-32" tabindex="-1"></a>                  title<span class="op">=</span>p9.element_text(ha<span class="op">=</span><span class="st">'center'</span>),</span>
<span id="cb194-33"><a aria-hidden="true" href="#cb194-33" tabindex="-1"></a>                  legend_title<span class="op">=</span>p9.element_text(x<span class="op">=</span><span class="dv">300</span>),</span>
<span id="cb194-34"><a aria-hidden="true" href="#cb194-34" tabindex="-1"></a>                  axis_text_x<span class="op">=</span>p9element_text(angle<span class="op">=</span><span class="dv">90</span>),</span>
<span id="cb194-35"><a aria-hidden="true" href="#cb194-35" tabindex="-1"></a>       )</span>
<span id="cb194-36"><a aria-hidden="true" href="#cb194-36" tabindex="-1"></a> )</span>
<span id="cb194-37"><a aria-hidden="true" href="#cb194-37" tabindex="-1"></a>g.save(<span class="st">'../output/figures/top_genera_blanks.svg'</span>)</span>
<span id="cb194-38"><a aria-hidden="true" href="#cb194-38" tabindex="-1"></a>g.draw()</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="521">
<div>
<figure class="figure">
<p><img class="figure-img" height="2250" src="microbial_diversity_analysis_files/figure-html/cell-183-output-1.png" width="2025"/></p>
</figure>
</div>
</div>
</div>
<div class="cell" id="cell-223">
<div class="sourceCode cell-code" id="cb195"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb195-1"><a aria-hidden="true" href="#cb195-1" tabindex="-1"></a>full_long_df <span class="op">=</span> (full_matrix_fun</span>
<span id="cb195-2"><a aria-hidden="true" href="#cb195-2" tabindex="-1"></a>                .reset_index()</span>
<span id="cb195-3"><a aria-hidden="true" href="#cb195-3" tabindex="-1"></a>                .melt(<span class="st">'sample_label'</span>, var_name<span class="op">=</span><span class="st">'Genus'</span>, value_name<span class="op">=</span><span class="st">'value'</span>)</span>
<span id="cb195-4"><a aria-hidden="true" href="#cb195-4" tabindex="-1"></a>)</span>
<span id="cb195-5"><a aria-hidden="true" href="#cb195-5" tabindex="-1"></a></span>
<span id="cb195-6"><a aria-hidden="true" href="#cb195-6" tabindex="-1"></a></span>
<span id="cb195-7"><a aria-hidden="true" href="#cb195-7" tabindex="-1"></a>top_30_genera_blanks <span class="op">=</span> (</span>
<span id="cb195-8"><a aria-hidden="true" href="#cb195-8" tabindex="-1"></a>    full_long_df</span>
<span id="cb195-9"><a aria-hidden="true" href="#cb195-9" tabindex="-1"></a>    .query(<span class="st">'Genus!="unknown"'</span>)</span>
<span id="cb195-10"><a aria-hidden="true" href="#cb195-10" tabindex="-1"></a>    .query(<span class="st">'sample_label.str.contains("Pool")'</span>)</span>
<span id="cb195-11"><a aria-hidden="true" href="#cb195-11" tabindex="-1"></a>    .groupby(<span class="st">'Genus'</span>)</span>
<span id="cb195-12"><a aria-hidden="true" href="#cb195-12" tabindex="-1"></a>    [<span class="st">'value'</span>]</span>
<span id="cb195-13"><a aria-hidden="true" href="#cb195-13" tabindex="-1"></a>    .<span class="bu">sum</span>()</span>
<span id="cb195-14"><a aria-hidden="true" href="#cb195-14" tabindex="-1"></a>    .sort_values(ascending<span class="op">=</span><span class="va">True</span>)</span>
<span id="cb195-15"><a aria-hidden="true" href="#cb195-15" tabindex="-1"></a>    .tail(<span class="dv">30</span>)</span>
<span id="cb195-16"><a aria-hidden="true" href="#cb195-16" tabindex="-1"></a>    .index</span>
<span id="cb195-17"><a aria-hidden="true" href="#cb195-17" tabindex="-1"></a>    .values</span>
<span id="cb195-18"><a aria-hidden="true" href="#cb195-18" tabindex="-1"></a>)</span>
<span id="cb195-19"><a aria-hidden="true" href="#cb195-19" tabindex="-1"></a></span>
<span id="cb195-20"><a aria-hidden="true" href="#cb195-20" tabindex="-1"></a>top_30_genera_samples <span class="op">=</span> (</span>
<span id="cb195-21"><a aria-hidden="true" href="#cb195-21" tabindex="-1"></a>    full_long_df</span>
<span id="cb195-22"><a aria-hidden="true" href="#cb195-22" tabindex="-1"></a>    .query(<span class="st">'Genus!="unknown"'</span>)</span>
<span id="cb195-23"><a aria-hidden="true" href="#cb195-23" tabindex="-1"></a>    .query(<span class="st">'sample_label.str.startswith("S")'</span>)</span>
<span id="cb195-24"><a aria-hidden="true" href="#cb195-24" tabindex="-1"></a>    .groupby(<span class="st">'Genus'</span>)</span>
<span id="cb195-25"><a aria-hidden="true" href="#cb195-25" tabindex="-1"></a>    [<span class="st">'value'</span>]</span>
<span id="cb195-26"><a aria-hidden="true" href="#cb195-26" tabindex="-1"></a>    .<span class="bu">sum</span>()</span>
<span id="cb195-27"><a aria-hidden="true" href="#cb195-27" tabindex="-1"></a>    .sort_values(ascending<span class="op">=</span><span class="va">True</span>)</span>
<span id="cb195-28"><a aria-hidden="true" href="#cb195-28" tabindex="-1"></a>    .tail(<span class="dv">30</span>)</span>
<span id="cb195-29"><a aria-hidden="true" href="#cb195-29" tabindex="-1"></a>    .index</span>
<span id="cb195-30"><a aria-hidden="true" href="#cb195-30" tabindex="-1"></a>    .values</span>
<span id="cb195-31"><a aria-hidden="true" href="#cb195-31" tabindex="-1"></a>)</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" id="cell-224">
<div class="sourceCode cell-code" id="cb196"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb196-1"><a aria-hidden="true" href="#cb196-1" tabindex="-1"></a>(full_long_df </span>
<span id="cb196-2"><a aria-hidden="true" href="#cb196-2" tabindex="-1"></a> .query(<span class="st">'Genus.notna()'</span>)</span>
<span id="cb196-3"><a aria-hidden="true" href="#cb196-3" tabindex="-1"></a>.assign(Genus<span class="op">=</span><span class="kw">lambda</span> dd: </span>
<span id="cb196-4"><a aria-hidden="true" href="#cb196-4" tabindex="-1"></a>        pd.Categorical(dd.Genus, categories<span class="op">=</span>top_30_genera_samples[<span class="dv">10</span>:], ordered<span class="op">=</span><span class="va">True</span>))</span>
<span id="cb196-5"><a aria-hidden="true" href="#cb196-5" tabindex="-1"></a>.dropna()</span>
<span id="cb196-6"><a aria-hidden="true" href="#cb196-6" tabindex="-1"></a>.replace({<span class="st">'Allorhizobium-Neorhizobium-Pararhizobium-Rhizobium'</span>: <span class="st">'Rhizobium'</span>})</span>
<span id="cb196-7"><a aria-hidden="true" href="#cb196-7" tabindex="-1"></a>.assign(sample_mode<span class="op">=</span><span class="kw">lambda</span> dd: np.where(dd.sample_label.<span class="bu">str</span>.contains(<span class="st">'Flight'</span>), <span class="st">'Flight'</span>, <span class="st">'Ground'</span>))</span>
<span id="cb196-8"><a aria-hidden="true" href="#cb196-8" tabindex="-1"></a>.assign(sample_label<span class="op">=</span><span class="kw">lambda</span> dd: dd.sample_label.<span class="bu">str</span>.replace(<span class="st">'</span><span class="ch">\n</span><span class="st">'</span>, <span class="st">' '</span>))</span>
<span id="cb196-9"><a aria-hidden="true" href="#cb196-9" tabindex="-1"></a>.query(<span class="st">'value &gt; 0'</span>)</span>
<span id="cb196-10"><a aria-hidden="true" href="#cb196-10" tabindex="-1"></a>.assign(value<span class="op">=</span><span class="kw">lambda</span> dd: np.where(dd.value <span class="op">&lt;</span> treshold, treshold, dd.value))</span>
<span id="cb196-11"><a aria-hidden="true" href="#cb196-11" tabindex="-1"></a>.assign(sample_label<span class="op">=</span><span class="kw">lambda</span> dd: dd.sample_label.<span class="bu">str</span>.split(<span class="st">' '</span>).<span class="bu">str</span>[<span class="dv">0</span>])</span>
<span id="cb196-12"><a aria-hidden="true" href="#cb196-12" tabindex="-1"></a> .pipe(<span class="kw">lambda</span> dd: p9.ggplot(dd)</span>
<span id="cb196-13"><a aria-hidden="true" href="#cb196-13" tabindex="-1"></a>       <span class="op">+</span> p9.aes(<span class="st">'sample_label'</span>, <span class="st">'Genus'</span>, size<span class="op">=</span><span class="st">'value'</span>, fill<span class="op">=</span><span class="st">'sample_mode'</span>)</span>
<span id="cb196-14"><a aria-hidden="true" href="#cb196-14" tabindex="-1"></a>       <span class="op">+</span> p9.geom_point(alpha<span class="op">=</span><span class="fl">.8</span>, stroke<span class="op">=</span><span class="dv">0</span>)</span>
<span id="cb196-15"><a aria-hidden="true" href="#cb196-15" tabindex="-1"></a>       <span class="op">+</span> p9.scale_size_continuous(trans<span class="op">=</span><span class="st">'log10'</span>,</span>
<span id="cb196-16"><a aria-hidden="true" href="#cb196-16" tabindex="-1"></a>                                  <span class="bu">range</span><span class="op">=</span>(<span class="dv">1</span>, <span class="dv">9</span>),</span>
<span id="cb196-17"><a aria-hidden="true" href="#cb196-17" tabindex="-1"></a>                                  labels<span class="op">=</span>percent_format(), </span>
<span id="cb196-18"><a aria-hidden="true" href="#cb196-18" tabindex="-1"></a>                                  limits<span class="op">=</span>(treshold <span class="op">-</span> <span class="fl">1e-5</span>, <span class="fl">.9</span>),</span>
<span id="cb196-19"><a aria-hidden="true" href="#cb196-19" tabindex="-1"></a>                                  breaks<span class="op">=</span>[<span class="fl">.001</span>, <span class="fl">.01</span>, <span class="fl">.05</span>, <span class="fl">.5</span>])</span>
<span id="cb196-20"><a aria-hidden="true" href="#cb196-20" tabindex="-1"></a>       <span class="op">+</span> p9.scale_y_discrete(expand<span class="op">=</span>(<span class="fl">.035</span>, <span class="fl">.035</span>))</span>
<span id="cb196-21"><a aria-hidden="true" href="#cb196-21" tabindex="-1"></a>       <span class="op">+</span> p9.scale_fill_manual(values<span class="op">=</span>[<span class="st">'#D3894C'</span>, <span class="st">'#435B97'</span>])</span>
<span id="cb196-22"><a aria-hidden="true" href="#cb196-22" tabindex="-1"></a>       <span class="op">+</span> p9.guides(size<span class="op">=</span><span class="va">False</span>, color<span class="op">=</span><span class="va">False</span>, fill<span class="op">=</span><span class="va">False</span>)</span>
<span id="cb196-23"><a aria-hidden="true" href="#cb196-23" tabindex="-1"></a>       <span class="op">+</span> p9.labs(x<span class="op">=</span><span class="st">''</span>, y<span class="op">=</span><span class="st">''</span>, size<span class="op">=</span><span class="st">'Relative Abundance'</span>, color<span class="op">=</span><span class="st">'Class'</span>,</span>
<span id="cb196-24"><a aria-hidden="true" href="#cb196-24" tabindex="-1"></a>                 title<span class="op">=</span><span class="st">''</span>,</span>
<span id="cb196-25"><a aria-hidden="true" href="#cb196-25" tabindex="-1"></a>                 fill<span class="op">=</span><span class="st">''</span></span>
<span id="cb196-26"><a aria-hidden="true" href="#cb196-26" tabindex="-1"></a>                 )</span>
<span id="cb196-27"><a aria-hidden="true" href="#cb196-27" tabindex="-1"></a>                  )</span>
<span id="cb196-28"><a aria-hidden="true" href="#cb196-28" tabindex="-1"></a>       <span class="op">+</span> p9.annotate(geom<span class="op">=</span><span class="st">'vline'</span>, linetype<span class="op">=</span><span class="st">'dashed'</span>, xintercept<span class="op">=</span><span class="fl">2.5</span>, color<span class="op">=</span><span class="st">'grey'</span>)</span>
<span id="cb196-29"><a aria-hidden="true" href="#cb196-29" tabindex="-1"></a>       <span class="op">+</span> p9.theme(figure_size<span class="op">=</span>(<span class="dv">5</span>, <span class="dv">3</span>),</span>
<span id="cb196-30"><a aria-hidden="true" href="#cb196-30" tabindex="-1"></a>                  strip_text<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">12</span>),</span>
<span id="cb196-31"><a aria-hidden="true" href="#cb196-31" tabindex="-1"></a>                  axis_text<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">10</span>),</span>
<span id="cb196-32"><a aria-hidden="true" href="#cb196-32" tabindex="-1"></a>                  legend_position<span class="op">=</span><span class="st">'bottom'</span>,</span>
<span id="cb196-33"><a aria-hidden="true" href="#cb196-33" tabindex="-1"></a>                  title<span class="op">=</span>p9.element_text(ha<span class="op">=</span><span class="st">'center'</span>),</span>
<span id="cb196-34"><a aria-hidden="true" href="#cb196-34" tabindex="-1"></a>                  legend_title<span class="op">=</span>p9.element_text(x<span class="op">=</span><span class="dv">300</span>),</span>
<span id="cb196-35"><a aria-hidden="true" href="#cb196-35" tabindex="-1"></a>                  axis_text_x<span class="op">=</span>p9.element_text(angle<span class="op">=</span><span class="dv">90</span>),</span>
<span id="cb196-36"><a aria-hidden="true" href="#cb196-36" tabindex="-1"></a>       )</span>
<span id="cb196-37"><a aria-hidden="true" href="#cb196-37" tabindex="-1"></a> )</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img class="figure-img" height="900" src="microbial_diversity_analysis_files/figure-html/cell-185-output-1.png" width="1500"/></p>
</figure>
</div>
</div>
</div>
<div class="cell" id="cell-225">
<div class="sourceCode cell-code" id="cb197"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb197-1"><a aria-hidden="true" href="#cb197-1" tabindex="-1"></a>f <span class="op">=</span> (full_long_df </span>
<span id="cb197-2"><a aria-hidden="true" href="#cb197-2" tabindex="-1"></a> .query(<span class="st">'Genus.notna()'</span>)</span>
<span id="cb197-3"><a aria-hidden="true" href="#cb197-3" tabindex="-1"></a>.assign(Genus<span class="op">=</span><span class="kw">lambda</span> dd: </span>
<span id="cb197-4"><a aria-hidden="true" href="#cb197-4" tabindex="-1"></a>        pd.Categorical(dd.Genus, categories<span class="op">=</span>top_30_genera_samples, ordered<span class="op">=</span><span class="va">True</span>))</span>
<span id="cb197-5"><a aria-hidden="true" href="#cb197-5" tabindex="-1"></a>.dropna()</span>
<span id="cb197-6"><a aria-hidden="true" href="#cb197-6" tabindex="-1"></a>.replace({<span class="st">'Allorhizobium-Neorhizobium-Pararhizobium-Rhizobium'</span>: <span class="st">'Rhizobium'</span>})</span>
<span id="cb197-7"><a aria-hidden="true" href="#cb197-7" tabindex="-1"></a>.assign(sample_mode<span class="op">=</span><span class="kw">lambda</span> dd: np.where(dd.sample_label.<span class="bu">str</span>.contains(<span class="st">'Flight'</span>), <span class="st">'Flight'</span>, <span class="st">'Ground'</span>))</span>
<span id="cb197-8"><a aria-hidden="true" href="#cb197-8" tabindex="-1"></a>.assign(sample_label<span class="op">=</span><span class="kw">lambda</span> dd: dd.sample_label.<span class="bu">str</span>.replace(<span class="st">'</span><span class="ch">\n</span><span class="st">'</span>, <span class="st">' '</span>))</span>
<span id="cb197-9"><a aria-hidden="true" href="#cb197-9" tabindex="-1"></a>.query(<span class="st">'value &gt; 0'</span>)</span>
<span id="cb197-10"><a aria-hidden="true" href="#cb197-10" tabindex="-1"></a>.assign(value<span class="op">=</span><span class="kw">lambda</span> dd: np.where(dd.value <span class="op">&lt;</span> treshold, treshold, dd.value))</span>
<span id="cb197-11"><a aria-hidden="true" href="#cb197-11" tabindex="-1"></a> .pipe(<span class="kw">lambda</span> dd: p9.ggplot(dd)</span>
<span id="cb197-12"><a aria-hidden="true" href="#cb197-12" tabindex="-1"></a>       <span class="op">+</span> p9.aes(<span class="st">'sample_label'</span>, <span class="st">'Genus'</span>, size<span class="op">=</span><span class="st">'value'</span>, fill<span class="op">=</span><span class="st">'sample_mode'</span>)</span>
<span id="cb197-13"><a aria-hidden="true" href="#cb197-13" tabindex="-1"></a>       <span class="op">+</span> p9.geom_point(alpha<span class="op">=</span><span class="fl">.8</span>, stroke<span class="op">=</span><span class="dv">0</span>)</span>
<span id="cb197-14"><a aria-hidden="true" href="#cb197-14" tabindex="-1"></a>       <span class="op">+</span> p9.scale_size_continuous(trans<span class="op">=</span><span class="st">'log10'</span>,</span>
<span id="cb197-15"><a aria-hidden="true" href="#cb197-15" tabindex="-1"></a>                                  <span class="bu">range</span><span class="op">=</span>(<span class="dv">1</span>, <span class="dv">9</span>),</span>
<span id="cb197-16"><a aria-hidden="true" href="#cb197-16" tabindex="-1"></a>                                  labels<span class="op">=</span>percent_format(), </span>
<span id="cb197-17"><a aria-hidden="true" href="#cb197-17" tabindex="-1"></a>                                  limits<span class="op">=</span>(treshold <span class="op">-</span> <span class="fl">1e-5</span>, <span class="fl">.9</span>),</span>
<span id="cb197-18"><a aria-hidden="true" href="#cb197-18" tabindex="-1"></a>                                  breaks<span class="op">=</span>[<span class="fl">.001</span>, <span class="fl">.01</span>, <span class="fl">.05</span>, <span class="fl">.5</span>])</span>
<span id="cb197-19"><a aria-hidden="true" href="#cb197-19" tabindex="-1"></a>       <span class="op">+</span> p9.scale_y_discrete(expand<span class="op">=</span>(<span class="fl">.035</span>, <span class="fl">.035</span>))</span>
<span id="cb197-20"><a aria-hidden="true" href="#cb197-20" tabindex="-1"></a>       <span class="op">+</span> p9.scale_fill_manual(values<span class="op">=</span>[<span class="st">'#D3894C'</span>, <span class="st">'#435B97'</span>])</span>
<span id="cb197-21"><a aria-hidden="true" href="#cb197-21" tabindex="-1"></a>       <span class="op">+</span> p9.guides(size<span class="op">=</span>p9.guide_legend(ncol<span class="op">=</span><span class="dv">4</span>))</span>
<span id="cb197-22"><a aria-hidden="true" href="#cb197-22" tabindex="-1"></a>       <span class="op">+</span> p9.labs(x<span class="op">=</span><span class="st">''</span>, y<span class="op">=</span><span class="st">''</span>, size<span class="op">=</span><span class="st">'Relative Abundance'</span>, color<span class="op">=</span><span class="st">'Class'</span>,</span>
<span id="cb197-23"><a aria-hidden="true" href="#cb197-23" tabindex="-1"></a>                 title<span class="op">=</span><span class="st">'Relative Abundance of Top 30 Genera in Samples'</span>,</span>
<span id="cb197-24"><a aria-hidden="true" href="#cb197-24" tabindex="-1"></a>                 fill<span class="op">=</span><span class="st">''</span></span>
<span id="cb197-25"><a aria-hidden="true" href="#cb197-25" tabindex="-1"></a>                 )</span>
<span id="cb197-26"><a aria-hidden="true" href="#cb197-26" tabindex="-1"></a>                  )</span>
<span id="cb197-27"><a aria-hidden="true" href="#cb197-27" tabindex="-1"></a>       <span class="op">+</span> p9.annotate(geom<span class="op">=</span><span class="st">'vline'</span>, linetype<span class="op">=</span><span class="st">'dashed'</span>, xintercept<span class="op">=</span><span class="fl">2.5</span>, color<span class="op">=</span><span class="st">'grey'</span>)</span>
<span id="cb197-28"><a aria-hidden="true" href="#cb197-28" tabindex="-1"></a>       <span class="op">+</span> p9.theme(figure_size<span class="op">=</span>(<span class="fl">6.25</span>, <span class="fl">7.5</span>),</span>
<span id="cb197-29"><a aria-hidden="true" href="#cb197-29" tabindex="-1"></a>                  strip_text<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">12</span>),</span>
<span id="cb197-30"><a aria-hidden="true" href="#cb197-30" tabindex="-1"></a>                  axis_text<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">10</span>),</span>
<span id="cb197-31"><a aria-hidden="true" href="#cb197-31" tabindex="-1"></a>                  legend_position<span class="op">=</span><span class="st">'bottom'</span>,</span>
<span id="cb197-32"><a aria-hidden="true" href="#cb197-32" tabindex="-1"></a>                  title<span class="op">=</span>p9.element_text(ha<span class="op">=</span><span class="st">'center'</span>),</span>
<span id="cb197-33"><a aria-hidden="true" href="#cb197-33" tabindex="-1"></a>                  legend_title<span class="op">=</span>p9.element_text(x<span class="op">=</span><span class="dv">300</span>),</span>
<span id="cb197-34"><a aria-hidden="true" href="#cb197-34" tabindex="-1"></a>                  axis_text_x<span class="op">=</span>p9.element_text(angle<span class="op">=</span><span class="dv">90</span>),</span>
<span id="cb197-35"><a aria-hidden="true" href="#cb197-35" tabindex="-1"></a>       )</span>
<span id="cb197-36"><a aria-hidden="true" href="#cb197-36" tabindex="-1"></a> )</span>
<span id="cb197-37"><a aria-hidden="true" href="#cb197-37" tabindex="-1"></a>f.save(<span class="st">'../output/figures/top_genera_samples_fun.svg'</span>)</span>
<span id="cb197-38"><a aria-hidden="true" href="#cb197-38" tabindex="-1"></a>f.draw()</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="528">
<div>
<figure class="figure">
<p><img class="figure-img" height="2250" src="microbial_diversity_analysis_files/figure-html/cell-186-output-1.png" width="1875"/></p>
</figure>
</div>
</div>
</div>
<div class="cell" id="cell-226">
<div class="sourceCode cell-code" id="cb198"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb198-1"><a aria-hidden="true" href="#cb198-1" tabindex="-1"></a>g <span class="op">=</span> (full_long_df </span>
<span id="cb198-2"><a aria-hidden="true" href="#cb198-2" tabindex="-1"></a> .query(<span class="st">'Genus.notna()'</span>)</span>
<span id="cb198-3"><a aria-hidden="true" href="#cb198-3" tabindex="-1"></a>.assign(Genus<span class="op">=</span><span class="kw">lambda</span> dd: </span>
<span id="cb198-4"><a aria-hidden="true" href="#cb198-4" tabindex="-1"></a>        pd.Categorical(dd.Genus, categories<span class="op">=</span>top_30_genera_blanks, ordered<span class="op">=</span><span class="va">True</span>))</span>
<span id="cb198-5"><a aria-hidden="true" href="#cb198-5" tabindex="-1"></a>.dropna()</span>
<span id="cb198-6"><a aria-hidden="true" href="#cb198-6" tabindex="-1"></a>.replace({<span class="st">'Genus'</span>: {<span class="st">'Malasseziaceae_gen_Incertae_sedis'</span>: <span class="st">'Malasseziaceae'</span>}})</span>
<span id="cb198-7"><a aria-hidden="true" href="#cb198-7" tabindex="-1"></a>.assign(sample_mode<span class="op">=</span><span class="kw">lambda</span> dd: np.where(dd.sample_label.<span class="bu">str</span>.contains(<span class="st">'Flight'</span>), <span class="st">'Flight'</span>, <span class="st">'Ground'</span>))</span>
<span id="cb198-8"><a aria-hidden="true" href="#cb198-8" tabindex="-1"></a>.assign(sample_label<span class="op">=</span><span class="kw">lambda</span> dd: dd.sample_label.<span class="bu">str</span>.replace(<span class="st">'</span><span class="ch">\n</span><span class="st">'</span>, <span class="st">' '</span>))</span>
<span id="cb198-9"><a aria-hidden="true" href="#cb198-9" tabindex="-1"></a>.assign(value<span class="op">=</span><span class="kw">lambda</span> dd: np.where(dd.value <span class="op">&lt;</span> treshold, treshold, dd.value))</span>
<span id="cb198-10"><a aria-hidden="true" href="#cb198-10" tabindex="-1"></a>.pipe(<span class="kw">lambda</span> dd: p9.ggplot(dd)</span>
<span id="cb198-11"><a aria-hidden="true" href="#cb198-11" tabindex="-1"></a>       <span class="op">+</span> p9.aes(<span class="st">'sample_label'</span>, <span class="st">'Genus'</span>, size<span class="op">=</span><span class="st">'value'</span>, fill<span class="op">=</span><span class="st">'sample_mode'</span>)</span>
<span id="cb198-12"><a aria-hidden="true" href="#cb198-12" tabindex="-1"></a>       <span class="op">+</span> p9.geom_point(alpha<span class="op">=</span><span class="fl">.8</span>, stroke<span class="op">=</span><span class="dv">0</span>)</span>
<span id="cb198-13"><a aria-hidden="true" href="#cb198-13" tabindex="-1"></a>       <span class="op">+</span> p9.scale_size_continuous(trans<span class="op">=</span><span class="st">'log10'</span>, <span class="bu">range</span><span class="op">=</span>(<span class="dv">1</span>, <span class="dv">9</span>),</span>
<span id="cb198-14"><a aria-hidden="true" href="#cb198-14" tabindex="-1"></a>                                  labels<span class="op">=</span>percent_format(), </span>
<span id="cb198-15"><a aria-hidden="true" href="#cb198-15" tabindex="-1"></a>                                  limits<span class="op">=</span>(treshold <span class="op">-</span> <span class="fl">1e-5</span>, <span class="fl">.9</span>),</span>
<span id="cb198-16"><a aria-hidden="true" href="#cb198-16" tabindex="-1"></a>                                  breaks<span class="op">=</span>[<span class="fl">.001</span>, <span class="fl">.01</span>, <span class="fl">.05</span>, <span class="fl">.5</span>])</span>
<span id="cb198-17"><a aria-hidden="true" href="#cb198-17" tabindex="-1"></a>       <span class="op">+</span> p9.scale_y_discrete(expand<span class="op">=</span>(<span class="fl">.035</span>, <span class="fl">.035</span>))</span>
<span id="cb198-18"><a aria-hidden="true" href="#cb198-18" tabindex="-1"></a>       <span class="op">+</span> p9.scale_fill_manual(values<span class="op">=</span>[<span class="st">'#D3894C'</span>, <span class="st">'#435B97'</span>])</span>
<span id="cb198-19"><a aria-hidden="true" href="#cb198-19" tabindex="-1"></a>       <span class="op">+</span> p9.guides(size<span class="op">=</span>p9.guide_legend(ncol<span class="op">=</span><span class="dv">4</span>))</span>
<span id="cb198-20"><a aria-hidden="true" href="#cb198-20" tabindex="-1"></a>       <span class="op">+</span> p9.labs(x<span class="op">=</span><span class="st">''</span>, y<span class="op">=</span><span class="st">''</span>, size<span class="op">=</span><span class="st">'Relative Abundance'</span>, color<span class="op">=</span><span class="st">'Class'</span>,</span>
<span id="cb198-21"><a aria-hidden="true" href="#cb198-21" tabindex="-1"></a>                 title<span class="op">=</span><span class="st">'Relative Abundance of Top 30 Genera in Blanks'</span>,</span>
<span id="cb198-22"><a aria-hidden="true" href="#cb198-22" tabindex="-1"></a>                 fill<span class="op">=</span><span class="st">''</span></span>
<span id="cb198-23"><a aria-hidden="true" href="#cb198-23" tabindex="-1"></a>                 )</span>
<span id="cb198-24"><a aria-hidden="true" href="#cb198-24" tabindex="-1"></a>                  )</span>
<span id="cb198-25"><a aria-hidden="true" href="#cb198-25" tabindex="-1"></a>       <span class="op">+</span> p9.annotate(geom<span class="op">=</span><span class="st">'vline'</span>, linetype<span class="op">=</span><span class="st">'dashed'</span>, xintercept<span class="op">=</span><span class="fl">2.5</span>, color<span class="op">=</span><span class="st">'grey'</span>)</span>
<span id="cb198-26"><a aria-hidden="true" href="#cb198-26" tabindex="-1"></a>       <span class="op">+</span> p9.theme(figure_size<span class="op">=</span>(<span class="fl">6.25</span>, <span class="fl">7.5</span>),</span>
<span id="cb198-27"><a aria-hidden="true" href="#cb198-27" tabindex="-1"></a>                  strip_text<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">12</span>),</span>
<span id="cb198-28"><a aria-hidden="true" href="#cb198-28" tabindex="-1"></a>                  axis_text<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">10</span>),</span>
<span id="cb198-29"><a aria-hidden="true" href="#cb198-29" tabindex="-1"></a>                  legend_position<span class="op">=</span><span class="st">'bottom'</span>,</span>
<span id="cb198-30"><a aria-hidden="true" href="#cb198-30" tabindex="-1"></a>                  title<span class="op">=</span>p9.element_text(ha<span class="op">=</span><span class="st">'center'</span>),</span>
<span id="cb198-31"><a aria-hidden="true" href="#cb198-31" tabindex="-1"></a>                  legend_title<span class="op">=</span>p9.element_text(x<span class="op">=</span><span class="dv">300</span>),</span>
<span id="cb198-32"><a aria-hidden="true" href="#cb198-32" tabindex="-1"></a>                  axis_text_x<span class="op">=</span>p9.element_text(angle<span class="op">=</span><span class="dv">90</span>),</span>
<span id="cb198-33"><a aria-hidden="true" href="#cb198-33" tabindex="-1"></a>       )</span>
<span id="cb198-34"><a aria-hidden="true" href="#cb198-34" tabindex="-1"></a> )</span>
<span id="cb198-35"><a aria-hidden="true" href="#cb198-35" tabindex="-1"></a>g.save(<span class="st">'../output/figures/top_genera_blanks_fun.svg'</span>)</span>
<span id="cb198-36"><a aria-hidden="true" href="#cb198-36" tabindex="-1"></a>g.draw()</span></code><button class="code-copy-button" title="Copy to Clipboard"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="527">
<div>
<figure class="figure">
<p><img class="figure-img" height="2250" src="microbial_diversity_analysis_files/figure-html/cell-187-output-1.png" width="1875"/></p>
</figure>
</div>
</div>
</div>
</section>
</section>
</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    // For code content inside modals, clipBoardJS needs to be initialized with a container option
    // TODO: Check when it could be a function (https://github.com/zenorocha/clipboard.js/issues/860)
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->
<script src="site_libs/quarto-html/zenscroll-min.js"></script>
</body></html>